<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="ERM" />

<meta name="date" content="2023-01-31" />

<title>GO analysis</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.1/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.10.1/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-2.11.1/plotly-latest.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cardiotoxicity</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/reneeisnowhere/Cardiotoxicity">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GO analysis</h1>
<h4 class="author">ERM</h4>
<h4 class="date">2023-01-31</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-01-31
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Cardiotoxicity/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230109code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230109)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230109code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230109)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereCardiotoxicitytree848eb1a472661b66bb74e2dfb3785359937f2236targetblank848eb1aa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/tree/848eb1a472661b66bb74e2dfb3785359937f2236" target="_blank">848eb1a</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereCardiotoxicitytree848eb1a472661b66bb74e2dfb3785359937f2236targetblank848eb1aa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/tree/848eb1a472661b66bb74e2dfb3785359937f2236" target="_blank">848eb1a</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    data/.txt
    Ignored:    data/allgenes.txt
    Ignored:    data/backGL.txt
    Ignored:    data/data_outline.txt
    Ignored:    data/filenameonly.txt
    Ignored:    data/mymatrix.RDS
    Ignored:    data/sigVDA24.txt
    Ignored:    data/sigVDA3.txt
    Ignored:    data/sigVDX24.txt
    Ignored:    data/sigVDX3.txt
    Ignored:    data/sigVEP24.txt
    Ignored:    data/sigVEP3.txt
    Ignored:    data/sigVMT24.txt
    Ignored:    data/sigVMT3.txt
    Ignored:    data/sigVTR24.txt
    Ignored:    data/sigVTR3.txt

Untracked files:
    Untracked:  reneebasecode.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/GO_analysis.Rmd</code>) and HTML
(<code>docs/GO_analysis.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/848eb1a472661b66bb74e2dfb3785359937f2236/analysis/GO_analysis.Rmd" target="_blank">848eb1a</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-31
</td>
<td>
updating GO plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/82b31a007623e98d5c87b454d1de4ea8fc98bd49/analysis/GO_analysis.Rmd" target="_blank">82b31a0</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-31
</td>
<td>
updating GEA
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/34a525cbeec7915aaeb7a5edd1aec222a950fb3b/analysis/GO_analysis.Rmd" target="_blank">34a525c</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-26
</td>
<td>
link update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/41b5f9c3819e894149c76aacf95afb117ff50106/docs/GO_analysis.html" target="_blank">41b5f9c</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/dc890ec898b2a62c0dbde4a99422a81adee1e7b7/analysis/GO_analysis.Rmd" target="_blank">dc890ec</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-23
</td>
<td>
Updating the source dir
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/a78eab0ed9e312aec423638afdba81ba2d963dce/analysis/GO_analysis.Rmd" target="_blank">a78eab0</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-23
</td>
<td>
updated Go analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/d9edc0612095d04009afd65ed97fa6fd1c703c83/analysis/GO_analysis.Rmd" target="_blank">d9edc06</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-20
</td>
<td>
try a little update later
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/b7eac762b2cfe321cf328b6f1f339bc3d9ba5caa/analysis/GO_analysis.Rmd" target="_blank">b7eac76</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-20
</td>
<td>
updating GO
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/6c21cf8cca434fdf262799bdc3b685ce23a91ad2/docs/GO_analysis.html" target="_blank">6c21cf8</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/443a9f393601e6c682c88350ddbe45f4e46f9617/analysis/GO_analysis.Rmd" target="_blank">443a9f3</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-20
</td>
<td>
adding in GO analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="go-analysis" class="section level1">
<h1>GO Analysis</h1>
<p>I have created several files from the RNA analysis that contain the
significant genes(determined by adj.P.val &lt; 0.1) from each Time and
Condition. The names of the files are in the following format:
‘sigV’+Drug(2 letters)+time.</p>
<p>example: ‘sigVDA3.txt’ means this file contains the significant DE
genes from the Daunorubicin 3 hour compared to Vehicle Control 3 hour
analysis</p>
<pre class="r"><code>library(gprofiler2)
library(tidyverse)
library(readr)
library(BiocGenerics)
library(gridExtra)
library(VennDiagram)</code></pre>
<p>The analysis is based on all genes that passed the rowMeans&gt;0 from
the previous page <a
href="https://reneeisnowhere.github.io/Cardiotoxicity/RNAseqrun_1_analysis.html">link</a></p>
<p>Below is the analysis of differentially expressed genes for each
treatment at 3 hours and 24 hours.</p>
<div class="plotly html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-d3736c7f9cf5f559c042" style="width:960px;height:1824px;"></div>
<script type="application/json" data-for="htmlwidget-d3736c7f9cf5f559c042">{"x":{"data":[{"x":[87.1877575065437,87.1596578126843,87.1355723608048,165.553789438535,113.878452431029,140.689574614913,87.1516293287245,184.2721997909,199.345678425499,96.5971407074754,103.678263560054,117.84452350719,110.84368549421,101.84375497523,104.489140439997,94.5980482014754,176.685282448852,92.4022578384594,128.642834433174,110.83164276827,161.33482111764,140.420620402258,128.594663529415,77.9750721626281,177.476088118897,132.239595247182,184.27621403288,181.349831629519,140.693588856893,92.3982435964794,96.6051691914352,132.400164926379,96.3643146726401,132.243609489162,184.28022827486,140.697603098873,181.353845871499,110.84769973619,96.6011549494553,94.6020624434553,110.85171397817,132.436293104198,94.6060766854352,132.404179168359,110.83565701025],"y":[30.8218061586451,30.2130593694013,25.8366358523072,25.8168142591415,25.4420477032675,24.1981302351936,24.0627104900628,24.0432452713857,23.8685082534042,21.4494150529683,21.066399688473,19.4137452104842,19.3024609804777,18.9891586490655,18.688275637665,18.4210634298515,17.7173321992548,17.2502658006679,14.1243289392766,13.4235836358112,11.1295841985649,10.9740191013083,10.2439728169115,10.1558558025663,9.59505063452015,9.56222755972271,9.52046256014854,9.47880969822171,9.30499437784419,9.19080705994493,7.42824430291614,7.34468903177387,6.94706733768896,6.93106565962437,6.93106565962437,6.88919167640844,6.83736910790214,6.43351260007848,6.34941958303546,5.95537310341777,5.82655472623194,5.43759488755844,5.35114407375997,5.00860645953812,4.43912434261483],"text":["GO:0006366 (2055) <br> transcription by RNA polymerase II <br> 1.507e-31","GO:0006357 (1978) <br> regulation of transcription by RNA polymerase II <br> 6.123e-31","GO:0006351 (2766) <br> DNA-templated transcription <br> 1.457e-26","GO:0097659 (2767) <br> nucleic acid-templated transcription <br> 1.525e-26","GO:0032774 (2782) <br> RNA biosynthetic process <br> 3.614e-26","GO:0051252 (2921) <br> regulation of RNA metabolic process <br> 6.337e-25","GO:0006355 (2662) <br> regulation of DNA-templated transcription <br> 8.655e-25","GO:1903506 (2663) <br> regulation of nucleic acid-templated transcription <br> 9.052e-25","GO:2001141 (2670) <br> regulation of RNA biosynthetic process <br> 1.354e-24","GO:0010556 (3088) <br> regulation of macromolecule biosynthetic process <br> 3.553e-22","GO:0019219 (3181) <br> regulation of nucleobase-containing compound metabolic process <br> 8.582e-22","GO:0034654 (3149) <br> nucleobase-containing compound biosynthetic process <br> 3.857e-20","GO:0031326 (3204) <br> regulation of cellular biosynthetic process <br> 4.984e-20","GO:0018130 (3210) <br> heterocycle biosynthetic process <br> 1.025e-19","GO:0019438 (3212) <br> aromatic compound biosynthetic process <br> 2.050e-19","GO:0009889 (3255) <br> regulation of biosynthetic process <br> 3.793e-19","GO:1901362 (3319) <br> organic cyclic compound biosynthetic process <br> 1.917e-18","GO:0009059 (3801) <br> macromolecule biosynthetic process <br> 5.620e-18","GO:0044271 (3820) <br> cellular nitrogen compound biosynthetic process <br> 7.511e-15","GO:0031323 (4432) <br> regulation of cellular metabolic process <br> 3.771e-14","GO:0080090 (4567) <br> regulation of primary metabolic process <br> 7.420e-12","GO:0051171 (4443) <br> regulation of nitrogen compound metabolic process <br> 1.062e-11","GO:0044249 (4569) <br> cellular biosynthetic process <br> 5.702e-11","GO:0000122 (761) <br> negative regulation of transcription by RNA polymerase II <br> 6.985e-11","GO:1901576 (4628) <br> organic substance biosynthetic process <br> 2.541e-10","GO:0045892 (1041) <br> negative regulation of DNA-templated transcription <br> 2.740e-10","GO:1903507 (1043) <br> negative regulation of nucleic acid-templated transcription <br> 3.017e-10","GO:1902679 (1044) <br> negative regulation of RNA biosynthetic process <br> 3.320e-10","GO:0051253 (1137) <br> negative regulation of RNA metabolic process <br> 4.955e-10","GO:0009058 (4691) <br> biosynthetic process <br> 6.445e-10","GO:0010558 (1237) <br> negative regulation of macromolecule biosynthetic process <br> 3.730e-08","GO:0045934 (1239) <br> negative regulation of nucleobase-containing compound metabolic process <br> 4.522e-08","GO:0010468 (3687) <br> regulation of gene expression <br> 1.130e-07","GO:0045893 (1342) <br> positive regulation of DNA-templated transcription <br> 1.172e-07","GO:1903508 (1342) <br> positive regulation of nucleic acid-templated transcription <br> 1.172e-07","GO:0051254 (1469) <br> positive regulation of RNA metabolic process <br> 1.291e-07","GO:1902680 (1348) <br> positive regulation of RNA biosynthetic process <br> 1.454e-07","GO:0031327 (1279) <br> negative regulation of cellular biosynthetic process <br> 3.685e-07","GO:0010557 (1537) <br> positive regulation of macromolecule biosynthetic process <br> 4.473e-07","GO:0009890 (1304) <br> negative regulation of biosynthetic process <br> 1.108e-06","GO:0031328 (1614) <br> positive regulation of cellular biosynthetic process <br> 1.491e-06","GO:0045944 (952) <br> positive regulation of transcription by RNA polymerase II <br> 3.651e-06","GO:0009891 (1640) <br> positive regulation of biosynthetic process <br> 4.455e-06","GO:0045935 (1641) <br> positive regulation of nucleobase-containing compound metabolic process <br> 9.804e-06","GO:0031324 (1830) <br> negative regulation of cellular metabolic process <br> 3.638e-05"],"key":["GO:0006366","GO:0006357","GO:0006351","GO:0097659","GO:0032774","GO:0051252","GO:0006355","GO:1903506","GO:2001141","GO:0010556","GO:0019219","GO:0034654","GO:0031326","GO:0018130","GO:0019438","GO:0009889","GO:1901362","GO:0009059","GO:0044271","GO:0031323","GO:0080090","GO:0051171","GO:0044249","GO:0000122","GO:1901576","GO:0045892","GO:1903507","GO:1902679","GO:0051253","GO:0009058","GO:0010558","GO:0045934","GO:0010468","GO:0045893","GO:1903508","GO:0051254","GO:1902680","GO:0031327","GO:0010557","GO:0009890","GO:0031328","GO:0045944","GO:0009891","GO:0045935","GO:0031324"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.8,"size":[19.9328889998923,19.7945336813753,20.9713442387267,20.9725693673668,20.9908829468851,21.1551665518898,20.840951144652,20.8422338544322,20.8511966676227,21.3406278723253,21.4388099164726,21.4054160010167,21.4625664648654,21.4687305614409,21.4707822236613,21.5145300343105,21.5783903620437,22.0169227868673,22.0328474320382,22.5012201858246,22.5943797406809,22.5089340071967,22.5957356700136,15.8277343250134,22.6354275307425,17.2581063646558,17.2664017066207,17.2705415054242,17.634214663173,22.6771653543307,17.9843260537941,17.9909516155099,21.9193671229383,18.3146966709246,18.3146966709246,18.6727776861116,18.3325694578545,18.1206413748947,18.8487828891077,18.1990817739118,19.0366358532793,16.86603051892,19.0975483417282,19.0998670059696,19.509021462839],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedDataa8945cba","name":"GO:BP","legendgroup":"GO:BP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[57.0618373107918,68.5889282104043],"y":[4.21361406646978,4.10453105744516],"text":["GO:0005667 (413) <br> transcription regulator complex <br> 6.115e-05","GO:0090575 (219) <br> RNA polymerase II transcription regulator complex <br> 7.861e-05"],"key":["GO:0005667","GO:0090575"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.8,"size":[12.3778381165886,3.77952755905512],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedDataa8945cba","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[3.60978804546664,49.193686440164,2.52990529177455,2.7346414272329,3.69007672603854,2.53391972580315,51.5742458191209,29.4105555472474,2.54596302788893,3.65796125380978,2.54997746191753,2.53793415983174,2.95142086477704,2.97149303492002,2.94740643074844,2.96747860089142,31.3615704851446],"y":[28.0099748427831,27.585010227732,26.6832303803185,26.6207985144869,26.6047605815763,25.7224197259945,25.4685915385632,24.6100747804592,23.8060664190773,23.1433130503246,20.9047468783541,20.7981338439732,5.51414599616135,5.26111233651397,5.15730375359616,4.56899837480342,4.12608104566222],"text":["GO:0003677 (1908) <br> DNA binding <br> 9.773e-29","GO:0140110 (1431) <br> transcription regulator activity <br> 2.600e-28","GO:0000976 (1082) <br> transcription cis-regulatory region binding <br> 2.074e-27","GO:0001067 (1084) <br> transcription regulatory region nucleic acid binding <br> 2.394e-27","GO:0003700 (975) <br> DNA-binding transcription factor activity <br> 2.485e-27","GO:0000977 (998) <br> RNA polymerase II transcription regulatory region sequence-specific DNA binding <br> 1.895e-26","GO:1990837 (1128) <br> sequence-specific double-stranded DNA binding <br> 3.399e-26","GO:0043565 (1220) <br> sequence-specific DNA binding <br> 2.454e-25","GO:0000981 (935) <br> DNA-binding transcription factor activity, RNA polymerase II-specific <br> 1.563e-24","GO:0003690 (1209) <br> double-stranded DNA binding <br> 7.189e-24","GO:0000987 (860) <br> cis-regulatory region sequence-specific DNA binding <br> 1.245e-21","GO:0000978 (843) <br> RNA polymerase II cis-regulatory region sequence-specific DNA binding <br> 1.592e-21","GO:0001217 (246) <br> DNA-binding transcription repressor activity <br> 3.061e-06","GO:0001228 (304) <br> DNA-binding transcription activator activity, RNA polymerase II-specific <br> 5.481e-06","GO:0001216 (308) <br> DNA-binding transcription activator activity <br> 6.961e-06","GO:0001227 (242) <br> DNA-binding transcription repressor activity, RNA polymerase II-specific <br> 2.698e-05","GO:0046872 (3328) <br> metal ion binding <br> 7.480e-05"],"key":["GO:0003677","GO:0140110","GO:0000976","GO:0001067","GO:0003700","GO:0000977","GO:1990837","GO:0043565","GO:0000981","GO:0003690","GO:0000987","GO:0000978","GO:0001217","GO:0001228","GO:0001216","GO:0001227","GO:0046872"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.8,"size":[19.6628950729895,18.5698805137589,17.4240870389712,17.4319713962061,16.9718997050129,17.0744865851853,17.6007501676041,17.9274452632219,16.7855515823337,17.8900919690294,16.4053806402216,16.3128978192306,7.4604324750387,9.9617919393928,10.0837982825388,7.19105305613139,21.58725367991],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedDataa8945cba","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,51.7629242184649],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[55.777034424036,73.7080647123221],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[77.7221749178932,200.004014110206],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.2835201328352,"r":6.6417600664176,"b":32.6669625382871,"l":55.1930261519303},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP"],"tickvals":[26.8814621092325,64.7425495681791,138.863094514049],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.6268161062682},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.132835201328352,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,36],"tickmode":"array","ticktext":["0","10","20","30"],"tickvals":[0,10,20,30],"categoryorder":"array","categoryarray":["0","10","20","30"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":0,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis","xshift":-31.5483603154836},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.9730178497302,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"1b9044031b3d":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"1b907c214a7c":{"x":{},"xend":{},"y":{},"yend":{}},"1b9024571b7":{"x":{},"xend":{},"y":{},"yend":{}},"1b907d76597a":{"x":{},"xend":{},"y":{},"yend":{}}},"cur_data":"1b9044031b3d","visdat":{"1b9044031b3d":["function (y) ","x"],"1b907c214a7c":["function (y) ","x"],"1b9024571b7":["function (y) ","x"],"1b907d76597a":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedDataa8945cba"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p><img src="figure/GO_analysis.Rmd/intial%20analysis-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>I first looked at the data with all genes from the sigDA3 dataset. I
used the list of all genes based on my rowMeans&gt;0 filtering as
background.</p>
<div id="analysis-of-up-versus-down" class="section level3">
<h3>Analysis of Up versus Down</h3>
<p>I then separated the VDA3 file by log2 Fold Change to see how the
gene sets are enriched. Nothing showed up in the GO-BP/CC/MG-down
regulated gene-set at a significant level, p&lt;0.05.</p>
<div class="plotly html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-1a0d15c6ca5bba774942" style="width:672px;height:480px;"></div>
<script type="application/json" data-for="htmlwidget-1a0d15c6ca5bba774942">{"x":{"data":[{"x":[87.1596578126843,87.1877575065437,87.1355723608048,165.553789438535,140.689574614913,113.878452431029,87.1516293287245,184.2721997909,199.345678425499,96.5971407074754,103.678263560054,110.84368549421,117.84452350719,94.5980482014754,101.84375497523,104.489140439997,176.685282448852,92.4022578384594,110.83164276827,128.642834433174,161.33482111764,140.420620402258,96.3643146726401,128.594663529415,177.476088118897,77.9750721626281,140.693588856893,92.3982435964794,132.239595247182,184.27621403288,181.349831629519,132.400164926379,96.6051691914352,146.004430996326,140.697603098873,110.84769973619,184.28022827486,132.243609489162,181.353845871499,94.6020624434553,96.6011549494553,110.85171397817,103.690306285993,94.6060766854352,132.404179168359,132.436293104198,110.83565701025,110.83967125223,140.428648886218,140.424634644238,96.7697531126119,146.598538809355,85.1404940967847,94.6141051693951,85.1364798548048],"y":[34.9926384078061,34.9719470285102,30.9562797196672,30.9368670038422,30.8413456744103,30.5696284769525,29.9789142970439,29.9594854202717,29.7850205960405,28.1557201804136,26.8175178433716,25.3559714522605,24.9241748436204,24.3555768455875,23.7315219537352,23.5549452561497,21.7403783516825,21.0552744725141,19.0474150763866,17.7455966752712,16.0705251267004,15.9675535386176,12.3606817076219,12.2087944488686,11.570257374882,11.0762336634655,10.9293920477093,10.8197444560198,10.8174483300484,10.7776345842476,10.7379194215558,8.66784927919169,8.65029412118999,7.91568405981336,7.8522195859478,7.69896710543731,7.50711159615415,7.50711159615415,7.41986999399017,7.2391908319336,7.08637665442872,6.77413557703562,6.41193781199028,6.32073101371132,6.22792392001201,6.09293051329191,6.02995999207308,5.03316946685607,4.23397383792113,4.12361191808874,3.17908940150795,2.8983696781421,2.70735392552274,2.35492305525189,2.08848372812711],"text":["GO:0006357 (1978) <br> regulation of transcription by RNA polymerase II <br> 1.017e-35","GO:0006366 (2055) <br> transcription by RNA polymerase II <br> 1.067e-35","GO:0006351 (2766) <br> DNA-templated transcription <br> 1.106e-31","GO:0097659 (2767) <br> nucleic acid-templated transcription <br> 1.156e-31","GO:0051252 (2921) <br> regulation of RNA metabolic process <br> 1.441e-31","GO:0032774 (2782) <br> RNA biosynthetic process <br> 2.694e-31","GO:0006355 (2662) <br> regulation of DNA-templated transcription <br> 1.050e-30","GO:1903506 (2663) <br> regulation of nucleic acid-templated transcription <br> 1.098e-30","GO:2001141 (2670) <br> regulation of RNA biosynthetic process <br> 1.641e-30","GO:0010556 (3088) <br> regulation of macromolecule biosynthetic process <br> 6.987e-29","GO:0019219 (3181) <br> regulation of nucleobase-containing compound metabolic process <br> 1.522e-27","GO:0031326 (3204) <br> regulation of cellular biosynthetic process <br> 4.406e-26","GO:0034654 (3149) <br> nucleobase-containing compound biosynthetic process <br> 1.191e-25","GO:0009889 (3255) <br> regulation of biosynthetic process <br> 4.410e-25","GO:0018130 (3210) <br> heterocycle biosynthetic process <br> 1.856e-24","GO:0019438 (3212) <br> aromatic compound biosynthetic process <br> 2.786e-24","GO:1901362 (3319) <br> organic cyclic compound biosynthetic process <br> 1.818e-22","GO:0009059 (3801) <br> macromolecule biosynthetic process <br> 8.805e-22","GO:0031323 (4432) <br> regulation of cellular metabolic process <br> 8.966e-20","GO:0044271 (3820) <br> cellular nitrogen compound biosynthetic process <br> 1.796e-18","GO:0080090 (4567) <br> regulation of primary metabolic process <br> 8.501e-17","GO:0051171 (4443) <br> regulation of nitrogen compound metabolic process <br> 1.078e-16","GO:0010468 (3687) <br> regulation of gene expression <br> 4.358e-13","GO:0044249 (4569) <br> cellular biosynthetic process <br> 6.183e-13","GO:1901576 (4628) <br> organic substance biosynthetic process <br> 2.690e-12","GO:0000122 (761) <br> negative regulation of transcription by RNA polymerase II <br> 8.390e-12","GO:0051253 (1137) <br> negative regulation of RNA metabolic process <br> 1.177e-11","GO:0009058 (4691) <br> biosynthetic process <br> 1.514e-11","GO:0045892 (1041) <br> negative regulation of DNA-templated transcription <br> 1.522e-11","GO:1903507 (1043) <br> negative regulation of nucleic acid-templated transcription <br> 1.669e-11","GO:1902679 (1044) <br> negative regulation of RNA biosynthetic process <br> 1.828e-11","GO:0045934 (1239) <br> negative regulation of nucleobase-containing compound metabolic process <br> 2.149e-09","GO:0010558 (1237) <br> negative regulation of macromolecule biosynthetic process <br> 2.237e-09","GO:0060255 (4772) <br> regulation of macromolecule metabolic process <br> 1.214e-08","GO:0051254 (1469) <br> positive regulation of RNA metabolic process <br> 1.405e-08","GO:0031327 (1279) <br> negative regulation of cellular biosynthetic process <br> 2.000e-08","GO:1903508 (1342) <br> positive regulation of nucleic acid-templated transcription <br> 3.111e-08","GO:0045893 (1342) <br> positive regulation of DNA-templated transcription <br> 3.111e-08","GO:1902680 (1348) <br> positive regulation of RNA biosynthetic process <br> 3.803e-08","GO:0009890 (1304) <br> negative regulation of biosynthetic process <br> 5.765e-08","GO:0010557 (1537) <br> positive regulation of macromolecule biosynthetic process <br> 8.196e-08","GO:0031328 (1614) <br> positive regulation of cellular biosynthetic process <br> 1.682e-07","GO:0019222 (5189) <br> regulation of metabolic process <br> 3.873e-07","GO:0009891 (1640) <br> positive regulation of biosynthetic process <br> 4.778e-07","GO:0045935 (1641) <br> positive regulation of nucleobase-containing compound metabolic process <br> 5.917e-07","GO:0045944 (952) <br> positive regulation of transcription by RNA polymerase II <br> 8.074e-07","GO:0031324 (1830) <br> negative regulation of cellular metabolic process <br> 9.333e-07","GO:0031325 (2446) <br> positive regulation of cellular metabolic process <br> 9.265e-06","GO:0051173 (2497) <br> positive regulation of nitrogen compound metabolic process <br> 5.835e-05","GO:0051172 (1905) <br> negative regulation of nitrogen compound metabolic process <br> 7.523e-05","GO:0010604 (2729) <br> positive regulation of macromolecule metabolic process <br> 6.621e-04","GO:0060411 (67) <br> cardiac septum morphogenesis <br> 1.264e-03","GO:0003281 (70) <br> ventricular septum development <br> 1.962e-03","GO:0009893 (2972) <br> positive regulation of metabolic process <br> 4.416e-03","GO:0003279 (102) <br> cardiac septum development <br> 8.157e-03"],"key":["GO:0006357","GO:0006366","GO:0006351","GO:0097659","GO:0051252","GO:0032774","GO:0006355","GO:1903506","GO:2001141","GO:0010556","GO:0019219","GO:0031326","GO:0034654","GO:0009889","GO:0018130","GO:0019438","GO:1901362","GO:0009059","GO:0031323","GO:0044271","GO:0080090","GO:0051171","GO:0010468","GO:0044249","GO:1901576","GO:0000122","GO:0051253","GO:0009058","GO:0045892","GO:1903507","GO:1902679","GO:0045934","GO:0010558","GO:0060255","GO:0051254","GO:0031327","GO:1903508","GO:0045893","GO:1902680","GO:0009890","GO:0010557","GO:0031328","GO:0019222","GO:0009891","GO:0045935","GO:0045944","GO:0031324","GO:0031325","GO:0051173","GO:0051172","GO:0010604","GO:0060411","GO:0003281","GO:0009893","GO:0003279"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.8,"size":[20.4509081370983,20.5446836280313,21.257106944793,21.2579559548973,21.3847089888028,21.2706494989905,21.1668556404719,21.1677423917556,21.1739390805522,21.5138786939824,21.5824319374815,21.5990370116592,21.5591022394644,21.63538169264,21.6033466370339,21.6047811577455,21.6800917613345,21.9884149987099,22.3314465781081,21.9996530364737,22.3977252086938,22.3369310759708,21.9196328809265,22.398690571078,22.4269580735081,17.9042694057372,19.0266390858246,22.4567006099302,18.7872549590907,18.7925043255409,18.7951245483966,19.2562227939711,19.2519370451829,22.4942916353457,19.7014801990592,19.3402752135322,19.4666128176827,19.4666128176827,19.4782825604275,19.3912614715106,19.8177268766811,19.9423659682431,22.6771653543307,19.9829045427899,19.9844488471854,18.5407626928052,20.2582916248577,20.9658947820798,21.0151162752745,20.3580517914707,21.2254464131897,3.77952755905512,5.67593973589361,21.4250245259244,9.65377369680358],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedDatad3736c7f","name":"GO:BP","legendgroup":"GO:BP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[57.0176722115596,56.1986103712528,68.5889282104043,57.0618373107918],"y":[4.15854866612688,3.23780674223291,3.18703919427696,3.06446214515811],"text":["GO:0005654 (3669) <br> nucleoplasm <br> 6.941e-05","GO:0000785 (884) <br> chromatin <br> 5.784e-04","GO:0090575 (219) <br> RNA polymerase II transcription regulator complex <br> 6.501e-04","GO:0005667 (413) <br> transcription regulator complex <br> 8.621e-04"],"key":["GO:0005654","GO:0000785","GO:0090575","GO:0005667"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.8,"size":[21.9085541544034,18.3332036292077,13.6406885144731,15.9994872973395],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedDatad3736c7f","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[49.193686440164,3.60978804546664,3.69007672603854,2.52990529177455,2.7346414272329,2.53391972580315,51.5742458191209,29.4105555472474,2.54596302788893,3.65796125380978,2.54997746191753,2.53793415983174,31.3615704851446,29.1456029013601,2.95142086477704,2.96747860089142,2.97149303492002,2.94740643074844,3.60577361143804,31.0323868947998,9.99273815093285,49.2820039887931],"y":[32.3626741405966,32.04452873889,29.2781254749774,28.569500751919,28.5109053511107,28.2449065330052,27.5240281369125,27.0898291701502,25.855734241005,25.4188771908728,25.3858106585396,25.3772829648675,7.19175409826462,6.36093657585566,6.25840197769239,5.19449660689582,4.66445241452034,4.57905970368937,4.57819818985417,2.39495836835457,2.385902276404,2.21310729614734],"text":["GO:0140110 (1431) <br> transcription regulator activity <br> 4.338e-33","GO:0003677 (1908) <br> DNA binding <br> 9.025e-33","GO:0003700 (975) <br> DNA-binding transcription factor activity <br> 5.271e-30","GO:0000976 (1082) <br> transcription cis-regulatory region binding <br> 2.695e-29","GO:0001067 (1084) <br> transcription regulatory region nucleic acid binding <br> 3.084e-29","GO:0000977 (998) <br> RNA polymerase II transcription regulatory region sequence-specific DNA binding <br> 5.690e-29","GO:1990837 (1128) <br> sequence-specific double-stranded DNA binding <br> 2.992e-28","GO:0043565 (1220) <br> sequence-specific DNA binding <br> 8.132e-28","GO:0000981 (935) <br> DNA-binding transcription factor activity, RNA polymerase II-specific <br> 1.394e-26","GO:0003690 (1209) <br> double-stranded DNA binding <br> 3.812e-26","GO:0000987 (860) <br> cis-regulatory region sequence-specific DNA binding <br> 4.113e-26","GO:0000978 (843) <br> RNA polymerase II cis-regulatory region sequence-specific DNA binding <br> 4.195e-26","GO:0046872 (3328) <br> metal ion binding <br> 6.431e-08","GO:0043169 (3392) <br> cation binding <br> 4.356e-07","GO:0001217 (246) <br> DNA-binding transcription repressor activity <br> 5.516e-07","GO:0001227 (242) <br> DNA-binding transcription repressor activity, RNA polymerase II-specific <br> 6.390e-06","GO:0001228 (304) <br> DNA-binding transcription activator activity, RNA polymerase II-specific <br> 2.165e-05","GO:0001216 (308) <br> DNA-binding transcription activator activity <br> 2.636e-05","GO:0003676 (3264) <br> nucleic acid binding <br> 2.641e-05","GO:0046332 (71) <br> SMAD binding <br> 4.028e-03","GO:0008134 (520) <br> transcription factor binding <br> 4.112e-03","GO:0140297 (423) <br> DNA-binding transcription factor binding <br> 6.122e-03"],"key":["GO:0140110","GO:0003677","GO:0003700","GO:0000976","GO:0001067","GO:0000977","GO:1990837","GO:0043565","GO:0000981","GO:0003690","GO:0000987","GO:0000978","GO:0046872","GO:0043169","GO:0001217","GO:0001227","GO:0001228","GO:0001216","GO:0003676","GO:0046332","GO:0008134","GO:0140297"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.8,"size":[19.633762214844,20.3619478278103,18.6070050879752,18.8925522997128,18.8975677920573,18.6714184844922,19.0052269779247,19.2151771458461,18.4905665351014,19.1910698856366,18.2553555477481,18.1986242805929,21.6863010294926,21.729916771193,14.113351312744,14.0480188207092,14.9226445813289,14.9706995141141,21.6417287051014,5.96149850414743,16.750367195814,16.079597677797],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedDatad3736c7f","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,51.7629242184649],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[55.777034424036,73.7080647123221],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[77.7221749178932,200.004014110206],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.2835201328352,"r":6.6417600664176,"b":55.8781661388202,"l":55.1930261519303},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP"],"tickvals":[26.8814621092325,64.7425495681791,138.863094514049],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.6268161062682},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.132835201328352,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,40],"tickmode":"array","ticktext":["0","10","20","30","40"],"tickvals":[0,10,20,30,40],"categoryorder":"array","categoryarray":["0","10","20","30","40"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":0,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis","xshift":-31.5483603154836},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.9730178497302,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"1b9040f918c2":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"1b9015c5666b":{"x":{},"xend":{},"y":{},"yend":{}},"1b903b9e7adb":{"x":{},"xend":{},"y":{},"yend":{}},"1b9072fecf5":{"x":{},"xend":{},"y":{},"yend":{}}},"cur_data":"1b9040f918c2","visdat":{"1b9040f918c2":["function (y) ","x"],"1b9015c5666b":["function (y) ","x"],"1b903b9e7adb":["function (y) ","x"],"1b9072fecf5":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedDatad3736c7f"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<div
id="i-next-wanted-to-see-what-happened-at-24-hours-with-daunorubicin.-i-used-the-sigvda24-file-to-do-this."
class="section level4">
<h4>I next wanted to see what happened at 24 hours with daunorubicin. I
used the sigVDA24 file to do this.</h4>
<p>unfortunately the enrichment below 0.0001</p>
<div class="plotly html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-001f7902152dbd8c1251" style="width:960px;height:960px;"></div>
<script type="application/json" data-for="htmlwidget-001f7902152dbd8c1251">{"x":{"data":[{"x":[87.1877575065437,87.1596578126843,87.1355723608048,165.553789438535,113.878452431029,140.689574614913,87.1516293287245,184.2721997909,199.345678425499,96.5971407074754,103.678263560054,117.84452350719,110.84368549421,101.84375497523,104.489140439997,94.5980482014754,176.685282448852,92.4022578384594,128.642834433174,110.83164276827,161.33482111764,140.420620402258,128.594663529415,77.9750721626281,177.476088118897,132.239595247182,184.27621403288,181.349831629519,140.693588856893,92.3982435964794,96.6051691914352,132.400164926379,96.3643146726401,132.243609489162,184.28022827486,140.697603098873,181.353845871499,110.84769973619,96.6011549494553,94.6020624434553,110.85171397817,132.436293104198,94.6060766854352,132.404179168359,110.83565701025],"y":[30.8218061586451,30.2130593694013,25.8366358523072,25.8168142591415,25.4420477032675,24.1981302351936,24.0627104900628,24.0432452713857,23.8685082534042,21.4494150529683,21.066399688473,19.4137452104842,19.3024609804777,18.9891586490655,18.688275637665,18.4210634298515,17.7173321992548,17.2502658006679,14.1243289392766,13.4235836358112,11.1295841985649,10.9740191013083,10.2439728169115,10.1558558025663,9.59505063452015,9.56222755972271,9.52046256014854,9.47880969822171,9.30499437784419,9.19080705994493,7.42824430291614,7.34468903177387,6.94706733768896,6.93106565962437,6.93106565962437,6.88919167640844,6.83736910790214,6.43351260007848,6.34941958303546,5.95537310341777,5.82655472623194,5.43759488755844,5.35114407375997,5.00860645953812,4.43912434261483],"text":["GO:0006366 (2055) <br> transcription by RNA polymerase II <br> 1.507e-31","GO:0006357 (1978) <br> regulation of transcription by RNA polymerase II <br> 6.123e-31","GO:0006351 (2766) <br> DNA-templated transcription <br> 1.457e-26","GO:0097659 (2767) <br> nucleic acid-templated transcription <br> 1.525e-26","GO:0032774 (2782) <br> RNA biosynthetic process <br> 3.614e-26","GO:0051252 (2921) <br> regulation of RNA metabolic process <br> 6.337e-25","GO:0006355 (2662) <br> regulation of DNA-templated transcription <br> 8.655e-25","GO:1903506 (2663) <br> regulation of nucleic acid-templated transcription <br> 9.052e-25","GO:2001141 (2670) <br> regulation of RNA biosynthetic process <br> 1.354e-24","GO:0010556 (3088) <br> regulation of macromolecule biosynthetic process <br> 3.553e-22","GO:0019219 (3181) <br> regulation of nucleobase-containing compound metabolic process <br> 8.582e-22","GO:0034654 (3149) <br> nucleobase-containing compound biosynthetic process <br> 3.857e-20","GO:0031326 (3204) <br> regulation of cellular biosynthetic process <br> 4.984e-20","GO:0018130 (3210) <br> heterocycle biosynthetic process <br> 1.025e-19","GO:0019438 (3212) <br> aromatic compound biosynthetic process <br> 2.050e-19","GO:0009889 (3255) <br> regulation of biosynthetic process <br> 3.793e-19","GO:1901362 (3319) <br> organic cyclic compound biosynthetic process <br> 1.917e-18","GO:0009059 (3801) <br> macromolecule biosynthetic process <br> 5.620e-18","GO:0044271 (3820) <br> cellular nitrogen compound biosynthetic process <br> 7.511e-15","GO:0031323 (4432) <br> regulation of cellular metabolic process <br> 3.771e-14","GO:0080090 (4567) <br> regulation of primary metabolic process <br> 7.420e-12","GO:0051171 (4443) <br> regulation of nitrogen compound metabolic process <br> 1.062e-11","GO:0044249 (4569) <br> cellular biosynthetic process <br> 5.702e-11","GO:0000122 (761) <br> negative regulation of transcription by RNA polymerase II <br> 6.985e-11","GO:1901576 (4628) <br> organic substance biosynthetic process <br> 2.541e-10","GO:0045892 (1041) <br> negative regulation of DNA-templated transcription <br> 2.740e-10","GO:1903507 (1043) <br> negative regulation of nucleic acid-templated transcription <br> 3.017e-10","GO:1902679 (1044) <br> negative regulation of RNA biosynthetic process <br> 3.320e-10","GO:0051253 (1137) <br> negative regulation of RNA metabolic process <br> 4.955e-10","GO:0009058 (4691) <br> biosynthetic process <br> 6.445e-10","GO:0010558 (1237) <br> negative regulation of macromolecule biosynthetic process <br> 3.730e-08","GO:0045934 (1239) <br> negative regulation of nucleobase-containing compound metabolic process <br> 4.522e-08","GO:0010468 (3687) <br> regulation of gene expression <br> 1.130e-07","GO:0045893 (1342) <br> positive regulation of DNA-templated transcription <br> 1.172e-07","GO:1903508 (1342) <br> positive regulation of nucleic acid-templated transcription <br> 1.172e-07","GO:0051254 (1469) <br> positive regulation of RNA metabolic process <br> 1.291e-07","GO:1902680 (1348) <br> positive regulation of RNA biosynthetic process <br> 1.454e-07","GO:0031327 (1279) <br> negative regulation of cellular biosynthetic process <br> 3.685e-07","GO:0010557 (1537) <br> positive regulation of macromolecule biosynthetic process <br> 4.473e-07","GO:0009890 (1304) <br> negative regulation of biosynthetic process <br> 1.108e-06","GO:0031328 (1614) <br> positive regulation of cellular biosynthetic process <br> 1.491e-06","GO:0045944 (952) <br> positive regulation of transcription by RNA polymerase II <br> 3.651e-06","GO:0009891 (1640) <br> positive regulation of biosynthetic process <br> 4.455e-06","GO:0045935 (1641) <br> positive regulation of nucleobase-containing compound metabolic process <br> 9.804e-06","GO:0031324 (1830) <br> negative regulation of cellular metabolic process <br> 3.638e-05"],"key":["GO:0006366","GO:0006357","GO:0006351","GO:0097659","GO:0032774","GO:0051252","GO:0006355","GO:1903506","GO:2001141","GO:0010556","GO:0019219","GO:0034654","GO:0031326","GO:0018130","GO:0019438","GO:0009889","GO:1901362","GO:0009059","GO:0044271","GO:0031323","GO:0080090","GO:0051171","GO:0044249","GO:0000122","GO:1901576","GO:0045892","GO:1903507","GO:1902679","GO:0051253","GO:0009058","GO:0010558","GO:0045934","GO:0010468","GO:0045893","GO:1903508","GO:0051254","GO:1902680","GO:0031327","GO:0010557","GO:0009890","GO:0031328","GO:0045944","GO:0009891","GO:0045935","GO:0031324"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.8,"size":[19.9328889998923,19.7945336813753,20.9713442387267,20.9725693673668,20.9908829468851,21.1551665518898,20.840951144652,20.8422338544322,20.8511966676227,21.3406278723253,21.4388099164726,21.4054160010167,21.4625664648654,21.4687305614409,21.4707822236613,21.5145300343105,21.5783903620437,22.0169227868673,22.0328474320382,22.5012201858246,22.5943797406809,22.5089340071967,22.5957356700136,15.8277343250134,22.6354275307425,17.2581063646558,17.2664017066207,17.2705415054242,17.634214663173,22.6771653543307,17.9843260537941,17.9909516155099,21.9193671229383,18.3146966709246,18.3146966709246,18.6727776861116,18.3325694578545,18.1206413748947,18.8487828891077,18.1990817739118,19.0366358532793,16.86603051892,19.0975483417282,19.0998670059696,19.509021462839],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedData9cf5f559","name":"GO:BP","legendgroup":"GO:BP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[57.0618373107918,68.5889282104043],"y":[4.21361406646978,4.10453105744516],"text":["GO:0005667 (413) <br> transcription regulator complex <br> 6.115e-05","GO:0090575 (219) <br> RNA polymerase II transcription regulator complex <br> 7.861e-05"],"key":["GO:0005667","GO:0090575"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.8,"size":[12.3778381165886,3.77952755905512],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedData9cf5f559","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[3.60978804546664,49.193686440164,2.52990529177455,2.7346414272329,3.69007672603854,2.53391972580315,51.5742458191209,29.4105555472474,2.54596302788893,3.65796125380978,2.54997746191753,2.53793415983174,2.95142086477704,2.97149303492002,2.94740643074844,2.96747860089142,31.3615704851446],"y":[28.0099748427831,27.585010227732,26.6832303803185,26.6207985144869,26.6047605815763,25.7224197259945,25.4685915385632,24.6100747804592,23.8060664190773,23.1433130503246,20.9047468783541,20.7981338439732,5.51414599616135,5.26111233651397,5.15730375359616,4.56899837480342,4.12608104566222],"text":["GO:0003677 (1908) <br> DNA binding <br> 9.773e-29","GO:0140110 (1431) <br> transcription regulator activity <br> 2.600e-28","GO:0000976 (1082) <br> transcription cis-regulatory region binding <br> 2.074e-27","GO:0001067 (1084) <br> transcription regulatory region nucleic acid binding <br> 2.394e-27","GO:0003700 (975) <br> DNA-binding transcription factor activity <br> 2.485e-27","GO:0000977 (998) <br> RNA polymerase II transcription regulatory region sequence-specific DNA binding <br> 1.895e-26","GO:1990837 (1128) <br> sequence-specific double-stranded DNA binding <br> 3.399e-26","GO:0043565 (1220) <br> sequence-specific DNA binding <br> 2.454e-25","GO:0000981 (935) <br> DNA-binding transcription factor activity, RNA polymerase II-specific <br> 1.563e-24","GO:0003690 (1209) <br> double-stranded DNA binding <br> 7.189e-24","GO:0000987 (860) <br> cis-regulatory region sequence-specific DNA binding <br> 1.245e-21","GO:0000978 (843) <br> RNA polymerase II cis-regulatory region sequence-specific DNA binding <br> 1.592e-21","GO:0001217 (246) <br> DNA-binding transcription repressor activity <br> 3.061e-06","GO:0001228 (304) <br> DNA-binding transcription activator activity, RNA polymerase II-specific <br> 5.481e-06","GO:0001216 (308) <br> DNA-binding transcription activator activity <br> 6.961e-06","GO:0001227 (242) <br> DNA-binding transcription repressor activity, RNA polymerase II-specific <br> 2.698e-05","GO:0046872 (3328) <br> metal ion binding <br> 7.480e-05"],"key":["GO:0003677","GO:0140110","GO:0000976","GO:0001067","GO:0003700","GO:0000977","GO:1990837","GO:0043565","GO:0000981","GO:0003690","GO:0000987","GO:0000978","GO:0001217","GO:0001228","GO:0001216","GO:0001227","GO:0046872"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.8,"size":[19.6628950729895,18.5698805137589,17.4240870389712,17.4319713962061,16.9718997050129,17.0744865851853,17.6007501676041,17.9274452632219,16.7855515823337,17.8900919690294,16.4053806402216,16.3128978192306,7.4604324750387,9.9617919393928,10.0837982825388,7.19105305613139,21.58725367991],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedData9cf5f559","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,51.7629242184649],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[55.777034424036,73.7080647123221],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[77.7221749178932,200.004014110206],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.2835201328352,"r":6.6417600664176,"b":45.251350032552,"l":55.1930261519303},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP"],"tickvals":[26.8814621092325,64.7425495681791,138.863094514049],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.6268161062682},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.132835201328352,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,36],"tickmode":"array","ticktext":["0","10","20","30"],"tickvals":[0,10,20,30],"categoryorder":"array","categoryarray":["0","10","20","30"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":0,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis","xshift":-31.5483603154836},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.9730178497302,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"1b90166d1073":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"1b9070df74dd":{"x":{},"xend":{},"y":{},"yend":{}},"1b9041393d93":{"x":{},"xend":{},"y":{},"yend":{}},"1b9067eb3535":{"x":{},"xend":{},"y":{},"yend":{}}},"cur_data":"1b90166d1073","visdat":{"1b90166d1073":["function (y) ","x"],"1b9070df74dd":["function (y) ","x"],"1b9041393d93":["function (y) ","x"],"1b9067eb3535":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedData9cf5f559"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p><img src="figure/GO_analysis.Rmd/DA24-2.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="graphing-specific-gene-expression" class="section level1">
<h1>Graphing specific gene expression</h1>
<p>First get a list of genes you want to see. There are multiple was to
“see” these. I used the word ‘apple’ to store my list</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] VennDiagram_1.7.3   futile.logger_1.4.3 gridExtra_2.3      
 [4] BiocGenerics_0.42.0 forcats_1.0.0       stringr_1.5.0      
 [7] dplyr_1.1.0         purrr_1.0.1         readr_2.1.3        
[10] tidyr_1.3.0         tibble_3.1.8        ggplot2_3.4.0      
[13] tidyverse_1.3.2     gprofiler2_0.2.1    workflowr_1.7.0    

loaded via a namespace (and not attached):
 [1] bitops_1.0-7         fs_1.6.0             bit64_4.0.5         
 [4] lubridate_1.9.1      httr_1.4.4           rprojroot_2.0.3     
 [7] tools_4.2.2          backports_1.4.1      bslib_0.4.2         
[10] utf8_1.2.2           R6_2.5.1             DBI_1.1.3           
[13] lazyeval_0.2.2       colorspace_2.1-0     withr_2.5.0         
[16] tidyselect_1.2.0     processx_3.8.0       bit_4.0.5           
[19] compiler_4.2.2       git2r_0.31.0         textshaping_0.3.6   
[22] cli_3.6.0            rvest_1.0.3          formatR_1.14        
[25] xml2_1.3.3           plotly_4.10.1        labeling_0.4.2      
[28] sass_0.4.5           scales_1.2.1         callr_3.7.3         
[31] systemfonts_1.0.4    digest_0.6.31        rmarkdown_2.20      
[34] pkgconfig_2.0.3      htmltools_0.5.4      highr_0.10          
[37] dbplyr_2.3.0         fastmap_1.1.0        htmlwidgets_1.6.1   
[40] rlang_1.0.6          readxl_1.4.1         rstudioapi_0.14     
[43] shiny_1.7.4          jquerylib_0.1.4      generics_0.1.3      
[46] jsonlite_1.8.4       crosstalk_1.2.0      vroom_1.6.1         
[49] googlesheets4_1.0.1  RCurl_1.98-1.10      magrittr_2.0.3      
[52] Rcpp_1.0.10          munsell_0.5.0        fansi_1.0.4         
[55] lifecycle_1.0.3      stringi_1.7.12       whisker_0.4.1       
[58] yaml_2.3.7           parallel_4.2.2       promises_1.2.0.1    
[61] crayon_1.5.2         haven_2.5.1          hms_1.1.2           
[64] knitr_1.42           ps_1.7.2             pillar_1.8.1        
[67] futile.options_1.0.1 reprex_2.0.2         glue_1.6.2          
[70] evaluate_0.20        getPass_0.2-2        lambda.r_1.2.4      
[73] data.table_1.14.6    modelr_0.1.10        vctrs_0.5.2         
[76] tzdb_0.3.0           httpuv_1.6.8         cellranger_1.1.0    
[79] gtable_0.3.1         assertthat_0.2.1     cachem_1.0.6        
[82] xfun_0.36            mime_0.12            xtable_1.8-4        
[85] broom_1.0.3          later_1.3.0          ragg_1.2.5          
[88] googledrive_2.0.0    viridisLite_0.4.1    gargle_1.2.1        
[91] timechange_0.2.0     ellipsis_0.3.2      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
