<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="ERM" />

<meta name="date" content="2023-06-15" />

<title>GTEx_genes</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cardiotoxicity</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/reneeisnowhere/Cardiotoxicity">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GTEx_genes</h1>
<h4 class="author">ERM</h4>
<h4 class="date">2023-06-15</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-06-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Cardiotoxicity/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230109code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230109)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230109code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230109)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereCardiotoxicitytreed5478771e9d452749ccdbd07bbe537ad93cf2597targetblankd547877a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/tree/d5478771e9d452749ccdbd07bbe537ad93cf2597" target="_blank">d547877</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereCardiotoxicitytreed5478771e9d452749ccdbd07bbe537ad93cf2597targetblankd547877a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/tree/d5478771e9d452749ccdbd07bbe537ad93cf2597" target="_blank">d547877</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/41588_2018_171_MOESM3_ESMeQTL_ST2_for paper.csv
    Ignored:    data/Arr_GWAS.txt
    Ignored:    data/Arr_geneset.RDS
    Ignored:    data/BC_cell_lines.csv
    Ignored:    data/CADGWASgene_table.csv
    Ignored:    data/CAD_geneset.RDS
    Ignored:    data/Clamp_Summary.csv
    Ignored:    data/Cormotif_24_k1-5_raw.RDS
    Ignored:    data/DAgostres24.RDS
    Ignored:    data/DAtable1.csv
    Ignored:    data/DDEMresp_list.csv
    Ignored:    data/DDE_reQTL.txt
    Ignored:    data/DDEresp_list.csv
    Ignored:    data/DEG-GO/
    Ignored:    data/DEG_cormotif.RDS
    Ignored:    data/DF_Plate_Peak.csv
    Ignored:    data/DRC48hoursdata.csv
    Ignored:    data/Da24counts.txt
    Ignored:    data/Dx24counts.txt
    Ignored:    data/Dx_reQTL_specific.txt
    Ignored:    data/Ep24counts.txt
    Ignored:    data/GOIsig.csv
    Ignored:    data/GOplots.R
    Ignored:    data/GTEX_setsimple.csv
    Ignored:    data/GTEx_gene_list.csv
    Ignored:    data/HFGWASgene_table.csv
    Ignored:    data/HF_geneset.RDS
    Ignored:    data/Heart_Left_Ventricle.v8.egenes.txt
    Ignored:    data/Hf_GWAS.txt
    Ignored:    data/K_cluster
    Ignored:    data/K_cluster_kisthree.csv
    Ignored:    data/K_cluster_kistwo.csv
    Ignored:    data/LDH48hoursdata.csv
    Ignored:    data/Mt24counts.txt
    Ignored:    data/RINsamplelist.txt
    Ignored:    data/Seonane2019supp1.txt
    Ignored:    data/TOP2Bi-24hoursGO_analysis.csv
    Ignored:    data/TR24counts.txt
    Ignored:    data/Top2biresp_cluster24h.csv
    Ignored:    data/Viabilitylistfull.csv
    Ignored:    data/allexpressedgenes.txt
    Ignored:    data/allgenes.txt
    Ignored:    data/allmatrix.RDS
    Ignored:    data/avgLD50.RDS
    Ignored:    data/backGL.txt
    Ignored:    data/cormotif_3hk1-8.RDS
    Ignored:    data/cormotif_initalK5.RDS
    Ignored:    data/cormotif_initialK5.RDS
    Ignored:    data/cormotif_initialall.RDS
    Ignored:    data/counts24hours.RDS
    Ignored:    data/cpmnorm_counts.csv
    Ignored:    data/crispr_genes.csv
    Ignored:    data/cvd_GWAS.txt
    Ignored:    data/dat_cpm.RDS
    Ignored:    data/data_outline.txt
    Ignored:    data/efit2.RDS
    Ignored:    data/efit2results.RDS
    Ignored:    data/ensembl_backup.RDS
    Ignored:    data/ensgtotal.txt
    Ignored:    data/filenameonly.txt
    Ignored:    data/filtered_cpm_counts.csv
    Ignored:    data/filtered_raw_counts.csv
    Ignored:    data/filtermatrix_x.RDS
    Ignored:    data/folder_05top/
    Ignored:    data/gene_corr_frame.RDS
    Ignored:    data/gene_prob_tran3h.RDS
    Ignored:    data/gene_probabilityk5.RDS
    Ignored:    data/gostresTop2bi_ER.RDS
    Ignored:    data/gostresTop2bi_LR
    Ignored:    data/gostresTop2bi_LR.RDS
    Ignored:    data/gostresTop2bi_TI.RDS
    Ignored:    data/gostrescoNR
    Ignored:    data/gtex/
    Ignored:    data/heartgenes.csv
    Ignored:    data/individualDRCfile.RDS
    Ignored:    data/individual_DRC48.RDS
    Ignored:    data/individual_LDH48.RDS
    Ignored:    data/knowfig4.csv
    Ignored:    data/knowfig5.csv
    Ignored:    data/knowles56.GMT
    Ignored:    data/knowlesGMT.GMT
    Ignored:    data/mymatrix.RDS
    Ignored:    data/nonresponse_cluster24h.csv
    Ignored:    data/norm_LDH.csv
    Ignored:    data/norm_counts.csv
    Ignored:    data/old_sets/
    Ignored:    data/plan2plot.png
    Ignored:    data/raw_counts.csv
    Ignored:    data/response_cluster24h.csv
    Ignored:    data/sigVDA24.txt
    Ignored:    data/sigVDA3.txt
    Ignored:    data/sigVDX24.txt
    Ignored:    data/sigVDX3.txt
    Ignored:    data/sigVEP24.txt
    Ignored:    data/sigVEP3.txt
    Ignored:    data/sigVMT24.txt
    Ignored:    data/sigVMT3.txt
    Ignored:    data/sigVTR24.txt
    Ignored:    data/sigVTR3.txt
    Ignored:    data/siglist.RDS
    Ignored:    data/table3a.omar
    Ignored:    data/toplistall.RDS
    Ignored:    data/tvl24hour.txt
    Ignored:    data/tvl24hourw.txt
    Ignored:    data/venn_code.R

Untracked files:
    Untracked:  .RDataTmp
    Untracked:  .RDataTmp1
    Untracked:  .RDataTmp2
    Untracked:  analysis/Seoane_chrom.Rmd
    Untracked:  cormotif_probability_genelist.csv
    Untracked:  individual-legenddark2.png
    Untracked:  installed_old.rda
    Untracked:  motif_ER.txt
    Untracked:  motif_LR.txt
    Untracked:  motif_NR.txt
    Untracked:  motif_TI.txt
    Untracked:  output/DNRmeQTLs.RDS
    Untracked:  output/DNRmeSNPs.RDS
    Untracked:  output/DNRreQTLs.RDS
    Untracked:  output/DOXmeQTLs.RDS
    Untracked:  output/DOXmeSNPs.RDS
    Untracked:  output/DOXreQTLs.RDS
    Untracked:  output/EPImeQTLs.RDS
    Untracked:  output/EPImeSNPs.RDS
    Untracked:  output/EPIreQTLs.RDS
    Untracked:  output/Fig_summary1.png
    Untracked:  output/GWAS_goi.csv
    Untracked:  output/MTXmeQTLs.RDS
    Untracked:  output/MTXmeSNPs.RDS
    Untracked:  output/MTXreQTLs.RDS
    Untracked:  output/Sup4seoane.csv
    Untracked:  output/egenes.csv
    Untracked:  output/egenes_hgnc.csv
    Untracked:  output/egenes_set.csv
    Untracked:  output/figure_1.Rmd
    Untracked:  output/heart_gtex.csv
    Untracked:  output/knowles4.RDS
    Untracked:  output/knowles5.RDS
    Untracked:  output/not_eqtls_GTEX.csv
    Untracked:  output/output-old/
    Untracked:  output/plan2plot.png
    Untracked:  output/plan48ldh.png
    Untracked:  reneebasecode.R

Unstaged changes:
    Modified:   analysis/DEG-GO_analysis.Rmd
    Modified:   analysis/Figure1.Rmd
    Modified:   analysis/Knowles2019.Rmd
    Modified:   code/eQTLcodes.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/GTEx_genes.Rmd</code>) and HTML
(<code>docs/GTEx_genes.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/d5478771e9d452749ccdbd07bbe537ad93cf2597/analysis/GTEx_genes.Rmd" target="_blank">d547877</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-15
</td>
<td>
update on GTEx
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/7600128faca947e3cc6662ac5fec29f57ab0d4f0/docs/GTEx_genes.html" target="_blank">7600128</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/ff0298978f92043090ea7528d639f966a42c930d/analysis/GTEx_genes.Rmd" target="_blank">ff02989</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-14
</td>
<td>
update for website
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/ae073bed9690f93cc29ba7ac91adc94fc7dca3e5/docs/GTEx_genes.html" target="_blank">ae073be</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/e9f1a70e9abda731deb9fb10bb50645b6be91a1a/analysis/GTEx_genes.Rmd" target="_blank">e9f1a70</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-14
</td>
<td>
update for website
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/7fc7ec79fc2645d132183614919dcf4be3639184/analysis/GTEx_genes.Rmd" target="_blank">7fc7ec7</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-14
</td>
<td>
updating code
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="gtex-egenes-and-analysis" class="section level1">
<h1>GTEx egenes and analysis</h1>
<pre class="r"><code>library(ComplexHeatmap)
library(tidyverse)
library(ggsignif)
library(biomaRt)
library(RColorBrewer)
library(cowplot)
library(scales)
library(sjmisc)
library(kableExtra)
library(broom)
library(ggstats)</code></pre>
<div id="data-import" class="section level2">
<h2>Data import</h2>
<pre class="r"><code>toplistall&lt;- read.csv(&quot;output/toplistall.csv&quot;, row.names = 1)
my_exp_genes &lt;- read.csv(&quot;data/backGL.txt&quot;)
egenes_set &lt;- read.csv(&quot;output/egenes_set.csv&quot;,row.names = 1)
egenes_hgnc &lt;- read.csv(&quot;output/egenes_hgnc.csv&quot;,row.names = 1)
GTEx_genes &lt;- read.csv(&quot;data/GTEx_gene_list.csv&quot;,row.names = 1)
not_eqtls &lt;- read.csv(&quot;output/not_eqtls_GTEX.csv&quot;,row.names = 1)
heart_gtex &lt;- read.csv(&quot;output/heart_gtex.csv&quot;,row.names = 1)
egenes &lt;- read.csv(&quot;output/egenes.csv&quot;,row.names = 1)</code></pre>
<div id="obtaining-the-gtex-data-set" class="section level3">
<h3>Obtaining the GTEx data set</h3>
<p>I downloaded the GTEx_Analysis_v8.metasoft.txt.gz files from the
Consortium at <a href="https://www.gtexportal.org/home/datasets"
class="uri">https://www.gtexportal.org/home/datasets</a> .</p>
<p>I the extracted the Heart_Left_Ventricle.v8.egenes.txt file and
uploaded into R under the data folder.</p>
<pre class="r"><code>heart_gtex &lt;-
  readr::read_delim(&quot;data/Heart_Left_Ventricle.v8.egenes.txt&quot;,
                    delim = &quot;\t&quot;,
                    escape_double = FALSE,
                    trim_ws = TRUE)
egenes &lt;- heart_gtex %&gt;% 
  dplyr::select(gene_id, gene_name, qval) %&gt;% 
  filter(qval&lt;0.05) %&gt;% 
  separate(gene_id, into =c(&#39;ensembl_gene_id&#39;, &#39;gene_version&#39;))

not_eqtl &lt;- heart_gtex %&gt;% 
  dplyr::select(gene_id, gene_name, qval) %&gt;% 
  filter(qval&gt;0.05) %&gt;% 
  separate(gene_id, into =c(&#39;ensembl_gene_id&#39;, &#39;gene_version&#39;))

egenes_set &lt;- getBM(attributes=my_attributes, 
                    filters =&#39;ensembl_gene_id&#39;,
                    values =egenes$ensembl_gene_id,
                    mart = ensembl)
egenes_hgnc &lt;- getBM(attributes=my_attributes,
                     filters =&#39;hgnc_symbol&#39;,
                     values =egenes$gene_name, 
                     mart = ensembl)
not_eqtl_set &lt;- getBM(attributes=my_attributes, 
                    filters =&#39;ensembl_gene_id&#39;,
                    values =not_eqtl$ensembl_gene_id,
                    mart = ensembl)

not_eqtls &lt;- not_eqtl_set %&gt;% 
  distinct(entrezgene_id,.keep_all = TRUE) %&gt;% 
  filter(entrezgene_id %in% my_exp_genes$ENTREZID)
##6711 not_eqtls
GTEx_genes &lt;- egenes_set %&gt;% 
  distinct(entrezgene_id,.keep_all = TRUE)</code></pre>
<p>This file contains several columns gene_id, gene_name, gene_chr,
gene_start, gene_end, strand, num_var, beta_shape1, beta_shape2,
true_df, pval_true_df, variant_id, tss_distance, chr, variant_pos, ref,
alt, num_alt_per_site, rs_id_dbSNP151_GRCh38p7, minor_allele_samples,
minor_allele_count, maf, ref_factor, pval_nominal, slope, slope_se,
pval_perm, pval_beta, qval, pval_nominal_threshold, log2_aFC,
log2_aFC_lower, log2_aFC_upper.</p>
<p>I then chose the the ‘gene_id’,‘gene_name’, and ‘qval’ columns. This
left me with 21353 genes. Next I filtered the tissue specific expressed
genes using a the ‘qval &lt; 0.05’ for a total of 9642. I then took the
gene_name column and used biomart to convert to ‘entrezgene_id’.<br />
Because results vary by which way I look up genes in BioMart, I tested
both egenes using ensemble_gene_id and hgnc_symbol columns. I found 7813
for the ensemble_gene_ set and 7271 for the hgnc_symbol set.<br />
I went with using ensemble_gene_id because I found ~600 more genes
overall than using the hgnc_symbol filter.</p>
<pre class="r"><code>GTEx &lt;- intersect(GTEx_genes$entrezgene_id,my_exp_genes$ENTREZID)
nQTL &lt;- not_eqtls
# &#39;%!in%&#39; &lt;- function(x,y)!(&#39;%in%&#39;(x,y))

nQTLmy &lt;- my_exp_genes %&gt;%
   dplyr:: filter(!ENTREZID %in%GTEx)

# testset &lt;- toplistall %&gt;% 
#   filter(adj.P.Val&lt;0.05) %&gt;% 
#  select(ENTREZID) %&gt;% distinct(ENTREZID) %&gt;% 
#   dplyr:: filter(ENTREZID %in%GTEx_genes$entrezgene_id)</code></pre>
<p>To find out how many genes from the gtex egenes were expressed in my
data, I intersected my expressed genes list of 14084 genes with the
GTEx_genes and found 6261 genes were shared between them. I called this
set ‘GTEx’. Using the other eGenes from GTEx, I made another set
intersected with my expressed gene set called ‘nQTL’. This nQTL set
contains 7823 genes. Next, I then took my DEG top list and filtered out
genes with an adj.P.value &lt; 0.05.</p>
<pre class="r"><code>drug_palspc &lt;- c(&quot;#8B006D&quot;,&quot;#DF707E&quot;,&quot;#8B006D&quot;,&quot;#DF707E&quot;)</code></pre>
<p>The next step is to wrangle the data so that I can test the
difference between the proportions of significantly DE genes found in
the GTEx and nQTLs.</p>
<pre class="r"><code>nQTLsum &lt;- toplistall %&gt;%
  mutate(id = as.factor(id)) %&gt;%
  mutate(time=factor(time, levels=c(&quot;3_hours&quot;,&quot;24_hours&quot;))) %&gt;%
  dplyr::filter(adj.P.Val &lt;0.05) %&gt;%
  mutate(nQTL=if_else(ENTREZID %in% nQTLmy$ENTREZID,&#39;nQTL_y&#39;,&#39;nQTL_no&#39;)) %&gt;% 
  group_by(id,time,nQTL) %&gt;% 
  tally() %&gt;% 
  separate(nQTL, into=c(&#39;set&#39;, &#39;group&#39;)) %&gt;% 
  mutate(total=length(nQTLmy$ENTREZID) - n) %&gt;% 
  dplyr::filter(group==&quot;y&quot;)

GTExsum &lt;- toplistall %&gt;%
  mutate(id = as.factor(id)) %&gt;%
  mutate(time=factor(time, levels=c(&quot;3_hours&quot;,&quot;24_hours&quot;))) %&gt;%
  dplyr::filter(adj.P.Val &lt;0.05) %&gt;%
   mutate(GTEx=if_else(ENTREZID %in%GTEx,&quot;GTEx_y&quot;,&quot;GTEx_no&quot;)) %&gt;% 
  group_by(id,time,GTEx) %&gt;% 
  tally() %&gt;% 
  separate(GTEx, into=c(&#39;set&#39;, &#39;group&#39;)) %&gt;% 
  mutate(total=length(GTEx) - n) %&gt;% 
  dplyr::filter(group==&quot;y&quot;)

GTEXcr8z &lt;- GTExsum %&gt;% 
  rbind(., nQTLsum) %&gt;% 
  dplyr::select(id,time,set, n,total) %&gt;% 
  pivot_longer(cols=n:total, names_to=&quot;group&quot;,values_to=&quot;total&quot;)  
  

GTEXcr8z %&gt;% 
  ggplot(., aes(x=set,y=total, fill=group))+
   geom_col(position=&#39;fill&#39;)+
    facet_wrap(time~id,nrow=2,ncol=4)+
    theme_classic()+
   # scale_color_manual(values=drug_palNoVeh[c(1,2)])+
    scale_fill_manual(values=drug_palspc)</code></pre>
<p><img src="figure/GTEx_genes.Rmd/still%20going-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>GTExsum %&gt;% 
  rbind(., nQTLsum) %&gt;% 
  dplyr::select(id,time,set, n, total) %&gt;% 
   mutate(time = case_match(time, &#39;3_hours&#39;~&#39;3 hrs&#39;,
                           &#39;24_hours&#39;~&#39;24 hrs&#39;,.default = time)) %&gt;% 
  mutate(id =case_match( id, 
                         &#39;Daunorubicin&#39;~&#39;DNR&#39;,   
                         &#39;Doxorubicin&#39;~&#39;DOX&#39; ,
                         &#39;Epirubicin&#39;~&#39;EPI&#39; , 
                         &#39;Mitoxantrone&#39; ~ &#39;MTX&#39;, .default = id)) #%&gt;% </code></pre>
<pre><code># A tibble: 16 × 5
# Groups:   id, time [8]
   id    time   set       n total
   &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt; &lt;int&gt; &lt;int&gt;
 1 DNR   3 hrs  GTEx    190  6071
 2 DNR   24 hrs GTEx   3032  3229
 3 DOX   3 hrs  GTEx      2  6259
 4 DOX   24 hrs GTEx   2877  3384
 5 EPI   3 hrs  GTEx     77  6184
 6 EPI   24 hrs GTEx   2686  3575
 7 MTX   3 hrs  GTEx     13  6248
 8 MTX   24 hrs GTEx    534  5727
 9 DNR   3 hrs  nQTL    365  7458
10 DNR   24 hrs nQTL   3832  3991
11 DOX   3 hrs  nQTL     14  7809
12 DOX   24 hrs nQTL   3639  4184
13 EPI   3 hrs  nQTL    143  7680
14 EPI   24 hrs nQTL   3516  4307
15 MTX   3 hrs  nQTL     45  7778
16 MTX   24 hrs nQTL    793  7030</code></pre>
<pre class="r"><code>testDNR3chix &lt;- matrix(c(365,190,7458, 6071), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c(&quot;nogtex&quot;,&quot;gtex&quot;),c( &quot;y&quot;, &quot;n&quot;))) 
    DNR_3chix &lt;- chisq.test(testDNR3chix,correct=TRUE)$p.value
testDNR24chix &lt;- matrix(c(3832,3032,3991,3229), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c(&quot;nogtex&quot;,&quot;gtex&quot;),c( &quot;y&quot;, &quot;n&quot;)))  
    DNR_24chix &lt;- chisq.test(testDNR24chix,correct=TRUE)$p.value

testDOX3chix &lt;- matrix(c(14,2,7809,6259), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c(&quot;nogtex&quot;,&quot;gtex&quot;),c( &quot;y&quot;, &quot;n&quot;)))  
    DOX_3chix &lt;- chisq.test(testDOX3chix,correct=TRUE)$p.value
testDOX24chix &lt;- matrix(c(3639,2877,4184,3384), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c(&quot;nogtex&quot;,&quot;gtex&quot;),c( &quot;y&quot;, &quot;n&quot;))) 
    DOX_24chix &lt;- chisq.test(testDOX24chix,correct=TRUE)$p.value

testEPI3chix &lt;- matrix(c(143,77,7680,6184), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c(&quot;nogtex&quot;,&quot;gtex&quot;),c( &quot;y&quot;, &quot;n&quot;))) 
    EPI_3chix &lt;- chisq.test(testEPI3chix)$p.value
testEPI24chix &lt;- matrix(c(3516,2686,4307,3575), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c(&quot;nogtex&quot;,&quot;gtex&quot;),c( &quot;y&quot;, &quot;n&quot;))) 
    EPI_24chix &lt;- chisq.test(testEPI24chix)$p.value
  
testMTX3chix &lt;- matrix(c(45,13,7778,6248), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c(&quot;nogtex&quot;,&quot;nogtex&quot;),c( &quot;y&quot;, &quot;n&quot;))) 
    MTX_3chix &lt;- chisq.test(testMTX3chix)$p.value
testMTX24chix &lt;- matrix(c(793,534,7030,5727), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c(&quot;nogtex&quot;,&quot;nogtex&quot;),c( &quot;y&quot;, &quot;n&quot;))) 
    MTX_24chix &lt;- chisq.test(testMTX24chix,correct=TRUE)$p.value
GTEX_table_chix &lt;- data.frame(treatment=c(&#39;DNR_3&#39;,&#39;DNR_24&#39;,&#39;DOX_3&#39;,&#39;DOX_24&#39;,&#39;EPI_3&#39;,&#39;EPI_24&#39;,&#39;MTX_3&#39;,&#39;MTX_24&#39;), chi_p.value=c(DNR_3chix,DNR_24chix,DOX_3chix,DOX_24chix,EPI_3chix,EPI_24chix,MTX_3chix,MTX_24chix))

GTEX_table_chix %&gt;% 
  separate(treatment, into= c(&#39;Drug&#39;,&#39;time&#39;)) %&gt;% 
  pivot_wider(id_cols = Drug, names_from = time, values_from = chi_p.value) %&gt;% 
  kable(., caption= &quot;Chi Square p. values from chi-square test between proportions of sig-DE meQTLs and reQTLS by time and treatment&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;, full_width = TRUE) %&gt;%  
  kable_styling(full_width = FALSE, font_size = 16) %&gt;% 
  scroll_box( height = &quot;500px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; ">
<table class=" lightable-paper lightable-striped table" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; margin-left: auto; margin-right: auto; font-size: 16px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Chi Square p. values from chi-square test between proportions of sig-DE
meQTLs and reQTLS by time and treatment
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Drug
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
3
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
24
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
DNR
</td>
<td style="text-align:right;">
0.0000010
</td>
<td style="text-align:right;">
0.5220483
</td>
</tr>
<tr>
<td style="text-align:left;">
DOX
</td>
<td style="text-align:right;">
0.0202323
</td>
<td style="text-align:right;">
0.5144642
</td>
</tr>
<tr>
<td style="text-align:left;">
EPI
</td>
<td style="text-align:right;">
0.0055023
</td>
<td style="text-align:right;">
0.0159127
</td>
</tr>
<tr>
<td style="text-align:left;">
MTX
</td>
<td style="text-align:right;">
0.0011436
</td>
<td style="text-align:right;">
0.0012975
</td>
</tr>
</tbody>
</table>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggstats_0.3.0         broom_1.0.5           kableExtra_1.3.4     
 [4] sjmisc_2.8.9          scales_1.2.1          cowplot_1.1.1        
 [7] RColorBrewer_1.1-3    biomaRt_2.52.0        ggsignif_0.6.4       
[10] lubridate_1.9.2       forcats_1.0.0         stringr_1.5.0        
[13] dplyr_1.1.2           purrr_1.0.1           readr_2.1.4          
[16] tidyr_1.3.0           tibble_3.2.1          ggplot2_3.4.2        
[19] tidyverse_2.0.0       ComplexHeatmap_2.12.1 workflowr_1.7.0      

loaded via a namespace (and not attached):
  [1] colorspace_2.1-0       rjson_0.2.21           sjlabelled_1.2.0      
  [4] rprojroot_2.0.3        circlize_0.4.15        XVector_0.36.0        
  [7] GlobalOptions_0.1.2    fs_1.6.2               clue_0.3-64           
 [10] rstudioapi_0.14        farver_2.1.1           bit64_4.0.5           
 [13] AnnotationDbi_1.58.0   fansi_1.0.4            xml2_1.3.4            
 [16] codetools_0.2-19       doParallel_1.0.17      cachem_1.0.8          
 [19] knitr_1.43             jsonlite_1.8.5         cluster_2.1.4         
 [22] dbplyr_2.3.2           png_0.1-8              compiler_4.2.2        
 [25] httr_1.4.6             backports_1.4.1        fastmap_1.1.1         
 [28] cli_3.6.1              later_1.3.1            htmltools_0.5.5       
 [31] prettyunits_1.1.1      tools_4.2.2            gtable_0.3.3          
 [34] glue_1.6.2             GenomeInfoDbData_1.2.8 rappdirs_0.3.3        
 [37] Rcpp_1.0.10            Biobase_2.56.0         jquerylib_0.1.4       
 [40] vctrs_0.6.2            Biostrings_2.64.1      svglite_2.1.1         
 [43] iterators_1.0.14       insight_0.19.2         xfun_0.39             
 [46] ps_1.7.5               rvest_1.0.3            timechange_0.2.0      
 [49] lifecycle_1.0.3        XML_3.99-0.14          getPass_0.2-2         
 [52] zlibbioc_1.42.0        hms_1.1.3              promises_1.2.0.1      
 [55] parallel_4.2.2         yaml_2.3.7             curl_5.0.1            
 [58] memoise_2.0.1          sass_0.4.6             stringi_1.7.12        
 [61] RSQLite_2.3.1          highr_0.10             S4Vectors_0.34.0      
 [64] foreach_1.5.2          BiocGenerics_0.42.0    filelock_1.0.2        
 [67] shape_1.4.6            GenomeInfoDb_1.32.4    rlang_1.1.1           
 [70] pkgconfig_2.0.3        systemfonts_1.0.4      matrixStats_1.0.0     
 [73] bitops_1.0-7           evaluate_0.21          labeling_0.4.2        
 [76] bit_4.0.5              processx_3.8.1         tidyselect_1.2.0      
 [79] magrittr_2.0.3         R6_2.5.1               IRanges_2.30.1        
 [82] generics_0.1.3         DBI_1.1.3              pillar_1.9.0          
 [85] whisker_0.4.1          withr_2.5.0            KEGGREST_1.36.3       
 [88] RCurl_1.98-1.12        crayon_1.5.2           utf8_1.2.3            
 [91] BiocFileCache_2.4.0    tzdb_0.4.0             rmarkdown_2.22        
 [94] GetoptLong_1.0.5       progress_1.2.2         blob_1.2.4            
 [97] callr_3.7.3            git2r_0.32.0           digest_0.6.31         
[100] webshot_0.5.4          httpuv_1.6.11          stats4_4.2.2          
[103] munsell_0.5.0          viridisLite_0.4.2      bslib_0.5.0           </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
