<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="ERM" />

<meta name="date" content="2023-06-20" />

<title>Knowles 2019 data</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cardiotoxicity</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/reneeisnowhere/Cardiotoxicity">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Knowles 2019 data</h1>
<h4 class="author">ERM</h4>
<h4 class="date">2023-06-20</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-06-20
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Cardiotoxicity/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230109code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230109)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230109code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230109)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereCardiotoxicitytree7376396d39594107d233e522158581b3d1b9ce05targetblank7376396a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/tree/7376396d39594107d233e522158581b3d1b9ce05" target="_blank">7376396</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereCardiotoxicitytree7376396d39594107d233e522158581b3d1b9ce05targetblank7376396a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/tree/7376396d39594107d233e522158581b3d1b9ce05" target="_blank">7376396</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/41588_2018_171_MOESM3_ESMeQTL_ST2_for paper.csv
    Ignored:    data/Arr_GWAS.txt
    Ignored:    data/Arr_geneset.RDS
    Ignored:    data/BC_cell_lines.csv
    Ignored:    data/CADGWASgene_table.csv
    Ignored:    data/CAD_geneset.RDS
    Ignored:    data/Clamp_Summary.csv
    Ignored:    data/Cormotif_24_k1-5_raw.RDS
    Ignored:    data/DAgostres24.RDS
    Ignored:    data/DAtable1.csv
    Ignored:    data/DDEMresp_list.csv
    Ignored:    data/DDE_reQTL.txt
    Ignored:    data/DDEresp_list.csv
    Ignored:    data/DEG-GO/
    Ignored:    data/DEG_cormotif.RDS
    Ignored:    data/DF_Plate_Peak.csv
    Ignored:    data/DRC48hoursdata.csv
    Ignored:    data/Da24counts.txt
    Ignored:    data/Dx24counts.txt
    Ignored:    data/Dx_reQTL_specific.txt
    Ignored:    data/Ep24counts.txt
    Ignored:    data/GOIsig.csv
    Ignored:    data/GOplots.R
    Ignored:    data/GTEX_setsimple.csv
    Ignored:    data/GTEx_gene_list.csv
    Ignored:    data/HFGWASgene_table.csv
    Ignored:    data/HF_geneset.RDS
    Ignored:    data/Heart_Left_Ventricle.v8.egenes.txt
    Ignored:    data/Hf_GWAS.txt
    Ignored:    data/K_cluster
    Ignored:    data/K_cluster_kisthree.csv
    Ignored:    data/K_cluster_kistwo.csv
    Ignored:    data/LDH48hoursdata.csv
    Ignored:    data/Mt24counts.txt
    Ignored:    data/RINsamplelist.txt
    Ignored:    data/Seonane2019supp1.txt
    Ignored:    data/TOP2Bi-24hoursGO_analysis.csv
    Ignored:    data/TR24counts.txt
    Ignored:    data/Top2biresp_cluster24h.csv
    Ignored:    data/Viabilitylistfull.csv
    Ignored:    data/allexpressedgenes.txt
    Ignored:    data/allgenes.txt
    Ignored:    data/allmatrix.RDS
    Ignored:    data/avgLD50.RDS
    Ignored:    data/backGL.txt
    Ignored:    data/cormotif_3hk1-8.RDS
    Ignored:    data/cormotif_initalK5.RDS
    Ignored:    data/cormotif_initialK5.RDS
    Ignored:    data/cormotif_initialall.RDS
    Ignored:    data/counts24hours.RDS
    Ignored:    data/cpmcount.RDS
    Ignored:    data/cpmnorm_counts.csv
    Ignored:    data/crispr_genes.csv
    Ignored:    data/cvd_GWAS.txt
    Ignored:    data/dat_cpm.RDS
    Ignored:    data/data_outline.txt
    Ignored:    data/efit2.RDS
    Ignored:    data/efit2results.RDS
    Ignored:    data/ensembl_backup.RDS
    Ignored:    data/ensgtotal.txt
    Ignored:    data/filenameonly.txt
    Ignored:    data/filtered_cpm_counts.csv
    Ignored:    data/filtered_raw_counts.csv
    Ignored:    data/filtermatrix_x.RDS
    Ignored:    data/folder_05top/
    Ignored:    data/geneDoxonlyQTL.csv
    Ignored:    data/gene_corr_frame.RDS
    Ignored:    data/gene_prob_tran3h.RDS
    Ignored:    data/gene_probabilityk5.RDS
    Ignored:    data/gostresTop2bi_ER.RDS
    Ignored:    data/gostresTop2bi_LR
    Ignored:    data/gostresTop2bi_LR.RDS
    Ignored:    data/gostresTop2bi_TI.RDS
    Ignored:    data/gostrescoNR
    Ignored:    data/gtex/
    Ignored:    data/heartgenes.csv
    Ignored:    data/individualDRCfile.RDS
    Ignored:    data/individual_DRC48.RDS
    Ignored:    data/individual_LDH48.RDS
    Ignored:    data/knowfig4.csv
    Ignored:    data/knowfig5.csv
    Ignored:    data/mymatrix.RDS
    Ignored:    data/nonresponse_cluster24h.csv
    Ignored:    data/norm_LDH.csv
    Ignored:    data/norm_counts.csv
    Ignored:    data/old_sets/
    Ignored:    data/plan2plot.png
    Ignored:    data/raw_counts.csv
    Ignored:    data/response_cluster24h.csv
    Ignored:    data/sigVDA24.txt
    Ignored:    data/sigVDA3.txt
    Ignored:    data/sigVDX24.txt
    Ignored:    data/sigVDX3.txt
    Ignored:    data/sigVEP24.txt
    Ignored:    data/sigVEP3.txt
    Ignored:    data/sigVMT24.txt
    Ignored:    data/sigVMT3.txt
    Ignored:    data/sigVTR24.txt
    Ignored:    data/sigVTR3.txt
    Ignored:    data/siglist.RDS
    Ignored:    data/slope_table.csv
    Ignored:    data/table3a.omar
    Ignored:    data/toplistall.RDS
    Ignored:    data/tvl24hour.txt
    Ignored:    data/tvl24hourw.txt
    Ignored:    data/venn_code.R

Untracked files:
    Untracked:  .RDataTmp
    Untracked:  .RDataTmp1
    Untracked:  .RDataTmp2
    Untracked:  OmicNavigator_learn.R
    Untracked:  code/DRC_plotfigure1.png
    Untracked:  code/cpm_boxplot.R
    Untracked:  code/fig1plot.png
    Untracked:  code/figurelegeddrc.png
    Untracked:  cormotif_probability_genelist.csv
    Untracked:  individual-legenddark2.png
    Untracked:  installed_old.rda
    Untracked:  motif_ER.txt
    Untracked:  motif_LR.txt
    Untracked:  motif_NR.txt
    Untracked:  motif_TI.txt
    Untracked:  output/Doxonly_deg.csv
    Untracked:  output/daplot.RDS
    Untracked:  output/dxplot.RDS
    Untracked:  output/epplot.RDS
    Untracked:  output/mtplot.RDS
    Untracked:  output/output-old/
    Untracked:  output/trplot.RDS
    Untracked:  output/veplot.RDS
    Untracked:  reneebasecode.R

Unstaged changes:
    Modified:   analysis/GTEx_genes.Rmd
    Modified:   analysis/LDH_analysis.Rmd
    Modified:   analysis/other_analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Knowles2019.Rmd</code>) and HTML
(<code>docs/Knowles2019.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/4c762a84645779f3b4b08ebb1505920b7222c5ff/analysis/Knowles2019.Rmd" target="_blank">4c762a8</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-20
</td>
<td>
adding Dox specific investigation
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/e94e8f2bc5fbab269f4708d3e5c34a77069b28dd/analysis/Knowles2019.Rmd" target="_blank">e94e8f2</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-19
</td>
<td>
new code from Monday
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/124e03653a4908f5f6037ef9e329ebd8514b308e/docs/Knowles2019.html" target="_blank">124e036</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/fd4fe274792cf77c4e417c605a3a414e0f1c89cc/analysis/Knowles2019.Rmd" target="_blank">fd4fe27</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-16
</td>
<td>
fix spelling
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/d6ad3fe1105e1853df5c7155ded22428078af5f6/docs/Knowles2019.html" target="_blank">d6ad3fe</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/4191e24cb50b45c18838e8af6846722b35eaffad/analysis/Knowles2019.Rmd" target="_blank">4191e24</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-16
</td>
<td>
adding lcpm plots for dox spec
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/bd0e45f05268110fedb5d45f08fdd77f653e6a61/docs/Knowles2019.html" target="_blank">bd0e45f</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/af934216165419b55b29d2303e558217fd2cef63/analysis/Knowles2019.Rmd" target="_blank">af93421</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-15
</td>
<td>
adding in seperate files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/1a1a0341faa5e86d69cf192a3d23d312c118f74c/analysis/Knowles2019.Rmd" target="_blank">1a1a034</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-15
</td>
<td>
updates galore
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/f8f511af8663414efe663a35368a3786c78607b4/analysis/Knowles2019.Rmd" target="_blank">f8f511a</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-15
</td>
<td>
updates and simplifications of code
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/7fc7ec79fc2645d132183614919dcf4be3639184/analysis/Knowles2019.Rmd" target="_blank">7fc7ec7</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-14
</td>
<td>
updating code
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="package-loading" class="section level6">
<h6>package loading</h6>
<p>##function load</p>
</div>
<div id="knowles-data" class="section level2">
<h2>Knowles data</h2>
<p>The code below is how I wrangled the knowles supplemental lists into
entrezgene_ids to overlap with my expressed gene lists</p>
</div>
<div id="enrichment-of-pairwise" class="section level2">
<h2>Enrichment of Pairwise</h2>
<pre class="r"><code>backGL &lt;- read.csv(&quot;data/backGL.txt&quot;,     row.names =1)
drug_palNoVeh &lt;- c(&quot;#8B006D&quot;,&quot;#DF707E&quot;,&quot;#F1B72B&quot;, &quot;#3386DD&quot;,&quot;#707031&quot;)
drug_palc &lt;- c(&quot;#8B006D&quot;,&quot;#DF707E&quot;,&quot;#F1B72B&quot;, &quot;#3386DD&quot;,&quot;#707031&quot;,&quot;#41B333&quot;)
time &lt;- rep((rep(c(&quot;3h&quot;, &quot;24h&quot;), c(6,6))), 6)
time &lt;- ordered(time, levels =c(&quot;3h&quot;, &quot;24h&quot;))
drug &lt;- rep(c(&quot;Daunorubicin&quot;,&quot;Doxorubicin&quot;,&quot;Epirubicin&quot;,&quot;Mitoxantrone&quot;,&quot;Trastuzumab&quot;, &quot;Vehicle&quot;),12)
mat_drug &lt;- c(&quot;Daunorubicin&quot;,&quot;Doxorubicin&quot;,&quot;Epirubicin&quot;,&quot;Mitoxantrone&quot;)
indv &lt;- as.factor(rep(c(1,2,3,4,5,6), c(12,12,12,12,12,12)))
labeltop &lt;- (interaction(substring(drug, 0, 2), indv, time))
level_order2 &lt;- c(&#39;75&#39;,&#39;87&#39;,&#39;77&#39;,&#39;79&#39;,&#39;78&#39;,&#39;71&#39;)


toplistall &lt;- readRDS(&quot;data/toplistall.RDS&quot;)
knowles4 &lt;-readRDS(&quot;output/knowles4.RDS&quot;)
knowles5 &lt;-readRDS(&quot;output/knowles5.RDS&quot;)
DOXmeSNPs &lt;- readRDS(&quot;output/DOXmeSNPs.RDS&quot;)
DOXreQTLs &lt;- readRDS(&quot;output/DOXreQTLs.RDS&quot;)
DNRmeSNPs &lt;- readRDS(&quot;output/DNRmeSNPs.RDS&quot;)
DNRreQTLs &lt;- readRDS(&quot;output/DNRreQTLs.RDS&quot;)
EPImeSNPs &lt;- readRDS(&quot;output/EPImeSNPs.RDS&quot;)
EPIreQTLs &lt;- readRDS(&quot;output/EPIreQTLs.RDS&quot;)
MTXmeSNPs &lt;- readRDS(&quot;output/MTXmeSNPs.RDS&quot;)
MTXreQTLs &lt;- readRDS(&quot;output/MTXreQTLs.RDS&quot;)

toplist24hr &lt;-   toplistall %&gt;%
  mutate(id = as.factor(id)) %&gt;%
  mutate(time=factor(time, levels=c(&quot;3_hours&quot;,&quot;24_hours&quot;))) %&gt;%
  filter(time==&quot;24_hours&quot;)

toplist3hr &lt;-   toplistall %&gt;%
  mutate(id = as.factor(id)) %&gt;%
  mutate(time=factor(time, levels=c(&quot;3_hours&quot;,&quot;24_hours&quot;))) %&gt;%
  filter(time==&quot;3_hours&quot;)</code></pre>
<p>Determining the genetic basis of anthracycline-cardiotoxicity by
molecular response QTL mapping in induced cardiomyocytes<br />
David A Knowles, Courtney K Burrows†, John D Blischak, Kristen M
Patterson, Daniel J Serie, Nadine Norton, Carole Ober, Jonathan K
Pritchard, Yoav Gilad</p>
<p>Knowles <span class="math inline">\(~~et~ al.~\)</span> eLife
2018;7:e33480. DOI: <a href="https://doi.org/10.7554/eLife.33480"
class="uri">https://doi.org/10.7554/eLife.33480</a></p>
<p>My first question was about transcriptional response at the 24 hour
mark with my treatments. 3 hour RNA-seq had low numbers of DEGs,so my
initial focus is at 24 hours. This time also happens to be when the
Knowles paper collected their RNA-seq data.</p>
<p>Supplementary 4 contains a list of 518 SNPs within 1 Mb of TSS, which
had a detectable marginal effect on expression (5% FDR). When converted
from ensembl gene id to entrez gene id, my list of genes for this
supplement = 521. I will call these meSNPS for marginal effect QTLs.</p>
<p>Supplementary 5 contains a list of 376 response eQTLs (reQTLs). These
are variants that were associated with modulation of transcriptomic
response to doxorubicin treatment. After database name conversion, I
have 377 genes.</p>
<pre class="r"><code>meSNPdf &lt;- toplistall %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  mutate(time=factor(time, levels=c(&quot;3_hours&quot;,&quot;24_hours&quot;))) %&gt;%
  mutate(sigcount = if_else(adj.P.Val &lt; 0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(meSNP = if_else( ENTREZID %in% knowles4$entrezgene_id, &quot;y&quot; , &quot;no&quot;)) %&gt;% 
  dplyr::select(ENTREZID,id,time,meSNP,sigcount) %&gt;%
  group_by(id,time,meSNP,sigcount) %&gt;% 
  tally() %&gt;% 
  pivot_wider(id_cols = c(id,time,sigcount), names_from = meSNP,names_glue = &quot;meSNP_{meSNP}&quot;, values_from = n)

meSNPdf %&gt;% 
  kable(., caption= &quot;Number of genes from minimally expressed SNPs found in DEGs (sig and nonsig) broken down by drug,time, and significance&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;, full_width = FALSE) %&gt;% 
  kable_styling(full_width = FALSE, position = &quot;left&quot;,bootstrap_options = c(&quot;striped&quot;),font_size = 18) %&gt;% 
  scroll_box(width = &quot;80%&quot;, height = &quot;400px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:80%; ">
<table class=" lightable-paper lightable-striped table table-striped" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 18px; width: auto !important; ">
<caption style="font-size: initial !important;">
Number of genes from minimally expressed SNPs found in DEGs (sig and
nonsig) broken down by drug,time, and significance
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
id
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
time
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
sigcount
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
meSNP_no
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
meSNP_y
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
13032
</td>
<td style="text-align:right;">
497
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:right;">
549
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
6916
</td>
<td style="text-align:right;">
304
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:right;">
6665
</td>
<td style="text-align:right;">
199
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
13565
</td>
<td style="text-align:right;">
503
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
7249
</td>
<td style="text-align:right;">
319
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:right;">
6332
</td>
<td style="text-align:right;">
184
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
13365
</td>
<td style="text-align:right;">
499
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:right;">
216
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
7551
</td>
<td style="text-align:right;">
331
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:right;">
6030
</td>
<td style="text-align:right;">
172
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
13524
</td>
<td style="text-align:right;">
502
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:right;">
57
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
12284
</td>
<td style="text-align:right;">
473
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:right;">
1297
</td>
<td style="text-align:right;">
30
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
13581
</td>
<td style="text-align:right;">
503
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
13581
</td>
<td style="text-align:right;">
503
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>reQTLdf &lt;- toplistall %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  mutate(time=factor(time, levels=c(&quot;3_hours&quot;,&quot;24_hours&quot;))) %&gt;%
  mutate(sigcount = if_else(adj.P.Val &lt; 0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(reQTL = if_else( ENTREZID %in% knowles5$entrezgene_id, &quot;y&quot; , &quot;no&quot;)) %&gt;% 
  dplyr::select(ENTREZID,id,time,reQTL,sigcount) %&gt;%
  group_by(id,time,reQTL,sigcount) %&gt;% 
  tally() %&gt;% 
  pivot_wider(id_cols = c(id,time,sigcount), names_from = reQTL,names_glue = &quot;reQTL_{reQTL}&quot;, values_from = n)

reQTLdf %&gt;% 
  kable(., caption= &quot;Number of genes from response eQTLS found in DEGs (sig and nonsig) broken down by drug,time, and significance&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;, full_width = FALSE) %&gt;% 
  kable_styling(full_width = FALSE, position = &quot;left&quot;,bootstrap_options = c(&quot;striped&quot;),font_size = 18) %&gt;% 
  scroll_box(width = &quot;80%&quot;, height = &quot;400px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:80%; ">
<table class=" lightable-paper lightable-striped table table-striped" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 18px; width: auto !important; ">
<caption style="font-size: initial !important;">
Number of genes from response eQTLS found in DEGs (sig and nonsig)
broken down by drug,time, and significance
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
id
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
time
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
sigcount
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
reQTL_no
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
reQTL_y
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
13168
</td>
<td style="text-align:right;">
361
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:right;">
542
</td>
<td style="text-align:right;">
13
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
7033
</td>
<td style="text-align:right;">
187
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:right;">
6677
</td>
<td style="text-align:right;">
187
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
13694
</td>
<td style="text-align:right;">
374
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
7374
</td>
<td style="text-align:right;">
194
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:right;">
6336
</td>
<td style="text-align:right;">
180
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
13494
</td>
<td style="text-align:right;">
370
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:right;">
216
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
7684
</td>
<td style="text-align:right;">
198
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:right;">
6026
</td>
<td style="text-align:right;">
176
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
13652
</td>
<td style="text-align:right;">
374
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:right;">
58
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
12423
</td>
<td style="text-align:right;">
334
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:right;">
1287
</td>
<td style="text-align:right;">
40
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
13710
</td>
<td style="text-align:right;">
374
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:right;">
13710
</td>
<td style="text-align:right;">
374
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>dataframK_45 &lt;- toplistall %&gt;%
  mutate(id = as.factor(id)) %&gt;%
  mutate(time=factor(time, levels=c(&quot;3_hours&quot;,&quot;24_hours&quot;))) %&gt;%
  mutate(K4 = if_else(ENTREZID %in% knowles4$entrezgene_id,&#39;y&#39;,&#39;no&#39;))%&gt;%
  mutate(K5 = if_else(ENTREZID %in% knowles5$entrezgene_id,&#39;y&#39;,&#39;no&#39;))%&gt;%
  filter(adj.P.Val&lt;0.05) %&gt;%
   group_by(time, id) %&gt;%
 dplyr::summarize(n=n(), meSNP=sum(if_else(K4==&quot;y&quot;,1,0)), reQTL=sum(if_else(K5==&quot;y&quot;,1,0))) %&gt;% 
  as.data.frame()


dataframK_45 %&gt;% 
kable(., caption= &quot;Summary of meSNPs and reQTLs found in my DEGs&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;, full_width = FALSE) %&gt;% 
  kable_styling(font_size = 18) %&gt;% 
  scroll_box(width = &quot;80%&quot;, height = &quot;400px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:80%; ">
<table class=" lightable-paper lightable-striped table" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 18px; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Summary of meSNPs and reQTLs found in my DEGs
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
time
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
id
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
n
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
meSNP
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
reQTL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:right;">
555
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
13
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:right;">
220
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
4
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:right;">
58
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:right;">
6864
</td>
<td style="text-align:right;">
199
</td>
<td style="text-align:right;">
187
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:right;">
6516
</td>
<td style="text-align:right;">
184
</td>
<td style="text-align:right;">
180
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:right;">
6202
</td>
<td style="text-align:right;">
172
</td>
<td style="text-align:right;">
176
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:right;">
1327
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
40
</td>
</tr>
</tbody>
</table>
</div>
<div id="visualization-of-proportions" class="section level3">
<h3>Visualization of proportions</h3>
<pre class="r"><code>toplistall %&gt;%
  mutate(id = as.factor(id)) %&gt;%
  mutate(time=factor(time, levels=c(&quot;3_hours&quot;,&quot;24_hours&quot;))) %&gt;%
  mutate(sigcount = if_else(adj.P.Val &lt; 0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,&quot;a&quot;,&quot;b&quot;)) %&gt;% 
  filter(time ==&quot;24_hours&quot;) %&gt;% 
  group_by(id,sigcount,meSNP) %&gt;% 
  summarize(K4count=n())%&gt;% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(meSNP), values_from=K4count) %&gt;% 
   mutate(SNPprop=(y/(y+no)*100)) %&gt;% 
       ggplot(., aes(x=id, y=SNPprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf(&quot;%.2f&quot;,SNPprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0(&quot;number&quot;~a,&quot;out of&quot;,~b))))+
       facet_wrap(~sigcount)+
       ggtitle(&quot;non-significant and significant enrichment proporitions of Knowles meSNPs &quot;)</code></pre>
<p><img src="figure/Knowles2019.Rmd/K4K5%20proportions-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>toplistall %&gt;%
  mutate(id = as.factor(id)) %&gt;%
  mutate(time=factor(time, levels=c(&quot;3_hours&quot;,&quot;24_hours&quot;))) %&gt;%
  mutate(sigcount = if_else(adj.P.Val &lt; 0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,&quot;a&quot;,&quot;b&quot;)) %&gt;% 
  filter(time ==&quot;24_hours&quot;) %&gt;% 
  group_by(id,sigcount,reQTL) %&gt;% 
  summarize(K5count=n())%&gt;% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(reQTL), values_from=K5count) %&gt;% 
   mutate(QTLprop=(a/(a+b)*100)) %&gt;% 
       ggplot(., aes(x=id, y=QTLprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf(&quot;%.2f&quot;,QTLprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0(&quot;number&quot;~a,&quot;out of&quot;,~b))))+
       facet_wrap(~sigcount)+
       ggtitle(&quot;non-significant and significant enrichment proporitions of Knowles reQTLs &quot;)</code></pre>
<p><img src="figure/Knowles2019.Rmd/K4K5%20proportions-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>chi_squarek4 &lt;- toplistall %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  dplyr::filter(id!=&quot;Trastuzumab&quot;) %&gt;% 
  mutate(sigcount = if_else(adj.P.Val &lt;0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  group_by(id,time) %&gt;% 
  summarise(pvalue= chisq.test(meSNP, sigcount)$p.value) 

chi_squarek5 &lt;-   toplistall %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  dplyr::filter(id!=&quot;Trastuzumab&quot;) %&gt;% 
  mutate(sigcount = if_else(adj.P.Val &lt;0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,&quot;a&quot;,&quot;b&quot;)) %&gt;% 
  group_by(id,time) %&gt;% 
  summarise(pvalue= chisq.test(reQTL, sigcount)$p.value) 

chi_squarek4 %&gt;% 
  kable(., caption= &quot;meSNPs (mininmally expressed QTLS) chi-squared pvalues&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;, full_width = FALSE) %&gt;%  
  kable_styling(full_width = FALSE, position = &quot;right&quot;,bootstrap_options = c(&quot;striped&quot;),font_size = 18) %&gt;%   scroll_box(width = &quot;50%&quot;, height = &quot;400px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:50%; ">
<table class=" lightable-paper lightable-striped table table-striped" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 18px; width: auto !important; margin-right: 0; margin-left: auto">
<caption style="font-size: initial !important;">
meSNPs (mininmally expressed QTLS) chi-squared pvalues
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
id
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
time
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
pvalue
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.0000338
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
0.0018777
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.0000113
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
0.9232984
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.0000074
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
0.2189641
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.0086490
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
0.6853643
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>chi_squarek5 %&gt;% 
  kable(., caption= &quot;reQTLS (response eQTLS) chi-squared pvalues&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;, full_width = FALSE) %&gt;%  
  kable_styling(full_width = FALSE, position = &quot;right&quot;,bootstrap_options = c(&quot;striped&quot;),font_size = 18) %&gt;%   scroll_box(width = &quot;50%&quot;, height = &quot;400px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:50%; ">
<table class=" lightable-paper lightable-striped table table-striped" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 18px; width: auto !important; margin-right: 0; margin-left: auto">
<caption style="font-size: initial !important;">
reQTLS (response eQTLS) chi-squared pvalues
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
id
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
time
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
pvalue
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.6576304
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
0.7387684
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.4965954
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
1.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.2539396
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
0.5705567
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.4445513
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
0.3946217
</td>
</tr>
</tbody>
</table>
</div>
<p>We can “see” the proportions of SNPs are not enriched significantly
in the DE genes compared to the non-DE genes (chi square test).<br />
We do not see significant enrichment of reQTLS in our DE gene sets over
the non-DE gene sets.</p>
<p>So what about enrichment of meSNPs in DE compared to reQTLs in
significantly DE gene sets?</p>
<pre class="r"><code>Knowles_count &lt;- 
  toplistall %&gt;%
  filter(id!=&#39;Trastuzumab&#39;) %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  mutate(time=factor(time, levels=c(&quot;3_hours&quot;,&quot;24_hours&quot;))) %&gt;%
  group_by(time, id) %&gt;%
  mutate(K4 = if_else(ENTREZID %in% knowles4$entrezgene_id,1,0))%&gt;%
  mutate(K5 = if_else(ENTREZID %in% knowles5$entrezgene_id,1,0))%&gt;%
  filter(adj.P.Val&lt;0.05) %&gt;%
  dplyr::summarize(n=n(), K4=sum(K4), K5=sum(K5)) %&gt;% 
  as.tibble() %&gt;% 
  dplyr::select(time,id,K4,K5) %&gt;% rename(&quot;K4_y&quot;=&#39;K4&#39;,&quot;K5_y&quot;=&#39;K5&#39;) %&gt;% 
  mutate(time = case_match(time, &#39;3_hours&#39;~&#39;3 hrs&#39;,
                           &#39;24_hours&#39;~&#39;24 hrs&#39;,.default = time)) %&gt;% 
  mutate(K4_n= 503-K4_y, K5_n=371-K5_y) %&gt;% 
  pivot_longer(!c(time,id), names_to=&#39;QTL&#39;,values_to=&quot;gene_count&quot;) %&gt;%
  separate(QTL,into=c(&quot;QTL_type&quot;,&#39;group&#39;),sep = &#39;_&#39;) %&gt;%  
  mutate(QTL_type =case_match(QTL_type, &#39;K4&#39;~&#39;meQTLs&#39;,
                         &#39;K5&#39;~&#39;reQTLs&#39;,.default = QTL_type)) %&gt;% 
  mutate(time=factor(time, levels=c(&quot;3 hrs&quot;,&quot;24 hrs&quot;))) %&gt;% 
  group_by(id,time,QTL_type) %&gt;% 
  mutate(percent=gene_count/sum(gene_count)*100)
  
  ggplot(Knowles_count, aes(x=QTL_type,y=gene_count, group=group,  fill=group))+
    geom_col(position=&#39;fill&#39;)+
    # guides(fill=&quot;none&quot;)+
    facet_wrap(time~id,nrow=2,ncol=4)+
    theme_classic()+
   scale_color_manual(values=drug_palNoVeh)+
    scale_fill_manual(values=drug_palNoVeh)+
       scale_y_continuous(label = scales::percent)</code></pre>
<p><img src="figure/Knowles2019.Rmd/Knowles%20comparison%20between%20sets-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  ###This works
  chi_frame &lt;- Knowles_count %&gt;%
  pivot_wider(id_cols = c(time, id,QTL_type), names_from = group, values_from = gene_count) 

testDNR3chi &lt;- matrix(c(6,13,497, 358), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c(&quot;k4&quot;, &quot;k5&quot;),c( &quot;y&quot;, &quot;n&quot;))) 
    DNR_3chi &lt;- chisq.test(testDNR3chi,correct=FALSE)$p.value
testDNR24chi &lt;- matrix(c(199,187,304, 184), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c(&quot;k4&quot;, &quot;k5&quot;),c( &quot;y&quot;, &quot;n&quot;)))  
    DNR_24chi &lt;- chisq.test(testDNR24chi,correct=FALSE)$p.value

testDOX3chi &lt;- matrix(c(0,0,503, 371), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c(&quot;k4&quot;, &quot;k5&quot;),c( &quot;y&quot;, &quot;n&quot;)))  
    DOX_3chi &lt;- chisq.test(testDOX3chi,correct=FALSE)$p.value
testDOX24chi &lt;- matrix(c(184,180,318, 191), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c(&quot;k4&quot;, &quot;k5&quot;),c( &quot;y&quot;, &quot;n&quot;))) 
    DOX_24chi &lt;- chisq.test(testDOX24chi,correct=FALSE)$p.value

testEPI3chi &lt;- matrix(c(4,4,499, 367), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c(&quot;k4&quot;, &quot;k5&quot;),c( &quot;y&quot;, &quot;n&quot;))) 
    EPI_3chi &lt;- chisq.test(testEPI3chi,correct=FALSE)$p.value
testEPI24chi &lt;- matrix(c(172,176,331, 195), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c(&quot;k4&quot;, &quot;k5&quot;),c( &quot;y&quot;, &quot;n&quot;))) 
    EPI_24chi &lt;- chisq.test(testEPI24chi,correct=FALSE)$p.value
  
testMTX3chi &lt;- matrix(c(1,0,502, 370), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c(&quot;k4&quot;, &quot;k5&quot;),c( &quot;y&quot;, &quot;n&quot;))) 
    MTX_3chi &lt;- chisq.test(testMTX3chi,correct=FALSE)$p.value
testMTX24chi &lt;- matrix(c(30,40,473, 331), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c(&quot;k4&quot;, &quot;k5&quot;),c( &quot;y&quot;, &quot;n&quot;))) 
    MTX_24chi &lt;- chisq.test(testMTX24chi,correct=FALSE)$p.value
K4nK5chi &lt;- data.frame(treatment=c(&#39;DNR_3&#39;,&#39;DNR_24&#39;,&#39;DOX_3&#39;,&#39;DOX_24&#39;,&#39;EPI_3&#39;,&#39;EPI_24&#39;,&#39;MTX_3&#39;,&#39;MTX_24&#39;), chi_p.value=c(DNR_3chi,DNR_24chi,DOX_3chi,DOX_24chi,EPI_3chi,EPI_24chi,MTX_3chi,MTX_24chi))

K4nK5chi %&gt;% 
  separate(treatment, into= c(&#39;Drug&#39;,&#39;time&#39;)) %&gt;% 
  pivot_wider(id_cols = Drug, names_from = time, values_from = chi_p.value) %&gt;% 
  kable(., caption= &quot;Chi Square p. values from chi-square test between proportions of sig-DE meQTLs and reQTLS by time and treatment&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;, full_width = TRUE) %&gt;%  
  kable_styling(full_width = FALSE, font_size = 16) %&gt;% 
  scroll_box( height = &quot;500px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; ">
<table class=" lightable-paper lightable-striped table" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; margin-left: auto; margin-right: auto; font-size: 16px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
Chi Square p. values from chi-square test between proportions of sig-DE
meQTLs and reQTLS by time and treatment
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Drug
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
3
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
24
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
DNR
</td>
<td style="text-align:right;">
0.0205682
</td>
<td style="text-align:right;">
0.0014217
</td>
</tr>
<tr>
<td style="text-align:left;">
DOX
</td>
<td style="text-align:right;">
NaN
</td>
<td style="text-align:right;">
0.0004405
</td>
</tr>
<tr>
<td style="text-align:left;">
EPI
</td>
<td style="text-align:right;">
0.6641977
</td>
<td style="text-align:right;">
0.0000770
</td>
</tr>
<tr>
<td style="text-align:left;">
MTX
</td>
<td style="text-align:right;">
0.3908069
</td>
<td style="text-align:right;">
0.0095036
</td>
</tr>
</tbody>
</table>
</div>
<p>Here we found that 24 hour reQTLs are significantly enriched in
sig-DEgens compared to meQTLS, with Daunorubicin 3 hour treatment also
showing significant enrichment.</p>
<pre class="r"><code>toplistall %&gt;% 
  filter(time==&quot;24_hours&quot;) %&gt;% 
  group_by(time, id) %&gt;% 
  mutate(sigcount = if_else(adj.P.Val &lt; 0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  ggplot(., aes(x=id, y=abs(logFC)))+
  geom_boxplot(aes(fill=id))+
  ggpubr::fill_palette(palette =drug_palNoVeh)+
  guides(fill=guide_legend(title = &quot;Treatment&quot;))+
  facet_wrap(sigcount~time)+
  theme_bw()+
  xlab(&quot;&quot;)+
  ylab(&quot;abs |Log Fold Change|&quot;)+
  theme_bw()+
  facet_wrap(~time)+
  #ggtitle(&quot;All QTLs from all expressed genes (n=507)&quot;)+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = &quot;black&quot;),
        # axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        strip.background = element_rect(fill = &quot;transparent&quot;),
        axis.text = element_text(size = 8, color = &quot;black&quot;, angle = 0),
        strip.text.x = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;))</code></pre>
<p><img src="figure/Knowles2019.Rmd/abs%20FC%20magnitude-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="dox-reqtls" class="section level2">
<h2>Dox-reQTLs</h2>
<p>When I look ot the reQTLs across treatments at 24 hours, I see that
dox has a total of 180 reQTLS. Because this gene set was specifically
about doxorubicin response eQTLs, I wanted to see if these reQTLs were
only Dox specific, AC specific, Top2i specific, or show any type of
pattern.</p>
<pre class="r"><code>siglist &lt;- readRDS(&quot;data/siglist.RDS&quot;)
list2env(siglist,.GlobalEnv)</code></pre>
<pre><code>&lt;environment: R_GlobalEnv&gt;</code></pre>
<pre class="r"><code>reQTLcombine=list(DNRreQTLs$ENTREZID,DOXreQTLs$ENTREZID,EPIreQTLs$ENTREZID,MTXreQTLs$ENTREZID)

length(unique(c(DNRreQTLs$ENTREZID,DOXreQTLs$ENTREZID,EPIreQTLs$ENTREZID,MTXreQTLs$ENTREZID)))</code></pre>
<pre><code>[1] 218</code></pre>
<pre class="r"><code>print(&quot;number of unique genes expressed in pairwise DE set&quot;)</code></pre>
<pre><code>[1] &quot;number of unique genes expressed in pairwise DE set&quot;</code></pre>
<pre class="r"><code>QTLoverlaps &lt;- VennDiagram::get.venn.partitions(reQTLcombine)
DoxonlyQTL &lt;-  QTLoverlaps$..values..[[14]]
     ggVennDiagram::ggVennDiagram(reQTLcombine, category.names = c(&quot;DNR-187\nsigDEG&quot;,&quot;DOX-180 \nsigDEG\n&quot;,&quot;EPI-176\nsigDEG\n&quot;,&quot;MTX-40\nsigDEG&quot;),label = &quot;count&quot;) +scale_x_continuous(expand = expansion(mult = .2))+scale_y_continuous(expand = expansion(mult = .1))</code></pre>
<p><img src="figure/Knowles2019.Rmd/Doxreqtl-1.png" width="672" style="display: block; margin: auto;" />
Here, I found 119 reQTLs that were related to all anthracyclines, with
31 reQTLS related to all Top2i drugs at 24 hours.</p>
<pre class="r"><code> DOXeQTL_table &lt;- toplistall %&gt;% 
  mutate(sigcount = if_else(adj.P.Val &lt; 0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(DOXreQTLs=if_else(ENTREZID %in%DOXreQTLs$ENTREZID,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  dplyr::filter(time ==&quot;24_hours&quot;) %&gt;% 
  dplyr::select(ENTREZID,id,DOXreQTLs,sigcount) %&gt;%
  group_by(id,DOXreQTLs,sigcount) %&gt;% 
  tally() %&gt;% as.data.frame() %&gt;% 
  pivot_wider(.,id_cols = c(id,DOXreQTLs),names_from = sigcount,values_from = n) %&gt;% 
  dplyr::select(id,DOXreQTLs,sig) 
DOXeQTL_table%&gt;% 
  kable(., caption= &quot;All 24 hour sidDEGs broken down by drug and DOXreQTL status&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;, full_width = FALSE) %&gt;% 
  kable_styling(full_width = FALSE, position = &quot;left&quot;,bootstrap_options = c(&quot;striped&quot;),font_size = 18) %&gt;% 
  scroll_box(width = &quot;80%&quot;, height = &quot;400px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:80%; ">
<table class=" lightable-paper lightable-striped table table-striped" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 18px; width: auto !important; ">
<caption style="font-size: initial !important;">
All 24 hour sidDEGs broken down by drug and DOXreQTL status
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
id
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
DOXreQTLs
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
sig
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
6695
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
169
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
6336
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
180
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
6048
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
154
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
1291
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
36
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
NA
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>DOXeQTL_table %&gt;%
  dplyr::filter(DOXreQTLs==&quot;y&quot;) %&gt;% 
  mutate(opp= 180-sig) %&gt;% 
  filter(id !=c(&#39;Trastuzumab&#39;,&#39;Doxorubicin&#39;)) %&gt;% 
  rename(&quot;yes&quot;=sig, &quot;no&quot;=opp) %&gt;% 
  pivot_longer(!c(id,DOXreQTLs), names_to=&quot;group&quot;, values_to = &quot;count&quot;) %&gt;% 
  mutate(id=factor(id, levels=c(&quot;Daunorubicin&quot;,&quot;Epirubicin&quot;, &quot;Mitoxantrone&quot;))) %&gt;% 
ggplot(., aes(y=id,x=count,fill=group))+
     geom_col(position=&#39;fill&#39;)+
    theme_classic()+
   scale_color_manual(values=c(&quot;red4&quot;,&quot;navy&quot;))+
    scale_fill_manual(values=c(&quot;#2297E6&quot;,&quot;red2&quot;))+
  ggtitle(&quot;DOXreQTLs overlaps&quot;)+

    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = &quot;black&quot;),
        axis.ticks = element_line(linewidth = 1.0),
        axis.line = element_line(linewidth = 1.0),
        axis.text = element_text(size = 10, color = &quot;black&quot;, angle = 0),
        strip.text.x = element_text(size = 15, color = &quot;black&quot;, face = &quot;bold&quot;))</code></pre>
<p><img src="figure/Knowles2019.Rmd/doxreqtl%20examine-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="dox-specific-dox-de-reqtl" class="section level4">
<h4>DOX specific-Dox dE-reqtl</h4>
<pre class="r"><code>DoxonlyQTL &lt;-  as.numeric(QTLoverlaps$..values..[[14]])
# geneDoxonlyQTL &lt;- getBM(attributes=my_attributes,filters =&#39;entrezgene_id&#39;,
#                   values =DoxonlyQTL, mart = ensembl)
# write.csv(geneDoxonlyQTL,&quot;data/geneDoxonlyQTL.csv&quot;)
# saveRDS(cpmcounts,&quot;data/cpmcount.RDS&quot;)
geneDoxonlyQTL &lt;- read.csv(&quot;data/geneDoxonlyQTL.csv&quot;,row.names = 1)
cpmcounts &lt;- readRDS(&quot;data/cpmcount.RDS&quot;)

for (g in seq(from=1, to=length(geneDoxonlyQTL$entrezgene_id))){
  a &lt;- geneDoxonlyQTL$hgnc_symbol[g]
  cpm_boxplot(cpmcounts,GOI=geneDoxonlyQTL[g,1],&quot;Dark2&quot;,drug_palc,
                ylab=bquote(~italic(.(a))~log[2]~&quot;cpm &quot;))

}</code></pre>
<p><img src="figure/Knowles2019.Rmd/DOX%20spec%20DEreQTL-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/DOX%20spec%20DEreQTL-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/DOX%20spec%20DEreQTL-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/DOX%20spec%20DEreQTL-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/DOX%20spec%20DEreQTL-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/DOX%20spec%20DEreQTL-6.png" width="672" style="display: block; margin: auto;" />
### Dox specific DEG examination</p>
<pre class="r"><code>#pull list
total24 &lt;-list(sigVDA24$ENTREZID,sigVDX24$ENTREZID,sigVEP24$ENTREZID,sigVMT24$ENTREZID)
## do venn partition and pull doxspe genes



venn_24h &lt;- VennDiagram::get.venn.partitions(total24)
DoxonlyDEG &lt;- venn_24h$..values..[[14]]
EpionlyDEG &lt;- venn_24h$..values..[[12]]
DnronlyDEG &lt;- venn_24h$..values..[[15]]
MtxonlyDEG &lt;- venn_24h$..values..[[8]]
  
  
Dox24_lfc &lt;- toplist24hr %&gt;% 
  filter(ENTREZID %in% DoxonlyDEG) %&gt;% 
  filter(adj.P.Val&lt;0.05) %&gt;% 
  mutate(logFC=logFC*(-1)) %&gt;% 
  ggplot(., aes(x= id, y=logFC))+
  geom_boxplot(aes(fill=id))+
  theme_classic()+
  fill_palette(palette = drug_palNoVeh)+
  ggtitle(&quot;LogFC of Dox specific DEGs&quot;)+
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5,face = &quot;bold&quot;),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(size = 1.5),
    axis.text = element_text(size = 8, color = &quot;black&quot;, angle = 20))
    # strip.text.x = element_text(size = 12, color = &quot;black&quot;, face = &quot;italic&quot;))
  
  toplist24hr %&gt;% 
  filter(ENTREZID %in% DoxonlyDEG) %&gt;% 
  # filter(adj.P.Val&lt;0.05) %&gt;%
  ggplot(., aes(x=adj.P.Val))+
  geom_histogram(aes(fill=id,position=&quot;dodge&quot;))+
    # geom_density(aes(fill=id))+
    facet_wrap(~id)+
    fill_palette(palette = drug_palNoVeh)</code></pre>
<p><img src="figure/Knowles2019.Rmd/Dox%20specific%20LFC-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>toplist24hr %&gt;% 
  filter(ENTREZID %in% DoxonlyDEG) %&gt;%
  ggplot(., aes(x=adj.P.Val))+
  geom_density(aes(fill=id, alpha= 0.8))+
  fill_palette(palette = drug_palNoVeh)</code></pre>
<p><img src="figure/Knowles2019.Rmd/Dox%20specific%20LFC-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>print(Dox24_lfc)</code></pre>
<p><img src="figure/Knowles2019.Rmd/Dox%20specific%20LFC-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="plots-of-some-genes" class="section level2">
<h2>Plots of some genes:</h2>
<pre class="r"><code>### making list
# 
# Doxonly_deg &lt;- getBM(attributes=my_attributes,filters =&#39;entrezgene_id&#39;,
#                   values =DoxonlyDEG, mart = ensembl)
# write.csv(Doxonly_deg,&quot;output/Doxonly_deg.csv&quot;)
Doxonly_deg &lt;- read.csv(&quot;output/Doxonly_deg.csv&quot;, row.names = 1)
set.seed(12345)
sampset &lt;- Doxonly_deg %&gt;% 
  distinct(entrezgene_id,.keep_all = TRUE) %&gt;% 
  sample_n(.,12)
for (g in seq(from=1, to=length(sampset$entrezgene_id))){
  a &lt;- sampset$hgnc_symbol[g]
  cpm_boxplot(cpmcounts,GOI=sampset[g,1],&quot;Dark2&quot;,drug_palc,
                ylab=bquote(~italic(.(a))~log[2]~&quot;cpm &quot;))

}</code></pre>
<p><img src="figure/Knowles2019.Rmd/lcpm%20of%20some%20dox%20spec%20genes-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/lcpm%20of%20some%20dox%20spec%20genes-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/lcpm%20of%20some%20dox%20spec%20genes-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/lcpm%20of%20some%20dox%20spec%20genes-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/lcpm%20of%20some%20dox%20spec%20genes-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/lcpm%20of%20some%20dox%20spec%20genes-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/lcpm%20of%20some%20dox%20spec%20genes-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/lcpm%20of%20some%20dox%20spec%20genes-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/lcpm%20of%20some%20dox%20spec%20genes-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/lcpm%20of%20some%20dox%20spec%20genes-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/lcpm%20of%20some%20dox%20spec%20genes-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/lcpm%20of%20some%20dox%20spec%20genes-12.png" width="672" style="display: block; margin: auto;" /></p>
<div id="more-investigation-of-24-hour-specific-genes"
class="section level3">
<h3>More investigation of 24 hour specific genes</h3>
<div id="doxorubicin" class="section level4">
<h4>Doxorubicin</h4>
<pre class="r"><code>DOXdeg_sp &lt;- toplist24hr %&gt;% 
  filter(if_else(id==&quot;Doxorubicin&quot;,adj.P.Val&lt;0.01,adj.P.Val&gt;0.01 )) %&gt;% 
  filter(ENTREZID %in% DoxonlyDEG) %&gt;% 
  filter(id==&quot;Doxorubicin&quot;) %&gt;% 
  dplyr::select(ENTREZID, SYMBOL)  

DOXdeg_sp %&gt;% 
  kable(., caption= &quot;DOX specific genes&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;, full_width = TRUE) %&gt;%  
  kable_styling(full_width = FALSE, font_size = 16) %&gt;% 
  scroll_box( height = &quot;500px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; ">
<table class=" lightable-paper lightable-striped table" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; margin-left: auto; margin-right: auto; font-size: 16px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
DOX specific genes
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ENTREZID
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
SYMBOL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
169200
</td>
<td style="text-align:left;">
169200
</td>
<td style="text-align:left;">
TMEM64
</td>
</tr>
<tr>
<td style="text-align:left;">
653082
</td>
<td style="text-align:left;">
653082
</td>
<td style="text-align:left;">
ZDHHC11B
</td>
</tr>
<tr>
<td style="text-align:left;">
1452
</td>
<td style="text-align:left;">
1452
</td>
<td style="text-align:left;">
CSNK1A1
</td>
</tr>
<tr>
<td style="text-align:left;">
23108
</td>
<td style="text-align:left;">
23108
</td>
<td style="text-align:left;">
RAP1GAP2
</td>
</tr>
<tr>
<td style="text-align:left;">
114882
</td>
<td style="text-align:left;">
114882
</td>
<td style="text-align:left;">
OSBPL8
</td>
</tr>
<tr>
<td style="text-align:left;">
8501
</td>
<td style="text-align:left;">
8501
</td>
<td style="text-align:left;">
SLC43A1
</td>
</tr>
<tr>
<td style="text-align:left;">
5000
</td>
<td style="text-align:left;">
5000
</td>
<td style="text-align:left;">
ORC4
</td>
</tr>
<tr>
<td style="text-align:left;">
202181
</td>
<td style="text-align:left;">
202181
</td>
<td style="text-align:left;">
LOC202181
</td>
</tr>
<tr>
<td style="text-align:left;">
135154
</td>
<td style="text-align:left;">
135154
</td>
<td style="text-align:left;">
SDHAF4
</td>
</tr>
<tr>
<td style="text-align:left;">
122553
</td>
<td style="text-align:left;">
122553
</td>
<td style="text-align:left;">
TRAPPC6B
</td>
</tr>
<tr>
<td style="text-align:left;">
79065
</td>
<td style="text-align:left;">
79065
</td>
<td style="text-align:left;">
ATG9A
</td>
</tr>
<tr>
<td style="text-align:left;">
8763
</td>
<td style="text-align:left;">
8763
</td>
<td style="text-align:left;">
CD164
</td>
</tr>
<tr>
<td style="text-align:left;">
284900
</td>
<td style="text-align:left;">
284900
</td>
<td style="text-align:left;">
TTC28-AS1
</td>
</tr>
<tr>
<td style="text-align:left;">
26994
</td>
<td style="text-align:left;">
26994
</td>
<td style="text-align:left;">
RNF11
</td>
</tr>
<tr>
<td style="text-align:left;">
256586
</td>
<td style="text-align:left;">
256586
</td>
<td style="text-align:left;">
LYSMD2
</td>
</tr>
<tr>
<td style="text-align:left;">
54165
</td>
<td style="text-align:left;">
54165
</td>
<td style="text-align:left;">
DCUN1D1
</td>
</tr>
<tr>
<td style="text-align:left;">
8515
</td>
<td style="text-align:left;">
8515
</td>
<td style="text-align:left;">
ITGA10
</td>
</tr>
<tr>
<td style="text-align:left;">
440944
</td>
<td style="text-align:left;">
440944
</td>
<td style="text-align:left;">
THUMPD3-AS1
</td>
</tr>
<tr>
<td style="text-align:left;">
57181
</td>
<td style="text-align:left;">
57181
</td>
<td style="text-align:left;">
SLC39A10
</td>
</tr>
<tr>
<td style="text-align:left;">
57338
</td>
<td style="text-align:left;">
57338
</td>
<td style="text-align:left;">
JPH3
</td>
</tr>
<tr>
<td style="text-align:left;">
2230
</td>
<td style="text-align:left;">
2230
</td>
<td style="text-align:left;">
FDX1
</td>
</tr>
<tr>
<td style="text-align:left;">
23548
</td>
<td style="text-align:left;">
23548
</td>
<td style="text-align:left;">
TTC33
</td>
</tr>
<tr>
<td style="text-align:left;">
81532
</td>
<td style="text-align:left;">
81532
</td>
<td style="text-align:left;">
MOB2
</td>
</tr>
<tr>
<td style="text-align:left;">
27430
</td>
<td style="text-align:left;">
27430
</td>
<td style="text-align:left;">
MAT2B
</td>
</tr>
<tr>
<td style="text-align:left;">
9524
</td>
<td style="text-align:left;">
9524
</td>
<td style="text-align:left;">
TECR
</td>
</tr>
<tr>
<td style="text-align:left;">
805
</td>
<td style="text-align:left;">
805
</td>
<td style="text-align:left;">
CALM2
</td>
</tr>
<tr>
<td style="text-align:left;">
147
</td>
<td style="text-align:left;">
147
</td>
<td style="text-align:left;">
ADRA1B
</td>
</tr>
<tr>
<td style="text-align:left;">
2551
</td>
<td style="text-align:left;">
2551
</td>
<td style="text-align:left;">
GABPA
</td>
</tr>
<tr>
<td style="text-align:left;">
26001
</td>
<td style="text-align:left;">
26001
</td>
<td style="text-align:left;">
RNF167
</td>
</tr>
<tr>
<td style="text-align:left;">
100131211
</td>
<td style="text-align:left;">
100131211
</td>
<td style="text-align:left;">
NEMP2
</td>
</tr>
<tr>
<td style="text-align:left;">
79624
</td>
<td style="text-align:left;">
79624
</td>
<td style="text-align:left;">
ARMT1
</td>
</tr>
<tr>
<td style="text-align:left;">
10190
</td>
<td style="text-align:left;">
10190
</td>
<td style="text-align:left;">
TXNDC9
</td>
</tr>
<tr>
<td style="text-align:left;">
7917
</td>
<td style="text-align:left;">
7917
</td>
<td style="text-align:left;">
BAG6
</td>
</tr>
<tr>
<td style="text-align:left;">
6653
</td>
<td style="text-align:left;">
6653
</td>
<td style="text-align:left;">
SORL1
</td>
</tr>
<tr>
<td style="text-align:left;">
9039
</td>
<td style="text-align:left;">
9039
</td>
<td style="text-align:left;">
UBA3
</td>
</tr>
<tr>
<td style="text-align:left;">
57862
</td>
<td style="text-align:left;">
57862
</td>
<td style="text-align:left;">
ZNF410
</td>
</tr>
<tr>
<td style="text-align:left;">
6498
</td>
<td style="text-align:left;">
6498
</td>
<td style="text-align:left;">
SKIL
</td>
</tr>
<tr>
<td style="text-align:left;">
112858
</td>
<td style="text-align:left;">
112858
</td>
<td style="text-align:left;">
TP53RK
</td>
</tr>
<tr>
<td style="text-align:left;">
101927720
</td>
<td style="text-align:left;">
101927720
</td>
<td style="text-align:left;">
ZNF793-AS1
</td>
</tr>
<tr>
<td style="text-align:left;">
56998
</td>
<td style="text-align:left;">
56998
</td>
<td style="text-align:left;">
CTNNBIP1
</td>
</tr>
<tr>
<td style="text-align:left;">
5431
</td>
<td style="text-align:left;">
5431
</td>
<td style="text-align:left;">
POLR2B
</td>
</tr>
<tr>
<td style="text-align:left;">
25842
</td>
<td style="text-align:left;">
25842
</td>
<td style="text-align:left;">
ASF1A
</td>
</tr>
<tr>
<td style="text-align:left;">
55664
</td>
<td style="text-align:left;">
55664
</td>
<td style="text-align:left;">
CDC37L1
</td>
</tr>
<tr>
<td style="text-align:left;">
57186
</td>
<td style="text-align:left;">
57186
</td>
<td style="text-align:left;">
RALGAPA2
</td>
</tr>
<tr>
<td style="text-align:left;">
5598
</td>
<td style="text-align:left;">
5598
</td>
<td style="text-align:left;">
MAPK7
</td>
</tr>
<tr>
<td style="text-align:left;">
5594
</td>
<td style="text-align:left;">
5594
</td>
<td style="text-align:left;">
MAPK1
</td>
</tr>
<tr>
<td style="text-align:left;">
58490
</td>
<td style="text-align:left;">
58490
</td>
<td style="text-align:left;">
RPRD1B
</td>
</tr>
<tr>
<td style="text-align:left;">
8848
</td>
<td style="text-align:left;">
8848
</td>
<td style="text-align:left;">
TSC22D1
</td>
</tr>
<tr>
<td style="text-align:left;">
29944
</td>
<td style="text-align:left;">
29944
</td>
<td style="text-align:left;">
PNMA3
</td>
</tr>
<tr>
<td style="text-align:left;">
2130
</td>
<td style="text-align:left;">
2130
</td>
<td style="text-align:left;">
EWSR1
</td>
</tr>
<tr>
<td style="text-align:left;">
6139
</td>
<td style="text-align:left;">
6139
</td>
<td style="text-align:left;">
RPL17
</td>
</tr>
<tr>
<td style="text-align:left;">
9446
</td>
<td style="text-align:left;">
9446
</td>
<td style="text-align:left;">
GSTO1
</td>
</tr>
<tr>
<td style="text-align:left;">
102723508
</td>
<td style="text-align:left;">
102723508
</td>
<td style="text-align:left;">
KANTR
</td>
</tr>
<tr>
<td style="text-align:left;">
4987
</td>
<td style="text-align:left;">
4987
</td>
<td style="text-align:left;">
OPRL1
</td>
</tr>
<tr>
<td style="text-align:left;">
84458
</td>
<td style="text-align:left;">
84458
</td>
<td style="text-align:left;">
LCOR
</td>
</tr>
<tr>
<td style="text-align:left;">
84752
</td>
<td style="text-align:left;">
84752
</td>
<td style="text-align:left;">
B3GNT9
</td>
</tr>
<tr>
<td style="text-align:left;">
54629
</td>
<td style="text-align:left;">
54629
</td>
<td style="text-align:left;">
MINDY2
</td>
</tr>
<tr>
<td style="text-align:left;">
54469
</td>
<td style="text-align:left;">
54469
</td>
<td style="text-align:left;">
ZFAND6
</td>
</tr>
<tr>
<td style="text-align:left;">
144363
</td>
<td style="text-align:left;">
144363
</td>
<td style="text-align:left;">
ETFRF1
</td>
</tr>
<tr>
<td style="text-align:left;">
9821
</td>
<td style="text-align:left;">
9821
</td>
<td style="text-align:left;">
RB1CC1
</td>
</tr>
<tr>
<td style="text-align:left;">
116068
</td>
<td style="text-align:left;">
116068
</td>
<td style="text-align:left;">
LYSMD3
</td>
</tr>
<tr>
<td style="text-align:left;">
66036
</td>
<td style="text-align:left;">
66036
</td>
<td style="text-align:left;">
MTMR9
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>toplist24hr %&gt;% 
  group_by(time,id) %&gt;% 
  filter(ENTREZID %in% DOXdeg_sp$ENTREZID) %&gt;% 
  filter(adj.P.Val &lt;0.05) %&gt;% 
  # filter(id==&quot;Doxorubicin&quot;)
  mutate(logFC=logFC*(-1)) %&gt;% 
  ggplot(., aes(x= id, y=logFC))+
  geom_boxplot(aes(fill=id))+
  theme_classic()+
  fill_palette(palette = drug_palNoVeh)+
  ggtitle(&quot;LogFC of n=62 Dox specific DEGs&quot;)+
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5,face = &quot;bold&quot;),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(size = 1.5),
    axis.text = element_text(size = 8, color = &quot;black&quot;, angle = 20))</code></pre>
<p><img src="figure/Knowles2019.Rmd/Dox%20investigation-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>    # strip.text.x = element_text(size = 12, color = &quot;black&quot;, face = &quot;italic&quot;))

toplist24hr %&gt;% 
  filter(if_else(id==&quot;Doxorubicin&quot;,adj.P.Val&lt;0.01,adj.P.Val&gt;0.01 )) %&gt;% 
  group_by(time,id) %&gt;% 
  filter(ENTREZID %in% DoxonlyDEG) %&gt;% 
  filter(adj.P.Val &lt;0.05) %&gt;% 
  ggplot(., aes(x=adj.P.Val))+
  geom_histogram(aes(fill=id))+
  geom_vline(xintercept=0.01,linetype=2)+
    # geom_density(aes(fill=id))+
    facet_wrap(~id)+
  ggtitle(&quot;DOX adj p. value &lt;0.05&quot;)+
    fill_palette(palette = drug_palNoVeh)+
  theme_bw()</code></pre>
<p><img src="figure/Knowles2019.Rmd/Dox%20investigation-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>toplist24hr %&gt;% 
  filter(ENTREZID %in% DoxonlyDEG) %&gt;% 
  # filter(adj.P.Val&lt;0.05) %&gt;%
  ggplot(., aes(x=adj.P.Val))+
  geom_density(aes(fill=id, alpha= 0.8))+
  fill_palette(palette = drug_palNoVeh)+
  theme_bw()</code></pre>
<p><img src="figure/Knowles2019.Rmd/Dox%20investigation-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>set.seed(12345)
sampset &lt;- DOXdeg_sp %&gt;% 
  sample_n(.,12)

for (g in seq(from=1, to=length(sampset$ENTREZID))){
  a &lt;- sampset$SYMBOL[g]
  cpm_boxplot(cpmcounts,GOI=sampset[g,1],&quot;Dark2&quot;,drug_palc,
                ylab=bquote(~italic(.(a))~log[2]~&quot;cpm &quot;))
}</code></pre>
<p><img src="figure/Knowles2019.Rmd/Dox%20investigation-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/Dox%20investigation-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/Dox%20investigation-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/Dox%20investigation-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/Dox%20investigation-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/Dox%20investigation-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/Dox%20investigation-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/Dox%20investigation-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/Dox%20investigation-12.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/Dox%20investigation-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/Dox%20investigation-14.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/Dox%20investigation-15.png" width="672" style="display: block; margin: auto;" />
#### Mitoxantrone</p>
<pre class="r"><code>MTXdeg_sp &lt;- toplist24hr %&gt;% 
  filter(if_else(id==&quot;Mitoxantrone&quot;,adj.P.Val&lt;0.01,adj.P.Val&gt;0.01 )) %&gt;% 
  filter(ENTREZID %in% MtxonlyDEG) %&gt;% 
  filter(id==&quot;Mitoxantrone&quot;) %&gt;% 
  dplyr::select(ENTREZID, SYMBOL)  

MTXdeg_sp %&gt;% 
  kable(., caption= &quot;MTX specific genes&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;, full_width = TRUE) %&gt;%  
  kable_styling(full_width = FALSE, font_size = 16) %&gt;% 
  scroll_box( height = &quot;500px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; ">
<table class=" lightable-paper lightable-striped table" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; margin-left: auto; margin-right: auto; font-size: 16px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
MTX specific genes
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ENTREZID
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
SYMBOL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
25894
</td>
<td style="text-align:left;">
25894
</td>
<td style="text-align:left;">
PLEKHG4
</td>
</tr>
<tr>
<td style="text-align:left;">
126432
</td>
<td style="text-align:left;">
126432
</td>
<td style="text-align:left;">
RINL
</td>
</tr>
<tr>
<td style="text-align:left;">
100128191
</td>
<td style="text-align:left;">
100128191
</td>
<td style="text-align:left;">
TMPO-AS1
</td>
</tr>
<tr>
<td style="text-align:left;">
63827
</td>
<td style="text-align:left;">
63827
</td>
<td style="text-align:left;">
BCAN
</td>
</tr>
<tr>
<td style="text-align:left;">
55147
</td>
<td style="text-align:left;">
55147
</td>
<td style="text-align:left;">
RBM23
</td>
</tr>
<tr>
<td style="text-align:left;">
253714
</td>
<td style="text-align:left;">
253714
</td>
<td style="text-align:left;">
MMS22L
</td>
</tr>
<tr>
<td style="text-align:left;">
23582
</td>
<td style="text-align:left;">
23582
</td>
<td style="text-align:left;">
CCNDBP1
</td>
</tr>
<tr>
<td style="text-align:left;">
4077
</td>
<td style="text-align:left;">
4077
</td>
<td style="text-align:left;">
NBR1
</td>
</tr>
<tr>
<td style="text-align:left;">
7014
</td>
<td style="text-align:left;">
7014
</td>
<td style="text-align:left;">
TERF2
</td>
</tr>
<tr>
<td style="text-align:left;">
25886
</td>
<td style="text-align:left;">
25886
</td>
<td style="text-align:left;">
POC1A
</td>
</tr>
<tr>
<td style="text-align:left;">
6240
</td>
<td style="text-align:left;">
6240
</td>
<td style="text-align:left;">
RRM1
</td>
</tr>
<tr>
<td style="text-align:left;">
83879
</td>
<td style="text-align:left;">
83879
</td>
<td style="text-align:left;">
CDCA7
</td>
</tr>
<tr>
<td style="text-align:left;">
79758
</td>
<td style="text-align:left;">
79758
</td>
<td style="text-align:left;">
DHRS12
</td>
</tr>
<tr>
<td style="text-align:left;">
23786
</td>
<td style="text-align:left;">
23786
</td>
<td style="text-align:left;">
BCL2L13
</td>
</tr>
<tr>
<td style="text-align:left;">
55723
</td>
<td style="text-align:left;">
55723
</td>
<td style="text-align:left;">
ASF1B
</td>
</tr>
<tr>
<td style="text-align:left;">
22950
</td>
<td style="text-align:left;">
22950
</td>
<td style="text-align:left;">
SLC4A1AP
</td>
</tr>
<tr>
<td style="text-align:left;">
54853
</td>
<td style="text-align:left;">
54853
</td>
<td style="text-align:left;">
WDR55
</td>
</tr>
<tr>
<td style="text-align:left;">
4001
</td>
<td style="text-align:left;">
4001
</td>
<td style="text-align:left;">
LMNB1
</td>
</tr>
<tr>
<td style="text-align:left;">
100996573
</td>
<td style="text-align:left;">
100996573
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
55780
</td>
<td style="text-align:left;">
55780
</td>
<td style="text-align:left;">
ERMARD
</td>
</tr>
<tr>
<td style="text-align:left;">
25981
</td>
<td style="text-align:left;">
25981
</td>
<td style="text-align:left;">
DNAH1
</td>
</tr>
<tr>
<td style="text-align:left;">
56970
</td>
<td style="text-align:left;">
56970
</td>
<td style="text-align:left;">
ATXN7L3
</td>
</tr>
<tr>
<td style="text-align:left;">
11152
</td>
<td style="text-align:left;">
11152
</td>
<td style="text-align:left;">
WDR45
</td>
</tr>
<tr>
<td style="text-align:left;">
9820
</td>
<td style="text-align:left;">
9820
</td>
<td style="text-align:left;">
CUL7
</td>
</tr>
<tr>
<td style="text-align:left;">
23558
</td>
<td style="text-align:left;">
23558
</td>
<td style="text-align:left;">
WBP2
</td>
</tr>
<tr>
<td style="text-align:left;">
6293
</td>
<td style="text-align:left;">
6293
</td>
<td style="text-align:left;">
VPS52
</td>
</tr>
<tr>
<td style="text-align:left;">
150962
</td>
<td style="text-align:left;">
150962
</td>
<td style="text-align:left;">
PUS10
</td>
</tr>
<tr>
<td style="text-align:left;">
26164
</td>
<td style="text-align:left;">
26164
</td>
<td style="text-align:left;">
MTG2
</td>
</tr>
<tr>
<td style="text-align:left;">
64799
</td>
<td style="text-align:left;">
64799
</td>
<td style="text-align:left;">
IQCH
</td>
</tr>
<tr>
<td style="text-align:left;">
27346
</td>
<td style="text-align:left;">
27346
</td>
<td style="text-align:left;">
TMEM97
</td>
</tr>
<tr>
<td style="text-align:left;">
55717
</td>
<td style="text-align:left;">
55717
</td>
<td style="text-align:left;">
WDR11
</td>
</tr>
<tr>
<td style="text-align:left;">
8723
</td>
<td style="text-align:left;">
8723
</td>
<td style="text-align:left;">
SNX4
</td>
</tr>
<tr>
<td style="text-align:left;">
4173
</td>
<td style="text-align:left;">
4173
</td>
<td style="text-align:left;">
MCM4
</td>
</tr>
<tr>
<td style="text-align:left;">
10432
</td>
<td style="text-align:left;">
10432
</td>
<td style="text-align:left;">
RBM14
</td>
</tr>
<tr>
<td style="text-align:left;">
655
</td>
<td style="text-align:left;">
655
</td>
<td style="text-align:left;">
BMP7
</td>
</tr>
<tr>
<td style="text-align:left;">
9391
</td>
<td style="text-align:left;">
9391
</td>
<td style="text-align:left;">
CIAO1
</td>
</tr>
<tr>
<td style="text-align:left;">
3148
</td>
<td style="text-align:left;">
3148
</td>
<td style="text-align:left;">
HMGB2
</td>
</tr>
<tr>
<td style="text-align:left;">
8317
</td>
<td style="text-align:left;">
8317
</td>
<td style="text-align:left;">
CDC7
</td>
</tr>
<tr>
<td style="text-align:left;">
441478
</td>
<td style="text-align:left;">
441478
</td>
<td style="text-align:left;">
NRARP
</td>
</tr>
<tr>
<td style="text-align:left;">
90120
</td>
<td style="text-align:left;">
90120
</td>
<td style="text-align:left;">
TMEM250
</td>
</tr>
<tr>
<td style="text-align:left;">
57699
</td>
<td style="text-align:left;">
57699
</td>
<td style="text-align:left;">
CPNE5
</td>
</tr>
<tr>
<td style="text-align:left;">
6273
</td>
<td style="text-align:left;">
6273
</td>
<td style="text-align:left;">
S100A2
</td>
</tr>
<tr>
<td style="text-align:left;">
55670
</td>
<td style="text-align:left;">
55670
</td>
<td style="text-align:left;">
PEX26
</td>
</tr>
<tr>
<td style="text-align:left;">
79027
</td>
<td style="text-align:left;">
79027
</td>
<td style="text-align:left;">
ZNF655
</td>
</tr>
<tr>
<td style="text-align:left;">
11176
</td>
<td style="text-align:left;">
11176
</td>
<td style="text-align:left;">
BAZ2A
</td>
</tr>
<tr>
<td style="text-align:left;">
100131067
</td>
<td style="text-align:left;">
100131067
</td>
<td style="text-align:left;">
CKMT2-AS1
</td>
</tr>
<tr>
<td style="text-align:left;">
84892
</td>
<td style="text-align:left;">
84892
</td>
<td style="text-align:left;">
POMGNT2
</td>
</tr>
<tr>
<td style="text-align:left;">
9134
</td>
<td style="text-align:left;">
9134
</td>
<td style="text-align:left;">
CCNE2
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>toplist24hr %&gt;% 
  filter(ENTREZID %in% MTXdeg_sp$ENTREZID) %&gt;% 
  filter(adj.P.Val &lt;0.05) %&gt;% 
  # filter(id==&quot;Mitoxantrone&quot;)
  mutate(logFC=logFC*(-1)) %&gt;% 
  add_row(time=&quot;24_hours&quot;,id=&quot;Epirubicin&quot;) %&gt;% 
  add_row(time=&quot;24_hours&quot;,id=&quot;Daunorubicin&quot;) %&gt;% 
  mutate(id= factor(id,levels= c(&quot;Daunorubicin&quot;, &quot;Doxorubicin&quot;, &quot;Epirubicin&quot;, &quot;Mitoxantrone&quot;))) %&gt;% 
  ggplot(., aes(x= id, y=logFC))+
  geom_boxplot(aes(fill=id))+
  theme_classic()+
  fill_palette(palette = drug_palNoVeh,drop = FALSE)+
  ggtitle(&quot;LogFC of n=48 MTX specific DEGs&quot;)+
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5,face = &quot;bold&quot;),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(size = 1.5),
    axis.text = element_text(size = 8, color = &quot;black&quot;, angle = 20))</code></pre>
<p><img src="figure/Knowles2019.Rmd/mitoonly-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>    # strip.text.x = element_text(size = 12, color = &quot;black&quot;, face = &quot;italic&quot;))

toplist24hr %&gt;% 
  filter(if_else(id==&quot;Mitoxantrone&quot;,adj.P.Val&lt;0.01,adj.P.Val&gt;0.01 )) %&gt;% 
  filter(ENTREZID %in% MtxonlyDEG) %&gt;% 
  filter(adj.P.Val &lt;0.05) %&gt;% 
   add_row(time=&quot;24_hours&quot;,id=&quot;Epirubicin&quot;) %&gt;% 
   mutate(id= factor(id, levels = c(&quot;Daunorubicin&quot;, &quot;Doxorubicin&quot;, &quot;Epirubicin&quot;, &quot;Mitoxantrone&quot;))) %&gt;% 
   ggplot(., aes(x=adj.P.Val))+
  geom_histogram(aes(fill=id))+
  geom_vline(xintercept=0.01, linetype=2)+
    # geom_density(aes(fill=id))+
    facet_wrap(~id)+
  ggtitle(&quot;MTX adj p. value &lt;0.05&quot;)+
    fill_palette(palette = drug_palNoVeh, drop = FALSE)+
  theme_bw()</code></pre>
<p><img src="figure/Knowles2019.Rmd/mitoonly-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>toplist24hr %&gt;% 
  filter(ENTREZID %in% MtxonlyDEG) %&gt;% 
  # filter(adj.P.Val&lt;0.05) %&gt;%
  ggplot(., aes(x=adj.P.Val))+
  geom_density(aes(fill=id, alpha= 0.8))+
  fill_palette(palette = drug_palNoVeh)+
  theme_bw()</code></pre>
<p><img src="figure/Knowles2019.Rmd/mitoonly-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>set.seed(12345)
sampset &lt;- MTXdeg_sp %&gt;% 
  sample_n(.,12)

for (g in seq(from=1, to=length(sampset$ENTREZID))){
  a &lt;- sampset$SYMBOL[g]
  cpm_boxplot(cpmcounts,GOI=sampset[g,1],&quot;Dark2&quot;,drug_palc,
                ylab=bquote(~italic(.(a))~log[2]~&quot;cpm &quot;))
}</code></pre>
<p><img src="figure/Knowles2019.Rmd/mitoonly-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/mitoonly-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/mitoonly-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/mitoonly-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/mitoonly-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/mitoonly-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/mitoonly-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/mitoonly-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/mitoonly-12.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/mitoonly-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/mitoonly-14.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/mitoonly-15.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="daunorubicin" class="section level4">
<h4>Daunorubicin</h4>
<pre class="r"><code>DNRdeg_sp &lt;- toplist24hr %&gt;% 
  filter(if_else(id==&quot;Daunorubicin&quot;,adj.P.Val&lt;0.01,adj.P.Val&gt;0.01 )) %&gt;% 
  filter(ENTREZID %in% DnronlyDEG) %&gt;% 
  filter(id==&quot;Daunorubicin&quot;) %&gt;% 
  dplyr::select(ENTREZID, SYMBOL)  

DNRdeg_sp %&gt;% 
  kable(., caption= &quot;DNR specific genes&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;, full_width = TRUE) %&gt;%  
  kable_styling(full_width = FALSE, font_size = 16) %&gt;% 
  scroll_box( height = &quot;500px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; ">
<table class=" lightable-paper lightable-striped table" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; margin-left: auto; margin-right: auto; font-size: 16px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
DNR specific genes
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ENTREZID
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
SYMBOL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
5742
</td>
<td style="text-align:left;">
5742
</td>
<td style="text-align:left;">
PTGS1
</td>
</tr>
<tr>
<td style="text-align:left;">
114826
</td>
<td style="text-align:left;">
114826
</td>
<td style="text-align:left;">
SMYD4
</td>
</tr>
<tr>
<td style="text-align:left;">
6117
</td>
<td style="text-align:left;">
6117
</td>
<td style="text-align:left;">
RPA1
</td>
</tr>
<tr>
<td style="text-align:left;">
54550
</td>
<td style="text-align:left;">
54550
</td>
<td style="text-align:left;">
NECAB2
</td>
</tr>
<tr>
<td style="text-align:left;">
55000
</td>
<td style="text-align:left;">
55000
</td>
<td style="text-align:left;">
TUG1
</td>
</tr>
<tr>
<td style="text-align:left;">
23150
</td>
<td style="text-align:left;">
23150
</td>
<td style="text-align:left;">
FRMD4B
</td>
</tr>
<tr>
<td style="text-align:left;">
128637
</td>
<td style="text-align:left;">
128637
</td>
<td style="text-align:left;">
TBC1D20
</td>
</tr>
<tr>
<td style="text-align:left;">
8864
</td>
<td style="text-align:left;">
8864
</td>
<td style="text-align:left;">
PER2
</td>
</tr>
<tr>
<td style="text-align:left;">
132430
</td>
<td style="text-align:left;">
132430
</td>
<td style="text-align:left;">
PABPC4L
</td>
</tr>
<tr>
<td style="text-align:left;">
54973
</td>
<td style="text-align:left;">
54973
</td>
<td style="text-align:left;">
INTS11
</td>
</tr>
<tr>
<td style="text-align:left;">
140465
</td>
<td style="text-align:left;">
140465
</td>
<td style="text-align:left;">
MYL6B
</td>
</tr>
<tr>
<td style="text-align:left;">
81628
</td>
<td style="text-align:left;">
81628
</td>
<td style="text-align:left;">
TSC22D4
</td>
</tr>
<tr>
<td style="text-align:left;">
53838
</td>
<td style="text-align:left;">
53838
</td>
<td style="text-align:left;">
C11orf24
</td>
</tr>
<tr>
<td style="text-align:left;">
387640
</td>
<td style="text-align:left;">
387640
</td>
<td style="text-align:left;">
SKIDA1
</td>
</tr>
<tr>
<td style="text-align:left;">
57654
</td>
<td style="text-align:left;">
57654
</td>
<td style="text-align:left;">
UVSSA
</td>
</tr>
<tr>
<td style="text-align:left;">
11273
</td>
<td style="text-align:left;">
11273
</td>
<td style="text-align:left;">
ATXN2L
</td>
</tr>
<tr>
<td style="text-align:left;">
56624
</td>
<td style="text-align:left;">
56624
</td>
<td style="text-align:left;">
ASAH2
</td>
</tr>
<tr>
<td style="text-align:left;">
255043
</td>
<td style="text-align:left;">
255043
</td>
<td style="text-align:left;">
TMEM86B
</td>
</tr>
<tr>
<td style="text-align:left;">
23428
</td>
<td style="text-align:left;">
23428
</td>
<td style="text-align:left;">
SLC7A8
</td>
</tr>
<tr>
<td style="text-align:left;">
4007
</td>
<td style="text-align:left;">
4007
</td>
<td style="text-align:left;">
PRICKLE3
</td>
</tr>
<tr>
<td style="text-align:left;">
5915
</td>
<td style="text-align:left;">
5915
</td>
<td style="text-align:left;">
RARB
</td>
</tr>
<tr>
<td style="text-align:left;">
79137
</td>
<td style="text-align:left;">
79137
</td>
<td style="text-align:left;">
RETREG2
</td>
</tr>
<tr>
<td style="text-align:left;">
57605
</td>
<td style="text-align:left;">
57605
</td>
<td style="text-align:left;">
PITPNM2
</td>
</tr>
<tr>
<td style="text-align:left;">
4241
</td>
<td style="text-align:left;">
4241
</td>
<td style="text-align:left;">
MELTF
</td>
</tr>
<tr>
<td style="text-align:left;">
10043
</td>
<td style="text-align:left;">
10043
</td>
<td style="text-align:left;">
TOM1
</td>
</tr>
<tr>
<td style="text-align:left;">
8766
</td>
<td style="text-align:left;">
8766
</td>
<td style="text-align:left;">
RAB11A
</td>
</tr>
<tr>
<td style="text-align:left;">
10233
</td>
<td style="text-align:left;">
10233
</td>
<td style="text-align:left;">
LRRC23
</td>
</tr>
<tr>
<td style="text-align:left;">
23140
</td>
<td style="text-align:left;">
23140
</td>
<td style="text-align:left;">
ZZEF1
</td>
</tr>
<tr>
<td style="text-align:left;">
7675
</td>
<td style="text-align:left;">
7675
</td>
<td style="text-align:left;">
ZNF121
</td>
</tr>
<tr>
<td style="text-align:left;">
83606
</td>
<td style="text-align:left;">
83606
</td>
<td style="text-align:left;">
GUCD1
</td>
</tr>
<tr>
<td style="text-align:left;">
6726
</td>
<td style="text-align:left;">
6726
</td>
<td style="text-align:left;">
SRP9
</td>
</tr>
<tr>
<td style="text-align:left;">
5026
</td>
<td style="text-align:left;">
5026
</td>
<td style="text-align:left;">
P2RX5
</td>
</tr>
<tr>
<td style="text-align:left;">
6550
</td>
<td style="text-align:left;">
6550
</td>
<td style="text-align:left;">
SLC9A3
</td>
</tr>
<tr>
<td style="text-align:left;">
10457
</td>
<td style="text-align:left;">
10457
</td>
<td style="text-align:left;">
GPNMB
</td>
</tr>
<tr>
<td style="text-align:left;">
22982
</td>
<td style="text-align:left;">
22982
</td>
<td style="text-align:left;">
DIP2C
</td>
</tr>
<tr>
<td style="text-align:left;">
1200
</td>
<td style="text-align:left;">
1200
</td>
<td style="text-align:left;">
TPP1
</td>
</tr>
<tr>
<td style="text-align:left;">
84922
</td>
<td style="text-align:left;">
84922
</td>
<td style="text-align:left;">
FIZ1
</td>
</tr>
<tr>
<td style="text-align:left;">
200014
</td>
<td style="text-align:left;">
200014
</td>
<td style="text-align:left;">
CC2D1B
</td>
</tr>
<tr>
<td style="text-align:left;">
4149
</td>
<td style="text-align:left;">
4149
</td>
<td style="text-align:left;">
MAX
</td>
</tr>
<tr>
<td style="text-align:left;">
55818
</td>
<td style="text-align:left;">
55818
</td>
<td style="text-align:left;">
KDM3A
</td>
</tr>
<tr>
<td style="text-align:left;">
3710
</td>
<td style="text-align:left;">
3710
</td>
<td style="text-align:left;">
ITPR3
</td>
</tr>
<tr>
<td style="text-align:left;">
5350
</td>
<td style="text-align:left;">
5350
</td>
<td style="text-align:left;">
PLN
</td>
</tr>
<tr>
<td style="text-align:left;">
22938
</td>
<td style="text-align:left;">
22938
</td>
<td style="text-align:left;">
SNW1
</td>
</tr>
<tr>
<td style="text-align:left;">
284612
</td>
<td style="text-align:left;">
284612
</td>
<td style="text-align:left;">
SYPL2
</td>
</tr>
<tr>
<td style="text-align:left;">
10992
</td>
<td style="text-align:left;">
10992
</td>
<td style="text-align:left;">
SF3B2
</td>
</tr>
<tr>
<td style="text-align:left;">
5463
</td>
<td style="text-align:left;">
5463
</td>
<td style="text-align:left;">
POU6F1
</td>
</tr>
<tr>
<td style="text-align:left;">
9757
</td>
<td style="text-align:left;">
9757
</td>
<td style="text-align:left;">
KMT2B
</td>
</tr>
<tr>
<td style="text-align:left;">
146880
</td>
<td style="text-align:left;">
146880
</td>
<td style="text-align:left;">
ARHGAP27P1
</td>
</tr>
<tr>
<td style="text-align:left;">
1827
</td>
<td style="text-align:left;">
1827
</td>
<td style="text-align:left;">
RCAN1
</td>
</tr>
<tr>
<td style="text-align:left;">
29763
</td>
<td style="text-align:left;">
29763
</td>
<td style="text-align:left;">
PACSIN3
</td>
</tr>
<tr>
<td style="text-align:left;">
10081
</td>
<td style="text-align:left;">
10081
</td>
<td style="text-align:left;">
PDCD7
</td>
</tr>
<tr>
<td style="text-align:left;">
56132
</td>
<td style="text-align:left;">
56132
</td>
<td style="text-align:left;">
PCDHB3
</td>
</tr>
<tr>
<td style="text-align:left;">
257144
</td>
<td style="text-align:left;">
257144
</td>
<td style="text-align:left;">
GCSAM
</td>
</tr>
<tr>
<td style="text-align:left;">
9929
</td>
<td style="text-align:left;">
9929
</td>
<td style="text-align:left;">
JOSD1
</td>
</tr>
<tr>
<td style="text-align:left;">
7052
</td>
<td style="text-align:left;">
7052
</td>
<td style="text-align:left;">
TGM2
</td>
</tr>
<tr>
<td style="text-align:left;">
11190
</td>
<td style="text-align:left;">
11190
</td>
<td style="text-align:left;">
CEP250
</td>
</tr>
<tr>
<td style="text-align:left;">
388722
</td>
<td style="text-align:left;">
388722
</td>
<td style="text-align:left;">
C1orf53
</td>
</tr>
<tr>
<td style="text-align:left;">
2188
</td>
<td style="text-align:left;">
2188
</td>
<td style="text-align:left;">
FANCF
</td>
</tr>
<tr>
<td style="text-align:left;">
375690
</td>
<td style="text-align:left;">
375690
</td>
<td style="text-align:left;">
WASH5P
</td>
</tr>
<tr>
<td style="text-align:left;">
28957
</td>
<td style="text-align:left;">
28957
</td>
<td style="text-align:left;">
MRPS28
</td>
</tr>
<tr>
<td style="text-align:left;">
644150
</td>
<td style="text-align:left;">
644150
</td>
<td style="text-align:left;">
WIPF3
</td>
</tr>
<tr>
<td style="text-align:left;">
4861
</td>
<td style="text-align:left;">
4861
</td>
<td style="text-align:left;">
NPAS1
</td>
</tr>
<tr>
<td style="text-align:left;">
55876
</td>
<td style="text-align:left;">
55876
</td>
<td style="text-align:left;">
GSDMB
</td>
</tr>
<tr>
<td style="text-align:left;">
6832
</td>
<td style="text-align:left;">
6832
</td>
<td style="text-align:left;">
SUPV3L1
</td>
</tr>
<tr>
<td style="text-align:left;">
90993
</td>
<td style="text-align:left;">
90993
</td>
<td style="text-align:left;">
CREB3L1
</td>
</tr>
<tr>
<td style="text-align:left;">
64761
</td>
<td style="text-align:left;">
64761
</td>
<td style="text-align:left;">
PARP12
</td>
</tr>
<tr>
<td style="text-align:left;">
7249
</td>
<td style="text-align:left;">
7249
</td>
<td style="text-align:left;">
TSC2
</td>
</tr>
<tr>
<td style="text-align:left;">
158431
</td>
<td style="text-align:left;">
158431
</td>
<td style="text-align:left;">
ZNF782
</td>
</tr>
<tr>
<td style="text-align:left;">
10580
</td>
<td style="text-align:left;">
10580
</td>
<td style="text-align:left;">
SORBS1
</td>
</tr>
<tr>
<td style="text-align:left;">
220869
</td>
<td style="text-align:left;">
220869
</td>
<td style="text-align:left;">
CBWD5
</td>
</tr>
<tr>
<td style="text-align:left;">
55140
</td>
<td style="text-align:left;">
55140
</td>
<td style="text-align:left;">
ELP3
</td>
</tr>
<tr>
<td style="text-align:left;">
5587
</td>
<td style="text-align:left;">
5587
</td>
<td style="text-align:left;">
PRKD1
</td>
</tr>
<tr>
<td style="text-align:left;">
80347
</td>
<td style="text-align:left;">
80347
</td>
<td style="text-align:left;">
COASY
</td>
</tr>
<tr>
<td style="text-align:left;">
6302
</td>
<td style="text-align:left;">
6302
</td>
<td style="text-align:left;">
TSPAN31
</td>
</tr>
<tr>
<td style="text-align:left;">
9564
</td>
<td style="text-align:left;">
9564
</td>
<td style="text-align:left;">
BCAR1
</td>
</tr>
<tr>
<td style="text-align:left;">
7760
</td>
<td style="text-align:left;">
7760
</td>
<td style="text-align:left;">
ZNF213
</td>
</tr>
<tr>
<td style="text-align:left;">
26267
</td>
<td style="text-align:left;">
26267
</td>
<td style="text-align:left;">
FBXO10
</td>
</tr>
<tr>
<td style="text-align:left;">
5570
</td>
<td style="text-align:left;">
5570
</td>
<td style="text-align:left;">
PKIB
</td>
</tr>
<tr>
<td style="text-align:left;">
6938
</td>
<td style="text-align:left;">
6938
</td>
<td style="text-align:left;">
TCF12
</td>
</tr>
<tr>
<td style="text-align:left;">
55132
</td>
<td style="text-align:left;">
55132
</td>
<td style="text-align:left;">
LARP1B
</td>
</tr>
<tr>
<td style="text-align:left;">
143689
</td>
<td style="text-align:left;">
143689
</td>
<td style="text-align:left;">
PIWIL4
</td>
</tr>
<tr>
<td style="text-align:left;">
6455
</td>
<td style="text-align:left;">
6455
</td>
<td style="text-align:left;">
SH3GL1
</td>
</tr>
<tr>
<td style="text-align:left;">
57610
</td>
<td style="text-align:left;">
57610
</td>
<td style="text-align:left;">
RANBP10
</td>
</tr>
<tr>
<td style="text-align:left;">
337876
</td>
<td style="text-align:left;">
337876
</td>
<td style="text-align:left;">
CHSY3
</td>
</tr>
<tr>
<td style="text-align:left;">
54520
</td>
<td style="text-align:left;">
54520
</td>
<td style="text-align:left;">
CCDC93
</td>
</tr>
<tr>
<td style="text-align:left;">
23476
</td>
<td style="text-align:left;">
23476
</td>
<td style="text-align:left;">
BRD4
</td>
</tr>
<tr>
<td style="text-align:left;">
84897
</td>
<td style="text-align:left;">
84897
</td>
<td style="text-align:left;">
TBRG1
</td>
</tr>
<tr>
<td style="text-align:left;">
81575
</td>
<td style="text-align:left;">
81575
</td>
<td style="text-align:left;">
APOLD1
</td>
</tr>
<tr>
<td style="text-align:left;">
728492
</td>
<td style="text-align:left;">
728492
</td>
<td style="text-align:left;">
SERF1B
</td>
</tr>
<tr>
<td style="text-align:left;">
285464
</td>
<td style="text-align:left;">
285464
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
960
</td>
<td style="text-align:left;">
960
</td>
<td style="text-align:left;">
CD44
</td>
</tr>
<tr>
<td style="text-align:left;">
25850
</td>
<td style="text-align:left;">
25850
</td>
<td style="text-align:left;">
ZNF345
</td>
</tr>
<tr>
<td style="text-align:left;">
6158
</td>
<td style="text-align:left;">
6158
</td>
<td style="text-align:left;">
RPL28
</td>
</tr>
<tr>
<td style="text-align:left;">
157773
</td>
<td style="text-align:left;">
157773
</td>
<td style="text-align:left;">
C8orf48
</td>
</tr>
<tr>
<td style="text-align:left;">
26156
</td>
<td style="text-align:left;">
26156
</td>
<td style="text-align:left;">
RSL1D1
</td>
</tr>
<tr>
<td style="text-align:left;">
1487
</td>
<td style="text-align:left;">
1487
</td>
<td style="text-align:left;">
CTBP1
</td>
</tr>
<tr>
<td style="text-align:left;">
23467
</td>
<td style="text-align:left;">
23467
</td>
<td style="text-align:left;">
NPTXR
</td>
</tr>
<tr>
<td style="text-align:left;">
91355
</td>
<td style="text-align:left;">
91355
</td>
<td style="text-align:left;">
LRP5L
</td>
</tr>
<tr>
<td style="text-align:left;">
26005
</td>
<td style="text-align:left;">
26005
</td>
<td style="text-align:left;">
C2CD3
</td>
</tr>
<tr>
<td style="text-align:left;">
11108
</td>
<td style="text-align:left;">
11108
</td>
<td style="text-align:left;">
PRDM4
</td>
</tr>
<tr>
<td style="text-align:left;">
25936
</td>
<td style="text-align:left;">
25936
</td>
<td style="text-align:left;">
NSL1
</td>
</tr>
<tr>
<td style="text-align:left;">
128826
</td>
<td style="text-align:left;">
128826
</td>
<td style="text-align:left;">
MIR1-1HG
</td>
</tr>
<tr>
<td style="text-align:left;">
254048
</td>
<td style="text-align:left;">
254048
</td>
<td style="text-align:left;">
UBN2
</td>
</tr>
<tr>
<td style="text-align:left;">
145165
</td>
<td style="text-align:left;">
145165
</td>
<td style="text-align:left;">
ST13P4
</td>
</tr>
<tr>
<td style="text-align:left;">
105371592
</td>
<td style="text-align:left;">
105371592
</td>
<td style="text-align:left;">
LOC105371592
</td>
</tr>
<tr>
<td style="text-align:left;">
6143
</td>
<td style="text-align:left;">
6143
</td>
<td style="text-align:left;">
RPL19
</td>
</tr>
<tr>
<td style="text-align:left;">
57139
</td>
<td style="text-align:left;">
57139
</td>
<td style="text-align:left;">
RGL3
</td>
</tr>
<tr>
<td style="text-align:left;">
6767
</td>
<td style="text-align:left;">
6767
</td>
<td style="text-align:left;">
ST13
</td>
</tr>
<tr>
<td style="text-align:left;">
6992
</td>
<td style="text-align:left;">
6992
</td>
<td style="text-align:left;">
PPP1R11
</td>
</tr>
<tr>
<td style="text-align:left;">
989
</td>
<td style="text-align:left;">
989
</td>
<td style="text-align:left;">
SEPTIN7
</td>
</tr>
<tr>
<td style="text-align:left;">
2115
</td>
<td style="text-align:left;">
2115
</td>
<td style="text-align:left;">
ETV1
</td>
</tr>
<tr>
<td style="text-align:left;">
643180
</td>
<td style="text-align:left;">
643180
</td>
<td style="text-align:left;">
CCT6P3
</td>
</tr>
<tr>
<td style="text-align:left;">
4152
</td>
<td style="text-align:left;">
4152
</td>
<td style="text-align:left;">
MBD1
</td>
</tr>
<tr>
<td style="text-align:left;">
57404
</td>
<td style="text-align:left;">
57404
</td>
<td style="text-align:left;">
CYP20A1
</td>
</tr>
<tr>
<td style="text-align:left;">
171391
</td>
<td style="text-align:left;">
171391
</td>
<td style="text-align:left;">
GATD1-DT
</td>
</tr>
<tr>
<td style="text-align:left;">
6612
</td>
<td style="text-align:left;">
6612
</td>
<td style="text-align:left;">
SUMO3
</td>
</tr>
<tr>
<td style="text-align:left;">
642799
</td>
<td style="text-align:left;">
642799
</td>
<td style="text-align:left;">
NPIPA2
</td>
</tr>
<tr>
<td style="text-align:left;">
8934
</td>
<td style="text-align:left;">
8934
</td>
<td style="text-align:left;">
RAB29
</td>
</tr>
<tr>
<td style="text-align:left;">
6722
</td>
<td style="text-align:left;">
6722
</td>
<td style="text-align:left;">
SRF
</td>
</tr>
<tr>
<td style="text-align:left;">
10628
</td>
<td style="text-align:left;">
10628
</td>
<td style="text-align:left;">
TXNIP
</td>
</tr>
<tr>
<td style="text-align:left;">
79894
</td>
<td style="text-align:left;">
79894
</td>
<td style="text-align:left;">
ZNF672
</td>
</tr>
<tr>
<td style="text-align:left;">
118987
</td>
<td style="text-align:left;">
118987
</td>
<td style="text-align:left;">
PDZD8
</td>
</tr>
<tr>
<td style="text-align:left;">
157697
</td>
<td style="text-align:left;">
157697
</td>
<td style="text-align:left;">
ERICH1
</td>
</tr>
<tr>
<td style="text-align:left;">
81502
</td>
<td style="text-align:left;">
81502
</td>
<td style="text-align:left;">
HM13
</td>
</tr>
<tr>
<td style="text-align:left;">
55787
</td>
<td style="text-align:left;">
55787
</td>
<td style="text-align:left;">
TXLNG
</td>
</tr>
<tr>
<td style="text-align:left;">
11135
</td>
<td style="text-align:left;">
11135
</td>
<td style="text-align:left;">
CDC42EP1
</td>
</tr>
<tr>
<td style="text-align:left;">
5493
</td>
<td style="text-align:left;">
5493
</td>
<td style="text-align:left;">
PPL
</td>
</tr>
<tr>
<td style="text-align:left;">
55102
</td>
<td style="text-align:left;">
55102
</td>
<td style="text-align:left;">
ATG2B
</td>
</tr>
<tr>
<td style="text-align:left;">
3925
</td>
<td style="text-align:left;">
3925
</td>
<td style="text-align:left;">
STMN1
</td>
</tr>
<tr>
<td style="text-align:left;">
147700
</td>
<td style="text-align:left;">
147700
</td>
<td style="text-align:left;">
KLC3
</td>
</tr>
<tr>
<td style="text-align:left;">
10475
</td>
<td style="text-align:left;">
10475
</td>
<td style="text-align:left;">
TRIM38
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>toplist24hr %&gt;% 
  group_by(time,id) %&gt;% 
  filter(ENTREZID %in% DNRdeg_sp$ENTREZID) %&gt;% 
  filter(adj.P.Val &lt;0.05) %&gt;% 
  # filter(id==&quot;Daunorubicin&quot;)
  mutate(logFC=logFC*(-1)) %&gt;% 
  ggplot(., aes(x= id, y=logFC))+
  geom_boxplot(aes(fill=id))+
  theme_classic()+
  fill_palette(palette = drug_palNoVeh)+
  ggtitle(&quot;LogFC of n=131 DNR specific DEGs&quot;)+
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5,face = &quot;bold&quot;),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(size = 1.5),
    axis.text = element_text(size = 8, color = &quot;black&quot;, angle = 20))</code></pre>
<p><img src="figure/Knowles2019.Rmd/dnr%20only-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>    # strip.text.x = element_text(size = 12, color = &quot;black&quot;, face = &quot;italic&quot;))

toplist24hr %&gt;% 
  filter(if_else(id==&quot;Daunorubicin&quot;,adj.P.Val&lt;0.01,adj.P.Val&gt;0.01 )) %&gt;% 
  group_by(time,id) %&gt;% 
  filter(ENTREZID %in% DnronlyDEG) %&gt;% 
  filter(adj.P.Val &lt;0.05) %&gt;% 
  ggplot(., aes(x=adj.P.Val))+
  geom_histogram(aes(fill=id))+
  geom_vline(xintercept=0.01,linetype=2)+
    # geom_density(aes(fill=id))+
    facet_wrap(~id)+
  ggtitle(&quot;DNR adj p. value &lt;0.05&quot;)+
    fill_palette(palette = drug_palNoVeh)+
  theme_bw()</code></pre>
<p><img src="figure/Knowles2019.Rmd/dnr%20only-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>toplist24hr %&gt;% 
  filter(ENTREZID %in% DnronlyDEG) %&gt;% 
  # filter(adj.P.Val&lt;0.05) %&gt;%
  ggplot(., aes(x=adj.P.Val))+
  geom_density(aes(fill=id, alpha= 0.8))+
  fill_palette(palette = drug_palNoVeh)+
  theme_bw()</code></pre>
<p><img src="figure/Knowles2019.Rmd/dnr%20only-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>set.seed(12345)
sampset &lt;- DNRdeg_sp %&gt;% 
  sample_n(.,12)

for (g in seq(from=1, to=length(sampset$ENTREZID))){
  a &lt;- sampset$SYMBOL[g]
  cpm_boxplot(cpmcounts,GOI=sampset[g,1],&quot;Dark2&quot;,drug_palc,
                ylab=bquote(~italic(.(a))~log[2]~&quot;cpm &quot;))
}</code></pre>
<p><img src="figure/Knowles2019.Rmd/dnr%20only-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/dnr%20only-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/dnr%20only-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/dnr%20only-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/dnr%20only-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/dnr%20only-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/dnr%20only-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/dnr%20only-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/dnr%20only-12.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/dnr%20only-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/dnr%20only-14.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/dnr%20only-15.png" width="672" style="display: block; margin: auto;" />
#### Epirubicin</p>
<pre class="r"><code>EPIdeg_sp &lt;- toplist24hr %&gt;% 
  filter(if_else(id==&quot;Epirubicin&quot;,adj.P.Val&lt;0.01,adj.P.Val&gt;0.01 )) %&gt;% 
  filter(ENTREZID %in% EpionlyDEG) %&gt;% 
  filter(id==&quot;Epirubicin&quot;) %&gt;% 
  dplyr::select(ENTREZID, SYMBOL)  

EPIdeg_sp %&gt;% 
  kable(., caption= &quot;EPI specific genes&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;, full_width = TRUE) %&gt;%  
  kable_styling(full_width = FALSE, font_size = 16) %&gt;% 
  scroll_box( height = &quot;500px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; ">
<table class=" lightable-paper lightable-striped table" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; margin-left: auto; margin-right: auto; font-size: 16px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
EPI specific genes
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ENTREZID
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
SYMBOL
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
79798
</td>
<td style="text-align:left;">
79798
</td>
<td style="text-align:left;">
ARMC5
</td>
</tr>
<tr>
<td style="text-align:left;">
9685
</td>
<td style="text-align:left;">
9685
</td>
<td style="text-align:left;">
CLINT1
</td>
</tr>
<tr>
<td style="text-align:left;">
55006
</td>
<td style="text-align:left;">
55006
</td>
<td style="text-align:left;">
TRMT61B
</td>
</tr>
<tr>
<td style="text-align:left;">
9373
</td>
<td style="text-align:left;">
9373
</td>
<td style="text-align:left;">
PLAA
</td>
</tr>
<tr>
<td style="text-align:left;">
54508
</td>
<td style="text-align:left;">
54508
</td>
<td style="text-align:left;">
EPB41L4A-DT
</td>
</tr>
<tr>
<td style="text-align:left;">
55702
</td>
<td style="text-align:left;">
55702
</td>
<td style="text-align:left;">
YJU2
</td>
</tr>
<tr>
<td style="text-align:left;">
79759
</td>
<td style="text-align:left;">
79759
</td>
<td style="text-align:left;">
ZNF668
</td>
</tr>
<tr>
<td style="text-align:left;">
64781
</td>
<td style="text-align:left;">
64781
</td>
<td style="text-align:left;">
CERK
</td>
</tr>
<tr>
<td style="text-align:left;">
57587
</td>
<td style="text-align:left;">
57587
</td>
<td style="text-align:left;">
CFAP97
</td>
</tr>
<tr>
<td style="text-align:left;">
23086
</td>
<td style="text-align:left;">
23086
</td>
<td style="text-align:left;">
EXPH5
</td>
</tr>
<tr>
<td style="text-align:left;">
90864
</td>
<td style="text-align:left;">
90864
</td>
<td style="text-align:left;">
SPSB3
</td>
</tr>
<tr>
<td style="text-align:left;">
11097
</td>
<td style="text-align:left;">
11097
</td>
<td style="text-align:left;">
NUP42
</td>
</tr>
<tr>
<td style="text-align:left;">
64863
</td>
<td style="text-align:left;">
64863
</td>
<td style="text-align:left;">
METTL4
</td>
</tr>
<tr>
<td style="text-align:left;">
10342
</td>
<td style="text-align:left;">
10342
</td>
<td style="text-align:left;">
TFG
</td>
</tr>
<tr>
<td style="text-align:left;">
51132
</td>
<td style="text-align:left;">
51132
</td>
<td style="text-align:left;">
RLIM
</td>
</tr>
<tr>
<td style="text-align:left;">
57325
</td>
<td style="text-align:left;">
57325
</td>
<td style="text-align:left;">
KAT14
</td>
</tr>
<tr>
<td style="text-align:left;">
285636
</td>
<td style="text-align:left;">
285636
</td>
<td style="text-align:left;">
RIMOC1
</td>
</tr>
<tr>
<td style="text-align:left;">
220963
</td>
<td style="text-align:left;">
220963
</td>
<td style="text-align:left;">
SLC16A9
</td>
</tr>
<tr>
<td style="text-align:left;">
7110
</td>
<td style="text-align:left;">
7110
</td>
<td style="text-align:left;">
TMF1
</td>
</tr>
<tr>
<td style="text-align:left;">
51434
</td>
<td style="text-align:left;">
51434
</td>
<td style="text-align:left;">
ANAPC7
</td>
</tr>
<tr>
<td style="text-align:left;">
55105
</td>
<td style="text-align:left;">
55105
</td>
<td style="text-align:left;">
GPATCH2
</td>
</tr>
<tr>
<td style="text-align:left;">
92140
</td>
<td style="text-align:left;">
92140
</td>
<td style="text-align:left;">
MTDH
</td>
</tr>
<tr>
<td style="text-align:left;">
1938
</td>
<td style="text-align:left;">
1938
</td>
<td style="text-align:left;">
EEF2
</td>
</tr>
<tr>
<td style="text-align:left;">
56252
</td>
<td style="text-align:left;">
56252
</td>
<td style="text-align:left;">
YLPM1
</td>
</tr>
<tr>
<td style="text-align:left;">
7592
</td>
<td style="text-align:left;">
7592
</td>
<td style="text-align:left;">
ZNF41
</td>
</tr>
<tr>
<td style="text-align:left;">
79038
</td>
<td style="text-align:left;">
79038
</td>
<td style="text-align:left;">
ZFYVE21
</td>
</tr>
<tr>
<td style="text-align:left;">
11232
</td>
<td style="text-align:left;">
11232
</td>
<td style="text-align:left;">
POLG2
</td>
</tr>
<tr>
<td style="text-align:left;">
339210
</td>
<td style="text-align:left;">
339210
</td>
<td style="text-align:left;">
C17orf67
</td>
</tr>
<tr>
<td style="text-align:left;">
5884
</td>
<td style="text-align:left;">
5884
</td>
<td style="text-align:left;">
RAD17
</td>
</tr>
<tr>
<td style="text-align:left;">
64860
</td>
<td style="text-align:left;">
64860
</td>
<td style="text-align:left;">
ARMCX5
</td>
</tr>
<tr>
<td style="text-align:left;">
493812
</td>
<td style="text-align:left;">
493812
</td>
<td style="text-align:left;">
HCG11
</td>
</tr>
<tr>
<td style="text-align:left;">
105371932
</td>
<td style="text-align:left;">
105371932
</td>
<td style="text-align:left;">
LOC105371932
</td>
</tr>
<tr>
<td style="text-align:left;">
130507
</td>
<td style="text-align:left;">
130507
</td>
<td style="text-align:left;">
UBR3
</td>
</tr>
<tr>
<td style="text-align:left;">
10111
</td>
<td style="text-align:left;">
10111
</td>
<td style="text-align:left;">
RAD50
</td>
</tr>
<tr>
<td style="text-align:left;">
26135
</td>
<td style="text-align:left;">
26135
</td>
<td style="text-align:left;">
SERBP1
</td>
</tr>
<tr>
<td style="text-align:left;">
27246
</td>
<td style="text-align:left;">
27246
</td>
<td style="text-align:left;">
RNF115
</td>
</tr>
<tr>
<td style="text-align:left;">
57609
</td>
<td style="text-align:left;">
57609
</td>
<td style="text-align:left;">
DIP2B
</td>
</tr>
<tr>
<td style="text-align:left;">
85457
</td>
<td style="text-align:left;">
85457
</td>
<td style="text-align:left;">
CIPC
</td>
</tr>
<tr>
<td style="text-align:left;">
60561
</td>
<td style="text-align:left;">
60561
</td>
<td style="text-align:left;">
RINT1
</td>
</tr>
<tr>
<td style="text-align:left;">
339122
</td>
<td style="text-align:left;">
339122
</td>
<td style="text-align:left;">
RAB43
</td>
</tr>
<tr>
<td style="text-align:left;">
54942
</td>
<td style="text-align:left;">
54942
</td>
<td style="text-align:left;">
ABITRAM
</td>
</tr>
<tr>
<td style="text-align:left;">
5533
</td>
<td style="text-align:left;">
5533
</td>
<td style="text-align:left;">
PPP3CC
</td>
</tr>
<tr>
<td style="text-align:left;">
10021
</td>
<td style="text-align:left;">
10021
</td>
<td style="text-align:left;">
HCN4
</td>
</tr>
<tr>
<td style="text-align:left;">
23122
</td>
<td style="text-align:left;">
23122
</td>
<td style="text-align:left;">
CLASP2
</td>
</tr>
<tr>
<td style="text-align:left;">
84967
</td>
<td style="text-align:left;">
84967
</td>
<td style="text-align:left;">
LSM10
</td>
</tr>
<tr>
<td style="text-align:left;">
340359
</td>
<td style="text-align:left;">
340359
</td>
<td style="text-align:left;">
KLHL38
</td>
</tr>
<tr>
<td style="text-align:left;">
79713
</td>
<td style="text-align:left;">
79713
</td>
<td style="text-align:left;">
IGFLR1
</td>
</tr>
<tr>
<td style="text-align:left;">
83852
</td>
<td style="text-align:left;">
83852
</td>
<td style="text-align:left;">
SETDB2
</td>
</tr>
<tr>
<td style="text-align:left;">
55937
</td>
<td style="text-align:left;">
55937
</td>
<td style="text-align:left;">
APOM
</td>
</tr>
<tr>
<td style="text-align:left;">
2043
</td>
<td style="text-align:left;">
2043
</td>
<td style="text-align:left;">
EPHA4
</td>
</tr>
<tr>
<td style="text-align:left;">
22796
</td>
<td style="text-align:left;">
22796
</td>
<td style="text-align:left;">
COG2
</td>
</tr>
<tr>
<td style="text-align:left;">
1032
</td>
<td style="text-align:left;">
1032
</td>
<td style="text-align:left;">
CDKN2D
</td>
</tr>
<tr>
<td style="text-align:left;">
8624
</td>
<td style="text-align:left;">
8624
</td>
<td style="text-align:left;">
PSMG1
</td>
</tr>
<tr>
<td style="text-align:left;">
26046
</td>
<td style="text-align:left;">
26046
</td>
<td style="text-align:left;">
LTN1
</td>
</tr>
<tr>
<td style="text-align:left;">
64844
</td>
<td style="text-align:left;">
64844
</td>
<td style="text-align:left;">
MARCHF7
</td>
</tr>
<tr>
<td style="text-align:left;">
1385
</td>
<td style="text-align:left;">
1385
</td>
<td style="text-align:left;">
CREB1
</td>
</tr>
<tr>
<td style="text-align:left;">
84541
</td>
<td style="text-align:left;">
84541
</td>
<td style="text-align:left;">
KBTBD8
</td>
</tr>
<tr>
<td style="text-align:left;">
55676
</td>
<td style="text-align:left;">
55676
</td>
<td style="text-align:left;">
SLC30A6
</td>
</tr>
<tr>
<td style="text-align:left;">
79797
</td>
<td style="text-align:left;">
79797
</td>
<td style="text-align:left;">
ZNF408
</td>
</tr>
<tr>
<td style="text-align:left;">
143282
</td>
<td style="text-align:left;">
143282
</td>
<td style="text-align:left;">
FGFBP3
</td>
</tr>
<tr>
<td style="text-align:left;">
55339
</td>
<td style="text-align:left;">
55339
</td>
<td style="text-align:left;">
WDR33
</td>
</tr>
<tr>
<td style="text-align:left;">
79230
</td>
<td style="text-align:left;">
79230
</td>
<td style="text-align:left;">
ZNF557
</td>
</tr>
<tr>
<td style="text-align:left;">
55223
</td>
<td style="text-align:left;">
55223
</td>
<td style="text-align:left;">
TRIM62
</td>
</tr>
<tr>
<td style="text-align:left;">
23466
</td>
<td style="text-align:left;">
23466
</td>
<td style="text-align:left;">
CBX6
</td>
</tr>
<tr>
<td style="text-align:left;">
201627
</td>
<td style="text-align:left;">
201627
</td>
<td style="text-align:left;">
DENND6A
</td>
</tr>
<tr>
<td style="text-align:left;">
284232
</td>
<td style="text-align:left;">
284232
</td>
<td style="text-align:left;">
ANKRD20A9P
</td>
</tr>
<tr>
<td style="text-align:left;">
7869
</td>
<td style="text-align:left;">
7869
</td>
<td style="text-align:left;">
SEMA3B
</td>
</tr>
<tr>
<td style="text-align:left;">
9972
</td>
<td style="text-align:left;">
9972
</td>
<td style="text-align:left;">
NUP153
</td>
</tr>
<tr>
<td style="text-align:left;">
10240
</td>
<td style="text-align:left;">
10240
</td>
<td style="text-align:left;">
MRPS31
</td>
</tr>
<tr>
<td style="text-align:left;">
63943
</td>
<td style="text-align:left;">
63943
</td>
<td style="text-align:left;">
FKBPL
</td>
</tr>
<tr>
<td style="text-align:left;">
55727
</td>
<td style="text-align:left;">
55727
</td>
<td style="text-align:left;">
BTBD7
</td>
</tr>
<tr>
<td style="text-align:left;">
79657
</td>
<td style="text-align:left;">
79657
</td>
<td style="text-align:left;">
RPAP3
</td>
</tr>
<tr>
<td style="text-align:left;">
23258
</td>
<td style="text-align:left;">
23258
</td>
<td style="text-align:left;">
DENND5A
</td>
</tr>
<tr>
<td style="text-align:left;">
84859
</td>
<td style="text-align:left;">
84859
</td>
<td style="text-align:left;">
LRCH3
</td>
</tr>
<tr>
<td style="text-align:left;">
51507
</td>
<td style="text-align:left;">
51507
</td>
<td style="text-align:left;">
RTF2
</td>
</tr>
<tr>
<td style="text-align:left;">
3665
</td>
<td style="text-align:left;">
3665
</td>
<td style="text-align:left;">
IRF7
</td>
</tr>
<tr>
<td style="text-align:left;">
138241
</td>
<td style="text-align:left;">
138241
</td>
<td style="text-align:left;">
C9orf85
</td>
</tr>
<tr>
<td style="text-align:left;">
727851
</td>
<td style="text-align:left;">
727851
</td>
<td style="text-align:left;">
RGPD8
</td>
</tr>
<tr>
<td style="text-align:left;">
140456
</td>
<td style="text-align:left;">
140456
</td>
<td style="text-align:left;">
ASB11
</td>
</tr>
<tr>
<td style="text-align:left;">
349136
</td>
<td style="text-align:left;">
349136
</td>
<td style="text-align:left;">
WDR86
</td>
</tr>
<tr>
<td style="text-align:left;">
2966
</td>
<td style="text-align:left;">
2966
</td>
<td style="text-align:left;">
GTF2H2
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>toplist24hr %&gt;% 
  group_by(time,id) %&gt;% 
  filter(ENTREZID %in% EPIdeg_sp$ENTREZID) %&gt;% 
  filter(adj.P.Val &lt;0.05) %&gt;% 
  # filter(id==&quot;Epirubicin&quot;)
  mutate(logFC=logFC*(-1)) %&gt;% 
  ggplot(., aes(x= id, y=logFC))+
  geom_boxplot(aes(fill=id))+
  theme_classic()+
  fill_palette(palette = drug_palNoVeh)+
  ggtitle(&quot;LogFC of n=81 EPI specific DEGs&quot;)+
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5,face = &quot;bold&quot;),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(size = 1.5),
    axis.text = element_text(size = 8, color = &quot;black&quot;, angle = 20))</code></pre>
<p><img src="figure/Knowles2019.Rmd/epi%20only-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>    # strip.text.x = element_text(size = 12, color = &quot;black&quot;, face = &quot;italic&quot;))

toplist24hr %&gt;% 
  filter(if_else(id==&quot;Epirubicin&quot;,adj.P.Val&lt;0.01,adj.P.Val&gt;0.01 )) %&gt;% 
  group_by(time,id) %&gt;% 
  filter(ENTREZID %in% EpionlyDEG) %&gt;% 
  filter(adj.P.Val &lt;0.05) %&gt;% 
  ggplot(., aes(x=adj.P.Val))+
  geom_histogram(aes(fill=id))+
  geom_vline(xintercept=0.01,linetype=2)+
    # geom_density(aes(fill=id))+
    facet_wrap(~id)+
  ggtitle(&quot;EPI adj p. value &lt;0.05&quot;)+
    fill_palette(palette = drug_palNoVeh)+
  theme_bw()</code></pre>
<p><img src="figure/Knowles2019.Rmd/epi%20only-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>toplist24hr %&gt;% 
  filter(ENTREZID %in% EpionlyDEG) %&gt;% 
  # filter(adj.P.Val&lt;0.05) %&gt;%
  ggplot(., aes(x=adj.P.Val))+
  geom_density(aes(fill=id, alpha= 0.8))+
  fill_palette(palette = drug_palNoVeh)+
  theme_bw()</code></pre>
<p><img src="figure/Knowles2019.Rmd/epi%20only-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>set.seed(12345)
sampset &lt;- EPIdeg_sp %&gt;% 
  sample_n(.,12)

for (g in seq(from=1, to=length(sampset$ENTREZID))){
  a &lt;- sampset$SYMBOL[g]
  cpm_boxplot(cpmcounts,GOI=sampset[g,1],&quot;Dark2&quot;,drug_palc,
                ylab=bquote(~italic(.(a))~log[2]~&quot;cpm &quot;))
}</code></pre>
<p><img src="figure/Knowles2019.Rmd/epi%20only-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/epi%20only-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/epi%20only-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/epi%20only-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/epi%20only-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/epi%20only-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/epi%20only-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/epi%20only-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/epi%20only-12.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/epi%20only-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/epi%20only-14.png" width="672" style="display: block; margin: auto;" /><img src="figure/Knowles2019.Rmd/epi%20only-15.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ComplexHeatmap_2.12.1 broom_1.0.5           kableExtra_1.3.4     
 [4] sjmisc_2.8.9          scales_1.2.1          ggpubr_0.6.0         
 [7] cowplot_1.1.1         RColorBrewer_1.1-3    biomaRt_2.52.0       
[10] ggsignif_0.6.4        lubridate_1.9.2       forcats_1.0.0        
[13] stringr_1.5.0         dplyr_1.1.2           purrr_1.0.1          
[16] readr_2.1.4           tidyr_1.3.0           tibble_3.2.1         
[19] ggplot2_3.4.2         tidyverse_2.0.0       limma_3.52.4         
[22] workflowr_1.7.0      

loaded via a namespace (and not attached):
  [1] backports_1.4.1        circlize_0.4.15        BiocFileCache_2.4.0   
  [4] systemfonts_1.0.4      GenomeInfoDb_1.32.4    digest_0.6.31         
  [7] foreach_1.5.2          htmltools_0.5.5        fansi_1.0.4           
 [10] magrittr_2.0.3         memoise_2.0.1          cluster_2.1.4         
 [13] doParallel_1.0.17      tzdb_0.4.0             Biostrings_2.64.1     
 [16] matrixStats_1.0.0      svglite_2.1.1          timechange_0.2.0      
 [19] prettyunits_1.1.1      RVenn_1.1.0            colorspace_2.1-0      
 [22] blob_1.2.4             rvest_1.0.3            rappdirs_0.3.3        
 [25] xfun_0.39              callr_3.7.3            crayon_1.5.2          
 [28] RCurl_1.98-1.12        jsonlite_1.8.5         iterators_1.0.14      
 [31] glue_1.6.2             gtable_0.3.3           zlibbioc_1.42.0       
 [34] XVector_0.36.0         webshot_0.5.4          GetoptLong_1.0.5      
 [37] car_3.1-2              shape_1.4.6            BiocGenerics_0.42.0   
 [40] abind_1.4-5            futile.options_1.0.1   DBI_1.1.3             
 [43] rstatix_0.7.2          Rcpp_1.0.10            viridisLite_0.4.2     
 [46] progress_1.2.2         units_0.8-2            clue_0.3-64           
 [49] proxy_0.4-27           bit_4.0.5              stats4_4.2.2          
 [52] httr_1.4.6             pkgconfig_2.0.3        XML_3.99-0.14         
 [55] farver_2.1.1           sass_0.4.6             dbplyr_2.3.2          
 [58] utf8_1.2.3             tidyselect_1.2.0       labeling_0.4.2        
 [61] rlang_1.1.1            later_1.3.1            AnnotationDbi_1.58.0  
 [64] munsell_0.5.0          tools_4.2.2            cachem_1.0.8          
 [67] cli_3.6.1              generics_0.1.3         RSQLite_2.3.1         
 [70] ggVennDiagram_1.2.2    sjlabelled_1.2.0       evaluate_0.21         
 [73] fastmap_1.1.1          yaml_2.3.7             processx_3.8.1        
 [76] knitr_1.43             bit64_4.0.5            fs_1.6.2              
 [79] KEGGREST_1.36.3        whisker_0.4.1          formatR_1.14          
 [82] xml2_1.3.4             compiler_4.2.2         rstudioapi_0.14       
 [85] filelock_1.0.2         curl_5.0.1             png_0.1-8             
 [88] e1071_1.7-13           bslib_0.5.0            stringi_1.7.12        
 [91] highr_0.10             ps_1.7.5               futile.logger_1.4.3   
 [94] classInt_0.4-9         vctrs_0.6.3            pillar_1.9.0          
 [97] lifecycle_1.0.3        jquerylib_0.1.4        GlobalOptions_0.1.2   
[100] bitops_1.0-7           insight_0.19.2         httpuv_1.6.11         
[103] R6_2.5.1               promises_1.2.0.1       KernSmooth_2.23-21    
[106] IRanges_2.30.1         codetools_0.2-19       lambda.r_1.2.4        
[109] rprojroot_2.0.3        rjson_0.2.21           withr_2.5.0           
[112] S4Vectors_0.34.0       GenomeInfoDbData_1.2.8 parallel_4.2.2        
[115] hms_1.1.3              VennDiagram_1.7.3      class_7.3-22          
[118] rmarkdown_2.22         carData_3.0-5          git2r_0.32.0          
[121] sf_1.0-13              getPass_0.2-2          Biobase_2.56.0        </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
