<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="ERM" />

<meta name="date" content="2023-06-29" />

<title>Comparisons with other data sets</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cardiotoxicity</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/reneeisnowhere/Cardiotoxicity">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Comparisons with other data sets</h1>
<h4 class="author">ERM</h4>
<h4 class="date">2023-06-29</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-06-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Cardiotoxicity/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230109code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230109)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230109code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230109)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereCardiotoxicitytree112c968fc7afa4c67de7d7d353eb533d4892b32etargetblank112c968a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/tree/112c968fc7afa4c67de7d7d353eb533d4892b32e" target="_blank">112c968</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereCardiotoxicitytree112c968fc7afa4c67de7d7d353eb533d4892b32etargetblank112c968a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/tree/112c968fc7afa4c67de7d7d353eb533d4892b32e" target="_blank">112c968</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/41588_2018_171_MOESM3_ESMeQTL_ST2_for paper.csv
    Ignored:    data/Arr_GWAS.txt
    Ignored:    data/Arr_geneset.RDS
    Ignored:    data/BC_cell_lines.csv
    Ignored:    data/CADGWASgene_table.csv
    Ignored:    data/CAD_geneset.RDS
    Ignored:    data/Clamp_Summary.csv
    Ignored:    data/Cormotif_24_k1-5_raw.RDS
    Ignored:    data/DAgostres24.RDS
    Ignored:    data/DAtable1.csv
    Ignored:    data/DDEMresp_list.csv
    Ignored:    data/DDE_reQTL.txt
    Ignored:    data/DDEresp_list.csv
    Ignored:    data/DEG-GO/
    Ignored:    data/DEG_cormotif.RDS
    Ignored:    data/DF_Plate_Peak.csv
    Ignored:    data/DRC48hoursdata.csv
    Ignored:    data/Da24counts.txt
    Ignored:    data/Dx24counts.txt
    Ignored:    data/Dx_reQTL_specific.txt
    Ignored:    data/Ep24counts.txt
    Ignored:    data/GOIsig.csv
    Ignored:    data/GOplots.R
    Ignored:    data/GTEX_setsimple.csv
    Ignored:    data/GTEX_sig24.RDS
    Ignored:    data/GTEx_gene_list.csv
    Ignored:    data/HFGWASgene_table.csv
    Ignored:    data/HF_geneset.RDS
    Ignored:    data/Heart_Left_Ventricle.v8.egenes.txt
    Ignored:    data/Hf_GWAS.txt
    Ignored:    data/K_cluster
    Ignored:    data/K_cluster_kisthree.csv
    Ignored:    data/K_cluster_kistwo.csv
    Ignored:    data/LD50_05via.csv
    Ignored:    data/LDH48hoursdata.csv
    Ignored:    data/Mt24counts.txt
    Ignored:    data/RINsamplelist.txt
    Ignored:    data/Seonane2019supp1.txt
    Ignored:    data/TMMnormed_x.RDS
    Ignored:    data/TOP2Bi-24hoursGO_analysis.csv
    Ignored:    data/TR24counts.txt
    Ignored:    data/Top2biresp_cluster24h.csv
    Ignored:    data/Viabilitylistfull.csv
    Ignored:    data/allexpressedgenes.txt
    Ignored:    data/allgenes.txt
    Ignored:    data/allmatrix.RDS
    Ignored:    data/annotation_data_frame.RDS
    Ignored:    data/averageviabilitytable.RDS
    Ignored:    data/avgLD50.RDS
    Ignored:    data/backGL.txt
    Ignored:    data/cormotif_3hk1-8.RDS
    Ignored:    data/cormotif_initalK5.RDS
    Ignored:    data/cormotif_initialK5.RDS
    Ignored:    data/cormotif_initialall.RDS
    Ignored:    data/counts24hours.RDS
    Ignored:    data/cpmcount.RDS
    Ignored:    data/cpmnorm_counts.csv
    Ignored:    data/crispr_genes.csv
    Ignored:    data/ctnnt_results.txt
    Ignored:    data/cvd_GWAS.txt
    Ignored:    data/dat_cpm.RDS
    Ignored:    data/data_outline.txt
    Ignored:    data/efit2.RDS
    Ignored:    data/efit2_final.RDS
    Ignored:    data/efit2results.RDS
    Ignored:    data/ensembl_backup.RDS
    Ignored:    data/ensgtotal.txt
    Ignored:    data/filcpm_counts.RDS
    Ignored:    data/filenameonly.txt
    Ignored:    data/filtered_cpm_counts.csv
    Ignored:    data/filtered_raw_counts.csv
    Ignored:    data/filtermatrix_x.RDS
    Ignored:    data/folder_05top/
    Ignored:    data/geneDoxonlyQTL.csv
    Ignored:    data/gene_corr_df.RDS
    Ignored:    data/gene_corr_frame.RDS
    Ignored:    data/gene_prob_tran3h.RDS
    Ignored:    data/gene_probabilityk5.RDS
    Ignored:    data/gostresTop2bi_ER.RDS
    Ignored:    data/gostresTop2bi_LR
    Ignored:    data/gostresTop2bi_LR.RDS
    Ignored:    data/gostresTop2bi_TI.RDS
    Ignored:    data/gostrescoNR
    Ignored:    data/gtex/
    Ignored:    data/heartgenes.csv
    Ignored:    data/individualDRCfile.RDS
    Ignored:    data/individual_DRC48.RDS
    Ignored:    data/individual_LDH48.RDS
    Ignored:    data/kegglistDEG.RDS
    Ignored:    data/knowfig4.csv
    Ignored:    data/knowfig5.csv
    Ignored:    data/label_list.RDS
    Ignored:    data/ld50_table.csv
    Ignored:    data/mymatrix.RDS
    Ignored:    data/nonresponse_cluster24h.csv
    Ignored:    data/norm_LDH.csv
    Ignored:    data/norm_counts.csv
    Ignored:    data/old_sets/
    Ignored:    data/plan2plot.png
    Ignored:    data/raw_counts.csv
    Ignored:    data/response_cluster24h.csv
    Ignored:    data/sigVDA24.txt
    Ignored:    data/sigVDA3.txt
    Ignored:    data/sigVDX24.txt
    Ignored:    data/sigVDX3.txt
    Ignored:    data/sigVEP24.txt
    Ignored:    data/sigVEP3.txt
    Ignored:    data/sigVMT24.txt
    Ignored:    data/sigVMT3.txt
    Ignored:    data/sigVTR24.txt
    Ignored:    data/sigVTR3.txt
    Ignored:    data/siglist.RDS
    Ignored:    data/siglist_final.RDS
    Ignored:    data/siglist_old.RDS
    Ignored:    data/slope_table.csv
    Ignored:    data/supp_normLDH48.RDS
    Ignored:    data/supp_pca_all_anno.RDS
    Ignored:    data/table3a.omar
    Ignored:    data/toplistall.RDS
    Ignored:    data/tvl24hour.txt
    Ignored:    data/tvl24hourw.txt
    Ignored:    data/venn_code.R
    Ignored:    data/viability.RDS

Untracked files:
    Untracked:  .RDataTmp
    Untracked:  .RDataTmp1
    Untracked:  .RDataTmp2
    Untracked:  OmicNavigator_learn.R
    Untracked:  analysis/Figure9.Rmd
    Untracked:  code/DRC_plotfigure1.png
    Untracked:  code/cpm_boxplot.R
    Untracked:  code/extracting_ggplot_data.R
    Untracked:  code/fig1plot.png
    Untracked:  code/figurelegeddrc.png
    Untracked:  cormotif_probability_genelist.csv
    Untracked:  individual-legenddark2.png
    Untracked:  installed_old.rda
    Untracked:  motif_ER.txt
    Untracked:  motif_LR.txt
    Untracked:  motif_NR.txt
    Untracked:  motif_TI.txt
    Untracked:  output/DNRvenn.RDS
    Untracked:  output/DOXvenn.RDS
    Untracked:  output/EPIvenn.RDS
    Untracked:  output/MTXvenn.RDS
    Untracked:  output/Volcanoplot_10
    Untracked:  output/Volcanoplot_10.RDS
    Untracked:  output/allfinal_sup10.RDS
    Untracked:  output/motif_ERrep.RDS
    Untracked:  output/motif_LRrep.RDS
    Untracked:  output/motif_NRrep.RDS
    Untracked:  output/motif_TI_rep.RDS
    Untracked:  output/output-old/
    Untracked:  output/supplementary_motif_list_GO.RDS
    Untracked:  output/toptablebydrug.RDS
    Untracked:  output/x_counts.RDS
    Untracked:  reneebasecode.R

Unstaged changes:
    Modified:   analysis/Cormotifcluster_analysis.Rmd
    Modified:   analysis/Figure1.Rmd
    Modified:   analysis/GTEx_genes.Rmd
    Modified:   analysis/Supplementary_figures.Rmd
    Modified:   analysis/run_all_analysis.Rmd
    Modified:   code/24hour_RNAeval.R
    Modified:   output/toplistall.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/other_analysis.Rmd</code>) and
HTML (<code>docs/other_analysis.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/112c968fc7afa4c67de7d7d353eb533d4892b32e/analysis/other_analysis.Rmd" target="_blank">112c968</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-29
</td>
<td>
with final data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/6e4c86749c984702be1a0040522ac54b71d22875/docs/other_analysis.html" target="_blank">6e4c867</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/028f18d407e4a6736284c99e824f8a959cb7825e/analysis/other_analysis.Rmd" target="_blank">028f18d</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-21
</td>
<td>
update on Fold Change plot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/46436008ea4f77a7becdbb7bc3123afc5cf066b7/docs/other_analysis.html" target="_blank">4643600</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/751239eee41b19a825e580fce86d935777ef6160/analysis/other_analysis.Rmd" target="_blank">751239e</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-16
</td>
<td>
updating and moving code
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/3d4ca6458d1a506bfb480498918285e886386178/analysis/other_analysis.Rmd" target="_blank">3d4ca64</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-16
</td>
<td>
updates on Friday
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/7ce5a2c0c142bb4d59f1cec4ef882c44761c1898/docs/other_analysis.html" target="_blank">7ce5a2c</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/9afd6a00b03f24ef204973a84e91a5cba0dbce25/analysis/other_analysis.Rmd" target="_blank">9afd6a0</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-15
</td>
<td>
fixing wflow error
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/6b03af2d301e424370c468bea4857b159cbb7df8/docs/other_analysis.html" target="_blank">6b03af2</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/e02ca18cfdba1a09e2f8024c8fe2ae3f4d2c8953/docs/other_analysis.html" target="_blank">e02ca18</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/9ad6b911145da135d780f778f7c27c67be082f00/analysis/other_analysis.Rmd" target="_blank">9ad6b91</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-15
</td>
<td>
showing code adding pvalue text
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/750ee45b49395e0a3b0b697c5cf6b321d329b47d/docs/other_analysis.html" target="_blank">750ee45</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/637531cd7f160fb39ee28043c8985c5e80852eb9/analysis/other_analysis.Rmd" target="_blank">637531c</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-15
</td>
<td>
moving out the knowles data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/f8f511af8663414efe663a35368a3786c78607b4/analysis/other_analysis.Rmd" target="_blank">f8f511a</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-15
</td>
<td>
updates and simplifications of code
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/7fc7ec79fc2645d132183614919dcf4be3639184/analysis/other_analysis.Rmd" target="_blank">7fc7ec7</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-14
</td>
<td>
updating code
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/4b6bd9b852567426b563c13fcf07bf97a2a5fa26/docs/other_analysis.html" target="_blank">4b6bd9b</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/4b62a1e1d8aa9186856b6186d2f9e99930401909/analysis/other_analysis.Rmd" target="_blank">4b62a1e</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-07
</td>
<td>
updated numbers for grant
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/d64a0aeeeb3c53f23ee14068fdb142d25d71982c/docs/other_analysis.html" target="_blank">d64a0ae</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/81f100c2b2c928fdd16d1632fd859f3928f81cad/analysis/other_analysis.Rmd" target="_blank">81f100c</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-07
</td>
<td>
add Dox reQTL grouping and AC shared numbers
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/47f85a29402175e486a4473027adccfd648c8279/docs/other_analysis.html" target="_blank">47f85a2</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/0ecede3a8e7f18601d7dc3ac18736f74f590678b/analysis/other_analysis.Rmd" target="_blank">0ecede3</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-07
</td>
<td>
data with CRispr set added and heatmap changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/9a62d7c730cc913b0dc1ae23c45f0d86884c5bb4/docs/other_analysis.html" target="_blank">9a62d7c</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/232d3b0851d71563aeb0f3019fca0f67cda23d1a/analysis/other_analysis.Rmd" target="_blank">232d3b0</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-06
</td>
<td>
Finally tested chisquare between knowles data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/10bcf05372016520ab1a55fa77cb798d7b2bf466/analysis/other_analysis.Rmd" target="_blank">10bcf05</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-06
</td>
<td>
updating the k4/k5 analysis of DEG
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/b4dd015c91f5dbdb669488df59cc850de8b323a4/docs/other_analysis.html" target="_blank">b4dd015</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/652d7e8f4bcb2c8cb09680fa0da4c52499ac67ac/analysis/other_analysis.Rmd" target="_blank">652d7e8</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-02
</td>
<td>
updated heatmap Seoane Chisqure for cormotif
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/5aeda279932ec0ca001860f9e3692ac7a4398278/docs/other_analysis.html" target="_blank">5aeda27</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/6524ecdeabd4029da1d1bd6690bc1615c9e9400e/analysis/other_analysis.Rmd" target="_blank">6524ecd</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-02
</td>
<td>
Adding in heatmaps of chi values
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/5dd9ddbeea19ed665ac90521d464d09e65d82e52/docs/other_analysis.html" target="_blank">5dd9ddb</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/8eaea47d4790ef579af20ee326255781ab41546d/analysis/other_analysis.Rmd" target="_blank">8eaea47</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-02
</td>
<td>
chi square updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/e4d118c6396a1114eb0dd5aebbc3481db3f2bb86/docs/other_analysis.html" target="_blank">e4d118c</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/573a477d2f2fc5fa6cf2ca5254188c1166f9fdd5/analysis/other_analysis.Rmd" target="_blank">573a477</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-01
</td>
<td>
Updateing supplement 1 seoan chi results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/cc3dfc371903fa4efcdae2cc65e248bd2096b47a/docs/other_analysis.html" target="_blank">cc3dfc3</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/522cce8d16b6b99e8b36416abbde636d26454652/analysis/other_analysis.Rmd" target="_blank">522cce8</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-01
</td>
<td>
Adding chisquare and other analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/4723cddd5dcb01e7ac6746376b27f6db714bae2d/docs/other_analysis.html" target="_blank">4723cdd</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/07a6e0634ed657b47c620bfe2e536bdc0aa35612/analysis/other_analysis.Rmd" target="_blank">07a6e06</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-31
</td>
<td>
adding in more data including Cormotif enrichment numbers
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/6fd877b906dff7762e1fb2e7f7e88c220d5e78dc/docs/other_analysis.html" target="_blank">6fd877b</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/b2ba05543ba33b808852ec2946715cc719826a60/analysis/other_analysis.Rmd" target="_blank">b2ba055</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-31
</td>
<td>
adding Seoane data with cormotif things
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/4c0812e96b1a6d77e01795568d9c8ad128f27355/docs/other_analysis.html" target="_blank">4c0812e</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/c7e0fcc1538ce9c279840ff78eef23e211f5e2ff/analysis/other_analysis.Rmd" target="_blank">c7e0fcc</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-26
</td>
<td>
adding in Gtex and chisquare values
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/e1bcef01efad375131f1487200ebf4c288290fa5/docs/other_analysis.html" target="_blank">e1bcef0</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/0f512c394c1a9e200dbdbe7a5930d5f34af81da1/analysis/other_analysis.Rmd" target="_blank">0f512c3</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-26
</td>
<td>
adding in Gtex and chisquare values
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/1f8c4835b54c2804bbfee0526b6bab3457265f13/analysis/other_analysis.Rmd" target="_blank">1f8c483</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-26
</td>
<td>
updating code with gtex and chisq
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/25d32daf7be2ae6723057c97caf2949a9428488e/analysis/other_analysis.Rmd" target="_blank">25d32da</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-26
</td>
<td>
Adding 3 hour and chisq test to populations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/56107490ac4b9dbdd9343ade9de73aac37bb15e7/docs/other_analysis.html" target="_blank">5610749</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/889832ac0ea578fe1d04c0de6c291f15b992a10e/analysis/other_analysis.Rmd" target="_blank">889832a</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-22
</td>
<td>
add Seoane data again
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/36cbdab1fb448aa132b0811799320f905e86f928/docs/other_analysis.html" target="_blank">36cbdab</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/de54fd50d3a5d518e1949560d9d81c3250c78ff8/analysis/other_analysis.Rmd" target="_blank">de54fd5</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-22
</td>
<td>
add Seoane data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/7243a18895c7608d714a09b4df5e3265419428c8/docs/other_analysis.html" target="_blank">7243a18</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/e2b32151e34bcb93bd9d8f364ef0ee2eeb2eaf82/analysis/other_analysis.Rmd" target="_blank">e2b3215</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-22
</td>
<td>
add Seoane data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/c3481d851bbd1051407745bbb9d62ed2c0e8b267/docs/other_analysis.html" target="_blank">c3481d8</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/acbd0a894a6a735d071ed66d74ef6af06ff4b70f/analysis/other_analysis.Rmd" target="_blank">acbd0a8</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-22
</td>
<td>
updates on GWAS enrichment
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/e8c82ec1a3ebfae76e09c0e71b7d6185a83f9fb0/analysis/other_analysis.Rmd" target="_blank">e8c82ec</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-05-18
</td>
<td>
adding other_analysis and genes of interest log2cpm
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(limma)
library(tidyverse)
library(ggsignif)
library(biomaRt)
library(RColorBrewer)
library(cowplot)
library(ggpubr)
library(scales)
library(sjmisc)
library(kableExtra)
library(broom)
library(ComplexHeatmap)</code></pre>
<div id="data-set-comparison" class="section level1">
<h1>Data set comparison</h1>
<div id="order" class="section level3">
<h3>order:</h3>
<p>ArrGWAS<br />
HFGWAS<br />
CADGWAS</p>
<p>Seaone 2019<br />
Supplemental 1 (408 genes)<br />
Supplemental 4 (54 genes)</p>
<p>Crispr sets</p>
</div>
<div id="gwas" class="section level2">
<h2>GWAS</h2>
<div id="arrgwas-to-24-hour-deg-genes-p-0.05" class="section level3">
<h3>ArrGWAS to 24 hour DEG genes p &lt; 0.05</h3>
<div id="hour-data-set" class="section level4">
<h4>24 hour data set</h4>
<pre class="r"><code># How I did the string split
# Arr_GWAS &lt;- ArrGWAS[,13]
# names(Arr_GWAS) &lt;- &quot;genesplit&quot;
# Arr_GWAS &lt;- Arr_GWAS %&gt;% 
#   separate_longer_delim(genesplit, delim = &quot;,&quot;)
#write.csv(Arr_GWAS,&quot;data/Arr_GWAS.txt&quot;)
arr_GWAS &lt;- read.csv(&quot;data/Arr_GWAS.txt&quot;, row.names = 1)
Arr_geneset &lt;- readRDS(&quot;data/Arr_geneset.RDS&quot;)
# Arr_geneset &lt;- getBM(attributes=my_attributes,filters =&#39;hgnc_symbol&#39;,
#                   values = arr_GWAS, mart = ensembl)
# #remove duplicates
# Arr_geneset &lt;- Arr_geneset %&gt;% distinct(entrezgene_id, .keep_all =TRUE)
# saveRDS(Arr_geneset,&quot;data/Arr_geneset.RDS&quot;)
#Apply sorting
toplist24hr %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  mutate(sigcount = if_else(adj.P.Val &lt;0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(ARR=if_else(ENTREZID %in%Arr_geneset$entrezgene_id,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  group_by(id,sigcount,ARR) %&gt;% 
  dplyr::summarize(ARRcount=n())%&gt;% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(ARR), values_from=ARRcount) %&gt;% 
   mutate(ARRprop=(y/(y+no)*100)) %&gt;% 
       ggplot(., aes(x=id, y=ARRprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf(&quot;%.2f&quot;,ARRprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0(&quot;number&quot;~a,&quot;out of&quot;,~b))))+
       facet_wrap(~sigcount)+
       ggtitle(&quot;24 hour non-significant and significant enrichment proporitions of Arrhythmia GWAS &quot;)</code></pre>
<p><img src="figure/other_analysis.Rmd/24%20hour%20ArrGWAS-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make table of numbers:


dataframARR &lt;- toplist24hr %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  mutate(sigcount = if_else(adj.P.Val &lt;0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(ARR=if_else(ENTREZID %in%Arr_geneset$entrezgene_id,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  group_by(id,sigcount,ARR) %&gt;% 
  dplyr::summarize(ARRcount=n()) %&gt;% 
  as.data.frame()

dataframARR %&gt;% 
  kable(., caption= &quot;Significant (adj. P value of &lt;0.05) and non-sig gene counts in Arrhythmia 24 hour GWAS&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;, full_width = FALSE) %&gt;%  
  kable_styling(full_width = FALSE, position = &quot;left&quot;,bootstrap_options = c(&quot;striped&quot;),font_size = 18) %&gt;% 
  scroll_box(width = &quot;60%&quot;, height = &quot;400px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:60%; ">
<table class=" lightable-paper lightable-striped table table-striped" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 18px; width: auto !important; ">
<caption style="font-size: initial !important;">
Significant (adj. P value of &lt;0.05) and non-sig gene counts in
Arrhythmia 24 hour GWAS
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
id
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
sigcount
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ARR
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
ARRcount
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
7016
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
51
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
6948
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
69
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
7382
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
57
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
6582
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
63
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
7699
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
57
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
6265
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
63
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
12863
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
106
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
1101
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
14
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13964
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
120
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="hour-data-set-1" class="section level4">
<h4>3 hour data set</h4>
<img src="figure/other_analysis.Rmd/three%20hour%20ArrGWAS-1.png" width="672" style="display: block; margin: auto;" />
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:60%; ">
<table class=" lightable-paper lightable-striped table table-striped" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 18px; width: auto !important; ">
<caption style="font-size: initial !important;">
Significant (adj. P value of &lt;0.05) and non-sig gene counts in
Arrhythmia 3 hour GWAS
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
id
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
sigcount
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ARR
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
ARRcount
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13440
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
112
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
524
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
8
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13945
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
120
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
19
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13757
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
117
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
207
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13894
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
115
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
70
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
5
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13964
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
120
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="chi-square-test-arr" class="section level4">
<h4>chi square test ARR</h4>
<pre class="r"><code>chi_funarr &lt;-  toplistall %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  dplyr::filter(id!=&quot;Trastuzumab&quot;) %&gt;% 
  mutate(sigcount = if_else(adj.P.Val &lt;0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(ARR=if_else(ENTREZID %in%Arr_geneset$entrezgene_id,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  group_by(id,time) %&gt;% 
  summarise(pvalue= chisq.test(ARR, sigcount)$p.value) 


chi_funarr %&gt;% 
  kable(., caption= &quot;after performing chi square test between DEgenes, and non DE genes&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;) %&gt;%  
  kable_styling(full_width = FALSE,font_size = 18) %&gt;% 
  scroll_box(width = &quot;60%&quot;, height = &quot;400px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:60%; ">
<table class=" lightable-paper lightable-striped table" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; margin-left: auto; margin-right: auto; font-size: 18px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
after performing chi square test between DEgenes, and non DE genes
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
id
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
time
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
pvalue
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.1101318
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
0.1536193
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.2799966
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
1.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.1136502
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
0.5908261
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.1744165
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
0.0000012
</td>
</tr>
</tbody>
</table>
</div>
<p><img src="figure/other_analysis.Rmd/ARR%20heatmap-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="hfgwas" class="section level3">
<h3>HFGWAS</h3>
<div id="hours-hf" class="section level4">
<h4>24 hours HF</h4>
<pre class="r"><code>##just like ARrGWAS- imported the total csv, the took the &quot;nearest&quot; column and separated out the gene info
# test &lt;- HFGWAS %&gt;% 
#   select(nearest) %&gt;% 
#   separate_wider_delim(nearest, delim = &quot;[&quot;, names_sep = &quot;&quot;, too_few = &quot;align_start&quot;)
# test2 &lt;- str_sub(test$nearest2,0,nchar(test$nearest2)-1)
# Hf_GWAS &lt;- test2
#write.csv(Hf_GWAS, &quot;data/Hf_GWAS.txt&quot;)
# HF_GWAS &lt;- read.csv(&quot;data/Hf_GWAS.txt&quot;, row.names =1)
# 
# HF_geneset &lt;- getBM(attributes=my_attributes,filters =&#39;hgnc_symbol&#39;,
#                   values = HF_GWAS, mart = ensembl)
# #remove duplicates
# HF_geneset &lt;- HF_geneset %&gt;% distinct(entrezgene_id, .keep_all =TRUE)
# saveRDS(HF_geneset,&quot;data/HF_geneset.RDS&quot;)
HF_geneset &lt;- readRDS(&quot;data/HF_geneset.RDS&quot;)
#Apply sorting
toplist24hr %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  mutate(sigcount = if_else(adj.P.Val &lt;0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(HF=if_else(ENTREZID %in%HF_geneset$entrezgene_id,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  group_by(id,sigcount,HF) %&gt;% 
  summarize(HFcount=n())%&gt;% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(HF), values_from=HFcount) %&gt;% 
   mutate(HFprop=(y/(y+no)*100)) %&gt;% 
       ggplot(., aes(x=id, y=HFprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf(&quot;%.2f&quot;,HFprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0(&quot;number&quot;~a,&quot;out of&quot;,~b))))+
       facet_wrap(~sigcount)+
       ggtitle(&quot;non-significant and significant enrichment proporitions of Heart Failure GWAS &quot;)</code></pre>
<p><img src="figure/other_analysis.Rmd/24%20hours%20HFGWAS-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make table of numbers:


dataframHF &lt;- toplist24hr %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  mutate(sigcount = if_else(adj.P.Val &lt;0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(HF=if_else(ENTREZID %in%HF_geneset$entrezgene_id,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  group_by(id,sigcount,HF) %&gt;% 
  summarize(HFcount=n()) %&gt;% 
  as.data.frame()

dataframHF %&gt;% #mutate_at(.vars = 6, .funs= scientific_format()) %&gt;% 
  kable(., caption= &quot;Significant (adj. P value of &lt;0.05) and non-sig gene counts in HFhythmia GWAS&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;, full_width = FALSE) %&gt;%  
  kable_styling(full_width = FALSE, position = &quot;left&quot;,bootstrap_options = c(&quot;striped&quot;),font_size = 18) %&gt;% 
  scroll_box(width = &quot;60%&quot;, height = &quot;400px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:60%; ">
<table class=" lightable-paper lightable-striped table table-striped" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 18px; width: auto !important; ">
<caption style="font-size: initial !important;">
Significant (adj. P value of &lt;0.05) and non-sig gene counts in
HFhythmia GWAS
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
id
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
sigcount
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
HF
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
HFcount
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
7056
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
11
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
6995
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
22
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
7427
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
12
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
6624
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
21
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
7742
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
14
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
6309
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
19
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
12939
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
30
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
1112
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
14051
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
33
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="hours-hf-1" class="section level3">
<h3>3 hours HF</h3>
<img src="figure/other_analysis.Rmd/3%20hours%20HFGWAS-1.png" width="672" style="display: block; margin: auto;" />
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:60%; ">
<table class=" lightable-paper lightable-striped table table-striped" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 18px; width: auto !important; ">
<caption style="font-size: initial !important;">
Significant (adj. P value of &lt;0.05) and non-sig gene counts in Three
hour HFhythmia GWAS
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
id
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
sigcount
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
HF
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
HFcount
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13521
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
31
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
530
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
14032
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
33
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
19
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13841
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
33
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
210
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13976
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
33
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
75
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
14051
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
33
</td>
</tr>
</tbody>
</table>
</div>
<div id="chi-square-test-hf" class="section level4">
<h4>chi square test HF</h4>
<pre class="r"><code>chi_funhf &lt;-  toplistall %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  dplyr::filter(id!=&quot;Trastuzumab&quot;) %&gt;% 
  mutate(sigcount = if_else(adj.P.Val &lt;0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(HF=if_else(ENTREZID %in%HF_geneset$entrezgene_id,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  group_by(id,time) %&gt;% 
  summarise(pvalue= chisq.test(HF, sigcount)$p.value) 

chi_funhf %&gt;% 
  kable(., caption= &quot;after performing chi square test between DEgenes, and non DE genes&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;, full_width = FALSE) %&gt;%  
  kable_styling(full_width = FALSE,font_size = 18) %&gt;% 
  scroll_box(width = &quot;60%&quot;, height = &quot;400px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:60%; ">
<table class=" lightable-paper lightable-striped table" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 18px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
after performing chi square test between DEgenes, and non DE genes
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
id
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
time
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
pvalue
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.0778586
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
0.8167557
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.0852093
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
1.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.1981312
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
1.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
1.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
1.0000000
</td>
</tr>
</tbody>
</table>
</div>
<p><img src="figure/other_analysis.Rmd/HF%20heatmap-1.png" width="672" style="display: block; margin: auto;" />
## CAD GWAS</p>
</div>
</div>
<div id="hour-data-set-2" class="section level3">
<h3>24 hour data set</h3>
<pre class="r"><code># test &lt;- CADGWAS %&gt;% 
#    select(nearest) %&gt;% 
#    separate_wider_delim(nearest, delim = &quot;[&quot;, names_sep = &quot;&quot;, too_few = &quot;align_start&quot;)
#  test2 &lt;- str_sub(test$nearest2,0,nchar(test$nearest2)-1)
# 
# test2[c(32,38,44,74,112,126,191,212)] &lt;- c(&quot;TPCN1&quot;,&quot;C2orf43&quot;,&quot;FAM222A&quot;, &quot;TDRD15&quot;  ,&quot;AGPAT4&quot;,&quot;SVOP&quot;,&quot;SVOP&quot;,&quot;PLG&quot;)
#    
#  test2 [c(218,226,228,233,239,245,256,270,281,322,324,332,335,338,347,351,352,358)] &lt;-  c(&quot;HPCAL1&quot;, &quot;KLHL29&quot;  , &quot;COL4A3BP&quot;  , &quot;ARAP1&quot; ,
#  &quot;VEGFA&quot;, &quot;TBPL1&quot;,&quot;SLC22A3&quot; ,&quot;C19orf38&quot;,&quot;LPA&quot;,&quot;VPS29&quot;,&quot;ATP2A2&quot; ,&quot;ATP2A2&quot;,&quot;KLHL29&quot;,&quot;GUCY1A3&quot;,&quot;KCNE2&quot;,  &quot;HOXB9&quot;,&quot;P2RY2&quot; ,&quot;CTC-236F12.4&quot;)
#  
#  CAD_GWAS &lt;- test2
#write.csv(CAD_GWAS, &quot;data/cvd_GWAS.txt&quot;)
CAD_GWAS &lt;- read.csv(&quot;data/cvd_GWAS.txt&quot;, row.names =1)

# CAD_geneset &lt;- getBM(attributes=my_attributes,filters =&#39;hgnc_symbol&#39;,
#                   values = CAD_GWAS, mart = ensembl)
# #remove duplicates
# CAD_geneset &lt;- CAD_geneset %&gt;% distinct(entrezgene_id, .keep_all =TRUE)
# 
# saveRDS(CAD_geneset,&quot;data/CAD_geneset.RDS&quot;)
CAD_geneset &lt;- readRDS(&quot;data/CAD_geneset.RDS&quot;)
#Apply sorting



toplist24hr %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  mutate(sigcount = if_else(adj.P.Val &lt;0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(CAD=if_else(ENTREZID %in%CAD_geneset$entrezgene_id,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  group_by(id,sigcount,CAD) %&gt;% 
  summarize(CADcount=n())%&gt;% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(CAD), values_from=CADcount) %&gt;% 
   mutate(CADprop=(y/(y+no)*100)) %&gt;% 
       ggplot(., aes(x=id, y=CADprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf(&quot;%.2f&quot;,CADprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0(&quot;number&quot;~a,&quot;out of&quot;,~b))))+
       facet_wrap(~sigcount)+
       ggtitle(&quot;non-significant and significant enrichment proporitions of CAD GWAS &quot;)</code></pre>
<p><img src="figure/other_analysis.Rmd/CAD%2024%20GWAS-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##make table of numbers:


dataframCAD &lt;- toplist24hr %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  mutate(sigcount = if_else(adj.P.Val &lt;0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(CAD=if_else(ENTREZID %in%CAD_geneset$entrezgene_id,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  group_by(id,sigcount,CAD) %&gt;% 
  summarize(CADcount=n()) %&gt;% 
  as.data.frame()

dataframCAD %&gt;% #mutate_at(.vars = 6, .funs= scientific_format()) %&gt;% 
  kable(., caption= &quot;Significant (adj. P value of &lt;0.05) and non-sig gene counts in CAD GWAS&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;, full_width = FALSE) %&gt;%  
  kable_styling(full_width = FALSE, position = &quot;left&quot;,bootstrap_options = c(&quot;striped&quot;),font_size = 18) %&gt;% 
  scroll_box(width = &quot;60%&quot;, height = &quot;400px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:60%; ">
<table class=" lightable-paper lightable-striped table table-striped" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 18px; width: auto !important; ">
<caption style="font-size: initial !important;">
Significant (adj. P value of &lt;0.05) and non-sig gene counts in CAD
GWAS
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
id
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
sigcount
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
CAD
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
CADcount
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
6956
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
111
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
6899
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
118
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
7318
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
121
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
6537
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
108
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
7636
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
120
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
6219
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
109
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
12756
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
213
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
1099
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
16
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13855
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
229
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="hour-data-set-3" class="section level3">
<h3>3 hour data set</h3>
<img src="figure/other_analysis.Rmd/CAD%203%20GWAS-1.png" width="672" style="display: block; margin: auto;" />
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:60%; ">
<table class=" lightable-paper lightable-striped table table-striped" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 18px; width: auto !important; ">
<caption style="font-size: initial !important;">
Significant (adj. P value of &lt;0.05) and non-sig gene counts in 3 hour
CAD GWAS
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
id
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
sigcount
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
CAD
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
CADcount
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13338
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
214
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
517
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
15
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13836
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
229
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
19
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13651
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
223
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
204
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13782
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
227
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
73
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13855
</td>
</tr>
<tr>
<td style="text-align:left;">
Trastuzumab
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
229
</td>
</tr>
</tbody>
</table>
</div>
<div id="chi-square-test-cad" class="section level4">
<h4>chi square test CAD</h4>
<pre class="r"><code>chi_funCAD &lt;-  toplistall %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  dplyr::filter(id!=&quot;Trastuzumab&quot;) %&gt;% 
  mutate(sigcount = if_else(adj.P.Val &lt;0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(CAD=if_else(ENTREZID %in%CAD_geneset$entrezgene_id,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  group_by(id,time) %&gt;% 
  summarise(pvalue= chisq.test(CAD, sigcount)$p.value) 

chi_funCAD %&gt;% 
  kable(., caption= &quot;after performing chi square test between DEgenes, and non DE genes&quot;) %&gt;% 
  kable_paper(&quot;striped&quot;, full_width = FALSE) %&gt;%  
  kable_styling(full_width = FALSE,font_size = 18) %&gt;% 
  scroll_box(width = &quot;60%&quot;, height = &quot;400px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:60%; ">
<table class=" lightable-paper lightable-striped table" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 18px; width: auto !important; margin-left: auto; margin-right: auto;">
<caption style="font-size: initial !important;">
after performing chi square test between DEgenes, and non DE genes
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
id
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
time
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
pvalue
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.6498845
</td>
</tr>
<tr>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
0.0409172
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
1.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
1.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.4524570
</td>
</tr>
<tr>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
0.2515982
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:right;">
0.6876234
</td>
</tr>
<tr>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:right;">
0.7973241
</td>
</tr>
</tbody>
</table>
</div>
<p><img src="figure/other_analysis.Rmd/CAD%20heatmap-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/other_analysis.Rmd/CAD%20heatmap-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>[1] &quot;This is for  GWAS 24 hours -log(chi square pvalue)&quot;</code></pre>
<p>The star represents chi square p.value &lt; 0.05.</p>
</div>
</div>
<div id="gwas-heatmap" class="section level3">
<h3>GWAS heatmap</h3>
<pre class="r"><code># GWAS_goi &lt;- c(&#39;RARG&#39;, &#39;ITGB7&#39;, &#39;TNS2&#39;,&#39;ZNF740&#39;,&#39;SLC28A3&#39;,&#39;RMI1&#39;,
# &#39;FEDORA&#39; ,&#39;GDF5&#39;,&#39;FRS2&#39;,&#39;HDDC2&#39;,&#39;EEF1B2&#39;)
# 
# library(biomaRt)
# ensembl &lt;- useMart(&quot;ensembl&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)
# my_chr &lt;- c(1:22, &#39;M&#39;, &#39;X&#39;, &#39;Y&#39;)  
# my_attributes &lt;- c(&#39;entrezgene_id&#39;, &#39;ensembl_gene_id&#39;, &#39;hgnc_symbol&#39;)
# 
# 
# GWAS_goi&lt;- getBM(attributes=my_attributes,filters =&#39;hgnc_symbol&#39;,
#          values = GWAS_goi, mart = ensembl)
# GWAS_goi&lt;-GWAS_goi %&gt;% distinct(entrezgene_id,.keep_all = TRUE) %&gt;% add_row(entrezgene_id=&#39;124903732&#39;,ensembl_gene_id=&#39;ENSG00000260788&#39;, hgnc_symbol=&quot;RP11-298D21.1
# &quot;)

  
# write.csv(GWAS_goi,&quot;output/GWAS_goi.csv&quot;)
GWAS_goi &lt;- read.csv(&quot;output/GWAS_goi.csv&quot;)
##get the abs FC of all GOI
GWASabsFCsig &lt;- 
  toplistall %&gt;% 
  # mutate(absFC=abs(logFC)) %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  filter(id !=&quot;Trastuzumab&quot;) %&gt;%
  mutate(time=factor(time, levels=c(&quot;3_hours&quot;,&quot;24_hours&quot;))) %&gt;%
  filter(ENTREZID %in% GWAS_goi$entrezgene_id) %&gt;% 
   filter(time ==&quot;24_hours&quot;) %&gt;% 
  dplyr::select(ENTREZID ,time, id,logFC, adj.P.Val, SYMBOL) %&gt;%
  mutate(id =case_match(id,
                        &#39;Daunorubicin&#39;~&#39;DNR&#39;,
                        &#39;Doxorubicin&#39;~&#39;DOX&#39;,
                        &#39;Epirubicin&#39;~&#39;EPI&#39;,
                        &#39;Mitoxantrone&#39;~&#39;MTX&#39;,
                        .default = id)) %&gt;% 
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from =adj.P.Val)
  
gwas_sig_mat &lt;- GWASabsFCsig %&gt;% 
   column_to_rownames(var=&quot;id&quot;) %&gt;%
  as.matrix()
 

GWASabsFC &lt;- toplistall %&gt;% 
  # mutate(absFC=abs(logFC)) %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  filter(id !=&quot;Trastuzumab&quot;) %&gt;% 
  filter(time==&quot;24_hours&quot;) %&gt;% 
  mutate(logFC= logFC*(-1)) %&gt;%
  filter(ENTREZID %in% GWAS_goi$entrezgene_id) %&gt;% 
  dplyr::select(SYMBOL ,time, id, logFC) %&gt;% 
  mutate(id =case_match(id,&#39;Daunorubicin&#39;~&#39;DNR&#39;, 
                        &#39;Doxorubicin&#39;~&#39;DOX&#39;,
                        &#39;Epirubicin&#39;~&#39;EPI&#39;,
                        &#39;Mitoxantrone&#39;~&#39;MTX&#39;,
                        .default = id)) %&gt;% 
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from = logFC) %&gt;% 
  column_to_rownames(var=&quot;id&quot;) %&gt;%
  as.matrix()

 

Heatmap(GWASabsFC, name = &quot;Fold change\nvalues&quot;, 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = &quot;left&quot;,
        column_title = &quot;Fold change values of GWAS and TWAS genes&quot;, 
        column_title_side = &quot;top&quot;,
        column_title_gp = gpar(fontsize = 16, fontface = &quot;bold&quot;),
        column_order= c(&#39;RARG&#39;,
                        &#39;TNS2&#39;, 
                        &#39;ZNF740&#39;,
                        &#39;SLC28A3&#39;,
                        &#39;RMI1&#39;,
                        &#39;EEF1B2&#39;,
                        &#39;FRS2&#39;, 
                        &#39;HDDC2&#39;),
        column_names_rot = 0, 
        column_names_gp = gpar(fontsize = 12),
        column_names_centered = TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(gwas_sig_mat[i, j] &lt;0.05)
            grid.text(&quot;*&quot;, x, y, gp = gpar(fontsize = 20))
})</code></pre>
<p><img src="figure/other_analysis.Rmd/GWAS%20genes%20of%20interest-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The stars represent all genes that have an adj. P. value of &lt; 0.05
(significantly differentially expressed)</p>
</div>
</div>
<div id="crispr-list" class="section level2">
<h2>Crispr list</h2>
<pre class="r"><code>DEG_cormotif &lt;- readRDS(&quot;data/DEG_cormotif.RDS&quot;)
list2env(DEG_cormotif,envir=.GlobalEnv)</code></pre>
<pre><code>&lt;environment: R_GlobalEnv&gt;</code></pre>
<pre class="r"><code># Crispr_list &lt;- read_excel(&quot;C:/Users/renee/Downloads/41598_2021_92988_MOESM2_ESM.xlsx&quot;)
#  View(Crispr_list)
# crispr_genes &lt;- Crispr_list %&gt;% 
#   dplyr::filter(p.value &lt;0.05) %&gt;% 
#   select(GeneName)
  

# crispr_genes &lt;- getBM(attributes=my_attributes,filters =&#39;hgnc_symbol&#39;,
                  # values =crispr_genes$GeneName, mart = ensembl)
# write.csv(crispr_genes,&#39;data/crispr_genes.csv&#39;)

crispr_genes &lt;- read.csv(&quot;data/crispr_genes.csv&quot;, row.names = 1)
print(&quot; number of unique crispr_genes after conversion from hgnc symbol to entrezid&quot;)</code></pre>
<pre><code>[1] &quot; number of unique crispr_genes after conversion from hgnc symbol to entrezid&quot;</code></pre>
<pre class="r"><code>length(unique(crispr_genes$entrezgene_id))</code></pre>
<pre><code>[1] 154</code></pre>
<pre class="r"><code>crisprunique &lt;- crispr_genes %&gt;% distinct(entrezgene_id,.keep_all = TRUE)

Doxcrispall &lt;- toplistall %&gt;%
  distinct(ENTREZID,.keep_all = TRUE) %&gt;% 
  dplyr::select(ENTREZID,id,time)
  

crispmotifsummary &lt;- Doxcrispall %&gt;% 
  mutate(ER=if_else(ENTREZID %in% motif_ER,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  mutate(LR=if_else(ENTREZID %in% motif_LR,&quot;y&quot;,&quot;no&quot;)) %&gt;%
  mutate(TI=if_else(ENTREZID %in% motif_TI,&quot;y&quot;,&quot;no&quot;)) %&gt;%
  mutate(NR=if_else(ENTREZID %in% motif_NR,&quot;y&quot;,&quot;no&quot;)) %&gt;%
  mutate(crisp = if_else(ENTREZID %in% crisprunique$entrezgene_id, &quot;y&quot;, &quot;no&quot;)) %&gt;% 
  group_by(crisp,ER,TI,LR,NR) %&gt;% 
  dplyr::summarize(n=n()) %&gt;% 
  as.tibble  %&gt;% 
  pivot_wider(id_cols = c(crisp), names_from = c(&#39;ER&#39;, &#39;TI&#39;, &#39;LR&#39;, &#39;NR&#39;), values_from= n) %&gt;% 
  rename(.,c(&quot;crisp&quot;=crisp,&quot;none&quot;= 2 , &quot;ER&quot; = 3 , &quot;TI&quot; = 4 , &quot;LR&quot; = 5 ,&quot;NR&quot; = 6)) 

cris_mat &lt;- crispmotifsummary %&gt;% dplyr::select(ER:NR) %&gt;% as.matrix()
chicheck &lt;- data.frame(one= c(&quot;LR&quot;,&quot;ER&quot;,&quot;TI&quot;),two=rep(&quot;NR&quot;,3),p.value=c(&quot;&quot;,&quot;&quot;,&quot;&quot;))
  
 chicheck$p.value[1] &lt;- chisq.test(cris_mat[,c(&#39;LR&#39;,&#39;NR&#39;)],correct = FALSE)$p.value

chicheck$p.value[2] &lt;- chisq.test(cris_mat[,c(&#39;ER&#39;,&#39;NR&#39;)],correct = FALSE)$p.value
chicheck$p.value[3] &lt;- chisq.test(cris_mat[,c(&#39;TI&#39;,&#39;NR&#39;)],correct = FALSE)$p.value

chicheck%&gt;% kable(., caption= &quot;chi square test p.values for encrichment of  Doxcrispr gene sets in motif sets&quot; )%&gt;% 
  kable_paper(&quot;striped&quot;, full_width = FALSE) %&gt;%  
  kable_styling(full_width = FALSE, position = &quot;left&quot;,bootstrap_options = c(&quot;striped&quot;),font_size = 18) %&gt;% 
  scroll_box(width = &quot;60%&quot;, height = &quot;400px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:60%; ">
<table class=" lightable-paper lightable-striped table table-striped" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 18px; width: auto !important; ">
<caption style="font-size: initial !important;">
chi square test p.values for encrichment of Doxcrispr gene sets in motif
sets
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
one
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
two
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
p.value
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
LR
</td>
<td style="text-align:left;">
NR
</td>
<td style="text-align:left;">
0.861985991471947
</td>
</tr>
<tr>
<td style="text-align:left;">
ER
</td>
<td style="text-align:left;">
NR
</td>
<td style="text-align:left;">
0.402681880154749
</td>
</tr>
<tr>
<td style="text-align:left;">
TI
</td>
<td style="text-align:left;">
NR
</td>
<td style="text-align:left;">
0.309642007916355
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>chicheck_1 &lt;- chicheck %&gt;% mutate(p.value=as.numeric(p.value)) %&gt;% 
  mutate(neg.logvalue=(-1*log(p.value))) %&gt;% column_to_rownames(&#39;one&#39;) %&gt;% dplyr::select(neg.logvalue) %&gt;% as.matrix
col_fun = circlize::colorRamp2(c(0, 2), c(&quot;white&quot;, &quot;purple&quot;))

Heatmap( chicheck_1, name = &quot;Doxcrispr enrichment \nchi square -log p values&quot;, cluster_rows = FALSE, cluster_columns = FALSE, col=col_fun,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(chicheck_1[i, j] &gt; -log(0.05))
            grid.text(&quot;*&quot;, x, y, gp = gpar(fontsize = 20))
})</code></pre>
<p><img src="figure/other_analysis.Rmd/Crispr%20data%20analysis-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>col_fun4 = circlize::colorRamp2(c(0, 5), c(&quot;white&quot;, &quot;purple&quot;))


pairwisecrispr &lt;- toplistall %&gt;%
  filter(id!=&#39;Trastuzumab&#39;) %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  mutate(time=factor(time, levels=c(&quot;3_hours&quot;,&quot;24_hours&quot;))) %&gt;%
  mutate(sigcount = if_else(adj.P.Val &lt;0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(crisp = if_else(ENTREZID %in% crisprunique$entrezgene_id, &quot;y&quot;, &quot;no&quot;)) %&gt;% 
  group_by(time, id) %&gt;%
  summarise(pvalue= chisq.test(crisp, sigcount, correct=FALSE)$p.value)
 
  
  crisprnumbers &lt;- toplistall %&gt;%
  filter(id!=&#39;Trastuzumab&#39;) %&gt;% 
  mutate(id = as.factor(id)) %&gt;%
  mutate(time=factor(time, levels=c(&quot;3_hours&quot;,&quot;24_hours&quot;))) %&gt;%
  mutate(sigcount = if_else(adj.P.Val &lt;0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  mutate(crisp = if_else(ENTREZID %in% crisprunique$entrezgene_id, &quot;y&quot;, &quot;no&quot;)) %&gt;% 
  group_by(time, id,sigcount,crisp) %&gt;%
  dplyr::summarize(n=n()) %&gt;% 
  as.tibble() #%&gt;% 
  
crisprnumbers %&gt;% kable(., caption= &quot;Summary of genes found in both sigDE and non sigDE by treatment&quot; )%&gt;% 
  kable_paper(&quot;striped&quot;, full_width = FALSE) %&gt;%  
  kable_styling(full_width = FALSE, position = &quot;left&quot;,bootstrap_options = c(&quot;striped&quot;),font_size = 18) %&gt;% 
  scroll_box(width = &quot;60%&quot;, height = &quot;400px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:60%; ">
<table class=" lightable-paper lightable-striped table table-striped" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 18px; width: auto !important; ">
<caption style="font-size: initial !important;">
Summary of genes found in both sigDE and non sigDE by treatment
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
time
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
id
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
sigcount
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
crisp
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
n
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13437
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
115
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
530
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13948
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
117
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
19
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13758
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
116
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
209
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
13892
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
117
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
75
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
7016
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
51
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
6951
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
66
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
7385
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
54
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
6582
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
63
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
7700
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
56
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
6267
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
61
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
12861
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
notsig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
108
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:right;">
1106
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:left;">
sig
</td>
<td style="text-align:left;">
y
</td>
<td style="text-align:right;">
9
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>pairwisecrispr%&gt;% kable(., caption= &quot;Summary of chisqure values between numbers of sigDE and non sigDE by treatment&quot; )%&gt;% 
  kable_paper(&quot;striped&quot;, full_width = FALSE) %&gt;%  
  kable_styling(full_width = FALSE, position = &quot;left&quot;,bootstrap_options = c(&quot;striped&quot;),font_size = 18) %&gt;% 
  scroll_box(width = &quot;60%&quot;, height = &quot;400px&quot;)</code></pre>
<div
style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:400px; overflow-x: scroll; width:60%; ">
<table class=" lightable-paper lightable-striped table table-striped" style="font-family: &quot;Arial Narrow&quot;, arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; font-size: 18px; width: auto !important; ">
<caption style="font-size: initial !important;">
Summary of chisqure values between numbers of sigDE and non sigDE by
treatment
</caption>
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
time
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
id
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
pvalue
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:right;">
0.2387271
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:right;">
0.6897318
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:right;">
0.5684613
</td>
</tr>
<tr>
<td style="text-align:left;">
3_hours
</td>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:right;">
0.4267580
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Daunorubicin
</td>
<td style="text-align:right;">
0.1523968
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Doxorubicin
</td>
<td style="text-align:right;">
0.1470074
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Epirubicin
</td>
<td style="text-align:right;">
0.1155814
</td>
</tr>
<tr>
<td style="text-align:left;">
24_hours
</td>
<td style="text-align:left;">
Mitoxantrone
</td>
<td style="text-align:right;">
0.9280448
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>crisp_pair_mat &lt;- pairwisecrispr %&gt;%
  mutate(neg.log.pvalue= (-1*log(pvalue))) %&gt;% 
  mutate(time= case_match(time, &#39;3_hours&#39;~&#39;3_hrs&#39;, &#39;24_hours&#39;~&#39;24_hrs&#39;,.default = id)) %&gt;% 
  mutate(id =case_match( id, &#39;Daunorubicin&#39;~&#39;DNR&#39;,   &#39;Doxorubicin&#39;~&#39;DOX&#39; ,&#39;Epirubicin&#39;~&#39;EPI&#39; , &#39;Mitoxantrone&#39; ~ &#39;MTX&#39;,.default = id)) %&gt;% 
  unite(&#39;pairset&#39;,time,id ) %&gt;% 
  column_to_rownames(&#39;pairset&#39;) %&gt;% dplyr::select(neg.log.pvalue) %&gt;% as.matrix()
    
Heatmap( crisp_pair_mat, name = &quot;Doxcrispr pairwise enrichment \nchi square -log p values&quot;, 
         cluster_rows = FALSE, 
         cluster_columns = FALSE, 
         col=col_fun5, column_names_rot = 0,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(crisp_pair_mat[i, j] &gt; -log(0.05))
            grid.text(&quot;*&quot;, x, y, gp = gpar(fontsize = 20))
})</code></pre>
<p><img src="figure/other_analysis.Rmd/crisper%20DEG-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ComplexHeatmap_2.12.1 broom_1.0.5           kableExtra_1.3.4     
 [4] sjmisc_2.8.9          scales_1.2.1          ggpubr_0.6.0         
 [7] cowplot_1.1.1         RColorBrewer_1.1-3    biomaRt_2.52.0       
[10] ggsignif_0.6.4        lubridate_1.9.2       forcats_1.0.0        
[13] stringr_1.5.0         dplyr_1.1.2           purrr_1.0.1          
[16] readr_2.1.4           tidyr_1.3.0           tibble_3.2.1         
[19] ggplot2_3.4.2         tidyverse_2.0.0       limma_3.52.4         
[22] workflowr_1.7.0      

loaded via a namespace (and not attached):
  [1] colorspace_2.1-0       rjson_0.2.21           sjlabelled_1.2.0      
  [4] rprojroot_2.0.3        circlize_0.4.15        XVector_0.36.0        
  [7] GlobalOptions_0.1.2    fs_1.6.2               clue_0.3-64           
 [10] rstudioapi_0.14        farver_2.1.1           bit64_4.0.5           
 [13] AnnotationDbi_1.58.0   fansi_1.0.4            xml2_1.3.4            
 [16] codetools_0.2-19       doParallel_1.0.17      cachem_1.0.8          
 [19] knitr_1.43             jsonlite_1.8.5         cluster_2.1.4         
 [22] dbplyr_2.3.2           png_0.1-8              compiler_4.2.2        
 [25] httr_1.4.6             backports_1.4.1        fastmap_1.1.1         
 [28] cli_3.6.1              later_1.3.1            htmltools_0.5.5       
 [31] prettyunits_1.1.1      tools_4.2.2            gtable_0.3.3          
 [34] glue_1.6.2             GenomeInfoDbData_1.2.8 rappdirs_0.3.3        
 [37] Rcpp_1.0.10            carData_3.0-5          Biobase_2.56.0        
 [40] jquerylib_0.1.4        vctrs_0.6.3            Biostrings_2.64.1     
 [43] svglite_2.1.1          iterators_1.0.14       insight_0.19.2        
 [46] xfun_0.39              ps_1.7.5               rvest_1.0.3           
 [49] timechange_0.2.0       lifecycle_1.0.3        rstatix_0.7.2         
 [52] XML_3.99-0.14          getPass_0.2-2          zlibbioc_1.42.0       
 [55] hms_1.1.3              promises_1.2.0.1       parallel_4.2.2        
 [58] yaml_2.3.7             curl_5.0.1             memoise_2.0.1         
 [61] sass_0.4.6             stringi_1.7.12         RSQLite_2.3.1         
 [64] highr_0.10             S4Vectors_0.34.0       foreach_1.5.2         
 [67] BiocGenerics_0.42.0    filelock_1.0.2         shape_1.4.6           
 [70] GenomeInfoDb_1.32.4    matrixStats_1.0.0      rlang_1.1.1           
 [73] pkgconfig_2.0.3        systemfonts_1.0.4      bitops_1.0-7          
 [76] evaluate_0.21          labeling_0.4.2         bit_4.0.5             
 [79] processx_3.8.1         tidyselect_1.2.0       magrittr_2.0.3        
 [82] R6_2.5.1               IRanges_2.30.1         generics_0.1.3        
 [85] DBI_1.1.3              pillar_1.9.0           whisker_0.4.1         
 [88] withr_2.5.0            KEGGREST_1.36.3        abind_1.4-5           
 [91] RCurl_1.98-1.12        crayon_1.5.2           car_3.1-2             
 [94] utf8_1.2.3             BiocFileCache_2.4.0    tzdb_0.4.0            
 [97] rmarkdown_2.22         GetoptLong_1.0.5       progress_1.2.2        
[100] blob_1.2.4             callr_3.7.3            git2r_0.32.0          
[103] digest_0.6.31          webshot_0.5.4          httpuv_1.6.11         
[106] stats4_4.2.2           munsell_0.5.0          viridisLite_0.4.2     
[109] bslib_0.5.0           </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
