<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="ERM" />


<title>Run1_analysis</title>

<script src="site_libs/header-attrs-2.19/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cardiotoxicity</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/reneeisnowhere/Cardiotoxicity">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Run1_analysis</h1>
<h4 class="author">ERM</h4>
<h4 class="date">20230109`</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-01-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Cardiotoxicity/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230109code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230109)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230109code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230109)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereCardiotoxicitytree93c440954bacc490b38f71f2247daa5f3b1f298etargetblank93c4409a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/tree/93c440954bacc490b38f71f2247daa5f3b1f298e" target="_blank">93c4409</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereCardiotoxicitytree93c440954bacc490b38f71f2247daa5f3b1f298etargetblank93c4409a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/tree/93c440954bacc490b38f71f2247daa5f3b1f298e" target="_blank">93c4409</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/data_outline.txt
    Ignored:    data/mymatrix.RDS

Untracked files:
    Untracked:  reneebasecode.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/RNAseqrun_1_analysis.Rmd</code>)
and HTML (<code>docs/RNAseqrun_1_analysis.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/93c440954bacc490b38f71f2247daa5f3b1f298e/analysis/RNAseqrun_1_analysis.Rmd" target="_blank">93c4409</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-12
</td>
<td>
try again
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/8ebcfd816046362034045a93dc8c4f48a3de8463/docs/RNAseqrun_1_analysis.html" target="_blank">8ebcfd8</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/56eab44892e7fa8fc5580e15e5a68b48e6c656bf/analysis/RNAseqrun_1_analysis.Rmd" target="_blank">56eab44</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-12
</td>
<td>
end of day
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/ae8266e811dcf8952b7e8770b17e97d68c4e0f80/analysis/RNAseqrun_1_analysis.Rmd" target="_blank">ae8266e</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-12
</td>
<td>
updating the rmarkdown
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/f6c386f7517825baaed8d426adf7669097464b37/docs/RNAseqrun_1_analysis.html" target="_blank">f6c386f</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/82fa5e5431108cb03183140f60e98e3c476fa0b0/analysis/RNAseqrun_1_analysis.Rmd" target="_blank">82fa5e5</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-12
</td>
<td>
addressing functional errors
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/83aef3a63fac20b3ccdd14638bae7a4ffe9d5d28/analysis/RNAseqrun_1_analysis.Rmd" target="_blank">83aef3a</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-12
</td>
<td>
addressing functional errors
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/b456c92f36637066999a4174435180efdca41ea1/analysis/RNAseqrun_1_analysis.Rmd" target="_blank">b456c92</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-11
</td>
<td>
wflow_git_commit(all = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/943e5659f697d60f7c33817cb25f30169f042a98/analysis/RNAseqrun_1_analysis.Rmd" target="_blank">943e565</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-11
</td>
<td>
updating the counts filtering
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/921e76f66838e153a0821c2328189b6ba8da8f1e/analysis/RNAseqrun_1_analysis.Rmd" target="_blank">921e76f</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-11
</td>
<td>
updating the counts filtering
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/98fe131fcc1e93a32fd9383a1cb6941a00cf7735/analysis/RNAseqrun_1_analysis.Rmd" target="_blank">98fe131</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-11
</td>
<td>
wflow_publish(c("analysis/license.Rmd", "analysis/index.Rmd"))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/364d73dc015fa6decf61337c3754634ee5fc29fe/analysis/RNAseqrun_1_analysis.Rmd" target="_blank">364d73d</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-11
</td>
<td>
wflow_git_commit(all = TRUE)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/be977ac06ad7e9eca72d1ac799267ab39c8f7238/analysis/RNAseqrun_1_analysis.Rmd" target="_blank">be977ac</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-10
</td>
<td>
testing interface
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/bdb802d6f2220ae88234bd9e6ec49484e386ce36/analysis/RNAseqrun_1_analysis.Rmd" target="_blank">bdb802d</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-10
</td>
<td>
Starting the editing
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/ec76d0d2a708607500b6c6cc414c5b3e24385f2c/analysis/RNAseqrun_1_analysis.Rmd" target="_blank">ec76d0d</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-01-09
</td>
<td>
Having issues with rmark
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Hello! I am first loading all the beautiful libraries I will
need.</p>
<pre class="r"><code>library(Biobase)
library(edgeR)
library(limma)
library(RColorBrewer)
library(mixOmics)
library(VennDiagram)
library(HTSFilter)
library(ggplot2)
library(gridExtra)
library(reshape2)
library(devtools)
library(AnnotationHub)
library(tidyverse)
library(scales)
library(biomaRt)
library(Homo.sapiens)
library(cowplot)</code></pre>
<p>The next step is to load all the data I will be using. Currently, I
am not posting the raw data, but I will release in the future.</p>
<pre class="r"><code>theme_set(theme_bw(base_size = 16))

pca_plot &lt;- function(df, col_var = NULL, shape_var = NULL, title = &quot;&quot;) {
  ggplot(df) + geom_point(aes_string(x = &quot;PC1&quot;, y = &quot;PC2&quot;, color = col_var,
                                     shape = shape_var), size = 5) +
    labs(title = title, x = &quot;PC1&quot;, y = &quot;PC2&quot;)
}

pca_var_plot &lt;- function(pca) {
  # x: class == prcomp
  pca.var &lt;- pca$sdev^2
  pca.prop &lt;- pca.var / sum(pca.var)
  var.plot &lt;- qplot(PC, prop, data = data.frame(PC = 1:length(pca.prop),
                                                prop = pca.prop)) +
    labs(title = &#39;Variance contributed by each PC&#39;,
         x = &#39;PC&#39;, y = &#39;Proportion of variance&#39;)
}

calc_pca &lt;- function(x) {
  # Performs principal components analysis with prcomp
  # x: a sample-by-gene numeric matrix
  prcomp(x, scale. = TRUE, retx = TRUE)
}

get_regr_pval &lt;- function(mod) {
  # Returns the p-value for the Fstatistic of a linear model
  # mod: class lm
  stopifnot(class(mod) == &quot;lm&quot;)
  fstat &lt;- summary(mod)$fstatistic
  pval &lt;- 1 - pf(fstat[1], fstat[2], fstat[3])
  return(pval)
}

plot_versus_pc &lt;- function(df, pc_num, fac) {
  # df: data.frame
  # pc_num: numeric, specific PC for plotting
  # fac: column name of df for plotting against PC
  pc_char &lt;- paste0(&quot;PC&quot;, pc_num)
  # Calculate F-statistic p-value for linear model
  pval &lt;- get_regr_pval(lm(df[, pc_char] ~ df[, fac]))
  if (is.numeric(df[, f])) {
    ggplot(df, aes_string(x = f, y = pc_char)) + geom_point() +
      geom_smooth(method = &quot;lm&quot;) + labs(title = sprintf(&quot;p-val: %.2f&quot;, pval))
  } else {
    ggplot(df, aes_string(x = f, y = pc_char)) + geom_boxplot() +
      labs(title = sprintf(&quot;p-val: %.2f&quot;, pval))
  }
}</code></pre>
<p>This is how I retrieved the gene symbols.</p>
<pre class="r"><code> ###now we add genenames to the geneid###
 
geneid &lt;- rownames(mymatrix) ### pulls the names we have in the counts file
genes &lt;- select(Homo.sapiens, keys=geneid, columns=c(&quot;SYMBOL&quot;),
  keytype=&quot;ENTREZID&quot;)</code></pre>
<pre><code>&#39;select()&#39; returned 1:1 mapping between keys and columns</code></pre>
<pre class="r"><code>genes &lt;- genes[!duplicated(genes$ENTREZID),]
 

mymatrix$genes &lt;- genes</code></pre>
<p>##okay, now to transform data</p>
<p>Filtering the genes that are lowly expressed using several
methods.</p>
<p>First method, removing only those rows with zero counts across all
samples.</p>
<pre class="r"><code>table(rowSums(mymatrix$counts==0)==72)</code></pre>
<pre><code>
FALSE  TRUE 
24931  3464 </code></pre>
<p>This filtering would leave 24931 genes and remove 3464, That is too
many leftover genes!</p>
<p>So now to try something a little more stringent using the built in
function from the edgeR package.</p>
<pre class="r"><code>keep &lt;- filterByExpr.DGEList(mymatrix, group = group)
filter_test &lt;- mymatrix[keep, , keep.lib.sizes=FALSE]
dim(filter_test)</code></pre>
<pre><code>[1] 14448    72</code></pre>
<p>This method effectively uses a cutoff off that leaves 14448 genes.
The cutoff is determined by keeping genes that have a count-per-million
(CPM) above 10, (the default minimum set) in 6 samples. A set is
determined using the design matrix.<br />
For my design, I grouped my 72 samples into sets of 6, one set includes
each individual + a specific treatment + a specific time.</p>
<p>The beginning cutoff-standard in our lab is to start by using the
rowMeans &gt;0 cutoff on the log10 of cpm.</p>
<pre class="r"><code>cpm &lt;- cpm(mymatrix)
lcpm &lt;- cpm(mymatrix, log=TRUE)  ### for determining the basic cutoffs
dim(lcpm)</code></pre>
<pre><code>[1] 28395    72</code></pre>
<pre class="r"><code>L &lt;- mean(mymatrix$samples$lib.size) * 1e-6
M &lt;- median(mymatrix$samples$lib.size) * 1e-6
c(L, M)</code></pre>
<pre><code>[1] 4.679061 4.494188</code></pre>
<pre class="r"><code>filcpm_matrix &lt;- subset(lcpm, (rowMeans(lcpm)&gt;0))

dim(filcpm_matrix)</code></pre>
<pre><code>[1] 14823    72</code></pre>
<pre class="r"><code>##method 2 with rowMeans
row_means &lt;- rowMeans(lcpm)
x &lt;- mymatrix[row_means &gt; 0,]
dim(x)</code></pre>
<pre><code>[1] 14823    72</code></pre>
<p>Both of the above methods leave 14823 genes from 28,395. I prefer the
second method, which keeps the DGEList format of the data.</p>
<p><img src="figure/RNAseqrun_1_analysis.Rmd/plots%20of%20before%20and%20after%20filter-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqrun_1_analysis.Rmd/plots%20of%20before%20and%20after%20filter-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqrun_1_analysis.Rmd/plots%20of%20before%20and%20after%20filter-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>x_axis_labels=function(labels,every_nth=1,...) {
    axis(side=1,at=seq_along(labels),labels=F)
    text(x=(seq_along(labels))[seq_len(every_nth)==1],
        y=par(&quot;usr&quot;)[3]-0.075*(par(&quot;usr&quot;)[4]-par(&quot;usr&quot;)[3]),
        labels=labels[seq_len(every_nth)==1],xpd=TRUE,...)
}
boxplot(lcpm, main = &quot;Boxplots of log cpm per sample&quot;,xaxt = &quot;n&quot;, xlab= &quot;&quot;)#, xlab=mymatrix$samples, srt =85,  adj=1, cex = 0.5)
x_axis_labels(labels = samplenames, every_nth = 1, adj=1, srt =90, cex =0.4)</code></pre>
<p><img src="figure/RNAseqrun_1_analysis.Rmd/boxplot%20counts%20per%20sample-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>boxplot(filcpm_matrix, main =&quot;boxplots of log cpm per sample filtered&quot;,xaxt = &quot;n&quot;, xlab=&quot;&quot;)
x_axis_labels(labels = samplenames, every_nth = 1, adj=1, srt =90, cex =0.4)</code></pre>
<p><img src="figure/RNAseqrun_1_analysis.Rmd/boxplot%20counts%20per%20sample-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>now I will produce the RIN x sample plots:###
<img src="figure/RNAseqrun_1_analysis.Rmd/sample%20quality%20checks-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqrun_1_analysis.Rmd/sample%20quality%20checks-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqrun_1_analysis.Rmd/sample%20quality%20checks-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqrun_1_analysis.Rmd/sample%20quality%20checks-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqrun_1_analysis.Rmd/sample%20quality%20checks-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>Principal Component Analysis</p>
<p>PCA was done using code adopted from J. Blischak.</p>
<pre><code>

###  Daunorubicin </code></pre>
<pre><code>Warning: `aes_string()` was deprecated in ggplot2 3.0.0.
ℹ Please use tidy evaluation ideoms with `aes()`</code></pre>
<pre><code>Warning: Using shapes for an ordinal variable is not advised</code></pre>
<p><img src="figure/RNAseqrun_1_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

###  Doxorubicin </code></pre>
<pre><code>Warning: Using shapes for an ordinal variable is not advised</code></pre>
<p><img src="figure/RNAseqrun_1_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

###  Epirubicin </code></pre>
<pre><code>Warning: Using shapes for an ordinal variable is not advised</code></pre>
<p><img src="figure/RNAseqrun_1_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

###  Mitoxantrone </code></pre>
<pre><code>Warning: Using shapes for an ordinal variable is not advised</code></pre>
<p><img src="figure/RNAseqrun_1_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

###  Trastuzumab </code></pre>
<pre><code>Warning: Using shapes for an ordinal variable is not advised</code></pre>
<p><img src="figure/RNAseqrun_1_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

###  Vehicle </code></pre>
<pre><code>Warning: Using shapes for an ordinal variable is not advised</code></pre>
<p><img src="figure/RNAseqrun_1_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Warning: `qplot()` was deprecated in ggplot2 3.4.0.</code></pre>
<p><img src="figure/RNAseqrun_1_analysis.Rmd/all-samples-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>facs &lt;- c(&quot;indv&quot;, &quot;drug&quot;, &quot;time&quot;) 
names(facs) &lt;- c(&quot;Individual&quot;,&quot;Treatment&quot;, &quot;Time&quot;) 
for (f in facs) { 
  # PC1 v PC2
  pca_plot(pca_all_anno,
    col_var = f,
    shape_var = indv,
    title =names(facs)[which(facs == f)]) 
  print(last_plot())
  f_v_pc1 &lt;-arrangeGrob(plot_versus_pc(pca_all_anno, 1, f)) 
  f_v_pc2 &lt;-arrangeGrob(plot_versus_pc(pca_all_anno, 2, f)) 
grid.arrange(f_v_pc1,f_v_pc2, ncol = 2)
}</code></pre>
<p><img src="figure/RNAseqrun_1_analysis.Rmd/variance%20by%20treatment%20time%20and%20individual-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqrun_1_analysis.Rmd/variance%20by%20treatment%20time%20and%20individual-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqrun_1_analysis.Rmd/variance%20by%20treatment%20time%20and%20individual-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqrun_1_analysis.Rmd/variance%20by%20treatment%20time%20and%20individual-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqrun_1_analysis.Rmd/variance%20by%20treatment%20time%20and%20individual-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqrun_1_analysis.Rmd/variance%20by%20treatment%20time%20and%20individual-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>{r Differential expression, eval=FALSE, include=FALSE} mm &lt;-
model.matrix(~0 +drug +time) colnames(mm) &lt;- gsub(“group”, ““,
colnames(mm)) mm2 &lt;- model.matrix(~0 + label) colnames(mm2) &lt;-
c(”A3”, “X3”, “E3”,“M3”,“T3”, “V3”,“A24”, “X24”, “E24”,“M24”,“T24”,
“V24”)</p>
<p>mm y &lt;- voom(x, mm, plot =TRUE) dim(y)</p>
<p>colnames(mm2) &lt;- c(“A24”, “X24”, “E24”,“M24”,“T24”, “V24”,“A3”,
“X3”, “E3”,“M3”,“T3”, “V3”) ### renamed to make contrasts easier 1-6 is
24 hour, 7-12 is 3 hour A X E M T V order</p>
<p>##now apply correlations and things I chose block of indv in the mm.
with my model only looking and drug and time contrasts y2 &lt;- voom(x,
mm2) y &lt;- voom(x, mm)##not plotting corfit &lt;-
duplicateCorrelation(y, mm, block = indv) corfit2 &lt;-
duplicateCorrelation(y2, mm2, block = indv) v &lt;- voom(x, mm, block =
indv, correlation = corfit<span class="math inline">\(consensus) v2
&lt;- voom(x, mm2, block = indv, correlation =
corfit2\)</span>consensus) fit &lt;- lmFit(v, mm, block = indv,
correlation = corfit<span class="math inline">\(consensus) fit2 &lt;-
lmFit(v2, mm2, block = indv, correlation = corfit2\)</span>consensus)
fit2 mm2 # Make contrasts cm &lt;- makeContrasts( VvDA = drugVehicle
-drugDaunorubicin, VvDX = drugVehicle-drugDoxorubicin, VvEP =
drugVehicle-drugEpirubicin, VvMT = drugVehicle-drugMitoxantrone, VvTR =
drugVehicle-drugTrastuzumab, # VvDA24 = drugVehicleA-B, # VvDX24=
drugVehicleA-C, # VvEP24= A-D, # VvMT24= B-C, # VvTR24= B-D, levels=mm)
cm2 &lt;- makeContrasts( V.DA = V3 - A3, V.DX = V3 - X3, V.EP = V3 - E3,
V.MT = V3 - M3, V.TR = V3 - T3, V.DA24 = V24-A24, V.DX24= V24-X24,
V.EP24= V24-E24, V.MT24= V24-M24, V.TR24= V24-T24, levels = mm2)</p>
<p>vfit &lt;- lmFit(y, mm) vfit2 &lt;- lmFit(y2, mm2) vfit &lt;-
contrasts.fit(vfit, contrasts=cm) vfit2&lt;- contrasts.fit(vfit2,
contrasts=cm2) efit &lt;- eBayes(vfit) efit2 &lt;- eBayes(vfit2)
plotSA(efit)</p>
<p>plotSA(efit2)</p>
<p>V.DA.top= topTable(efit2, coef=1, adjust=“BH”, number=Inf,
sort.by=“p”) V.DX.top= topTable(efit2, coef=2, adjust=“BH”, number=Inf,
sort.by=“p”) V.EP.top= topTable(efit2, coef=3, adjust=“BH”, number=Inf,
sort.by=“p”) V.MT.top= topTable(efit2, coef=4, adjust=“BH”, number=Inf,
sort.by=“p”) V.TR.top= topTable(efit2, coef=5, adjust=“BH”, number=Inf,
sort.by=“p”) V.DA24.top= topTable(efit2, coef=6, adjust=“BH”,
number=Inf, sort.by=“p”) V.DX24.top= topTable(efit2, coef=7,
adjust=“BH”, number=Inf, sort.by=“p”) V.EP24.top= topTable(efit2,
coef=8, adjust=“BH”, number=Inf, sort.by=“p”) V.MT24.top=
topTable(efit2, coef=9, adjust=“BH”, number=Inf, sort.by=“p”)
V.TR24.top= topTable(efit2, coef=10, adjust=“BH”, number=Inf,
sort.by=“p”)</p>
<p>sigVDA3 = V.DA.top[V.DA.top<span class="math inline">\(adj.P.Val &lt;
.1 , ] sigVDX3 = V.DX.top[V.DX.top\)</span>adj.P.Val &lt; .1 , ] sigVEP3
= V.EP.top[V.EP.top<span class="math inline">\(adj.P.Val &lt; .1 , ]
sigVMT3 = V.MT.top[V.MT.top\)</span>adj.P.Val &lt; .1 , ] sigVTR3 =
V.TR.top[V.TR.top<span class="math inline">\(adj.P.Val &lt; .1 , ]
sigVDA24 = V.DA24.top[V.DA24.top\)</span>adj.P.Val &lt; .1 , ] sigVDX24
= V.DX24.top[V.DX24.top<span class="math inline">\(adj.P.Val &lt; .1 , ]
sigVEP24 = V.EP24.top[V.EP24.top\)</span>adj.P.Val &lt; .1 , ] sigVMT24
= V.MT24.top[V.MT24.top<span class="math inline">\(adj.P.Val &lt; .1 , ]
sigVTR24 = V.TR24.top[V.TR24.top\)</span>adj.P.Val &lt; .1 , ] siglist
&lt;- c(“sigVDA3”, “sigVDX3”, “sigVEP3”, “sigVMT3”, “sigVTR3”,
“sigVDA24”, “sigVDX24”, “sigVEP24”, “sigVMT24”, “sigVTR24”)</p>
<div id="trying-to-graph-the-genes-from-siglists"
class="section level3">
<h3>trying to graph the genes from siglists</h3>
<p>##first I will pull all data into a frame for ggplots ##the
voomtransformed data is in y2 (line 226ish above) countstotal &lt;-
y2<span class="math inline">\(E colnames(countstotal) &lt;- smlabel
boxplot(countstotal) boxplot(Da24counts\)</span>Tags)</p>
</div>
<div id="i-made-an-object-called-countstotal-from-the-elist-y2"
class="section level3">
<h3>I made an object called countstotal from the EList y2</h3>
</div>
<div
id="subsetting-out-specific-columns-for-ggplot-graphing-of-specific-counts-comparisons"
class="section level3">
<h3>(subsetting out specific columns for ggplot graphing of specific
counts comparisons)</h3>
<p>Da24counts &lt;-
as.data.frame(as.table(countstotal[,c(7,12,19,24,31,36,43,48,55,60,67,72)]))
Dx24counts &lt;-
as.data.frame(as.table(countstotal[,c(8,12,20,24,32,36,44,48,56,60,68,72)]))
Ep24counts &lt;-
as.data.frame(as.table(countstotal[,c(9,12,21,24,33,36,45,48,57,60,69,72)]))
Mt24counts &lt;-
as.data.frame(as.table(countstotal[,c(10,12,22,24,34,36,46,48,58,60,70,72)]))
TR24counts &lt;-
as.data.frame(as.table(countstotal[,c(11,12,23,24,35,36,47,48,59,60,71,72)]))
Da3counts &lt;-
as.data.frame(as.table(countstotal[,c(1,6,13,18,25,30,37,42,49,54,61,66)]))
Dx3counts &lt;-
as.data.frame(as.table(countstotal[,c(2,6,14,18,26,30,38,42,50,54,62,66)]))
Ep3counts &lt;-
as.data.frame(as.table(countstotal[,c(3,6,15,18,27,30,39,42,51,54,63,66)]))
Mt3counts &lt;-
as.data.frame(as.table(countstotal[,c(4,6,16,18,28,30,40,42,52,54,64,66)]))
TR3counts &lt;-
as.data.frame(as.table(countstotal[,c(5,6,17,18,29,30,41,42,53,54,65,66)]))
library(cowplot) ###then convert to data frame I think
colnames(Da24counts) &lt;- c(“ENTREZID”, “Samples”, “Counts”) #
grep(’da<span class="math inline">\(&#39; %in%
Da24counts\)</span>Samples,) # Da24counts$Samples[grep(“da”)]</p>
<div id="graph-the-gene-expression" class="section level4">
<h4>graph the gene expression</h4>
<p>#make the list of gene wanted</p>
<p>apple &lt;- c(‘CDKN1A’,“BBC3”, “MDM2”, “BCL2”, ‘IGFBP3’, “BAX”,
“GPX1”, ‘MFN2’,“HAS3”,‘CYBA’,‘TOP2B’, ‘TP53’, ‘ABCC1’, ‘ABCC5’, ‘RRAGD’,
‘DUSP13’, ‘NDUFAF1’, ‘TDP2’, ‘TXNIP’,‘BRCA1’, ‘CTCF’,‘RAD21’,‘RYR2’)
##find the index number for each gene indices &lt;- match(apple,
mymatrix<span class="math inline">\(genes\)</span>SYMBOL)</p>
<p>###subset the matrix entreset &lt;- mymatrix<span
class="math inline">\(genes\)</span>ENTREZID[indices] entreset gnames
&lt;- cbind(entreset,apple) ###voila!</p>
<p>colnames(gnames) &lt;- c(“ENTREZID”,“SYMBOL” ) #which(gnames<span
class="math inline">\(SYMBOL== sigVDA24\)</span>SYMBOL) #
match(Da24counts[%in% x,]) #i &lt;- which(x<span
class="math inline">\(genes\)</span>ENTREZID %in%
Da24counts$ENTREZID)</p>
<p>###now to ggplot### for (gn in indices){ print(Da24counts %&gt;%
filter(Tags == gn) %&gt;% ggplot(aes(x=as.factor(Samples), y=Freq))+
geom_boxplot(aes(fill = as.factor(Samples)))+ theme_cowplot(font_size
=12,)+</p>
<pre><code>ggtitle(paste0(gnames$gnames,&quot; expression in Daunorubicin at 24 hours&quot;))+

labs(y = &quot;Log2(cpm)&quot;,x= &quot;&quot;, fill = &quot;Treatment&quot;))</code></pre>
<p>}</p>
<p>plot_function &lt;- function(index, df) { for (i in index) { data
&lt;- df %&gt;% filter(Tags == i) if (!nrow(data)) next print(data
%&gt;% ggplot(aes(x=as.factor(Samples), y=Freq))+ geom_boxplot(aes(fill
= as.factor(Samples)))+ theme_cowplot(font_size =12,)+</p>
<pre><code>        ggtitle(paste0(i,&quot; expression &quot;))+
        
        labs(y = &quot;Log2(cpm)&quot;,x= &quot;&quot;, fill = &quot;Treatment&quot;)) 
}</code></pre>
<p>} gname &lt;- gnames$ENTREZID gnames &lt;- as.data.frame(gnames)
plotDA_function(gname,TR3counts)</p>
<p>plot_saving(gname, Da3counts)</p>
<p>print(gnames[2,2])</p>
<p>plot_saving &lt;- function(index, df) { for (i in index) { data &lt;-
df %&gt;% filter(Tags==i) if (!nrow(data)) next p &lt;- data %&gt;%
ggplot(aes(x=as.factor(Samples), y=Freq))+ geom_boxplot(aes(fill =
as.factor(Samples)))+ theme_cowplot(font_size =12,)+</p>
<pre><code>        ggtitle(paste0(i,&quot; expression &quot;))+
        
        labs(y = &quot;Log2(cpm)&quot;,x= &quot;&quot;, fill = &quot;Treatment&quot;)

plot_name &lt;- c(paste(i,substr(df,1,5),sep=&quot;_&quot;))
ggsave(file.path(&quot;~/Ward Lab/Cardiotoxicity/Data/R code output&quot;), filename= (plot_name,&quot;.pdf&quot;), plot = p)
}</code></pre>
<p>} plotDA_function &lt;- function(indices, Da24counts) { for (i in
indices) { data &lt;- Da24counts %&gt;% filter(Tags==i) if (!nrow(data))
next p &lt;- data %&gt;% ggplot(aes(x=as.factor(Samples), y=Freq))+
geom_boxplot(aes(fill = as.factor(Samples)))+ theme_cowplot(font_size
=12,)+</p>
<pre><code>  ggtitle(paste0(i,&quot; expression&quot;))+
  
  labs(y = &quot;Log2(cpm)&quot;,x= &quot;&quot;, fill = &quot;Treatment&quot;)
ggsave(file.path(&quot;~/Ward Lab/Cardiotoxicity/Data/R_code_output/3hours&quot;, paste0(deparse(substitute(Da24counts)), &quot;_&quot;, i, &quot;.pdf&quot;)), plot = p)</code></pre>
<p>} }</p>
<p>Da24counts %&gt;% filter(Tags==gnames$ENTREZID) %&gt;%
ggplot(aes(x=as.factor(Samples), y=Freq))+ geom_boxplot(aes(fill =
as.factor(Samples)))+ #ggplot(Da24counts, aes(x=as.factor(Samples),
y=Freq))+ #geom_boxplot(aes(fill = as.factor(Samples))) +
theme_cowplot(font_size =12,)+</p>
<p>ggtitle(paste0(“BRCA1 expression in Daunorubicin at 24 hours”))+</p>
<p>labs(y = “Log2(cpm)”,x= ““, fill =”Treatment”)</p>
<p>Dx24counts %&gt;% filter(Tags==7155) %&gt;%
ggplot(aes(x=as.factor(Samples), y=Freq))+ geom_boxplot(aes(fill =
as.factor(Samples)))+ #ggplot(Da24counts, aes(x=as.factor(Samples),
y=Freq))+ #geom_boxplot(aes(fill = as.factor(Samples))) +
theme_cowplot(font_size =12,)+</p>
<p>ggtitle(“Top2b expression in Doxorubicin at 24 hours”)+</p>
<p>labs(y = “Log2(cpm)”,x= ““, fill =”Treatment”)</p>
<p>Ep24counts %&gt;% filter(Tags==7155) %&gt;%
ggplot(aes(x=as.factor(Samples), y=Freq))+ geom_boxplot(aes(fill =
as.factor(Samples)))+ #ggplot(Da24counts, aes(x=as.factor(Samples),
y=Freq))+ #geom_boxplot(aes(fill = as.factor(Samples))) +
theme_cowplot(font_size =12,)+</p>
<p>ggtitle(“Top2b expression in Epirubicin at 24 hours”)+</p>
<p>labs(y = “Log2(cpm)”,x= ““, fill =”Treatment”)</p>
<p>Mt24counts %&gt;% filter(Tags==7155) %&gt;%
ggplot(aes(x=as.factor(Samples), y=Freq))+ geom_boxplot(aes(fill =
as.factor(Samples)))+ #ggplot(Da24counts, aes(x=as.factor(Samples),
y=Freq))+ #geom_boxplot(aes(fill = as.factor(Samples))) +
theme_cowplot(font_size =12,)+</p>
<p>ggtitle(“Top2b expression in Mito at 24 hours”)+</p>
<p>labs(y = “Log2(cpm)”,x= ““, fill =”Treatment”)</p>
<p>TR24counts %&gt;% filter(Tags==7155) %&gt;%
ggplot(aes(x=as.factor(Samples), y=Freq))+ geom_boxplot(aes(fill =
as.factor(Samples)))+ #ggplot(Da24counts, aes(x=as.factor(Samples),
y=Freq))+ #geom_boxplot(aes(fill = as.factor(Samples))) +
theme_cowplot(font_size =12,)+</p>
<p>ggtitle(“Top2b expression in Trastuzumab at 24 hours”)+</p>
<p>labs(y = “Log2(cpm)”,x= ““, fill =”Treatment”)</p>
<p>plot(Mt24counts) plot(Da24counts) venn.diagram(x =
list(“Daunorubicin” = rownames(sigVDA24), “Filtering” =
rownames(sigVDX24)), fill = brewer.pal(3, “Set2”)[1:2], filename = NULL)
x<span class="math inline">\(samples\)</span>group vd</p>
<p>coef(efit2)</p>
<p>dim(efit2) sigVDA24</p>
<p>V.DA24.top sum &lt;- summary(decideTests(efit2)) sum</p>
</div>
</div>
<div id="next-is-way-to-add-the-names-to-each-sample-in-the-lcpm-matrix"
class="section level3">
<h3>next is way to add the names to each sample in the lcpm matrix</h3>
<p>colnames(lcpm) &lt;- label # de.common &lt;- which(#addname of the
list here using sig…)## # [,1]!=0 &amp; dt[,2]!=0) #
length(de.common)</p>
<p>####volcanoplot code: (go lower for ENHANCED!)</p>
<p>V.DA.volp &lt;- volcanoplot(efit2, coef = 1,style = “p-value”,
highlight = 0.05, names = fit<span
class="math inline">\(genes\)</span>SYMBOL, hl.col = “red”, xlab = “Log2
Fold Change”, ylab = NULL, pch = 16, cex = 0.35, main = “3 hour V DA” )
V.DA.volp + geom_hline(yintercept = 2, linetype = “dashed”) +
geom_vline(xintercept = c(log2(0.5), log2(2)), linetype = “dashed”)
V.DX.volp &lt;- volcanoplot(efit2,coef = 2, style = “p-value”, highlight
= 40, names = fit<span class="math inline">\(genes\)</span>SYMBOL,
hl.col = “red”, xlab = “Log2 Fold Change”, ylab = NULL, pch = 16, cex =
0.35, main = “3 hour V DX” ) V.EP.volp &lt;- volcanoplot(efit2,coef = 3,
style = “p-value”, highlight = 40, names = fit<span
class="math inline">\(genes\)</span>SYMBOL, hl.col = “red”, xlab = “Log2
Fold Change”, ylab = NULL, pch = 16, cex = 0.35, main = “3 hour V EP” )
V.MT.volp = volcanoplot(efit2,coef = 4, style = “p-value”, highlight =
100, names = fit<span class="math inline">\(genes\)</span>SYMBOL, hl.col
= “red”, xlab = “Log2 Fold Change”, ylab = NULL, pch = 16, cex = 0.35,
main = “3 hour V MT” ) V.TR.volp = volcanoplot(efit2,coef = 5, style =
“p-value”, highlight = 100, names = fit<span
class="math inline">\(genes\)</span>SYMBOL, hl.col = “red”, xlab = “Log2
Fold Change”, ylab = NULL, pch = 16, cex = 0.35, main = “3 hour V TR” )
V.DA24.volp = volcanoplot(efit2,coef = 6, style = “p-value”, highlight =
100, names = fit<span class="math inline">\(genes\)</span>SYMBOL, hl.col
= “red”, xlab = “Log2 Fold Change”, ylab = NULL, pch = 16, cex = 0.35,
main = “24 hour V DA” ) V.DX24.volp = volcanoplot(efit2,coef = 7, style
= “p-value”, highlight = 100, names = fit<span
class="math inline">\(genes\)</span>SYMBOL, hl.col = “red”, xlab = “Log2
Fold Change”, ylab = NULL, pch = 16, cex = 0.35, main = “24 hour V DX” )
V.EP24.volp = volcanoplot(efit2,coef = 8, style = “p-value”, highlight =
100, names = fit<span class="math inline">\(genes\)</span>SYMBOL, hl.col
= “red”, xlab = “Log2 Fold Change”, ylab = NULL, pch = 16, cex = 0.35,
main = “24 hour V EP” ) V.MT24.volp = volcanoplot(efit2,coef = 9, style
= “p-value”,<br />
highlight = 100, names = fit<span
class="math inline">\(genes\)</span>SYMBOL, hl.col = “red”, xlab = “Log2
Fold Change”, ylab = NULL, pch = 16, cex = 0.35, main = “24 hour V MT” )
V.TR24.volp = volcanoplot(efit2,coef = 10,style = “p-value”, highlight =
20, names = fit<span class="math inline">\(genes\)</span>SYMBOL, hl.col
= “red”, xlab = “Log2 Fold Change”, ylab = NULL, pch = 16, cex = 0.35,
main = “24 hour V TR” )</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] cowplot_1.1.1                          
 [2] Homo.sapiens_1.3.1                     
 [3] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
 [4] org.Hs.eg.db_3.15.0                    
 [5] GO.db_3.15.0                           
 [6] OrganismDbi_1.38.1                     
 [7] GenomicFeatures_1.48.4                 
 [8] GenomicRanges_1.48.0                   
 [9] GenomeInfoDb_1.32.4                    
[10] AnnotationDbi_1.58.0                   
[11] IRanges_2.30.1                         
[12] S4Vectors_0.34.0                       
[13] biomaRt_2.52.0                         
[14] scales_1.2.1                           
[15] forcats_0.5.2                          
[16] stringr_1.5.0                          
[17] dplyr_1.0.10                           
[18] purrr_1.0.0                            
[19] readr_2.1.3                            
[20] tidyr_1.2.1                            
[21] tibble_3.1.8                           
[22] tidyverse_1.3.2                        
[23] AnnotationHub_3.4.0                    
[24] BiocFileCache_2.4.0                    
[25] dbplyr_2.2.1                           
[26] devtools_2.4.5                         
[27] usethis_2.1.6                          
[28] reshape2_1.4.4                         
[29] gridExtra_2.3                          
[30] HTSFilter_1.36.0                       
[31] VennDiagram_1.7.3                      
[32] futile.logger_1.4.3                    
[33] mixOmics_6.20.0                        
[34] ggplot2_3.4.0                          
[35] lattice_0.20-45                        
[36] MASS_7.3-58.1                          
[37] RColorBrewer_1.1-3                     
[38] edgeR_3.38.4                           
[39] limma_3.52.4                           
[40] Biobase_2.56.0                         
[41] BiocGenerics_0.42.0                    
[42] workflowr_1.7.0                        

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                    tidyselect_1.2.0             
  [3] RSQLite_2.2.20                htmlwidgets_1.6.1            
  [5] BiocParallel_1.30.4           munsell_0.5.0                
  [7] codetools_0.2-18              miniUI_0.1.1.1               
  [9] withr_2.5.0                   colorspace_2.0-3             
 [11] filelock_1.0.2                highr_0.10                   
 [13] knitr_1.41                    rstudioapi_0.14              
 [15] labeling_0.4.2                MatrixGenerics_1.8.1         
 [17] git2r_0.30.1                  GenomeInfoDbData_1.2.8       
 [19] farver_2.1.1                  bit64_4.0.5                  
 [21] rprojroot_2.0.3               vctrs_0.5.1                  
 [23] generics_0.1.3                lambda.r_1.2.4               
 [25] xfun_0.36                     timechange_0.1.1             
 [27] R6_2.5.1                      locfit_1.5-9.7               
 [29] bitops_1.0-7                  cachem_1.0.6                 
 [31] DelayedArray_0.22.0           assertthat_0.2.1             
 [33] promises_1.2.0.1              BiocIO_1.6.0                 
 [35] googlesheets4_1.0.1           gtable_0.3.1                 
 [37] processx_3.8.0                rlang_1.0.6                  
 [39] genefilter_1.78.0             splines_4.2.2                
 [41] rtracklayer_1.56.1            gargle_1.2.1                 
 [43] broom_1.0.2                   BiocManager_1.30.19          
 [45] yaml_2.3.6                    modelr_0.1.10                
 [47] backports_1.4.1               httpuv_1.6.7                 
 [49] RBGL_1.72.0                   tools_4.2.2                  
 [51] ellipsis_0.3.2                jquerylib_0.1.4              
 [53] sessioninfo_1.2.2             Rcpp_1.0.9                   
 [55] plyr_1.8.8                    progress_1.2.2               
 [57] zlibbioc_1.42.0               RCurl_1.98-1.9               
 [59] ps_1.7.2                      prettyunits_1.1.1            
 [61] urlchecker_1.0.1              SummarizedExperiment_1.26.1  
 [63] haven_2.5.1                   ggrepel_0.9.2                
 [65] fs_1.5.2                      magrittr_2.0.3               
 [67] RSpectra_0.16-1               futile.options_1.0.1         
 [69] reprex_2.0.2                  googledrive_2.0.0            
 [71] whisker_0.4.1                 matrixStats_0.63.0           
 [73] pkgload_1.3.2                 hms_1.1.2                    
 [75] mime_0.12                     evaluate_0.19                
 [77] xtable_1.8-4                  XML_3.99-0.13                
 [79] readxl_1.4.1                  compiler_4.2.2               
 [81] ellipse_0.4.3                 crayon_1.5.2                 
 [83] htmltools_0.5.4               corpcor_1.6.10               
 [85] later_1.3.0                   tzdb_0.3.0                   
 [87] geneplotter_1.74.0            lubridate_1.9.0              
 [89] DBI_1.1.3                     formatR_1.13                 
 [91] rappdirs_0.3.3                Matrix_1.5-3                 
 [93] cli_3.5.0                     parallel_4.2.2               
 [95] igraph_1.3.5                  pkgconfig_2.0.3              
 [97] getPass_0.2-2                 GenomicAlignments_1.32.1     
 [99] xml2_1.3.3                    rARPACK_0.11-0               
[101] annotate_1.74.0               bslib_0.4.2                  
[103] XVector_0.36.0                rvest_1.0.3                  
[105] callr_3.7.3                   digest_0.6.31                
[107] graph_1.74.0                  Biostrings_2.64.1            
[109] rmarkdown_2.19                cellranger_1.1.0             
[111] restfulr_0.0.15               curl_4.3.3                   
[113] shiny_1.7.4                   Rsamtools_2.12.0             
[115] rjson_0.2.21                  lifecycle_1.0.3              
[117] jsonlite_1.8.4                viridisLite_0.4.1            
[119] fansi_1.0.3                   pillar_1.8.1                 
[121] KEGGREST_1.36.3               fastmap_1.1.0                
[123] httr_1.4.4                    pkgbuild_1.4.0               
[125] survival_3.4-0                interactiveDisplayBase_1.34.0
[127] glue_1.6.2                    remotes_2.4.2                
[129] png_0.1-8                     BiocVersion_3.15.2           
[131] bit_4.0.5                     stringi_1.7.8                
[133] sass_0.4.4                    profvis_0.3.7                
[135] blob_1.2.3                    DESeq2_1.36.0                
[137] memoise_2.0.1                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
