<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="ERM" />

<meta name="date" content="2023-01-18" />

<title>Run1_analysis</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cardiotoxicity</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/reneeisnowhere/Cardiotoxicity">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Run1_analysis</h1>
<h4 class="author">ERM</h4>
<h4 class="date">20230118</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-02-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Cardiotoxicity/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230109code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230109)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230109code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230109)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereCardiotoxicitytree7a6d5d52a002366168c3e7b36d95429c1de478f4targetblank7a6d5d5a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/tree/7a6d5d52a002366168c3e7b36d95429c1de478f4" target="_blank">7a6d5d5</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereCardiotoxicitytree7a6d5d52a002366168c3e7b36d95429c1de478f4targetblank7a6d5d5a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/tree/7a6d5d52a002366168c3e7b36d95429c1de478f4" target="_blank">7a6d5d5</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.txt
    Ignored:    data/allgenes.txt
    Ignored:    data/backGL.txt
    Ignored:    data/data_outline.txt
    Ignored:    data/filenameonly.txt
    Ignored:    data/individualDRCfile.RDS
    Ignored:    data/mymatrix.RDS
    Ignored:    data/norm_LDH.csv
    Ignored:    data/norm_counts.csv
    Ignored:    data/response_cluster24h.csv
    Ignored:    data/sigVDA24.txt
    Ignored:    data/sigVDA3.txt
    Ignored:    data/sigVDX24.txt
    Ignored:    data/sigVDX3.txt
    Ignored:    data/sigVEP24.txt
    Ignored:    data/sigVEP3.txt
    Ignored:    data/sigVMT24.txt
    Ignored:    data/sigVMT3.txt
    Ignored:    data/sigVTR24.txt
    Ignored:    data/sigVTR3.txt

Untracked files:
    Untracked:  .RDataTmp
    Untracked:  firstvenn24h.png
    Untracked:  mcor.csv
    Untracked:  reneebasecode.R

Unstaged changes:
    Modified:   code/Corrscripts.R
    Modified:   venn_code.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/RNAseqanalysis.Rmd</code>) and
HTML (<code>docs/RNAseqanalysis.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/7a6d5d52a002366168c3e7b36d95429c1de478f4/analysis/RNAseqanalysis.Rmd" target="_blank">7a6d5d5</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-02-10
</td>
<td>
updates for the week part 500
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/cdd9374fe035ea7d70f2e099cc46d4543a18bb5d/docs/RNAseqanalysis.html" target="_blank">cdd9374</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-02-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/40fb3405b34002a284af9905504291b1f6d870bd/analysis/RNAseqanalysis.Rmd" target="_blank">40fb340</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-02-10
</td>
<td>
updates for the week
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/accc2412d4b3a5da3d043dbb62116296492e33e2/analysis/RNAseqanalysis.Rmd" target="_blank">accc241</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-02-10
</td>
<td>
updates for the week
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/1fc5c19512ece9d333dec0ef8e8ab5fe58c735e6/analysis/RNAseqanalysis.Rmd" target="_blank">1fc5c19</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-02-09
</td>
<td>
Git update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Hello! I am first loading all the beautiful libraries I will
need.</p>
<pre class="r"><code>library(Biobase)
library(edgeR)
library(limma)
library(RColorBrewer)
library(mixOmics)
library(VennDiagram)
library(HTSFilter)
library(ggplot2)
library(gridExtra)
library(reshape2)
library(devtools)
library(AnnotationHub)
library(tidyverse)
library(scales)
library(biomaRt)
library(Homo.sapiens)
library(cowplot)
library(ggrepel)
library(corrplot)
library(Hmisc)</code></pre>
<p>The next step is to load all the data I will be using. Currently, I
am not posting the raw data, but I will release in the future.</p>
<p>This is how I retrieved the gene symbols.</p>
<pre class="r"><code> ###now we add genenames to the geneid###
 
geneid &lt;- rownames(mymatrix) ### pulls the names we have in the counts file
genes &lt;- select(Homo.sapiens, keys=geneid, columns=c(&quot;SYMBOL&quot;),
  keytype=&quot;ENTREZID&quot;)
genes &lt;- genes[!duplicated(genes$ENTREZID),]
mymatrix$genes &lt;- genes</code></pre>
<div id="filtering-out-low-count-genes" class="section level2">
<h2>Filtering out low count genes</h2>
<p>Filtering the genes that are lowly expressed using several
methods.</p>
<p>First method, removing only those rows with zero counts across all
samples.</p>
<pre class="r"><code># 
# old.par &lt;- par(mar = c(0, 0, 0, 0))
# par(old.par)
# boxplot(data =RNAseqreads, total~Sample, main = &quot;Boxplots of total reads&quot;,xaxt = &quot;n&quot;, xlab= &quot;&quot;)
# x_axis_labels(labels = samplenames, every_nth = 1, adj=1, srt =90, cex =0.4)
# ggplot(RNAseqreads, x = Sample, y = total)+
#   geom_boxplot()

table(rowSums(mymatrix$counts==0)==72)</code></pre>
<pre><code>
FALSE  TRUE 
24931  3464 </code></pre>
<p>This filtering would leave 24931 genes and remove 3464, That is too
many leftover genes!</p>
<p>So now to try something a little more stringent using the built in
function from the edgeR package.</p>
<pre class="r"><code>keep &lt;- filterByExpr.DGEList(mymatrix, group = group)
filter_test &lt;- mymatrix[keep, , keep.lib.sizes=FALSE]
dim(filter_test)</code></pre>
<pre><code>[1] 14448    72</code></pre>
<p>This method effectively uses a cutoff off that leaves 14448 genes.
The cutoff is determined by keeping genes that have a count-per-million
(CPM) above 10, (the default minimum set) in 6 samples. A set is
determined using the design matrix.<br />
For my design, I grouped my 72 samples into sets of 6, one set includes
each individual + a specific treatment + a specific time.</p>
<p>The beginning cutoff-standard in our lab is to start by using the
rowMeans &gt;0 cutoff on the log10 of cpm.</p>
<pre class="r"><code>cpm &lt;- cpm(mymatrix)
lcpm &lt;- cpm(mymatrix, log=TRUE)  ### for determining the basic cutoffs
dim(lcpm)</code></pre>
<pre><code>[1] 28395    72</code></pre>
<pre class="r"><code>L &lt;- mean(mymatrix$samples$lib.size) * 1e-6
M &lt;- median(mymatrix$samples$lib.size) * 1e-6
c(L, M)</code></pre>
<pre><code>[1] 4.679061 4.494188</code></pre>
<pre class="r"><code>filcpm_matrix &lt;- subset(lcpm, (rowMeans(lcpm)&gt;0))

dim(filcpm_matrix)</code></pre>
<pre><code>[1] 14823    72</code></pre>
<pre class="r"><code>##method 2 with rowMeans
row_means &lt;- rowMeans(lcpm)
x &lt;- mymatrix[row_means &gt; 0,]
dim(x)</code></pre>
<pre><code>[1] 14823    72</code></pre>
<pre class="r"><code>write.csv(x$counts, &quot;data/norm_counts.csv&quot;)</code></pre>
<p>Both of the above methods leave 14823 genes from 28,395. I prefer the
second method, which keeps the DGEList format of the data.</p>
<p><img src="figure/RNAseqanalysis.Rmd/plots%20of%20before%20and%20after%20filter-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/plots%20of%20before%20and%20after%20filter-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/plots%20of%20before%20and%20after%20filter-3.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="figure/RNAseqanalysis.Rmd/boxplot%20counts%20per%20sample-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/boxplot%20counts%20per%20sample-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>now I will produce the RIN x sample plots:###
<img src="figure/RNAseqanalysis.Rmd/sample%20quality%20checks-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/sample%20quality%20checks-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/sample%20quality%20checks-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/sample%20quality%20checks-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/sample%20quality%20checks-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="principal-component-analysis" class="section level1">
<h1>Principal Component Analysis</h1>
<p>PCA was done using code adopted from J. Blischak.</p>
<pre><code>

###  Daunorubicin </code></pre>
<p><img src="figure/RNAseqanalysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre><code>

###  Doxorubicin </code></pre>
<p><img src="figure/RNAseqanalysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-2.png" width="576" style="display: block; margin: auto;" /></p>
<pre><code>

###  Epirubicin </code></pre>
<p><img src="figure/RNAseqanalysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-3.png" width="576" style="display: block; margin: auto;" /></p>
<pre><code>

###  Mitoxantrone </code></pre>
<p><img src="figure/RNAseqanalysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-4.png" width="576" style="display: block; margin: auto;" /></p>
<pre><code>

###  Trastuzumab </code></pre>
<p><img src="figure/RNAseqanalysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-5.png" width="576" style="display: block; margin: auto;" /></p>
<pre><code>

###  Vehicle </code></pre>
<p><img src="figure/RNAseqanalysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-6.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="pca-of-all-72-samples" class="section level1">
<h1>PCA of all 72 samples</h1>
<pre><code>Warning: `qplot()` was deprecated in ggplot2 3.4.0.</code></pre>
<p><img src="figure/RNAseqanalysis.Rmd/all-samples-1.png" width="672" style="display: block; margin: auto;" />
## Variance contribution from treatment, extraction time, or individual
on PC1 and PC2</p>
<p><img src="figure/RNAseqanalysis.Rmd/variance%20by%20treatment%20time%20and%20individual-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/variance%20by%20treatment%20time%20and%20individual-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/variance%20by%20treatment%20time%20and%20individual-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/variance%20by%20treatment%20time%20and%20individual-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/variance%20by%20treatment%20time%20and%20individual-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/variance%20by%20treatment%20time%20and%20individual-6.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="differential-expression" class="section level1">
<h1>Differential Expression</h1>
<p><img src="figure/RNAseqanalysis.Rmd/Differential%20expression-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>mm2 &lt;- model.matrix(~0 + group1)
##made the matrix model using the interaction between Treatment and Time
colnames(mm2) &lt;- c(&quot;A3&quot;, &quot;X3&quot;, &quot;E3&quot;,&quot;M3&quot;,&quot;T3&quot;, &quot;V3&quot;,&quot;A24&quot;, &quot;X24&quot;, &quot;E24&quot;,&quot;M24&quot;,&quot;T24&quot;, &quot;V24&quot;)
y2 &lt;- voom(x, mm2)

corfit2 &lt;- duplicateCorrelation(y2, mm2, block = indv)

v2 &lt;- voom(x, mm2, block = indv, correlation = corfit2$consensus)

fit2 &lt;- lmFit(v2, mm2, block = indv, correlation = corfit2$consensus)

vfit2 &lt;- lmFit(y2, mm2)

vfit2&lt;- contrasts.fit(vfit2, contrasts=cm2)

efit2 &lt;- eBayes(vfit2)

V.DA.top= topTable(efit2, coef=1, adjust=&quot;BH&quot;, number=Inf, sort.by=&quot;p&quot;)
### sorting all top expressed genes for the Vehicle and Daunorubicin 3 hour treatments
sigVDA3 = V.DA.top[V.DA.top$adj.P.Val &lt; .1 , ]
### this helped pull only those files that were at an adjusted p value of less than 0.1
### This p-value was used as the beginning examination of the data, considering I will run multiple runs of this RNA seq library.</code></pre>
<p>This is the example code I used to process my data. I used two model
matrix initially, one set up was /~0 +drug +time and the second was
/~0+group1, then blocking by individual. That is why you see the number
2 in the code above.</p>
<p>#DEG summary</p>
<pre><code>Importance of components:
                           PC1     PC2     PC3      PC4     PC5      PC6
Standard deviation     60.2067 44.5108 32.3037 30.98489 27.1431 24.07762
Proportion of Variance  0.2445  0.1337  0.0704  0.06477  0.0497  0.03911
Cumulative Proportion   0.2445  0.3782  0.4486  0.51337  0.5631  0.60218
                            PC7      PC8      PC9     PC10     PC11     PC12
Standard deviation     23.26775 21.37712 20.13091 18.19509 17.63059 16.30004
Proportion of Variance  0.03652  0.03083  0.02734  0.02233  0.02097  0.01792
Cumulative Proportion   0.63870  0.66953  0.69687  0.71921  0.74018  0.75810
                           PC13     PC14     PC15     PC16     PC17     PC18
Standard deviation     15.87695 14.24554 13.51806 12.67177 11.84085 11.44481
Proportion of Variance  0.01701  0.01369  0.01233  0.01083  0.00946  0.00884
Cumulative Proportion   0.77511  0.78880  0.80113  0.81196  0.82142  0.83025
                           PC19    PC20    PC21    PC22    PC23    PC24    PC25
Standard deviation     11.08484 10.4711 9.86347 9.53287 9.25138 9.06027 8.70778
Proportion of Variance  0.00829  0.0074 0.00656 0.00613 0.00577 0.00554 0.00512
Cumulative Proportion   0.83854  0.8459 0.85250 0.85863 0.86441 0.86995 0.87506
                          PC26    PC27    PC28    PC29    PC30    PC31    PC32
Standard deviation     8.40220 8.26732 8.04037 7.85926 7.76676 7.67958 7.48930
Proportion of Variance 0.00476 0.00461 0.00436 0.00417 0.00407 0.00398 0.00378
Cumulative Proportion  0.87982 0.88444 0.88880 0.89296 0.89703 0.90101 0.90480
                          PC33    PC34    PC35   PC36    PC37    PC38    PC39
Standard deviation     7.34178 7.28566 7.22861 7.1043 6.99972 6.94542 6.83650
Proportion of Variance 0.00364 0.00358 0.00353 0.0034 0.00331 0.00325 0.00315
Cumulative Proportion  0.90843 0.91201 0.91554 0.9189 0.92225 0.92550 0.92866
                          PC40    PC41    PC42    PC43    PC44   PC45    PC46
Standard deviation     6.75717 6.73038 6.57835 6.54307 6.53123 6.4395 6.37432
Proportion of Variance 0.00308 0.00306 0.00292 0.00289 0.00288 0.0028 0.00274
Cumulative Proportion  0.93174 0.93479 0.93771 0.94060 0.94348 0.9463 0.94902
                          PC47    PC48    PC49   PC50    PC51   PC52    PC53
Standard deviation     6.34772 6.23190 6.17748 6.0816 6.01946 5.9644 5.89730
Proportion of Variance 0.00272 0.00262 0.00257 0.0025 0.00244 0.0024 0.00235
Cumulative Proportion  0.95173 0.95435 0.95693 0.9594 0.96187 0.9643 0.96661
                         PC54    PC55    PC56    PC57    PC58    PC59    PC60
Standard deviation     5.8378 5.80173 5.72099 5.69367 5.65008 5.60128 5.47568
Proportion of Variance 0.0023 0.00227 0.00221 0.00219 0.00215 0.00212 0.00202
Cumulative Proportion  0.9689 0.97118 0.97339 0.97558 0.97773 0.97985 0.98187
                          PC61    PC62    PC63    PC64    PC65   PC66    PC67
Standard deviation     5.46660 5.38710 5.31872 5.25710 5.11760 5.0166 4.83245
Proportion of Variance 0.00202 0.00196 0.00191 0.00186 0.00177 0.0017 0.00158
Cumulative Proportion  0.98389 0.98585 0.98776 0.98962 0.99139 0.9931 0.99466
                          PC68    PC69    PC70    PC71      PC72
Standard deviation     4.73475 4.57831 4.51242 3.92680 4.923e-14
Proportion of Variance 0.00151 0.00141 0.00137 0.00104 0.000e+00
Cumulative Proportion  0.99617 0.99759 0.99896 1.00000 1.000e+00</code></pre>
<pre><code>        V.DA  V.DX  V.EP  V.MT  V.TR V.DA24 V.DX24 V.EP24 V.MT24 V.TR24
Down      63     3    11    13     0   3342   3013   2871    320      0
NotSig 14456 14814 14714 14780 14823   8279   8993   9034  13970  14823
Up       304     6    98    30     0   3202   2817   2918    533      0</code></pre>
<div id="graphing-de-genes" class="section level2">
<h2>Graphing DE genes</h2>
<p>I then created a counts table for each set of genes. Luckily, the
counts are stored in the y2 object, which is an EList class object. I
can ‘simplify’ this process because I kept the DEGList format initially.
I first made an object called ‘countstotal’ from the EList y2. For
ggploting later, I subsetted ‘countstotal’ by treatments.</p>
<pre class="r"><code>countstotal &lt;- y2$E
colnames(countstotal) &lt;- smlabel
boxplot(countstotal, xaxt = &quot;n&quot;, xlab=&quot;&quot;)
Da3counts &lt;- as.data.frame(as.table(countstotal[,c(1,6,13,18,25,30,37,42,49,54,61,66)]))
x_axis_labels(labels = label, every_nth = 1, adj=1, srt =90, cex =0.4)</code></pre>
<p><img src="figure/RNAseqanalysis.Rmd/counts%20files-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>library(cowplot)</p>
</div>
</div>
<div id="volcano-plots" class="section level1">
<h1>Volcano Plots</h1>
<p><img src="figure/RNAseqanalysis.Rmd/Volcanoplots-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/Volcanoplots-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/Volcanoplots-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/Volcanoplots-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/Volcanoplots-5.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="figure/RNAseqanalysis.Rmd/cormotif%20things-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/cormotif%20things-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/RNAseqanalysis.Rmd/cormotif%20things-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="go-analysis" class="section level1">
<h1><a
href="https://reneeisnowhere.github.io/Cardiotoxicity/GO_analysis.html">GO
analysis</a></h1>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] pheatmap_1.0.12                        
 [2] Cormotif_1.42.0                        
 [3] affy_1.74.0                            
 [4] Hmisc_4.7-2                            
 [5] Formula_1.2-4                          
 [6] survival_3.5-0                         
 [7] corrplot_0.92                          
 [8] ggrepel_0.9.3                          
 [9] cowplot_1.1.1                          
[10] Homo.sapiens_1.3.1                     
[11] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
[12] org.Hs.eg.db_3.15.0                    
[13] GO.db_3.15.0                           
[14] OrganismDbi_1.38.1                     
[15] GenomicFeatures_1.48.4                 
[16] GenomicRanges_1.48.0                   
[17] GenomeInfoDb_1.32.4                    
[18] AnnotationDbi_1.58.0                   
[19] IRanges_2.30.1                         
[20] S4Vectors_0.34.0                       
[21] biomaRt_2.52.0                         
[22] scales_1.2.1                           
[23] forcats_1.0.0                          
[24] stringr_1.5.0                          
[25] dplyr_1.1.0                            
[26] purrr_1.0.1                            
[27] readr_2.1.3                            
[28] tidyr_1.3.0                            
[29] tibble_3.1.8                           
[30] tidyverse_1.3.2                        
[31] AnnotationHub_3.4.0                    
[32] BiocFileCache_2.4.0                    
[33] dbplyr_2.3.0                           
[34] devtools_2.4.5                         
[35] usethis_2.1.6                          
[36] reshape2_1.4.4                         
[37] gridExtra_2.3                          
[38] HTSFilter_1.36.0                       
[39] VennDiagram_1.7.3                      
[40] futile.logger_1.4.3                    
[41] mixOmics_6.20.0                        
[42] ggplot2_3.4.0                          
[43] lattice_0.20-45                        
[44] MASS_7.3-58.2                          
[45] RColorBrewer_1.1-3                     
[46] edgeR_3.38.4                           
[47] limma_3.52.4                           
[48] Biobase_2.56.0                         
[49] BiocGenerics_0.42.0                    
[50] workflowr_1.7.0                        

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3                rtracklayer_1.56.1           
  [3] bit64_4.0.5                   knitr_1.42                   
  [5] DelayedArray_0.22.0           data.table_1.14.6            
  [7] rpart_4.1.19                  KEGGREST_1.36.3              
  [9] RCurl_1.98-1.10               generics_0.1.3               
 [11] preprocessCore_1.58.0         callr_3.7.3                  
 [13] lambda.r_1.2.4                RSQLite_2.2.20               
 [15] bit_4.0.5                     tzdb_0.3.0                   
 [17] xml2_1.3.3                    lubridate_1.9.1              
 [19] httpuv_1.6.8                  SummarizedExperiment_1.26.1  
 [21] assertthat_0.2.1              gargle_1.3.0                 
 [23] xfun_0.37                     hms_1.1.2                    
 [25] jquerylib_0.1.4               evaluate_0.20                
 [27] promises_1.2.0.1              fansi_1.0.4                  
 [29] restfulr_0.0.15               progress_1.2.2               
 [31] readxl_1.4.1                  igraph_1.3.5                 
 [33] DBI_1.1.3                     geneplotter_1.74.0           
 [35] htmlwidgets_1.6.1             rARPACK_0.11-0               
 [37] googledrive_2.0.0             ellipsis_0.3.2               
 [39] RSpectra_0.16-1               backports_1.4.1              
 [41] annotate_1.74.0               deldir_1.0-6                 
 [43] MatrixGenerics_1.8.1          vctrs_0.5.2                  
 [45] remotes_2.4.2                 cachem_1.0.6                 
 [47] withr_2.5.0                   checkmate_2.1.0              
 [49] GenomicAlignments_1.32.1      prettyunits_1.1.1            
 [51] cluster_2.1.4                 crayon_1.5.2                 
 [53] ellipse_0.4.3                 genefilter_1.78.0            
 [55] pkgconfig_2.0.3               labeling_0.4.2               
 [57] pkgload_1.3.2                 nnet_7.3-18                  
 [59] rlang_1.0.6                   lifecycle_1.0.3              
 [61] miniUI_0.1.1.1                filelock_1.0.2               
 [63] affyio_1.66.0                 modelr_0.1.10                
 [65] cellranger_1.1.0              rprojroot_2.0.3              
 [67] matrixStats_0.63.0            graph_1.74.0                 
 [69] Matrix_1.5-3                  reprex_2.0.2                 
 [71] base64enc_0.1-3               whisker_0.4.1                
 [73] processx_3.8.0                googlesheets4_1.0.1          
 [75] png_0.1-8                     viridisLite_0.4.1            
 [77] rjson_0.2.21                  bitops_1.0-7                 
 [79] getPass_0.2-2                 Biostrings_2.64.1            
 [81] blob_1.2.3                    jpeg_0.1-10                  
 [83] memoise_2.0.1                 magrittr_2.0.3               
 [85] plyr_1.8.8                    zlibbioc_1.42.0              
 [87] compiler_4.2.2                BiocIO_1.6.0                 
 [89] DESeq2_1.36.0                 Rsamtools_2.12.0             
 [91] cli_3.6.0                     XVector_0.36.0               
 [93] urlchecker_1.0.1              ps_1.7.2                     
 [95] htmlTable_2.4.1               formatR_1.14                 
 [97] tidyselect_1.2.0              stringi_1.7.12               
 [99] highr_0.10                    yaml_2.3.7                   
[101] locfit_1.5-9.7                latticeExtra_0.6-30          
[103] sass_0.4.5                    tools_4.2.2                  
[105] timechange_0.2.0              parallel_4.2.2               
[107] rstudioapi_0.14               foreign_0.8-84               
[109] git2r_0.31.0                  farver_2.1.1                 
[111] digest_0.6.31                 BiocManager_1.30.19          
[113] shiny_1.7.4                   Rcpp_1.0.10                  
[115] broom_1.0.3                   BiocVersion_3.15.2           
[117] later_1.3.0                   httr_1.4.4                   
[119] colorspace_2.1-0              rvest_1.0.3                  
[121] XML_3.99-0.13                 fs_1.6.1                     
[123] splines_4.2.2                 RBGL_1.72.0                  
[125] statmod_1.5.0                 sessioninfo_1.2.2            
[127] xtable_1.8-4                  jsonlite_1.8.4               
[129] futile.options_1.0.1          corpcor_1.6.10               
[131] R6_2.5.1                      profvis_0.3.7                
[133] pillar_1.8.1                  htmltools_0.5.4              
[135] mime_0.12                     glue_1.6.2                   
[137] fastmap_1.1.0                 BiocParallel_1.30.4          
[139] interactiveDisplayBase_1.34.0 codetools_0.2-19             
[141] pkgbuild_1.4.0                utf8_1.2.3                   
[143] bslib_0.4.2                   curl_5.0.0                   
[145] interp_1.1-3                  rmarkdown_2.20               
[147] munsell_0.5.0                 GenomeInfoDbData_1.2.8       
[149] haven_2.5.1                   gtable_0.3.1                 </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
