<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="ERM" />

<meta name="date" content="2023-04-10" />

<title>Run_all_analysis</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cardiotoxicity</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/reneeisnowhere/Cardiotoxicity">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Run_all_analysis</h1>
<h4 class="author">ERM</h4>
<h4 class="date">20230410</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-04-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Cardiotoxicity/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230109code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230109)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230109code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230109)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereCardiotoxicitytreefa2cbeb73c1ad79d45735208ec4b4fe897acfb4ctargetblankfa2cbeba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/tree/fa2cbeb73c1ad79d45735208ec4b4fe897acfb4c" target="_blank">fa2cbeb</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereCardiotoxicitytreefa2cbeb73c1ad79d45735208ec4b4fe897acfb4ctargetblankfa2cbeba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/tree/fa2cbeb73c1ad79d45735208ec4b4fe897acfb4c" target="_blank">fa2cbeb</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/ACresponse_cluster24h.csv
    Ignored:    data/Clamp_Summary.csv
    Ignored:    data/Cormotif_24_k1-5_raw.RDS
    Ignored:    data/DAtable1.csv
    Ignored:    data/DDE_reQTL.txt
    Ignored:    data/DDEresp_list.csv
    Ignored:    data/DEG_cormotif.RDS
    Ignored:    data/DF_Plate_Peak.csv
    Ignored:    data/Da24counts.txt
    Ignored:    data/Dx24counts.txt
    Ignored:    data/Dx_reQTL_specific.txt
    Ignored:    data/Ep24counts.txt
    Ignored:    data/GOplots.R
    Ignored:    data/K_cluster
    Ignored:    data/K_cluster_kisthree.csv
    Ignored:    data/K_cluster_kistwo.csv
    Ignored:    data/Mt24counts.txt
    Ignored:    data/RINsamplelist.txt
    Ignored:    data/Seonane2019supp1.txt
    Ignored:    data/TOP2Bi-24hoursGO_analysis.csv
    Ignored:    data/TR24counts.txt
    Ignored:    data/Top2biresp_cluster24h.csv
    Ignored:    data/Viabilitylistfull.csv
    Ignored:    data/allexpressedgenes.txt
    Ignored:    data/allgenes.txt
    Ignored:    data/allmatrix.RDS
    Ignored:    data/avgLD50.RDS
    Ignored:    data/backGL.txt
    Ignored:    data/cormotif_3hk1-8.RDS
    Ignored:    data/cormotif_ER_respint.txt
    Ignored:    data/cormotif_ER_respset.txt
    Ignored:    data/cormotif_LR_respint.txt
    Ignored:    data/cormotif_LR_respset.txt
    Ignored:    data/cormotif_NRset.txt
    Ignored:    data/cormotif_TI_respint.txt
    Ignored:    data/cormotif_TI_respset.txt
    Ignored:    data/cormotif_initalK5.RDS
    Ignored:    data/cormotif_initialK5.RDS
    Ignored:    data/counts24hours.RDS
    Ignored:    data/cpmnorm_counts.csv
    Ignored:    data/dat_cpm.RDS
    Ignored:    data/data_outline.txt
    Ignored:    data/efit2results.RDS
    Ignored:    data/ensembl_backup.RDS
    Ignored:    data/ensgtotal.txt
    Ignored:    data/filenameonly.txt
    Ignored:    data/filtered_cpm_counts.csv
    Ignored:    data/filtermatrix_x.RDS
    Ignored:    data/folder_05top/
    Ignored:    data/gene_prob_tran3h.RDS
    Ignored:    data/gene_probabilityk5.RDS
    Ignored:    data/heartgenes.csv
    Ignored:    data/individualDRCfile.RDS
    Ignored:    data/knowles56.GMT
    Ignored:    data/knowlesGMT.GMT
    Ignored:    data/mymatrix.RDS
    Ignored:    data/nonresponse_cluster24h.csv
    Ignored:    data/norm_LDH.csv
    Ignored:    data/norm_counts.csv
    Ignored:    data/plan2plot.png
    Ignored:    data/raw_counts.csv
    Ignored:    data/response_cluster24h.csv
    Ignored:    data/sigVDA24.txt
    Ignored:    data/sigVDA3.txt
    Ignored:    data/sigVDX24.txt
    Ignored:    data/sigVDX3.txt
    Ignored:    data/sigVEP24.txt
    Ignored:    data/sigVEP3.txt
    Ignored:    data/sigVMT24.txt
    Ignored:    data/sigVMT3.txt
    Ignored:    data/sigVTR24.txt
    Ignored:    data/sigVTR3.txt
    Ignored:    data/table3a.omar
    Ignored:    data/tvl24hour.txt
    Ignored:    data/tvl24hourw.txt
    Ignored:    data/venn_code.R

Untracked files:
    Untracked:  .RDataTmp
    Untracked:  .RDataTmp1
    Untracked:  code/extra_code.R
    Untracked:  output/output-old/
    Untracked:  reneebasecode.R

Unstaged changes:
    Modified:   analysis/DRC_analysis.Rmd
    Modified:   analysis/RNAseqanalysis.Rmd
    Modified:   code/eQTLcodes.R
    Deleted:    output/Cormotif.svg
    Deleted:    output/Ctrxn24-3-23.svg
    Deleted:    output/Ctxnrate3-23.png
    Deleted:    output/Decay_Slope3-23.svg
    Deleted:    output/ERmotif5_LFC.svg
    Deleted:    output/GOBP_motif345.svg
    Deleted:    output/KEGGmotif_345.svg
    Deleted:    output/LD503-21-23.png
    Deleted:    output/LDH_24-3-23.svg
    Deleted:    output/LDH_243-23.svg
    Deleted:    output/LFCbytreatment3-25.svg
    Deleted:    output/LFCmotif4_LR.svg
    Deleted:    output/LR_RespMoti4.svg
    Deleted:    output/MeanAmp243-23.svg
    Deleted:    output/NRmotif1_LFC.svg
    Deleted:    output/Rise_Slope3-23.svg
    Deleted:    output/TI_LFC.svg
    Deleted:    output/TVLcorr3-23.svg
    Deleted:    output/TropI3-23.svg
    Deleted:    output/Venn24DEG-3-24.png
    Deleted:    output/motif1NR_LFC.svg
    Deleted:    output/motif3TIlfc3-25.svg
    Deleted:    output/motif4LR3-25LFC.svg
    Deleted:    output/motif5ER-LFC-3-25.svg
    Deleted:    output/nolegendLDH.svg
    Deleted:    output/resultsigVDA24.csv
    Deleted:    output/tropI_24-3-23.svg

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/run_all_analysis.Rmd</code>) and
HTML (<code>docs/run_all_analysis.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/fa2cbeb73c1ad79d45735208ec4b4fe897acfb4c/analysis/run_all_analysis.Rmd" target="_blank">fa2cbeb</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
monday end
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/85526c530073e51f8a1887b4682a80266bb7cfe7/docs/run_all_analysis.html" target="_blank">85526c5</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/1444a8597d566a517ec99258027b3ae8b599fee6/analysis/run_all_analysis.Rmd" target="_blank">1444a85</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
update push of new data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/b266b76acd8dac6d46d06572cf74173c443a67d8/docs/run_all_analysis.html" target="_blank">b266b76</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/d3f8cf7ecab5d70e7d00b13a0fdd8341a3f0002b/analysis/run_all_analysis.Rmd" target="_blank">d3f8cf7</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
update push of new data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/f0a75e110e3d2b1efc7ce936134b1eea7eda38fc/docs/run_all_analysis.html" target="_blank">f0a75e1</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/8ca4c7e6edd12340f8cc00426c21007648daa48a/analysis/run_all_analysis.Rmd" target="_blank">8ca4c7e</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
first rmd commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/2e6996907c969ccc32e0039f388a9054262f2d1d/analysis/run_all_analysis.Rmd" target="_blank">2e69969</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
adding data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/0f1f1da326a9786ad2180b7d8e6400b8d7c1a892/analysis/run_all_analysis.Rmd" target="_blank">0f1f1da</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
final run analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This starts the documentation of the RNA-seq cardiotoxicity analysis
for my manuscript</p>
<pre class="r"><code>library(edgeR)#
library(limma)#
library(RColorBrewer)
library(mixOmics)
library(gridExtra)#
library(reshape2)#
#library(devtools)
library(AnnotationHub)
library(tidyverse)
library(scales)
library(biomaRt)#
library(Homo.sapiens)
library(cowplot)#
library(ggrepel)#
library(corrplot)
library(Hmisc)</code></pre>
<pre class="r"><code> ###now we add genenames to the geneid###
 
geneid &lt;- rownames(mymatrix) ### pulls the names we have in the counts file
genes &lt;- select(Homo.sapiens, keys=geneid, columns=c(&quot;SYMBOL&quot;),
  keytype=&quot;ENTREZID&quot;)
genes &lt;- genes[!duplicated(genes$ENTREZID),]
mymatrix$genes &lt;- genes</code></pre>
<pre><code>[1] 28395    72</code></pre>
<pre><code>[1] 14086    72</code></pre>
<p><img src="figure/run_all_analysis.Rmd/readinfo-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/readinfo-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/readinfo-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/readinfo-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

###  Vehicle </code></pre>
<p><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

###  Daunorubicin </code></pre>
<p><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

###  Doxorubicin </code></pre>
<p><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

###  Epirubicin </code></pre>
<p><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

###  Mitoxantrone </code></pre>
<p><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

###  Trastuzumab </code></pre>
<p><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>        samplenames indv         drug time RIN group       PC1      PC2
Da.1.3h MCW_RM_R_11    1 Daunorubicin   3h 9.3     1 -18.53902 62.11914
Do.1.3h MCW_RM_R_12    1  Doxorubicin   3h 9.8     2 -12.37874 73.96424
Ep.1.3h MCW_RM_R_13    1   Epirubicin   3h 9.8     3 -11.19226 66.47554
Mi.1.3h MCW_RM_R_14    1 Mitoxantrone   3h  10     4 -10.24538 73.46900
Tr.1.3h MCW_RM_R_15    1  Trastuzumab   3h 9.6     5 -12.21578 79.97646
Ve.1.3h MCW_RM_R_16    1      Vehicle   3h 9.9     6 -15.01396 76.61739
              PC3        PC4       PC5       PC6
Da.1.3h 44.793907  -4.229684  25.33474 -35.96203
Do.1.3h 24.565270  -8.522490 -19.86738 -19.00810
Ep.1.3h 33.103318  -9.146883  18.17321 -43.12911
Mi.1.3h 19.048074 -14.540036  -9.00047 -24.33769
Tr.1.3h  2.646249 -17.004856 -34.18428 -11.83222
Ve.1.3h 12.636907  -4.140009 -39.84065 -17.16481</code></pre>
<p><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Importance of components:
                           PC1     PC2      PC3      PC4      PC5      PC6
Standard deviation     63.9774 47.1102 34.19844 32.56486 28.23410 23.90465
Proportion of Variance  0.2906  0.1576  0.08303  0.07529  0.05659  0.04057
Cumulative Proportion   0.2906  0.4481  0.53117  0.60645  0.66304  0.70361
                            PC7      PC8      PC9     PC10     PC11     PC12
Standard deviation     21.54760 20.74762 19.52266 17.46914 16.25667 15.49063
Proportion of Variance  0.03296  0.03056  0.02706  0.02166  0.01876  0.01704
Cumulative Proportion   0.73657  0.76713  0.79419  0.81586  0.83462  0.85165
                           PC13     PC14     PC15     PC16     PC17    PC18
Standard deviation     14.02995 13.01851 11.78389 11.40612 10.15585 9.76592
Proportion of Variance  0.01397  0.01203  0.00986  0.00924  0.00732 0.00677
Cumulative Proportion   0.86563  0.87766  0.88752  0.89675  0.90408 0.91085
                          PC19    PC20    PC21   PC22    PC23    PC24    PC25
Standard deviation     8.79138 8.66072 8.09173 7.8730 7.46626 7.09040 6.61801
Proportion of Variance 0.00549 0.00533 0.00465 0.0044 0.00396 0.00357 0.00311
Cumulative Proportion  0.91633 0.92166 0.92631 0.9307 0.93466 0.93823 0.94134
                          PC26    PC27    PC28    PC29    PC30    PC31    PC32
Standard deviation     6.31712 6.05845 5.86317 5.72293 5.54398 5.49527 5.27437
Proportion of Variance 0.00283 0.00261 0.00244 0.00233 0.00218 0.00214 0.00197
Cumulative Proportion  0.94418 0.94678 0.94922 0.95155 0.95373 0.95587 0.95785
                          PC33    PC34    PC35    PC36    PC37    PC38    PC39
Standard deviation     5.18844 5.11708 4.86120 4.83415 4.71966 4.52799 4.50677
Proportion of Variance 0.00191 0.00186 0.00168 0.00166 0.00158 0.00146 0.00144
Cumulative Proportion  0.95976 0.96162 0.96330 0.96495 0.96654 0.96799 0.96943
                          PC40    PC41    PC42    PC43    PC44    PC45    PC46
Standard deviation     4.39472 4.34878 4.30170 4.23851 4.15240 4.12290 4.05711
Proportion of Variance 0.00137 0.00134 0.00131 0.00128 0.00122 0.00121 0.00117
Cumulative Proportion  0.97080 0.97215 0.97346 0.97474 0.97596 0.97717 0.97834
                          PC47   PC48    PC49    PC50    PC51    PC52    PC53
Standard deviation     4.01633 3.9421 3.90038 3.83199 3.81069 3.80580 3.76258
Proportion of Variance 0.00115 0.0011 0.00108 0.00104 0.00103 0.00103 0.00101
Cumulative Proportion  0.97948 0.9806 0.98166 0.98271 0.98374 0.98477 0.98577
                          PC54    PC55    PC56    PC57   PC58    PC59    PC60
Standard deviation     3.71747 3.69618 3.64557 3.59456 3.5595 3.51279 3.46169
Proportion of Variance 0.00098 0.00097 0.00094 0.00092 0.0009 0.00088 0.00085
Cumulative Proportion  0.98675 0.98772 0.98866 0.98958 0.9905 0.99136 0.99221
                          PC61    PC62    PC63    PC64    PC65   PC66    PC67
Standard deviation     3.45216 3.40460 3.34569 3.27664 3.21965 3.1371 3.12182
Proportion of Variance 0.00085 0.00082 0.00079 0.00076 0.00074 0.0007 0.00069
Cumulative Proportion  0.99305 0.99388 0.99467 0.99543 0.99617 0.9969 0.99756
                          PC68    PC69    PC70   PC71      PC72
Standard deviation     3.09899 3.05675 2.88458 2.6632 8.662e-14
Proportion of Variance 0.00068 0.00066 0.00059 0.0005 0.000e+00
Cumulative Proportion  0.99824 0.99891 0.99950 1.0000 1.000e+00</code></pre>
<p><img src="figure/run_all_analysis.Rmd/typical%20gene%20expression-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>##correlation heatmap of counts matrix
<img src="figure/run_all_analysis.Rmd/pheatmap%20clustering%20of%20gene%20counts-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>###now to get the counts set for DEG!!</p>
<p><img src="figure/run_all_analysis.Rmd/counts%20to%20DEG%20set%20up-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>        V.DA  V.DX  V.EP  V.MT  V.TR V.DA24 V.DX24 V.EP24 V.MT24 V.TR24
Down     108     3    29    22     0   3542   3341   3111    430      0
NotSig 13569 14067 13876 14014 14086   7062   7431   7751  12964  14086
Up       409    16   181    50     0   3482   3314   3224    692      0</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.2.2 (2022-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] Hmisc_4.8-0                            
 [2] Formula_1.2-5                          
 [3] survival_3.5-3                         
 [4] corrplot_0.92                          
 [5] ggrepel_0.9.3                          
 [6] cowplot_1.1.1                          
 [7] Homo.sapiens_1.3.1                     
 [8] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
 [9] org.Hs.eg.db_3.15.0                    
[10] GO.db_3.15.0                           
[11] OrganismDbi_1.38.1                     
[12] GenomicFeatures_1.48.4                 
[13] GenomicRanges_1.48.0                   
[14] GenomeInfoDb_1.32.4                    
[15] AnnotationDbi_1.58.0                   
[16] IRanges_2.30.1                         
[17] S4Vectors_0.34.0                       
[18] Biobase_2.56.0                         
[19] biomaRt_2.52.0                         
[20] scales_1.2.1                           
[21] lubridate_1.9.2                        
[22] forcats_1.0.0                          
[23] stringr_1.5.0                          
[24] dplyr_1.1.0                            
[25] purrr_1.0.1                            
[26] readr_2.1.4                            
[27] tidyr_1.3.0                            
[28] tibble_3.1.8                           
[29] tidyverse_2.0.0                        
[30] AnnotationHub_3.4.0                    
[31] BiocFileCache_2.4.0                    
[32] dbplyr_2.3.1                           
[33] BiocGenerics_0.42.0                    
[34] reshape2_1.4.4                         
[35] gridExtra_2.3                          
[36] mixOmics_6.20.0                        
[37] ggplot2_3.4.1                          
[38] lattice_0.20-45                        
[39] MASS_7.3-58.2                          
[40] RColorBrewer_1.1-3                     
[41] edgeR_3.38.4                           
[42] limma_3.52.4                           
[43] workflowr_1.7.0                        

loaded via a namespace (and not attached):
  [1] backports_1.4.1               plyr_1.8.8                   
  [3] igraph_1.4.1                  splines_4.2.2                
  [5] BiocParallel_1.30.4           digest_0.6.31                
  [7] htmltools_0.5.4               fansi_1.0.4                  
  [9] checkmate_2.1.0               magrittr_2.0.3               
 [11] memoise_2.0.1                 cluster_2.1.4                
 [13] tzdb_0.3.0                    Biostrings_2.64.1            
 [15] matrixStats_0.63.0            rARPACK_0.11-0               
 [17] timechange_0.2.0              prettyunits_1.1.1            
 [19] jpeg_0.1-10                   colorspace_2.1-0             
 [21] blob_1.2.3                    rappdirs_0.3.3               
 [23] xfun_0.37                     callr_3.7.3                  
 [25] crayon_1.5.2                  RCurl_1.98-1.10              
 [27] jsonlite_1.8.4                graph_1.74.0                 
 [29] glue_1.6.2                    gtable_0.3.1                 
 [31] zlibbioc_1.42.0               XVector_0.36.0               
 [33] DelayedArray_0.22.0           pheatmap_1.0.12              
 [35] DBI_1.1.3                     Rcpp_1.0.10                  
 [37] htmlTable_2.4.1               xtable_1.8-4                 
 [39] progress_1.2.2                foreign_0.8-84               
 [41] bit_4.0.5                     htmlwidgets_1.6.1            
 [43] httr_1.4.5                    ellipsis_0.3.2               
 [45] farver_2.1.1                  pkgconfig_2.0.3              
 [47] XML_3.99-0.13                 nnet_7.3-18                  
 [49] deldir_1.0-6                  sass_0.4.5                   
 [51] locfit_1.5-9.7                utf8_1.2.3                   
 [53] labeling_0.4.2                tidyselect_1.2.0             
 [55] rlang_1.0.6                   later_1.3.0                  
 [57] munsell_0.5.0                 BiocVersion_3.15.2           
 [59] tools_4.2.2                   cachem_1.0.7                 
 [61] cli_3.6.0                     generics_0.1.3               
 [63] RSQLite_2.3.0                 evaluate_0.20                
 [65] fastmap_1.1.1                 yaml_2.3.7                   
 [67] processx_3.8.0                knitr_1.42                   
 [69] bit64_4.0.5                   fs_1.6.1                     
 [71] KEGGREST_1.36.3               RBGL_1.72.0                  
 [73] whisker_0.4.1                 mime_0.12                    
 [75] xml2_1.3.3                    compiler_4.2.2               
 [77] rstudioapi_0.14               filelock_1.0.2               
 [79] curl_5.0.0                    png_0.1-8                    
 [81] interactiveDisplayBase_1.34.0 statmod_1.5.0                
 [83] bslib_0.4.2                   stringi_1.7.12               
 [85] highr_0.10                    ps_1.7.2                     
 [87] RSpectra_0.16-1               Matrix_1.5-3                 
 [89] vctrs_0.5.2                   pillar_1.8.1                 
 [91] lifecycle_1.0.3               BiocManager_1.30.20          
 [93] jquerylib_0.1.4               data.table_1.14.8            
 [95] bitops_1.0-7                  corpcor_1.6.10               
 [97] httpuv_1.6.9                  rtracklayer_1.56.1           
 [99] latticeExtra_0.6-30           R6_2.5.1                     
[101] BiocIO_1.6.0                  promises_1.2.0.1             
[103] codetools_0.2-19              SummarizedExperiment_1.26.1  
[105] rprojroot_2.0.3               rjson_0.2.21                 
[107] withr_2.5.0                   GenomicAlignments_1.32.1     
[109] Rsamtools_2.12.0              GenomeInfoDbData_1.2.8       
[111] parallel_4.2.2                hms_1.1.2                    
[113] rpart_4.1.19                  grid_4.2.2                   
[115] rmarkdown_2.20                MatrixGenerics_1.8.1         
[117] git2r_0.31.0                  getPass_0.2-2                
[119] base64enc_0.1-3               shiny_1.7.4                  
[121] ellipse_0.4.3                 interp_1.1-3                 
[123] restfulr_0.0.15              </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
