<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="ERM" />

<meta name="date" content="2023-04-14" />

<title>Run_all_analysis</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cardiotoxicity</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/reneeisnowhere/Cardiotoxicity">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Run_all_analysis</h1>
<h4 class="author">ERM</h4>
<h4 class="date">20230414</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-04-17
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Cardiotoxicity/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230109code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230109)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230109code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230109)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereCardiotoxicitytreee1c63d978170fe9544c74c1ef195627e0a1ececbtargetblanke1c63d9a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/tree/e1c63d978170fe9544c74c1ef195627e0a1ececb" target="_blank">e1c63d9</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomreneeisnowhereCardiotoxicitytreee1c63d978170fe9544c74c1ef195627e0a1ececbtargetblanke1c63d9a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/tree/e1c63d978170fe9544c74c1ef195627e0a1ececb" target="_blank">e1c63d9</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/ACresponse_cluster24h.csv
    Ignored:    data/Clamp_Summary.csv
    Ignored:    data/Cormotif_24_k1-5_raw.RDS
    Ignored:    data/DAtable1.csv
    Ignored:    data/DDE_reQTL.txt
    Ignored:    data/DDEresp_list.csv
    Ignored:    data/DEG_cormotif.RDS
    Ignored:    data/DF_Plate_Peak.csv
    Ignored:    data/Da24counts.txt
    Ignored:    data/Dx24counts.txt
    Ignored:    data/Dx_reQTL_specific.txt
    Ignored:    data/Ep24counts.txt
    Ignored:    data/GOplots.R
    Ignored:    data/K_cluster
    Ignored:    data/K_cluster_kisthree.csv
    Ignored:    data/K_cluster_kistwo.csv
    Ignored:    data/Mt24counts.txt
    Ignored:    data/RINsamplelist.txt
    Ignored:    data/Seonane2019supp1.txt
    Ignored:    data/TOP2Bi-24hoursGO_analysis.csv
    Ignored:    data/TR24counts.txt
    Ignored:    data/Top2biresp_cluster24h.csv
    Ignored:    data/Viabilitylistfull.csv
    Ignored:    data/allexpressedgenes.txt
    Ignored:    data/allgenes.txt
    Ignored:    data/allmatrix.RDS
    Ignored:    data/avgLD50.RDS
    Ignored:    data/backGL.txt
    Ignored:    data/cormotif_3hk1-8.RDS
    Ignored:    data/cormotif_ER_cluster.txt
    Ignored:    data/cormotif_ER_respint.txt
    Ignored:    data/cormotif_ER_respset.txt
    Ignored:    data/cormotif_LR_cluster.txt
    Ignored:    data/cormotif_LR_respint.txt
    Ignored:    data/cormotif_LR_respset.txt
    Ignored:    data/cormotif_NRset.txt
    Ignored:    data/cormotif_TI_cluster.txt
    Ignored:    data/cormotif_TI_respint.txt
    Ignored:    data/cormotif_TI_respset.txt
    Ignored:    data/cormotif_initalK5.RDS
    Ignored:    data/cormotif_initialK5.RDS
    Ignored:    data/cormotif_initialall.RDS
    Ignored:    data/counts24hours.RDS
    Ignored:    data/cpmnorm_counts.csv
    Ignored:    data/dat_cpm.RDS
    Ignored:    data/data_outline.txt
    Ignored:    data/efit2.RDS
    Ignored:    data/efit2results.RDS
    Ignored:    data/ensembl_backup.RDS
    Ignored:    data/ensgtotal.txt
    Ignored:    data/filenameonly.txt
    Ignored:    data/filtered_cpm_counts.csv
    Ignored:    data/filtered_raw_counts.csv
    Ignored:    data/filtermatrix_x.RDS
    Ignored:    data/folder_05top/
    Ignored:    data/gene_prob_tran3h.RDS
    Ignored:    data/gene_probabilityk5.RDS
    Ignored:    data/gostresTop2bi_ER.RDS
    Ignored:    data/gostresTop2bi_LR
    Ignored:    data/gostresTop2bi_LR.RDS
    Ignored:    data/gostresTop2bi_TI.RDS
    Ignored:    data/gostrescoNR
    Ignored:    data/heartgenes.csv
    Ignored:    data/individualDRCfile.RDS
    Ignored:    data/knowles56.GMT
    Ignored:    data/knowlesGMT.GMT
    Ignored:    data/mymatrix.RDS
    Ignored:    data/nonresponse_cluster24h.csv
    Ignored:    data/norm_LDH.csv
    Ignored:    data/norm_counts.csv
    Ignored:    data/plan2plot.png
    Ignored:    data/raw_counts.csv
    Ignored:    data/response_cluster24h.csv
    Ignored:    data/sigVDA24.txt
    Ignored:    data/sigVDA3.txt
    Ignored:    data/sigVDX24.txt
    Ignored:    data/sigVDX3.txt
    Ignored:    data/sigVEP24.txt
    Ignored:    data/sigVEP3.txt
    Ignored:    data/sigVMT24.txt
    Ignored:    data/sigVMT3.txt
    Ignored:    data/sigVTR24.txt
    Ignored:    data/sigVTR3.txt
    Ignored:    data/siglist.RDS
    Ignored:    data/table3a.omar
    Ignored:    data/tvl24hour.txt
    Ignored:    data/tvl24hourw.txt
    Ignored:    data/venn_code.R

Untracked files:
    Untracked:  .RDataTmp
    Untracked:  .RDataTmp1
    Untracked:  .RDataTmp2
    Untracked:  code/extra_code.R
    Untracked:  corMotifcustom.R
    Untracked:  output/output-old/
    Untracked:  output/plan48ldh.png
    Untracked:  output/sequencing_info.txt
    Untracked:  output/tableNR.csv
    Untracked:  output/tabletop2Bi_ER.csv
    Untracked:  output/tabletop2Bi_LR.csv
    Untracked:  output/tabletop2Bi_TI.csv
    Untracked:  reneebasecode.R

Unstaged changes:
    Modified:   code/Corrscripts.R
    Deleted:    output/Cormotif.svg
    Deleted:    output/Ctrxn24-3-23.svg
    Deleted:    output/Ctxnrate3-23.png
    Deleted:    output/Decay_Slope3-23.svg
    Deleted:    output/ERmotif5_LFC.svg
    Deleted:    output/GOBP_motif345.svg
    Deleted:    output/KEGGmotif_345.svg
    Deleted:    output/LD503-21-23.png
    Deleted:    output/LDH_24-3-23.svg
    Deleted:    output/LDH_243-23.svg
    Deleted:    output/LFCbytreatment3-25.svg
    Deleted:    output/LFCmotif4_LR.svg
    Deleted:    output/LR_RespMoti4.svg
    Deleted:    output/MeanAmp243-23.svg
    Deleted:    output/NRmotif1_LFC.svg
    Deleted:    output/Rise_Slope3-23.svg
    Deleted:    output/TI_LFC.svg
    Deleted:    output/TVLcorr3-23.svg
    Deleted:    output/TropI3-23.svg
    Deleted:    output/Venn24DEG-3-24.png
    Deleted:    output/motif1NR_LFC.svg
    Deleted:    output/motif3TIlfc3-25.svg
    Deleted:    output/motif4LR3-25LFC.svg
    Deleted:    output/motif5ER-LFC-3-25.svg
    Deleted:    output/nolegendLDH.svg
    Deleted:    output/resultsigVDA24.csv
    Deleted:    output/tropI_24-3-23.svg

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/run_all_analysis.Rmd</code>) and
HTML (<code>docs/run_all_analysis.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/e1c63d978170fe9544c74c1ef195627e0a1ececb/analysis/run_all_analysis.Rmd" target="_blank">e1c63d9</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-17
</td>
<td>
wflow_publish("analysis/run_all_analysis.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/8221ec34ab3768ad8c1185e329ed9b44233f21a9/docs/run_all_analysis.html" target="_blank">8221ec3</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/9e88c228099cffe9113cbf68024bc65ac5c3f926/analysis/run_all_analysis.Rmd" target="_blank">9e88c22</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-16
</td>
<td>
updated run data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/6d925a2a619d80bf5196a37a22b293714e3d546d/analysis/run_all_analysis.Rmd" target="_blank">6d925a2</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-16
</td>
<td>
updating cormotif with updated RNAseq counts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/8d08bd233d078f431ecc293e3ff642d4c3bc0181/docs/run_all_analysis.html" target="_blank">8d08bd2</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/0aaa63da25dab300af8f0facfeb2fa6a1c6c92a1/analysis/run_all_analysis.Rmd" target="_blank">0aaa63d</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-11
</td>
<td>
cormotif analysis update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/575fd81f58cd36d6408f0734092b219162b1f4be/analysis/run_all_analysis.Rmd" target="_blank">575fd81</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-11
</td>
<td>
updating cormotif
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/4cd8ac4a07466bdf159970d0502a5edf602cacc8/docs/run_all_analysis.html" target="_blank">4cd8ac4</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/08936e73a08d8db556009edcce82d976cc4a3e2b/docs/run_all_analysis.html" target="_blank">08936e7</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/fa2cbeb73c1ad79d45735208ec4b4fe897acfb4c/analysis/run_all_analysis.Rmd" target="_blank">fa2cbeb</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
monday end
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/85526c530073e51f8a1887b4682a80266bb7cfe7/docs/run_all_analysis.html" target="_blank">85526c5</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/1444a8597d566a517ec99258027b3ae8b599fee6/analysis/run_all_analysis.Rmd" target="_blank">1444a85</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
update push of new data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/b266b76acd8dac6d46d06572cf74173c443a67d8/docs/run_all_analysis.html" target="_blank">b266b76</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/d3f8cf7ecab5d70e7d00b13a0fdd8341a3f0002b/analysis/run_all_analysis.Rmd" target="_blank">d3f8cf7</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
update push of new data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/reneeisnowhere/Cardiotoxicity/f0a75e110e3d2b1efc7ce936134b1eea7eda38fc/docs/run_all_analysis.html" target="_blank">f0a75e1</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/8ca4c7e6edd12340f8cc00426c21007648daa48a/analysis/run_all_analysis.Rmd" target="_blank">8ca4c7e</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
first rmd commit
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/2e6996907c969ccc32e0039f388a9054262f2d1d/analysis/run_all_analysis.Rmd" target="_blank">2e69969</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
adding data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/reneeisnowhere/Cardiotoxicity/blob/0f1f1da326a9786ad2180b7d8e6400b8d7c1a892/analysis/run_all_analysis.Rmd" target="_blank">0f1f1da</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-04-10
</td>
<td>
final run analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This starts the documentation of the RNA-seq cardiotoxicity analysis
for my manuscript</p>
<pre class="r"><code>library(edgeR)#
library(limma)#
library(RColorBrewer)
library(mixOmics)
library(gridExtra)#
library(reshape2)#
library(data.table)
library(AnnotationHub)
library(tidyverse)
library(scales)
library(biomaRt)#
library(Homo.sapiens)
library(cowplot)#
library(ggrepel)#
library(corrplot)
library(Hmisc)
library(ggpubr)</code></pre>
<pre class="r"><code> ###now we add genenames to the geneid###
 
geneid &lt;- rownames(mymatrix) ### pulls the names we have in the counts file
genes &lt;- select(Homo.sapiens, keys=geneid, columns=c(&quot;SYMBOL&quot;),
  keytype=&quot;ENTREZID&quot;)
genes &lt;- genes[!duplicated(genes$ENTREZID),]
mymatrix$genes &lt;- genes
#saveRDS(mymatrix, &quot;data/allmatrix.RDS&quot;)
##note-not filtered!</code></pre>
<div id="initial-rna-seq-quality-checks" class="section level1">
<h1>Initial RNA-seq quality checks</h1>
<p><img src="figure/run_all_analysis.Rmd/seq%20quals-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/seq%20quals-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/seq%20quals-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/seq%20quals-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/seq%20quals-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="initial-histograms-from-count-matrix" class="section level1">
<h1>Initial histograms from count matrix</h1>
<pre><code>[1] 28395    72</code></pre>
<pre><code>[1] 14084    72</code></pre>
<p><img src="figure/run_all_analysis.Rmd/readinfo-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/readinfo-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/readinfo-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/readinfo-4.png" width="672" style="display: block; margin: auto;" /></p>
<div id="pca-by-treatment-and-as-a-whole" class="section level2">
<h2>PCA by treatment and as a whole</h2>
<pre><code>

###  Vehicle </code></pre>
<p><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

###  Daunorubicin </code></pre>
<p><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

###  Doxorubicin </code></pre>
<p><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

###  Epirubicin </code></pre>
<p><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

###  Mitoxantrone </code></pre>
<p><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>

###  Trastuzumab </code></pre>
<p><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>        samplenames indv         drug time RIN group       PC1      PC2
Da.1.3h MCW_RM_R_11    1 Daunorubicin   3h 9.3     1 -18.33154 61.71013
Do.1.3h MCW_RM_R_12    1  Doxorubicin   3h 9.8     2 -12.36280 73.97678
Ep.1.3h MCW_RM_R_13    1   Epirubicin   3h 9.8     3 -11.16205 66.48794
Mi.1.3h MCW_RM_R_14    1 Mitoxantrone   3h  10     4 -10.19948 73.48343
Tr.1.3h MCW_RM_R_15    1  Trastuzumab   3h 9.6     5 -12.17619 80.01454
Ve.1.3h MCW_RM_R_16    1      Vehicle   3h 9.9     6 -14.98226 76.62199
              PC3        PC4        PC5       PC6
Da.1.3h 44.039139  -4.547031  24.642107 -35.03245
Do.1.3h 24.576395  -8.626528 -19.908580 -18.97447
Ep.1.3h 33.025628  -9.349549  18.083569 -43.06551
Mi.1.3h 19.016766 -14.639651  -9.065324 -24.29908
Tr.1.3h  2.640624 -17.019296 -34.253925 -11.77881
Ve.1.3h 12.706808  -4.173412 -39.846595 -17.16213</code></pre>
<p><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/individual%20PCA%20by%20drug%20and%20time%20steps-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Importance of components:
                           PC1     PC2      PC3     PC4      PC5      PC6
Standard deviation     63.9885 47.1161 34.21502 32.5878 28.22245 23.90977
Proportion of Variance  0.2907  0.1576  0.08312  0.0754  0.05655  0.04059
Cumulative Proportion   0.2907  0.4483  0.53146  0.6069  0.66342  0.70401
                            PC7      PC8      PC9     PC10     PC11     PC12
Standard deviation     21.56133 20.74271 19.52290 17.47711 16.26239 15.49112
Proportion of Variance  0.03301  0.03055  0.02706  0.02169  0.01878  0.01704
Cumulative Proportion   0.73702  0.76757  0.79463  0.81632  0.83510  0.85213
                           PC13     PC14     PC15     PC16     PC17    PC18
Standard deviation     14.03690 13.01039 11.78935 11.40888 10.15831 9.76192
Proportion of Variance  0.01399  0.01202  0.00987  0.00924  0.00733 0.00677
Cumulative Proportion   0.86612  0.87814  0.88801  0.89725  0.90458 0.91135
                          PC19    PC20    PC21   PC22    PC23    PC24   PC25
Standard deviation     8.77010 8.66331 8.08961 7.8745 7.45127 7.07877 6.6073
Proportion of Variance 0.00546 0.00533 0.00465 0.0044 0.00394 0.00356 0.0031
Cumulative Proportion  0.91681 0.92214 0.92678 0.9312 0.93513 0.93869 0.9418
                          PC26    PC27    PC28    PC29    PC30    PC31    PC32
Standard deviation     6.30345 6.04266 5.85249 5.72714 5.52443 5.49059 5.26497
Proportion of Variance 0.00282 0.00259 0.00243 0.00233 0.00217 0.00214 0.00197
Cumulative Proportion  0.94461 0.94720 0.94963 0.95196 0.95413 0.95627 0.95824
                          PC33    PC34    PC35    PC36    PC37    PC38    PC39
Standard deviation     5.18567 5.07323 4.83532 4.80110 4.65658 4.52528 4.46361
Proportion of Variance 0.00191 0.00183 0.00166 0.00164 0.00154 0.00145 0.00141
Cumulative Proportion  0.96014 0.96197 0.96363 0.96527 0.96681 0.96826 0.96968
                          PC40    PC41    PC42    PC43    PC44    PC45    PC46
Standard deviation     4.39495 4.30843 4.22660 4.20932 4.12976 4.07652 4.04834
Proportion of Variance 0.00137 0.00132 0.00127 0.00126 0.00121 0.00118 0.00116
Cumulative Proportion  0.97105 0.97237 0.97363 0.97489 0.97610 0.97728 0.97845
                          PC47    PC48    PC49    PC50    PC51    PC52   PC53
Standard deviation     3.99248 3.89519 3.86841 3.81632 3.80862 3.78710 3.7466
Proportion of Variance 0.00113 0.00108 0.00106 0.00103 0.00103 0.00102 0.0010
Cumulative Proportion  0.97958 0.98066 0.98172 0.98275 0.98378 0.98480 0.9858
                          PC54    PC55    PC56    PC57   PC58    PC59    PC60
Standard deviation     3.70669 3.65110 3.64619 3.59910 3.5568 3.50638 3.46507
Proportion of Variance 0.00098 0.00095 0.00094 0.00092 0.0009 0.00087 0.00085
Cumulative Proportion  0.98677 0.98772 0.98866 0.98958 0.9905 0.99136 0.99221
                          PC61    PC62    PC63    PC64    PC65   PC66    PC67
Standard deviation     3.44803 3.40615 3.34532 3.27967 3.22481 3.1318 3.11900
Proportion of Variance 0.00084 0.00082 0.00079 0.00076 0.00074 0.0007 0.00069
Cumulative Proportion  0.99305 0.99388 0.99467 0.99543 0.99617 0.9969 0.99756
                          PC68    PC69    PC70   PC71     PC72
Standard deviation     3.10497 3.05639 2.88429 2.6597 9.29e-14
Proportion of Variance 0.00068 0.00066 0.00059 0.0005 0.00e+00
Cumulative Proportion  0.99824 0.99891 0.99950 1.0000 1.00e+00</code></pre>
</div>
<div id="typical-genes-expressed-in-ipsc-cms" class="section level2">
<h2>Typical genes expressed in iPSC-CMS</h2>
<p><img src="figure/run_all_analysis.Rmd/typical%20gene%20expression-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="correlation-heatmap-of-counts-matrix" class="section level2">
<h2>correlation heatmap of counts matrix</h2>
<p><img src="figure/run_all_analysis.Rmd/pheatmap%20clustering%20of%20gene%20counts-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="now-to-get-the-counts-set-for-deg" class="section level3">
<h3>now to get the counts set for DEG!!</h3>
<p><img src="figure/run_all_analysis.Rmd/counts%20to%20DEG%20set%20up-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="deg-analysis" class="section level1">
<h1>DEG analysis</h1>
<div id="summary" class="section level2">
<h2>summary</h2>
<pre><code>        V.DA  V.DX  V.EP  V.MT  V.TR V.DA24 V.DX24 V.EP24 V.MT24 V.TR24
Down     109     3    30    24     0   3540   3336   3105    428      0
NotSig 13552 14065 13874 14009 14084   7067   7439   7756  12969  14084
Up       423    16   180    51     0   3477   3309   3223    687      0</code></pre>
</div>
<div id="cormotif-analysis" class="section level2">
<h2>cormotif analysis</h2>
<p>Response sets look similar to previous results. This data is based on
the filtered count matrix (using rowmeans&gt;0 of cpm(log=true)).
Classification of patterns appear to be:</p>
<p>motif 1- No Response set: 7362 (gene list made by filtering each
column by posterior probability of &lt;0.55)</p>
<p>motif 2- Time-independent Top2Bi response cluster: 432</p>
<p>motif 3- Early Top2Bi response cluster: 481</p>
<p>motif 4- All BC drug response set (only 1!) (I could not really
isolate this gene from the p.post data)</p>
<p>motif 5- Late Top2Bi response cluster: 4850</p>
<p>NOTE: these are based on the most recent counts (motif numbers have
changed a little)</p>
<p>More analysis on corMotif (Baysian can be found on this page: <a
href="https://reneeisnowhere.github.io/Cardiotoxicity/Cormotifcluster_analysis.html">CorMotif</a></p>
<p>Pairwise gene analysis</p>
<p><img src="figure/run_all_analysis.Rmd/Volcanoplots-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/Volcanoplots-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/Volcanoplots-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/Volcanoplots-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/run_all_analysis.Rmd/Volcanoplots-5.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.2.2 (2022-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggpubr_0.6.0                           
 [2] Hmisc_4.8-0                            
 [3] Formula_1.2-5                          
 [4] survival_3.5-3                         
 [5] corrplot_0.92                          
 [6] ggrepel_0.9.3                          
 [7] cowplot_1.1.1                          
 [8] Homo.sapiens_1.3.1                     
 [9] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
[10] org.Hs.eg.db_3.15.0                    
[11] GO.db_3.15.0                           
[12] OrganismDbi_1.38.1                     
[13] GenomicFeatures_1.48.4                 
[14] GenomicRanges_1.48.0                   
[15] GenomeInfoDb_1.32.4                    
[16] AnnotationDbi_1.58.0                   
[17] IRanges_2.30.1                         
[18] S4Vectors_0.34.0                       
[19] Biobase_2.56.0                         
[20] biomaRt_2.52.0                         
[21] scales_1.2.1                           
[22] lubridate_1.9.2                        
[23] forcats_1.0.0                          
[24] stringr_1.5.0                          
[25] dplyr_1.1.0                            
[26] purrr_1.0.1                            
[27] readr_2.1.4                            
[28] tidyr_1.3.0                            
[29] tibble_3.1.8                           
[30] tidyverse_2.0.0                        
[31] AnnotationHub_3.4.0                    
[32] BiocFileCache_2.4.0                    
[33] dbplyr_2.3.1                           
[34] BiocGenerics_0.42.0                    
[35] data.table_1.14.8                      
[36] reshape2_1.4.4                         
[37] gridExtra_2.3                          
[38] mixOmics_6.20.0                        
[39] ggplot2_3.4.1                          
[40] lattice_0.20-45                        
[41] MASS_7.3-58.2                          
[42] RColorBrewer_1.1-3                     
[43] edgeR_3.38.4                           
[44] limma_3.52.4                           
[45] workflowr_1.7.0                        

loaded via a namespace (and not attached):
  [1] utf8_1.2.3                    tidyselect_1.2.0             
  [3] RSQLite_2.3.0                 htmlwidgets_1.6.1            
  [5] grid_4.2.2                    BiocParallel_1.30.4          
  [7] munsell_0.5.0                 codetools_0.2-19             
  [9] statmod_1.5.0                 interp_1.1-3                 
 [11] withr_2.5.0                   colorspace_2.1-0             
 [13] filelock_1.0.2                highr_0.10                   
 [15] knitr_1.42                    rstudioapi_0.14              
 [17] ggsignif_0.6.4                MatrixGenerics_1.8.1         
 [19] labeling_0.4.2                git2r_0.31.0                 
 [21] GenomeInfoDbData_1.2.8        pheatmap_1.0.12              
 [23] farver_2.1.1                  bit64_4.0.5                  
 [25] rprojroot_2.0.3               vctrs_0.5.2                  
 [27] generics_0.1.3                xfun_0.37                    
 [29] timechange_0.2.0              R6_2.5.1                     
 [31] locfit_1.5-9.7                bitops_1.0-7                 
 [33] cachem_1.0.7                  DelayedArray_0.22.0          
 [35] promises_1.2.0.1              BiocIO_1.6.0                 
 [37] nnet_7.3-18                   gtable_0.3.1                 
 [39] processx_3.8.0                rlang_1.0.6                  
 [41] splines_4.2.2                 rtracklayer_1.56.1           
 [43] rstatix_0.7.2                 broom_1.0.3                  
 [45] checkmate_2.1.0               BiocManager_1.30.20          
 [47] yaml_2.3.7                    abind_1.4-5                  
 [49] backports_1.4.1               httpuv_1.6.9                 
 [51] RBGL_1.72.0                   tools_4.2.2                  
 [53] ellipsis_0.3.2                jquerylib_0.1.4              
 [55] Rcpp_1.0.10                   plyr_1.8.8                   
 [57] base64enc_0.1-3               progress_1.2.2               
 [59] zlibbioc_1.42.0               RCurl_1.98-1.10              
 [61] ps_1.7.2                      prettyunits_1.1.1            
 [63] rpart_4.1.19                  deldir_1.0-6                 
 [65] SummarizedExperiment_1.26.1   cluster_2.1.4                
 [67] fs_1.6.1                      magrittr_2.0.3               
 [69] RSpectra_0.16-1               whisker_0.4.1                
 [71] matrixStats_0.63.0            hms_1.1.2                    
 [73] mime_0.12                     evaluate_0.20                
 [75] xtable_1.8-4                  XML_3.99-0.13                
 [77] jpeg_0.1-10                   compiler_4.2.2               
 [79] ellipse_0.4.3                 crayon_1.5.2                 
 [81] htmltools_0.5.4               corpcor_1.6.10               
 [83] later_1.3.0                   tzdb_0.3.0                   
 [85] DBI_1.1.3                     rappdirs_0.3.3               
 [87] Matrix_1.5-3                  car_3.1-1                    
 [89] cli_3.6.0                     parallel_4.2.2               
 [91] igraph_1.4.1                  pkgconfig_2.0.3              
 [93] getPass_0.2-2                 GenomicAlignments_1.32.1     
 [95] foreign_0.8-84                xml2_1.3.3                   
 [97] rARPACK_0.11-0                bslib_0.4.2                  
 [99] XVector_0.36.0                callr_3.7.3                  
[101] digest_0.6.31                 graph_1.74.0                 
[103] Biostrings_2.64.1             rmarkdown_2.20               
[105] htmlTable_2.4.1               restfulr_0.0.15              
[107] curl_5.0.0                    shiny_1.7.4                  
[109] Rsamtools_2.12.0              rjson_0.2.21                 
[111] lifecycle_1.0.3               jsonlite_1.8.4               
[113] carData_3.0-5                 fansi_1.0.4                  
[115] pillar_1.8.1                  KEGGREST_1.36.3              
[117] fastmap_1.1.1                 httr_1.4.5                   
[119] interactiveDisplayBase_1.34.0 glue_1.6.2                   
[121] png_0.1-8                     BiocVersion_3.15.2           
[123] bit_4.0.5                     stringi_1.7.12               
[125] sass_0.4.5                    blob_1.2.3                   
[127] latticeExtra_0.6-30           memoise_2.0.1                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
