<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="ERM" />

<meta name="date" content="2024-02-12" />

<title>Supplementary figures</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cardiotoxicity</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/reneeisnowhere/Cardiotoxicity">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Supplementary figures</h1>
<h4 class="author">ERM</h4>
<h4 class="date">2024-02-12</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-02-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Cardiotoxicity/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230109code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230109)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230109code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230109)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommwardlabMatthewsTOP2icardiotox2023tree83b8420b9da0addcdc1572a0298923cf59e38347targetblank83b8420a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/tree/83b8420b9da0addcdc1572a0298923cf59e38347" target="_blank">83b8420</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommwardlabMatthewsTOP2icardiotox2023tree83b8420b9da0addcdc1572a0298923cf59e38347targetblank83b8420a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/tree/83b8420b9da0addcdc1572a0298923cf59e38347" target="_blank">83b8420</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/variance_values by gene.png
    Ignored:    data/41588_2018_171_MOESM3_ESMeQTL_ST2_for paper.csv
    Ignored:    data/Arr_GWAS.txt
    Ignored:    data/Arr_geneset.RDS
    Ignored:    data/BC_cell_lines.csv
    Ignored:    data/BurridgeDOXTOX.RDS
    Ignored:    data/CADGWASgene_table.csv
    Ignored:    data/CAD_geneset.RDS
    Ignored:    data/CALIMA_Data/
    Ignored:    data/CMD04_75DRCviability.csv
    Ignored:    data/CMD04_87DRCviability.csv
    Ignored:    data/CMD05_75DRCviability.csv
    Ignored:    data/CMD05_87DRCviability.csv
    Ignored:    data/Clamp_Summary.csv
    Ignored:    data/Cormotif_24_k1-5_raw.RDS
    Ignored:    data/Counts_RNA_ERMatthews.txt
    Ignored:    data/DAgostres24.RDS
    Ignored:    data/DAtable1.csv
    Ignored:    data/DDEMresp_list.csv
    Ignored:    data/DDE_reQTL.txt
    Ignored:    data/DDEresp_list.csv
    Ignored:    data/DEG-GO/
    Ignored:    data/DEG_cormotif.RDS
    Ignored:    data/DF_Plate_Peak.csv
    Ignored:    data/DRC48hoursdata.csv
    Ignored:    data/Da24counts.txt
    Ignored:    data/Dx24counts.txt
    Ignored:    data/Dx_reQTL_specific.txt
    Ignored:    data/EPIstorelist24.RDS
    Ignored:    data/Ep24counts.txt
    Ignored:    data/FC_necela.RDS
    Ignored:    data/FC_necela_names.RDS
    Ignored:    data/Full_LD_rep.csv
    Ignored:    data/GOIsig.csv
    Ignored:    data/GOplots.R
    Ignored:    data/GTEX_setsimple.csv
    Ignored:    data/GTEX_sig24.RDS
    Ignored:    data/GTEx_gene_list.csv
    Ignored:    data/HFGWASgene_table.csv
    Ignored:    data/HF_geneset.RDS
    Ignored:    data/Heart_Left_Ventricle.v8.egenes.txt
    Ignored:    data/Heatmap_mat.RDS
    Ignored:    data/Heatmap_sig.RDS
    Ignored:    data/Hf_GWAS.txt
    Ignored:    data/K_cluster
    Ignored:    data/K_cluster_kisthree.csv
    Ignored:    data/K_cluster_kistwo.csv
    Ignored:    data/Knowles_log2cpm_real.RDS
    Ignored:    data/Knowles_variation_data.RDS
    Ignored:    data/Knowles_variation_data_conc.RDS
    Ignored:    data/Knowlesvarlist.RDS
    Ignored:    data/LD50_05via.csv
    Ignored:    data/LDH48hoursdata.csv
    Ignored:    data/Mt24counts.txt
    Ignored:    data/NoRespDEG_final.csv
    Ignored:    data/RINsamplelist.txt
    Ignored:    data/RNA_seq_trial.RDS
    Ignored:    data/Schneider_GWAS.txt
    Ignored:    data/Seonane2019supp1.txt
    Ignored:    data/Sup_replicate_values.csv
    Ignored:    data/TMMnormed_x.RDS
    Ignored:    data/TOP2Bi-24hoursGO_analysis.csv
    Ignored:    data/TR24counts.txt
    Ignored:    data/TableS10.csv
    Ignored:    data/TableS11.csv
    Ignored:    data/TableS9.csv
    Ignored:    data/Top2_expression.RDS
    Ignored:    data/Top2biresp_cluster24h.csv
    Ignored:    data/Var_test_list.RDS
    Ignored:    data/Var_test_list24.RDS
    Ignored:    data/Var_test_list24alt.RDS
    Ignored:    data/Var_test_list3.RDS
    Ignored:    data/Vargenes.RDS
    Ignored:    data/Viabilitylistfull.csv
    Ignored:    data/allexpressedgenes.txt
    Ignored:    data/allfinal3hour.RDS
    Ignored:    data/allgenes.txt
    Ignored:    data/allmatrix.RDS
    Ignored:    data/allmymatrix.RDS
    Ignored:    data/annotation_data_frame.RDS
    Ignored:    data/averageviabilitytable.RDS
    Ignored:    data/averageviabilitytable.csv
    Ignored:    data/avgLD50.RDS
    Ignored:    data/avg_LD50.RDS
    Ignored:    data/avg_via_table.csv
    Ignored:    data/backGL.txt
    Ignored:    data/burr_genes.RDS
    Ignored:    data/calcium_data.RDS
    Ignored:    data/clamp_summary.RDS
    Ignored:    data/cormotif_3hk1-8.RDS
    Ignored:    data/cormotif_initalK5.RDS
    Ignored:    data/cormotif_initialK5.RDS
    Ignored:    data/cormotif_initialall.RDS
    Ignored:    data/cormotifprobs.csv
    Ignored:    data/counts24hours.RDS
    Ignored:    data/cpmcount.RDS
    Ignored:    data/cpmnorm_counts.csv
    Ignored:    data/crispr_genes.csv
    Ignored:    data/ctnnt_results.txt
    Ignored:    data/cvd_GWAS.txt
    Ignored:    data/dat_cpm.RDS
    Ignored:    data/data_outline.txt
    Ignored:    data/drug_noveh1.csv
    Ignored:    data/efit2.RDS
    Ignored:    data/efit2_final.RDS
    Ignored:    data/efit2results.RDS
    Ignored:    data/ensembl_backup.RDS
    Ignored:    data/ensgtotal.txt
    Ignored:    data/filcpm_counts.RDS
    Ignored:    data/filenameonly.txt
    Ignored:    data/filtered_cpm_counts.csv
    Ignored:    data/filtered_raw_counts.csv
    Ignored:    data/filtermatrix_x.RDS
    Ignored:    data/folder_05top/
    Ignored:    data/framefun24.RDS
    Ignored:    data/geneDoxonlyQTL.csv
    Ignored:    data/gene_corr_df.RDS
    Ignored:    data/gene_corr_frame.RDS
    Ignored:    data/gene_prob_tran3h.RDS
    Ignored:    data/gene_probabilityk5.RDS
    Ignored:    data/geneset_24.RDS
    Ignored:    data/gostresTop2bi_ER.RDS
    Ignored:    data/gostresTop2bi_LR
    Ignored:    data/gostresTop2bi_LR.RDS
    Ignored:    data/gostresTop2bi_TI.RDS
    Ignored:    data/gostrescoNR
    Ignored:    data/gtex/
    Ignored:    data/heartgenes.csv
    Ignored:    data/highly_var_genelist.RDS
    Ignored:    data/hsa_kegg_anno.RDS
    Ignored:    data/individualDRCfile.RDS
    Ignored:    data/individual_DRC48.RDS
    Ignored:    data/individual_LDH48.RDS
    Ignored:    data/indv_noveh1.csv
    Ignored:    data/kegglistDEG.RDS
    Ignored:    data/kegglistDEG24.RDS
    Ignored:    data/kegglistDEG3.RDS
    Ignored:    data/knowfig4.csv
    Ignored:    data/knowfig5.csv
    Ignored:    data/label_list.RDS
    Ignored:    data/ld50_table.csv
    Ignored:    data/mean_vardrug1.csv
    Ignored:    data/mean_varframe.csv
    Ignored:    data/mymatrix.RDS
    Ignored:    data/new_ld50avg.RDS
    Ignored:    data/nonresponse_cluster24h.csv
    Ignored:    data/norm_LDH.csv
    Ignored:    data/norm_counts.csv
    Ignored:    data/old_sets/
    Ignored:    data/organized_drugframe.csv
    Ignored:    data/pca_all_anno.csv
    Ignored:    data/plan2plot.png
    Ignored:    data/plot_intv_list.RDS
    Ignored:    data/plot_list_DRC.RDS
    Ignored:    data/qval24hr.RDS
    Ignored:    data/qval3hr.RDS
    Ignored:    data/qvalueEPItemp.RDS
    Ignored:    data/raw_counts.csv
    Ignored:    data/response_cluster24h.csv
    Ignored:    data/sampsettrz.RDS
    Ignored:    data/schneider_closest_output.RDS
    Ignored:    data/sigVDA24.txt
    Ignored:    data/sigVDA3.txt
    Ignored:    data/sigVDX24.txt
    Ignored:    data/sigVDX3.txt
    Ignored:    data/sigVEP24.txt
    Ignored:    data/sigVEP3.txt
    Ignored:    data/sigVMT24.txt
    Ignored:    data/sigVMT3.txt
    Ignored:    data/sigVTR24.txt
    Ignored:    data/sigVTR3.txt
    Ignored:    data/siglist.RDS
    Ignored:    data/siglist_final.RDS
    Ignored:    data/siglist_old.RDS
    Ignored:    data/slope_table.csv
    Ignored:    data/supp10_24hlist.RDS
    Ignored:    data/supp10_3hlist.RDS
    Ignored:    data/supp_normLDH48.RDS
    Ignored:    data/supp_pca_all_anno.RDS
    Ignored:    data/supp_viadata.csv
    Ignored:    data/table3a.omar
    Ignored:    data/test_run_sample_list.txt
    Ignored:    data/testlist.txt
    Ignored:    data/toplistall.RDS
    Ignored:    data/trtonly_24h_genes.RDS
    Ignored:    data/trtonly_3h_genes.RDS
    Ignored:    data/tvl24hour.txt
    Ignored:    data/tvl24hourw.txt
    Ignored:    data/venn_code.R
    Ignored:    data/viability.RDS

Untracked files:
    Untracked:  .RDataTmp
    Untracked:  .RDataTmp1
    Untracked:  .RDataTmp2
    Untracked:  .RDataTmp3
    Untracked:  3hr all.pdf
    Untracked:  Code_files_list.csv
    Untracked:  Data_files_list.csv
    Untracked:  Doxorubicin_vehicle_3_24.csv
    Untracked:  Doxtoplist.csv
    Untracked:  EPIqvalue_analysis.Rmd
    Untracked:  Final.sup.pdf
    Untracked:  GWAS_list_of_interest.xlsx
    Untracked:  KEGGpathwaylist.R
    Untracked:  NA
    Untracked:  OmicNavigator_learn.R
    Untracked:  SNP_egenes_allfiles.RDS
    Untracked:  SNP_frame_pdf
    Untracked:  SNP_frame_pdf.pdf
    Untracked:  SigDoxtoplist.csv
    Untracked:  analysis/DRC_viability_check.Rmd
    Untracked:  analysis/New_code_dec-23.R
    Untracked:  analysis/cellcycle_kegg_genes.R
    Untracked:  analysis/ciFIT.R
    Untracked:  analysis/export_to_excel.R
    Untracked:  analysis/featureCountsPLAY.R
    Untracked:  cleanupfiles_script.R
    Untracked:  code/biomart_gene_names.R
    Untracked:  code/constantcode.R
    Untracked:  code/corMotifcustom.R
    Untracked:  code/cpm_boxplot.R
    Untracked:  code/extracting_ggplot_data.R
    Untracked:  code/movingfilesto_ppl.R
    Untracked:  code/pearson_extract_func.R
    Untracked:  code/pearson_tox_extract.R
    Untracked:  code/plot1C.fun.R
    Untracked:  code/spearman_extract_func.R
    Untracked:  code/venndiagramcolor_control.R
    Untracked:  cormotif_p.post.list_4.csv
    Untracked:  figS1024h.pdf
    Untracked:  final.pdf
    Untracked:  individual-legenddark2.png
    Untracked:  installed_old.rda
    Untracked:  listoftranscripts
    Untracked:  motif_ER.txt
    Untracked:  motif_LR.txt
    Untracked:  motif_NR.txt
    Untracked:  motif_TI.txt
    Untracked:  output/ABHD8_dif_values.RDS
    Untracked:  output/C3orf18_dif_values.RDS
    Untracked:  output/Cardiotox_dif_values.RDS
    Untracked:  output/DNR_DEGlist.csv
    Untracked:  output/DNRvenn.RDS
    Untracked:  output/DOX_DEGlist.csv
    Untracked:  output/DOX_de_goi.csv
    Untracked:  output/DOXvenn.RDS
    Untracked:  output/EEF1B2_dif_values.RDS
    Untracked:  output/EEIG1_dif_values.RDS
    Untracked:  output/EPI_DEGlist.csv
    Untracked:  output/EPIvenn.RDS
    Untracked:  output/ESGN_rds.RDS
    Untracked:  output/FC_necela.RDS
    Untracked:  output/FC_necela_names.RDS
    Untracked:  output/FRS2_dif_values.RDS
    Untracked:  output/Figures/
    Untracked:  output/GTEXv8_gene_median_tpm.RDS
    Untracked:  output/GTEXv8_gene_tpm_heart_left_ventricle.RDS
    Untracked:  output/HDDC2_dif_values.RDS
    Untracked:  output/HER2_gene.RDS
    Untracked:  output/KEGGcellcyclegenes.RDS
    Untracked:  output/Knowles_S13.csv
    Untracked:  output/Knowles_log2cpm.csv
    Untracked:  output/Knowles_supp13.csv
    Untracked:  output/LD50tox_table.RDS
    Untracked:  output/MTX_DEGlist.csv
    Untracked:  output/MTXvenn.RDS
    Untracked:  output/PEX16_dif_values.RDS
    Untracked:  output/RASIP1_dif_values.RDS
    Untracked:  output/RMI1_dif_values.RDS
    Untracked:  output/RSID_QTL_list_full.txt
    Untracked:  output/SETA_analysis_reyes.RDS
    Untracked:  output/SGWAS_top50_order.csv
    Untracked:  output/SLC27A1_dif_values.RDS
    Untracked:  output/SLC28A3_dif_values.RDS
    Untracked:  output/SNP_egenes_allfiles.RDS
    Untracked:  output/SNP_list_ID.RDS
    Untracked:  output/SNP_list_full.txt
    Untracked:  output/SNP_supp.RDS
    Untracked:  output/TGFBR3L_dif_values.RDS
    Untracked:  output/TNS2_dif_values.RDS
    Untracked:  output/TOP_50SNPreffile.csv
    Untracked:  output/TRZ_DEGlist.csv
    Untracked:  output/TableS8.csv
    Untracked:  output/Volcanoplot_10
    Untracked:  output/Volcanoplot_10.RDS
    Untracked:  output/ZNF740_dif_values.RDS
    Untracked:  output/allfinal_sup10.RDS
    Untracked:  output/counts_v8_heart_left_ventricle_gct.RDS
    Untracked:  output/crisprfoldchange.RDS
    Untracked:  output/endocytosisgenes.csv
    Untracked:  output/expre7k.csv
    Untracked:  output/expressed_egenes_by_RSID.csv
    Untracked:  output/gene_corr_fig9.RDS
    Untracked:  output/genes.RDS
    Untracked:  output/legend_b.RDS
    Untracked:  output/motif_ERrep.RDS
    Untracked:  output/motif_LRrep.RDS
    Untracked:  output/motif_NRrep.RDS
    Untracked:  output/motif_TI_rep.RDS
    Untracked:  output/near_genes_SNP1.RDS
    Untracked:  output/necela_list_test.RDS
    Untracked:  output/necela_val_genes.RDS
    Untracked:  output/output-old/
    Untracked:  output/rank24genes.csv
    Untracked:  output/rank3genes.csv
    Untracked:  output/sequencinginformationforsupp.csv
    Untracked:  output/sequencinginformationforsupp.prn
    Untracked:  output/sigVDA24.txt
    Untracked:  output/sigVDA3.txt
    Untracked:  output/sigVDX24.txt
    Untracked:  output/sigVDX3.txt
    Untracked:  output/sigVEP24.txt
    Untracked:  output/sigVEP3.txt
    Untracked:  output/sigVMT24.txt
    Untracked:  output/sigVMT3.txt
    Untracked:  output/sigVTR24.txt
    Untracked:  output/sigVTR3.txt
    Untracked:  output/supplementary_motif_list_GO.RDS
    Untracked:  output/test_biomart_run.RDS
    Untracked:  output/toptablebydrug.RDS
    Untracked:  output/trop_knowles_fun.csv
    Untracked:  output/tvl24hour.txt
    Untracked:  output/x_counts.RDS
    Untracked:  reneebasecode.R

Unstaged changes:
    Modified:   analysis/DRC_analysis.Rmd
    Modified:   analysis/Figure4.Rmd
    Modified:   analysis/Figure5.Rmd
    Modified:   analysis/GTEx_genes.Rmd
    Deleted:    analysis/Knowles2019.Rmd
    Modified:   output/daplot.RDS
    Modified:   output/dxplot.RDS
    Modified:   output/epplot.RDS
    Modified:   output/mtplot.RDS
    Modified:   output/plan2plot.png
    Modified:   output/trplot.RDS
    Modified:   output/veplot.RDS

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Supplementary_figures.Rmd</code>)
and HTML (<code>docs/Supplementary_figures.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/83b8420b9da0addcdc1572a0298923cf59e38347/analysis/Supplementary_figures.Rmd" target="_blank">83b8420</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-12
</td>
<td>
correction to markdown
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mward-lab/Matthews_TOP2i_cardiotox_2023/a08048ce6f157e2de0013a13380df6fed9af6176/docs/Supplementary_figures.html" target="_blank">a08048c</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/2d6ed9bc0f92fe436f9ce0a8d550f1cc994f7a91/analysis/Supplementary_figures.Rmd" target="_blank">2d6ed9b</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-06
</td>
<td>
TRZ GO added
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mward-lab/Matthews_TOP2i_cardiotox_2023/652df932402672208f615b21dc6dd91860a833aa/docs/Supplementary_figures.html" target="_blank">652df93</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/b5afd08da5b18716904aaeb69ee8273051393712/analysis/Supplementary_figures.Rmd" target="_blank">b5afd08</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-06
</td>
<td>
updates to figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mward-lab/Matthews_TOP2i_cardiotox_2023/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/Supplementary_figures.html" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/c553b1c48664977510194c16216ebb9f5adb2909/analysis/Supplementary_figures.Rmd" target="_blank">c553b1c</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
<td>
rearrange and add CSS and links
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mward-lab/Matthews_TOP2i_cardiotox_2023/a7737443a7d5abc68a3213e3ae14b59c484463b1/docs/Supplementary_figures.html" target="_blank">a773744</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-08-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/b30e0e4a511b9a81071bd4d942e612a03045bae9/analysis/Supplementary_figures.Rmd" target="_blank">b30e0e4</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-08-10
</td>
<td>
adding fig3
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mward-lab/Matthews_TOP2i_cardiotox_2023/021ae6df9b1a6106f052163e95f11f7d3af834aa/docs/Supplementary_figures.html" target="_blank">021ae6d</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-08-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/04026cea6afd636419951f1510b0fcca4d60df34/analysis/Supplementary_figures.Rmd" target="_blank">04026ce</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-08-08
</td>
<td>
adding svglite
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/b0c1fb85df3c903a9d7d0a124e758ef642c1726d/analysis/Supplementary_figures.Rmd" target="_blank">b0c1fb8</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-08-01
</td>
<td>
updated code but not published yet
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mward-lab/Matthews_TOP2i_cardiotox_2023/fa4c610dc54bb403987e8af13f1573f060f25e51/docs/Supplementary_figures.html" target="_blank">fa4c610</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/b2ef59a34a9946b525a05c75e40be37101473870/analysis/Supplementary_figures.Rmd" target="_blank">b2ef59a</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-28
</td>
<td>
new figures update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mward-lab/Matthews_TOP2i_cardiotox_2023/b31f5962c4b2eb28ce7f5c42fdff2edfe8c5621e/docs/Supplementary_figures.html" target="_blank">b31f596</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/fdb94fb9f9cb372b820c3b5fa8e24cf4a4f7d569/analysis/Supplementary_figures.Rmd" target="_blank">fdb94fb</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-28
</td>
<td>
new figures update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/62286c39df588d70d06650c1569865a76958bc09/analysis/Supplementary_figures.Rmd" target="_blank">62286c3</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-28
</td>
<td>
Updateing figure code
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mward-lab/Matthews_TOP2i_cardiotox_2023/680af3aae7258383999e2b9aa8cc74449bd91cfe/docs/Supplementary_figures.html" target="_blank">680af3a</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d7b9ff1f1a8e0542b9b4a951b1e7f0de6298d0fc/analysis/Supplementary_figures.Rmd" target="_blank">d7b9ff1</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
<td>
Adding supplementary figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mward-lab/Matthews_TOP2i_cardiotox_2023/537bc2eaae49e2ce7fd9561bcf6d0667227a10a8/docs/Supplementary_figures.html" target="_blank">537bc2e</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/6e50959c3929749167be6933504c7238536acf64/analysis/Supplementary_figures.Rmd" target="_blank">6e50959</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
<td>
Adding supplementary figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mward-lab/Matthews_TOP2i_cardiotox_2023/dace8badd39879fefd85529bd1688e75a35d5adc/docs/Supplementary_figures.html" target="_blank">dace8ba</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/2b109c3cc369bba77a84b6a33681e8f9ff0bfbe4/analysis/Supplementary_figures.Rmd" target="_blank">2b109c3</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-23
</td>
<td>
adding some supp graphs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/c1d667f03ba3bb1dc3cfb40559c8991a31f74974/analysis/Supplementary_figures.Rmd" target="_blank">c1d667f</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-23
</td>
<td>
updating the codes at Friday start.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<style type="text/css">
pre {
  max-height: 400px;
  overflow-y: auto;
}

pre[class] {
  max-height: 200px;
}
</style>
<pre class="r"><code>library(tidyverse)
library(ggpubr)
library(rstatix)
library(zoo)
library(ggsignif)
library(RColorBrewer)
library(grid)
library(scales)
library(ComplexHeatmap)
library(gridExtra)
library(cowplot)
library(drc)

library(kableExtra)</code></pre>
<pre><code>Error: package or namespace load failed for &#39;kableExtra&#39;:
 .onLoad failed in loadNamespace() for &#39;kableExtra&#39;, details:
  call: !is.null(rmarkdown::metadata$output) &amp;&amp; rmarkdown::metadata$output %in% 
  error: &#39;length = 3&#39; in coercion to &#39;logical(1)&#39;</code></pre>
<pre class="r"><code>library(broom)
library(ggVennDiagram)</code></pre>
<div id="function-to-graph-gene-expression-across-treatments"
class="section level4">
<h4>Function to graph gene expression across treatments</h4>
<pre class="r"><code>cpm_boxplot_time &lt;-function(cpmcounts,timex, GOI,brewer_palette, fill_colors, ylab) {
  ##GOI needs to be ENTREZID
  df &lt;- cpmcounts
    df_plot &lt;- df %&gt;% 
      dplyr::filter(rownames(.)==GOI) %&gt;%
      pivot_longer(everything(),
                   names_to = &quot;treatment&quot;,
                   values_to = &quot;counts&quot;) %&gt;%
      separate(treatment, c(&quot;drug&quot;,&quot;indv&quot;,&quot;time&quot;)) %&gt;%
      dplyr::filter(time == paste(timex)) %&gt;% 
      mutate(time=factor(time, levels =c(&quot;3h&quot;, &quot;24h&quot;), labels=c(&quot;3 hours&quot;, &quot;24 hours&quot;))) %&gt;%
      
      mutate(indv=factor(indv, levels = c(1,2,3,4,5,6))) %&gt;%
      mutate(drug =case_match(drug, &quot;Da&quot;~&quot;DNR&quot;,
                            &quot;Do&quot;~&quot;DOX&quot;,
                            &quot;Ep&quot;~&quot;EPI&quot;,
                            &quot;Mi&quot;~&quot;MTX&quot;,
                            &quot;Tr&quot;~&quot;TRZ&quot;,
                            &quot;Ve&quot;~&quot;VEH&quot;, .default = drug)) %&gt;% 
      mutate(drug=factor(drug, levels = c(&#39;DOX&#39;,&#39;EPI&#39;,&#39;DNR&#39;,&#39;MTX&#39;,&#39;TRZ&#39;,&#39;VEH&#39;)))
    plot &lt;- ggplot2::ggplot(df_plot, aes(x=drug, y=counts))+
      geom_boxplot(position=&quot;identity&quot;,aes(fill=drug))+
      geom_point(aes(col=indv, size=1.5, alpha=0.5))+
      guides(alpha= &quot;none&quot;, size= &quot;none&quot;)+
      scale_color_brewer(palette = brewer_palette, guide = &quot;none&quot;)+
      scale_fill_manual(values=fill_colors)+
      # try(facet_wrap(&quot;time&quot;, nrow=1, ncol=2))+
      theme_bw()+
      ylab(ylab)+
      xlab(&quot;&quot;)+
      theme(strip.background = element_rect(fill = &quot;white&quot;,linetype=1, linewidth = 0.5),
          plot.title = element_text(size=10,hjust = 0.5,face=&quot;bold&quot;),
          axis.title = element_text(size = 10, color = &quot;black&quot;),
          axis.ticks = element_line(linewidth = 1.0),
          axis.line = element_line(linewidth = 1.0),
          axis.text.x = element_blank(),
          strip.text.x = element_text(margin = margin(2,0,2,0, &quot;pt&quot;),face = &quot;bold&quot;))
    print(plot)
}

drug_pal &lt;- c(&quot;#707031&quot;,&quot;#41B333&quot;)
# use &quot;Dark2&quot; for individuals, and drug_pal for drug fill

cpm_TRZ &lt;-
  function(cpmcounts,
           GOI,
           ylab,
           titlename) {
    ##GOI needs to be ENTREZID
    df &lt;- trz_veh_count %&gt;% 
      column_to_rownames(&quot;ENTREZID&quot;)
    df_plot &lt;- df %&gt;%
      dplyr::filter(rownames(.) == GOI) %&gt;%
      pivot_longer(everything(),
                   names_to = &quot;treatment&quot;,
                   values_to = &quot;counts&quot;) %&gt;%
      separate(treatment, c(&quot;drug&quot;, &quot;indv&quot;, &quot;time&quot;)) %&gt;%
      mutate(time = factor(time, levels = c(&quot;3h&quot;, &quot;24h&quot;))) %&gt;%
      mutate(indv = factor(indv, levels = c(1, 2, 3, 4, 5, 6))) %&gt;%
      mutate(
        drug = case_match(
          drug,
          &quot;Da&quot; ~ &quot;DNR&quot;,
          &quot;Do&quot; ~ &quot;DOX&quot;,
          &quot;Ep&quot; ~ &quot;EPI&quot;,
          &quot;Mi&quot; ~ &quot;MTX&quot;,
          &quot;Tr&quot; ~ &quot;TRZ&quot;,
          &quot;Ve&quot; ~ &quot;VEH&quot;,
          .default = drug
        )
      ) %&gt;%
      mutate(drug = factor(drug, levels = c(&#39;DOX&#39;, &#39;EPI&#39;, &#39;DNR&#39;, &#39;MTX&#39;, &#39;TRZ&#39;, &#39;VEH&#39;)))
    plot &lt;-
    ggplot2::ggplot(df_plot, aes(x = drug, y = counts)) +
      geom_boxplot(position = &quot;identity&quot;, aes(fill = drug)) +
      geom_point(aes(
        col = indv,
        size = 1.5,
        alpha = 0.5
      )) +
      guides(alpha = &quot;none&quot;, size = &quot;none&quot;) +
      # scale_color_brewer(palette = &quot;Dark2&quot;, guide = &quot;none&quot;) +
      scale_fill_manual(values = drug_pal) +
      facet_wrap(&quot;time&quot;, nrow = 1, ncol = 2) +
      theme_bw() +
      ylab(ylab) +
      xlab(&quot;&quot;) +
      ggtitle(titlename)+
      theme(
        strip.background = element_rect(
          fill = &quot;white&quot;,
          linetype = 1,
          linewidth = 0.5
        ),
        plot.title = element_text(
          size = 10,
          hjust = 0.5,
          face = &quot;bold&quot;
        ),
        axis.title = element_text(size = 10, color = &quot;black&quot;),
        axis.ticks = element_line(linewidth = 1.0),
        axis.line = element_line(linewidth = 1.0),
        axis.text.x = element_blank(),
        strip.text.x = element_text(margin = margin(2, 0, 2, 0, &quot;pt&quot;), face = &quot;bold&quot;)
      )
    print(plot)
  }</code></pre>
</div>
<div
id="s1-fig-cardiomyocytes-can-be-generated-at-high-purity-across-six-individuals."
class="section level2">
<h2>S1 Fig: Cardiomyocytes can be generated at high purity across six
individuals.</h2>
<div id="s1-figa-flow-cytometry-results" class="section level3">
<h3>S1 FigA Flow cytometry results</h3>
<p>Please see the paper for this figure. It was generated using
FloJo.</p>
</div>
<div id="s1-figb" class="section level3">
<h3>S1 FigB</h3>
<pre class="r"><code>ctnnt &lt;- read.csv(&quot;data/ctnnt_results.txt&quot;, row.names = 1)
ctnnt %&gt;% 
  mutate(Individual=fct_inorder(Individual)) %&gt;% 
  ggplot(., aes(Individual,Percent , fill=Individual))+
  geom_boxplot()+
  geom_point()+
  geom_hline(yintercept =70,linetype=&quot;dashed&quot;, alpha=0.75)+###adds a line indicating high positivity +
coord_cartesian(ylim = c(0,105))+ ##set those limits
  theme_bw()+  ##white background
  labs(title=&quot;Cardiomyocyte Purity&quot;)+ #subtitle = &quot;from  n&gt;3 differentiations&quot;)+
  geom_boxplot(color=&quot;black&quot;,alpha =0.2, fill=NA, fatten=0, show.legend = FALSE)+
  scale_fill_brewer(palette = &quot;Dark2&quot;,name=&quot;&quot; )+
  xlab(NULL)+ 
  ylab(&quot;% TNNT2+ &quot;)+
  guides(fill = NULL)+
  theme(plot.title = element_text(hjust = 0.5, size =20, face= &quot;bold&quot;), 
        axis.title.x=element_blank(),
        axis.text.x=element_blank(),###removes all axis names and tick names etc.####
        axis.ticks.x=element_blank(),
        # legend.text=element_text(size=15), 
        axis.title.y=element_text(size=15),
        axis.ticks.y=element_line(size =2),
        axis.text.y=element_text(size=10, face = &quot;bold&quot;),
       panel.grid.major = element_line(colour = &#39;grey&#39;),
       panel.border=element_rect(fill = NA, size = 3),
       plot.subtitle=element_text(size=18, hjust=0.5, face=&quot;italic&quot;, color=&quot;black&quot;)) </code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S1%20troponinT-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S1-troponinT-1">
Past versions of "Fig S1 troponinT-1.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S1-troponinT-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S1 troponinT-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/b31f5962c4b2eb28ce7f5c42fdff2edfe8c5621e/docs/figure/Supplementary_figures.Rmd/Fig S1 troponinT-1.png" target="_blank">b31f596</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># ctnnt %&gt;%
#   summary() %&gt;%
#   kable(., caption= &quot;Stats summary of cTNNT+ FACs readings&quot;) %&gt;%
#   kable_paper(&quot;striped&quot;, full_width = FALSE) #%&gt;%
#   kable_styling(full_width = FALSE,font_size = 18) #%&gt;%
#   # scroll_box(width = &quot;60%&quot;, height = &quot;400px&quot;)</code></pre>
</div>
</div>
<div
id="s2-fig-dose-response-curves-are-reproducible-across-replicate-cardiomyocyte-differentiations-from-the-same-individual."
class="section level2">
<h2>S2 Fig: Dose-response curves are reproducible across replicate
cardiomyocyte differentiations from the same individual.</h2>
<pre class="r"><code>drug_pal_fact &lt;- c(&quot;#8B006D&quot;,&quot;#DF707E&quot;,&quot;#F1B72B&quot;, &quot;#3386DD&quot;,&quot;#707031&quot;,&quot;#41B333&quot;)
library(data.table)
conf_int &lt;- readRDS(&quot;data/plot_intv_list.RDS&quot;)
DRC_list &lt;- readRDS(&quot;data/plot_list_DRC.RDS&quot;)
pull_drc2 &lt;- data.frame(&quot;ind1&quot;, &quot;ind2&quot;,&quot;ind3&quot;,&quot;ind4&quot;,&quot;ind5&quot;,&quot;ind6&quot;)
doubl_plot &lt;- data.frame(&quot;ind3a&quot;, &quot;ind3b&quot;, &quot;ind5a&quot;, &quot;ind5b&quot;)
color_order &lt;- c(&#39;1&#39;,&#39;2&#39;,&#39;5&#39;,&#39;4&#39;,&#39;3&#39;,&#39;6&#39;)
intervals &lt;- rbindlist(conf_int,idcol=&quot;trt&quot;)
drug_list &lt;- c(&quot;DNR&quot;,&quot;DOX&quot;,&quot;EPI&quot;,&quot;MTX&quot;, &quot;TRZ&quot;, &quot;VEH&quot;)

for (each in 1:6){
  newdata &lt;- intervals %&gt;% 
    separate(&quot;trt&quot;, into=c(&quot;sDrug&quot;,NA)) %&gt;% 
    dplyr::filter(sDrug ==drug_list[each]) %&gt;% 
    mutate(SampleID=indv) %&gt;% 
    mutate(indv=substr(indv,4,4)) %&gt;% 
    mutate(indv=factor(indv, levels=color_order)) %&gt;% 
    dplyr::filter(SampleID %in% doubl_plot)
    
drug_plot &lt;- DRC_list[[each]]
     f &lt;-
       drug_plot %&gt;% 
         filter(SampleID %in% doubl_plot) %&gt;% 
         ggplot(., aes(x=Conc, y= Percent, group=SampleID,linetype=SampleID, color= indv,alpha =0.6 )) +
         guides(color=&quot;none&quot;, alpha = &quot;none&quot;)+
         stat_smooth(method = &quot;drm&quot;,
                     method.args = list(fct = L.4(c(NA,NA,1,NA))),
                     se = FALSE)+
         geom_ribbon(data = newdata, 
                     aes(x = Conc, y = Prediction, 
                         ymin = Lower, 
                         ymax = Upper, 
                         fill=indv),
                         alpha = 0.1, 
                         color = &quot;transparent&quot;)+
         # ylim(-.2,1.5)+
        coord_cartesian(ylim = c(-0.1, 1.5)) +
         scale_linetype_manual(values = c(&quot;dotted&quot;,&quot;solid&quot;,&quot;dotted&quot;,&quot;solid&quot;),
                               name=&quot;replicate&quot;,
                               labels=c(&quot;Rep 1&quot;, &quot;Rep 2&quot;,&quot;Rep 1&quot;, &quot;Rep 2&quot;))+
         scale_color_brewer(palette = &quot;Dark2&quot;)+
         scale_fill_brewer(palette = &quot;Dark2&quot;)+
         scale_x_log10() +  # Change the x-axis scale to log 10 scale
         theme_classic() +
         xlab(NULL)+
         ylab(NULL)+
        # scale_y_continuous(oob=scales::rescale_max, limits = -.4, 1.5)+
        ggtitle(drug_list[each])+
      theme(plot.title = element_text(hjust = 0.5, size =15, face =&quot;bold&quot;),
            axis.title=element_text(size=10),
            axis.ticks=element_line(linewidth = 2),
            axis.text=element_text(size=10, face = &quot;bold&quot;, color=&quot;black&quot;),
            panel.grid.major = element_line(colour = &#39;lightgrey&#39;),
            panel.border=element_rect(fill = NA, linewidth = 2),
            plot.background = element_rect(fill = &quot;white&quot;, colour = NA))
       
  print(f)
} </code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S2%20DRC%20reps-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S2-DRC-reps-1">
Past versions of "Fig S2 DRC reps-1.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S2-DRC-reps-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S2 DRC reps-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/021ae6df9b1a6106f052163e95f11f7d3af834aa/docs/figure/Supplementary_figures.Rmd/Fig S2 DRC reps-1.png" target="_blank">021ae6d</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-08-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S2%20DRC%20reps-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S2-DRC-reps-2">
Past versions of "Fig S2 DRC reps-2.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S2-DRC-reps-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S2 DRC reps-2.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/021ae6df9b1a6106f052163e95f11f7d3af834aa/docs/figure/Supplementary_figures.Rmd/Fig S2 DRC reps-2.png" target="_blank">021ae6d</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-08-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S2%20DRC%20reps-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S2-DRC-reps-3">
Past versions of "Fig S2 DRC reps-3.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S2-DRC-reps-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S2 DRC reps-3.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/021ae6df9b1a6106f052163e95f11f7d3af834aa/docs/figure/Supplementary_figures.Rmd/Fig S2 DRC reps-3.png" target="_blank">021ae6d</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-08-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S2%20DRC%20reps-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S2-DRC-reps-4">
Past versions of "Fig S2 DRC reps-4.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S2-DRC-reps-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S2 DRC reps-4.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/021ae6df9b1a6106f052163e95f11f7d3af834aa/docs/figure/Supplementary_figures.Rmd/Fig S2 DRC reps-4.png" target="_blank">021ae6d</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-08-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S2%20DRC%20reps-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S2-DRC-reps-5">
Past versions of "Fig S2 DRC reps-5.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S2-DRC-reps-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S2 DRC reps-5.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/021ae6df9b1a6106f052163e95f11f7d3af834aa/docs/figure/Supplementary_figures.Rmd/Fig S2 DRC reps-5.png" target="_blank">021ae6d</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-08-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S2%20DRC%20reps-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S2-DRC-reps-6">
Past versions of "Fig S2 DRC reps-6.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S2-DRC-reps-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S2 DRC reps-6.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/021ae6df9b1a6106f052163e95f11f7d3af834aa/docs/figure/Supplementary_figures.Rmd/Fig S2 DRC reps-6.png" target="_blank">021ae6d</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-08-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="s3-fig-cancer-drugs-that-decrease-cardiomyocyte-viability-induce-cellular-stress."
class="section level2">
<h2>S3 Fig: Cancer drugs that decrease cardiomyocyte viability induce
cellular stress.</h2>
<pre class="r"><code>viability &lt;- readRDS(&quot;data/viability.RDS&quot;)
norm_LDH48 &lt;- readRDS(&quot;data/supp_normLDH48.RDS&quot;)
viability %&gt;% 
  left_join(., norm_LDH48,by = c(&quot;indv&quot;,&quot;Drug&quot;,&quot;Conc&quot;)) %&gt;% 
  ggplot(., aes(x=per.live, y=ldh))+
  geom_point(aes(col=indv))+
  geom_smooth(method=&quot;lm&quot;)+
  facet_wrap(~Drug)+
  theme_bw()+
  xlab(&quot;Average viability of cardiomyocytes/100&quot;) +
  ylab(&quot;Average LDH&quot;) +
  ggtitle(&quot;Cancer drugs that decrease cardiomyocyte viability induce cellular stress&quot;)+
  scale_color_brewer(palette = &quot;Dark2&quot;,name = &quot;Individual&quot;, label = c(&quot;1&quot;,&quot;2&quot;,&quot;3&quot;,&quot;4&quot;,&quot;5&quot;,&quot;6&quot;))+
  ggpubr::stat_cor(method=&quot;pearson&quot;,
                   aes(label = paste(..r.label.., ..p.label.., sep = &quot;*`,`~&quot;)),
           color = &quot;red&quot;)+
  
  theme(plot.title = element_text(size = rel(1), hjust = 0.5,face = &quot;bold&quot;),
        axis.title = element_text(size = 12, color = &quot;black&quot;),
        axis.ticks = element_line(size = 1.5),
        axis.text = element_text(size = 8, color = &quot;black&quot;, angle = 0),
        strip.text.x = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;),
        strip.background = element_rect(fill = &quot;transparent&quot;)) </code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S3-1">
Past versions of "Fig S3-1.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/652df932402672208f615b21dc6dd91860a833aa/docs/figure/Supplementary_figures.Rmd/Fig S3-1.png" target="_blank">652df93</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S3-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/a7737443a7d5abc68a3213e3ae14b59c484463b1/docs/figure/Supplementary_figures.Rmd/Fig S3-1.png" target="_blank">a773744</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-08-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/021ae6df9b1a6106f052163e95f11f7d3af834aa/docs/figure/Supplementary_figures.Rmd/Fig S3-1.png" target="_blank">021ae6d</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-08-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/b31f5962c4b2eb28ce7f5c42fdff2edfe8c5621e/docs/figure/Supplementary_figures.Rmd/Fig S3-1.png" target="_blank">b31f596</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/680af3aae7258383999e2b9aa8cc74449bd91cfe/docs/figure/Supplementary_figures.Rmd/Fig S3-1.png" target="_blank">680af3a</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/537bc2eaae49e2ce7fd9561bcf6d0667227a10a8/docs/figure/Supplementary_figures.Rmd/Fig S3-1.png" target="_blank">537bc2e</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/dace8badd39879fefd85529bd1688e75a35d5adc/docs/figure/Supplementary_figures.Rmd/Fig S3-1.png" target="_blank">dace8ba</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="s4-fig-treatment-with-acs-at-a-dose-of-0.5-mum-for-48-hours-induces-effects-on-cardiomyocyte-viability."
class="section level2">
<h2>S4 Fig: Treatment with ACs at a dose of 0.5 <span
class="math inline">\(\mu\)</span>M for 48 hours induces effects on
cardiomyocyte viability.</h2>
<pre class="r"><code>viabilitytable &lt;- readRDS(&quot;data/averageviabilitytable.RDS&quot;)


viabilitytable %&gt;% 
  ungroup() %&gt;% 
  mutate(indv=substr(SampleID,4,4)) %&gt;% 
  mutate(indv=factor(indv, levels= c(&#39;1&#39;,&#39;2&#39;,&#39;3&#39;,&#39;4&#39;,&#39;5&#39;,&#39;6&#39;))) %&gt;% 
  dplyr::filter(Conc &lt;5) %&gt;%
  mutate(Conc= factor(as.numeric(Conc))) %&gt;%
  group_by(indv,sDrug,Conc) %&gt;% 
  dplyr::summarize(Viability=mean(Mean)) %&gt;%
  ggplot(.,  aes(x=sDrug, y= Viability*100 )) +
  geom_boxplot(position=&quot;dodge&quot;, 
                 aes(fill=sDrug))+
  geom_point(aes(color=indv))+
  guides(alpha = &quot;none&quot;)+
  ylim(0,150.5)+
  scale_color_brewer(palette = &quot;Dark2&quot;,
                       guide=&quot;legend&quot;,
                       name =&quot;individual&quot;, 
                       labels(c(1,2,3,4,5,6)))+
  scale_fill_manual(values=drug_pal_fact, name =&quot;treatment&quot;)+
  theme_classic() +
  xlab(&quot;&quot;)+
  ylab(&quot;% Viability&quot;) +
  facet_wrap(~Conc)+ 
  theme(axis.title=element_text(size=10),
        axis.ticks=element_line(size =2),
        axis.text.y=element_text(size=9, face = &quot;bold&quot;),
        axis.text.x=element_blank(),
        panel.grid.major = element_line(colour = &#39;darkgrey&#39;),
        panel.border=element_rect(fill = NA, size = 2),
        plot.title = element_text(hjust = 0.5, size =15, face = &quot;bold&quot;))</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S4-1">
Past versions of "Fig S4-1.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S4-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/021ae6df9b1a6106f052163e95f11f7d3af834aa/docs/figure/Supplementary_figures.Rmd/Fig S4-1.png" target="_blank">021ae6d</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-08-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/b31f5962c4b2eb28ce7f5c42fdff2edfe8c5621e/docs/figure/Supplementary_figures.Rmd/Fig S4-1.png" target="_blank">b31f596</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/680af3aae7258383999e2b9aa8cc74449bd91cfe/docs/figure/Supplementary_figures.Rmd/Fig S4-1.png" target="_blank">680af3a</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/537bc2eaae49e2ce7fd9561bcf6d0667227a10a8/docs/figure/Supplementary_figures.Rmd/Fig S4-1.png" target="_blank">537bc2e</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/dace8badd39879fefd85529bd1688e75a35d5adc/docs/figure/Supplementary_figures.Rmd/Fig S4-1.png" target="_blank">dace8ba</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="s5-fig-rna-seq-sample-quality-is-equivalent-across-individuals-treatments-and-time-points."
class="section level2">
<h2>S5 Fig: RNA-seq sample quality is equivalent across individuals,
treatments, and time points.</h2>
<div id="s5-figa-rin-vs-treatment-time-by-drug" class="section level3">
<h3>S5 FigA RIN vs treatment time by drug</h3>
<pre class="r"><code>library(limma)
library(edgeR)
library(cowplot)
filcpm_matrix &lt;- readRDS(&quot;data/filcpm_counts.RDS&quot;)


x &lt;- readRDS(&quot;data/filtermatrix_x.RDS&quot;)
x$samples %&gt;%
  mutate(
    drug = case_match(
      drug,
      &quot;Daunorubicin&quot; ~ &quot;DNR&quot;,
      &quot;Doxorubicin&quot; ~ &quot;DOX&quot;,
      &quot;Epirubicin&quot; ~ &quot;EPI&quot;,
      &quot;Mitoxantrone&quot; ~ &quot;MTX&quot;,
      &quot;Trastuzumab&quot; ~ &quot;TRZ&quot;,
      &quot;Vehicle&quot; ~ &quot;VEH&quot;,
      .default = drug
    )
  ) %&gt;%
  mutate(drug = factor(drug, levels = c(&#39;DOX&#39;, &#39;EPI&#39;, &#39;DNR&#39;, &#39;MTX&#39;, &#39;TRZ&#39;, &#39;VEH&#39;))) %&gt;%
  mutate(time = factor(time, labels = c(&quot;3 hours&quot;, &quot;24 hours&quot;))) %&gt;%
  ggplot(., aes(x = as.factor(time), y = RIN)) +
  geom_boxplot(aes(fill = as.factor(time))) +
  theme_bw() +
  ylim(c(0, 10)) +
  labs(x = &quot;&quot;, fill = &quot;Time in hours&quot;, y = &quot;RNA Integrity Number&quot;) +
  ggtitle(&quot;Boxplot of RIN by time and drug&quot;) +
  facet_wrap( ~ drug) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.text.y = element_text(
      size = 10,
      color = &quot;black&quot;,
      angle = 0,
      hjust = 0.8,
      vjust = 0.5
    ),
    axis.text.x = element_text(
      size = 10,
      color = &quot;black&quot;,
      angle = 0,
      hjust = 1,
      vjust = 0.2
    ),
    strip.text.x = element_text(
      size = 15,
      color = &quot;black&quot;,
      face = &quot;bold&quot;
    ),
    strip.background = element_rect(fill = &quot;white&quot;)
  )</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S5A-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S5A-1">
Past versions of "Fig S5A-1.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S5A-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S5A-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="s5-figb-total-number-of-reads-by-treatment"
class="section level3">
<h3>S5 FigB Total number of reads by treatment</h3>
<pre class="r"><code>seq_info &lt;- read.csv(&quot;output/sequencing_info.txt&quot;, row.names = 1)
seq_info %&gt;%
  
  filter(type == &quot;Total_reads&quot;) %&gt;%
  mutate(
    drug = case_match(
      drug,
      &quot;Daunorubicin&quot; ~ &quot;DNR&quot;,
      &quot;Doxorubicin&quot; ~ &quot;DOX&quot;,
      &quot;Epirubicin&quot; ~ &quot;EPI&quot;,
      &quot;Mitoxantrone&quot; ~ &quot;MTX&quot;,
      &quot;Trastuzumab&quot; ~ &quot;TRZ&quot;,
      &quot;Vehicle&quot; ~ &quot;VEH&quot;,
      .default = drug
    )
  ) %&gt;%
  mutate(drug = factor(drug, levels = c(&#39;DOX&#39;, &#39;EPI&#39;, &#39;DNR&#39;, &#39;MTX&#39;, &#39;TRZ&#39;, &#39;VEH&#39;))) %&gt;%
  ggplot(., aes (x = drug, y = count, fill = drug)) +
  geom_boxplot() +
  scale_fill_manual(values = drug_pal_fact) +
  ggtitle(expression(&quot;Total number of reads by treatment&quot;)) +
  xlab(&quot; &quot;) +
  ylab(expression(&quot;RNA -sequencing reads&quot;)) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 12, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(
      size = 10,
      color = &quot;black&quot;,
      angle = 0,
      hjust = 0.8,
      vjust = 0.5
    ),
    axis.text.x = element_text(size = 10, color = &quot;white&quot;),
    #strip.text.x = element_text(size = 15, color = &quot;black&quot;, face = &quot;bold&quot;),
    strip.text.y = element_text(color = &quot;white&quot;)
  )</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/S5B-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-S5B-1">
Past versions of S5B-1.png
</button>
</p>
<div id="fig-S5B-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/S5B-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="s5-figc-total-number-of-reads-by-sample"
class="section level3">
<h3>S5 FigC Total number of reads by sample</h3>
<pre class="r"><code>seq_info %&gt;%
  mutate(
    drug = case_match(
      drug,
      &quot;Daunorubicin&quot; ~ &quot;DNR&quot;,
      &quot;Doxorubicin&quot; ~ &quot;DOX&quot;,
      &quot;Epirubicin&quot; ~ &quot;EPI&quot;,
      &quot;Mitoxantrone&quot; ~ &quot;MTX&quot;,
      &quot;Trastuzumab&quot; ~ &quot;TRZ&quot;,
      &quot;Vehicle&quot; ~ &quot;VEH&quot;,
      .default = drug
    )
  ) %&gt;%
  mutate(drug = factor(drug, levels = c(&#39;DOX&#39;, &#39;EPI&#39;, &#39;DNR&#39;, &#39;MTX&#39;, &#39;TRZ&#39;, &#39;VEH&#39;))) %&gt;%
  # separate(samplenames, into=c(NA,NA,NA,&quot;samplenames&quot;)) %&gt;%
  # mutate(shortnames = paste(&quot;Sample&quot;,str_trim(samplenames))) %&gt;%
  filter(type == &quot;Total_reads&quot;) %&gt;%
  mutate(sampleID = colnames(filcpm_matrix)) %&gt;%
  ggplot(., aes (
    x = sampleID,
    y = count,
    fill = drug,
    group_by = indv
  )) +
  geom_col() +
  geom_hline(aes(yintercept = 20000000)) +
  scale_fill_manual(values = drug_pal_fact) +
  ggtitle(expression(&quot;Total number of reads by sample&quot;)) +
  xlab(&quot;&quot;) +
  ylab(expression(&quot;RNA -sequencing reads&quot;)) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(2), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(
      size = 10,
      color = &quot;black&quot;,
      angle = 0,
      hjust = 0.8,
      vjust = 0.5
    ),
    axis.text.x = element_text(
      size = 6,
      color = &quot;black&quot;,
      angle = 90,
      hjust = 1,
      vjust = 0.2
    ),
    #strip.text.x = element_text(size = 15, color = &quot;black&quot;, face = &quot;bold&quot;),
    strip.text.y = element_text(color = &quot;white&quot;)
  )</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/S5C-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-S5C-1">
Past versions of S5C-1.png
</button>
</p>
<div id="fig-S5C-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/S5C-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>To see more RNA-seq related analysis <a
href="https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/run_all_analysis.html">click
here</a></p>
</div>
</div>
<div
id="s6-fig-rna-seq-samples-cluster-by-treatment-type-timepoint-and-individual.-pearson-correlation-of-log2-cpm-values-across-all-pairs-of-samples."
class="section level2">
<h2>S6 Fig: RNA-seq samples cluster by treatment type, timepoint, and
individual. Pearson correlation of log2 cpm values across all pairs of
samples.</h2>
<pre class="r"><code>filcpm_matrix &lt;- readRDS(&quot;data/filcpm_counts.RDS&quot;)

mcor &lt;- cor(filcpm_matrix) 

filmat_groupmat_col &lt;- data.frame(timeset = colnames(filcpm_matrix))
counts_corr_mat &lt;- filmat_groupmat_col %&gt;%
  separate(timeset, into = c(&quot;drug&quot;, &quot;indv&quot;, &quot;time&quot;)) %&gt;%
  mutate(class = if_else(drug == &quot;DNR&quot;, &quot;AC&quot;, if_else(
    drug == &quot;DOX&quot;, &quot;AC&quot;, if_else(drug == &quot;EPI&quot;, &quot;AC&quot;, &quot;nAC&quot;)
  ))) %&gt;%
  mutate(TOP2i = if_else(drug == &quot;DNR&quot;, &quot;yes&quot;, if_else(
    drug == &quot;DOX&quot;, &quot;yes&quot;, if_else(drug == &quot;EPI&quot;, &quot;yes&quot;, if_else(drug == &quot;MTX&quot;, &quot;yes&quot;, &quot;no&quot;))
  ))) 
                         
 mat_colors &lt;- list( 
   drug= c(&quot;#F1B72B&quot;,&quot;#8B006D&quot;,&quot;#DF707E&quot;,&quot;#3386DD&quot;,&quot;#707031&quot;,&quot;#41B333&quot;),
   indv=c(&quot;#1B9E77&quot;, &quot;#D95F02&quot; ,&quot;#7570B3&quot;, &quot;#E7298A&quot; ,&quot;#66A61E&quot;, &quot;#E6AB02&quot;),
   time=c(&quot;pink&quot;, &quot;chocolate4&quot;),
   class=c(&quot;yellow1&quot;,&quot;darkorange1&quot;), 
   TOP2i =c(&quot;darkgreen&quot;,&quot;lightgreen&quot;))                        
                         
names(mat_colors$drug)   &lt;- unique(counts_corr_mat$drug)                      
names(mat_colors$indv) &lt;- unique(counts_corr_mat$indv)
names(mat_colors$time) &lt;- unique(counts_corr_mat$time)
names(mat_colors$class) &lt;- unique(counts_corr_mat$class)
names(mat_colors$TOP2i) &lt;- unique(counts_corr_mat$TOP2i)


ComplexHeatmap::pheatmap(mcor,
                         # column_title=(paste0(&quot;RNA-seq log&quot;[2]~&quot;cpm correlation&quot;)),
        annotation_col = counts_corr_mat,
        annotation_colors = mat_colors,
        heatmap_legend_param = mat_colors,
        fontsize=10,
        fontsize_row = 8,
        angle_col=&quot;90&quot;,
        treeheight_row=25,
        fontsize_col = 8,
        treeheight_col = 20)</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S6-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S6-1">
Past versions of "Fig S6-1.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S6-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/b31f5962c4b2eb28ce7f5c42fdff2edfe8c5621e/docs/figure/Supplementary_figures.Rmd/Fig S6-1.png" target="_blank">b31f596</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/680af3aae7258383999e2b9aa8cc74449bd91cfe/docs/figure/Supplementary_figures.Rmd/Fig S6-1.png" target="_blank">680af3a</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/537bc2eaae49e2ce7fd9561bcf6d0667227a10a8/docs/figure/Supplementary_figures.Rmd/Fig S6-1.png" target="_blank">537bc2e</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/dace8badd39879fefd85529bd1688e75a35d5adc/docs/figure/Supplementary_figures.Rmd/Fig S6-1.png" target="_blank">dace8ba</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>To see more RNA-seq related analysis <a
href="https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/run_all_analysis.html#correlation_heatmap_of_counts_matrix">click
here</a></p>
</div>
<div
id="s7-fig-pc1-associates-with-drug-treatment-and-treatment-time-while-pc2-associates-with-individual."
class="section level2">
<h2>S7 Fig: PC1 associates with drug treatment and treatment time, while
PC2 associates with individual.</h2>
<pre class="r"><code>pca_all_anno &lt;- readRDS(&quot;data/supp_pca_all_anno.RDS&quot;)
pca_all_anno &lt;- pca_all_anno %&gt;%
  mutate(
    drug = case_match(
      drug,
      &quot;Daunorubicin&quot; ~ &quot;DNR&quot;,
      &quot;Doxorubicin&quot; ~ &quot;DOX&quot;,
      &quot;Epirubicin&quot; ~ &quot;EPI&quot;,
      &quot;Mitoxantrone&quot; ~ &quot;MTX&quot;,
      &quot;Trastuzumab&quot; ~ &quot;TRX&quot;,
      &quot;Vehicle&quot; ~ &quot;VEH&quot;,
      .default = drug
    )
  )
facs &lt;- c(&quot;indv&quot;, &quot;drug&quot;, &quot;time&quot;)
names(facs) &lt;- c(&quot;Individual&quot;, &quot;Treatment&quot;, &quot;Time&quot;)

get_regr_pval &lt;- function(mod) {
  # Returns the p-value for the Fstatistic of a linear model
  # mod: class lm
  stopifnot(class(mod) == &quot;lm&quot;)
  fstat &lt;- summary(mod)$fstatistic
  pval &lt;- 1 - pf(fstat[1], fstat[2], fstat[3])
  return(pval)
}

plot_versus_pc &lt;- function(df, pc_num, fac) {
  # df: data.frame
  # pc_num: numeric, specific PC for plotting
  # fac: column name of df for plotting against PC
  pc_char &lt;- paste0(&quot;PC&quot;, pc_num)
  # Calculate F-statistic p-value for linear model
  pval &lt;- get_regr_pval(lm(df[, pc_char] ~ df[, fac]))
  if (is.numeric(df[, f])) {
    ggplot(df, aes_string(x = f, y = pc_char)) + geom_point() +
      geom_smooth(method = &quot;lm&quot;) + labs(title = sprintf(&quot;p-val: %.2f&quot;, pval))
  } else {
    ggplot(df, aes_string(x = f, y = pc_char)) + geom_boxplot() +
      labs(title = sprintf(&quot;p-val: %.2f&quot;, pval))
  }
}

for (f in facs) {
  # Plot f versus PC1 and PC2
  f_v_pc1 &lt;-
    arrangeGrob(plot_versus_pc(pca_all_anno, 1, f) + theme_bw())
  f_v_pc2 &lt;-
    arrangeGrob(plot_versus_pc(pca_all_anno, 2, f) + theme_bw())
  grid.arrange(f_v_pc1, f_v_pc2, ncol = 2, top = names(facs)[which(facs == f)])
}</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S7-1">
Past versions of "Fig S7-1.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/652df932402672208f615b21dc6dd91860a833aa/docs/figure/Supplementary_figures.Rmd/Fig S7-1.png" target="_blank">652df93</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S7-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/b31f5962c4b2eb28ce7f5c42fdff2edfe8c5621e/docs/figure/Supplementary_figures.Rmd/Fig S7-1.png" target="_blank">b31f596</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/680af3aae7258383999e2b9aa8cc74449bd91cfe/docs/figure/Supplementary_figures.Rmd/Fig S7-1.png" target="_blank">680af3a</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/537bc2eaae49e2ce7fd9561bcf6d0667227a10a8/docs/figure/Supplementary_figures.Rmd/Fig S7-1.png" target="_blank">537bc2e</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/dace8badd39879fefd85529bd1688e75a35d5adc/docs/figure/Supplementary_figures.Rmd/Fig S7-1.png" target="_blank">dace8ba</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S7-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S7-2">
Past versions of "Fig S7-2.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/652df932402672208f615b21dc6dd91860a833aa/docs/figure/Supplementary_figures.Rmd/Fig S7-2.png" target="_blank">652df93</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S7-2.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/b31f5962c4b2eb28ce7f5c42fdff2edfe8c5621e/docs/figure/Supplementary_figures.Rmd/Fig S7-2.png" target="_blank">b31f596</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/680af3aae7258383999e2b9aa8cc74449bd91cfe/docs/figure/Supplementary_figures.Rmd/Fig S7-2.png" target="_blank">680af3a</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/537bc2eaae49e2ce7fd9561bcf6d0667227a10a8/docs/figure/Supplementary_figures.Rmd/Fig S7-2.png" target="_blank">537bc2e</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/dace8badd39879fefd85529bd1688e75a35d5adc/docs/figure/Supplementary_figures.Rmd/Fig S7-2.png" target="_blank">dace8ba</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S7-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S7-3">
Past versions of "Fig S7-3.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S7-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/652df932402672208f615b21dc6dd91860a833aa/docs/figure/Supplementary_figures.Rmd/Fig S7-3.png" target="_blank">652df93</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S7-3.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/b31f5962c4b2eb28ce7f5c42fdff2edfe8c5621e/docs/figure/Supplementary_figures.Rmd/Fig S7-3.png" target="_blank">b31f596</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/680af3aae7258383999e2b9aa8cc74449bd91cfe/docs/figure/Supplementary_figures.Rmd/Fig S7-3.png" target="_blank">680af3a</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/537bc2eaae49e2ce7fd9561bcf6d0667227a10a8/docs/figure/Supplementary_figures.Rmd/Fig S7-3.png" target="_blank">537bc2e</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/dace8badd39879fefd85529bd1688e75a35d5adc/docs/figure/Supplementary_figures.Rmd/Fig S7-3.png" target="_blank">dace8ba</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>To see more RNA-seq related analysis <a
href="https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/run_all_analysis.html">click
here</a></p>
</div>
<div
id="s8-fig-thousands-of-gene-expression-changes-are-induced-in-response-to-top2i-treatment-over-24-hours."
class="section level2">
<h2>S8 Fig: Thousands of gene expression changes are induced in response
to TOP2i treatment over 24 hours.</h2>
<pre class="r"><code>Volcanoplots &lt;- readRDS(&quot;output/Volcanoplot_10.RDS&quot;)
Volcanoplots</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S8%20volcano-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S8-volcano-1">
Past versions of "Fig S8 volcano-1.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S8-volcano-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S8 volcano-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/b31f5962c4b2eb28ce7f5c42fdff2edfe8c5621e/docs/figure/Supplementary_figures.Rmd/Fig S8 volcano-1.png" target="_blank">b31f596</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/680af3aae7258383999e2b9aa8cc74449bd91cfe/docs/figure/Supplementary_figures.Rmd/Fig S8 volcano-1.png" target="_blank">680af3a</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/537bc2eaae49e2ce7fd9561bcf6d0667227a10a8/docs/figure/Supplementary_figures.Rmd/Fig S8 volcano-1.png" target="_blank">537bc2e</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/dace8badd39879fefd85529bd1688e75a35d5adc/docs/figure/Supplementary_figures.Rmd/Fig S8 volcano-1.png" target="_blank">dace8ba</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>To see more RNA-seq related analysis <a
href="https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/run_all_analysis.html#Volcano_plots_from_pairwise_gene_analysis">click
here</a></p>
</div>
<div
id="s9-fig-acs-affect-expression-of-nearly-half-of-all-expressed-genes-after-24-hours-of-treatment."
class="section level2">
<h2>S9 Fig: ACs affect expression of nearly half of all expressed genes
after 24 hours of treatment.</h2>
<div id="s9-figa-proportion-of-genes-de-in-response-to-drug-treatment"
class="section level3">
<h3>S9 FigA Proportion of genes DE in response to drug treatment</h3>
<pre class="r"><code>toplistall &lt;- readRDS(&quot;data/toplistall.RDS&quot;)
drug_pal_fact &lt;- c(&quot;#8B006D&quot;,&quot;#DF707E&quot;,&quot;#F1B72B&quot;, &quot;#3386DD&quot;,&quot;#707031&quot;,&quot;#41B333&quot;)
toplistall %&gt;%
    mutate(id=factor(id, levels = c(&#39;DOX&#39;, &#39;EPI&#39;, &#39;DNR&#39;, &#39;MTX&#39;, &#39;TRZ&#39;,&#39;VEH&#39;))) %&gt;% 
  mutate(time= factor(time,
     levels=c(&quot;3_hours&quot;,&quot;24_hours&quot;),
              labels=c(&quot;3 hours&quot;,&quot;24 hours&quot;))) %&gt;%
  group_by(time, id) %&gt;% 
  mutate(sigcount = if_else(adj.P.Val &lt; 0.05,&#39;sig&#39;,&#39;notsig&#39;))%&gt;%
  count(sigcount) %&gt;% 
  pivot_wider(id_cols = c(time,id), names_from=sigcount, values_from=n) %&gt;% 
  mutate(prop = sig/(sig+notsig)*100) %&gt;% 
  mutate(prop=if_else(is.na(prop),0,prop)) %&gt;% 
  ggplot(., aes(x=id, y= prop))+
  geom_col(aes(fill=id))+
  geom_text(aes(label = sprintf(&quot;%.2f&quot;,prop)),
            position=position_dodge(0.9),vjust=-.2 )+
  scale_fill_manual(values =drug_pal_fact)+
  guides(fill=guide_legend(title = &quot;Treatment&quot;))+
  facet_wrap(~time)+#labeller = (time = facettimelabel) )+
  theme_bw()+
  xlab(&quot;&quot;)+
  ylab(&quot;Percentage of expressed genes&quot;)+
  theme_bw()+
  ggtitle(&quot;Percent DEGs (adj. P value &lt;0.05)&quot;)+
  scale_y_continuous(expand=expansion(c(0.02,.2)))+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = &quot;black&quot;),
        # axis.ticks = element_line(linewidth = 1.5),
        # axis.line = element_line(linewidth = 1.5),
        strip.background = element_rect(fill = &quot;transparent&quot;),
        axis.text.x = element_text(size = 8, color = &quot;white&quot;, angle = 0),
        axis.text.y = element_text(size = 8, color = &quot;black&quot;, angle = 0),
        strip.text.x = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;))</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S9-1">
Past versions of "Fig S9-1.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S9-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/021ae6df9b1a6106f052163e95f11f7d3af834aa/docs/figure/Supplementary_figures.Rmd/Fig S9-1.png" target="_blank">021ae6d</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-08-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/b31f5962c4b2eb28ce7f5c42fdff2edfe8c5621e/docs/figure/Supplementary_figures.Rmd/Fig S9-1.png" target="_blank">b31f596</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/680af3aae7258383999e2b9aa8cc74449bd91cfe/docs/figure/Supplementary_figures.Rmd/Fig S9-1.png" target="_blank">680af3a</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/537bc2eaae49e2ce7fd9561bcf6d0667227a10a8/docs/figure/Supplementary_figures.Rmd/Fig S9-1.png" target="_blank">537bc2e</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/dace8badd39879fefd85529bd1688e75a35d5adc/docs/figure/Supplementary_figures.Rmd/Fig S9-1.png" target="_blank">dace8ba</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-06-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="s9-figb-magnitude-of-response-to-drug-treatment"
class="section level3">
<h3>S9 FigB Magnitude of response to drug treatment</h3>
<pre class="r"><code>toplistall %&gt;% 
  group_by(time, id) %&gt;% 
  mutate(id=factor(id, levels = c(&#39;DOX&#39;, &#39;EPI&#39;, &#39;DNR&#39;, &#39;MTX&#39;, &#39;TRZ&#39;,&#39;VEH&#39;))) %&gt;% 
  mutate(time= factor(time,
     levels=c(&quot;3_hours&quot;,&quot;24_hours&quot;),
              labels=c(&quot;3 hours&quot;,&quot;24 hours&quot;))) %&gt;%
  ggplot(., aes(x=id, y=logFC))+
  geom_boxplot(aes(fill=id))+
  ggpubr::fill_palette(palette =drug_pal_fact)+
  guides(fill=guide_legend(title = &quot;Treatment&quot;))+
  # facet_wrap(sigcount~time)+
  theme_bw()+
  xlab(&quot;&quot;)+
  ylab(expression(&quot;Log&quot;[2]*&quot; fold change&quot;))+
  theme_bw()+
  facet_wrap(~time)+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = &quot;black&quot;),
        # axis.ticks = element_line(linewidth = 1.5),
        # axis.line = element_line(linewidth = 1.5),
        strip.background = element_rect(fill = &quot;transparent&quot;),
        axis.text.x = element_blank(),
        strip.text.x = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;))</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/FIGS9B-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-FIGS9B-1">
Past versions of FIGS9B-1.png
</button>
</p>
<div id="fig-FIGS9B-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/FIGS9B-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="s10-fig-a-small-number-of-genes-respond-to-a-single-drug-only."
class="section level2">
<h2>S10 Fig: A small number of genes respond to a single drug only.</h2>
<div id="hours" class="section level3">
<h3>3 hours</h3>
<pre class="r"><code>supp10_3hlist &lt;- readRDS(&quot;data/supp10_3hlist.RDS&quot;)
list2env(supp10_3hlist, envir = .GlobalEnv)</code></pre>
<pre><code>&lt;environment: R_GlobalEnv&gt;</code></pre>
<pre class="r"><code>mito3pg &lt;- plot_grid(densityMTX3sp, MTX3Bplot, MTX3genesp_example, 
                    nrow = 1, 
                    rel_heights = c(.8,2,1), 
                    rel_widths=c(1.2,1,1.5),
                    scale=c(1,0.8,0.8))


Daun3pg &lt;- plot_grid(densityDNR3sp, DNR3Bplot, DNR3genesp_example, nrow = 1, rel_heights = c(.8,2,1), rel_widths=c(1.2,1,1.5),scale=c(1,0.8,0.8))
Doxo3pg &lt;- plot_grid(NA, NA, NA, nrow = 1, rel_heights = c(.8,2,1), rel_widths=c(1.2,1,1.5),scale=c(1,0.8,0.8))
Epi3pg &lt;- plot_grid(densityEPI3sp, NA, NA, nrow = 1, rel_heights = c(.8,2,1), rel_widths=c(1.2,1,1.5),scale=c(1,0.8,0.8))

allfinal3hour &lt;- plot_grid(Doxo3pg,Epi3pg,Daun3pg,mito3pg,nrow=4, rel_heights = c(1,1,1,1))
# allfinal3hour &lt;- readRDS(&quot;data/allfinal3hour.RDS&quot;)

allfinal3hour</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S10%203hr%20Drug%20specific%20genes-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S10-3hr-Drug-specific-genes-1">
Past versions of "Fig S10 3hr Drug specific genes-1.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S10-3hr-Drug-specific-genes-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S10 3hr Drug specific genes-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="hours-1" class="section level3">
<h3>24 hours</h3>
<pre class="r"><code>supp10_24hlist &lt;- readRDS(&quot;data/supp10_24hlist.RDS&quot;)
list2env(supp10_24hlist, envir = .GlobalEnv)</code></pre>
<pre><code>&lt;environment: R_GlobalEnv&gt;</code></pre>
<pre class="r"><code>mitopg &lt;- plot_grid(densityMTXsp, MtxBplot, Mtxgenesp_example, 
                    nrow = 1, 
                    rel_heights = c(.8,2,1), 
                    rel_widths=c(1.2,1,1.5),
                    scale=c(1,0.8,0.8))


Daunpg &lt;- plot_grid(densityDNRsp, DnrBplot, Dnrgenesp_example, nrow = 1, rel_heights = c(.8,2,1), rel_widths=c(1.2,1,1.5),scale=c(1,0.8,0.8))
Doxopg &lt;- plot_grid(densityDOXsp, DoxBplot, Doxgenesp_example, nrow = 1, rel_heights = c(.8,2,1), rel_widths=c(1.2,1,1.5),scale=c(1,0.8,0.8))
Epipg &lt;- plot_grid(densityEPIsp, EpiBplot, Epigenesp_example, nrow = 1, rel_heights = c(.8,2,1), rel_widths=c(1.2,1,1.5),scale=c(1,0.8,0.8))
allfinal &lt;- plot_grid(Doxopg,Epipg,Daunpg,mitopg,nrow=4, labels = &quot;AUTO&quot;)


# allfinal &lt;- readRDS(&quot;output/allfinal_sup10.RDS&quot;) 
plot(allfinal)</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S1024hr%20Drug%20specific%20genes-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S1024hr-Drug-specific-genes-1">
Past versions of "Fig S1024hr Drug specific genes-1.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S1024hr-Drug-specific-genes-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S1024hr Drug specific genes-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div
id="s11-fig-stringently-identified-drug-specific-response-genes-are-enriched-in-biological-processes."
class="section level2">
<h2>S11 Fig: Stringently-identified drug-specific response genes are
enriched in biological processes.</h2>
<div id="hours-2" class="section level3">
<h3>3 hours</h3>
<pre class="r"><code>gostres3Dnrdeg_sp &lt;- readRDS(&quot;data/DEG-GO/gostres3Dnrdeg_sp.RDS&quot;)

Dnr3_sp_DEGtable &lt;- gostres3Dnrdeg_sp$result %&gt;%
  dplyr::select(c(source, term_id,term_name,intersection_size, 
                   term_size, p_value))


Dnr3_sp_DEGtable  %&gt;% 
    dplyr::filter(source==&quot;GO:BP&quot;) %&gt;% 
    dplyr::select(p_value,term_name,intersection_size) %&gt;%
    slice_min(., n=10 ,order_by=p_value) %&gt;%
    mutate(log_val = -log10(p_value)) %&gt;%
   # slice_max(., n=10,order_by = p_value) %&gt;%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels =scales::label_wrap(30))+
    guides(col=&quot;none&quot;, size=guide_legend(title = &quot;# of intersected \n terms&quot;))+
    ggtitle(&#39;DNR 3 hour specific(stringent)\n gene set GO:BP terms&#39;) +
    xlab(expression(&quot; -&quot;~log[10]~(&quot;adj. p-value&quot;)))+
    ylab(&quot;GO: BP term&quot;)+
    theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = &quot;black&quot;),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = &quot;black&quot;, angle = 0),
        strip.text.x = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;))</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S11%20stringent-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S11-stringent-1">
Past versions of "Fig S11 stringent-1.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S11-stringent-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S11 stringent-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>gostres3Mtxdeg_sp &lt;- readRDS(&quot;data/DEG-GO/gostres3Mtxdeg_sp.RDS&quot;)
Mtx3_sp_DEGtable &lt;- gostres3Mtxdeg_sp$result %&gt;%
  dplyr::select(c(source, term_id,term_name,intersection_size, 
                   term_size, p_value))
Mtx3_sp_DEGtable  %&gt;% 
    dplyr::filter(source==&quot;GO:BP&quot;) %&gt;% 
    dplyr::select(p_value,term_name,intersection_size) %&gt;%
    slice_min(., n=5,order_by=p_value) %&gt;%
    mutate(log_val = -log10(p_value)) %&gt;%
   # slice_max(., n=10,order_by = p_value) %&gt;%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = scales::label_wrap(30))+
  geom_vline(xintercept = (-log10(0.05)))+
    guides(col=&quot;none&quot;, size=guide_legend(title = &quot;# of intersected \n terms&quot;))+
    ggtitle(&#39;MTX 3 hour specific(stringent)\n gene set GO:BP terms&#39;) +
    xlab(expression(&quot; -&quot;~log[10]~(&quot;adj. p-value&quot;)))+
    ylab(&quot;GO: BP term&quot;)+
    theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = &quot;black&quot;),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = &quot;black&quot;, angle = 0),
        strip.text.x = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;))</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S11%20stringent-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S11-stringent-2">
Past versions of "Fig S11 stringent-2.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S11-stringent-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S11 stringent-2.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="hours-3" class="section level3">
<h3>24 hours</h3>
<pre class="r"><code>DX_sp_DEGgostres &lt;- readRDS(&quot;data/DEG-GO/gostresDOXdeg_sp.RDS&quot;)
MT_sp_DEGgostres &lt;- readRDS(&quot;data/DEG-GO/gostresMTXdeg_sp.RDS&quot;)

DX_sp_DEGtable &lt;- DX_sp_DEGgostres$result %&gt;%
  dplyr::select(c(source, term_id,term_name,intersection_size, 
                   term_size, p_value))
MT_sp_DEGtable &lt;- MT_sp_DEGgostres$result %&gt;%
  dplyr::select(c(source, term_id,term_name,intersection_size, 
                   term_size, p_value))
  
  
DX_sp_DEGtable  %&gt;% 
    dplyr::filter(source==&quot;GO:BP&quot;) %&gt;% 
    dplyr::select(p_value,term_name,intersection_size) %&gt;%
    slice_min(., n=10 ,order_by=p_value) %&gt;%
    mutate(log_val = -log10(p_value)) %&gt;%
   # slice_max(., n=10,order_by = p_value) %&gt;%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = scales::wrap_format(30))+
    guides(col=&quot;none&quot;, size=guide_legend(title = &quot;# of intersected \n terms&quot;))+
    ggtitle(&#39;DOX specific 24 hour gene set GO:BP terms&#39;) +
    xlab(expression(&quot; -&quot;~log[10]~(&quot;adj. p-value&quot;)))+
    ylab(&quot;GO: BP term&quot;)+
    theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = &quot;black&quot;),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = &quot;black&quot;, angle = 0),
        strip.text.x = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;))</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/FigS11%2024h-%20drug%20spec%20path-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-FigS11-24h--drug-spec-path-1">
Past versions of "FigS11 24h- drug spec path-1.png"
</button>
</p>
<div id="fig-no-spaces-FigS11-24h--drug-spec-path-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/FigS11 24h- drug spec path-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>MT_sp_DEGtable  %&gt;% 
    dplyr::filter(source==&quot;GO:BP&quot;) %&gt;% 
    dplyr::select(p_value,term_name,intersection_size) %&gt;%
    slice_min(., n=10 ,order_by=p_value) %&gt;%
    mutate(log_val = -log10(p_value)) %&gt;%
   # slice_max(., n=10,order_by = p_value) %&gt;%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = scales::wrap_format(30))+
    guides(col=&quot;none&quot;, size=guide_legend(title = &quot;# of intersected \n terms&quot;))+
    ggtitle(&#39;MTX specific 24 hour gene set GO:BP terms&#39;) +
    xlab(expression(&quot; -&quot;~log[10]~(&quot;adj. p-value&quot;)))+
    ylab(&quot;GO: BP term&quot;)+
    theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = &quot;black&quot;),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = &quot;black&quot;, angle = 0),
        strip.text.x = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;))</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/FigS11%2024h-%20drug%20spec%20path-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-FigS11-24h--drug-spec-path-2">
Past versions of "FigS11 24h- drug spec path-2.png"
</button>
</p>
<div id="fig-no-spaces-FigS11-24h--drug-spec-path-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/FigS11 24h- drug spec path-2.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div
id="s12-fig-nominally-significant-trz-response-genes-show-enrichment-in-cancer-related-pathways."
class="section level2">
<h2>S12 Fig: Nominally significant TRZ response genes show enrichment in
cancer-related pathways.</h2>
<pre class="r"><code>FC_mine &lt;- readRDS(&quot;data/toplistall.RDS&quot;)

TRZ_list &lt;- FC_mine %&gt;% 
  dplyr::filter(id==&quot;TRZ&quot;,P.Value &lt; 0.05) 

TRZ_list %&gt;% 
  group_by(time) %&gt;% 
 count() %&gt;% 
  mutate(time = factor(time, levels = c(&quot;3_hours&quot;, &quot;24_hours&quot;), labels = c(&quot;3 hours&quot;, &quot;24 hours&quot;)))%&gt;% 
  
  ggplot(., aes(x=time, y=n))+
  geom_col()+
   theme_bw() +
  ggtitle(&quot;Number of nominal pvalue TRZ genes by time&quot;
  )+
  ylab(expression(&quot;count&quot;)) +
  theme(
    strip.background = element_rect(
      fill = &quot;white&quot;,
      linetype = 1,
      linewidth = 0.5
    ),
    plot.title = element_text(size = 12, hjust = 0.5, face = &quot;bold&quot;),
    axis.title = element_text(size = 10, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.0),
    panel.background = element_rect(colour = &quot;black&quot;, size = 1),
    # axis.text.x = element_blank(),
    strip.text.x = element_text(margin = margin(2, 0, 2, 0, &quot;mm&quot;), face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/TRZ%20expression-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-TRZ-expression-1">
Past versions of "TRZ expression-1.png"
</button>
</p>
<div id="fig-no-spaces-TRZ-expression-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/652df932402672208f615b21dc6dd91860a833aa/docs/figure/Supplementary_figures.Rmd/TRZ expression-1.png" target="_blank">652df93</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>all_cpmcount &lt;-   readRDS(&quot;data/cpmcount.RDS&quot;) %&gt;% rownames_to_column(var = &quot;ENTREZID&quot;)
##pulled all my logcpm, now choosing only TRZ and VEH for plotting

trz_veh_count &lt;- all_cpmcount %&gt;% 
  dplyr::select(ENTREZID, starts_with(&quot;Tr&quot;), starts_with(&quot;Ve&quot;))



sampset &lt;- readRDS(&quot;data/sampsettrz.RDS&quot;)
sub_samp &lt;- sampset %&gt;% 
  dplyr::filter(SYMBOL==&quot;PHLDA1&quot;|SYMBOL== &quot;ANKRD2&quot;)
#  &lt;- TRZ_list %&gt;% 
#   distinct(ENTREZID,.keep_all = TRUE) %&gt;% 
#   sample_n(.,6)
for (g in seq(from=1, to=length(sub_samp$ENTREZID))){
  a &lt;- sub_samp$SYMBOL[g]
  cpm_TRZ(trz_veh_count,GOI=sub_samp[g,3],
          ylab=bquote(~italic(.(a))~log[2]~&quot;cpm &quot;), 
          titlename=bquote(~italic(.(a))~&quot;gene &quot;))

}</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/cpmTRZ-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cpmTRZ-1">
Past versions of cpmTRZ-1.png
</button>
</p>
<div id="fig-cpmTRZ-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/652df932402672208f615b21dc6dd91860a833aa/docs/figure/Supplementary_figures.Rmd/cpmTRZ-1.png" target="_blank">652df93</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/Supplementary_figures.Rmd/cpmTRZ-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cpmTRZ-2">
Past versions of cpmTRZ-2.png
</button>
</p>
<div id="fig-cpmTRZ-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/652df932402672208f615b21dc6dd91860a833aa/docs/figure/Supplementary_figures.Rmd/cpmTRZ-2.png" target="_blank">652df93</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>list3hr &lt;- readRDS(&quot;data/DEG-GO/list3hr.RDS&quot;)
list24hr &lt;- readRDS(&quot;data/DEG-GO/list24hr.RDS&quot;)
KEGGtrz3 &lt;- c(&quot;KEGG:05202&quot;, &quot;KEGG:04115&quot;, &quot;KEGG:05223&quot;, &quot;KEGG:05212&quot;,&quot;KEGG:05222&quot;)
KEGGtrz24 &lt;- c(&quot;KEGG:05202&quot;, &quot;KEGG:05215&quot;, &quot;KEGG:00510&quot;)
# testframe &lt;- lapply(list3hr, `[`,&#39;result&#39;)

hr3_results &lt;- list3hr |&gt; 
  map(~ .x$result |&gt; dplyr::select(p_value,term_id,term_name,intersection_size))
hr24_results &lt;- list24hr |&gt; 
  map(~ .x$result |&gt; dplyr::select(p_value,term_id,term_name,intersection_size))

table_3hr &lt;- list_rbind(hr3_results,names_to = &quot;set&quot;)%&gt;%
  mutate(log_val = -log10(p_value)) %&gt;% 
  dplyr::filter(term_id %in% KEGGtrz3) %&gt;% 
  mutate(treatment = case_when(set ==&quot;DAgostres3&quot;~&quot;DNR&quot;,set==&quot;DXgostres&quot; ~ &quot;DOX&quot;,set==&quot;EPgostres&quot;~&quot;EPI&quot;,set==&quot;MTgostres&quot;~&quot;MTX&quot;,set==&quot;TRZresgenes3h&quot;~&quot;TRZ&quot;,set==&quot;DDEgostres23only&quot;~&quot;AC&quot;,set==&quot;DDEMgostres3&quot;~&quot;TOP2i&quot;))


table_24hr &lt;- list_rbind(hr24_results,names_to = &quot;set&quot;) %&gt;%
  mutate(log_val = -log10(p_value))%&gt;% 
  dplyr::filter(term_id %in% KEGGtrz24) %&gt;% 
  mutate(treatment = case_when(set ==&quot;DAgostres24&quot;~&quot;DNR&quot;,set==&quot;DXgostres24&quot; ~ &quot;DOX&quot;,set==&quot;EPgostres24&quot;~&quot;EPI&quot;,set==&quot;MTgostres24&quot;~&quot;MTX&quot;,set==&quot;TRZresgenes243h&quot;~&quot;TRZ&quot;,set==&quot;DDEgostres24only&quot;~&quot;AC&quot;,set==&quot;DDEMgostres24&quot;~&quot;TOP2i&quot;))


hr3_results[[&quot;TRZresgenes3h&quot;]]  %&gt;% 
    mutate(log_val = -log10(p_value)) %&gt;% 
  dplyr::filter(str_detect(term_id,&quot;^KEGG:&quot;)) %&gt;% 
  slice_min(., n=5,order_by=p_value) %&gt;%
    ggplot(., aes(x = log_val, y =reorder(term_name,p_value),col=intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = scales::label_wrap(30))+
  geom_vline(xintercept = (-log10(0.05)), col=&quot;red&quot;)+
    guides(col=&quot;none&quot;, size=guide_legend(title = &quot;# of intersected \n terms&quot;))+
    ggtitle(&#39;TRZ 3 hour nominal DE\n KEGG pathway enrichment&#39;) +
    xlab(expression(&quot; -&quot;~log[10]~(&quot;nominal p-value&quot;)))+
    ylab(&quot;KEGG&quot;)+
    theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = &quot;black&quot;),
        axis.ticks = element_line(linewidth = 1.5),
        # axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = &quot;black&quot;, angle = 0),
        strip.text.x = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;))</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/TRZpathways-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-TRZpathways-1">
Past versions of TRZpathways-1.png
</button>
</p>
<div id="fig-TRZpathways-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/a08048ce6f157e2de0013a13380df6fed9af6176/docs/figure/Supplementary_figures.Rmd/TRZpathways-1.png" target="_blank">a08048c</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>hr24_results[[&quot;TRZresgenes243h&quot;]]  %&gt;% 
    mutate(log_val = -log10(p_value)) %&gt;% 
  dplyr::filter(str_detect(term_id,&quot;^KEGG:&quot;)) %&gt;% 
  slice_min(., n=5,order_by=p_value) %&gt;%
    ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col=intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = scales::label_wrap(30))+
   geom_vline(xintercept = (-log10(0.05)), col=&quot;red&quot;)+
    guides(col=&quot;none&quot;, size=guide_legend(title = &quot;# of intersected \n terms&quot;))+
    ggtitle(&#39;TRZ 24 hour nominal DE\n KEGG pathway enrichment&#39;) +
    xlab(expression(&quot; -&quot;~log[10]~(&quot;nominal p-value&quot;)))+
    ylab(&quot;KEGG&quot;)+
    theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = &quot;black&quot;),
        axis.ticks = element_line(linewidth = 1.5),
        # axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = &quot;black&quot;, angle = 0),
        strip.text.x = element_text(size = 12, color = &quot;black&quot;, face = &quot;bold&quot;))</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/TRZpathways-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-TRZpathways-2">
Past versions of TRZpathways-2.png
</button>
</p>
<div id="fig-TRZpathways-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/a08048ce6f157e2de0013a13380df6fed9af6176/docs/figure/Supplementary_figures.Rmd/TRZpathways-2.png" target="_blank">a08048c</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>More on TRZ expression <a
href="https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/after_comments.html">click
here</a><br />
OR GO <a
href="https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/GO_after_comments.html">analysis
here</a></p>
</div>
<div
id="s13-fig-four-gene-expression-signatures-capture-the-response-to-top2i-over-time."
class="section level2">
<h2>S13 Fig: Four gene expression signatures capture the response to
TOP2i over time.</h2>
<div id="s13-figa" class="section level3">
<h3>S13 FigA</h3>
<pre class="r"><code>cormotif_initial &lt;- readRDS(&quot;data/cormotif_initialall.RDS&quot;)
Cormotif::plotIC(cormotif_initial)</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S13A%20Cormotif-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S13A-Cormotif-1">
Past versions of "Fig S13A Cormotif-1.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S13A-Cormotif-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S13A Cormotif-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="s13-figb" class="section level3">
<h3>S13 FigB</h3>
<pre class="r"><code>motif_NRrep &lt;-  readRDS(&quot;output/motif_NRrep.RDS&quot;)
motif_ERrep &lt;-  readRDS(&quot;output/motif_ERrep.RDS&quot;)
motif_TIrep &lt;-  readRDS(&quot;output/motif_TI_rep.RDS&quot;)
motif_LRrep &lt;-  readRDS(&quot;output/motif_LRrep.RDS&quot;)
motif_NRrep &lt;- motif_NRrep +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1))+
  theme(legend.position = &quot;NULL&quot;, axis.title.y =element_text(size =12))

motif_ERrep &lt;- motif_ERrep+
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1))+
  theme(legend.position = &quot;NULL&quot;, axis.title.y =element_text(size =12))

motif_TIrep &lt;- motif_TIrep+
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1))+ 
  theme(legend.position = &quot;NULL&quot;, axis.title.y =element_text(size =12))

motif_LRrep &lt;- motif_LRrep+
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1))+
  theme(legend.position = &quot;NULL&quot;, axis.title.y =element_text(size =12))
plot_grid(motif_ERrep,motif_TIrep,motif_LRrep,motif_NRrep,nrow = 4,ncol = 1)</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S13B%20Cormotif%20genes-1.png" width="80%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S13B-Cormotif-genes-1">
Past versions of "Fig S13B Cormotif genes-1.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S13B-Cormotif-genes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S13B Cormotif genes-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="s13-figc" class="section level3">
<h3>S13 FigC</h3>
<pre class="r"><code>motif_list_GO &lt;- readRDS(&quot;output/supplementary_motif_list_GO.RDS&quot;)
list2env(motif_list_GO, envir = .GlobalEnv)</code></pre>
<pre><code>&lt;environment: R_GlobalEnv&gt;</code></pre>
<pre class="r"><code>motifcol &lt;-  c(&quot;#F8766D&quot;, &quot;#00BFC4&quot;, &quot;#7CAE00&quot;,  &quot;#C77CFF&quot;)
GOmotiflong &lt;- list(
  &quot;EAR&quot; = ER3f,
  &quot;ESR&quot; = TI3f,
  &quot;LR&quot; = LR3f,
  &quot;NR&quot; = NR1f
)
GOBPlong &lt;- data.table::rbindlist(GOmotiflong, idcol = &quot;motif&quot;)
p &lt;-
  GOBPlong %&gt;% mutate(motif = factor(motif, levels = c(&quot;EAR&quot;, &quot;ESR&quot;, &quot;LR&quot;, &quot;NR&quot;))) %&gt;%
  
  ggplot(., aes(
    x = log_val,
    y = reorder(term_name, order_val, desc = FALSE),
    col = motif
  )) +
  geom_point(aes(size = intersection_size, col = motif)) +
  scale_y_discrete(labels = scales::label_wrap(40)) +
  facet_grid(motif ~ ., scales = &quot;free_y&quot;) +
  scale_color_discrete(type = motifcol) +
  guides(col = guide_legend(title = &quot;Motif&quot;),
         size = guide_legend(title = &quot;# of intersected \n terms&quot;)) +
  ggtitle(&#39;Top 10 enriched GO:BP terms for each motif&#39;) +
  xlab(expression(&quot;-log&quot;[10] ~ &quot;(p-value)&quot;)) +
  ylab(&quot;GO:BP term&quot;) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text = element_text(
      size = 8,
      color = &quot;black&quot;,
      angle = 0
    ),
    strip.text.x = element_text(
      size = 15,
      color = &quot;black&quot;,
      face = &quot;bold&quot;
    )
  )

g &lt;- ggplot_gtable(ggplot_build(p))
stripr &lt;- which(grepl(&#39;strip-r&#39;, g$layout$name))
fills &lt;- c(&quot;#F8766D&quot;, &quot;#00BFC4&quot;, &quot;#7CAE00&quot;,  &quot;#C77CFF&quot;)
k &lt;- 1
for (i in stripr) {
  j &lt;- which(grepl(&#39;rect&#39;, g$grobs[[i]]$grobs[[1]]$childrenOrder))
  g$grobs[[i]]$grobs[[1]]$children[[j]]$gp$fill &lt;- fills[k]
  k &lt;- k + 1
}

grid.draw(g)</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S13C%20GO%20Cormotif-1.png" width="768" height="80%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S13C-GO-Cormotif-1">
Past versions of "Fig S13C GO Cormotif-1.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S13C-GO-Cormotif-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S13C GO Cormotif-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>More on Baysian gene analysis <a
href="https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/Cormotifcluster_analysis.html">click
here</a>.</p>
</div>
</div>
<div
id="s14-fig-gene-expression-variance-across-45-individuals-increases-in-ipsc-cms-treated-with-dox."
class="section level2">
<h2>S14 Fig: Gene expression variance across 45 individuals increases in
iPSC-CMs treated with DOX.</h2>
<pre class="r"><code>store_var &lt;- readRDS(&quot;data/Knowles_variation_data.RDS&quot;)

knowlesdrug&lt;- store_var %&gt;% 
  dplyr::select(&quot;ESGN&quot;,&quot;mean_DOX&quot;,&quot;var_DOX&quot;,&quot;mean_NT&quot;, &quot;var_NT&quot;) %&gt;% 
  pivot_longer(cols = !&quot;ESGN&quot;, names_to = &quot;short&quot;, values_to = &quot;values&quot;) %&gt;% 
  separate(short, into=c(&quot;calc&quot;,&quot;treatment&quot;)) #%&gt;% 
knowlesdrug %&gt;% 
  as.data.frame() %&gt;% 
  dplyr::filter(calc == &quot;mean&quot;) %&gt;% 
  ggplot(., aes(x= treatment, y=values))+
  geom_boxplot()+
  ggtitle(&quot;Knowles Means across all genes&quot;)+
  geom_signif(
    comparisons = list(
      c(&quot;DOX&quot;, &quot;NT&quot;)),
    test = &quot;t.test&quot;,
    tip_length = 0.01,
    map_signif_level = FALSE
    # textsize = 4,
    # # y_position = 11,
    # step_increase = 0.05
  )+
  theme_bw() +
  # xlab(&quot; &quot;) +
  ylab(expression(&quot;counts log&quot;[2] ~ &quot;cpm &quot;)) +
  theme(
    strip.background = element_rect(
      fill = &quot;white&quot;,
      linetype = 1,
      linewidth = 0.5
    ),
    plot.title = element_text(size = 12, hjust = 0.5, face = &quot;bold&quot;),
    axis.title = element_text(size = 10, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.0),
    panel.background = element_rect(colour = &quot;black&quot;, size = 1),
    # axis.text.x = element_blank(),
    strip.text.x = element_text(margin = margin(2, 0, 2, 0, &quot;mm&quot;), face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Knowlesvariance-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Knowlesvariance-1">
Past versions of Knowlesvariance-1.png
</button>
</p>
<div id="fig-Knowlesvariance-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/652df932402672208f615b21dc6dd91860a833aa/docs/figure/Supplementary_figures.Rmd/Knowlesvariance-1.png" target="_blank">652df93</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>knowlesdrug %&gt;% 
  as.data.frame() %&gt;% 
  dplyr::filter(calc == &quot;var&quot;) %&gt;% 
  ggplot(., aes(x= treatment, y=values))+
  geom_boxplot(outlier.shape= NA)+
  ggtitle(&quot; Knowles Variance across all genes&quot;)+
  geom_signif(
    comparisons = list(
      c(&quot;DOX&quot;, &quot;NT&quot;)),
    test = &quot;var.test&quot;,
    tip_length = 0.01,
    y_position = 0.5,
    # vjust=1,
    map_signif_level = FALSE)+
   coord_cartesian(ylim = c(NA, 1.5), expand = TRUE)+
theme_bw() +
  # xlab(&quot; &quot;) +
  ylab(expression(&quot;counts log&quot;[2] ~ &quot;cpm &quot;)) +
  theme(
    strip.background = element_rect(
      fill = &quot;white&quot;,
      linetype = 1,
      linewidth = 0.5
    ),
    plot.title = element_text(size = 12, hjust = 0.5, face = &quot;bold&quot;),
    axis.title = element_text(size = 10, color = &quot;black&quot;),
    axis.ticks = element_line(linewidth = 1.0),
    panel.background = element_rect(colour = &quot;black&quot;, size = 1),
    # axis.text.x = element_blank(),
    strip.text.x = element_text(margin = margin(2, 0, 2, 0, &quot;mm&quot;), face = &quot;bold&quot;)
  )</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Knowlesvariance-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Knowlesvariance-2">
Past versions of Knowlesvariance-2.png
</button>
</p>
<div id="fig-Knowlesvariance-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/652df932402672208f615b21dc6dd91860a833aa/docs/figure/Supplementary_figures.Rmd/Knowlesvariance-2.png" target="_blank">652df93</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>More on Knowles gene analysis <a
href="https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/Knowels_trop_analysis.html#Replication_of_variance_figures_using_Knowels_data">click
here</a>.</p>
</div>
<div
id="s15-fig-replication-of-gene-expression-response-in-ac-induced-cardiotoxicity-loci."
class="section level2">
<h2>S15 Fig: Replication of gene expression response in AC-induced
cardiotoxicity loci.</h2>
<pre class="r"><code>DOX_de_genes_interest &lt;- read.csv(&quot;output/DOX_de_goi.csv&quot;, row.names = 1)


all_cpmcounts &lt;-  read_table(&quot;data/Counts_RNA_ERMatthews.txt&quot;)
##all my count data
Knowles_log2cpm &lt;- readRDS(&quot;data/Knowles_log2cpm_real.RDS&quot;) 
##all Knowles count data, down with my method in analysis/after_comments.Rmd
store_box &lt;- Knowles_log2cpm%&gt;% 
  dplyr::select( &#39;ESGN&#39;,ends_with(c(&#39;0.625&#39;, &#39;0&#39;))) %&gt;%
  dplyr::filter(ESGN %in% DOX_de_genes_interest$ensembl_gene_id) %&gt;% 
  pivot_longer(cols=!ESGN, names_to = &quot;ind&quot;, values_to = &quot;counts&quot;) %&gt;% 
  separate(ind,into=c(&quot;cell_line&quot;,&quot;dosage&quot;), sep = &quot;:&quot;)%&gt;%
  left_join(., DOX_de_genes_interest, by = c(&quot;ESGN&quot; = &quot;ensembl_gene_id&quot;)) %&gt;% 
  mutate(expr=&quot;K&quot;)


lcpm_24h &lt;- all_cpmcounts %&gt;% 
  column_to_rownames(&quot;ENTREZID&quot;) %&gt;% 
  cpm(., log=TRUE) %&gt;% 
  as.data.frame() %&gt;% 
  rownames_to_column(var = &quot;ENTREZID&quot;) %&gt;% 
  # dplyr::select(ENTREZID, all_of(starts_with(c(&quot;DOX&quot;,&quot;VEH&quot;)))) %&gt;%
  dplyr::select(ENTREZID, all_of(ends_with(&quot;24h&quot;)))  %&gt;% 
  dplyr::filter(ENTREZID %in% DOX_de_genes_interest$ENTREZID) %&gt;% 
  pivot_longer(cols=!ENTREZID, names_to = &quot;ind&quot;, values_to = &quot;counts&quot;) %&gt;%   mutate(ENTREZID = as.numeric(ENTREZID)) %&gt;% 
  full_join(., DOX_de_genes_interest, by = c(&quot;ENTREZID&quot;=&quot;ENTREZID&quot;)) %&gt;% 
  mutate(expr=&quot;ME&quot;) %&gt;% 
  rename(&quot;ESGN&quot;=&quot;ensembl_gene_id&quot;) %&gt;% 
  separate(ind, into = c(&quot;dosage&quot;,&quot;cell_line&quot;,NA)) %&gt;% 
  mutate(dosage=case_match(dosage,&quot;DOX&quot;~&quot;0.5&quot;, .default = dosage)) 

my_fill_colors &lt;-  c(&quot;#41B333&quot;,&quot;#8B006D&quot;,&quot;#41B333&quot;,&quot;#8B006D&quot;,&quot;#DF707E&quot;,&quot;#F1B72B&quot;, &quot;#3386DD&quot;,&quot;#707031&quot;)

lcpm_24h %&gt;% 
  rbind(.,store_box) %&gt;% 
  mutate(dosage=factor(dosage, levels=c(&#39;0&#39;,&#39;0.625&#39;,&quot;1.25&quot;,&quot;2.5&quot;,&quot;5&quot;,&quot;VEH&quot;,&quot;0.5&quot;,&quot;EPI&quot;,&quot;DNR&quot;, &quot;MTX&quot;,&quot;TRZ&quot;))) %&gt;% 
  ggplot(., aes(x=dosage,y=counts))+
  geom_boxplot(aes(fill=dosage))+
  facet_wrap(~hgnc_symbol, scales=&quot;free_y&quot;, nrow=10, ncol = 3 )+
  theme_bw()+
  scale_fill_manual(values=my_fill_colors)</code></pre>
<p><img src="figure/Supplementary_figures.Rmd/Fig%20S15-1.png" width="80%" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Fig-S15-1">
Past versions of "Fig S15-1.png"
</button>
</p>
<div id="fig-no-spaces-Fig-S15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/652df932402672208f615b21dc6dd91860a833aa/docs/figure/Supplementary_figures.Rmd/Fig S15-1.png" target="_blank">652df93</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/d570170a7bf5239d19f52d37928e09bfe113d44d/docs/figure/Supplementary_figures.Rmd/Fig S15-1.png" target="_blank">d570170</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-09-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/021ae6df9b1a6106f052163e95f11f7d3af834aa/docs/figure/Supplementary_figures.Rmd/Fig S15-1.png" target="_blank">021ae6d</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-08-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] edgeR_3.42.4          limma_3.56.2          data.table_1.14.8    
 [4] ggVennDiagram_1.5.0   broom_1.0.5           drc_3.0-1            
 [7] MASS_7.3-60           cowplot_1.1.1         gridExtra_2.3        
[10] ComplexHeatmap_2.16.0 scales_1.3.0          RColorBrewer_1.1-3   
[13] ggsignif_0.6.4        zoo_1.8-12            rstatix_0.7.2        
[16] ggpubr_0.6.0          lubridate_1.9.3       forcats_1.0.0        
[19] stringr_1.5.0         dplyr_1.1.3           purrr_1.0.2          
[22] readr_2.1.4           tidyr_1.3.0           tibble_3.2.1         
[25] ggplot2_3.4.4         tidyverse_2.0.0       workflowr_1.7.1      

loaded via a namespace (and not attached):
  [1] rstudioapi_0.15.0     jsonlite_1.8.7        shape_1.4.6          
  [4] magrittr_2.0.3        TH.data_1.1-2         magick_2.8.1         
  [7] farver_2.1.1          rmarkdown_2.25        GlobalOptions_0.1.2  
 [10] fs_1.6.3              zlibbioc_1.46.0       vctrs_0.6.4          
 [13] webshot_0.5.5         htmltools_0.5.7       plotrix_3.8-4        
 [16] sass_0.4.7            bslib_0.6.1           sandwich_3.1-0       
 [19] cachem_1.0.8          whisker_0.4.1         lifecycle_1.0.4      
 [22] iterators_1.0.14      pkgconfig_2.0.3       Matrix_1.6-2         
 [25] R6_2.5.1              fastmap_1.1.1         clue_0.3-65          
 [28] digest_0.6.33         colorspace_2.1-0      S4Vectors_0.38.2     
 [31] ps_1.7.5              rprojroot_2.0.4       labeling_0.4.3       
 [34] fansi_1.0.5           timechange_0.2.0      httr_1.4.7           
 [37] abind_1.4-5           mgcv_1.9-1            compiler_4.3.1       
 [40] withr_3.0.0           doParallel_1.0.17     backports_1.4.1      
 [43] carData_3.0-5         highr_0.10            rjson_0.2.21         
 [46] gtools_3.9.4          tools_4.3.1           httpuv_1.6.12        
 [49] Cormotif_1.46.0       glue_1.6.2            callr_3.7.3          
 [52] nlme_3.1-164          promises_1.2.1        getPass_0.2-2        
 [55] cluster_2.1.4         generics_0.1.3        gtable_0.3.4         
 [58] tzdb_0.4.0            preprocessCore_1.62.1 hms_1.1.3            
 [61] xml2_1.3.5            car_3.1-2             utf8_1.2.4           
 [64] BiocGenerics_0.46.0   foreach_1.5.2         pillar_1.9.0         
 [67] later_1.3.1           circlize_0.4.15       splines_4.3.1        
 [70] lattice_0.22-5        survival_3.5-7        tidyselect_1.2.0     
 [73] locfit_1.5-9.8        knitr_1.45            git2r_0.32.0         
 [76] IRanges_2.34.1        svglite_2.1.2         stats4_4.3.1         
 [79] xfun_0.41             Biobase_2.60.0        matrixStats_1.1.0    
 [82] stringi_1.7.12        yaml_2.3.7            evaluate_0.23        
 [85] codetools_0.2-19      BiocManager_1.30.22   cli_3.6.1            
 [88] affyio_1.70.0         systemfonts_1.0.5     munsell_0.5.0        
 [91] processx_3.8.2        jquerylib_0.1.4       Rcpp_1.0.11          
 [94] png_0.1-8             parallel_4.3.1        viridisLite_0.4.2    
 [97] mvtnorm_1.2-3         affy_1.78.2           crayon_1.5.2         
[100] GetoptLong_1.0.5      rlang_1.1.2           rvest_1.0.3          
[103] multcomp_1.4-25      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
