<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="ERM" />

<meta name="date" content="2024-02-05" />

<title>Highly variable genes</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cardiotoxicity</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/reneeisnowhere/Cardiotoxicity">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Highly variable genes</h1>
<h4 class="author">ERM</h4>
<h4 class="date">2024-02-05</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-02-05
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Cardiotoxicity/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230109code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230109)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230109code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230109)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommwardlabMatthewsTOP2icardiotox2023treedf08393e94c7f554a1293ba0935e1de7778d9137targetblankdf08393a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/tree/df08393e94c7f554a1293ba0935e1de7778d9137" target="_blank">df08393</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommwardlabMatthewsTOP2icardiotox2023treedf08393e94c7f554a1293ba0935e1de7778d9137targetblankdf08393a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/tree/df08393e94c7f554a1293ba0935e1de7778d9137" target="_blank">df08393</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/variance_values by gene.png
    Ignored:    data/41588_2018_171_MOESM3_ESMeQTL_ST2_for paper.csv
    Ignored:    data/Arr_GWAS.txt
    Ignored:    data/Arr_geneset.RDS
    Ignored:    data/BC_cell_lines.csv
    Ignored:    data/BurridgeDOXTOX.RDS
    Ignored:    data/CADGWASgene_table.csv
    Ignored:    data/CAD_geneset.RDS
    Ignored:    data/CALIMA_Data/
    Ignored:    data/CMD04_75DRCviability.csv
    Ignored:    data/CMD04_87DRCviability.csv
    Ignored:    data/CMD05_75DRCviability.csv
    Ignored:    data/CMD05_87DRCviability.csv
    Ignored:    data/Clamp_Summary.csv
    Ignored:    data/Cormotif_24_k1-5_raw.RDS
    Ignored:    data/Counts_RNA_ERMatthews.txt
    Ignored:    data/DAgostres24.RDS
    Ignored:    data/DAtable1.csv
    Ignored:    data/DDEMresp_list.csv
    Ignored:    data/DDE_reQTL.txt
    Ignored:    data/DDEresp_list.csv
    Ignored:    data/DEG-GO/
    Ignored:    data/DEG_cormotif.RDS
    Ignored:    data/DF_Plate_Peak.csv
    Ignored:    data/DRC48hoursdata.csv
    Ignored:    data/Da24counts.txt
    Ignored:    data/Dx24counts.txt
    Ignored:    data/Dx_reQTL_specific.txt
    Ignored:    data/EPIstorelist24.RDS
    Ignored:    data/Ep24counts.txt
    Ignored:    data/FC_necela.RDS
    Ignored:    data/FC_necela_names.RDS
    Ignored:    data/Full_LD_rep.csv
    Ignored:    data/GOIsig.csv
    Ignored:    data/GOplots.R
    Ignored:    data/GTEX_setsimple.csv
    Ignored:    data/GTEX_sig24.RDS
    Ignored:    data/GTEx_gene_list.csv
    Ignored:    data/HFGWASgene_table.csv
    Ignored:    data/HF_geneset.RDS
    Ignored:    data/Heart_Left_Ventricle.v8.egenes.txt
    Ignored:    data/Heatmap_mat.RDS
    Ignored:    data/Heatmap_sig.RDS
    Ignored:    data/Hf_GWAS.txt
    Ignored:    data/K_cluster
    Ignored:    data/K_cluster_kisthree.csv
    Ignored:    data/K_cluster_kistwo.csv
    Ignored:    data/Knowles_log2cpm_real.RDS
    Ignored:    data/Knowles_variation_data.RDS
    Ignored:    data/Knowles_variation_data_conc.RDS
    Ignored:    data/Knowlesvarlist.RDS
    Ignored:    data/LD50_05via.csv
    Ignored:    data/LDH48hoursdata.csv
    Ignored:    data/Mt24counts.txt
    Ignored:    data/NoRespDEG_final.csv
    Ignored:    data/RINsamplelist.txt
    Ignored:    data/RNA_seq_trial.RDS
    Ignored:    data/Schneider_GWAS.txt
    Ignored:    data/Seonane2019supp1.txt
    Ignored:    data/Sup_replicate_values.csv
    Ignored:    data/TMMnormed_x.RDS
    Ignored:    data/TOP2Bi-24hoursGO_analysis.csv
    Ignored:    data/TR24counts.txt
    Ignored:    data/TableS10.csv
    Ignored:    data/TableS11.csv
    Ignored:    data/TableS9.csv
    Ignored:    data/Top2_expression.RDS
    Ignored:    data/Top2biresp_cluster24h.csv
    Ignored:    data/Var_test_list.RDS
    Ignored:    data/Var_test_list24.RDS
    Ignored:    data/Var_test_list24alt.RDS
    Ignored:    data/Var_test_list3.RDS
    Ignored:    data/Vargenes.RDS
    Ignored:    data/Viabilitylistfull.csv
    Ignored:    data/allexpressedgenes.txt
    Ignored:    data/allfinal3hour.RDS
    Ignored:    data/allgenes.txt
    Ignored:    data/allmatrix.RDS
    Ignored:    data/allmymatrix.RDS
    Ignored:    data/annotation_data_frame.RDS
    Ignored:    data/averageviabilitytable.RDS
    Ignored:    data/averageviabilitytable.csv
    Ignored:    data/avgLD50.RDS
    Ignored:    data/avg_LD50.RDS
    Ignored:    data/avg_via_table.csv
    Ignored:    data/backGL.txt
    Ignored:    data/burr_genes.RDS
    Ignored:    data/calcium_data.RDS
    Ignored:    data/clamp_summary.RDS
    Ignored:    data/cormotif_3hk1-8.RDS
    Ignored:    data/cormotif_initalK5.RDS
    Ignored:    data/cormotif_initialK5.RDS
    Ignored:    data/cormotif_initialall.RDS
    Ignored:    data/cormotifprobs.csv
    Ignored:    data/counts24hours.RDS
    Ignored:    data/cpmcount.RDS
    Ignored:    data/cpmnorm_counts.csv
    Ignored:    data/crispr_genes.csv
    Ignored:    data/ctnnt_results.txt
    Ignored:    data/cvd_GWAS.txt
    Ignored:    data/dat_cpm.RDS
    Ignored:    data/data_outline.txt
    Ignored:    data/drug_noveh1.csv
    Ignored:    data/efit2.RDS
    Ignored:    data/efit2_final.RDS
    Ignored:    data/efit2results.RDS
    Ignored:    data/ensembl_backup.RDS
    Ignored:    data/ensgtotal.txt
    Ignored:    data/filcpm_counts.RDS
    Ignored:    data/filenameonly.txt
    Ignored:    data/filtered_cpm_counts.csv
    Ignored:    data/filtered_raw_counts.csv
    Ignored:    data/filtermatrix_x.RDS
    Ignored:    data/folder_05top/
    Ignored:    data/framefun24.RDS
    Ignored:    data/geneDoxonlyQTL.csv
    Ignored:    data/gene_corr_df.RDS
    Ignored:    data/gene_corr_frame.RDS
    Ignored:    data/gene_prob_tran3h.RDS
    Ignored:    data/gene_probabilityk5.RDS
    Ignored:    data/geneset_24.RDS
    Ignored:    data/gostresTop2bi_ER.RDS
    Ignored:    data/gostresTop2bi_LR
    Ignored:    data/gostresTop2bi_LR.RDS
    Ignored:    data/gostresTop2bi_TI.RDS
    Ignored:    data/gostrescoNR
    Ignored:    data/gtex/
    Ignored:    data/heartgenes.csv
    Ignored:    data/highly_var_genelist.RDS
    Ignored:    data/hsa_kegg_anno.RDS
    Ignored:    data/individualDRCfile.RDS
    Ignored:    data/individual_DRC48.RDS
    Ignored:    data/individual_LDH48.RDS
    Ignored:    data/indv_noveh1.csv
    Ignored:    data/kegglistDEG.RDS
    Ignored:    data/kegglistDEG24.RDS
    Ignored:    data/kegglistDEG3.RDS
    Ignored:    data/knowfig4.csv
    Ignored:    data/knowfig5.csv
    Ignored:    data/label_list.RDS
    Ignored:    data/ld50_table.csv
    Ignored:    data/mean_vardrug1.csv
    Ignored:    data/mean_varframe.csv
    Ignored:    data/mymatrix.RDS
    Ignored:    data/new_ld50avg.RDS
    Ignored:    data/nonresponse_cluster24h.csv
    Ignored:    data/norm_LDH.csv
    Ignored:    data/norm_counts.csv
    Ignored:    data/old_sets/
    Ignored:    data/organized_drugframe.csv
    Ignored:    data/pca_all_anno.csv
    Ignored:    data/plan2plot.png
    Ignored:    data/plot_intv_list.RDS
    Ignored:    data/plot_list_DRC.RDS
    Ignored:    data/qval24hr.RDS
    Ignored:    data/qval3hr.RDS
    Ignored:    data/qvalueEPItemp.RDS
    Ignored:    data/raw_counts.csv
    Ignored:    data/response_cluster24h.csv
    Ignored:    data/sampsettrz.RDS
    Ignored:    data/schneider_closest_output.RDS
    Ignored:    data/sigVDA24.txt
    Ignored:    data/sigVDA3.txt
    Ignored:    data/sigVDX24.txt
    Ignored:    data/sigVDX3.txt
    Ignored:    data/sigVEP24.txt
    Ignored:    data/sigVEP3.txt
    Ignored:    data/sigVMT24.txt
    Ignored:    data/sigVMT3.txt
    Ignored:    data/sigVTR24.txt
    Ignored:    data/sigVTR3.txt
    Ignored:    data/siglist.RDS
    Ignored:    data/siglist_final.RDS
    Ignored:    data/siglist_old.RDS
    Ignored:    data/slope_table.csv
    Ignored:    data/supp10_24hlist.RDS
    Ignored:    data/supp10_3hlist.RDS
    Ignored:    data/supp_normLDH48.RDS
    Ignored:    data/supp_pca_all_anno.RDS
    Ignored:    data/supp_viadata.csv
    Ignored:    data/table3a.omar
    Ignored:    data/test_run_sample_list.txt
    Ignored:    data/testlist.txt
    Ignored:    data/toplistall.RDS
    Ignored:    data/trtonly_24h_genes.RDS
    Ignored:    data/trtonly_3h_genes.RDS
    Ignored:    data/tvl24hour.txt
    Ignored:    data/tvl24hourw.txt
    Ignored:    data/venn_code.R
    Ignored:    data/viability.RDS

Untracked files:
    Untracked:  .RDataTmp
    Untracked:  .RDataTmp1
    Untracked:  .RDataTmp2
    Untracked:  .RDataTmp3
    Untracked:  3hr all.pdf
    Untracked:  Code_files_list.csv
    Untracked:  Data_files_list.csv
    Untracked:  Doxorubicin_vehicle_3_24.csv
    Untracked:  Doxtoplist.csv
    Untracked:  EPIqvalue_analysis.Rmd
    Untracked:  Final.sup.pdf
    Untracked:  GWAS_list_of_interest.xlsx
    Untracked:  KEGGpathwaylist.R
    Untracked:  NA
    Untracked:  OmicNavigator_learn.R
    Untracked:  SNP_egenes_allfiles.RDS
    Untracked:  SNP_frame_pdf
    Untracked:  SNP_frame_pdf.pdf
    Untracked:  SigDoxtoplist.csv
    Untracked:  analysis/DRC_viability_check.Rmd
    Untracked:  analysis/New_code_dec-23.R
    Untracked:  analysis/cellcycle_kegg_genes.R
    Untracked:  analysis/ciFIT.R
    Untracked:  analysis/export_to_excel.R
    Untracked:  analysis/featureCountsPLAY.R
    Untracked:  cleanupfiles_script.R
    Untracked:  code/biomart_gene_names.R
    Untracked:  code/constantcode.R
    Untracked:  code/corMotifcustom.R
    Untracked:  code/cpm_boxplot.R
    Untracked:  code/extracting_ggplot_data.R
    Untracked:  code/movingfilesto_ppl.R
    Untracked:  code/pearson_extract_func.R
    Untracked:  code/pearson_tox_extract.R
    Untracked:  code/plot1C.fun.R
    Untracked:  code/spearman_extract_func.R
    Untracked:  code/venndiagramcolor_control.R
    Untracked:  cormotif_p.post.list_4.csv
    Untracked:  figS1024h.pdf
    Untracked:  final.pdf
    Untracked:  individual-legenddark2.png
    Untracked:  installed_old.rda
    Untracked:  listoftranscripts
    Untracked:  motif_ER.txt
    Untracked:  motif_LR.txt
    Untracked:  motif_NR.txt
    Untracked:  motif_TI.txt
    Untracked:  output/ABHD8_dif_values.RDS
    Untracked:  output/C3orf18_dif_values.RDS
    Untracked:  output/Cardiotox_dif_values.RDS
    Untracked:  output/DNR_DEGlist.csv
    Untracked:  output/DNRvenn.RDS
    Untracked:  output/DOX_DEGlist.csv
    Untracked:  output/DOX_de_goi.csv
    Untracked:  output/DOXvenn.RDS
    Untracked:  output/EEF1B2_dif_values.RDS
    Untracked:  output/EEIG1_dif_values.RDS
    Untracked:  output/EPI_DEGlist.csv
    Untracked:  output/EPIvenn.RDS
    Untracked:  output/ESGN_rds.RDS
    Untracked:  output/FC_necela.RDS
    Untracked:  output/FC_necela_names.RDS
    Untracked:  output/FRS2_dif_values.RDS
    Untracked:  output/Figures/
    Untracked:  output/GTEXv8_gene_median_tpm.RDS
    Untracked:  output/GTEXv8_gene_tpm_heart_left_ventricle.RDS
    Untracked:  output/HDDC2_dif_values.RDS
    Untracked:  output/HER2_gene.RDS
    Untracked:  output/KEGGcellcyclegenes.RDS
    Untracked:  output/Knowles_S13.csv
    Untracked:  output/Knowles_log2cpm.csv
    Untracked:  output/Knowles_supp13.csv
    Untracked:  output/LD50tox_table.RDS
    Untracked:  output/MTX_DEGlist.csv
    Untracked:  output/MTXvenn.RDS
    Untracked:  output/PEX16_dif_values.RDS
    Untracked:  output/RASIP1_dif_values.RDS
    Untracked:  output/RMI1_dif_values.RDS
    Untracked:  output/RSID_QTL_list_full.txt
    Untracked:  output/SETA_analysis_reyes.RDS
    Untracked:  output/SGWAS_top50_order.csv
    Untracked:  output/SLC27A1_dif_values.RDS
    Untracked:  output/SLC28A3_dif_values.RDS
    Untracked:  output/SNP_egenes_allfiles.RDS
    Untracked:  output/SNP_list_ID.RDS
    Untracked:  output/SNP_list_full.txt
    Untracked:  output/SNP_supp.RDS
    Untracked:  output/TGFBR3L_dif_values.RDS
    Untracked:  output/TNS2_dif_values.RDS
    Untracked:  output/TOP_50SNPreffile.csv
    Untracked:  output/TRZ_DEGlist.csv
    Untracked:  output/TableS8.csv
    Untracked:  output/Volcanoplot_10
    Untracked:  output/Volcanoplot_10.RDS
    Untracked:  output/ZNF740_dif_values.RDS
    Untracked:  output/allfinal_sup10.RDS
    Untracked:  output/counts_v8_heart_left_ventricle_gct.RDS
    Untracked:  output/crisprfoldchange.RDS
    Untracked:  output/endocytosisgenes.csv
    Untracked:  output/expre7k.csv
    Untracked:  output/expressed_egenes_by_RSID.csv
    Untracked:  output/gene_corr_fig9.RDS
    Untracked:  output/genes.RDS
    Untracked:  output/legend_b.RDS
    Untracked:  output/motif_ERrep.RDS
    Untracked:  output/motif_LRrep.RDS
    Untracked:  output/motif_NRrep.RDS
    Untracked:  output/motif_TI_rep.RDS
    Untracked:  output/near_genes_SNP1.RDS
    Untracked:  output/necela_list_test.RDS
    Untracked:  output/necela_val_genes.RDS
    Untracked:  output/output-old/
    Untracked:  output/rank24genes.csv
    Untracked:  output/rank3genes.csv
    Untracked:  output/sequencinginformationforsupp.csv
    Untracked:  output/sequencinginformationforsupp.prn
    Untracked:  output/sigVDA24.txt
    Untracked:  output/sigVDA3.txt
    Untracked:  output/sigVDX24.txt
    Untracked:  output/sigVDX3.txt
    Untracked:  output/sigVEP24.txt
    Untracked:  output/sigVEP3.txt
    Untracked:  output/sigVMT24.txt
    Untracked:  output/sigVMT3.txt
    Untracked:  output/sigVTR24.txt
    Untracked:  output/sigVTR3.txt
    Untracked:  output/supplementary_motif_list_GO.RDS
    Untracked:  output/test_biomart_run.RDS
    Untracked:  output/toptablebydrug.RDS
    Untracked:  output/trop_knowles_fun.csv
    Untracked:  output/tvl24hour.txt
    Untracked:  output/x_counts.RDS
    Untracked:  reneebasecode.R

Unstaged changes:
    Modified:   analysis/DRC_analysis.Rmd
    Modified:   analysis/GOI_plots.Rmd
    Modified:   analysis/GTEx_genes.Rmd
    Deleted:    analysis/Knowles2019.Rmd
    Modified:   output/daplot.RDS
    Modified:   output/dxplot.RDS
    Modified:   output/epplot.RDS
    Modified:   output/mtplot.RDS
    Modified:   output/plan2plot.png
    Modified:   output/trplot.RDS
    Modified:   output/veplot.RDS

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Var_genes.Rmd</code>) and HTML
(<code>docs/Var_genes.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/df08393e94c7f554a1293ba0935e1de7778d9137/analysis/Var_genes.Rmd" target="_blank">df08393</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2024-02-05
</td>
<td>
updates to scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/62286c39df588d70d06650c1569865a76958bc09/analysis/Var_genes.Rmd" target="_blank">62286c3</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-28
</td>
<td>
Updateing figure code
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/06800c9a0d5bf629ca2c406711ee7d699ca02279/analysis/Var_genes.Rmd" target="_blank">06800c9</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-26
</td>
<td>
Commits to small changes and edits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mward-lab/Matthews_TOP2i_cardiotox_2023/ee8be4c9b3aff2ee758b025cd3c808ec02f89cf7/docs/Var_genes.html" target="_blank">ee8be4c</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/b94104b2dd9628dc93469cd870f8b597bf0fd812/analysis/Var_genes.Rmd" target="_blank">b94104b</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-21
</td>
<td>
first plot update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Goals for this page:</p>
<p>I will examine the AC-shared variable genes within AC-shared response
genes</p>
<pre class="r"><code>library(tidyverse)
library(VennDiagram)
library(paletteer)
library(ggVennDiagram)
library(gridtext)
library(scales)
library(kableExtra)
library(ComplexHeatmap)
library(data.table)


# library(tidyverse)
library(ggpubr)
library(ggsignif)

# library(paletteer)
# library(ggVennDiagram)
# library(gridtext)
# library(scales)
# library(kableExtra)
library(qvalue)
# library(data.table)
# library(ComplexHeatmap)</code></pre>
<pre class="r"><code>drug_palc &lt;- c(&quot;#8B006D&quot;,&quot;#DF707E&quot;,&quot;#F1B72B&quot;, &quot;#3386DD&quot;,&quot;#707031&quot;,&quot;#41B333&quot;)
Vargenes &lt;- readRDS(&quot;data/geneset_24.RDS&quot;)
expressedgenes &lt;- read.csv(&quot;data/backGL.txt&quot;)
venn24part &lt;- VennDiagram::get.venn.partitions(Vargenes)
backGL &lt;- read.csv(&quot;data/backGL.txt&quot;, row.names = 1)


toplistall &lt;- readRDS(&quot;data/toplistall.RDS&quot;)
  #get sig files made with 0.05 data this way(data created on run)
siglist &lt;- readRDS(&quot;data/siglist_final.RDS&quot;)


siglist24 &lt;- siglist[6:9]
test &lt;- siglist24[1][&quot;ENTREZID&quot;]
sig_24_ENTREZID &lt;- sapply(siglist24,&quot;[[&quot;,1)
sig_24_venn&lt;- VennDiagram::get.venn.partitions(sig_24_ENTREZID)</code></pre>
<div id="ac-shared" class="section level2">
<h2>AC shared</h2>
<pre class="r"><code>AC_share_var &lt;- venn24part$..values..[[25]] #203
not_AC_shared &lt;- setdiff(expressedgenes$ENTREZID,AC_share_var)
AC_shared_sig &lt;- sig_24_venn$..values..[[9]] ##4435
length(intersect(AC_share_var,AC_shared_sig))  #(88 overlap)</code></pre>
<pre><code>[1] 88</code></pre>
<pre class="r"><code>length(intersect(not_AC_shared,AC_shared_sig)) #4347</code></pre>
<pre><code>[1] 4347</code></pre>
<pre class="r"><code># 
# stored &lt;-
  expressedgenes %&gt;% 
  mutate(var_AC= if_else(ENTREZID %in% AC_share_var, &quot;y&quot;,&quot;no&quot;)) %&gt;% 
  mutate(not_AC=if_else(ENTREZID %in% not_AC_shared,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  mutate(AC_shared_sig=if_else(ENTREZID %in% AC_shared_sig,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  # group_by(AC_shared_sig,not_AC)  %&gt;% 
    summarise(&quot;var_share-n&quot; = sum(not_AC==&quot;no&quot;&amp;AC_shared_sig==&quot;no&quot;), &quot;var_share-y&quot;=sum(not_AC==&quot;no&quot;&amp;AC_shared_sig==&quot;y&quot;),&quot;not_var-n&quot;=sum(not_AC==&quot;y&quot;&amp;AC_shared_sig==&quot;no&quot;), &quot;not_var-y&quot;=sum(not_AC==&quot;y&quot;&amp;AC_shared_sig==&quot;y&quot;)) %&gt;% 
    pivot_longer(everything(), names_to = &quot;set&quot;, values_to = &quot;count&quot;) %&gt;% 
    separate(set, into = c(&quot;set&quot;, &quot;color&quot;),sep=&quot;-&quot; ) %&gt;% 
    mutate(color, factor(color, levels = c(&quot;y&quot;,&quot;no&quot;))) %&gt;% 
    mutate(set=case_match(set,&quot;not_var&quot;~&quot;not AC shared\nvar. gene&quot;,&quot;var_share&quot;~&quot;AC shared\nvar. gene&quot;)) %&gt;% 
   ggplot( ., aes(x=set, y=count, group=color,fill=color))+
     geom_col(position=&#39;fill&#39;)+
      theme_classic()+
   scale_color_manual(values=drug_palc)+
     scale_fill_manual(values=c(&quot;y&quot;=&quot;cornflowerblue&quot;,&quot;n&quot;=&quot;darkblue&quot;), 
                    labels=c(&quot;not an AC-sp gene&quot;,&quot;shared with\nAC sp&quot;))+
       scale_y_continuous(expand=expansion(0.001))+
    theme(strip.text=element_text(size=10, face = &quot;bold&quot;),
          axis.text.x = element_text(margin = margin(2,0,2,0, &quot;pt&quot;), size=12),
  strip.background = element_rect (linetype=1, linewidth = 0.5))</code></pre>
<p><img src="figure/Var_genes.Rmd/enrichment%20checks-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-enrichment-checks-1">
Past versions of "enrichment checks-1.png"
</button>
</p>
<div id="fig-no-spaces-enrichment-checks-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/ee8be4c9b3aff2ee758b025cd3c808ec02f89cf7/docs/figure/Var_genes.Rmd/enrichment checks-1.png" target="_blank">ee8be4c</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>testmat &lt;- matrix(c(88,4347,115,9534),nrow = 2,byrow = TRUE)

chisq.test(testmat)#$p.value</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  testmat
X-squared = 12.877, df = 1, p-value = 0.0003326</code></pre>
</div>
<div id="top2i-shared" class="section level2">
<h2>Top2i shared</h2>
<pre class="r"><code>TOP2i_share_var &lt;- venn24part$..values..[[17]] # 32
not_T2_shared &lt;- setdiff(expressedgenes$ENTREZID,TOP2i_share_var)#14052
TOP2i_shared_sig &lt;- sig_24_venn$..values..[[1]] ##882
length(intersect(TOP2i_share_var,TOP2i_shared_sig))  #(3 overlap)</code></pre>
<pre><code>[1] 3</code></pre>
<pre class="r"><code>length(intersect(not_T2_shared,TOP2i_shared_sig)) #879</code></pre>
<pre><code>[1] 879</code></pre>
<pre class="r"><code># stored &lt;-
  expressedgenes %&gt;% 
  mutate(var_t2= if_else(ENTREZID %in% TOP2i_share_var, &quot;y&quot;,&quot;no&quot;)) %&gt;% 
  mutate(not_t2= if_else(ENTREZID %in% not_T2_shared,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  mutate(T2_shared_sig=if_else(ENTREZID %in% TOP2i_shared_sig,&quot;y&quot;,&quot;no&quot;)) %&gt;% 
  # group_by(AC_shared_sig,not_t2)  %&gt;%
    summarise(&quot;var_share-n&quot; = sum(not_t2==&quot;no&quot;&amp;T2_shared_sig==&quot;no&quot;), &quot;var_share-y&quot;=sum(not_t2==&quot;no&quot;&amp;T2_shared_sig==&quot;y&quot;),&quot;not_var-n&quot;=sum(not_t2==&quot;y&quot;&amp;T2_shared_sig==&quot;no&quot;), &quot;not_var-y&quot;=sum(not_t2==&quot;y&quot;&amp;T2_shared_sig==&quot;y&quot;))%&gt;% 
    pivot_longer(everything(), names_to = &quot;set&quot;, values_to = &quot;count&quot;) %&gt;% 
    separate(set, into = c(&quot;set&quot;, &quot;color&quot;),sep=&quot;-&quot; ) %&gt;% 
    mutate(color, factor(color, levels = c(&quot;y&quot;,&quot;no&quot;))) %&gt;% 
    mutate(set=case_match(set,&quot;not_var&quot;~&quot;not TOP2i shared\nvar. gene&quot;,&quot;var_share&quot;~&quot;TOP2i shared\nvar. gene&quot;)) %&gt;% 
   ggplot( ., aes(x=set, y=count, group=color,fill=color))+
     geom_col(position=&#39;fill&#39;)+
      theme_classic()+
    xlab(&quot; &quot;)+
    ggtitle(&quot;TOP2i variable genes in TOP2i specific n= 4435&quot;)+
   scale_color_manual(values=drug_palc)+
     scale_fill_manual(values=c(&quot;y&quot;=&quot;cornflowerblue&quot;,&quot;n&quot;=&quot;darkblue&quot;), 
                    labels=c(&quot;not a TOP2i-sp gene&quot;,&quot;shared with\nTOP2i-sp&quot;))+
       scale_y_continuous(expand=expansion(0.001))+
    theme(strip.text=element_text(size=10, face = &quot;bold&quot;),
          axis.text.x = element_text(margin = margin(2,0,2,0, &quot;pt&quot;), size=12),
  strip.background = element_rect (linetype=1, linewidth = 0.5))</code></pre>
<p><img src="figure/Var_genes.Rmd/enrichment%20top2bi%20checks-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-enrichment-top2bi-checks-1">
Past versions of "enrichment top2bi checks-1.png"
</button>
</p>
<div id="fig-no-spaces-enrichment-top2bi-checks-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/ee8be4c9b3aff2ee758b025cd3c808ec02f89cf7/docs/figure/Var_genes.Rmd/enrichment top2bi checks-1.png" target="_blank">ee8be4c</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>testmatTOP &lt;- matrix(c(3,879,29,13173),nrow = 2,byrow = TRUE)

chisq.test(testmatTOP)#$p.value</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  testmatTOP
X-squared = 0.13128, df = 1, p-value = 0.7171</code></pre>
</div>
<div id="go-heatmaps" class="section level2">
<h2>GO heatmaps</h2>
<pre class="r"><code>DOX_var24gost &lt;- readRDS(&quot;data/DEG-GO/var/DOX_var24gost.RDS&quot;)
DOX_table &lt;- DOX_var24gost$result %&gt;% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value))
EPI_var24gost &lt;- readRDS(&quot;data/DEG-GO/var/EPI_var24gost.RDS&quot;)
EPI_table &lt;- EPI_var24gost$result %&gt;% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value))
DNR_var24gost &lt;- readRDS(&quot;data/DEG-GO/var/DNR_var24gost.RDS&quot;)
DNR_table &lt;- DNR_var24gost$result %&gt;% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value))
MTX_var24gost &lt;- readRDS(&quot;data/DEG-GO/var/MTX_var24gost.RDS&quot;)
MTX_table &lt;- MTX_var24gost$result %&gt;% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value))
TRZ_var24gost &lt;- readRDS(&quot;data/DEG-GO/var/TRZ_var24gost.RDS&quot;)
TRZ_table &lt;- TRZ_var24gost$result %&gt;% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value))
AC_share_var24gost &lt;- readRDS(&quot;data/DEG-GO/var/AC_share_var24gost.RDS&quot;)
AC_share_var24_table &lt;- AC_share_var24gost$result %&gt;% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value))
TOP2i_var24gost &lt;- readRDS(&quot;data/DEG-GO/var/TOP2i_var24gost.RDS&quot;)
TOP2i_var24_table &lt;- TOP2i_var24gost$result %&gt;% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value))
terms &lt;- list()
terms[1] &lt;- DOX_table %&gt;% dplyr::filter(source==&quot;GO:BP&quot;) %&gt;% slice_min(.,p_value, n=3) %&gt;% list()
terms[2] &lt;- EPI_table %&gt;% dplyr::filter(source==&quot;GO:BP&quot;) %&gt;%slice_min(.,p_value, n=3) %&gt;% list()
terms[3] &lt;- DNR_table %&gt;% dplyr::filter(source==&quot;GO:BP&quot;) %&gt;%slice_min(.,p_value, n=3) %&gt;% list()
terms[4] &lt;- MTX_table %&gt;% dplyr::filter(source==&quot;GO:BP&quot;) %&gt;%slice_min(.,p_value, n=3) %&gt;% list()
terms[5] &lt;- TRZ_table %&gt;% dplyr::filter(source==&quot;GO:BP&quot;) %&gt;%slice_min(.,p_value, n=3) %&gt;% list()
terms[6] &lt;- AC_share_var24_table %&gt;% dplyr::filter(source==&quot;GO:BP&quot;) %&gt;%slice_min(.,p_value, n=3) %&gt;% list()
terms[7] &lt;- TOP2i_var24_table %&gt;% dplyr::filter(source==&quot;GO:BP&quot;) %&gt;%slice_min(.,p_value, n=3) %&gt;% list()
names(terms) &lt;- c(&quot;DOX&quot;, &quot;EPI&quot;,&quot;DNR&quot;, &quot;MTX&quot;, &quot;TRZ&quot;, &quot;AC_shared&quot;, &quot;TOP2i_shared&quot;)
termlist &lt;- rbindlist(terms)
termlistid &lt;- c(&quot;GO:0010867&quot;,&quot;GO:0043508&quot;,&quot;GO:0070932&quot;)</code></pre>
<pre class="r"><code>P_valueterm &lt;- list()
P_valueterm[1] &lt;- DOX_table %&gt;% dplyr::filter(term_id %in% termlistid) %&gt;% list()
P_valueterm[2] &lt;- EPI_table %&gt;% dplyr::filter(term_id %in% termlistid) %&gt;% list()
P_valueterm[3] &lt;- DNR_table %&gt;% dplyr::filter(term_id %in% termlistid) %&gt;% list()
P_valueterm[4] &lt;- MTX_table %&gt;% dplyr::filter(term_id %in% termlistid) %&gt;% list()
P_valueterm[5] &lt;- TRZ_table %&gt;% dplyr::filter(term_id %in% termlistid) %&gt;% list()
P_valueterm[6] &lt;- AC_share_var24_table %&gt;% dplyr::filter(term_id %in% termlistid) %&gt;% list()
P_valueterm[7] &lt;- TOP2i_var24_table %&gt;% dplyr::filter(term_id %in% termlistid) %&gt;% list()

names(P_valueterm) &lt;- c(&quot;DOX&quot;, &quot;EPI&quot;,&quot;DNR&quot;, &quot;MTX&quot;, &quot;TRZ&quot;, &quot;AC_shared&quot;, &quot;TOP2i_shared&quot;)

GO_heatmapdata &lt;- rbindlist(P_valueterm,idcol= &quot;deg&quot;)
col_funkegg= circlize::colorRamp2(c(0, 5), c(&quot;white&quot;, &quot;darkred&quot;))


GO_sig_mat &lt;- GO_heatmapdata %&gt;% 
  dplyr::select(deg,p_value,term_name) %&gt;%
# mutate(term_name= case_match(term_name,&quot;Cell cycle&quot;~&quot;Cell\ncycle&quot;,&quot;p53 signaling pathway&quot;~&quot;p53\nsig.\npath.&quot;,&quot;Base excision repair&quot;~&quot;Base\nexcision\nrepair&quot;, 
# &quot;DNA replication&quot;~&quot;DNA\nrep.&quot;,.default = term_name)) %&gt;% 
  pivot_wider(id_cols = everything(),
              names_from=&quot;term_name&quot;,
              values_from=&quot;p_value&quot;,
              values_fill = list(p_value = 1)) %&gt;% 
  column_to_rownames(&#39;deg&#39;) %&gt;% 
  as.matrix()# 

GO_mat&lt;- GO_heatmapdata%&gt;%
  mutate(log_val= (-log10(p_value))) %&gt;% 
  dplyr::select(deg,log_val,term_name) %&gt;%
  mutate(term_name= case_match(term_name,&quot;histone H3 deacetylation&quot;~&quot;histone H3\n deacetylation&quot;,&quot;negative regulation of JUN kinase activity
&quot;~&quot;neg. reg. of\nJUN kinase\nactivity&quot;,&quot;positive regulation of triglyceride biosynthetic process&quot;~&quot;pos. reg.\nof triglyceride\nbiosynthetic\nprocess&quot;,.default = term_name)) %&gt;%
  pivot_wider(id_cols = everything(),
              names_from=&quot;term_name&quot;,values_from=&quot;log_val&quot;) %&gt;% 
  column_to_rownames(&#39;deg&#39;) %&gt;% 
  as.matrix()# 

Heatmap(GO_mat,
        column_title = &quot;GO -log10 p values&quot;,
        name = &quot;-log10 (p value)&quot;,
        cluster_rows = FALSE, 
        cluster_columns = FALSE,
        column_names_rot = 0,
        column_dend_side = &quot;bottom&quot;,
        
        column_names_max_height = unit(12,&quot;cm&quot;),
        column_names_centered = TRUE,
        row_names_max_width = max_text_width(
        rownames(GO_mat), 
        gp = gpar(fontsize = 10)),
        col = col_funkegg,
        na_col=&quot;lightyellow&quot;,
        column_labels = paste0(c(&quot;histone H3\n deacetylation&quot;,
                                    &quot;neg. reg. of\nJUN kinase\nactivity&quot;,
                                    &quot;pos. reg.of\ntriglyceride\nbiosynthetic\nprocess&quot;)),
        cell_fun = function(j, i, x, y, width, height, fill) {
        if(GO_sig_mat[i, j]&lt; 0.05)
            grid.text(&quot;*&quot;, x, y, gp = gpar(fontsize = 20))
})</code></pre>
<p><img src="figure/Var_genes.Rmd/actual%20heatmap-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-actual-heatmap-1">
Past versions of "actual heatmap-1.png"
</button>
</p>
<div id="fig-no-spaces-actual-heatmap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mward-lab/Matthews_TOP2i_cardiotox_2023/blob/ee8be4c9b3aff2ee758b025cd3c808ec02f89cf7/docs/figure/Var_genes.Rmd/actual heatmap-1.png" target="_blank">ee8be4c</a>
</td>
<td>
reneeisnowhere
</td>
<td>
2023-07-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="seoane-data-investigation-of-variability"
class="section level3">
<h3>Seoane data investigation of variability</h3>
<p>I will examine the Seoane gene sets and overlap with variable
data</p>
<pre class="r"><code>list2env(Vargenes, envir = .GlobalEnv)</code></pre>
<pre><code>&lt;environment: R_GlobalEnv&gt;</code></pre>
<pre class="r"><code>chrom_reg_Seoane &lt;- read_csv(file = &quot;data/Seonane2019supp1.txt&quot;,col_types = cols(...1 = col_skip()))
Seoane_2019 &lt;- chrom_reg_Seoane[,2]
names(Seoane_2019) &lt;- &quot;ENTREZID&quot;
Sup1seoane &lt;- (unique(Seoane_2019$ENTREZID))
Sup4genes &lt;- read.csv(&quot;output/Sup4seoane.csv&quot;, row.names = 1)
Sup4seoane &lt;- Sup4genes  %&gt;% 
  dplyr::filter(pval.expAnth&lt;0.05) %&gt;% 
  distinct(entrez, .keep_all = TRUE) %&gt;% 
  dplyr::select(entrez) %&gt;% 
  rename(&quot;ENTREZID&quot;=&#39;entrez&#39;)</code></pre>
<pre class="r"><code>intersect(Sup1seoane,AC_share_var)</code></pre>
<pre><code>[1] &quot;387893&quot; &quot;79723&quot;  &quot;86&quot;     &quot;1105&quot;   &quot;8473&quot;   &quot;3012&quot;  </code></pre>
<pre class="r"><code># 
intersect(Sup4seoane$ENTREZID,as.numeric(AC_share_var))</code></pre>
<pre><code>numeric(0)</code></pre>
<pre class="r"><code>intersect(Sup4seoane$ENTREZID,as.numeric(not_AC_shared))</code></pre>
<pre><code> [1] 11176 10284  8819 23522  7786  2146  4297 79913  8242 51780  6872 23135
[13]  6877 23030 64324 79885 10847 51773  5253  9126  3054  9734 53335 27350
[25]  6601  1108  8289   890 64151 10445  7150  8110 54531 51409 27097  9739
[37]  6595  9555 22823 54556 10592  7528  9031 51377  7799  6602  8202 51564
[49] 79858 10856</code></pre>
<pre class="r"><code># Vargenes[[length(Vargenes)+1]] &lt;- list(Sup4seoane$ENTREZID)

library(paletteer)
assignInNamespace(x=&quot;plot_venn&quot;, value=plot_venn, ns=&quot;ggVennDiagram&quot;)

sup1overlap &lt;- list(as.numeric(DOX_24_var),as.numeric(EPI_24_var),as.numeric(DNR_24_var),as.numeric(MTX_24_var), Sup1seoane)
# re_incommon &lt;- c(DOXreQTLs$ENTREZID,sigVDA24$ENTREZID, sigVEP24$ENTREZID, sigVMT24$ENTREZID)
# 
# names(reQTL_overlapDE24) &lt;- c(&quot;Dox_reQTLS&quot;, &quot;DNR DEGs&quot;,&quot;EPI DEGs&quot;,&quot;MTX DEGs&quot;)

ggVennDiagram::ggVennDiagram(sup1overlap,
                             category.names = c(&quot;DOX var&quot;,
                                     &quot;EPI var&quot;,
                                      &quot;DNR var&quot;,
                                 &quot;MTX var&quot;,
                                 &quot;Seoane S1&quot;),
              show_intersect = FALSE,
              set_color = &quot;black&quot;,
              # category_size = c(6,6,6,6),
              label = &quot;count&quot;,
              # color =  c(&quot;DOX\negenes&quot; = &quot;yellow&quot;,&quot;DNR DEGs&quot; =&quot;steelblue&quot;,&quot;EPI DEGs&quot; = &#39;red&#39;, &quot;MTX DEGs&quot; = &#39;black&#39;) ,
          
              label_percent_digit = 1,
              label_size = 4,
              label_alpha = 0,
              label_color = &quot;black&quot;,
              edge_lty = &quot;solid&quot;, set_size = 4.5)+
  scale_x_continuous(expand = expansion(mult = .3))+
  scale_y_continuous(expand = expansion(mult = .2))+
  # scale_color_paletteer_d(palette = &quot;fishualize::Bodianus_pulchellus&quot;)+
  # scale_colour_gradientn(colours = cols, 
  #                      values = rescale(c(0, 20, 75, 1000, 5000)),
  #                      guide = &quot;colorbar&quot;, limits=c(0, 100)) +
  scale_fill_distiller(palette=&quot;Spectral&quot;, direction = -1, limits= c(1,200),oob=scales::squish)#+</code></pre>
<p><img src="figure/Var_genes.Rmd/venn%20seoane-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # scale_fill_manual(values = cbp1)+
  labs(title = &quot;24 hour DOX egenes in other DEG sets&quot;)+
  theme(plot.title = element_text(size = rel(1.6), hjust = 0.5, vjust =1))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>sup4overlap &lt;- list(as.numeric(DOX_24_var),as.numeric(EPI_24_var),as.numeric(DNR_24_var),as.numeric(MTX_24_var), as.numeric(Sup4seoane$ENTREZID))
ggVennDiagram::ggVennDiagram(sup4overlap,
                             category.names = c(&quot;DOX var&quot;,
                                     &quot;EPI var&quot;,
                                      &quot;DNR var&quot;,
                                 &quot;MTX var&quot;,
                                 &quot;Seoane S4&quot;),
              show_intersect = FALSE,
              set_color = &quot;black&quot;,
              # category_size = c(6,6,6,6),
              label = &quot;count&quot;,
              # color =  c(&quot;DOX\negenes&quot; = &quot;yellow&quot;,&quot;DNR DEGs&quot; =&quot;steelblue&quot;,&quot;EPI DEGs&quot; = &#39;red&#39;, &quot;MTX DEGs&quot; = &#39;black&#39;) ,
          
              label_percent_digit = 1,
              label_size = 4,
              label_alpha = 0,
              label_color = &quot;black&quot;,
              edge_lty = &quot;solid&quot;, set_size = 4.5)+
  scale_x_continuous(expand = expansion(mult = .3))+
  scale_y_continuous(expand = expansion(mult = .2))+
  # scale_color_paletteer_d(palette = &quot;fishualize::Bodianus_pulchellus&quot;)+
  # scale_colour_gradientn(colours = cols, 
  #                      values = rescale(c(0, 20, 75, 1000, 5000)),
  #                      guide = &quot;colorbar&quot;, limits=c(0, 100)) +
  scale_fill_distiller(palette=&quot;Spectral&quot;, direction = -1, limits= c(1,200),oob=scales::squish)#+</code></pre>
<p><img src="figure/Var_genes.Rmd/supp4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # scale_fill_manual(values = cbp1)+
  labs(title = &quot;24 hour DOX egenes in other DEG sets&quot;)+
  theme(plot.title = element_text(size = rel(1.6), hjust = 0.5, vjust =1))</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>DEG &lt;- Vargenes
siglist24 &lt;- siglist[6:9]
test &lt;- siglist24[1][&quot;ENTREZID&quot;]
sig_24_ENTREZID &lt;- sapply(siglist24,&quot;[[&quot;,1)
all_unique_DEG &lt;- unique(rbindlist(lapply(sig_24_ENTREZID, as.data.table))) ##8188
all_unique_var &lt;- unique(rbindlist((lapply(Vargenes, as.data.table))))

length(intersect(all_unique_DEG$V1,all_unique_var$V1))##2190</code></pre>
<pre><code>[1] 2190</code></pre>
<pre class="r"><code>DEG_var &lt;- intersect(all_unique_DEG$V1,all_unique_var$V1)
non_DEG_var &lt;- setdiff(expressedgenes$ENTREZID,DEG_var)

ggVennDiagram::ggVennDiagram(list(all_unique_var$V1, all_unique_DEG$V1),
                             category.names = c(&quot;all var genes&quot;, &quot;all DEGs&quot;),
              show_intersect = FALSE,
              set_color = &quot;black&quot;,
              label = &quot;count&quot;,
              label_percent_digit = 1,
              label_size = 4,
              label_alpha = 0,
              label_color = &quot;black&quot;,
              edge_lty = &quot;solid&quot;, set_size = 4.5)+
  scale_x_continuous(expand = expansion(mult = .3))+
  scale_y_continuous(expand = expansion(mult = .2))+
  # scale_color_paletteer_d(palette = &quot;fishualize::Bodianus_pulchellus&quot;)+
  # scale_colour_gradientn(colours = cols, 
  #                      values = rescale(c(0, 20, 75, 1000, 5000)),
  #                      guide = &quot;colorbar&quot;, limits=c(0, 100)) +
  scale_fill_distiller(palette=&quot;Spectral&quot;, direction = -1, limits= c(1000,8000),oob=scales::squish)#+</code></pre>
<p><img src="figure/Var_genes.Rmd/overlap%20with%20degs-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>holdlist &lt;-  list(DEG_var, as.numeric(non_DEG_var), Sup1seoane)
ggVennDiagram::ggVennDiagram(list( as.numeric(DEG_var), as.numeric(non_DEG_var), Sup1seoane),
                             category.names = c(&quot;DEG var&quot;, &quot;non DEG var&quot;,&quot;Seoane supp 1&quot;),
              show_intersect = FALSE,
              set_color = &quot;black&quot;,
              label = &quot;count&quot;,
              label_percent_digit = 1,
              label_size = 4,
              label_alpha = 0,
              label_color = &quot;black&quot;,
              edge_lty = &quot;solid&quot;, set_size = 4.5)+
  scale_x_continuous(expand = expansion(mult = .3))+
  scale_y_continuous(expand = expansion(mult = .2))+
  # scale_color_paletteer_d(palette = &quot;fishualize::Bodianus_pulchellus&quot;)+
  # scale_colour_gradientn(colours = cols, 
  #                      values = rescale(c(0, 20, 75, 1000, 5000)),
  #                      guide = &quot;colorbar&quot;, limits=c(0, 100)) +
  scale_fill_distiller(palette=&quot;Spectral&quot;, direction = -1, limits= c(1000,8000),oob=scales::squish)#+</code></pre>
<p><img src="figure/Var_genes.Rmd/overlap%20with%20degs-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>testS1 &lt;- matrix(c(59,271,2131,11623),nrow = 2,byrow = TRUE)

chisq.test(testS1)</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test with Yates&#39; continuity correction

data:  testS1
X-squared = 1.2204, df = 1, p-value = 0.2693</code></pre>
</div>
</div>
<div id="burridge-data" class="section level2">
<h2>Burridge data</h2>
<pre class="r"><code>DOXreQTLs &lt;- readRDS(&quot;output/DOXreQTLs.RDS&quot;)
burr_genes &lt;-  readRDS(&quot;data/BurridgeDOXTOX.RDS&quot;)
# BurridgeDOXTOX &lt;-  getBM(attributes=my_attributes,filters =&#39;hgnc_symbol&#39;,
#          values = burr_genes, mart = ensembl)
# BurridgeDOXTOX &lt;- BurridgeDOXTOX %&gt;% distinct(entrezgene_id, .keep_all = TRUE)
# saveRDS(BurridgeDOXTOX, &quot;data/BurridgeDOXTOX.RDS&quot;) 


storeEPI &lt;- readRDS(&quot;data/qvalueEPItemp.RDS&quot;)
VennDiagram::get.venn.partitions(list(storeEPI$ENTREZID, DOXreQTLs$ENTREZID))</code></pre>
<pre><code>     X1    X2   ..set..   ..values.. ..count..
1  TRUE  TRUE     X1∩X2 5119, 93....         4
2 FALSE  TRUE (X2)∖(X1) 29965, 2....       138
3  TRUE FALSE (X1)∖(X2) 49856, 2....       504</code></pre>
<pre class="r"><code>intersect(DOXreQTLs$ENTREZID, burr_genes$entrezgene_id)</code></pre>
<pre><code>character(0)</code></pre>
<pre class="r"><code># DOX_reqtls &lt;-$ENTREZID %&gt;% as.integer()
  DOXreQTLs %&gt;% dplyr::filter(ENTREZID %in% storeEPI$ENTREZID) %&gt;%  tally( )</code></pre>
<pre><code>  n
1 4</code></pre>
<pre class="r"><code> burr_genes %&gt;% 
   dplyr::filter(entrezgene_id%in% backGL$ENTREZID) %&gt;% 
   dplyr::filter(entrezgene_id %in% storeEPI$ENTREZID)</code></pre>
<pre><code>[1] entrezgene_id   ensembl_gene_id hgnc_symbol    
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre class="r"><code>### now for burr genes and epi508
VennDiagram::get.venn.partitions(list(storeEPI$ENTREZID, burr_genes$entrezgene_id))</code></pre>
<pre><code>     X1    X2   ..set..   ..values.. ..count..
1  TRUE  TRUE     X1∩X2                      0
2 FALSE  TRUE (X2)∖(X1) 220, 790....        22
3  TRUE FALSE (X1)∖(X2) 49856, 2....       508</code></pre>
<pre class="r"><code>### looking at the overlap of egenes and  EPI 508 var

storeEPI %&gt;% 
  dplyr::filter (ENTREZID %in% DOXreQTLs$ENTREZID) %&gt;% 
  left_join(., backGL, by=c(&quot;ENTREZID&quot;))</code></pre>
<pre><code>  ENTREZID  EPI.VEH.24    qvalues SYMBOL ensembl_gene_id hgnc_symbol
1     5119 0.001617342 0.07437143 CHMP1A ENSG00000131165      CHMP1A
2    93134 0.002341605 0.08331969 ZNF561 ENSG00000171469      ZNF561
3   132001 0.002319133 0.08331969 TAMM41 ENSG00000279643      TAMM41
4    10592 0.003429113 0.08968566   SMC2 ENSG00000136824        SMC2</code></pre>
<div id="looking-at-heatmaps" class="section level3">
<h3>looking at heatmaps</h3>
<pre class="r"><code>mean_vardrug1 &lt;- read.csv(&quot;data/mean_vardrug1.csv&quot;, row.names = 1)


drug_frame &lt;- mean_vardrug1 %&gt;%
  rownames_to_column(var = &quot;entrezid&quot;) %&gt;%
  pivot_longer(cols = mean.Da.3:var.Ve.24,
               names_to = &quot;short&quot;,
               values_to = &quot;values&quot;) %&gt;%
  separate(short, into = c(&quot;calc&quot;, &quot;treatment&quot;, &quot;time&quot;)) %&gt;%
  # mutate(treatment = factor(
  #   treatment,
  #   levels = c(&quot;Do&quot;, &quot;Ep&quot;, &quot;Da&quot;, &quot;Mi&quot;, &quot;Tr&quot;, &quot;Ve&quot;),
  #   labels = c(&quot;DOX&quot;, &quot;EPI&quot;, &quot;DNR&quot;, &quot;MTX&quot;, &quot;TRZ&quot;, &quot;VEH&quot;)
  # )) %&gt;%
  # mutate(time = factor(
  #   time,
  #   levels = c(&quot;3&quot;, &quot;24&quot;),
  #   labels = c(&quot;3 hours&quot;, &quot;24 hours&quot;)
  # )) %&gt;%
  # group_by(treatment, time, calc) %&gt;%
  as.data.frame</code></pre>
<pre class="r"><code>v_DNR_24 &lt;- drug_frame %&gt;% 
  dplyr::filter(calc !=&quot;mean&quot;, treatment ==&quot;Da&quot;, time==&quot;24&quot;) %&gt;% 
  select(entrezid,values)%&gt;% 
  rename(&quot;DNR&quot;=&quot;values&quot;)
  
v_DOX_24 &lt;- drug_frame %&gt;% 
  dplyr::filter(calc !=&quot;mean&quot;, treatment ==&quot;Do&quot;, time==&quot;24&quot;) %&gt;% 
  select(entrezid,values)%&gt;% 
  rename(&quot;DOX&quot;=&quot;values&quot;)

v_EPI_24 &lt;- drug_frame %&gt;% 
  dplyr::filter(calc !=&quot;mean&quot;, treatment ==&quot;Ep&quot;, time==&quot;24&quot;) %&gt;% 
  select(entrezid,values)%&gt;% 
  rename(&quot;EPI&quot;=&quot;values&quot;)

v_MTX_24 &lt;- drug_frame %&gt;% 
  dplyr::filter(calc !=&quot;mean&quot;, treatment ==&quot;Mi&quot;, time==&quot;24&quot;) %&gt;% 
  select(entrezid,values)%&gt;% 
  rename(&quot;MTX&quot;=&quot;values&quot;)

v_TRZ_24 &lt;- drug_frame %&gt;% 
  dplyr::filter(calc !=&quot;mean&quot;, treatment ==&quot;Tr&quot;, time==&quot;24&quot;) %&gt;% 
  select(entrezid,values)%&gt;% 
  rename(&quot;TRZ&quot;=&quot;values&quot;)

v_VEH_24 &lt;- drug_frame %&gt;% 
  dplyr::filter(calc !=&quot;mean&quot;, treatment ==&quot;Ve&quot;, time==&quot;24&quot;) %&gt;% 
  select(entrezid,values)%&gt;% 
  rename(&quot;VEH&quot;=&quot;values&quot;)
        
val_mat_24 &lt;- left_join(v_DNR_24,v_DOX_24,
                        by=c(&quot;entrezid&quot;))%&gt;%
  left_join(.,v_EPI_24, by=c(&quot;entrezid&quot;)) %&gt;%
  left_join(.,v_MTX_24, by=c(&quot;entrezid&quot;)) %&gt;% 
  left_join(.,v_TRZ_24, by=c(&quot;entrezid&quot;)) %&gt;% 
  left_join(.,v_VEH_24, by=c(&quot;entrezid&quot;)) %&gt;% 
  column_to_rownames(&quot;entrezid&quot;) %&gt;% 
  as.matrix
summary(val_mat_24)</code></pre>
<pre><code>      DNR                 DOX                EPI                 MTX           
 Min.   : 0.000902   Min.   : 0.00139   Min.   : 0.001492   Min.   : 0.000579  
 1st Qu.: 0.057038   1st Qu.: 0.07892   1st Qu.: 0.124501   1st Qu.: 0.045321  
 Median : 0.117373   Median : 0.16358   Median : 0.266216   Median : 0.095530  
 Mean   : 0.249433   Mean   : 0.32816   Mean   : 0.488970   Mean   : 0.235354  
 3rd Qu.: 0.252505   3rd Qu.: 0.35258   3rd Qu.: 0.571324   3rd Qu.: 0.230010  
 Max.   :28.118788   Max.   :33.54419   Max.   :26.410869   Max.   :27.443180  
      TRZ                VEH          
 Min.   : 0.00076   Min.   : 0.00057  
 1st Qu.: 0.04751   1st Qu.: 0.05105  
 Median : 0.10385   Median : 0.11627  
 Mean   : 0.27199   Mean   : 0.30214  
 3rd Qu.: 0.24733   3rd Qu.: 0.28361  
 Max.   :32.76541   Max.   :30.87884  </code></pre>
<pre class="r"><code>col_fun = circlize::colorRamp2(c(0, 0.4, 2), c(&quot;#377EB8&quot;, &quot;white&quot;, &quot;#E41A1C&quot;))
Heatmap(val_mat_24, col=col_fun,
        show_row_names = FALSE, 
        # width = unit(5, &quot;mm&quot;),
        cluster_columns = FALSE)</code></pre>
<p><img src="figure/Var_genes.Rmd/individual%20matrix-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] qvalue_2.32.0         ggsignif_0.6.4        ggpubr_0.6.0         
 [4] data.table_1.14.8     ComplexHeatmap_2.16.0 kableExtra_1.3.4     
 [7] scales_1.3.0          gridtext_0.1.5        ggVennDiagram_1.5.0  
[10] paletteer_1.6.0       VennDiagram_1.7.3     futile.logger_1.4.3  
[13] lubridate_1.9.3       forcats_1.0.0         stringr_1.5.0        
[16] dplyr_1.1.3           purrr_1.0.2           readr_2.1.4          
[19] tidyr_1.3.0           tibble_3.2.1          ggplot2_3.4.4        
[22] tidyverse_2.0.0       workflowr_1.7.1      

loaded via a namespace (and not attached):
 [1] formatR_1.14         rematch2_2.1.2       rlang_1.1.2         
 [4] magrittr_2.0.3       clue_0.3-65          GetoptLong_1.0.5    
 [7] git2r_0.32.0         matrixStats_1.1.0    compiler_4.3.1      
[10] getPass_0.2-2        png_0.1-8            systemfonts_1.0.5   
[13] callr_3.7.3          vctrs_0.6.4          reshape2_1.4.4      
[16] rvest_1.0.3          pkgconfig_2.0.3      shape_1.4.6         
[19] crayon_1.5.2         fastmap_1.1.1        magick_2.8.1        
[22] backports_1.4.1      labeling_0.4.3       utf8_1.2.4          
[25] promises_1.2.1       rmarkdown_2.25       tzdb_0.4.0          
[28] ps_1.7.5             bit_4.0.5            xfun_0.41           
[31] cachem_1.0.8         jsonlite_1.8.7       highr_0.10          
[34] later_1.3.1          broom_1.0.5          parallel_4.3.1      
[37] cluster_2.1.4        R6_2.5.1             bslib_0.6.1         
[40] stringi_1.7.12       RColorBrewer_1.1-3   car_3.1-2           
[43] jquerylib_0.1.4      Rcpp_1.0.11          iterators_1.0.14    
[46] knitr_1.45           IRanges_2.34.1       splines_4.3.1       
[49] httpuv_1.6.12        timechange_0.2.0     tidyselect_1.2.0    
[52] rstudioapi_0.15.0    abind_1.4-5          yaml_2.3.7          
[55] doParallel_1.0.17    codetools_0.2-19     processx_3.8.2      
[58] plyr_1.8.9           withr_3.0.0          evaluate_0.23       
[61] lambda.r_1.2.4       xml2_1.3.5           circlize_0.4.15     
[64] pillar_1.9.0         carData_3.0-5        whisker_0.4.1       
[67] foreach_1.5.2        stats4_4.3.1         generics_0.1.3      
[70] vroom_1.6.5          rprojroot_2.0.4      S4Vectors_0.38.2    
[73] hms_1.1.3            munsell_0.5.0        glue_1.6.2          
[76] tools_4.3.1          webshot_0.5.5        fs_1.6.3            
[79] colorspace_2.1-0     cli_3.6.1            futile.options_1.0.1
[82] fansi_1.0.5          viridisLite_0.4.2    svglite_2.1.2       
[85] gtable_0.3.4         rstatix_0.7.2        sass_0.4.7          
[88] digest_0.6.33        BiocGenerics_0.46.0  farver_2.1.1        
[91] rjson_0.2.21         htmltools_0.5.7      lifecycle_1.0.4     
[94] httr_1.4.7           GlobalOptions_0.1.2  bit64_4.0.5         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
