---
title: "After_comments"
author: "ERM"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE
)
```

```{css, echo=FALSE}
pre {
  max-height: 400px;
  overflow-y: auto;
}

pre[class] {
  max-height: 200px;
}
```

```{r functions}
drug_pal <- c("#707031","#41B333")
# use "Dark2" for individuals, and drug_pal for drug fill

cpm_TRZ <-
  function(cpmcounts,
           GOI,
           ylab,
           titlename) {
    ##GOI needs to be ENTREZID
    df <- trz_veh_count %>% 
      column_to_rownames("ENTREZID")
    df_plot <- df %>%
      dplyr::filter(rownames(.) == GOI) %>%
      pivot_longer(everything(),
                   names_to = "treatment",
                   values_to = "counts") %>%
      separate(treatment, c("drug", "indv", "time")) %>%
      mutate(time = factor(time, levels = c("3h", "24h"))) %>%
      mutate(indv = factor(indv, levels = c(1, 2, 3, 4, 5, 6))) %>%
      mutate(
        drug = case_match(
          drug,
          "Da" ~ "DNR",
          "Do" ~ "DOX",
          "Ep" ~ "EPI",
          "Mi" ~ "MTX",
          "Tr" ~ "TRZ",
          "Ve" ~ "VEH",
          .default = drug
        )
      ) %>%
      mutate(drug = factor(drug, levels = c('DOX', 'EPI', 'DNR', 'MTX', 'TRZ', 'VEH')))
    plot <-
    ggplot2::ggplot(df_plot, aes(x = drug, y = counts)) +
      geom_boxplot(position = "identity", aes(fill = drug)) +
      geom_point(aes(
        col = indv,
        size = 1.5,
        alpha = 0.5
      )) +
      guides(alpha = "none", size = "none") +
      # scale_color_brewer(palette = "Dark2", guide = "none") +
      scale_fill_manual(values = drug_pal) +
      facet_wrap("time", nrow = 1, ncol = 2) +
      theme_bw() +
      ylab(ylab) +
      xlab("") +
      ggtitle(titlename)+
      theme(
        strip.background = element_rect(
          fill = "white",
          linetype = 1,
          linewidth = 0.5
        ),
        plot.title = element_text(
          size = 10,
          hjust = 0.5,
          face = "bold"
        ),
        axis.title = element_text(size = 10, color = "black"),
        axis.ticks = element_line(linewidth = 1.0),
        axis.line = element_line(linewidth = 1.0),
        axis.text.x = element_blank(),
        strip.text.x = element_text(margin = margin(2, 0, 2, 0, "pt"), face = "bold")
      )
    print(plot)
  }



```

```{r package loading}
library(tidyverse)
library(ggsignif)
library(cowplot)
library(ggpubr)
library(scales)
library(sjmisc)
library(kableExtra)
library(broom)
library(ComplexHeatmap)
library(ggVennDiagram)
library(biomaRt)
library(edgeR)
library(RColorBrewer)


```

### Loading of data from Necela, supplemental 1.

```{r load_data}
backGL <- read.csv("data/backGL.txt")
FC_necela <- readRDS("output/FC_necela.RDS")
FC_mine <- readRDS("data/toplistall.RDS")

TRZ_list <- FC_mine %>% 
  dplyr::filter(id=="TRZ",P.Value < 0.05) #%>% 

TRZ_list %>% 
kable(., caption = "List of nominal pvalue TRZ genes in my data") %>%
  kable_paper("striped", full_width = FALSE) %>%
  kable_styling(
    full_width = FALSE,
    position = "left"
  ) %>%
  scroll_box(width = "100%", height = "400px")


necela_list <- FC_necela %>% 
  dplyr::filter(PValue<0.05) %>% 
  dplyr::select(Gene) 

FC_necela %>%# mutate_at(.vars = 6, .funs = scientific_format()) %>%
  kable(., caption = "List of nominal pvalue Necela TRZ genes") %>%
  kable_paper("striped", full_width = FALSE) %>%
  kable_styling(
    full_width = FALSE,
    position = "left"
  ) %>%
  scroll_box(width = "100%", height = "400px")

ggVennDiagram::ggVennDiagram(list(necela_list$Gene, TRZ_list$SYMBOL))
intersect(necela_list$Gene,TRZ_list$SYMBOL)
ven_list <- VennDiagram::get.venn.partitions(list(necela_list$Gene, TRZ_list$SYMBOL))
 TRZ_list %>% 
  group_by(time) %>% 
 count() %>% 
  mutate(time = factor(time, levels = c("3_hours", "24_hours"), labels = c("3 hours", "24 hours")))%>% 
  
  ggplot(., aes(x=time, y=n))+
  geom_col()+
   theme_bw() +
  ggtitle("Number of nominal pvalue TRZ genes by time"
  )+
  ylab(expression("count")) +
  theme(
    strip.background = element_rect(
      fill = "white",
      linetype = 1,
      linewidth = 0.5
    ),
    plot.title = element_text(size = 12, hjust = 0.5, face = "bold"),
    axis.title = element_text(size = 10, color = "black"),
    axis.ticks = element_line(linewidth = 1.0),
    panel.background = element_rect(colour = "black", size = 1),
    # axis.text.x = element_blank(),
    strip.text.x = element_text(margin = margin(2, 0, 2, 0, "mm"), face = "bold")
  )

```

```{r}
FC_mine %>% 
  separate(time, into=c("number","unit",sep="_"))
```

```{r basicBiomart}
ensembl <- useMart(biomart="ensembl",dataset ="hsapiens_gene_ensembl")

listogene <- necela_list$Gene
my_attributes <- c('entrezgene_id', 'ensembl_gene_id', 'hgnc_symbol')
# devtools::install_version("dbplyr", version = "2.3.4")
# necela_list_test <- getBM(attributes = my_attributes,
#                      filters ="hgnc_symbol",
#                      values =listogene,
#                      mart = ensembl)
# # 
# necela_val_genes <- getBM(attributes = my_attributes, 
#                      filters ="hgnc_symbol",   
#                      values =c("RGS4","EGR1", "SLC6A6", "PHLDA1"), 
#                      mart = ensembl)
# HER2_gene <- getBM(attributes = my_attributes,
#                    filters = "entrezgene_id",
#                    values= 2064,
#                    mart = ensembl)
# FC_necela_names <- getBM(attributes = my_attributes,
#                          filters = "hgnc_symbol",
#                          values=FC_necela$Gene,
#                          mart = ensembl)
HER2_gene <- readRDS("output/HER2_gene.RDS")
FC_necela_names <- readRDS("output/FC_necela_names.RDS")
necela_val_genes <- readRDS("output/necela_val_genes.RDS")
necela_list_test <- readRDS("output/necela_list_test.RDS")
length(intersect(FC_necela_names$entrezgene_id, backGL$ENTREZID))
hr3trz <- TRZ_list %>% 
  dplyr::filter(time=="3_hours")
hr24trz <- TRZ_list %>% 
  dplyr::filter(time=="24_hours")
intersect(hr24trz$SYMBOL, hr3trz$SYMBOL)



```



Necela's Supplemental 1 contains the column names `r colnames(FC_necela)`. To find the nominally expressed number of genes, I used the PValue column and filtered for all genes with a PValue of \< 0.05. This gave me a list of `r length(necela_list)` genes.\
I then did the same with my data, filtering for the nominal pvalue (P \<0.05) genes and getting a total of `r length(TRZ_list$ENTREZID)`. When I intersected their nominal pvalue genes with mine, I had 3 out of 36 nominally expressed genes in common. The genes that are shared are `r intersect(necela_list$Gene, TRZ_list$SYMBOL)`

## Gene expression of TRZ_3 and TRZ_24 hour nominal genes

using Pvalue\<0.05

These genes are examples of the 36 observed nominally DE genes.

```{r TRZnominal}

# use "Dark2" for individuals, and drug_pal for drug fill
all_cpmcount <-   readRDS("data/cpmcount.RDS") %>% rownames_to_column(var = "ENTREZID")
##pulled all my logcpm, now choosing only TRZ and VEH for plotting

trz_veh_count <- all_cpmcount %>% 
  dplyr::select(ENTREZID, starts_with("Tr"), starts_with("Ve"))


set.seed(12345)
sampset <- readRDS("data/sampsettrz.RDS")
#  <- TRZ_list %>% 
#   distinct(ENTREZID,.keep_all = TRUE) %>% 
#   sample_n(.,6)
for (g in seq(from=1, to=length(sampset$ENTREZID))){
  a <- sampset$SYMBOL[g]
  cpm_TRZ(trz_veh_count,GOI=sampset[g,3],
          ylab=bquote(~italic(.(a))~log[2]~"cpm "), 
          titlename=bquote(~italic(.(a))~"gene "))

}


```

```{r val_genes}

trz_veh_count <- all_cpmcount %>% 
  dplyr::select(ENTREZID, starts_with("Tr"), starts_with("Ve"))
necela_val_genes <- rbind(HER2_gene[1,],necela_val_genes)
set.seed(12345)

#  <- TRZ_list %>% 
#   distinct(ENTREZID,.keep_all = TRUE) %>% 
#   sample_n(.,6)
for (g in seq(from=1, to=length(necela_val_genes$entrezgene_id))){
  a <- necela_val_genes$hgnc_symbol[g]
  cpm_TRZ(trz_veh_count,GOI=necela_val_genes[g,1],
          ylab=bquote(~italic(.(a))~log[2]~"cpm "), 
          titlename=bquote(~italic(.(a))~"gene "))
}

```

### Adding in TRZ and pathway analysis


```{r}


```


### Cell cycle genes

```{r cellcycle,  out.width="50%"}

cell_cycle_list <- readRDS("output/KEGGcellcyclegenes.RDS")

cell_cycle_genes <- cell_cycle_list %>% 
  dplyr::distinct(entrezgene_id,.keep_all = TRUE) %>% 
  dplyr::filter(entrezgene_id %in% trz_veh_count$ENTREZID)

FC_necela_names <- readRDS("data/FC_necela_names.RDS")
necela_cc_shortlist <- FC_necela_names %>% 
  distinct(entrezgene_id,.keep_all = TRUE) %>% 
  dplyr::filter(entrezgene_id %in% cell_cycle_genes$entrezgene_id) %>% 
  # mutate(entrezgene_id_chr=as.charater(entrezgene_id)) %>% 
  full_join(., FC_necela, by=c("hgnc_symbol"="Gene"))



for (g in seq(from=1, to=44)){
  a <- cell_cycle_genes$hgnc_symbol[g]
  cpm_TRZ(trz_veh_count,GOI=cell_cycle_genes[g,1],
          ylab=bquote(~italic(.(a))~log[2]~"cpm "), 
          titlename=bquote(~italic(.(a))~"gene "))

}



```

```{r necela_cellcycle, eval=FALSE, include=FALSE}

df_nc <- necela_cc_shortlist %>% as_tibble() %>% 
        dplyr::select(entrezgene_id,ensembl_gene_id,hgnc_symbol,logCPM) %>% 
  dplyr::filter(entrezgene_id%in% cell_cycle_list$entrezgene_id)
  

# 
#     ggplot2::ggplot(df_nc, aes(x = drug, y = counts)) +
#       geom_bar(position = "identity")) +
#       geom_point(aes(
#         col = indv,
#         size = 1.5,
#         alpha = 0.5
#       )) +
#       guides(alpha = "none", size = "none") +
#       # scale_color_brewer(palette = "Dark2", guide = "none") +
#       scale_fill_manual(values = drug_pal) +
#       facet_wrap("time", nrow = 1, ncol = 2) +
#       theme_bw() +
#       ylab(ylab) +
#       xlab("") +
#       theme(
#         strip.background = element_rect(
#           fill = "white",
#           linetype = 1,
#           linewidth = 0.5
#         ),
#         plot.title = element_text(
#           size = 10,
#           hjust = 0.5,
#           face = "bold"
#         ),
#         axis.title = element_text(size = 10, color = "black"),
#         axis.ticks = element_line(linewidth = 1.0),
#         axis.line = element_line(linewidth = 1.0),
#         axis.text.x = element_blank(),
#         strip.text.x = element_text(margin = margin(2, 0, 2, 0, "pt"), face = "bold")
#       )
#     print(plot)
#   }


necela_cc_genes <- necela_cc_shortlist %>% 
  dplyr::distinct(entrezgene_id,.keep_all = TRUE) %>% 
  dplyr::filter(entrezgene_id %in% cell_cycle_genes$entrezgene_id)

# for (g in seq(from=1, to=44)){
#   a <- cell_cycle_genes$hgnc_symbol[g]
#   cpm_TRZ(necela_cc_genes,GOI=necela_cc_genes[g,1],
#           ylab=bquote(~italic(.(a))~log[2]~"cpm "))+
#     ggtitle(bquote(~italic(.(a))))
# 
# }

backGL %>% 
  dplyr::filter(ENTREZID %in% necela_list$ENTREZID)


```

Just to write down the cell cycle ENTREZID genes from the Kegg Cell cycle pathway

```{r}
Kegg_cellcycle<- c(595,894,896,1019,1021,5925,5933,5934,25,3065,3066,
                   1869,1870,1871,1874,1875,7027,7029,2932,7040,7042,
                   7043,4087,4088,4089,4609,7709,1029,1030,1031,1032,
                   1027,1028,1026,898,9134,1017,6500,8454,9978,6502,
                   890,8900,990,81620,4998,4999,23595,5000,5001,23594,
                   4171,4172,4173,4174,4175,4176,8318, 90381,27085,
                   8317,10926,80174,983, 891,9133,85417,994,995,7534,
                   7529 ,10971,7531,7533,7532,5347,7465 ,494551,9088,
                   902 ,1022,64682,29882,996, 29945,51433,888,51434,
                   8697,10393,51529,246184,25847, 25906,119504,991, 
                   9232,10744,9700,8243,27127,9126,5885,10735,10274,
                   546,1663,25836,23383,55869,114799,157570,23244,23047,
                   23063 ,113130,9212, 151648,5515,5516, 5519, 5518,
                   5526, 5527,5528, 5529,	5525,10403, 57082, 7272,
                   8379,4085,10459,701,9184,699,9319,9587,51343,
                   26271,8555,8556,1387,2033,5591,545,472,4193,7157,
                   1647,4616,10912,5111,2810,1111,11200,993)

```



### GTEX Genes


Here I am intersecting Necela, et al. GTeX V8 log2 cpm heart left ventricle, my expressed genes, and the Knowles, 2019 expressed genes with the KEGG cellcycle gene list

```{r egeneGTeX}
library(readr)

# Heart_Left_Ventricle_v8_egenes <- read_delim("data/Heart_Left_Ventricle.v8.egenes.txt", 
#     delim = "\t", escape_double = FALSE, 
#     trim_ws = TRUE)

# saveRDS(gene_tpm_heart_left_ventricle,"output/GTEXv8_gene_tpm_heart_left_ventricle.RDS")
# saveRDS(gene_median_tpm,"output/GTEXv8_gene_median_tpm.RDS")
# saveRDS(counts_v8_heart_left_ventricle_gct,"output/counts_v8_heart_left_ventricle_gct.RDS")
counts_v8_heart_left_ventricle_gct <- readRDS("output/counts_v8_heart_left_ventricle_gct.RDS")
# getex_ccc<- gene_median_tpm %>% 
#   select(Name, Description,`Heart - Left Ventricle`,`Heart - Atrial Appendage`) %>%
#   # dplyr::filter(grepl('Top*',gene_name))
#   dplyr::filter(Description %in% cell_cycle_list$hgnc_symbol)%>% 
#   mutate(log2tpm= log2(`Heart - Left Ventricle`)) %>% 
#   filter(log2tpm>0)
# intersect(getex_ccc$gene_name, backGL$SYMBOL)

intersect(cell_cycle_list$hgnc_symbol, backGL$SYMBOL)
Knowles_log2cpm <- readRDS("data/Knowles_log2cpm_real.RDS")

Knowles_log2cpm %>% 
  dplyr::filter(ESGN %in%cell_cycle_list$ensembl_gene_id)
FC_necela %>% 
  dplyr::filter(Gene %in% cell_cycle_list$hgnc_symbol)
FC_necela %>% 
  dplyr::filter(PValue<0.05, FC >1.5) %>% 
  dplyr::select(Gene) %>% 
  dplyr::filter(Gene %in% TRZ_list$SYMBOL)

print("147 in necela")

print("146 cc genes in Knowles_log2cpm")
print("150 ccgenes in our data")
gtex_counts_matrix<- counts_v8_heart_left_ventricle_gct %>% 
  column_to_rownames("Name") %>% 
  dplyr::select(!id) %>% 
  dplyr::select(!Description) %>% 
  as.matrix()
  
# cpm <- cpm(gtex_counts_matrix)
lcpm <- cpm(gtex_counts_matrix, log=TRUE)  ### for determining the basic cutoffs
dim(lcpm)

row_means <- rowMeans(lcpm)
gtex_x <- lcpm[row_means > 0,]
dim(gtex_x)
names_gtex_genes <- counts_v8_heart_left_ventricle_gct %>% 
  dplyr::select(Name, Description)
GTEX_heartLV_log2cpm <- gtex_x %>% 
  as.data.frame() %>%
  mutate(Name = rownames(gtex_x)) %>% 
  left_join(.,names_gtex_genes, by =c("Name"))

cellcycle_gtex_expression <- GTEX_heartLV_log2cpm %>% 
  dplyr::filter(Description %in% cell_cycle_list$hgnc_symbol)
length(cellcycle_gtex_expression$Name)


Top2_expression <- lcpm %>% 
  as.data.frame() %>%
  mutate(Name = rownames(lcpm)) %>% 
  left_join(.,names_gtex_genes, by =c("Name")) %>% 
  dplyr::filter(Description == "TOP2A"| Description == "TOP2B")

# saveRDS(Top2_expression,"data/Top2_expression.RDS")

```


```{r Top2expression}
Top2_expression %>% 
  pivot_longer(cols=!c(Name,Description), names_to = "indv", values_to = "counts") %>% 
  ggplot(., aes(x=Description, y=counts))+
  geom_boxplot()+
  ggtitle("GTEX log 2 cpm  of TOP2A and TOP2B")

Top2_expression %>% 
  pivot_longer(cols=!c(Name,Description), names_to = "indv", values_to = "counts") %>% 
  dplyr::filter(Description=="TOP2A") %>% 
  summary()
Top2_expression %>% 
  pivot_longer(cols=!c(Name,Description), names_to = "indv", values_to = "counts") %>% 
  dplyr::filter(Description=="TOP2B") %>% 
  summary()

```





## DOX CRISPR paper

Sapp, V., Aguirre, A., Mainkar, G. et al. Genome-wide CRISPR/Cas9 screening in human iPS derived cardiomyocytes uncovers novel mediators of doxorubicin cardiotoxicity. Sci Rep 11, 13866 (2021). <https://doi.org/10.1038/s41598-021-92988-1>

using this paper, I want to examine the expression of genes they implicate as contributors to Dox-induced cardiotoxicity--- 

I have intersected the GOI gene list which contains all 8 of the Figure 9 GWAS and TWAS genes, with the CRispr genes in supplemental #2.

```{r dOxCrisPr}
interest_genes <- read.csv("output/GOI_genelist.txt", row.names = 1)
crispy_data <- readRDS("output/crisprfoldchange.RDS")

crispy_data %>% 
  dplyr::filter(GeneName %in% interest_genes$hgnc_symbol) %>% 
  
kable(., caption = "List of CRISPR study genes that show up in the CRISPR KO data") %>%
  kable_paper("striped", full_width = FALSE) %>%
  kable_styling(
    full_width = FALSE,
    position = "left"
  ) %>%
  scroll_box(width = "100%", height = "400px")

```




random genes from SETa of Reyes, et al. data in the supplemental lists.

This is the intersection between my 24 hour DOX DEGs and the SETA genes (240 listed) that are said to be upregulated in the Reyes, et al. data.


```{r Reyes}
SETA_analysis_reyes <- readRDS("output/SETA_analysis_reyes.RDS")
toplistall <- readRDS("data/toplistall.RDS")
reyes_setA <- SETA_analysis_reyes %>% 
  pivot_longer(., cols=everything(), names_to="collist", values_to="HGNC") %>% 
  distinct(HGNC,.keep_all = TRUE)
  
DOXtop <- toplistall %>% 
  dplyr::filter(time == "24_hours", id=="DOX", adj.P.Val<0.05)

length(intersect(DOXtop$SYMBOL, reyes_setA$HGNC))

DOXtop_upreg<- toplistall %>% 
  dplyr::filter(time == "24_hours", id=="DOX", adj.P.Val<0.05, logFC<0) 

length(intersect(DOXtop_upreg$SYMBOL, reyes_setA$HGNC))

DOXtop_downreg <- toplistall %>% ##use opp logFC because of how DEGs were processed
  dplyr::filter(time == "24_hours", id=="DOX", adj.P.Val<0.05,logFC>0)
length(intersect(DOXtop_downreg$SYMBOL, reyes_setA$HGNC))

intersect(DOXtop_upreg$SYMBOL, reyes_setA$HGNC)
intersect(DOXtop_downreg$SYMBOL, reyes_setA$HGNC)

```

I found `r length(intersect(DOXtop_upreg$SYMBOL, reyes_setA$HGNC))` up-regulated DEGs in my data set to intersect with Set A.

I found `r length(intersect(DOXtop_downreg$SYMBOL, reyes_setA$HGNC))` down regulated DEGS in my data set that intersected with Set A.  A closer look yielded NPPB to be the only down-regulated DEG that intersected with SetA from the Reyes paper.



## DRC viability at 0 and 48 hours

only two sets of 75-1 and 87-1 available.
```{r DRCviability}

CMD04_75 <- read.csv("data/CMD04_75DRCviability.csv", row.names = 1)
CMD05_75 <- read.csv("data/CMD05_75DRCviability.csv", row.names = 1)
CMD04_87 <- read.csv("data/CMD04_87DRCviability.csv", row.names = 1)
CMD05_87 <- read.csv("data/CMD05_87DRCviability.csv", row.names = 1)


Veh_4_75 <- CMD04_75 %>% 
      separate(sample, into=c("drug","conc"),sep="_") %>% 
  dplyr::filter(drug=="Veh") %>% 
  mutate(percent= if_else(time=="0",RFLU/26609,RFLU/55596.67))%>% 
  group_by(time) %>% 
  summarize(as.tibble(rbind(summary(percent))))
# wilcox.test(percent~time, paired=TRUE, data=Veh_4_75,alternative= "less")

Veh_5_75 <- CMD05_75 %>% 
      separate(sample, into=c("drug","conc"),sep="_") %>% 
  dplyr::filter(drug=="Veh") %>% 
  mutate(percent= if_else(time=="0",RFLU/70958.5,RFLU/32740))%>% 
  group_by(time) %>% 
  summarize(as.tibble(rbind(summary(percent))))
# wilcox.test(percent~time, paired=TRUE, data=Veh_5_75,alternative= "less")


Veh_4_87 <- CMD04_87 %>% 
      separate(sample, into=c("drug","conc"),sep="_") %>% 
  dplyr::filter(drug=="Veh") %>% 
  mutate(percent= if_else(time=="0",RFLU/73528.67,RFLU/72454.83))%>% 
  group_by(time) %>% 
  summarize(as.tibble(rbind(summary(percent))))
# wilcox.test(percent~time, paired=TRUE, data=Veh_4_87,alternative= "less")

Veh_5_87 <- CMD05_87 %>% 
      separate(sample, into=c("drug","conc"),sep="_") %>% 
  dplyr::filter(drug=="Veh") %>% 
  mutate(percent= if_else(time=="0",RFLU/59901.33,RFLU/50557.33)) %>% 
  group_by(time) %>% 
  summarize(as.tibble(rbind(summary(percent))))
# wilcox.test(percent~time, paired=TRUE, data=Veh_5_87,alternative= "less")
Veh_via_table <- Veh_4_75 %>% 
  bind_rows(Veh_5_75,Veh_4_87,Veh_5_87, .id="DRC") %>% 
  dplyr::select(DRC,time,Mean)



```



```{r plotsof DRC}
library(PairedData)
before <- subset(Veh_via_table,  time == "0", Mean,
                 drop = TRUE)
# subset weight data after treatment
after <- subset(Veh_via_table,  time == "48", Mean,
                 drop = TRUE)


pd <- paired(before, after)
plot(pd, type = "profile") + theme_bw()+
  ggtitle("DRC vehicles before and after")

t.test(before,after, paired = TRUE)

# 
# before <- subset(Veh_4_75,  time == "0", percent,
#                  drop = TRUE)
# # subset weight data after treatment
# after <- subset(Veh_4_75,  time == "48", percent,
#                  drop = TRUE)
# # Plot paired data
# library(PairedData)
# pd <- paired(before, after)
# plot(pd, type = "profile") + theme_bw()+
#   ggtitle("Veh_4_75")
# 
# 
# before_1 <- subset(Veh_5_75,  time == "0", percent,
#                  drop = TRUE)
# # subset weight data after treatment
# after_1 <- subset(Veh_5_75,  time == "48", percent,
#                  drop = TRUE)
# # Plot paired data
# 
# pd_1 <- paired(before_1, after_1)
# plot(pd_1, type = "profile") + theme_bw()+
#   ggtitle("Veh_5_75")
# 
# 
# 
# before_2 <- subset(Veh_5_87,  time == "0", percent,
#                  drop = TRUE)
# # subset weight data after treatment
# after_2 <- subset(Veh_5_87,  time == "48", percent,
#                  drop = TRUE)
# # Plot paired data
# 
# pd_2 <- paired(before_2, after_2)
# plot(pd_2, type = "profile") + theme_bw()+
#   ggtitle("Veh_5_87")
# 
# 
# 
# before_3 <- subset(Veh_4_87,  time == "0", percent,
#                  drop = TRUE)
# # subset weight data after treatment
# after_3 <- subset(Veh_4_87,  time == "48", percent,
#                  drop = TRUE)
# # Plot paired data
# 
# pd_3 <- paired(before_3, after_3)
# plot(pd_3, type = "profile") + theme_bw()+
#   ggtitle("Veh_4_87")
# 






```

### SNP from another GWAS

paper ref: [https://aacrjournals.org/clincancerres/article/23/1/43/122971/Genome-Wide-Association-Study-for-Anthracycline]




```{r  SnPupload}
library(readr)
# near_genes_SNP1 <- read_table("~/Ward Lab/Cardiotoxicity/Manuscript/SNP_work/near_genes_SNP1.txt",
#     col_types = cols(CLIN_SIG = col_skip()))
# list_snp_IDS <- readRDS("output/SNP_list_ID.RDS")##list to assign ensemble/hgnc/entrezid
RSID_QTL_list_full <- read.csv("output/RSID_QTL_list_full.txt", row.names = 1)##closest annotated gene to complete rsid list

SNP_supp <- readRDS("output/SNP_supp.RDS")## has the pvalue
near_genes_SNP1 <- readRDS("output/near_genes_SNP1.RDS")
SNP_gene_list <- near_genes_SNP1 %>% 
  rename("RSID"=1) %>% 
  dplyr::select(RSID,SYMBOL,Gene) %>%
  distinct(RSID,.keep_all = TRUE) %>% 
  left_join(.,SNP_supp,by="RSID") %>% 
  dplyr::select(RSID, SYMBOL,Gene,Ch,BP,OR,Pvalue) #
  # distinct(Gene,.keep_all = TRUE)# %>%
  # dplyr::select(SYMBOL,Gene) %>%

full_SNP_list <- near_genes_SNP1 %>% 
  rename("RSID"=1) %>% 
  distinct(RSID,.keep_all = TRUE) 

RSID_QTL_list_full %>% 
  full_join(., (SNP_supp %>% dplyr::select(RSID,Pvalue)))

# write.csv(full_SNP_list,"output/SNP_list_full.txt")

# ensembl <- useMart(biomart="ensembl",dataset ="hsapiens_gene_ensembl")
# 
# 

# length(unique(RSID_QTL_list_full$Gene))
# 

#   list_snp_IDS <- getBM(attributes = my_attributes,
#                      filters ="ensembl_gene_id",
#                      values =RSID_QTL_list_full$Gene,
#                      mart = ensembl)
# saveRDS(list_snp_IDS,"output/SNP_list_ID.RDS")
```


old tSS method...
```{r retrieving TSSstartstop and IDs, eval=FALSE, include=TRUE}
# genes = getBM(c( "chromosome_name","start_position","end_position","strand","entrezgene_id","ensembl_gene_id"), filters =
# c("chromosome_name","with_entrezgene"),
# values=list(c(1:22,"X","Y"),TRUE), mart=ensembl)
# saveRDS(genes,"output/genes.RDS")
genes <- readRDS("output/genes.RDS")
genes %>% 
  dplyr::select("chromosome_name","start_position","end_position","entrezgene_id","ensembl_gene_id") %>%  
  
  mutate(chromosome_name=paste("chr",chromosome_name,sep="")) %>% 
  
  write_tsv(.,"~/ATAC_downloads/BEDtoolplay/biomart_genes.bed",col_names = FALSE)
  


```


```{r bedtools output}

##making a newfile## used bedtools to map closest transcript (ENST)
# schneider_closest_output <- read_delim("~/ATAC_downloads/BEDtoolplay/output.bed",
#     delim = "\t", escape_double = FALSE,
#    col_names = c("chrom","start","stop","RSID","Pvalue","chrom_gene","start_gene","end_gene","ENTREZID","ensembl_gene_id"), trim_ws = TRUE)
# saveRDS(schneider_closest_output,"data/schneider_closest_output.RDS")

schneider_closest_output <- readRDS("data/schneider_closest_output.RDS")
##below was from the first alignment attempt then I retrieved TSS from biomart and redid the mapping in bedtools
# listoftranscripts <- schneider_closest_output %>% 
#   dplyr::select(chrom:transcript) %>% 
#   distinct(RSID,.keep_all = TRUE) %>% 
#   # dplyr::select(transcript) %>% 
#   separate(transcript, into=c("transcript", "version"))
# 
# 
# my_filter <- c("ensembl_transcript_id")
# # test_biomart_run <- getBM(attributes=my_attributes, 
# #       filters = "ensembl_transcript_id", 
# #       values = listoftranscripts$transcript, 
# #       mart = ensembl)
# 
# test_biomart_run <- readRDS("output/test_biomart_run.RDS")
# 


# SNP_bedtools_data <- test_biomart_run %>% 
#   distinct(ensembl_gene_id,.keep_all = TRUE) %>% 
#   left_join(., (listoftranscripts %>% dplyr::select(transcript,RSID,chrom,start,stop,Pvalue)), by=c("ensembl_transcript_id" = "transcript")) %>% 
#   arrange(., ensembl_transcript_id, Pvalue) %>% 
#   mutate(gene_ID=coalesce(as.character(entrezgene_id),ensembl_gene_id)) 

just_the_SNPs <- schneider_closest_output %>% 
  distinct(.,ensembl_gene_id,.keep_all = TRUE) %>% 
  arrange(., Pvalue)

backGL <- read.csv("data/backGL.txt", row.names = 1)

# backGL <- getBM(attributes=my_attributes, 
#       filters = "entrezgene_id", 
#       values = backGL$ENTREZID, 
#       mart = ensembl)
# backGL2 <- backGL %>% 
#   dplyr::select(entrezgene_id:hgnc_symbol) %>% 
#   distinct(ensembl_gene_id,.keep_all = TRUE)
# backGL <- backGL %>% 
#   left_join(.,backGL2, by=c("ENTREZID"="entrezgene_id")) %>% 
#   distinct(ENTREZID,.keep_all = TRUE)
# write.csv(backGL,"data/backGL.txt")

length(intersect(just_the_SNPs$ensembl_gene_id,backGL$ensembl_gene_id))
length(intersect(just_the_SNPs$ENTREZID,backGL$ENTREZID))
# length(intersect(egene_SNP$ensembl_gene_id,backGL$ensembl_gene_id))
#      length(intersect(egene_SNP$entrezgene_id,backGL$ENTREZID))     
```


```{r graph with my data}
order_frame <- just_the_SNPs %>% 
  dplyr::filter(ENTREZID %in% backGL$ENTREZID) %>% 
  left_join(., (backGL %>% dplyr::select(ENTREZID,ensembl_gene_id,SYMBOL,hgnc_symbol)), by=c("ENTREZID"="ENTREZID","ensembl_gene_id"="ensembl_gene_id")) %>% 
  mutate(SYMBOL= if_else(is.na(SYMBOL),"NDN",SYMBOL))
##had to add NDN name to missing ENTREZID


new_Gwas_genes_sig <- FC_mine %>% 
  dplyr::filter(ENTREZID %in% just_the_SNPs$ENTREZID) %>% 
  mutate(drug=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>%
 dplyr::filter(time =="24_hours") %>% 
  dplyr::select(ENTREZID , id,logFC, adj.P.Val, SYMBOL) %>%
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from =adj.P.Val) %>% 
  column_to_rownames(var="id") %>%
  dplyr::select(order_frame$SYMBOL) %>% 
  as.matrix()

  
new_gwas_mat <-  FC_mine %>% 
  dplyr::filter(time=="24_hours") %>% 
  mutate(logFC= logFC*(-1)) %>%
  dplyr::filter(ENTREZID %in% just_the_SNPs$ENTREZID) %>% 
  dplyr::select(SYMBOL ,time, id, logFC) %>% 
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from = logFC) %>% 
  column_to_rownames(var="id") %>%
   dplyr::select(order_frame$SYMBOL) %>% 
  as.matrix()

ComplexHeatmap::Heatmap(new_gwas_mat, name = "Fold change\nvalues", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        # col = col_funFC,
        row_order = c('DOX','EPI','DNR', 'MTX','TRZ'),
        column_title = "Schneider GWAS SNP-closest gene FC", 
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_names_rot = 90, 
        column_names_gp = gpar(fontsize = 8,fontface="italic"),
        column_names_centered =TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(new_Gwas_genes_sig[i, j] <0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})
```


```{r heatmaplog10pvalue with my data}


framefun24 <- readRDS("data/framefun24.RDS")


logFtestSNP_mat <- framefun24 %>% 
  mutate(DOX_plog=(log10(DOX.VEH.24)*(-1)),
         EPI_plog=(log10(EPI.VEH.24)*(-1)),
         DNR_plog=(log10(DNR.VEH.24)*(-1)),
         MTX_plog=(log10(MTX.VEH.24)*(-1)),
         TRZ_plog=(log10(TRZ.VEH.24)*(-1)),) %>% 
  dplyr::select(ENTREZID,DOX_plog,EPI_plog, DNR_plog, MTX_plog,TRZ_plog) %>% 
  dplyr::filter(ENTREZID %in% just_the_SNPs$ENTREZID) %>% 
  pivot_longer(cols = !ENTREZID, names_to = "treatment", values_to = "plog") %>% 
  separate(treatment, into=c("treatment",NA)) %>% 
  mutate(ENTREZID=as.numeric(ENTREZID)) %>% 
  left_join(., (backGL)) %>% 
  pivot_wider(id_cols=treatment, 
              names_from = SYMBOL, 
              values_from = plog) %>% 
  column_to_rownames(var="treatment") %>%
   dplyr::select(order_frame$SYMBOL) %>% 
  as.matrix()


Heatmap(logFtestSNP_mat, name = "-Log10 \n p value", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        # col = col_funFC,
        row_order = c('DOX','EPI','DNR', 'MTX', 'TRZ'),
        column_title = "-log10 Ftestpvalue SNP nearest-genes", 
        column_title_side = "top",
        # column_order= c("LRRTM4","CLYBL",	"ADCY2","NEDD4L",	"PLEKHA5"	,	"PAG1",	"PPARA","PRODH", "CALD1","PAM",  "CDPF1",	"DUBR","PRR5L","RSU1",	"WDHD1"	,	"PRDX6","MSI1",	"SIRT4","GAREM1",	"ADAMTSL1","DENND3"	, "ESD"	,	"TRIB2"	,"KTN1","NFE2L3",	"SULF1"), 
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_names_rot = 90, 
        column_names_gp = gpar(fontsize = 8,fontface="italic"),
        column_names_centered =TRUE,
        cell_fun = function(j, i, x, y, width, height, fill) {
        if(logFtestSNP_mat[i, j] > log10(0.05)*(-1))
            grid.text("*", x, y, gp = gpar(fontsize = 16))
})
  
```


The new set is in order of  the pvalue on GWAS genes in common.
```{r  egenes of new GWAS}
# saveRDS(SNP_egenes_allfiles,"output/SNP_egenes_allfiles.RDS")

SNP_egenes_allfiles <- readRDS("output/SNP_egenes_allfiles.RDS")
# SNP_egenes_allfiles <- allfiles %>% 
#   rename("whoops"="X1","SYMBOL" = "X2", "Variant_ID"="X3", "RSID"="X4", "P_value"="X5", "NES"="X6", "Tissue"="X7") %>% 
#   separate(whoops, into = c("filenameref",NA,NA,NA,"ensembl_gene_id"))

egene_SNP <- SNP_egenes_allfiles %>% 
  distinct(ensembl_gene_id,RSID,.keep_all = TRUE) 


new_egenes_sig <- FC_mine %>% 
  dplyr::filter(ENTREZID %in% egene_SNP$entrezgene_id) %>% 
  mutate(drug=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>%
 dplyr::filter(time =="24_hours") %>% 
  dplyr::select(ENTREZID , id,logFC, adj.P.Val, SYMBOL) %>%
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from =adj.P.Val) %>% 
  column_to_rownames(var="id") %>%
  as.matrix()

  
new_egenes_mat <-  FC_mine %>% 
  dplyr::filter(time=="24_hours") %>% 
  mutate(logFC= logFC*(-1)) %>%
  dplyr::filter(ENTREZID %in% egene_SNP$entrezgene_id) %>% 
  dplyr::select(SYMBOL ,time, id, logFC) %>% 
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from = logFC) %>% 
  column_to_rownames(var="id") %>%
  as.matrix()

ComplexHeatmap::Heatmap(new_egenes_mat, name = "Fold change\nvalues", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        # col = col_funFC,
        row_order = c('DOX','EPI','DNR', 'MTX','TRZ'),
        column_title = "Fold change values Schneider/GTEx eGenes", 
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_order = sort(colnames(new_egenes_mat)),
        column_names_rot = 90, 
        column_names_gp = gpar(fontsize = 8,fontface="italic"),
        column_names_centered =TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(new_egenes_sig[i, j] <0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})
```
### egene log10 pvalue Ftest

```{r  log10pvalue of ftest egenes}


LogPvaluelist <- SNP_egenes_allfiles %>% 
  distinct(ensembl_gene_id,.keep_all = TRUE) 

logFtestegene_mat <- framefun24 %>% 
  mutate(DOX_plog=(log10(DOX.VEH.24)*(-1)),
         EPI_plog=(log10(EPI.VEH.24)*(-1)),
         DNR_plog=(log10(DNR.VEH.24)*(-1)),
         MTX_plog=(log10(MTX.VEH.24)*(-1)),
         TRZ_plog=(log10(TRZ.VEH.24)*(-1)),) %>% 
  dplyr::select(ENTREZID,DOX_plog,EPI_plog, DNR_plog, MTX_plog,TRZ_plog) %>% 
  dplyr::filter(ENTREZID %in% LogPvaluelist$entrezgene_id) %>% 
  pivot_longer(cols = !ENTREZID, names_to = "treatment", values_to = "plog") %>% 
  separate(treatment, into=c("treatment",NA)) %>% 
  mutate(ENTREZID=as.numeric(ENTREZID)) %>% 
  left_join(., (backGL)) %>% 
  pivot_wider(id_cols=treatment, 
              names_from = SYMBOL, 
              values_from = plog) %>% 
  column_to_rownames(var="treatment") %>%
   # dplyr::select(order_frame$SYMBOL) %>% 
  as.matrix()

Heatmap(logFtestegene_mat, name = "-Log10 \n p value", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        # col = col_funFC,
        row_order = c('DOX','EPI','DNR', 'MTX', 'TRZ'),
        column_title = "-log10 Ftestpvalue Schneider/GTEx eGenes", 
        column_title_side = "top",
         column_order = sort(colnames(new_egenes_mat)),
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_names_rot = 90, 
        column_names_gp = gpar(fontsize = 8,fontface="italic"),
        column_names_centered =TRUE,
        cell_fun = function(j, i, x, y, width, height, fill) {
        if(logFtestegene_mat[i, j] > log10(0.05)*(-1))
            grid.text("*", x, y, gp = gpar(fontsize = 16))
})


```


```{r making Schneider_completeSNP_egenes,  include=TRUE,eval=FALSE}

SNP_supp <- readRDS("output/SNP_supp.RDS")
## has the pvalue (this is the supplement from that paper)
# alternate file is the schneider_closest_output for paper
egene_pvalue_SNP <- SNP_egenes_allfiles %>% 
  distinct(ensembl_gene_id,RSID,.keep_all = TRUE) %>% 
  left_join(., (SNP_supp %>% 
                  dplyr::select(RSID,Pvalue)),by = join_by(RSID))%>% 
  dplyr::select(SYMBOL, ensembl_gene_id, entrezgene_id,Pvalue, RSID)
# contains the eGenes to every SNP with pvalue of associated RSID




SNPXgenes <- schneider_closest_output %>% 
# dplyr::filter(ENTREZID %in% backGL$ENTREZID) %>% 
  left_join(., (backGL %>% dplyr::select(ENTREZID,ensembl_gene_id,SYMBOL,hgnc_symbol)), by=c("ENTREZID"="ENTREZID","ensembl_gene_id"="ensembl_gene_id"))%>% 
  mutate(SYMBOL= if_else(is.na(SYMBOL)& ENTREZID==4692,"NDN",SYMBOL)) %>% 
  dplyr::select(SYMBOL, ensembl_gene_id, ENTREZID,Pvalue, RSID) #%>% 
  # dplyr::rename("ENTREZID"="entrezgene_id" )

complete_list <- bind_rows(list(SNP_genes=SNPXgenes,egene_SNP=egene_pvalue_SNP),.id ="id" ) 
complete_list <- complete_list%>% 
  mutate(entrezgeneid= if_else(is.na(entrezgene_id),ENTREZID,entrezgene_id)) %>% 
  dplyr::select(!entrezgene_id) %>% 
  rename("entrezgene_id"="entrezgeneid") %>% 
  dplyr::select(!ENTREZID)
 
# write.csv(complete_list,"data/Schneider_GWAS.txt")
# write.csv(egene_SNP, "output/expressed_egenes_by_RSID.csv")

```


```{r Schneider_orderHeatmap}

Schneider_GWAS <- read.csv("data/Schneider_GWAS.txt", row.names = 1)

SGWAS_sig_order <- Schneider_GWAS %>% 
  dplyr::filter(entrezgene_id %in%backGL$ENTREZID) %>% 
  arrange(Pvalue) %>% 
  dplyr::select(!SYMBOL) %>% 
  left_join(.,backGL, by = c("entrezgene_id"= "ENTREZID"))
  # dplyr::select(SYMBOL) #%>% 
  # distinct()
SGWAS_sig <- FC_mine %>% 
  dplyr::filter(ENTREZID %in% Schneider_GWAS$entrezgene_id) %>% 
  mutate(drug=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>%
 dplyr::filter(time =="24_hours") %>% 
  dplyr::select(ENTREZID , id,logFC, adj.P.Val, SYMBOL) %>%
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from =adj.P.Val) %>% 
  column_to_rownames(var="id") %>%
  dplyr::select(SGWAS_sig_order$SYMBOL) %>% 
  as.matrix()

SGWAS_mat <-  FC_mine %>% 
  dplyr::filter(time=="24_hours") %>% 
  mutate(logFC= logFC*(-1)) %>%
  dplyr::filter(ENTREZID %in% Schneider_GWAS$entrezgene_id) %>% 
  dplyr::select(SYMBOL ,time, id, logFC) %>% 
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from = logFC) %>% 
  column_to_rownames(var="id") %>%
  dplyr::select(SGWAS_sig_order$SYMBOL) %>% 
  as.matrix()



ComplexHeatmap::Heatmap(SGWAS_mat, name = "Fold change\nvalues", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        # col = col_funFC,
        row_order = c('DOX','EPI','DNR', 'MTX','TRZ'),
        column_title = "Full Schneider SNP-genes and eGenes n = 91",
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        # column_order = 
            column_names_rot = 90, 
        column_names_gp = gpar(fontsize = 8,fontface="italic"),
        column_names_centered =TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(SGWAS_sig[i, j] <0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 16))
})


framefun24 <- readRDS("data/framefun24.RDS")


logFtestSGWAS <- framefun24 %>% 
  mutate(DOX_plog=(log10(DOX.VEH.24)*(-1)),
         EPI_plog=(log10(EPI.VEH.24)*(-1)),
         DNR_plog=(log10(DNR.VEH.24)*(-1)),
         MTX_plog=(log10(MTX.VEH.24)*(-1)),
         TRZ_plog=(log10(TRZ.VEH.24)*(-1)),) %>% 
  dplyr::select(ENTREZID,DOX_plog,EPI_plog, DNR_plog, MTX_plog,TRZ_plog) %>% 
  dplyr::filter(ENTREZID %in% SGWAS_sig_order$entrezgene_id) %>% 
  pivot_longer(cols = !ENTREZID, names_to = "treatment", values_to = "plog") %>% 
  separate(treatment, into=c("treatment",NA)) %>% 
  mutate(ENTREZID=as.numeric(ENTREZID)) %>% 
  left_join(., (backGL)) %>% 
  pivot_wider(id_cols=treatment, 
              names_from = SYMBOL, 
              values_from = plog) %>% 
  column_to_rownames(var="treatment") %>%
  dplyr::select(SGWAS_sig_order$SYMBOL) %>% 
  as.matrix()


ComplexHeatmap::Heatmap(logFtestSGWAS, name = "-log10\np value", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        # col = col_funFC,
        row_order = c('DOX','EPI','DNR', 'MTX','TRZ'),
        column_title = " Variance Full Schneider SNP-genes and eGenes",
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        # column_order = 
            column_names_rot = 90, 
        column_names_gp = gpar(fontsize = 8,fontface="italic"),
        column_names_centered =TRUE,
        cell_fun = function(j, i, x, y, width, height, fill) {
        if(logFtestSGWAS[i, j] > log10(0.05)*(-1))
            grid.text("*", x, y, gp = gpar(fontsize = 16))
})

```
###  Top 50 SNP and eGenes from Schneider GWAS


```{r top50 p value}


schneider_closest_output <- readRDS("data/schneider_closest_output.RDS")
# egenes_full <- read.csv("output/expressed_egenes_by_RSID.csv",row.names = 1)
Schneider_GWAS <- read.csv("data/Schneider_GWAS.txt", row.names = 1)
top50RSID <- schneider_closest_output %>% 
  arrange(., Pvalue) %>% 
  slice_min(.,Pvalue,n=50) %>% 
 dplyr::select(RSID)
# 
# top40RSID <- schneider_closest_output %>% 
#   arrange(., Pvalue) %>% 
#   slice_min(.,Pvalue,n=40) %>% 
#  dplyr::select(RSID)
# 
# top30RSID <- schneider_closest_output %>% 
#   arrange(., Pvalue) %>% 
#   slice_min(.,Pvalue,n=30) %>% 
#  dplyr::select(RSID)
# 
# top25RSID <- schneider_closest_output %>% 
#   arrange(., Pvalue) %>% 
#   slice_min(.,Pvalue,n=25) %>% 
#  dplyr::select(RSID)

SNP_genestop50 <- Schneider_GWAS %>% 
  dplyr::filter(RSID %in% top50RSID$RSID) %>% 
  distinct(entrezgene_id,.keep_all = TRUE) 
    
  # left_join(., (Schneider_GWAS %>%
  #                 dplyr::select(ensembl_gene_id,entrezgene_id,SYMBOL)), 
  #           by = c("ensembl_gene_id"="ensembl_gene_id","ENTREZID"="entrezgene_id"))  %>% 
# mutate(SYMBOL= if_else(is.na(SYMBOL)& ENTREZID =="4692","NDN",SYMBOL))
 

 # write.csv(SNP_genestop50,"output/TOP_50SNPreffile.csv")
# egenes_top50SNP <- SNP_egenes_allfiles%>% 
#   dplyr::filter(RSID %in%SNP_genestop50$RSID) %>% 
#   distinct(ensembl_gene_id,.keep_all = TRUE) %>% 
#   left_join(., (SNP_supp %>% dplyr::select(RSID,Pvalue)),by = join_by(RSID)) %>% dplyr::select(SYMBOL, ensembl_gene_id, entrezgene_id,Pvalue)
###24 separate egenes (pvalue is associated with the RSID and that gene)

# top50full <- bind_rows(list(SNP_genes=SNP_genestop50,egene_SNP=egenes_top50SNP),.id ="id" )
# max(SNP_genestop50$Pvalue)
# 
# completetop50 <- Schneider_GWAS %>% 
#   dplyr::filter(id=="SNP_genes"& 
#                   Pvalue <= max(SNP_genestop50$Pvalue) | id=="egene_SNP")
#  

SGWAS_top50_order <- SNP_genestop50 %>% 
  dplyr::filter(entrezgene_id %in% backGL$ENTREZID) %>% 
  arrange(Pvalue) %>% 
  dplyr::select(!SYMBOL) %>% 
  left_join(.,backGL, by = c("entrezgene_id"= "ENTREZID"))
  
# write.csv(SGWAS_top50_order,"output/SGWAS_top50_order.csv")

SGWAS50_sig <- FC_mine %>% 
  dplyr::filter(ENTREZID %in% SNP_genestop50$entrezgene_id) %>% 
  mutate(drug=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>%
 dplyr::filter(time =="24_hours") %>% 
  dplyr::select(ENTREZID , id,logFC, adj.P.Val, SYMBOL) %>%
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from =adj.P.Val) %>% 
  column_to_rownames(var="id") %>%
  dplyr::select(SGWAS_top50_order$SYMBOL) %>%
  as.matrix()

SGWAS50_mat <-  FC_mine %>% 
  dplyr::filter(time=="24_hours") %>% 
  mutate(logFC= logFC*(-1)) %>%
  dplyr::filter(ENTREZID %in% SNP_genestop50$entrezgene_id) %>% 
  dplyr::select(SYMBOL ,time, id, logFC) %>% 
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from = logFC) %>% 
  column_to_rownames(var="id") %>%
  dplyr::select(SGWAS_top50_order$SYMBOL) %>% 
  as.matrix()

ComplexHeatmap::Heatmap(SGWAS50_mat, name = "Fold change\nvalues", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        # col = col_funFC,
        row_order = c('DOX','EPI','DNR', 'MTX','TRZ'),
        column_title = "Top50 SNPs from Schneider",
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        # column_order = 
            column_names_rot = 90, 
        column_names_gp = gpar(fontsize = 8,fontface="italic"),
        column_names_centered =TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(SGWAS50_sig[i, j] <0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 16))
})


framefun24 <- readRDS("data/framefun24.RDS")


logFtest50SGWAS <- framefun24 %>% 
  mutate(DOX_plog=(log10(DOX.VEH.24)*(-1)),
         EPI_plog=(log10(EPI.VEH.24)*(-1)),
         DNR_plog=(log10(DNR.VEH.24)*(-1)),
         MTX_plog=(log10(MTX.VEH.24)*(-1)),
         TRZ_plog=(log10(TRZ.VEH.24)*(-1)),) %>% 
  dplyr::select(ENTREZID,DOX_plog,EPI_plog, DNR_plog, MTX_plog,TRZ_plog) %>% 
  dplyr::filter(ENTREZID %in% SGWAS_top50_order$entrezgene_id) %>% 
  pivot_longer(cols = !ENTREZID, 
               names_to = "treatment", values_to = "plog") %>% 
  separate(treatment, into=c("treatment",NA)) %>% 
  mutate(ENTREZID=as.numeric(ENTREZID)) %>% 
  left_join(., (backGL)) %>% 
  pivot_wider(id_cols=treatment, 
              names_from = SYMBOL, 
              values_from = plog) %>% 
  column_to_rownames(var="treatment") %>%
  dplyr::select(SGWAS_top50_order$SYMBOL) %>% 
  as.matrix()


ComplexHeatmap::Heatmap(logFtest50SGWAS, name = "-log10\np value", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        # col = col_funFC,
        row_order = c('DOX','EPI','DNR', 'MTX','TRZ'),
        column_title = " Top50 SNPs and Variance in Schneider GWAS",
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        # column_order = 
            column_names_rot = 90, 
        column_names_gp = gpar(fontsize = 8,fontface="italic"),
        column_names_centered =TRUE,
        cell_fun = function(j, i, x, y, width, height, fill) {
        if(logFtestSGWAS[i, j] > log10(0.05)*(-1))
            grid.text("*", x, y, gp = gpar(fontsize = 16))
})

```


### Knowles troponin genes
```{r importKnowles, fig.width=10}
Knowles_supp13 <- read.csv("output/Knowles_supp13.csv", row.names = 1)

Knowles_S13 <- read.csv("output/Knowles_S13.csv", row.names = 1)
Knowles_supp13 %>% 
  left_join(., Knowles_S13, by=c("gene"="ensembl_gene_id", "hugo"="hgnc_symbol"))


# my_attributes <- c('entrezgene_id', 'ensembl_gene_id', 'hgnc_symbol')
# expre7k <- getBM(attributes = my_attributes,
                     # filters ="ensembl_gene_id",
                     # values=top7K$gene,
                     # mart = ensembl)

# write.csv(expre7k,"output/expre7k.csv")

Knowles_13trop_sig <- FC_mine %>% 
  dplyr::filter(ENTREZID %in% Knowles_S13$entrezgene_id) %>% 
  mutate(drug=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>%
 dplyr::filter(time =="24_hours") %>% 
  dplyr::select(ENTREZID , id,logFC, adj.P.Val, SYMBOL) %>%
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from =adj.P.Val) %>% 
  column_to_rownames(var="id") %>%
  as.matrix()

  
Knowles_13trop_mat <-  FC_mine %>% 
  dplyr::filter(time=="24_hours") %>% 
  mutate(logFC= logFC*(-1)) %>%
  dplyr::filter(ENTREZID %in% Knowles_S13$entrezgene_id) %>% 
  dplyr::select(SYMBOL ,time, id, logFC) %>% 
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from = logFC) %>% 
  column_to_rownames(var="id") %>%
  as.matrix()

ComplexHeatmap::Heatmap(Knowles_13trop_mat, name = "Fold change\nvalues", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        # col = col_funFC,
        row_order = c('DOX','EPI','DNR', 'MTX','TRZ'),
        column_title = "Fold change of all Knowles-Troponin variable genes in my data", 
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_order = sort(colnames(Knowles_13trop_mat)),
            column_names_rot = 90, 
        column_names_gp = gpar(fontsize = 8,fontface="italic"),
        column_names_centered =TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(Knowles_13trop_sig[i, j] <0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 16))
})

# intersect(Knowles_S13$entrezgene_id, EPI508_list$ENTREZID)
```


```{r top troponin7, fig.width=10}
###Top 7:
top7kgene <- read.csv("output/expre7k.csv", row.names = 1)
# top7K <- elife_33480_supp14 %>% 
#   dplyr::filter(conc=="0.625") #%>% 
#   select(gene)
trop7_sig <- FC_mine %>% 
  dplyr::filter(ENTREZID %in%top7kgene$entrezgene_id) %>% 
  mutate(drug=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>%
 dplyr::filter(time =="24_hours") %>% 
  dplyr::select(ENTREZID , id,logFC, adj.P.Val, SYMBOL) %>%
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from =adj.P.Val) %>% 
  column_to_rownames(var="id") %>%
  as.matrix()

  
trop7_mat <-  FC_mine %>% 
  dplyr::filter(time=="24_hours") %>% 
  mutate(logFC= logFC*(-1)) %>%
  dplyr::filter(ENTREZID %in% top7kgene$entrezgene_id) %>% 
  dplyr::select(SYMBOL ,time, id, logFC) %>% 
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from = logFC) %>% 
  column_to_rownames(var="id") %>%
  as.matrix()

ComplexHeatmap::Heatmap(trop7_mat, name = "Fold change\nvalues", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        # col = col_funFC,
        row_order = c('DOX','EPI','DNR', 'MTX','TRZ'),
        column_title = "Fold change 0.625 top 7 Knowles-Troponin genes in my data", 
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        # column_order= c('RARG',
        #                 'TNS2', 
        #                 'ZNF740',
        #                 'SLC28A3',
        #                 'RMI1',
        #                 'EEF1B2',
        #                 'FRS2', 
        #                 'HDDC2'),
        # bottom_annotation = ht,
            column_names_rot = 90, 
        column_names_gp = gpar(fontsize = 8,fontface="italic"),
        column_names_centered =TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(trop7_sig[i, j] <0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 16))
})



```



```{r qvalueheatmap}
my_qval_list24 <- readRDS("data/qval24hr.RDS") 
# list2env(my_qval_list24, envir = .GlobalEnv)
# organized_drugframe <- read.csv("data/organized_drugframe.csv",
#                                 row.names = 1)
# 
# Var_test_list24 <- readRDS("data/Var_test_list24.RDS")
# set24 <- names(Var_test_list24) 
# 
# framefun24 <- data.frame(ENTREZID=rownames(organized_drugframe))
# for (i in 1:length(set24)){
#   a <- set24[i]
#     s_list<- Var_test_list24[[i]][[13]]
#   names(s_list) <- rownames(organized_drugframe)
#   hold<- map_df(s_list, ~as.data.frame(.x$p.value), .id="ENTREZID")
#    framefun24[paste0(a)] <- hold$`.x$p.value`
#  
# }
# saveRDS(framefun24, "data/framefun24.RDS")

framefun24 <- readRDS("data/framefun24.RDS")


logFtestval_mat <- framefun24 %>% 
  mutate(DOX_plog=(log10(DOX.VEH.24)*(-1)),
         EPI_plog=(log10(EPI.VEH.24)*(-1)),
         DNR_plog=(log10(DNR.VEH.24)*(-1)),
         MTX_plog=(log10(MTX.VEH.24)*(-1)),
         TRZ_plog=(log10(TRZ.VEH.24)*(-1)),) %>% 
  dplyr::select(ENTREZID,DOX_plog,EPI_plog, DNR_plog, MTX_plog,TRZ_plog) %>% 
  dplyr::filter(ENTREZID %in% egene_SNP$entrezgene_id) %>% 
  pivot_longer(cols = !ENTREZID, names_to = "treatment", values_to = "plog") %>% 
  separate(treatment, into=c("treatment",NA)) %>% 
  mutate(ENTREZID=as.numeric(ENTREZID)) %>% 
  left_join(., (backGL)) %>% 
  pivot_wider(id_cols=treatment, 
              names_from = SYMBOL, 
              values_from = plog) %>% 
  column_to_rownames(var="treatment") %>%
  as.matrix()


Heatmap(logFtestval_mat, name = "-Log10 \n p value", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        # col = col_funFC,
        row_order = c('DOX','EPI','DNR', 'MTX', 'TRZ'),
        column_title = "-log10 Ftestpvalue eGenes", 
        column_title_side = "top",
        column_order = sort(colnames(logFtestval_mat)),
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_names_rot = 90, 
        column_names_gp = gpar(fontsize = 8,fontface="italic"),
        column_names_centered =TRUE,
        cell_fun = function(j, i, x, y, width, height, fill) {
        if(logFtestval_mat[i, j] > log10(0.05)*(-1))
            grid.text("*", x, y, gp = gpar(fontsize = 16))
})
  
```
```{r Cardiotoxval}

Cardiotox_table <- readRDS("output/Cardiotox_dif_values.RDS")
EEF1B2_table <- readRDS("output/EEF1B2_dif_values.RDS")
Cardiotox_table %>% 
  ggplot(., aes(x=tox, y=cpm))+
  geom_point(aes(color=indv))+
  geom_smooth(method="lm")+
  # stat_regline_equation(label.x.npc = "center")+
  facet_wrap(~drug, scales = "free_x")+
  labs(title="RARG ", y = "delta cpm", x = "delta tox")+
  scale_color_brewer(palette = "Dark2",guide="legend",name = "Individual",labels(c(1,2,3,4,5,6)))+
  theme_bw()+
  ggpubr:: stat_cor(method="pearson",
                       # cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "black",
               # label.x.npc = 0.01,
               # label.y.npc=0.01, 
               size = 3)+
ggpubr:: stat_cor(method="spearman",
                       cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "red",
               label.x.npc = 0.01,
               label.y.npc=0.01,
               size = 3)

EEF1B2_table %>% 
ggplot(., aes(x=tox, y=cpm))+
  geom_point(aes(color=indv))+
  geom_smooth(method="lm")+
  # stat_regline_equation(label.x.npc = "center")+
  facet_wrap(~drug, scales = "free_x")+
  labs(title="EEF1B2 ", y = "delta cpm", x = "delta tox")+
  scale_color_brewer(palette = "Dark2",guide="legend",name = "Individual",labels(c(1,2,3,4,5,6)))+
  theme_bw()+
  ggpubr:: stat_cor(method="pearson",
                       # cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "black",
               # label.x.npc = 0.01,
               # label.y.npc=0.01, 
               size = 3)+
ggpubr:: stat_cor(method="spearman",
                       cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "red",
               label.x.npc = 0.01,
               label.y.npc=0.01,
               size = 3)
```

### FRS2
```{r FRS2}
FRS2_table <- readRDS("output/FRS2_dif_values.RDS")

FRS2_table %>% 
ggplot(., aes(x=tox, y=cpm))+
  geom_point(aes(color=indv))+
  geom_smooth(method="lm")+
  # stat_regline_equation(label.x.npc = "center")+
  facet_wrap(~drug, scales = "free_x")+
  labs(title="FRS2 ", y = "delta cpm", x = "delta tox")+
  scale_color_brewer(palette = "Dark2",guide="legend",name = "Individual",labels(c(1,2,3,4,5,6)))+
  theme_bw()+
  ggpubr:: stat_cor(method="pearson",
                       # cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "black",
               # label.x.npc = 0.01,
               # label.y.npc=0.01, 
               size = 3)+
ggpubr:: stat_cor(method="spearman",
                       cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "red",
               label.x.npc = 0.01,
               label.y.npc=0.01,
               size = 3)

```

### HDDC2
```{r HDDC2}

HDDC2_table <- readRDS("output/FRS2_dif_values.RDS")

HDDC2_table %>% 
ggplot(., aes(x=tox, y=cpm))+
  geom_point(aes(color=indv))+
  geom_smooth(method="lm")+
  # stat_regline_equation(label.x.npc = "center")+
  facet_wrap(~drug, scales = "free_x")+
  labs(title="HDDC2 ", y = "delta cpm", x = "delta tox")+
  scale_color_brewer(palette = "Dark2",guide="legend",name = "Individual",labels(c(1,2,3,4,5,6)))+
  theme_bw()+
  ggpubr:: stat_cor(method="pearson",
                       # cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "black",
               # label.x.npc = 0.01,
               # label.y.npc=0.01, 
               size = 3)+
ggpubr:: stat_cor(method="spearman",
                       cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "red",
               label.x.npc = 0.01,
               label.y.npc=0.01,
               size = 3)

```

### RMI1
```{r RMI1}

RMI1_table <- readRDS("output/RMI1_dif_values.RDS")

RMI1_table %>% 
ggplot(., aes(x=tox, y=cpm))+
  geom_point(aes(color=indv))+
  geom_smooth(method="lm")+
  # stat_regline_equation(label.x.npc = "center")+
  facet_wrap(~drug, scales = "free_x")+
  labs(title="RMI1 ", y = "delta cpm", x = "delta tox")+
  scale_color_brewer(palette = "Dark2",guide="legend",name = "Individual",labels(c(1,2,3,4,5,6)))+
  theme_bw()+
  ggpubr:: stat_cor(method="pearson",
                       # cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "black",
               # label.x.npc = 0.01,
               # label.y.npc=0.01, 
               size = 3)+
ggpubr:: stat_cor(method="spearman",
                       cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "red",
               label.x.npc = 0.01,
               label.y.npc=0.01,
               size = 3)

```

### SLC28A3
```{r SLC28A3}



SLC28A3_table <- readRDS("output/SLC28A3_dif_values.RDS")
SLC28A3_table %>% 
ggplot(., aes(x=tox, y=cpm))+
  geom_point(aes(color=indv))+
  geom_smooth(method="lm")+
  # stat_regline_equation(label.x.npc = "center")+
  facet_wrap(~drug, scales = "free_x")+
  labs(title="SLC28A3 ", y = "delta cpm", x = "delta tox")+
  scale_color_brewer(palette = "Dark2",guide="legend",name = "Individual",labels(c(1,2,3,4,5,6)))+
  theme_bw()+
  ggpubr:: stat_cor(method="pearson",
                       # cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "black",
               # label.x.npc = 0.01,
               # label.y.npc=0.01, 
               size = 3)+
ggpubr:: stat_cor(method="spearman",
                       cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "red",
               label.x.npc = 0.01,
               label.y.npc=0.01,
               size = 3)


```

### TNS2
```{r TNS2}

  

TNS2_table <- readRDS("output/TNS2_dif_values.RDS")

TNS2_table %>% 
ggplot(., aes(x=tox, y=cpm))+
  geom_point(aes(color=indv))+
  geom_smooth(method="lm")+
  # stat_regline_equation(label.x.npc = "center")+
  facet_wrap(~drug, scales = "free_x")+
  labs(title="TNS2 ", y = "delta cpm", x = "delta tox")+
  scale_color_brewer(palette = "Dark2",guide="legend",name = "Individual",labels(c(1,2,3,4,5,6)))+
  theme_bw()+
  ggpubr:: stat_cor(method="pearson",
                       # cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "black",
               # label.x.npc = 0.01,
               # label.y.npc=0.01, 
               size = 3)+
ggpubr:: stat_cor(method="spearman",
                       cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "red",
               label.x.npc = 0.01,
               label.y.npc=0.01,
               size = 3)



```
### ZNF740

```{r ZNF740}


ZNF740_table <- readRDS("output/ZNF740_dif_values.RDS")
ZNF740_table %>% 
ggplot(., aes(x=tox, y=cpm))+
  geom_point(aes(color=indv))+
  geom_smooth(method="lm")+
  # stat_regline_equation(label.x.npc = "center")+
  facet_wrap(~drug, scales = "free_x")+
  labs(title="ZNF740 ", y = "delta cpm", x = "delta tox")+
  scale_color_brewer(palette = "Dark2",guide="legend",name = "Individual",labels(c(1,2,3,4,5,6)))+
  theme_bw()+
  ggpubr:: stat_cor(method="pearson",
                       # cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "black",
               # label.x.npc = 0.01,
               # label.y.npc=0.01, 
               size = 3)+
ggpubr:: stat_cor(method="spearman",
                       cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "red",
               label.x.npc = 0.01,
               label.y.npc=0.01,
               size = 3)

```
### C3orf18

```{r C3orf18}

C3orf18_table <- readRDS("output/C3orf18_dif_values.RDS")

C3orf18_table %>% 
ggplot(., aes(x=tox, y=cpm))+
  geom_point(aes(color=indv))+
  geom_smooth(method="lm")+
  # stat_regline_equation(label.x.npc = "center")+
  facet_wrap(~drug, scales = "free_x")+
  labs(title="C3orf18 ", y = "delta cpm", x = "delta tox")+
  scale_color_brewer(palette = "Dark2",guide="legend",name = "Individual",labels(c(1,2,3,4,5,6)))+
  theme_bw()+
  ggpubr:: stat_cor(method="pearson",
                       # cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "black",
               # label.x.npc = 0.01,
               # label.y.npc=0.01, 
               size = 3)+
ggpubr:: stat_cor(method="spearman",
                       cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "red",
               label.x.npc = 0.01,
               label.y.npc=0.01,
               size = 3)


```
###RASIP1

```{r RASIP1}


RASIP1_table <- readRDS("output/RASIP1_dif_values.RDS")

RASIP1_table %>% 
ggplot(., aes(x=tox, y=cpm))+
  geom_point(aes(color=indv))+
  geom_smooth(method="lm")+
  # stat_regline_equation(label.x.npc = "center")+
  facet_wrap(~drug, scales = "free_x")+
  labs(title="RASIP1 ", y = "delta cpm", x = "delta tox")+
  scale_color_brewer(palette = "Dark2",guide="legend",name = "Individual",labels(c(1,2,3,4,5,6)))+
  theme_bw()+
  ggpubr:: stat_cor(method="pearson",
                       # cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "black",
               # label.x.npc = 0.01,
               # label.y.npc=0.01, 
               size = 3)+
ggpubr:: stat_cor(method="spearman",
                       cor.coef.name="rho",
               aes(label = paste(..r.label.., ..p.label.., sep = "~`,\n`~")),
               color = "red",
               label.x.npc = 0.01,
               label.y.npc=0.01,
               size = 3)


```
other graphs are on the webpage ending in diff_tox.html

## Making highly variable supersets and overlapping with Seoane and Knowles data sets in figure 8
 
 
 
```{r loading highly vargene sets}

all_expressed <-read_csv("data/backGL.txt", 
    col_types = cols(...1 = col_skip()))


base_egene <-readRDS("output/knowles4.RDS")
resp_egene <-readRDS("output/knowles5.RDS")

highly_var_genelist<- readRDS("data/highly_var_genelist.RDS")

chrom_reg_Seoane <- read_csv(file = "data/Seonane2019supp1.txt",col_types = cols(...1 = col_skip()))
Seoane_2019 <- chrom_reg_Seoane[,2]
names(Seoane_2019) <- "ENTREZID"
Chrom_reg <- data.frame("entrez"=(unique(Seoane_2019$ENTREZID)))

Sup4seoane <- read.csv("output/Sup4seoane.csv", row.names = 1)
DOX_reg <- Sup4seoane  %>% 
  filter(pval.expAnth<0.05) %>% 
  distinct(entrez, .keep_all = TRUE) %>% 
  dplyr::select(entrez)

```
 
### enrichment test


```{r getting the plot}

Hvar_y <- all_expressed %>%
  mutate(Seoane_1 = if_else(ENTREZID %in% Chrom_reg$entrez,1,0)) %>%
  mutate(Seoane_4 = if_else(ENTREZID %in% DOX_reg$entrez, 1,0)) %>%
  mutate(base_egene = if_else(ENTREZID %in% base_egene$entrezgene_id,1,0))%>%
  mutate(resp_egene= if_else(ENTREZID %in% resp_egene$entrezgene_id,1,0))%>%
  mutate(Highvar= if_else(ENTREZID %in% highly_var_genelist,"y","n")) %>% 
  group_by(Highvar) %>% 
  dplyr::summarize(n=n(), K4=sum(base_egene), K5=sum(resp_egene),S1=sum(Seoane_1), S4=sum(Seoane_4)) %>% 
  aggregate(.~Highvar,sum)  
  

Hvar_y %>% 
  pivot_longer(cols = K4:S4, names_to ="class", values_to = "Hvar_y") %>% 
   mutate("Hvar_n"= n-Hvar_y) %>% 
  dplyr::select(class, Hvar_y,Hvar_n, Highvar) %>% 
  pivot_longer(cols= c(Hvar_y, Hvar_n), names_to = "group", values_to = "counts") %>% 
  mutate(class=factor(class, levels = c("K4", "K5", "S1", "S4"), labels=c("base eGenes set", "response genes set", "chromatin reg. gene set","AC-sensitive chrom reg gene set"))) %>% 
  mutate(Highvar=factor(Highvar, levels = c("y","n"))) %>%
  mutate(group=factor(group, levels = c("Hvar_y", "Hvar_n"))) %>% 
  ggplot(., aes(x=group,y=counts, group=group, fill=Highvar))+
   geom_col(position='fill')+
  facet_wrap(.~class)+
  ylab("")+
  xlab("Type of geneset")+
  guides(fill=guide_legend("Grouping var"))+
           theme_classic()

```



```{r Highvar chisquare}
Knowsvar_base <- matrix(
  c(5, 395, 362, 13322),
  nrow = 2,
  ncol = 2,
  byrow = FALSE,
  dimnames = list(c("Hvar", "nHvar"), c("y", "n")))

base_egeneset<- chisq.test(Knowsvar_base, correct = FALSE)
base_egeneset

Knowsvar_respE <- matrix(
  c(9, 262,358, 13455),
  nrow = 2,
  ncol = 2,
  byrow = FALSE,
  dimnames = list(c("Hvar", "nHvar"), c("y", "n")))

resp_egeneset<- chisq.test(Knowsvar_respE, correct = FALSE)
resp_egeneset

Seoane_Var <-  matrix(c(13, 317, 354, 13400),
  nrow = 2,
  ncol = 2,
  byrow = FALSE,
  dimnames = list(c("Hvar", "nHvar"), c("y", "n")))
all_chrom_regs<- chisq.test(Seoane_Var, correct = FALSE)
all_chrom_regs

# Seoane_Var2 <-  matrix(c(3, , 364,47 1   3670),
#   nrow = 2,
#   ncol = 2,
#   byrow = FALSE,
#   dimnames = list(c("Hvar", "nHvar"), c("y", "n")))
# AC_sens_chrom_regs<- chisq.test(Seoane_Var2, correct = FALSE)
# AC_sens_chrom_regs


baseVegene <- matrix(
  c(5, 9, 362, 358),
  nrow = 2,
  ncol = 2,
  byrow = FALSE,
  dimnames = list(c("base", "nbase"), c("egene", "negene")))

baseVegeneset<- chisq.test(baseVegene, correct = FALSE)
baseVegeneset


chromVactox <- matrix(
  c(5, 9, 362, 358),
  nrow = 2,
  ncol = 2,
  byrow = FALSE,
  dimnames = list(c("chrom", "nchrom"), c("ACtox", "nACtox")))

chisq.test(chromVactox, correct = FALSE)


```


### Combine heatmaps for final figure:

```{r Finfig_hm}
drug_palc <- c("#8B006D","#DF707E","#F1B72B", "#3386DD","#707031","#41B333")
drug_pal_fact <- c("#8B006D" ,"#DF707E", "#F1B72B" ,"#3386DD", "#707031","#41B333")


col_fun1 = circlize::colorRamp2(c(-1, 3), c("white", "purple"))
col_funFC= circlize::colorRamp2(c(-2,0, 2), c("darkgreen","white", "darkorange2"))
col_funTOX = circlize::colorRamp2(c(-1,0, 1), c("darkviolet", "white","firebrick4"))

toplistall <- readRDS("data/toplistall.RDS")
DEG_cormotif <- readRDS("data/DEG_cormotif.RDS")
list2env(DEG_cormotif,envir=.GlobalEnv)

SGWAS_order_frame <- read.csv("output/SGWAS_top50_order.csv", row.names = 1)
orderit <- data.frame(SYMBOL=c(SGWAS_order_frame$SYMBOL,'RARG',
                        'TNS2', 
                        'ZNF740',
                        'SLC28A3',
                        'RMI1',
                        'EEF1B2',
                        'FRS2', 
                        'HDDC2')) %>% 
  left_join(.,backGL) %>% 
  mutate(motif= if_else(ENTREZID %in% motif_ER, "ER", if_else(ENTREZID %in% motif_LR, "LR",if_else(ENTREZID %in% motif_TI, "TI", "NR" ))))




toplistall %>% 
    mutate(drug=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>% 
  filter(ENTREZID %in% orderit$ENTREZID) %>% 
   filter(time =="24_hours") %>% 
  dplyr::select(ENTREZID , id,logFC, adj.P.Val, SYMBOL) 
 

GWASabsFCsig <- 
  toplistall %>% 
    mutate(drug=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>% 
  filter(ENTREZID %in% orderit$ENTREZID) %>% 
   filter(time =="24_hours") %>% 
  dplyr::select(ENTREZID , id,logFC, adj.P.Val, SYMBOL)  %>%
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from =adj.P.Val)
  
gwas_sig_mat <- GWASabsFCsig %>% 
   column_to_rownames(var="id") %>%
  dplyr::select(orderit$SYMBOL) %>%
  as.matrix()
 

GWASabsFC <- toplistall %>% 
  
  # filter(id !="TRZ") %>% 
  filter(time=="24_hours") %>% 
  mutate(logFC= logFC*(-1)) %>%
  filter(ENTREZID %in% orderit$ENTREZID) %>% 
  dplyr::select(SYMBOL ,time, id, logFC) %>% 
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from = logFC) %>% 
  column_to_rownames(var="id") %>%
  dplyr::select(orderit$SYMBOL) %>%
  as.matrix()

study_anno <- data.frame(Study=c(rep("GWAS",35),rep("TWAS",3)),motif=orderit$motif)
rownames(study_anno) <- colnames(GWASabsFC)
ht <- HeatmapAnnotation(df = study_anno,
              col = list(Study=c("GWAS"="darkorange","TWAS"= "blueviolet"),
                         motif = c("LR"="#00BFC4", "NR"="#C77CFF", "ER"="#F8766D", "TI"="#7CAE00"), just = "left"))


Heatmap(GWASabsFC, name = "Fold change\nvalues", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        col = col_funFC,
        row_order = c('DOX','EPI','DNR', 'MTX','TRZ'),
        column_title = "Fold change values of GWAS and TWAS genes n = 38", 
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_order= orderit$SYMBOL,
        bottom_annotation = ht,
        
        column_names_rot = 90, 
        column_names_gp = gpar(fontsize = 12,fontface="italic"),
        column_names_centered =TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(gwas_sig_mat[i, j] <0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})

```

```{r final map var heatmap}

framefun24 <- readRDS("data/framefun24.RDS")

col_pvalue = circlize::colorRamp2(c(0,1 ,2), c("blue","white", "red"))
logFtestfullSGWAS <- framefun24 %>% 
  mutate(DOX_plog=(log10(DOX.VEH.24)*(-1)),
         EPI_plog=(log10(EPI.VEH.24)*(-1)),
         DNR_plog=(log10(DNR.VEH.24)*(-1)),
         MTX_plog=(log10(MTX.VEH.24)*(-1)),
         TRZ_plog=(log10(TRZ.VEH.24)*(-1)),) %>% 
  dplyr::select(ENTREZID,DOX_plog,EPI_plog, DNR_plog, MTX_plog,TRZ_plog) %>% 
  dplyr::filter(ENTREZID %in% orderit$ENTREZID) %>% 
  pivot_longer(cols = !ENTREZID, 
               names_to = "treatment", values_to = "plog") %>% 
  separate(treatment, into=c("treatment",NA)) %>% 
  mutate(ENTREZID=as.numeric(ENTREZID)) %>% 
  left_join(., (backGL)) %>% 
  pivot_wider(id_cols=treatment, 
              names_from = SYMBOL, 
              values_from = plog) %>% 
  column_to_rownames(var="treatment") %>%
  dplyr::select(orderit$SYMBOL) %>% 
  as.matrix()


ComplexHeatmap::Heatmap(logFtestfullSGWAS, name = "-log10\np value", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        col = col_pvalue,
        row_order = c('DOX','EPI','DNR', 'MTX','TRZ'),
        column_title = " Last Heatmap Variance",
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_names_rot = 90, 
        column_names_gp = gpar(fontsize = 8,fontface="italic"),
        column_names_centered =TRUE,
        cell_fun = function(j, i, x, y, width, height, fill) {
        if(logFtestfullSGWAS[i, j] > log10(0.05)*(-1))
            grid.text("*", x, y, gp = gpar(fontsize = 16))
})



```


## Combined Expression
```{r KNowles expression}
DOX_de_genes_interest <- read.csv("output/DOX_de_goi.csv", row.names = 1)


all_cpmcounts <-  read_table("data/Counts_RNA_ERMatthews.txt")
Knowles_log2cpm <- readRDS("data/Knowles_log2cpm_real.RDS") 
store_box <- Knowles_log2cpm%>% 
  # dplyr::select( 'ESGN',ends_with(c('0.625', '0'))) %>% 
  dplyr::filter(ESGN %in% DOX_de_genes_interest$ensembl_gene_id) %>% 
  pivot_longer(cols=!ESGN, names_to = "ind", values_to = "counts") %>% 
  separate(ind,into=c("cell_line","dosage"), sep = ":")%>%
  # mutate(dosage = as.double(dosage, length = 3)) #%>% 
  # full_join(., trop0.625, by=c("cell_line", "dosage")) %>%   group_by(cell_line) %>% 
  left_join(., DOX_de_genes_interest, by = c("ESGN" = "ensembl_gene_id")) %>% 
  mutate(expr="K")

store_box %>% 
  mutate(dosage=factor(dosage, levels=c('0', '0.625','1.25', '2.5','5'))) %>% 
  ggplot(., aes(x=dosage,y=counts), group=dosage)+
  geom_boxplot()+
  facet_wrap(~hgnc_symbol)
```


```{r KNowles expression2, fig.height=11, fig.width=8}
lcpm_24h <- all_cpmcounts %>% 
  column_to_rownames("ENTREZID") %>% 
  cpm(., log=TRUE) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "ENTREZID") %>% 
  # dplyr::select(ENTREZID, all_of(starts_with(c("DOX","VEH")))) %>%
  dplyr::select(ENTREZID, all_of(ends_with("24h")))  %>% 
  dplyr::filter(ENTREZID %in% DOX_de_genes_interest$ENTREZID) %>% 
  pivot_longer(cols=!ENTREZID, names_to = "ind", values_to = "counts") %>%   mutate(ENTREZID = as.numeric(ENTREZID)) %>% 
  full_join(., DOX_de_genes_interest, by = c("ENTREZID"="ENTREZID")) %>% 
  mutate(expr="ME") %>% 
  rename("ESGN"="ensembl_gene_id") %>% 
  separate(ind, into = c("dosage","cell_line",NA)) %>% 
  mutate(dosage=case_match(dosage,"DOX"~"0.5", .default = dosage)) 

my_fill_colors <-  c("#41B333","#8B006D","chocolate1","chocolate","chocolate4","#41B333","#8B006D","#DF707E","#F1B72B", "#3386DD","#707031")

lcpm_24h %>% 
  rbind(.,store_box) %>% 
  mutate(dosage=factor(dosage, levels=c('0','0.625',"1.25","2.5","5","VEH","0.5","EPI","DNR", "MTX","TRZ"))) %>% 
  ggplot(., aes(x=dosage,y=counts))+
  geom_boxplot(aes(fill=dosage))+
  facet_wrap(~hgnc_symbol, scales="free_y", nrow=10, ncol = 2 )+
  theme_bw()+
  scale_fill_manual(values=my_fill_colors)



```

### Link to other pages here:

[Knowles and troponin](https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/Knowels_trop_analysis.html)

GO analysis on after_comments data [here](https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/GO_after_comments.html)

