---
title: "GO analysis"
author: "ERM"
date: "`r Sys.Date()`"
output: 
  html_document: 
    theme: flatly
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE)
```

# GO Analysis

I have created several files from the RNA analysis that contain the significant genes(determined by adj.P.val \< 0.05) from each Time and Condition. The names of the files are in the following format: 'sigV'+Drug(2 letters)+time.

example: 'sigVDA3.txt' means this file contains the significant DE genes from the DNR 3 hour compared to Vehicle Control 3 hour analysis

```{r Import libraries, message=FALSE, warning=FALSE}
library(tidyverse)
library(gprofiler2)
library(readr)
library(BiocGenerics)
library(gridExtra)
library(kableExtra)
library(scales)
library(ggVennDiagram)
library(cowplot)
library(ggpubr)
library(rstatix)
library(zoo)
library(ggsignif)
```

```{r Import data, echo=FALSE, message=FALSE, warning=FALSE}
  toplistall <- readRDS("data/toplistall.RDS")
  #get sig files made with 0.05 data this way(data created on run)
siglist <- readRDS("data/siglist_final.RDS")
list2env(siglist,envir=.GlobalEnv)

```


```{r creation of toplist, eval=FALSE, echo=TRUE, message=FALSE, warning=FALSE}
##creation of the toplistall file:
efit2 <- readRDS("data/efit2_final.RDS")

##define 3hr and 24 hour toplist
V.DA.top= topTable(efit2, coef=1, adjust="BH", number=Inf, sort.by="p")
V.DX.top= topTable(efit2, coef=2, adjust="BH", number=Inf, sort.by="p")
V.EP.top= topTable(efit2, coef=3, adjust="BH", number=Inf, sort.by="p")
V.MT.top= topTable(efit2, coef=4, adjust="BH", number=Inf, sort.by="p")
V.TR.top= topTable(efit2, coef=5, adjust="BH", number=Inf, sort.by="p")

V.DA24.top= topTable(efit2, coef=6, adjust="BH", number=Inf, sort.by="p")
V.DX24.top= topTable(efit2, coef=7, adjust="BH", number=Inf, sort.by="p")
V.EP24.top= topTable(efit2, coef=8, adjust="BH", number=Inf, sort.by="p")
V.MT24.top= topTable(efit2, coef=9, adjust="BH", number=Inf, sort.by="p")
V.TR24.top= topTable(efit2, coef=10, adjust="BH", number=Inf, sort.by="p")

toplist3hours <-list(V.DA.top[,c(1:3,4,6:7)],
                              V.DX.top[,c(1:3,4,6:7)],
                              V.EP.top[,c(1:3,4,6:7)],
                              V.MT.top[,c(1:3,4,6:7)],
                              V.TR.top[,c(1:3,4,6:7)])
names(toplist3hours) <- c("DNR","DOX", "EPI","MTX", "TRZ")

toplist3hours <- map_df(toplist3hours, ~as.data.frame(.x), .id="id")


toplist24hours <-list(V.DA24.top[,c(1:3,4,6:7)],
                    V.DX24.top[,c(1:3,4,6:7)],
                    V.EP24.top[,c(1:3,4,6:7)],
                    V.MT24.top[,c(1:3,4,6:7)],
                    V.TR24.top[,c(1:3,4,6:7)])
names(toplist24hours) <- c("DNR","DOX", "EPI","MTX", "TRZ")

toplist24hours <- map_df(toplist24hours, ~as.data.frame(.x), .id="id")

###Combine 3 and 24 for toplist all

toplistall <- list(toplist24hours,toplist3hours)
names(toplistall) <- c("24_hours", "3_hours")
toplistall <-map_df(toplistall, ~as.data.frame(.x), .id="time")
saveRDS(toplistall, "data/toplistall.RDS")
write.csv(toplistall, "output/toplistall.csv")

```

The analysis is based on all genes that passed the rowMeans\>0 from the previous page which are about 14084 genes expressed in my RNA-seq data [link](https://reneeisnowhere.github.io/Cardiotoxicity/run_all_analysis.html)

```{r uploading the background genes, echo=FALSE,  message=FALSE, warning=FALSE}
# backGL <- efit2$genes  # ***making the list

# write.csv(backGL, "data/backGL.txt", row.names = FALSE)
##read the  .csv then render into txt in the data file!
 backGL <- read_csv("data/backGL.txt", 
    col_types = cols(...1 = col_skip()))
hsa_kegg_anno <- readRDS("data/hsa_kegg_anno.RDS")
```

All analysis is completed with an adjusted p value of 0.05
This is updated from the 0.1 value I had been using.

2023-04-19 RM
ran final RNA-seq DEG code on 6-30-23 RM

```{r tras had no DEGs}

drug_palNoVeh <- c("#8B006D" ,"#DF707E", "#F1B72B" ,"#3386DD", "#707031")

toplistall %>%
  mutate(id = factor(id,levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  group_by(time, id) %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  count(sigcount) %>% 
  pivot_wider(id_cols = c(time,id), names_from=sigcount, values_from=n) %>% 
 
  mutate(prop = sig/(sig+notsig)*100) %>% 
  mutate(prop=if_else(is.na(prop),0,prop)) %>% 
  ggplot(., aes(x=id, y= prop))+
  geom_col(aes(fill=id))+
  geom_text(aes(label = sprintf("%.2f",prop)),
            position=position_dodge(0.9),vjust=-.2 )+
  scale_fill_manual(values =drug_palNoVeh)+
  guides(fill=guide_legend(title = "Treatment"))+
  facet_wrap(id~time)+#labeller = (time = facettimelabel) )+
  theme_bw()+
  xlab("")+
  ylab("Percentage of expressed genes")+
  theme_bw()+
  facet_wrap(~time)+
  ggtitle("Percent DEGs (adj. P value <0.05)")+
  scale_y_continuous(expand= expansion(c(0.02,.1)))+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        # axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        strip.background = element_rect(fill = "transparent"),
        axis.text.x = element_text(size = 8, color = "white", angle = 0),
        axis.text.y = element_text(size = 8, color = "black", angle = 0),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))


```
Knowels, et al. mentioned a relationship between transcriptional change level indicating cardiotoxicity.  This may be an observation that supports that claim as well.
## DNR

This is analysis on the significantly differentially expressed genes for DNR by 3 and 24 hours.
```{r intial analysis DA3 and DA24, echo=FALSE, message=FALSE, warning=FALSE}
# DAgostres3 <- gost(query = sigVDA3$ENTREZID,
#                    organism = "hsapiens",
#                    significant = FALSE,
#                     ordered_query = TRUE,
#                     domain_scope = "custom",
#                     measure_underrepresentation = FALSE,
#                     evcodes = FALSE,
#                    numeric_ns = "ENTREZGENE",
#                     user_threshold = 0.05,
#                     correction_method = c("fdr"),
#                     custom_bg = backGL$ENTREZID,
#                     sources=c("GP:BP","KEGG"))
# saveRDS(DAgostres3,"data/DEG-GO/DAgostres3.RDS")
DAgostres3 <- readRDS("data/DEG-GO/DAgostres3.RDS")
DAp3 <- gostplot(DAgostres3, capped = FALSE, interactive = TRUE)
# publish_gostplot(DAp3, highlight_terms = c("KEGG:04115", "KEGG:05168", "KEGG:04110", "KEGG:03030", "KEGG:03410"))
# overlapKEGG <- hsa_kegg_anno %>% 
#   filter(eg %in%backGL$ENTREZID) #%>% 
#   filter(pathway %in% Path_of_interest)
# unique(overlapKEGG$pathway)

DAtable3 <- DAgostres3$result %>%
  dplyr::select(c(source, term_id,term_name,intersection_size, 
                   term_size, p_value))
# testframe <- DAgostres3star$meta
# testframeresult <- DAgostres3star$result
# testframeresult%>%
#   filter(term_id %in% kegg)
#   mutate_at(.vars = 6, .funs= scientific_format()) %>% 
#   kable(., captions=" fdr at 1") %>% 
#   kable_paper("striped", full_width = FALSE) %>%  
#   kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
#   scroll_box(width = "100%", height = "400px")
DAtable3 %>% 
    dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('DNR 3 hour GO:BP terms') +
    xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
    theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))
DAtable3 %>% mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,) %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")
```

###  DNR specific genes at 3 hours 
    How they may relate to 24 hour genes using GO analysis
 3 hours specifically (see venn below too):

```{r daun 3 hour sig DEG,}

total24 <-list(sigVDA24$ENTREZID,sigVDX24$ENTREZID,sigVEP24$ENTREZID,sigVMT24$ENTREZID)
total3 <- list(sigVDA3$ENTREZID,sigVDX3$ENTREZID,sigVEP3$ENTREZID,sigVMT3$ENTREZID)

list3totvenn3 <- VennDiagram::get.venn.partitions(total3)
list24totvenn24 <- VennDiagram::get.venn.partitions(total24)

Daun3sp <- list3totvenn3$..values..[[15]]
Daun24sp <- list24totvenn24$..values..[[15]]


# DAspgostres3 <- gost(query = Daun3sp,
#                     organism = "hsapiens", 
#                     significant = FALSE,
#                     ordered_query = TRUE,
#                     domain_scope = "custom",
#                     measure_underrepresentation = FALSE,
#                     evcodes = FALSE,
#                     user_threshold = 0.05,
#                     correction_method = c("fdr"),
#                     custom_bg = backGL$ENTREZID,
#                     sources=c("GO:BP","KEGG"))
# saveRDS(DAspgostres3,"data/DEG-GO/DAspgostres3.RDS")
DAspgostres3 <- readRDS("data/DEG-GO/DAspgostres3.RDS")

DAspp3 <- gostplot(DAspgostres3, capped = FALSE, interactive = TRUE)
DAspp3

DAsptable3 <- DAspgostres3$result %>% 
  dplyr::select(c(source, term_id,
                  term_name,intersection_size, 
                   term_size, p_value))# %>% 

```

###  DNR specific genes at 24 hours 

```{r DA24sp }
###no enrichment
# Daun24spgost <- gost(query = Daun24sp,
#                    organism = "hsapiens", 
#                    significant = FALSE,
#                     ordered_query = TRUE,
#                      domain_scope = "custom",
#                      measure_underrepresentation = FALSE,
#                      evcodes = FALSE,
#                      user_threshold = 0.05,
#                      correction_method = c("fdr"),
#                      custom_bg = backGL$ENTREZID,
#                      sources=c("GO:BP","KEGG"))
# saveRDS(Daun24spgost,"data/DEG-GO/Daun24spgost.RDS")
Daun24spgost <- readRDS("data/DEG-GO/Daun24spgost.RDS")
# 
# DAspp24h<- gostplot(Daun24spgost, capped = FALSE, interactive = TRUE)
# DAspp24h

DAspp24h <- Daun24spgost$result %>%
  dplyr::select(c(source, term_id,
                  term_name,intersection_size,
                   term_size, p_value))# %>%
DAspp24h %>% 
  mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,) %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")
#   
# 

```




### All DNR specifc genes adj. p value < 0.05

```{r daun specific Venn genes}
# 3 hour DNR specific genes (venn)
 


DAsptable3 <- DAspgostres3$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value))
  DAsptable3 %>% 
    dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('DNR specific 3 hour GO:BP \nterms n = 335') +
    xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))
  
  
DAsptable3 %>% mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,) %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")



length(union(Daun3sp,sigVDX3$ENTREZID))
ggVennDiagram::ggVennDiagram(list(Daun3sp, sigVDA24$ENTREZID),
              category.names = c("3 hour\n DNR- specific  ","24 hours\n DEG p < 0.05"),
              show_intersect = FALSE,
              set_color = "black",
              label = "count",
              label_percent_digit = 1,
              label_size = 5,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid") +
  scale_x_continuous(expand = expansion(mult = .2))+
  scale_y_continuous(expand = expansion(mult = .1))+
  scale_fill_gradient(low = "red4", high = "tan1")+
  labs(title = "3 hr DNR-sp with 24 hour DNR", caption = paste("n =", length(union(Daun3sp,sigVDX3$ENTREZID)),"genes"))+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5))


```
Interestingly, Over half of the DNR specific genes are in the DNR 24 hour group.

When I look at those  genes, I do not see a significant enrichment in GO:BP terms, only the KEGG pathway connection to Herpes simplex virus

```{r 3hour24hour daun overlap}
list324venn <- VennDiagram::get.venn.partitions(list(Daun3sp, sigVDA24$ENTREZID))

over3_24daun <- list324venn$..values..[[1]]
# 
# over3_24gost <- gost(query = over3_24daun,
#                   organism = "hsapiens", 
#                   significant = FALSE,
#                   ordered_query = TRUE,
#                   domain_scope = "custom",
#                   measure_underrepresentation = FALSE,
#                   evcodes = FALSE,
#                   user_threshold = 0.05,
#                   correction_method = c("fdr"),
#                   custom_bg = backGL$ENTREZID,
#                   sources=c("GO:BP","KEGG"))
# saveRDS(over3_24gost, "data/DEG-GO/over3_24gost.RDS")
over3_24gost <- readRDS("data/DEG-GO/over3_24gost.RDS")
DA3_24table <- gostplot(over3_24gost, capped = FALSE, interactive = TRUE)
DA3_24table
plDA3_24table <- over3_24gost$result%>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value))

plDA3_24table%>% 
  mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,  caption = "Overlap of 3hr and 24 hour DNR sigDEGs and all GO:BP and KEGG pathways") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")

```


### DNR 24 sigDEG

Looking at the DNR 24 hour geneset using adj. P value of 0.05, I have enrichment  in the terms cell cycle and chromosome segregation. 
```{r DA sig deg 24}
# DAgostres24 <- gost(query = sigVDA24$ENTREZID,
#                   organism = "hsapiens", 
#                   significant = FALSE,
#                   ordered_query = TRUE,
#                   domain_scope = "custom",
#                   measure_underrepresentation = FALSE,
#                   evcodes = FALSE,
#                   user_threshold = 0.05,
#                   correction_method = c("fdr"),
#                   custom_bg = backGL$ENTREZID,
#                   sources=c("GO:BP","KEGG"))
# saveRDS(DAgostres24, "data/DAgostres24.RDS")
DAgostres24 <- readRDS("data/DAgostres24.RDS")
DAp24 <- gostplot(DAgostres24, capped = FALSE, interactive = TRUE)
DAp24

DAtable124 <- DAgostres24$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value))  
  

  DAtable124 %>% 
  dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('DNR 24 hour GO:BP terms') +
   xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
      theme_bw()+ 
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))
  
  DAtable124 %>% mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,caption = "DNR 24 hour sig DEGs enrichment, all KEGG and GO:BP") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")
  
```
  
###DNR 3 v 24 intersection

All `r length(Daun3sp)` 3 hour DNR-specific genes are in the intersection of the DNR sigDEGs for 3 hours and 24 hours. This could support the idea that DNR's more toxic effect is related to earlier impacts on cell function.  Note the enriched go terms for the 304 genes that they share.  I need to see if there is a trend in EP and DX.  I do see in Mito 3 hour (looked a little later).

```{r DA 3v24 int, echo=FALSE, message=FALSE, warning=FALSE}
daun3v24int <- intersect(sigVDA3$ENTREZID,sigVDA24$ENTREZID)
length(union(sigVDA3$ENTREZID,sigVDA24$ENTREZID))
ggVennDiagram::ggVennDiagram(list(sigVDA3$ENTREZID, sigVDA24$ENTREZID),
              category.names = c("3 hour DNR\nn = 532","24 hour DNR\n   n = 7017"),
              show_intersect = FALSE,
              set_color = "black",
              label = "count",
              label_percent_digit = 1,
              label_size = 4,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid") +
  scale_x_continuous(expand = expansion(mult = .2))+
  scale_y_continuous(expand = expansion(mult = .1))+
  scale_fill_gradient(low = "red4", high = "tan1")+
  labs(title = "DEG p < 0.05, All DNR 3hour and 24 hours sets", 
       caption = paste("n =", length(union(sigVDA3$ENTREZID,sigVDA24$ENTREZID)),"genes"))+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5))


length(intersect(over3_24daun, intersect(sigVDA3$ENTREZID, sigVDA24$ENTREZID)))
# DAgostresint <- gost(query = daun3v24int,
#                     organism = "hsapiens", significant = FALSE,
#                     ordered_query = TRUE,
#                     domain_scope = "custom",
#                     measure_underrepresentation = FALSE,
#                     evcodes = FALSE,
#                     user_threshold = 0.05,
#                     correction_method = c("fdr"),
#                     custom_bg = backGL$ENTREZID,
#                     sources=c("GO:BP","KEGG"))

# saveRDS(DAgostresint,"data/DEG-GO/DAgostresint.RDS")
DAgostresint <- readRDS("data/DEG-GO/DAgostresint.RDS")
DApint <- gostplot(DAgostresint, capped = FALSE, interactive = TRUE)
DApint

DAtable_int <- DAgostresint$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value))  
 
  DAtable_int %>% 
    dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
    ggplot(., aes(x = log_val, y =reorder(term_name,p_value), 
                  col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", 
           size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('DNR intersect 3 v 24 hour GO:BP terms') +
     xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
    theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = 15, color = "black"),
          axis.ticks = element_line(linewidth = 1.5),
          axis.line = element_line(linewidth = 1.5),
          axis.text = element_text(size = 10, color = "black", angle = 0),
          strip.text.x = element_text(size = 15, color = "black", face = "bold"))
  
  DAtable_int %>% 
    mutate_at(.vars = 6, .funs= scientific_format()) %>% 
    kable(.,) %>% 
    kable_paper("striped", full_width = FALSE) %>%
    kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>%
    scroll_box(width = "100%", height = "400px") 
  
  
  
  
```

## DOX 3 hour and 24 hour
```{r intial analysis DX3 , echo=FALSE, message=FALSE, warning=FALSE}
# DXgostres <- gost(query = sigVDX3$ENTREZID,
#                   organism = "hsapiens", 
#                   significant = FALSE,
#                   ordered_query = TRUE,
#                   domain_scope = "custom",
#                   measure_underrepresentation = FALSE,
#                   evcodes = FALSE,
#                   user_threshold = 0.05,
#                   correction_method = c("fdr"),
#                   custom_bg = backGL$ENTREZID,
# sources=c("GO:BP","KEGG"))
# saveRDS(DXgostres,"data/DEG-GO/DXgostres.RDS")

DXgostres <- readRDS("data/DEG-GO/DXgostres.RDS")
DXp <- gostplot(DXgostres, capped = FALSE, interactive = TRUE)
DXp

DXtable1 <- DXgostres$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) 


  DXtable1 %>% 
    dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=8 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('DOX 3 hour GO:BP terms') +
   xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 9, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))

  DXtable1%>% mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,caption= "3 hour list of DOX KEGG and GO:BP terms") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")
```
 
```{r intial analysis DX24 , echo=FALSE, message=FALSE, warning=FALSE}
# ---------------------------------------------------------
  #
  # DXgostres24 <- gost(query = sigVDX24$ENTREZID,
  #                   organism = "hsapiens", 
  #                   significant = FALSE,
  #                   ordered_query = TRUE,
  #                   domain_scope = "custom",
  #                   measure_underrepresentation = FALSE,
  #                   evcodes = FALSE,
  #                   user_threshold = 0.05,
  #                   correction_method = c("fdr"),
  #                   custom_bg = backGL$ENTREZID,
  #                   sources=c("GO:BP","KEGG"))
# saveRDS(DXgostres24,"data/DEG-GO/gostres24.RDS")
  DXgostres24 <- readRDS("data/DEG-GO/gostres24.RDS")
# gostres24 <- readRDS("data/DEG-GO/gostres24.RDS")  
  p24 <- gostplot(DXgostres24, capped = FALSE, interactive = TRUE)
p24

DXtable124 <- DXgostres24$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) 


  DXtable124 %>% 
    dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=8 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected\n terms"))+
    ggtitle('DOX 24 hour GO:BP terms') +
   xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))
DXtable124 %>% 
  mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,caption= "24 hour list of DOX KEGG and GO:BP terms") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")
```
 
## EPI  
All Epi DEGs adj. p value < 0.05

EPspecific genes set now does not show enrichment in any terms.
```{r intial analysis EP3 , echo=FALSE, message=FALSE, warning=FALSE}


# EPsp3 <- list3totvenn3$..values..[[12]]
# EPspgostres <- gost(query = EPsp3,
#                   organism = "hsapiens", 
#                   significant = FALSE,
#                   ordered_query = TRUE,
#                   domain_scope = "custom",
#                   measure_underrepresentation = FALSE,
#                   evcodes = FALSE,
#                   user_threshold = 0.05,
#                   correction_method = c("fdr"),
#                   custom_bg = backGL$ENTREZID,
#                   sources=c("GO:BP","KEGG"))
 # saveRDS(EPspgostres,"data/DEG-GO/EPspgostres.RDS")

# EPgostres <- gost(query = sigVEP3$ENTREZID,
#                   organism = "hsapiens",
#                   significant = FALSE,
#                   ordered_query = TRUE,
#                   domain_scope = "custom",
#                   measure_underrepresentation = FALSE,
#                   evcodes = FALSE,
#                   user_threshold = 0.05,
#                   correction_method = c("fdr"),
#                   custom_bg = backGL$ENTREZID,
#                   sources=c("GO:BP","KEGG"))
# 
# saveRDS(EPgostres,"data/DEG-GO/EPgostres.RDS")

EPspc3h <- readRDS("data/DEG-GO/EPspgostres.RDS")
EPgostres <- readRDS("data/DEG-GO/EPgostres.RDS")
EPp <- gostplot(EPgostres, capped = FALSE, interactive = TRUE)
EPp 

EPtable1 <- EPgostres$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value))  




  EPtable1 %>% 
    dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('EPI 3 hour top ten GO:BP terms') +
   xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))
  EPtable1%>%
    mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,caption="3hour EPI full enrichment list") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")
  
 # ---------------------------------------------------------
```


```{r intial analysis EP24 , echo=FALSE, message=FALSE, warning=FALSE}
# EPgostres24 <- gost(query = sigVEP24$ENTREZID,
#                   organism = "hsapiens", 
#                   significant = FALSE,
#                   ordered_query = TRUE,
#                   domain_scope = "custom",
#                   measure_underrepresentation = FALSE,
#                   evcodes = FALSE,
#                   user_threshold = 0.05,
#                   correction_method = c("fdr"),
#                   custom_bg = backGL$ENTREZID,
#                   sources=c("GO:BP","KEGG"))
# saveRDS(EPgostres24,"data/DEG-GO/EPgostres24.RDS")
EPgostres24 <- readRDS("data/DEG-GO/EPgostres24.RDS")
  
  
  EPp24 <- gostplot(EPgostres24, capped = FALSE, interactive = TRUE)
EPp24

EPtable124 <- EPgostres24$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) 
EPtable124 %>% 
    dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('EPI 24 hour top 10 GO:BP terms') +
    xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))
EPtable124%>% 
  mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,caption= "EPI 24 hour lists") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")



##epi 24 specifi n = 533
epsp24 <- list24totvenn24$..values..[[12]]
##ep 3 hour above



# 
# EPspgostres24sp <- gost(query = epsp24,
#                     organism = "hsapiens", significant = FALSE,
#                     ordered_query = TRUE,
#                     domain_scope = "custom",
#                     measure_underrepresentation = FALSE,
#                     evcodes = FALSE,
#                     user_threshold = 0.05,
#                     correction_method = c("fdr"),
#                     custom_bg = backGL$ENTREZID,
#                     sources=c("GO:BP","KEGG"))
# saveRDS(EPspgostres,"data/DEG-GO/EPspgostres.RDS")
# saveRDS(EPgostres24sp,"data/DEG-GO/EPgostres24sp.RDS")
EPspgostres<- readRDS("data/DEG-GO/EPspgostres.RDS")
# 
# EPsptable <- EPspgostres$result %>%
#   dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value))
# EPsptable %>%
#     dplyr::filter(source=="KEGG") %>%
#     dplyr::select(p_value,term_name,intersection_size) %>%
#     slice_min(., n=10 ,order_by=p_value) %>%
#     mutate(log_val = -log10(p_value)) %>%
#    # slice_max(., n=10,order_by = p_value) %>%
#    ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
#     geom_point(aes(size = intersection_size)) +
#     scale_y_discrete(labels = label_wrap(30))+
#     guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
#     ggtitle("EPI specific 3 hour top KEGG terms") +
#      xlab(expression(" -"~log[10]~("adj. p-value")))+
#     ylab("KEGG term")+
#       theme_bw()+
#     theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
#         axis.title = element_text(size = 15, color = "black"),
#         axis.ticks = element_line(linewidth = 1.5),
#         axis.line = element_line(linewidth = 1.5),
#         axis.text = element_text(size = 10, color = "black", angle = 0),
#         strip.text.x = element_text(size = 15, color = "black", face = "bold"))
```



```{r mito specific at 24 hours}

mtsp24 <- list24totvenn24$..values..[[8]]
# MTgostres24sp <- gost(query = mtsp24,
#                     organism = "hsapiens", significant = FALSE,
#                     ordered_query = TRUE,
#                     domain_scope = "custom",
#                     measure_underrepresentation = FALSE,
#                     evcodes = FALSE,
#                     user_threshold = 0.05,
#                     correction_method = c("fdr"),
#                     custom_bg = backGL$ENTREZID,
#                     sources=c("GO:BP","KEGG"))
# 
# saveRDS(MTgostres24sp,"data/DEG-GO/MTgostres24sp.RDS")
MTgostres24sp <- readRDS("data/DEG-GO/MTgostres24sp.RDS")


MTsptable124 <- MTgostres24sp$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) 


MTsptable124%>% 
    dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle("MTX specific 24 hour top GO:BP terms") +
     xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))



```
MTX-specific 24 hour genes show an enrichment for DNA repair. (n=188)

## MTX DEGs (adj. p value < 0.05)
  3 hour  = `r length(sigVMT3$ENTREZID)`
  24 hour = `r length(sigVMT24$ENTREZID)`

```{r intial analysis MT3, echo=FALSE, message=FALSE, warning=FALSE}
# MTgostresstar <- gost(query = sigVMT3$ENTREZID,
#                   organism = "hsapiens",
#                   significant = FALSE,
#                   ordered_query = TRUE,
#                   domain_scope = "custom",
#                   measure_underrepresentation = FALSE,
#                   evcodes = FALSE,
#                   numeric_ns="ENTREZGENE",
#                   user_threshold = 1,
#                   correction_method = c("fdr"),
#                   custom_bg = backGL$ENTREZID,
#                   sources=c("KEGG"))

# saveRDS(MTgostres,"data/DEG-GO/MTgostre.RDS")
MTgostres <- readRDS("data/DEG-GO/MTgostre.RDS")

MTp <- gostplot(MTgostres, capped = FALSE, interactive = TRUE)
MTp

MTtable1 <- MTgostres$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value))
MTtable1 %>% 
 dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('MTX 3 hour GO:BP terms') +
    xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold")) 
  MTtable1 %>% 
    mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,caption = "MTX 3 hour table of enrichment") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")
```

```{r intial analysis MT24, echo=FALSE, message=FALSE, warning=FALSE}  
   # ---------------------------------------------------------
#   
# MTgostres24 <- gost(query = sigVMT24$ENTREZID,
#                   organism = "hsapiens", 
#                   significant = FALSE,
#                   ordered_query = TRUE,
#                   domain_scope = "custom",
#                   measure_underrepresentation = FALSE,
#                   evcodes = FALSE,
#                   user_threshold = 0.05,
#                   correction_method = c("fdr"),
#                   custom_bg = backGL$ENTREZID,
#                   sources=c("GO:BP","KEGG"))
# saveRDS(MTgostres24,"data/DEG-GO/MTgostres24.RDS")
MTgostres24 <- readRDS("data/DEG-GO/MTgostres24.RDS")
  
  
  MTp24 <- gostplot(MTgostres24, capped = FALSE, interactive = TRUE)
MTp24

MTtable124 <- MTgostres24$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) 

MTtable124 %>% dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('MTX 24 hour GO:BP terms') +
    xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))

  
  MTtable124 %>% 
    mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,caption = "MTX 24hour enrichment of terms") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")
  

```




## Venn Diagrams  

### All DEGs adj. p value < 0.05, 24 hour  

```{r venndiagram, echo=FALSE, message=FALSE, warning=FALSE}
library(paletteer)

total24 <-list(sigVDA24$ENTREZID,sigVDX24$ENTREZID,sigVEP24$ENTREZID,sigVMT24$ENTREZID)
in_common24 <-c(sigVDA24$ENTREZID,sigVDX24$ENTREZID,sigVEP24$ENTREZID,sigVMT24$ENTREZID)


ggVennDiagram::ggVennDiagram(total24,
              category.names = c("DNR\nn = 7017    ",
                                 "DOX\n n = 6645\n",
                                 "EPI\n n = 6328\n",
                                 " MTX\n   n = 1115"),
              show_intersect = FALSE,
              set_color = "black",
              category_size = c(6,6,6,6),
              label = "count",
              label_percent_digit = 1,
              label_size = 4,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = 4.5)+
  scale_x_continuous(expand = expansion(mult = .3))+
  scale_y_continuous(expand = expansion(mult = .2))+
  scale_color_paletteer_d(palette = "fishualize::Bodianus_pulchellus")+
  scale_fill_distiller(palette="Spectral", direction = -1)+
  labs(title = "24 hour comparison of DE genes p.adj. <0.05",
       caption = paste("n =", length(unique(in_common24)),"genes"))+
  
  theme(plot.title = element_text(size = rel(1.6), hjust = 0.5, vjust =1))





#paletteer_d("fishualize::Bodianus_pulchellus")
```

### All DEGs adj. p value < 0.05, 3 hour

```{r 3 hours, echo=FALSE, message=FALSE, warning=FALSE}
# total 3 ----------------------------------------------------------------
drug_palNoVeh <- c("#8B006D","#DF707E",
                   "#F1B72B","#3386DD","#707031","#41B333")
library(paletteer)
total3 <- list(sigVDA3$ENTREZID,sigVDX3$ENTREZID, sigVEP3$ENTREZID,sigVMT3$ENTREZID)
totalin_common3 <- c(sigVDA3$SYMBOL,sigVDX3$SYMBOL, sigVEP3$SYMBOL,sigVMT3$SYMBOL)


ggVennDiagram::ggVennDiagram(total3,
              category.names = c("DNR\nn = 532   ",
                                 "DOX\nn = 19\n",
                                 "EPI\nn = 210\n",
                                 "MTX\n  n = 75"),
              show_intersect = FALSE,
              set_color = "black",
              category_size = c(6,6,6,6),
              label = "count",
              label_percent_digit = 1,
              label_size = 4,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = 4.5)+
  scale_x_continuous(expand = expansion(mult = .3))+
  scale_y_continuous(expand = expansion(mult = .2))+
  scale_color_paletteer_d(palette = "fishualize::Bodianus_pulchellus")+
  scale_fill_distiller(palette="Spectral", direction = -1,limits=c(0,4500))+
  labs(title = "3 hour comparison of DE genes p.adj. <0.05",
       caption = paste("n =", length(unique(totalin_common3)),"genes"))+
  theme(plot.title = element_text(size = rel(1.6), hjust = 0.5, vjust =1))
```
Because the 3 hour DEGs are low (expecially in DOX),  we have chosen to focus on analyzing the 24 hour DEGs for GO: BP analysis
### Venn 3 and 24 hour sigDEGs by treatment (adj. p value <0.01) similarities

```{r 3V 24 BY DRUG, echo=FALSE}
# Dauno comp --------------------------------------------------------------

Dauncomp <- list(sigVDA24$ENTREZID,sigVDA3$ENTREZID)
in_commonDa <- c(sigVDA24$ENTREZID,sigVDA3$ENTREZID)
length(unique(in_commonDa))
# DNRvenn <-
  ggVennDiagram(Dauncomp,
              category.names = c("DNR-24","DNR-3"),
              show_intersect = FALSE,
              set_color = "black",
              label = "count",
              label_percent_digit = 1,
              label_size = 4,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = )+
  scale_x_continuous(expand = expansion(mult = .2))+
  scale_fill_gradient(low = "lightblue", high = "yellow")+
  labs(title = "Comparison of DNR 3h v 24h", caption = paste("n =", length(unique(in_commonDa)),"genes"))+
  theme(plot.title = element_text(size = rel(1), hjust = 0.5))
#Davenlist <- intersect(Dauncomp[[1]],Dauncomp[[2]])
# saveRDS(DNRvenn,"output/DNRvenn.RDS")

# Doxocomp ----------------------------------------------------------------
Doxcomp <- list(sigVDX24$ENTREZID,sigVDX3$ENTREZID)
in_commonDx <- c(sigVDX24$ENTREZID,sigVDX3$ENTREZID)
length(unique(in_commonDx))
# DOXvenn <-
  ggVennDiagram(Doxcomp,
              category.names = c("DOX-24","DOX-3"),
              show_intersect = FALSE,
              set_color = "black",
              label = "count",
              label_percent_digit = 1,
              label_size = 4,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = )+
  scale_x_continuous(expand = expansion(mult = .2))+
  scale_fill_gradient(low = "light blue", high = "yellow")+
  labs(title = "Comparison of Doxo 3h v 24h", 
       caption = paste("n =", length(unique(in_commonDx)),"genes"))+
  theme(plot.title = element_text(size = rel(1), hjust = 0.5))
# saveRDS(DOXvenn,"output/DOXvenn.RDS")

# Dxvenlist <- intersect(Doxcomp[[1]],Doxcomp[[2]])
#length(intersect(Dxvenlist,Davenlist))## 7 of DX are in DA
# Epi Comp ----------------------------------------------------------------

Epicomp <- list(sigVEP24$ENTREZID,sigVEP3$ENTREZID)
in_commonEp <- c(sigVEP24$ENTREZID,sigVEP3$ENTREZID)
length(unique(in_commonEp))

# EPIvenn <-
  ggVennDiagram(Epicomp,
              category.names = c("EPI-24","EPI-3"),
              show_intersect = FALSE,
              set_color = "black",
              label = "count",
              label_percent_digit = 1,
              label_size = 4,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = )+
  scale_x_continuous(expand = expansion(mult = .2))+
  scale_fill_gradient(low = "light blue", high = "yellow")+
  labs(title = "Comparison of EPI 3h v 24h", caption = paste("n =", length(unique(in_commonEp)),"genes"))+
  theme(plot.title = element_text(size = rel(1), hjust = 0.5))
# saveRDS(EPIvenn,"output/EPIvenn.RDS")
#
# Mito comp ---------------------------------------------------------------
Mitocomp <- list(sigVMT24$ENTREZID,sigVMT3$ENTREZID)
in_commonMt <-c(sigVMT24$ENTREZID,sigVMT3$ENTREZID)
length(unique(in_commonMt))

# MTXvenn<-
  ggVennDiagram(Mitocomp,
              category.names = c("MTX-24","MTX-3"),
              show_intersect = FALSE,
              set_color = "black",
              label = "count",
              label_percent_digit = 1,
              label_size = 4,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = )+
  scale_x_continuous(expand = expansion(mult = .2))+
  scale_fill_gradient(low = "light blue", high = "yellow")+
  labs(title = "Comparison of Mito 3h v 24h", 
       caption = paste("n =", length(unique(in_commonMt)),"genes"))+
  theme(plot.title = element_text(size = rel(1), hjust = 0.5))

# saveRDS(MTXvenn,"output/MTXvenn.RDS")

```

## GO analysis on 24 hour Venn, adj. p value < 0.05

```{r Making Go lists from SigDE genes, message=FALSE, warning=FALSE, paged.print=TRUE}

# > list24totvenn$..count..
#  [1]  977   15   17   14   58   24   34  188 4178  225  287  489  697  342  616
# > list24totvenn$..values..[?]

##pairwise
DDEresp<- list24totvenn24$..values..[[9]]
DDEresp3<- list3totvenn3$..values..[[9]]

#DDEresp1<- read.csv("data/DDEresp_list.csv",row.names = 1)
# write.csv(DDEresp,"data/DDEresp_list.csv")
DDEMresp<- list24totvenn24$..values..[[1]]
DDEMresp3<- list3totvenn3$..values..[[1]]
# DDEMresp<-read.csv("data/DDEMresp_list.csv", row.names = 1)
# write.csv(DDEMresp,"data/DDEMresp_list.csv")
# DDEall <-(intersect(sigVEP24$ENTREZID,intersect(sigVDA24$ENTREZID,sigVDX24$ENTREZID)))
# length(intersect(DDEresp,sigVMT24))
# 
# print("Number of AC drug DEG genes in common") 
# length(intersect(sigVEP24$ENTREZID,intersect(sigVDA24$ENTREZID,sigVDX24$ENTREZID)))
# #5317
DXsprespon<- list24totvenn24$..values..[[14]]
## (342)
print("Number of DOX specific DEGs, adj. P value < 0.05") 
length(DXsprespon)
complete <- c(sigVDA24$ENTREZID,sigVDX24$ENTREZID,sigVEP24$ENTREZID,sigVMT24$ENTREZID, sigVDA3$ENTREZID, sigVDX3$ENTREZID, sigVEP3$ENTREZID, sigVMT3$ENTREZID)
complete <- as.data.frame(unique(complete))
colnames(complete) <- "ENTREZID"  #9047 genes
print("Number of All genes that respond to some condition in DEG set, adj. P value  < 0.05")
length(complete$ENTREZID)
#8368
      
NoRespDEG <- setdiff(backGL$ENTREZID ,complete$ENTREZID) #
print("Number of No response genes in DEG set, adj. P value  < 0.05")
length(NoRespDEG) 
# write.csv(NoRespDEG,"data/NoRespDEG_final.csv")
 #5716
```
## DDE response set

adj. P value  < 0.05, intersection of DNR-DOX-EPI, excluding those in common with MTX, n =
`r length(DDEresp)` genes.

no significant enrichment found!  

AC specific enrichment at 24 hours did not occur.  The next step was Top2 inhibitior enrichment.  This would be the intersection of all Top2bi drugs. This was `r length( DDEMresp)` genes long.  Below is that analysis.  

```{r GO DDEresp 3 hour genelist of SDEGs,  message=FALSE, warning=FALSE, include=FALSE}
##No enrichment for DDE only resp!

##overlap of DDEM + AC only gives: 

DDEresp3All <- (union(DDEresp3,DDEMresp3))
# (just the overlap of DDE, i use the dataframe DDEresp3)
# DDEgostres3only <- gost(query = DDEresp3,
#                      organism = "hsapiens",
#                     significant = FALSE,
#                      ordered_query = TRUE,
#                      domain_scope = "custom",
#                      measure_underrepresentation = FALSE,
#                      evcodes = FALSE,
#                      user_threshold = 0.05,
#                      correction_method = c("fdr"),
#                      custom_bg = backGL$ENTREZID,
#                      sources=c("GO:BP","KEGG"))
# # 
# 
# saveRDS(DDEgostres3,"data/DEG-GO/DDE_allgostres3.RDS")
DDEgostres3 <- readRDS("data/DEG-GO/DDE_allgostres3.RDS")
# saveRDS(DDEgostres3only,"data/DEG-GO/DDEgostres3only.RDS")
DDEgostres3only <- readRDS("data/DEG-GO/DDEgostres3only.RDS")  
# saveRDS(DDEgostres24kegg,"data/DEG-GO/DDEgostres24kegg.RDS")

DDEp3 <- gostplot(DDEgostres3, capped = FALSE, interactive = TRUE)
DDEp3
DDEp3only <- gostplot(DDEgostres3only, capped = FALSE, interactive = TRUE)
DDEp3only

# kegg_term <- c("p53 signaling pathway","DNA replication","Cell cycle" ,"Base excision repair")  


DDEtable3 <- DDEgostres3$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) 
DDEtable3only<- DDEgostres3only$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) 


DDEtable3only %>% dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    dplyr::slice_min(., n=5 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('Anthracycline DEGs 3 hr GO:BP terms (not significantly enriched)') +
    xlab(expression(" -"~log[10]~"(adj. p-value)"))+
    ylab("GO: BP term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.2), hjust = 0.5),
        axis.title = element_text(size = 12, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 8, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))

DDEtable3only %>% dplyr::filter(source=="KEGG") %>%
    dplyr::select(p_value,term_name,intersection_size) %>%
  
    slice_min(., n= 10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
     geom_point(aes(size = intersection_size, col = "red")) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('Anthracycline DEGs 3 hr KEGG pathway terms ') +
    xlab(expression(" -"~log[10]~"(adj. p-value)"))+
    ylab("KEGG term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.2), hjust = 0.5),
        axis.title = element_text(size = 12, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 8, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))


  
  DDEtable3 %>% 
    mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,caption="DDE all overlapped genes (including those that overlapped with MTX),n = 11+ n = 8") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")

  DDEtable3only %>% 
    mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,caption="DDE only the overlapped n= 11") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")
  
``` 



```{r GO DDEresp 24 hour genelist of SDEGs,  message=FALSE, warning=FALSE, include=FALSE}
##No enrichment for DDE only resp!

##overlap of DDEM + AC only gives: 

DDErespAll <- (union(DDEresp,DDEMresp))## for the "24 hour all version (4435+882)
# DDEgostres24only <- gost(query = DDEresp,
#                      organism = "hsapiens", 
#                      significant = FALSE,
#                      ordered_query = TRUE,
#                      domain_scope = "custom",
#                      measure_underrepresentation = FALSE,
#                      evcodes = FALSE,
#                      user_threshold = 0.05,
#                      correction_method = c("fdr"),
#                      custom_bg = backGL$ENTREZID,
#                      sources=c("GO:BP","KEGG"))
# saveRDS(DDEgostres24all,"data/DEG-GO/DDE_allgostres24.RDS")
DDEgostres24all <- readRDS("data/DEG-GO/DDE_allgostres24.RDS")
# saveRDS(DDEgostres24only,"data/DEG-GO/DDEgostres24only.RDS")
DDEgostres24only <- readRDS("data/DEG-GO/DDE_allgostres24.RDS")
# saveRDS(DDEgostres24kegg,"data/DEG-GO/DDEgostres24kegg.RDS")

DDEp24all <- gostplot(DDEgostres24all, capped = FALSE, interactive = TRUE)
DDEp24all
DDEp24only <- gostplot(DDEgostres24only, capped = FALSE, interactive = TRUE)
DDEp24only



kegg_term <- c("p53 signaling pathway","DNA replication","Cell cycle" ,"Base excision repair")


DDEtable24only <- DDEgostres24only$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) 
DDEtable24all<- DDEgostres24all$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) 


DDEtable24only %>% dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    dplyr::slice_min(., n= 5 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   slice_max(., n=10,order_by = log_val) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('Anthracycline DEGs 24 hr GO:BP terms, n = 4435 genes') +
    xlab(expression(" -"~log[10]~"(adj. p-value)"))+
    ylab("GO: BP term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.2), hjust = 0.5),
        axis.title = element_text(size = 12, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 8, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))

DDEtable24only %>% dplyr::filter(source=="KEGG") %>%
    dplyr::select(p_value,term_name,intersection_size) %>%
  filter(term_name %in% kegg_term) %>% 
    slice_min(., n= 10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
     geom_point(aes(size = intersection_size, col = "red")) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('Anthracycline DEGs 24 hr KEGG pathway terms') +
    xlab(expression(" -"~log[10]~"(adj. p-value)"))+
    ylab("KEGG term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.2), hjust = 0.5),
        axis.title = element_text(size = 12, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 8, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))


  
  DDEtable24only %>% 
    mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,caption = "24 hour DDE only enrichment values") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")

``` 

of the AC genes that overlapped without Mito (n= `r length(DDEresp)`
and all AC gens plus those which overlapped with Mito (n= `r length(DDEMresp)`, the significantly  enriched GO: BP terms are the same.  This may support the idea that the specific AC set can tell us more about mechanism in AC induced cardiotoxicity.
## DDEM 24 hour response of `r length(DDEMresp3)`

```{r GO DDEM 3 hourlist of SDEGs, echo=FALSE, message=FALSE, warning=FALSE}


# DDEMgostres3 <- gost(query = DDEMresp3,
#                     organism = "hsapiens",
#                     significant = FALSE,
#                     ordered_query = TRUE,
#                     domain_scope = "custom",
#                     measure_underrepresentation = FALSE,
#                     evcodes = FALSE,
#                     user_threshold = 0.05,
#                     correction_method = c("fdr"),
#                     custom_bg = backGL$ENTREZID,
#                     sources=c("GO:BP","KEGG"))
# 
# saveRDS(DDEMgostres3,"data/DEG-GO/DDEMgostres3.RDS")
DDEMgostres3 <- readRDS("data/DEG-GO/DDEMgostres3.RDS")

DDEMp3 <- gostplot(DDEMgostres3, capped = FALSE, interactive = TRUE)
DDEMp3

DDEMtable3 <- DDEMgostres3$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) 

DDEMtable3 %>% dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n= 10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('Top2i DEGs (n=8) 3 hr GO:BP terms') +
    xlab(expression(" -"~log[10]~"(adj. p-value)"))+
    ylab("GO: BP term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.2), hjust = 0.5),
        axis.title = element_text(size = 12, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 8, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))

# 
# DDEMtable3 %>% dplyr::filter(source=="KEGG") %>% 
#     dplyr::select(p_value,term_name,intersection_size) %>%
#     slice_min(., n= 10 ,order_by=p_value) %>%
#     mutate(log_val = -log10(p_value)) %>%
#    # slice_max(., n=10,order_by = p_value) %>%
#    ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
#     geom_point(aes(size = intersection_size, col = "red")) +
#     scale_y_discrete(labels = label_wrap(30))+
#     guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
#     ggtitle('Top2 inhibitor DEGs (n=8) 3 hr KEGG terms') +
#     xlab(expression(" -"~log[10]~"(adj. p-value)"))+
#     ylab("KEGG term")+
#       theme_bw()+
#     theme(plot.title = element_text(size = rel(1.2), hjust = 0.5),
#         axis.title = element_text(size = 12, color = "black"),
#         axis.ticks = element_line(linewidth = 1.5),
#         axis.line = element_line(linewidth = 1.5),
#         axis.text = element_text(size = 8, color = "black", angle = 0),
#         strip.text.x = element_text(size = 15, color = "black", face = "bold"))
  
  DDEMtable3 %>% 
    mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,caption = "Top2 inhibitor enrichment list, n=8") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")
  
  
  
  
  
```
## DDEM 24 hour response of `r length(DDEMresp)`

```{r GO DDEM 24 hour list of SDEGs, echo=FALSE, message=FALSE, warning=FALSE}


# DDEMgostres24 <- gost(query = DDEMresp,
#                     organism = "hsapiens", 
#                     significant = FALSE,
#                     ordered_query = TRUE,
#                     domain_scope = "custom",
#                     measure_underrepresentation = FALSE,
#                     evcodes = FALSE,
#                     user_threshold = 0.05,
#                     correction_method = c("fdr"),
#                     custom_bg = backGL$ENTREZID,
#                     sources=c("GO:BP","KEGG"))
# 
# saveRDS(DDEMgostres24,"data/DEG-GO/DDEMgostres24.RDS")
DDEMgostres24 <- readRDS("data/DEG-GO/DDEMgostres24.RDS")

DDEMp24 <- gostplot(DDEMgostres24, capped = FALSE, interactive = TRUE)
DDEMp24

DDEMtable24 <- DDEMgostres24$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) 

DDEMtable24 %>% dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n= 10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('Top2i DEGs (n=832) 24 hr GO:BP terms') +
    xlab(expression(" -"~log[10]~"(adj. p-value)"))+
    ylab("GO: BP term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.2), hjust = 0.5),
        axis.title = element_text(size = 12, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 8, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))


DDEMtable24 %>% dplyr::filter(source=="KEGG") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n= 10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size, col = "red")) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('Top2 inhibitor DEGs (n=832) 24 hr KEGG terms') +
    xlab(expression(" -"~log[10]~"(adj. p-value)"))+
    ylab("KEGG term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.2), hjust = 0.5),
        axis.title = element_text(size = 12, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 8, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))
  
  DDEMtable24 %>% 
    mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,caption = "Top2 inhibitor enrichment list, n=832") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")
  
```

What I will do is take the terms from the top three KEGG pathways and look at count changes or at FC values.  Updated later.
 

## DXspecific response (only the subset unique to Dox)


```{r GO DXsprespon$ENTREZID genelist of SDEGs, echo=FALSE, message=FALSE, warning=FALSE}


# DXspgostres24 <- gost(query = DXsprespon,
#                     organism = "hsapiens", significant = FALSE,
#                     ordered_query = TRUE,
#                     domain_scope = "custom",
#                     measure_underrepresentation = FALSE,
#                     evcodes = FALSE,
#                     user_threshold = 0.05,
#                     correction_method = c("fdr"),
#                     custom_bg = backGL$ENTREZID,
#                     sources=c("GO:BP","KEGG"))
# 
# saveRDS(DXspgostres24,"data/DEG-GO/DXspgostres24.RDS")
DXspgostres24 <- readRDS("data/DEG-GO/DXspgostres24.RDS")


DXspp24 <- gostplot(DXspgostres24, capped = FALSE, interactive = TRUE)
DXspp24

DXsptable24 <- DXspgostres24$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) 

DXsptable24 %>% dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n= 10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = scales::label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle(paste0("DOX specific DEGs (n = ",length(DXsprespon),") 24 hr GO:BP terms")) +
   xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.2), hjust = 0.5),
        axis.title = element_text(size = 12, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 8, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))

# 
# DXsptable24 %>% dplyr::filter(source=="KEGG") %>%
#     dplyr::select(p_value,term_name,intersection_size) %>%
#     slice_min(., n= 10 ,order_by=p_value) %>%
#     mutate(log_val = -log10(p_value)) %>%
#    # slice_max(., n=10,order_by = p_value) %>%
#    ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
#     geom_point(aes(size = intersection_size, col = "red")) +
#     scale_y_discrete(labels = label_wrap(30))+
#     guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
#     ggtitle('DOX specific DEGs (n=400) 24 hr KEGG terms') +
#     xlab(expression(" -"~log[10]~("adj. p-value")))+
#     ylab("KEGG term")+
#       theme_bw()+
#     theme(plot.title = element_text(size = rel(1.2), hjust = 0.5),
#         axis.title = element_text(size = 12, color = "black"),
#         axis.ticks = element_line(linewidth = 1.5),
#         axis.line = element_line(linewidth = 1.5),
#         axis.text = element_text(size = 8, color = "black", angle = 0),
#         strip.text.x = element_text(size = 15, color = "black", face = "bold"))
  
  DXsptable24 %>% 
    mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,) %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")
  



``` 
 
 
  As above, I will pull the specific genes and look at logCPM of these genes soon to examine if the change is seen in the other AC genes at 24 hours.


## No Response genes n=5727
 

```{r no response cluster, echo=FALSE, message=FALSE, warning=FALSE}
# 
# gostresNoResp <- gost(query = NoRespDEG,  
#                     organism = "hsapiens", significant = FALSE,
#                     ordered_query = FALSE,
#                     domain_scope = "custom",
#                     measure_underrepresentation = FALSE,
#                     evcodes = FALSE,
#                     user_threshold = 0.05,
#                     correction_method = c("fdr"),
#                     custom_bg = backGL$ENTREZID,
#                     sources=c("GO:BP","KEGG"))
# 
#saveRDS(gostresNoResp,"data/DEG-GO/gostresNoResp.RDS")
gostresNoResp <- readRDS("data/DEG-GO/gostresNoResp.RDS")

Noresp24 <- gostplot(gostresNoResp, capped = FALSE, interactive = TRUE)
Noresp24

Noresp24table <- gostresNoResp$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) 

Noresp24table %>% dplyr::filter(source=="GO:BP") %>%
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n= 10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
     ggtitle(paste0("No response genes(n = ",length(NoRespDEG),") GO:BP terms")) +
    xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.2), hjust = 0.5),
        axis.title = element_text(size = 12, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 8, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))


Noresp24table %>% dplyr::filter(source=="KEGG") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n= 10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size, col = "red")) +
    scale_y_discrete(labels = label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle(paste0("No response genes(n = ",length(NoRespDEG),") 24 hr KEGG terms")) +
     xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("KEGG term")+
      theme_bw()+
    theme(plot.title = element_text(size = rel(1.2), hjust = 0.5),
        axis.title = element_text(size = 12, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 8, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))
  
  Noresp24table %>% 
    mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(.,) %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")
  

  
```
##GO:BP full figure 24 DEG analysis


```{r GOBP dEGlong, echo=FALSE, fig.height=12, fig.width=8,message=FALSE, warning=FALSE}

DDEdeg <- DDEtable24only %>% dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=5 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
  mutate(order_val=rev(as.numeric(rownames(.)))) 

DDEMdeg <-  DDEMtable24 %>% dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=5 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
  mutate(order_val=rev(as.numeric(rownames(.))))

DAsp3deg <- DAsptable3%>% 
  dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=5 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
  mutate(order_val=rev(as.numeric(rownames(.))))

DXsp24deg <- DXsptable24 %>% 
  dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=5 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
  mutate(order_val=rev(as.numeric(rownames(.))))

# EPsp24deg <- EPsptable124 %>%
#   dplyr::filter(source=="GO:BP") %>%
#     dplyr::select(p_value,term_name,intersection_size) %>%
#     slice_min(., n=10 ,order_by=p_value) %>%
#     mutate(log_val = -log10(p_value)) %>%
#   mutate(order_val=rev(as.numeric(rownames(.))))

MTsp24deg <-   MTsptable124 %>% 
    dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=5 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
  mutate(order_val=rev(as.numeric(rownames(.))))
  

DAdeg <-   DAtable124 %>% dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=5 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
  mutate(order_val=rev(as.numeric(rownames(.))))

DXdeg <-  DXtable124 %>% dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=5 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
  mutate(order_val=rev(as.numeric(rownames(.))))

  EPdeg <-   EPtable124 %>% dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=5 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
    mutate(order_val=rev(as.numeric(rownames(.))))
  
  MTdeg <-   MTtable124 %>% dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=5 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
    mutate(order_val=rev(as.numeric(rownames(.))))

#filter<- dplyr::filter 
#select <- dplyr::select
  
GODEGlong <-list("DXdeg"=DXdeg,
                 "DXsp24deg"=DXsp24deg,
                 "DAdeg"=DAdeg,
                 "EPdeg"=EPdeg,
                 "MTdeg"=MTdeg,
                 "MTsp24deg"= MTsp24deg,
                 "DDEdeg"=DDEdeg,
                 "DDEMdeg"=DDEMdeg)

GODEGlongtable <- data.table::rbindlist(GODEGlong, idcol = "deg")
 p <- GODEGlongtable%>% 
   mutate(deg = fct_inorder(deg)) %>% 
ggplot(., aes(x = log_val, y=reorder(term_name, order_val, desc=TRUE),col=deg)) +
   geom_point(aes(size = intersection_size, col=deg)) +
   scale_y_discrete(labels = label_wrap(40))+
   scale_color_manual(values=c("DXdeg"="#8B006D",
                 "DXsp24deg"="#b0028a",
                 "DAdeg"="#DF707E",
                 "EPdeg"="#F1B72B",
                 "MTdeg"="#33a1dd",
                 "MTsp24deg"="#33a1dd",
                 "DDEdeg"="red",
                 "DDEMdeg"="blue", alpha= 0.8))+
  facet_grid(deg ~.,scales="free_y" , labeller=label_bquote(.(as.expression(
               eval(parse(text = paste0('facet_deg', '$`', deg, '`')))
               )))) +
  
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('Top enriched GO:BP terms for each 24 hour DEG list') +
    xlab("-log 10 (adj. p-value)")+
    
    ylab("GO:BP term")+
      theme_bw()+
 theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 8, color = "black", angle = 0),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

 
 
 facet_deg=(
   c(DXdeg = expression(atop("DOX 24 hr",~italic("n = 6645"))),
     DXsp24deg =expression(atop("DOX-specific 24 hr",~italic("n = 355"))),
     DAdeg = expression(atop("DNR 24 hr",~italic("n = 7017"))),
     EPdeg = expression(atop("EPI 24 hr",~italic("n = 6328"))),
     MTdeg = expression(atop( "MTX 24 hr",~italic("n = 1115"))),
     MTsp24deg =expression(atop( "MTX-specific 24 hr",~italic("n = 122"))),
     DDEdeg = expression(atop("AC 24 hr",~italic("n = 4435,"))),
     DDEMdeg = expression(atop( "TOP2i 24 hr",~italic("n = 882")))))

#  title_labeller_function <- function(nuisance_parameter) {
#   return(facet_degnames() )
# }
 
 
#https://stackoverflow.com/questions/41631806/change-facet-label-text-and-background-colour/60046113#60046113
 
# drug_palNoVeh <- c("#8B006D","#DF707E","#F1B72B", "#3386DD","#707031","#41B333")
 
 
g <- ggplot_gtable(ggplot_build(p))
stripr <- which(grepl('strip-r', g$layout$name))
fills <- c("#8B006D","#b0028a","#DF707E","#F1B72B","#3386DD","#33a1dd","red", "blue", alpha= 0.8)
k <- 1
for (i in stripr) {
  j <- which(grepl('rect', g$grobs[[i]]$grobs[[1]]$childrenOrder))
  g$grobs[[i]]$grobs[[1]]$children[[j]]$gp$fill <- fills[k]
  k <- k+1
}
print(grid::grid.draw(g))
grid::grid.draw(g)
#hex <- hue_pal()(4)

#hex
#> hex (red/green/blue/purple default in ggplot)
#[1] "#F8766D" "#7CAE00" "#00BFC4" "#C77CFF"
#p



```



## CVRI poster figures not run

## Visualization of FC in cormotif sets
```{r Correlation of FC values in all values and in DEG}
library(limma)

library(pheatmap)

efit2<- readRDS(file ="data/efit2_final.RDS")


FCmatrix <- subset(efit2$coefficients)
is.matrix(subset(FCmatrix))
colnames(FCmatrix) <- c("DNR-3h","DOX-3h", "EPI-3h","MTX-3h", "TRZ-3h","DNR-24h","DOX-24h", "EPI-24h","MTX-24h", "TRZ-24h")


mat_col <- data.frame(time= c(rep("3 hours",5), rep("24 hours",5)))
rownames(mat_col) <- colnames(FCmatrix)
# list(group=c("#C25757", "#A77E69")
# names(mat_colors$group) <-c("3 hours", "24 hours")
mat_colors <- list(time=c("#C25757", "#A77E69"))
names(mat_colors$time) <- unique(mat_col$time)
corrFC <- cor(FCmatrix)
# corrplot(corrFC, method = 'square', order = 'AOE', addCoef.col = 'black')
# corrplot(corrFC,type = 'full', order = 'hclust', tl.col = 'black',
#          cl.ratio = 0.2, tl.srt = 45, col = COL2('PuOr', 20), col.lim=c(-0.1,1), addrect=3)


ComplexHeatmap::pheatmap(corrFC, 
                         display_numbers = TRUE, 
                         number_format = "%.2f",
                         main=("Correlation of FC values across all expressed genes, n=14084"),
                         annotation_col = mat_col,
                         annotation_colors = mat_colors)




#making of the side columns for class
```


```{r Poster DEG GOBP, eval=FALSE, include=FALSE}
DDEdeg <- DDEtable24only %>% dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=6 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%mutate(order_val=rev(as.numeric(rownames(.)))) %>% slice_max(., n=10, order_by=order_val)

DDEMdeg <-  DDEMtable24 %>% dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%mutate(order_val=rev(as.numeric(rownames(.))))
DAsp3deg <- DAsptable3%>% 
  dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
  mutate(order_val=rev(as.numeric(rownames(.))))
MTsp24 <- MTsptable124 %>% 
  dplyr::filter(source=="GO:BP") %>% 
  dplyr::select(p_value,term_name,intersection_size) %>%
  slice_min(., n=10 ,order_by=p_value) %>%
  mutate(log_val = -log10(p_value)) %>%
  mutate(order_val=rev(as.numeric(rownames(.))))





# GODEGposter <-list("DAsp3deg"=DAsp3deg,"DDEdeg"=DDEdeg,"DDEMdeg"=DDEMdeg)
GODEGposter <-list("MTsp24"=MTsp24,"DDEdeg"=DDEdeg,"DDEMdeg"=DDEMdeg)
GODEGpostertable <- data.table::rbindlist(GODEGposter, idcol = "deg")
facet_degp=(
   c(MTsp24 = expression(atop("MTX-specific 24 hours",~italic("n = 126"))),
    DDEdeg=expression(atop("Shared Anthracycline genes 24 hours",~italic("n = 4400"))),
     DDEMdeg=expression(atop("Shared Top2"*beta*"i genes 24 hours",~italic("n = 952")))))

DEG_posterp <- GODEGpostertable%>% mutate(deg = fct_inorder(deg)) %>% 
  ggplot(., aes(x = log_val, y=reorder(term_name, order_val, desc=TRUE),col=deg)) +
    geom_point(aes(size = intersection_size, col=deg)) +
  scale_y_discrete(labels = label_wrap(40))+
   scale_color_manual(values=c("#3386DD","red3", "orange"))+
   facet_grid(deg ~.,scales="free_y" , labeller=label_bquote(.(as.expression(
               eval(parse(text = paste0('facet_degp', '$`', deg, '`')))
               )))) +
  guides(color= "none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('Top 10 enriched GO:BP terms\n in specific overlaping sets') +
     xlab(expression(" -"~log[10]~"(adj. p-value)"))+
      ylab("GO:BP term")+
      theme_bw()+
 theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 8, color = "black", angle = 0),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "black"))

 
 g <- ggplot_gtable(ggplot_build(DEG_posterp))
stripr <- which(grepl('strip-r', g$layout$name))
fills <- c("#3386DD","red3", "orange", alpha= 0.8)
k <- 1
for (i in stripr) {
  j <- which(grepl('rect', g$grobs[[i]]$grobs[[1]]$childrenOrder))
  g$grobs[[i]]$grobs[[1]]$children[[j]]$gp$fill <- fills[k]
  k <- k+1
}
 

plot_grid(g)

```


###  KEGG pathway intersection of pairwise DEG sets


note:  grey box means these terms were not found to be enriched in the data set terms.  I will try to change the levels to pull out the pvalue of the term for each (6-30-23)

```{r kegg 24 hour DEG heat, message=FALSE, warning=FALSE}
# kegglistDEG <- list(DDEdegk,DDEMdegk,DAdegk,DXdegk,EPdegk, MTdegk)
# names(kegglistDEG) <-c("DDEdegk","DDEMdegk","DAdegk","DXdegk","EPdegk", "MTdegk")
# saveRDS(kegglistDEG,"data/kegglistDEG24.RDS")
kegglistDEG <- readRDS("data/kegglistDEG24.RDS")
list2env(kegglistDEG, envir=.GlobalEnv)
kegg_term24 <- data.frame(term=c("p53 signaling pathway","DNA replication","Cell cycle" ,"Base excision repair"),termid=c("KEGG:04115","KEGG:03030","KEGG:04110","KEGG:03410"))

DDEdegk <- DDEtable24only %>% dplyr::filter(source=="KEGG") %>% 
  dplyr::select(p_value,term_id,term_name,intersection_size) %>%
  filter(term_id %in% kegg_term24$termid) %>% 
    # slice_min(., n=4 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
  mutate(order_val=rev(as.numeric(rownames(.)))) 

DDEMdegk <-  DDEMtable24 %>% dplyr::filter(source=="KEGG") %>% 
    dplyr::select(p_value,term_id,term_name,intersection_size) %>%
  filter(term_id %in% kegg_term24$termid) %>% 
    # slice_min(., n=3 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
  mutate(order_val=rev(as.numeric(rownames(.))))

DAdegk <-   DAtable124 %>% dplyr::filter(source=="KEGG") %>% 
    dplyr::select(p_value,term_id,term_name,intersection_size) %>%
  filter(term_id %in% kegg_term24$termid) %>% 
    slice_min(., n=3 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
  mutate(order_val=rev(as.numeric(rownames(.))))

 DXdegk <-  DXtable124 %>%
   dplyr::filter(source=="KEGG") %>% 
    dplyr::select(p_value,term_id,term_name,intersection_size) %>%
  filter(term_id %in% kegg_term24$termid) %>% 
    # slice_min(., n=3 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
  mutate(order_val=rev(as.numeric(rownames(.))))

  EPdegk <-   EPtable124 %>% dplyr::filter(source=="KEGG") %>% 
    dplyr::select(p_value,term_id,term_name,intersection_size) %>%
  filter(term_id %in% kegg_term24$termid) %>% 
    # slice_min(., n=3 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
    mutate(order_val=rev(as.numeric(rownames(.))))
  
  MTdegk <-   MTtable124 %>%
    dplyr::filter(source=="KEGG") %>% 
    dplyr::select(p_value,term_id,term_name,intersection_size) %>%
  filter(term_id %in% kegg_term24$termid) %>% 
    # slice_min(., n=3 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
    mutate(order_val=rev(as.numeric(rownames(.))))
library(ComplexHeatmap)
  
keggDEGlong <-list("DOX"=DXdegk,
                 "EPI"=EPdegk,
                 "DNR"=DAdegk,
                 "MTX"=MTdegk,
                 "ACs"=DDEdegk,
                 "TOP2i"=DDEMdegk)

keggtable <- data.table::rbindlist(keggDEGlong, idcol = "deg")
kegg_sig_mat <- keggtable %>% 
  dplyr::select(deg,p_value,term_name) %>%
  pivot_wider(id_cols = everything(),
              names_from="term_name",
              values_from="p_value",
              values_fill = list(p_value = 1)) %>% 
  column_to_rownames('deg') %>% 
  as.matrix()# 

kegg_mat<- keggtable%>%
  dplyr::select(deg,log_val,term_name) %>% 
  pivot_wider(id_cols = everything(),
              names_from="term_name",values_from="log_val") %>% 
  column_to_rownames('deg') %>% 
  as.matrix()# 

Heatmap(kegg_mat,
        column_title = "KEGG Pathway -log p values",
        name = "-log (p value)",
        cluster_rows = FALSE, 
        cluster_columns = FALSE,
        cell_fun = function(j, i, x, y, width, height, fill) {
        if(kegg_sig_mat[i, j]< 0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})
  
```


```{r kegg 3 hour DEG heat, message=FALSE, warning=FALSE}
# kegglistDEG3 <- list(DDEdeg3k,DDEMdeg3k,DAdeg3k,DXdeg3k,EPdeg3k, MTdeg3k)
# names(kegglistDEG3) <-c("DDEdeg3k","DDEMdeg3k","DAdeg3k","DXdeg3k","EPdeg3k", "MTdeg3k")
# saveRDS(kegglistDEG3,"data/kegglistDEG3.RDS")
# kegglistDEG3 <- readRDS("data/kegglistDEG3.RDS")
# list2env(kegglistDEG, envir=.GlobalEnv)
kegg_term3 <- data.frame(term=c("p53 signaling pathway","Herpes simplex virus 1 infection"),termid=c("KEGG:04115","KEGG:05168"))

DDEdeg3k <- DDEtable3only %>% 
  dplyr::filter(source=="KEGG") %>% 
    dplyr::select(p_value,term_id,term_name,intersection_size) %>%
  filter(term_id %in% kegg_term3$termid) %>% 
    mutate(log_val = -log10(p_value)) %>%
  mutate(order_val=rev(as.numeric(rownames(.)))) 

DDEMdeg3k <-  DDEMtable3 %>% 
  dplyr::filter(source=="KEGG") %>% 
    dplyr::select(p_value,term_id,term_name,intersection_size) %>%
  filter(term_id %in% kegg_term3$termid) %>% 
    mutate(log_val = -log10(p_value)) %>%
  mutate(order_val=rev(as.numeric(rownames(.))))

DAdeg3k <-   DAtable3 %>% 
  dplyr::filter(source=="KEGG") %>% 
    dplyr::select(p_value,term_id,term_name,intersection_size) %>%
  filter(term_id %in% kegg_term3$termid) %>% 
    mutate(log_val = -log10(p_value)) %>%
  mutate(order_val=rev(as.numeric(rownames(.))))

DXdeg3k <-  DXtable1 %>% 
  dplyr::filter(source=="KEGG") %>% 
    dplyr::select(p_value,term_id,term_name,intersection_size) %>%
  filter(term_id %in% kegg_term3$termid) %>% 
    mutate(log_val = -log10(p_value)) %>%
  mutate(order_val=rev(as.numeric(rownames(.))))

  EPdeg3k <-   EPtable1 %>% 
    dplyr::filter(source=="KEGG") %>% 
    dplyr::select(p_value,term_id,term_name,intersection_size) %>%
  filter(term_id %in% kegg_term3$termid) %>% 
    slice_min(., n=3 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
    mutate(order_val=rev(as.numeric(rownames(.))))
  
  MTdeg3k <-   MTtable1 %>% 
    dplyr::filter(source=="KEGG") %>% 
    dplyr::select(p_value,term_id,term_name,intersection_size) %>%
  filter(term_id %in% kegg_term3$termid) %>% 
    slice_min(., n=3 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
    mutate(order_val=rev(as.numeric(rownames(.))))
library(ComplexHeatmap)
  
keggDEGlong3 <-list("DOX"=DXdeg3k,
                 "EPI"=EPdeg3k,
                 "DNR"=DAdeg3k,
                 "MTX"=MTdeg3k,
                 "ACs"=DDEdeg3k,
                 "TOP2i"=DDEMdeg3k)

keggtable3 <- data.table::rbindlist(keggDEGlong3, idcol = "deg")
kegg_sig_mat3 <- keggtable3 %>% 
  dplyr::select(deg,p_value,term_name)%>%
  add_row(deg ="TOP2i", p_value = 1,term_name="Herpes simplex virus 1 infection") %>% 
  pivot_wider(id_cols = everything(),
              names_from="term_name",
              values_from="p_value",
              values_fill = list(p_value = 1)) %>% 
  column_to_rownames('deg') %>% 
  as.matrix()# 

kegg_mat3<- keggtable3%>%
  dplyr::select(deg,log_val,term_name) %>%
  add_row(deg ="TOP2i", log_val = NA,term_name="Herpes simplex virus 1 infection") %>% 
  pivot_wider(id_cols = everything(),
              names_from="term_name",values_from="log_val") %>% 
  column_to_rownames('deg') %>% 
  as.matrix()# 

Heatmap(kegg_mat3,
        column_title = "KEGG Pathway -log p values",
        name = "-log (p value)",
        cluster_rows = FALSE, 
        cluster_columns = FALSE,
        na_col="grey",
        cell_fun = function(j, i, x, y, width, height, fill) {
        if(kegg_sig_mat3[i, j]< 0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})
  
```
