---
title: "Figure 9"
author: "ERM"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE
)
```

```{r packages loading}
library(edgeR)
library(tidyverse)
library(ggsignif)
library(RColorBrewer)
library(cowplot)
library(ggpubr)
library(scales)
library(sjmisc)
library(kableExtra)
library(broom)
library(ComplexHeatmap)
```
```{r data loading}
toplistall <- readRDS("data/toplistall.RDS")
siglist <- readRDS("data/siglist_final.RDS")
list2env(siglist, envir = .GlobalEnv)
cpmcounts <- readRDS("data/cpmcount.RDS")
backGL <- read.csv("data/backGL.txt",     row.names =1)

drug_palc <- c("#8B006D","#DF707E","#F1B72B", "#3386DD","#707031","#41B333")
drug_pal_fact <- c("#8B006D" ,"#DF707E", "#F1B72B" ,"#3386DD", "#707031","#41B333")

Arr_geneset <- readRDS("data/Arr_geneset.RDS")
HF_geneset <- readRDS("data/HF_geneset.RDS")
CAD_geneset <- readRDS("data/CAD_geneset.RDS")
col_fun1 = circlize::colorRamp2(c(-1, 3), c("white", "purple"))
col_funFC= circlize::colorRamp2(c(-2,0, 2), c("darkgreen","white", "darkorange2"))

```
```{r function load}

pearson_extract <- function(corr_df,ENTREZID) {


  ld50_plot <- corr_df %>%
    dplyr::filter(entrezgene_id == ENTREZID) %>%
    ggplot(., aes(x=LD50, y=counts))+
    geom_point(aes(col=indv))+
    geom_smooth(method="lm")+
    facet_wrap(hgnc_symbol~Drug, scales="free")+
    theme_classic()+
    stat_cor(method="pearson",
             aes(label = paste(..r.label.., ..p.label.., sep = "~`,`~")),
             color = "red",
             label.x.npc = 0,
             label.y.npc=1,
             size = 3)

  tnni_plot <-  corr_df %>%
    dplyr::filter(entrezgene_id == ENTREZID) %>%
    ggplot(., aes(x=rtnni, y=counts))+
    geom_point(aes(col=indv))+
    geom_smooth(method="lm")+
    facet_wrap(hgnc_symbol~Drug, scales="free")+
    theme_classic()+
    stat_cor(method="pearson",
             aes(label = paste(..r.label.., ..p.label.., sep = "~`,`~")),
             color = "red",
             label.x.npc = 0,
             label.y.npc=1,
             size = 3)

  ldh_plot <-   corr_df %>%
    dplyr::filter(entrezgene_id == ENTREZID) %>%
    ggplot(., aes(x=rldh, y=counts))+
    geom_point(aes(col=indv))+
    geom_smooth(method="lm")+
    facet_wrap(hgnc_symbol~Drug, scales="free")+
    theme_classic()+
    stat_cor(method="pearson",
             aes(label = paste(..r.label.., ..p.label.., sep = "~`,`~")),
             color = "red",
             label.x.npc = 0,
             label.y.npc=1,
             size = 3)

  ##ggbuild to get model:
  ld50_build <- ggplot_build(ld50_plot)
  ld50_data <- data.frame('rho_LD50'= c(ld50_build$data[[3]]$r,NA,NA),
                          'sig_LD50'=c(ld50_build$data[[3]]$p.value,NA,NA),
                          'rowname'=c("DOX","EPI","DNR","MTX", "TRZ", "VEH"))

  tnni_build <- ggplot_build(tnni_plot)
  tnni_data <- data.frame('rho_tnni'= c(tnni_build$data[[3]]$r),
                          'sig_tnni'=c(tnni_build$data[[3]]$p.value),
                          'rowname'=c("DOX","EPI","DNR","MTX", "TRZ", "VEH"))
  ldh_build <- ggplot_build(ldh_plot)
  ldh_data <- data.frame('rho_ldh'= c(ldh_build$data[[3]]$r),
                         'sig_ldh'=c(ldh_build$data[[3]]$p.value),
                         'rowname'=c("DOX","EPI","DNR","MTX", "TRZ", "VEH"))

  results <- cbind(ldh_data[,c(3,1:2)],tnni_data[,1:2],ld50_data[,1:2])

  return(results)

}


cpm_boxplot24h <-function(cpmcounts, GOI,brewer_palette, fill_colors, ylab) {
  ##GOI needs to be ENTREZID
  df <- cpmcounts
  df_plot <- df %>% 
      dplyr::filter(rownames(.)==GOI) %>%
      pivot_longer(everything(),
                   names_to = "treatment",
                   values_to = "counts") %>%
      separate(treatment, c("drug","indv","time")) %>%
      mutate(time = case_match(time,"24h"~"24 hours", "3h"~"3 hours")) %>% 
      mutate(indv=factor(indv, levels = c(1,2,3,4,5,6))) %>%
      mutate(drug =case_match(drug, "Da"~"DNR",
                            "Do"~"DOX",
                            "Ep"~"EPI",
                            "Mi"~"MTX",
                            "Tr"~"TRZ",
                            "Ve"~"VEH", .default = drug)) %>% 
      mutate(drug=factor(drug, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>% 
      dplyr::filter(time=="24 hours")
    plot <- ggplot2::ggplot(df_plot, aes(x=drug, y=counts))+
      geom_boxplot(position="identity",aes(fill=drug))+
      geom_point(aes(col=indv, size=2, alpha=0.5))+
      guides(alpha= "none", size= "none")+
      scale_color_brewer(palette = brewer_palette, guide = "none")+
      scale_fill_manual(values=fill_colors)+
      # facet_wrap("time", nrow=1, ncol=2)+
      theme_bw()+
      ylab(ylab)+
      xlab("")+
      xlab("")+
      ggtitle("24 hours")+
      theme(strip.background = element_rect(fill = "white",linetype=1, linewidth = 0.5),
          plot.title = element_text(size=12,hjust = 0.5,face="bold"),
          axis.title = element_text(size = 10, color = "black"),
          axis.ticks = element_line(linewidth = 1.0),
          panel.background = element_rect(colour = "black", size=1),
          axis.text.x = element_blank(),
          strip.text.x = element_text(margin = margin(2,0,2,0, "pt"),face = "bold"))
    print(plot)
}

```

# Figure 9

Genes in CVD and AC toxicity respond to Top2i

## A. GWAS chi square results
```{r GWAS enrichment}
toplist24hr <- toplistall %>% 
  filter(time=="24_hours")
chi_funarr <-  toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  dplyr::filter(id!="TRZ") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(ARR=if_else(ENTREZID %in%Arr_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id) %>% 
  summarise(pvalue= chisq.test(ARR, sigcount, correct =FALSE)$p.value) 
  

chi_funhf <-  toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
   dplyr::filter(id!="TRZ") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(HF=if_else(ENTREZID %in%HF_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id) %>% 
  summarise(pvalue= chisq.test(HF, sigcount, correct =FALSE)$p.value) 
chi_funCAD <-  toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  dplyr::filter(id!="TRZ") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(CAD=if_else(ENTREZID %in%CAD_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id) %>% 
  summarise(pvalue= chisq.test(CAD, sigcount, correct =FALSE)$p.value) 

HFmat <- chi_funhf %>% ungroup() %>% 
  mutate(HF= log(pvalue)*(-1)) %>% 
  dplyr::select(id,HF) %>% 
  column_to_rownames("id")

ARRmat <- chi_funarr %>% ungroup() %>% 
  mutate(ARR= log(pvalue)*(-1)) %>% 
  dplyr::select(id,ARR) %>% 
  column_to_rownames("id")


CADmat <- chi_funCAD %>% ungroup() %>% 
  mutate(CAD= -log(pvalue)) %>% 
  dplyr::select(id,CAD) %>% 
  column_to_rownames("id")

GWASmat <- bind_cols(CADmat$CAD,HFmat$HF,ARRmat$ARR) 
colnames(GWASmat) <- c('CAD','HF','ARR')
rownames(GWASmat) <- c('DNR','DOX','EPI','MTX')

 Heatmap(as.matrix(GWASmat), name = "GWAS chi square\n -log p values", 
         cluster_rows = FALSE, 
         column_title = 'This is for GWAS 24 hours -log(chi square pvalue)', 
         cluster_columns = FALSE,
         row_names_side = 'left',
         column_names_rot = 0, col_fun1,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(GWASmat[i, j] > -log(0.05))
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})

```



## B. GWAS and TWAS associated genes 
```{r  GWAS-TWAS FC heatmap}
GWAS_goi <- read.csv("output/GWAS_goi.csv",row.names = 1)
##get the abs FC of all GOI
GWASabsFCsig <- 
  toplistall %>% 
  mutate(id = as.factor(id)) %>%
  filter(id !="TRZ") %>%
  filter(ENTREZID %in% GWAS_goi$entrezgene_id) %>% 
   filter(time =="24_hours") %>% 
  dplyr::select(ENTREZID , id,logFC, adj.P.Val, SYMBOL) %>%
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from =adj.P.Val)
  
gwas_sig_mat <- GWASabsFCsig %>% 
   column_to_rownames(var="id") %>%
  as.matrix()
 

GWASabsFC <- toplistall %>% 
  mutate(id = as.factor(id)) %>%
  filter(id !="TRZ") %>% 
  filter(time=="24_hours") %>% 
  mutate(logFC= logFC*(-1)) %>%
  filter(ENTREZID %in% GWAS_goi$entrezgene_id) %>% 
  dplyr::select(SYMBOL ,time, id, logFC) %>% 
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from = logFC) %>% 
  column_to_rownames(var="id") %>%
  as.matrix()
# col_study_fun = circlize::colorRamp2(c(0, 5, 10), c("blue", "white", "red"))
# col_study_motif=circlize::colorRamp2(c(1,2,3,4), c("#7CAE00", "#C77CFF"))
 
study_anno <- data.frame(Study=c("GWAS","GWAS","TWAS","TWAS","GWAS","GWAS","GWAS","TWAS"),motif=c(rep("LR",7),"NR"))
rownames(study_anno) <- colnames(GWASabsFC)
ht <- HeatmapAnnotation(df = study_anno,
              col = list(Study=c("GWAS"="darkorange","TWAS"= "blueviolet"),motif = c("LR"="#7CAE00","NR"="#C77CFF")))


Heatmap(GWASabsFC, name = "Fold change\nvalues", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        col = col_funFC,
        row_order = c('DOX','DNR','EPI', 'MTX'),
        column_title = "Fold change values of GWAS and TWAS genes", 
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_order= c('RARG',
                        'TNS2', 
                        'ZNF740',
                        'SLC28A3',
                        'RMI1',
                        'EEF1B2',
                        'FRS2', 
                        'HDDC2'),
        bottom_annotation = ht,
        
        column_names_rot = 0, 
        column_names_gp = gpar(fontsize = 12,fontface="italic"),
        column_names_centered = TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(gwas_sig_mat[i, j] <0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})

```





## C. RARG expression and correlation to TNNI,  LD50, and LDH values
```{r  rarg data, fig.width=4, fig.height=3}

cpm_boxplot24h(cpmcounts,GOI='5916',"Dark2",drug_pal_fact,
  ylab=(expression(atop(" ",italic("RARG")~log[2]~"cpm "))))

```


```{r RARG corrdata, fig.height=4,fig.width=8}
gene_corr_fig9 <- readRDS("output/gene_corr_fig9.RDS")
RARG_data <- pearson_extract(gene_corr_fig9,5916)

rarg_mat <- RARG_data %>% 
  column_to_rownames('rowname') %>% 
  select(rho_tnni,rho_ldh,rho_LD50,) %>% 
  t() %>% 
  as.matrix()

rarg_mat_sig <- RARG_data %>% 
  column_to_rownames('rowname') %>% 
  select(sig_tnni,sig_ldh,sig_LD50,) %>% 
  mutate_all(~replace(., is.na(.), 1)) %>% 
  t() %>% 
  as.matrix()
Heatmap(rarg_mat, name = "correlation value",
        cluster_rows = FALSE, 
        cluster_columns = FALSE,
        column_names_rot = 0,
        column_names_centered = TRUE,
        na_col = "lightgrey",
        cell_fun = function(j, i, x, y, width, height, fill) {
           if(rarg_mat_sig[i, j]<0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
          })
```


## D. SLC28A3 expression and correlation to TNNI, LD50, and LDH values
```{r  SLC data, fig.width=4, fig.height=3}


cpm_boxplot24h(cpmcounts,GOI='64078',"Dark2",drug_pal_fact,
  ylab=(expression(atop(" ",italic("SLC28A3")~log[2]~"cpm "))))


```
```{r SCL cor data, fig.height=4, fig.width=8}

SLC_data <- pearson_extract(gene_corr_fig9,64078)

slc_mat <- SLC_data %>% 
  column_to_rownames('rowname') %>% 
  select(rho_tnni,rho_ldh,rho_LD50,) %>% 
  t() %>% 
  as.matrix()

slc_mat_sig <- SLC_data %>% 
  column_to_rownames('rowname') %>% 
  select(sig_tnni,sig_ldh,sig_LD50,) %>% 
  mutate_all(~replace(., is.na(.), 1)) %>% 
  t() %>% 
  as.matrix()
Heatmap(slc_mat, name = "correlation value",
        cluster_rows = FALSE, 
        cluster_columns = FALSE,
        column_names_rot = 0,
        column_names_centered = TRUE,
        na_col = "lightgrey",
        cell_fun = function(j, i, x, y, width, height, fill) {
           if(slc_mat_sig[i, j]<0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
          })
```


## E. TNS2 expression and correlation to TNNI, LD50, and LDH values
```{r  TNS2 data,fig.width=4, fig.height=3}

cpm_boxplot24h(cpmcounts,GOI='23371',"Dark2",drug_pal_fact,
  ylab=(expression(atop(" ",italic("TNS2")~log[2]~"cpm "))))

```

```{r TNS2 cor data, fig.height=4, fig.width=8}

TNS2_data <- pearson_extract(gene_corr_fig9,23371)

tns2_mat <- TNS2_data %>% 
  column_to_rownames('rowname') %>% 
  select(rho_tnni,rho_ldh,rho_LD50,) %>% 
  t() %>% 
  as.matrix()

tns2_mat_sig <- TNS2_data %>% 
  column_to_rownames('rowname') %>% 
  select(sig_tnni,sig_ldh,sig_LD50,) %>% 
  mutate_all(~replace(., is.na(.), 1)) %>% 
  t() %>% 
  as.matrix()
Heatmap(tns2_mat, name = "correlation value",
        cluster_rows = FALSE, 
        cluster_columns = FALSE,
        column_names_rot = 0,
        column_names_centered = TRUE,
        na_col = "lightgrey",
        cell_fun = function(j, i, x, y, width, height, fill) {
           if(tns2_mat_sig[i, j]<0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
          })
```

## F. ZNF740 expression and correlation to TNNI, LD50, and LDH values
```{r  ZNF740 data, fig.width=4, fig.height=3}

cpm_boxplot24h(cpmcounts,GOI='283337',"Dark2",drug_pal_fact,
  ylab=(expression(atop(" ",italic("ZNF740")~log[2]~"cpm "))))


```

```{r ZNF cor data, fig.height=4, fig.width=8}

ZNF_data <- pearson_extract(gene_corr_fig9,283337)

ZNF_mat <- ZNF_data %>% 
  column_to_rownames('rowname') %>% 
  select(rho_tnni,rho_ldh,rho_LD50,) %>% 
  t() %>% 
  as.matrix()

ZNF_mat_sig <- ZNF_data %>% 
  column_to_rownames('rowname') %>% 
  select(sig_tnni,sig_ldh,sig_LD50,) %>% 
  mutate_all(~replace(., is.na(.), 1)) %>% 
  t() %>% 
  as.matrix()
Heatmap(ZNF_mat, name = "correlation value",
        cluster_rows = FALSE, 
        cluster_columns = FALSE,
        column_names_rot = 0,
        column_names_centered = TRUE,
        na_col = "lightgrey",
        cell_fun = function(j, i, x, y, width, height, fill) {
           if(ZNF_mat_sig[i, j]<0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
          })
```








