---
title: "Figure 9: Genes in AC toxicity-associated loci respond to TOP2i"
author: "ERM"
date: "`r Sys.Date()`"
output: pdf_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE
)
```

```{css, echo=FALSE}
pre {
  max-height: 400px;
  overflow-y: auto;
}

pre[class] {
  max-height: 200px;
}
```

```{r packages loading,echo=TRUE}
library(edgeR)
library(tidyverse)
library(ggsignif)
library(RColorBrewer)
library(cowplot)
library(ggpubr)
library(scales)
library(sjmisc)
library(kableExtra)
library(broom)
library(ComplexHeatmap)
```
```{r data loading,echo=TRUE}
toplistall <- readRDS("data/toplistall.RDS")
siglist <- readRDS("data/siglist_final.RDS")
list2env(siglist, envir = .GlobalEnv)
cpmcounts <- readRDS("data/cpmcount.RDS")
backGL <- read.csv("data/backGL.txt",     row.names =1)

drug_palc <- c("#8B006D","#DF707E","#F1B72B", "#3386DD","#707031","#41B333")
drug_pal_fact <- c("#8B006D" ,"#DF707E", "#F1B72B" ,"#3386DD", "#707031","#41B333")


col_fun1 = circlize::colorRamp2(c(-1, 3), c("white", "purple"))
col_funFC= circlize::colorRamp2(c(-2,0, 2), c("darkgreen","white", "darkorange2"))
col_funTOX = circlize::colorRamp2(c(-1,0, 1), c("darkviolet", "white","firebrick4"))

```
```{r function load,echo=TRUE}

pearson_extract <- function(corr_df,ENTREZID) {
  ld50_plot <- corr_df %>%
    dplyr::filter(entrezgene_id == ENTREZID) %>%
    ggplot(., aes(x=LD50, y=counts))+
    geom_point(aes(col=indv))+
    geom_smooth(method="lm")+
    facet_wrap(hgnc_symbol~Drug, scales="free")+
    theme_classic()+
    stat_cor(method="pearson",
             aes(label = paste(..r.label.., ..p.label.., sep = "~`,`~")),
             color = "red",
             label.x.npc = 0,
             label.y.npc=1,
             size = 3)

  tnni_plot <-  corr_df %>%
    dplyr::filter(entrezgene_id == ENTREZID) %>%
    ggplot(., aes(x=rtnni, y=counts))+
    geom_point(aes(col=indv))+
    geom_smooth(method="lm")+
    facet_wrap(hgnc_symbol~Drug, scales="free")+
    theme_classic()+
    stat_cor(method="pearson",
             aes(label = paste(..r.label.., ..p.label.., sep = "~`,`~")),
             color = "red",
             label.x.npc = 0,
             label.y.npc=1,
             size = 3)

  ldh_plot <-   corr_df %>%
    dplyr::filter(entrezgene_id == ENTREZID) %>%
    ggplot(., aes(x=rldh, y=counts))+
    geom_point(aes(col=indv))+
    geom_smooth(method="lm")+
    facet_wrap(hgnc_symbol~Drug, scales="free")+
    theme_classic()+
    stat_cor(method="pearson",
             aes(label = paste(..r.label.., ..p.label.., sep = "~`,`~")),
             color = "red",
             label.x.npc = 0,
             label.y.npc=1,
             size = 3)

  ##ggbuild to get model:
  ld50_build <- ggplot_build(ld50_plot)
  ld50_data <- data.frame('rho_LD50'= c(ld50_build$data[[3]]$r,NA,NA),
                          'sig_LD50'=c(ld50_build$data[[3]]$p.value,NA,NA),
                          'rowname'=c("DOX","EPI","DNR","MTX", "TRZ", "VEH"))

  tnni_build <- ggplot_build(tnni_plot)
  tnni_data <- data.frame('rho_tnni'= c(tnni_build$data[[3]]$r),
                          'sig_tnni'=c(tnni_build$data[[3]]$p.value),
                          'rowname'=c("DOX","EPI","DNR","MTX", "TRZ", "VEH"))
  ldh_build <- ggplot_build(ldh_plot)
  ldh_data <- data.frame('rho_ldh'= c(ldh_build$data[[3]]$r),
                         'sig_ldh'=c(ldh_build$data[[3]]$p.value),
                         'rowname'=c("DOX","EPI","DNR","MTX", "TRZ", "VEH"))

  results <- cbind(ldh_data[,c(3,1:2)],tnni_data[,1:2],ld50_data[,1:2])

  return(results)

}


cpm_boxplot24h <-function(cpmcounts, GOI,brewer_palette, fill_colors, ylab) {
  ##GOI needs to be ENTREZID
  df <- cpmcounts
  df_plot <- df %>% 
      dplyr::filter(rownames(.)==GOI) %>%
      pivot_longer(everything(),
                   names_to = "treatment",
                   values_to = "counts") %>%
      separate(treatment, c("drug","indv","time")) %>%
      mutate(time = case_match(time,"24h"~"24 hours", "3h"~"3 hours")) %>% 
      mutate(indv=factor(indv, levels = c(1,2,3,4,5,6))) %>%
      mutate(drug =case_match(drug, "Da"~"DNR",
                            "Do"~"DOX",
                            "Ep"~"EPI",
                            "Mi"~"MTX",
                            "Tr"~"TRZ",
                            "Ve"~"VEH", .default = drug)) %>% 
      mutate(drug=factor(drug, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>% 
      dplyr::filter(time=="24 hours")
    plot <- ggplot2::ggplot(df_plot, aes(x=drug, y=counts))+
      geom_boxplot(position="identity",aes(fill=drug))+
      geom_point(aes(col=indv, size=2, alpha=0.5))+
      guides(alpha= "none", size= "none")+
      scale_color_brewer(palette = brewer_palette, guide = "none")+
      scale_fill_manual(values=fill_colors)+
      # facet_wrap("time", nrow=1, ncol=2)+
      theme_bw()+
      ylab(ylab)+
      xlab("")+
      ggtitle("24 hours")+
      theme(strip.background = element_rect(fill = "white",linetype=1, linewidth = 0.5),
          plot.title = element_text(size=12,hjust = 0.5,face="bold"),
          axis.title = element_text(size = 10, color = "black"),
          axis.ticks = element_line(linewidth = 1.0),
          panel.background = element_rect(colour = "black", size=1),
          axis.text.x = element_blank(),
          strip.text.x = element_text(margin = margin(2,0,2,0, "pt"),face = "bold"))
    print(plot)
}

pearson_cardiotox <- function(corr_df,ENTREZID) {

  full_plot <- corr_df %>%
    dplyr::filter(entrezgene_id == ENTREZID) %>%
    ggplot(., aes(x=tox_score, y=counts))+
    geom_point(aes(col=indv))+
    geom_smooth(method="lm")+
    facet_wrap(hgnc_symbol~Drug, scales="free")+
    theme_classic()+
    stat_cor(method="pearson",
             aes(label = paste(..r.label.., ..p.label.., sep = "~`,`~")),
             color = "red",
             label.x.npc = 0,
             label.y.npc=1,
             size = 3)


  ##ggbuild to get model:
  tox_build <- ggplot_build(full_plot)
  tox_data <- data.frame('rowname'=c("DOX","EPI","DNR","MTX", "TRZ", "VEH"),
                         'tox_val'= c(tox_build$data[[3]]$r,NA,NA),
                          'sig_tox'=c(tox_build$data[[3]]$p.value,NA,NA)
                          )


    return(tox_data)

}


```

## Figure 9: Genes in AC toxicity-associated loci respond to TOP2i


### A. GWAS and TWAS associated genes 
```{r  GWAS-TWAS FC heatmap,echo=TRUE}
GWAS_goi <- read.csv("output/GWAS_goi.csv",row.names = 1)
##get the abs FC of all GOI
GWASabsFCsig <- 
  toplistall %>% 
    mutate(drug=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>% 
  filter(ENTREZID %in% GWAS_goi$entrezgene_id) %>% 
   filter(time =="24_hours") %>% 
  dplyr::select(ENTREZID , id,logFC, adj.P.Val, SYMBOL) %>%
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from =adj.P.Val)
  
gwas_sig_mat <- GWASabsFCsig %>% 
   column_to_rownames(var="id") %>%
  as.matrix()
 

GWASabsFC <- toplistall %>% 
  
  # filter(id !="TRZ") %>% 
  filter(time=="24_hours") %>% 
  mutate(logFC= logFC*(-1)) %>%
  filter(ENTREZID %in% GWAS_goi$entrezgene_id) %>% 
  dplyr::select(SYMBOL ,time, id, logFC) %>% 
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from = logFC) %>% 
  column_to_rownames(var="id") %>%
  as.matrix()

study_anno <- data.frame(Study=c("GWAS","GWAS","TWAS","TWAS","GWAS","GWAS","GWAS","TWAS"),motif=c(rep("LR",7),"NR"))
rownames(study_anno) <- colnames(GWASabsFC)
ht <- HeatmapAnnotation(df = study_anno,
              col = list(Study=c("GWAS"="darkorange","TWAS"= "blueviolet"),motif = c("LR"="#7CAE00","NR"="#C77CFF"), just = "left"))


Heatmap(GWASabsFC, name = "Fold change\nvalues", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        col = col_funFC,
        row_order = c('DOX','EPI','DNR', 'MTX','TRZ'),
        column_title = "Fold change values of GWAS and TWAS genes", 
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_order= c('RARG',
                        'TNS2', 
                        'ZNF740',
                        'SLC28A3',
                        'RMI1',
                        'EEF1B2',
                        'FRS2', 
                        'HDDC2'),
        bottom_annotation = ht,
        
        column_names_rot = 0, 
        column_names_gp = gpar(fontsize = 12,fontface="italic"),
        column_names_centered =TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(gwas_sig_mat[i, j] <0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})

```

### B. Lactate dehydrogenase release
```{r LDH assay24hr}
RNAnormlist <- read.csv("output/TNNI_LDH_RNAnormlist.txt")
level_order2 <- c('75','87','77','79','78','71')
RNAnormlist <- RNAnormlist %>% 
  mutate(indv= factor(indv,levels= level_order2))
  
RNAnormlist %>% 
  mutate(Drug = factor(Drug, levels = c(  "DOX", 
                                          "EPI",
                                          "DNR",
                                          "MTX",
                                          "TRZ",
                                          "VEH"))) %>%
  ggplot(., aes(x=Drug, y=rldh))+
  geom_boxplot(position = "identity", fill = drug_pal_fact)+
  geom_point(aes(col=indv, size =3,alpha=0.5))+
  geom_signif(comparisons =list(c("VEH","DOX"),
                                c("VEH","EPI"),
                                c("VEH","DNR"),
                                c("VEH","MTX"),
                                c("VEH","TRZ")),
              test="t.test",
              map_signif_level=TRUE,
              textsize =4,
              step_increase = 0.1)+
  theme_classic()+
  guides(size = "none",alpha="none")+
  scale_color_brewer(palette = "Dark2", name = "Individual")+
  xlab("")+
  ylab("Relative LDH activity ")+
  ggtitle("Lactate dehydrogenase release at 24 hours")+
  theme_classic()+
  theme(strip.background = element_rect(fill = "transparent")) +
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        legend.position = "none",
         axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 12, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))
```
### C. Troponin I release
```{r TropI release 24}

RNAnormlist %>% 
  mutate(Drug = factor(Drug, levels = c(  "DOX", 
                                          "EPI",
                                          "DNR",
                                          "MTX",
                                          "TRZ",
                                          "VEH"))) %>%
  ggplot(., aes(x=Drug, y=rtnni))+
  geom_boxplot(position = "identity", fill = drug_pal_fact)+
  geom_point(aes(col=indv, size =3,alpha=0.5))+
  geom_signif(comparisons =list(c("VEH","DOX"),
                                c("VEH","EPI"),
                                c("VEH","DNR"),
                                c("VEH","MTX"),
                                c("VEH","TRZ")),
              test="t.test",
              map_signif_level=TRUE,
              textsize =4,
              step_increase = 0.1)+
  theme_classic()+
  guides(size = "none",alpha="none")+
  scale_color_brewer(palette = "Dark2", name = "Individual")+
  xlab("")+
  ylab("Relative Troponin I levels ")+
  ggtitle("Troponin I release at 24 hours")+
  theme_classic()+
  theme(strip.background = element_rect(fill = "transparent")) +
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        legend.position = "none",
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 12, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))

```

For more Troponin I and LDH release analysis, [check this link out](https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/LDH_analysis.html#Normalization_of_TNNI_and_LDH_to_RNA_concentration)


#### Cardiotoxicity Score code


```{r ctxscore, message=FALSE, warning=FALSE, include=TRUE,echo=TRUE}

gene_corr_fig9 <- readRDS("output/gene_corr_fig9.RDS")
toxlist <- data.frame(ENTREZID = c(5916, 23371, 283337, 64078, 80010))
toxtest_2pt <- gene_corr_fig9 %>%
  rowwise() %>%
  # mutate_all(~replace(., is.na(.), 0)) %>%
  mutate(tox_score = mean(c(rldh, rtnni), na.rm = TRUE)) %>%
  dplyr::select(entrezgene_id, hgnc_symbol, Drug, indv, time, counts, tox_score) %>%
  filter(entrezgene_id %in% toxlist$ENTREZID)


toxdata2pt <- list()


for (hay in 1:5) {
  data <- toxtest_2pt %>%
    dplyr::filter(entrezgene_id == toxlist$ENTREZID[hay])
  
  dataname <- unique(data$hgnc_symbol)
  
  p <- ggplot(data, aes(x = tox_score, y = counts)) +
    geom_point(aes(col = indv)) +
    geom_smooth(method = "lm") +
    facet_wrap(hgnc_symbol ~ Drug, scales = "free") +
    theme_classic() +
    scale_color_brewer(
      palette = "Dark2",
      name = "Individual",
      label = c("1", "2", "3", "4", "5", "6")
    ) +
    ggpubr::stat_cor(
      method = "pearson",
      aes(label = paste(..r.label.., ..p.label.., sep = "~`,`~")),
      color = "red",
      label.x.npc = 0,
      label.y.npc = 1,
      size = 3
    )
  tox_build <- ggplot_build(p)
  plot(p)
  
  toxdata2pt[[dataname]] <- list(
    'tox_val' = c(tox_build$data[[3]]$r),
    'sig_tox' = c(tox_build$data[[3]]$p.value)
  )
}

```

extraction code for heatmap below gene expression boxes
### D. SNP-related gene expression and cardiotoxicity score


The extraction code for the horizontal heatmap below gene expression boxplots in the paper is located here.



#### RARG


```{r  RARG data,fig.width=4, fig.height=3,echo=TRUE}

cpm_boxplot24h(cpmcounts,GOI='5916',"Dark2",drug_pal_fact,
  ylab=(expression(atop(" ",italic("RARG")~log[2]~"cpm "))))

```

```{r heatmap long RARG,echo=TRUE}

toxdata2ptr <- map_df(toxdata2pt, ~as.data.frame(.x), .id="gene")

RARG_sig_mat2ptr <- toxdata2ptr %>% 
  dplyr::filter(gene=="RARG") %>% 
  dplyr::select(sig_tox) %>% 
  t() %>% 
  matrix()
RARG_mat2ptr <- toxdata2ptr%>% 
  dplyr::filter(gene=="RARG") %>% 
  dplyr::select(tox_val) %>% 
    t() %>% 
  matrix()

Heatmap(RARG_mat2ptr, name = " RARG 2pt Cardiotox score", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_names_rot = 0, 
        col= col_funTOX,
        row_labels = c("DOX","EPI","DNR","MTX","TRZ","VEH"),
        column_names_gp = gpar(fontsize = 12,fontface="italic"),
        column_names_centered = TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(RARG_sig_mat2ptr[i, j] <0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})


```




#### TNS2


```{r  TNS2 data,fig.width=4, fig.height=3,echo=TRUE}

cpm_boxplot24h(cpmcounts,GOI='23371',"Dark2",drug_pal_fact,
  ylab=(expression(atop(" ",italic("TNS2")~log[2]~"cpm "))))

```


```{r heatmap long TNS2,echo=TRUE}

TNS2_sig_mat2ptr <- toxdata2ptr %>% 
  dplyr::filter(gene=="TNS2") %>% 
  dplyr::select(sig_tox) %>% 
  t() %>% 
  matrix()
TNS2_mat2ptr <- toxdata2ptr%>% 
  dplyr::filter(gene=="TNS2") %>% 
  dplyr::select(tox_val) %>% 
    t() %>% 
  matrix()

Heatmap(TNS2_mat2ptr, name = " TNS2 2pt Cardiotox score", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_names_rot = 0, 
        col= col_funTOX,
        row_labels = c("DOX","EPI","DNR","MTX","TRZ","VEH"),
        column_names_gp = gpar(fontsize = 12,fontface="italic"),
        column_names_centered = TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(TNS2_sig_mat2ptr[i, j] <0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})


```




#### ZNF740


```{r ZNF740 data, eval=TRUE, fig.height=3, fig.width=4, include=TRUE,echo=TRUE}


cpm_boxplot24h(cpmcounts,GOI='283337',"Dark2",drug_pal_fact,
  ylab=(expression(atop(" ",italic("ZNF740")~log[2]~"cpm "))))


```
```{r heatmap long ZF740,echo=TRUE}
ZNF740_sig_mat2ptr <- toxdata2ptr %>% 
  dplyr::filter(gene=="ZNF740") %>% 
  dplyr::select(sig_tox) %>% 
  t() %>% 
  matrix()
ZNF740_mat2ptr <- toxdata2ptr%>% 
  dplyr::filter(gene=="ZNF740") %>% 
  dplyr::select(tox_val) %>% 
    t() %>% 
  matrix()

Heatmap(ZNF740_mat2ptr, name = " ZNF740 2pt Cardiotox score", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_names_rot = 0, 
        col= col_funTOX,
        row_labels = c("DOX","EPI","DNR","MTX","TRZ","VEH"),
        column_names_gp = gpar(fontsize = 12,fontface="italic"),
        column_names_centered = TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(ZNF740_sig_mat2ptr[i, j] <0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})


```






#### RMI1



```{r RMI1 data, eval=TRUE, fig.height=3, fig.width=4, include=TRUE,echo=TRUE}


cpm_boxplot24h(cpmcounts,GOI='80010',"Dark2",drug_pal_fact,
  ylab=(expression(atop(" ",italic("RMI1")~log[2]~"cpm "))))


```


```{r RMI1,echo=TRUE}
toxdata2ptr <- map_df(toxdata2pt, ~as.data.frame(.x), .id="gene")

RMI1_sig_mat2ptr <- toxdata2ptr %>% 
  dplyr::filter(gene=="RMI1") %>% 
  dplyr::select(sig_tox) %>% 
  t() %>% 
  matrix()
RMI1_mat2ptr <- toxdata2ptr%>% 
  dplyr::filter(gene=="RMI1") %>% 
  dplyr::select(tox_val) %>% 
    t() %>% 
  matrix()



Heatmap(RMI1_mat2ptr, name = " RMI1 2pt Cardiotox score", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_names_rot = 0, 
        col= col_funTOX,
        row_labels = c("DOX","EPI","DNR","MTX","TRZ","VEH"),
        column_names_gp = gpar(fontsize = 12,fontface="italic"),
        column_names_centered = TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(RMI1_sig_mat2ptr[i, j] <0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})

```



#### SLC28A3



```{r SLC data, eval=TRUE, fig.height=3, fig.width=4, include=TRUE,echo=TRUE}


cpm_boxplot24h(cpmcounts,GOI='64078',"Dark2",drug_pal_fact,
  ylab=(expression(atop(" ",italic("SLC28A3")~log[2]~"cpm "))))


```


```{r SLC28A3,echo=TRUE}
toxdata2ptr <- map_df(toxdata2pt, ~as.data.frame(.x), .id="gene")

SLC28A3_sig_mat2ptr <- toxdata2ptr %>% 
  dplyr::filter(gene=="SLC28A3") %>% 
  dplyr::select(sig_tox) %>% 
  t() %>% 
  matrix()
SLC28A3_mat2ptr <- toxdata2ptr%>% 
  dplyr::filter(gene=="SLC28A3") %>% 
  dplyr::select(tox_val) %>% 
    t() %>% 
  matrix()



Heatmap(SLC28A3_mat2ptr, name = " SLC28A3 2pt Cardiotox score", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_names_rot = 0, 
        col= col_funTOX,
        row_labels = c("DOX","EPI","DNR","MTX","TRZ","VEH"),
        column_names_gp = gpar(fontsize = 12,fontface="italic"),
        column_names_centered = TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(SLC28A3_sig_mat2ptr[i, j] <0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})

```


