---
title: "Figure 7: Genes in AC toxicity-associated loci respond to TOP2i."
author: "ERM"
date: "`r Sys.Date()`"
output:
  html_document:
    code_folding: 'show'
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE
)
```

```{css, echo=FALSE}
pre {
  max-height: 400px;
  overflow-y: auto;
}

pre[class] {
  max-height: 200px;
}
```


```{r packages loading}

library(tidyverse)
library(gridtext)
library(scales)
library(kableExtra)
library(qvalue)
library(data.table)
library(ComplexHeatmap)
library(readr)
library(limma)
library(edgeR)



```

```{r data loading, message=FALSE, warning=FALSE}
drug_palc <- c("#8B006D","#DF707E","#F1B72B", "#3386DD","#707031","#41B333")
drug_pal_fact <- c("#8B006D" ,"#DF707E", "#F1B72B" ,"#3386DD", "#707031","#41B333")

cpmcounts <- readRDS("data/cpmcount.RDS")
DEG_cormotif <- readRDS("data/DEG_cormotif.RDS")
list2env(DEG_cormotif,envir=.GlobalEnv)
backGL <- read.csv("data/backGL.txt", row.names =1)
col_fun1 = circlize::colorRamp2(c(-1, 3), c("white", "purple"))
col_funFC= circlize::colorRamp2(c(-2,0, 2), c("darkgreen","white", "darkorange2"))
col_funTOX = circlize::colorRamp2(c(-1,0, 1), c("darkviolet", "white","firebrick4"))


DOXrespGene <- readRDS("output/DOXreQTLs.RDS")
toplistall <- readRDS("data/toplistall.RDS")

list2env(DEG_cormotif,envir=.GlobalEnv)

counts_v8_heart_left_ventricle_gct <- readRDS("output/counts_v8_heart_left_ventricle_gct.RDS")
names_gtex_genes <- counts_v8_heart_left_ventricle_gct %>% 
  dplyr::select(Name, Description)
gtex_counts_matrix<- counts_v8_heart_left_ventricle_gct %>% 
  column_to_rownames("Name") %>% 
  dplyr::select(!id) %>% 
  dplyr::select(!Description) %>% 
  as.matrix()
  
# cpm <- cpm(gtex_counts_matrix)
lcpm <- cpm(gtex_counts_matrix, log=TRUE)  ### for determining the basic cutoffs
dim(lcpm)

row_means <- rowMeans(lcpm)
gtex_x <- lcpm[row_means > 0,]
GTEX_heartLV_log2cpm <- gtex_x %>% 
  as.data.frame() %>%
  mutate(Name = rownames(gtex_x)) %>% 
  left_join(.,names_gtex_genes, by =c("Name"))





SGWAS_order_frame <- read.csv("output/SGWAS_top50_order.csv", row.names = 1)

col_fun5 = circlize::colorRamp2(c(0, 5), c("white", "purple"))
## supplement 1
chrom_reg_Seoane <- read_csv(file = "data/Seonane2019supp1.txt",col_types = cols(...1 = col_skip()))
Seoane_2019 <- chrom_reg_Seoane[,2]
names(Seoane_2019) <- "ENTREZID"
Chrom_reg <- data.frame("entrez"=(unique(Seoane_2019$ENTREZID)))

Sup4seoane <- read.csv("output/Sup4seoane.csv", row.names = 1)
DOX_reg <- Sup4seoane  %>% 
  filter(pval.expAnth<0.05) %>% 
  distinct(entrez, .keep_all = TRUE) %>% 
  dplyr::select(entrez)
base_egene <-readRDS("output/knowles4.RDS")
resp_egene <-readRDS("output/knowles5.RDS")


```

## Figure 7: Genes in AC toxicity-associated loci respond to TOP2i. 



```{r Finfig_hm 24 hours}

orderit <- data.frame(SYMBOL=c(SGWAS_order_frame$SYMBOL,'RARG',
                        'TNS2', 
                        'ZNF740',
                        'SLC28A3',
                        'RMI1',
                        'EEF1B2',
                        'FRS2', 
                        'HDDC2')) %>% 
  left_join(.,backGL) %>% 
  mutate(motif= if_else(ENTREZID %in% motif_ER, "ER", if_else(ENTREZID %in% motif_LR, "LR",if_else(ENTREZID %in% motif_TI, "TI", "NR" )))) %>% 
  mutate(chrom_gene= if_else(ENTREZID %in% Chrom_reg$entrez,"yes","no")) %>% 
  mutate(Dox_respegene= if_else(ENTREZID %in% resp_egene$entrezgene_id, "yes","no")) %>% 
  mutate(ACchrom_gene= if_else(ENTREZID %in%  DOX_reg$entrez,"yes","no")) %>%  
  mutate(gTEX_heart=if_else(SYMBOL %in% GTEX_heartLV_log2cpm$Description,  "yes","no"))




toplistall %>% 
    mutate(drug=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>% 
  filter(ENTREZID %in% orderit$ENTREZID) %>% 
   filter(time =="24_hours") %>% 
  dplyr::select(ENTREZID , id,logFC, adj.P.Val, SYMBOL) 
 

GWASabsFCsig <- 
  toplistall %>% 
    mutate(drug=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>% 
  filter(ENTREZID %in% orderit$ENTREZID) %>% 
   filter(time =="24_hours") %>% 
  dplyr::select(ENTREZID , id,logFC, adj.P.Val, SYMBOL)  %>%
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from =adj.P.Val)
  
gwas_sig_mat <- GWASabsFCsig %>% 
   column_to_rownames(var="id") %>%
  dplyr::select(orderit$SYMBOL) %>%
  as.matrix()
 

GWASabsFC <- toplistall %>% 
  
  # filter(id !="TRZ") %>% 
  filter(time=="24_hours") %>% 
  mutate(logFC= logFC*(-1)) %>%
  filter(ENTREZID %in% orderit$ENTREZID) %>% 
  dplyr::select(SYMBOL ,time, id, logFC) %>% 
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from = logFC) %>% 
  column_to_rownames(var="id") %>%
  dplyr::select(orderit$SYMBOL) %>%
  as.matrix()

study_anno <- data.frame(Study=c(rep("GWAS",35),rep("TWAS",3)),motif=orderit$motif,chrom_gene=orderit$chrom_gene,Dox_respegene=orderit$Dox_respegene, ACchrom_gene=orderit$ACchrom_gene,gTEX_heart=orderit$gTEX_heart)
rownames(study_anno) <- colnames(GWASabsFC)
ht <- HeatmapAnnotation(df = study_anno,
              col = list(Study=c("GWAS"="darkorange","TWAS"= "blueviolet"),
                         motif = c("LR"="#00BFC4", "NR"="#C77CFF", "ER"="#F8766D", "TI"="#7CAE00"),
chrom_gene=c("yes"="darkred","no"="pink"),
Dox_respegene=c("yes"="darkred","no"="pink"),
ACchrom_gene=c("yes"="darkred","no"="pink"),
gTEX_heart=c("yes"="darkred","no"="pink")))

Heatmap(GWASabsFC, name = "Fold change\nvalues", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        col = col_funFC,
        row_order = c('DOX','EPI','DNR', 'MTX','TRZ'),
        column_title = "Fold change values of GWAS and TWAS genes n = 38", 
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_order= orderit$SYMBOL,
        bottom_annotation = ht,
        
        column_names_rot = 90, 
        column_names_gp = gpar(fontsize = 12,fontface="italic"),
        column_names_centered =TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(gwas_sig_mat[i, j] <0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})
```


```{r Finfig_hm 3 hours}

GWASabsFCsig_3 <- 
  toplistall %>% 
    mutate(drug=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>% 
  filter(ENTREZID %in% orderit$ENTREZID) %>% 
   filter(time =="3_hours") %>% 
  dplyr::select(ENTREZID , id,logFC, adj.P.Val, SYMBOL)  %>%
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from =adj.P.Val)
  
gwas_sig_mat_3 <- GWASabsFCsig %>% 
   column_to_rownames(var="id") %>%
  dplyr::select(orderit$SYMBOL) %>%
  as.matrix()
 

GWASabsFC_3 <- toplistall %>% 
  
  # filter(id !="TRZ") %>% 
  filter(time=="24_hours") %>% 
  mutate(logFC= logFC*(-1)) %>%
  filter(ENTREZID %in% orderit$ENTREZID) %>% 
  dplyr::select(SYMBOL ,time, id, logFC) %>% 
  pivot_wider(id_cols=id, 
              names_from = SYMBOL, 
              values_from = logFC) %>% 
  column_to_rownames(var="id") %>%
  dplyr::select(orderit$SYMBOL) %>%
  as.matrix()

# study_anno <- data.frame(Study=c(rep("GWAS",35),rep("TWAS",3)),motif=orderit$motif,chrom_gene=orderit$chrom_gene,Dox_respegene=orderit$Dox_respegene, ACchrom_gene=orderit$ACchrom_gene,gTEX_heart=orderit$gTEX_heart)
# rownames(study_anno) <- colnames(GWASabsFC)
# ht <- HeatmapAnnotation(df = study_anno,
#               col = list(Study=c("GWAS"="darkorange","TWAS"= "blueviolet"),
#                          motif = c("LR"="#00BFC4", "NR"="#C77CFF", "ER"="#F8766D", "TI"="#7CAE00"),
# chrom_gene=c("yes"="darkred","no"="pink"),
# Dox_respegene=c("yes"="darkred","no"="pink"),
# ACchrom_gene=c("yes"="darkred","no"="pink"),
# gTEX_heart=c("yes"="darkred","no"="pink")))

Heatmap(GWASabsFC_3, name = "Fold change\nvalues", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        col = col_funFC,
        row_order = c('DOX','EPI','DNR', 'MTX','TRZ'),
        column_title = "Fold change values of GWAS and TWAS genes\n 3 hours n = 38", 
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_order= orderit$SYMBOL,
        bottom_annotation = ht,
        
        column_names_rot = 90, 
        column_names_gp = gpar(fontsize = 12,fontface="italic"),
        column_names_centered =TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(gwas_sig_mat_3[i, j] <0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})



```

More analysis GWAS and TWAS SNPs found in the Schneider data [at this link](https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/after_comments.html#Top_50_SNP_and_eGenes_from_Schneider_GWAS).
