---
title: "GO analysis"
author: "ERM"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE)
```

# GO Analysis

I have created several files from the RNA analysis that contain the significant genes(determined by adj.P.val \< 0.1) from each Time and Condition. The names of the files are in the following format: 'sigV'+Drug(2 letters)+time.

example: 'sigVDA3.txt' means this file contains the significant DE genes from the Daunorubicin 3 hour compared to Vehicle Control 3 hour analysis

```{r Import libraries, message=FALSE, warning=FALSE}
library(tidyverse)
library(gprofiler2)
library(readr)
library(BiocGenerics)
library(gridExtra)
library(VennDiagram)
library(kableExtra)
library(scales)
library(ggVennDiagram)

```

```{r Import data, echo=FALSE, message=FALSE, warning=FALSE}
 
##note, in this chunk the .rmd file is in the cm/analysis folder.   need to back out with .. 
##this code checks the directory and searches for the pattern, returning the full name of the file
file.names <- list.files(path = "data/", pattern = "sig*", ignore.case = TRUE,full.names = TRUE)

##Next I use the list of names and lapply to read all files into a list
##made the csv file to reimport
#write.csv(filenameonly,"data/filenameonly.txt", col.names = FALSE,row.names = FALSE)
##readin the filenameonly.txt file from data to use for naming and filtering.
filenameonly <- read_csv("data/filenameonly.txt")
#loop through the list of files and make a separate dataframe for each file under the 'real'  name of the data set
for (k in 1:length(file.names)){
  
  assign(paste0(filenameonly$x[k]) , read.csv(file.names[k]))
}

##rename the columns to the previous names

colnames(sigVDA24)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVDX24)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVEP24)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVMT24)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVTR24)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVDA3)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVDX3)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVEP3)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVMT3)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVTR3)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
```

The analysis is based on all genes that passed the rowMeans\>0 from the previous page [link](https://reneeisnowhere.github.io/Cardiotoxicity/RNAseqrun_1_analysis.html)

```{r uploading the background genes, echo=FALSE,  message=FALSE, warning=FALSE, paged.print=FALSE}
#backGL <- efit2$genes   ***making the list

#write.csv(backGL, "data/backGL.txt", row.names = FALSE)
##read the  .csv the render into txt in the data file!
 backGL <- read_csv("data/backGL.txt", 
    col_types = cols(...1 = col_skip()))
```

Below is the analysis of differentially expressed genes for each treatment at 3 hours and 24 hours.

```{r intial analysis, echo=FALSE, message=FALSE, warning=FALSE}
gostres <- gost(query = sigVDA3$ENTREZID,  organism = "hsapiens",
  ordered_query = TRUE,
  domain_scope = "custom",
  measure_underrepresentation = FALSE,
  evcodes = FALSE,
  user_threshold = 0.0001,
  correction_method = c("fdr"),
  custom_bg = backGL$ENTREZID,
  sources=c("GO:BP","GO:MF", "GO:CC"))
p <- gostplot(gostres, capped = FALSE, interactive = TRUE)
p

table1 <- gostres$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) %>% 
   mutate_at(.cols = 6, .funs= scientific_format()) %>% 
  kable(.,) %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover"))
  table1
# publish_gosttable(
#   gostres,
#   highlight_terms = NULL,
#   use_colors = TRUE,
#   show_columns = c("source", "term_name", "term_size", "intersection_size"),
#   filename = "../output/test.pdf",
#   ggplot = TRUE
# )
```

I first looked at the data with all genes from the sigDA3 dataset. I used the list of all genes based on my rowMeans\>0 filtering as background.

### Analysis of Up versus Down

I then separated the VDA3 file by log2 Fold Change to see how the gene sets are enriched. Nothing showed up in the GO-BP/CC/MG-down regulated gene-set at a significant level, p\<0.05.

```{r only the Da3 data3, echo=FALSE, message=FALSE, warning=FALSE}
#results_sig = subset(sigVDA3,adj.P.Val < 0.05)
# get the significant up-regulated genes
  up = subset(sigVDA3, logFC > 0)
# get the significant down-regulated genes
down = subset(sigVDA3, logFC < 0)
gp_up = gost(query = up$ENTREZID,  organism = "hsapiens",
  ordered_query = TRUE,
  domain_scope = "custom",
  measure_underrepresentation = FALSE,
  evcodes = FALSE,
  user_threshold = 0.001,
  correction_method = c("fdr"),
  custom_bg = backGL$ENTREZID,
  sources=c("GO:BP","GO:MF", "GO:CC"))
  
  
#gp_down = gost(query = down$ENTREZID,  organism = "hsapiens",
  # ordered_query = TRUE,
  # domain_scope = "custom",
  # measure_underrepresentation = FALSE,
  # evcodes = FALSE,
  # user_threshold = 0.05,
  # correction_method = c("bonferroni"),
  # custom_bg = backGL$ENTREZID,
  # sources=c("GO:BP","GO:MF", "GO:CC"))
p2_up <- gostplot(gp_up, capped = FALSE, interactive = TRUE)

p2_up #+ ggtitle("Daunorubicin up regulated gene enrichment at 3 hours")

```

Note this is at three hour for Daunorubicin

```{r showing it all, eval = FALSE, echo=FALSE, message=FALSE, warning=FALSE}

#p2_down <- gostplot(gp_down, capped = FALSE, interactive = TRUE)

#p2_down #+ ggtitle("Daunorubicin down regulated gene enrichment at 3 hours")


```

#### I next wanted to see what happened at 24 hours with daunorubicin. I used the sigVDA24 file to do this.

```{r DA24, eval=TRUE, echo=FALSE, message=FALSE, warning=FALSE, paged.print=TRUE}
gostresDA24 <- gost(query = sigVDA24$SYMBOL,  organism = "hsapiens",
  ordered_query = TRUE,
  domain_scope = "custom",
  measure_underrepresentation = FALSE,
  evcodes = FALSE,
  user_threshold = 0.0001,
  correction_method = c("fdr"),
  custom_bg = backGL$ENTREZID,
  sources=c("GO:BP","GO:MF", "GO:CC"))
pDA24 <- gostplot(gostres, capped = FALSE, interactive = TRUE)
pDA24

# (gostres$result$p_value)
table2 <- gostresDA24$result %>%
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) %>% 
  mutate_at(.cols = 6, .funs= scientific_format()) %>% 
  kable(.,) %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover"))
  table2
# publish_gosttable(gostresDA24,
#   highlight_terms = gostresDA24$result,
#   use_colors = TRUE,
#   show_columns = c("source", "term_name", "term_size", "intersection_size"),
#   filename = "../output/test.pdf",
#   ggplot = TRUE
# )

```
Here I was able to visualize the 24 hour GO analysis of daunorubicin.

##Venn Diagrams
```{r venndiagram, echo=FALSE, message=FALSE, warning=FALSE}
total24 <-list(sigVDA24$ENTREZID,sigVDX24$ENTREZID,sigVEP24$ENTREZID,sigVMT24$ENTREZID)
in_common24 <-c(sigVDA24$ENTREZID,sigVDX24$ENTREZID,sigVEP24$ENTREZID,sigVMT24$ENTREZID)

length(unique(in_common24))

ggVennDiagram(total24,
              category.names = c("Daunorubicin","Doxorubicin","Epirubicin","Mitoxantrone"),
              show_intersect = FALSE,
              set_color = "black",
              label = "both",
              label_percent_digit = 1,
              label_size = 3,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = )+
  scale_x_continuous(expand = expansion(mult = .2))+
  scale_fill_gradient(low = "red2", high = "yellow")+
  labs(title = "24 hour comparison of significant genes", caption = "n = 8887 genes")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5))


# total 3 ----------------------------------------------------------------

total3 <- list(sigVDA3$ENTREZID,sigVDX3$ENTREZID,sigVEP3$ENTREZID,sigVMT3$ENTREZID)
totalin_common3 <- c(sigVDA3$SYMBOL,sigVDX3$SYMBOL,sigVEP3$SYMBOL,sigVMT3$SYMBOL)
length(unique(in_common3))

ggVennDiagram(total3,
              category.names = c("Daunorubicin","Doxorubicin","Epirubicin","Mitoxantrone"),
              show_intersect = FALSE,
              set_color = "black",
              label = "both",
              label_percent_digit = 1,
              label_size = 3.5,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid") +
  scale_x_continuous(expand = expansion(mult = .2))+
  scale_fill_gradient(low = "red2", high = "yellow")+
  labs(title = "3 hour comparison of significant genes", caption = "n = 554 genes")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5))

# Dauno comp --------------------------------------------------------------

Dauncomp <- list(sigVDA24$ENTREZID,sigVDA3$ENTREZID)
in_commonDa <- c(sigVDA24$ENTREZID,sigVDA3$ENTREZID)
length(unique(in_commonDa))
ggVennDiagram(Dauncomp,
              category.names = c("Daunorubicin-24","Daunorubicin-3"),
              show_intersect = FALSE,
              set_color = "black",
              label = "both",
              label_percent_digit = 1,
              label_size = 3,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = )+
  scale_x_continuous(expand = expansion(mult = .2))+
  scale_fill_gradient(low = "light blue", high = "yellow")+
  labs(title = "Comparision of Dauno 3h v 24h", caption = "n = 7732 genes")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5))
#Davenlist <- intersect(Dauncomp[[1]],Dauncomp[[2]])


# Doxocomp ----------------------------------------------------------------
Doxcomp <- list(sigVDX24$ENTREZID,sigVDX3$ENTREZID)
in_commonDx <- c(sigVDX24$ENTREZID,sigVDX3$ENTREZID)
length(unique(in_commonDx))
ggVennDiagram(Doxcomp,
              category.names = c("Doxorubicin-24","Doxorubicin-3"),
              show_intersect = FALSE,
              set_color = "black",
              label = "both",
              label_percent_digit = 1,
              label_size = 3,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = )+
  scale_x_continuous(expand = expansion(mult = .2))+
  scale_fill_gradient(low = "light blue", high = "yellow")+
  labs(title = "Comparision of Doxo 3h v 24h", caption = "n = 6808 genes")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5))
#Dxvenlist <- intersect(Doxcomp[[1]],Doxcomp[[2]])
#length(intersect(Dxvenlist,Davenlist))## 7 of DX are in DA
# Epi Comp ----------------------------------------------------------------

Epicomp <- list(sigVEP24$ENTREZID,sigVEP3$ENTREZID)
in_commonEp <- c(sigVEP24$ENTREZID,sigVEP3$ENTREZID)
length(unique(in_commonEp))

ggVennDiagram(Epicomp,
              category.names = c("Epirubicin-24","Epirubicin-3"),
              show_intersect = FALSE,
              set_color = "black",
              label = "both",
              label_percent_digit = 1,
              label_size = 3,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = )+
  scale_x_continuous(expand = expansion(mult = .2))+
  scale_fill_gradient(low = "light blue", high = "yellow")+
  labs(title = "Comparision of Epi 3h v 24h", caption = "n = 6858 genes")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5))

#Epvenlist <- intersect(Epicomp[[1]],Epicomp[[2]])
# qA <- (intersect(Epvenlist , Davenlist))##76 are are in da
# qB <- (intersect(Dxvenlist, Epvenlist))#y are in Ep from Dx
# #ACintersect <- intersect(qA,qB)##total of 6 are in all 3 comparisons
# #ggVennDiagram(qA,qB)
#               
# qA
# qB
# Mito comp ---------------------------------------------------------------
Mitocomp <- list(sigVMT24$ENTREZID,sigVMT3$ENTREZID)
in_commonMt <-c(sigVMT24$ENTREZID,sigVMT3$ENTREZID)
length(unique(in_commonMt))

ggVennDiagram(Mitocomp,
              category.names = c("Mitoxantrone-24","Mitoxantrone-3"),
              show_intersect = FALSE,
              set_color = "black",
              label = "both",
              label_percent_digit = 1,
              label_size = 3,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = )+
  scale_x_continuous(expand = expansion(mult = .21))+
  scale_fill_gradient(low = "light blue", high = "yellow")+
  labs(title = "Comparision of Mito 3h v 24h", caption = "n = 1251 genes")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5))
#Mtvenlist <- intersect(Mitocomp[[1]],Mitocomp[[2]])
#intersect(Mtvenlist,ACintersect)
#

# AC comparison -----------------------------------------------------------

ACcomp <- list(sigVDA24$ENTREZID,sigVDX24$ENTREZID,sigVEP24$ENTREZID,sigVDA3$ENTREZID,sigVDX3$ENTREZID,sigVEP3$ENTREZID)
in_commonAC <-c(sigVDA24$ENTREZID,sigVDX24$ENTREZID,sigVEP24$ENTREZID,sigVDA3$ENTREZID,sigVDX3$ENTREZID,sigVEP3$ENTREZID)
length(unique(in_commonAC))

ggVennDiagram(ACcomp,
              category.names = c("Daunorubicin-24","Doxorubicin-24", "Epirubicin-24","Daunorubicin-3","Doxorubicin-3", "Epirubicin-3"),
              show_intersect = FALSE,
              set_color = "black",
              label = "both",
              label_percent_digit = 1,
              label_size = 3,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = )+
  scale_x_continuous(expand = expansion(mult = .21))+
  scale_fill_gradient(low = "purple", high = "yellow")+
  labs(title = "Comparision AC 3h v 24h", caption = "n = 8925 genes")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5))



```
Tell me more!  what are those genes at 24 hours??  give me some GO!
# Graphing GO

```{r graphing GO of 24 hour top2bi intersect, message=FALSE, warning=FALSE, paged.print=TRUE}
list24totvenn <- get.venn.partitions(total24)
# > list24totvenn$..count..
#  [1]  960   13   19   15   37   18   25  126 4440  265  538  533  661  399  838
# > list24totvenn$..values..[?]
# ##From Cormotif
##pairwise
DDEresp <- list24totvenn$..values..[[9]]
DDEMresp <- list24totvenn$..values..[[1]]
Dxresp <- sigVDX24$ENTREZID
DXsprespon <- list24totvenn$..values..[[14]]

top2bi <-  read_csv("data/response_cluster24h.csv",col_select = x, col_types = "c")
NRresp <- read_csv("data/nonresponse_cluster24h.csv",col_select =x,col_types = "c")
colnames(top2bi) <-"ENTREZID"
length(top2bi$ENTREZID)

complete <- c(sigVDA24$ENTREZID,sigVDX24$ENTREZID,sigVEP24$ENTREZID,sigVMT24$ENTREZID)
complete <- as.data.frame(unique(complete))
colnames(complete) <- "ENTREZID"

NoResp <- backGL %>% anti_join(.,(complete), by = "ENTREZID")
# ACresponse_cluster24h <-  read_csv("data/ACresponse_cluster24h.csv", 
#     col_types = cols(...1 = col_skip()))
# length(ACresponse_cluster24h$x)


# nonresponse_cluster24h <-  read_csv("data/nonresponse_cluster24h.csv", 
#     col_types = cols(...1 = col_skip()))
# length(nonresponse_cluster24h$x)
length(intersect(top2bi$ENTREZID,DDEMresp))
       

gostrescorm24 <- gost(query = c(NRresp$x),  
                      organism = "hsapiens",
                      ordered_query = FALSE,
                      domain_scope = "custom",
                      measure_underrepresentation = FALSE,
                      evcodes = FALSE,
                      user_threshold = 0.05,
                      correction_method = c("fdr"),
                      custom_bg = backGL$ENTREZID,
                      sources=c("GO:BP"))
cormotifcluster <- gostplot(gostrescorm24, capped = FALSE, interactive = TRUE)
cormotifcluster

# (gostres$result$p_value)
tableNR <- gostrescorm24$result %>%
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) #%>% 
  # mutate_at(.cols = 6, .funs= scientific_format()) %>% 
  # kable(.,) %>% 
  # kable_paper("striped", full_width = FALSE) %>%  
  # kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover"))
  tableNR
  
  
  tableNR %>% 
    mutate(log_pval=(-log10(p_value))) %>%
     ggplot(.,aes(x=log_pval, y=term_name))
    theme_bw()+
  geom_point( aes(size= intersection_size,col=p_value))+ 
    ggtitle('Non-resoponse genesenriched GO:BP terms')
  table3[1:40,] %>% 
    mutate(log_pval=(-log10(p_value))) %>%
    # filter(term_id,log_pval>15 %>% 
    ggplot(.,aes(x=log_pval, y=term_name))
     theme_bw()+
  geom_point( aes(size= intersection_size,col=p_value))+ 
    ggtitle('Top2Bi enriched GO:BP terms')
log10(table3$p_value)
  
  
```
  
  
  
  # Knowles Comparison data:
  
  pulled from 2018 eLife, supplementary 5
  
  First I will get my data organized into several sets.
  
  Set 1: 'Dx' -Doxorubicin 24 hour significant DEGs p<0.1  
  Set 2: 'DDE' -Union of Daunorubicin, Doxorubicin, Epirubicin 24 hour significant DEGs p<0.1
  Set 3: 'DDEM'-Union of Daunorubicin, Doxorubicin, Epirubicin, and Mitoxantrone 24 hour significant DEGs p<0.1
  set 4: 'NoResp' - all genes at 24 hours that are not expressed significantly in any treatment  
  set 5: 'TOP2Bi' -cormotif intersection of 24 hour samples compared to Vehicle with p>0.5
  set 6: Knowles list from supp 5 in 2018 eLife paper  eQTL
```{r  knowles data comparisons}
library(biomaRt)
# library(org.Hs.eg.db)
# library(Homo.sapiens)
# library("AnnotationHub")
# library(EnsDb.Hsapiens.v79)
# if (!require("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
entrez <- useMart("entrez",dataset="hsapiens_gene_entrez")
ensembl <- useMart("ensembl", dataset="hsapiens_gene_ensembl")
my_chr <- c(1:22, 'M', 'X', 'Y')  ## creates a filter for each database
attributes <- listAttributes((ensembl))

Knowles_2018.elife.33480.supp5.v2 <- read.delim("~/Ward Lab/Cardiotoxicity/Manuscript/Knowles_2018-elife-33480-supp5-v2/Knowles_2018-elife-33480-supp5-v2")
# Knowles_2018.elife.33480.supp6.v2 <- read.delim("~/Ward Lab/Cardiotoxicity/Manuscript/Knowles_2018-elife-33480-supp6-v2/Knowles_2018-elife-33480-supp6-v2")
# # #  read in supp file
# supp5list <- Knowles_2018.elife.33480.supp5.v2[,1]
# supp6list <- Knowles_2018.elife.33480.supp6.v2[,1]

my_attributes <- c('entrezgene_id', 'ensembl_gene_id', 'hgnc_symbol')
knowles5 <- Knowles_2018.elife.33480.supp5.v2[,1]

knowles5 <- getBM(attributes=my_attributes,filters ='ensembl_gene_id',
                 values = knowles5, mart = ensembl)
length(unique(knowles5$ensembl_gene_id))#376+
length(unique(knowles5$entrezgene_id))#377
# 
# in_common24AC <- intersect(sigVDA24$ENTREZID,sigVDX24$ENTREZID)
# in_common24AC <- intersect(in_common24AC,sigVEP24$ENTREZID)
# AC24hoursig <- (unique(in_common24AC))  ### 5400 total genes from all 24 h AC
# AConly24hsig <- getBM(attributes=my_attributes,filters ='entrezgene_id',
#                   values = AC24hoursig, mart = ensembl)
# length(unique(AConly24hsig$ensembl_gene_id))#5736
# length(unique(AConly24hsig$entrezgene_id))#5245

totalgenes <- getBM(attributes=my_attributes,filters ='entrezgene_id',
                  values = backGL$ENTREZID, mart = ensembl)
length(unique(totalgenes$entrezgene_id))#14184
length(unique(totalgenes$ensembl_gene_id))#15596
DX24Respset<- intersect(totalgenes$ensembl_gene_id, sigDox24$ensembl_gene_id)
length(unique(DX24Respset))  #7182 (#185 in common with knowls5)
15596-7182  #=8414

comp1 <- list(knowles5$entrezgene_id,top2bi$entrezgene_id)
  ggVennDiagram(comp1,
              category.names = c("Knowles", "Top2Bi-24hours"),
              show_intersect = FALSE,
              set_color = "black",
              label = "both",
              label_percent_digit = 1,
              label_size = 3,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = )+
  scale_x_continuous(expand = expansion(mult = .21))+
  scale_fill_gradient(low = "red", high = "yellow")+
  labs(title = "Comparision Knowles v 24h", caption = "n = 376")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5))
  

  
  
  
  
  
  

```
  
  
  
  
  
  
  Seone, Jose  Chromatin gene comparison:   comes from supp data NAT. MED 2019



```{r gene comparison}
library(readxl)

# Seoane_chromatinregs <- read_excel("C:/Users/renee/Downloads/Supplements folde manuscriptr/NIHMS1539805-supplement-SuppTables.xlsx", 
#     range = "A13:H469")

#write.csv(Seoane_chromatinregs, "data/Seonane2019supp1.txt")

chrom_reg_Seoane <- read_csv(file = "data/Seonane2019supp1.txt",col_types = cols(...1 = col_skip()))
                            
Seoane_2019 <- chrom_reg_Seoane[,2]
names(Seoane_2019) <- "ENTREZID"







```
## GO analysisl of DEG from Cormotif
```{r go of NR genes}

NoResp <- backGL %>% 
  anti_join(.,(complete), by ="ENTREZID")


     length(NoResp$ENTREZID) 
     
DEG_cormotif <- readRDS("data/DEG_cormotif.RDS")

gostresNR <- gost(query =DEG_cormotif[[1]], 
                  organism = "hsapiens",
                  ordered_query = FALSE,
                  domain_scope = "custom",
                  measure_underrepresentation = FALSE,
                  evcodes = TRUE,
                  user_threshold = 0.05,
                  correction_method = c("fdr"),
                  custom_bg = backGL$ENTREZID,
                  sources="GO:BP", significant = FALSE)


wNo_response_set <- gostplot(gostresNR, capped = FALSE, interactive = TRUE)
 #sample(NoResp$ENTREZID,3)
exampNR <- c(4287,56946,4478)
# (gostres$result$p_value)
table4 <- No_response_set$result %>%
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) #%>% 
  # mutate_at(.cols = 6, .funs= scientific_format()) %>% 
  # kable(.,) %>% 
  # kable_paper("striped", full_width = FALSE) %>%  
  # kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover"))
  table4


table4[1:20,] %>% mutate(gene_ratio = intersection_size/term_size) %>%
  ggplot(.,aes(x=gene_ratio, y=term_name))+
  geom_point(aes(size= intersection_size, color = p_value))
  
  table4[1:20,] %>% mutate(gene_ratio = intersection_size/term_size) %>%
  ggplot(.,aes(x=gene_ratio, y=term_name))+
  geom_point(aes(size= intersection_size, color = p_value))+ggtitle("No Response Gene set")

```





```{r running gene lists}
### list of genes I want to see expression from
#sample(complete$ENTREZID, 3)
library(readr)
# ensembl <- useMart("ensembl", dataset="hsapiens_gene_ensembl")
my_attributes <- c('entrezgene_id', 'ensembl_gene_id', 'hgnc_symbol')
Response_24list <- c(1119,2950,6556)
Resp24_list <- getBM(attributes=my_attributes,
                     filters ='entrezgene_id',
                     values = Response_24list, 
                     mart = ensembl)
lookupforlevel <- getBM(attributes=my_attributes,
                     filters ='entrezgene_id',
                     values =renamelvl, 
                     mart = ensembl)
lookupforlevel <- lookupforlevelunique()
NRresp_list <- getBM(attributes=my_attributes,
                     filters ='entrezgene_id',
                     values = exampNR, 
                     mart = ensembl)

exampNR <- c(4287,56946,4478)

#find the "indecies for those genes in the matrix"
exampNR,x
counts24hours <- read_rds("data/counts24hours.RDS")
counts24da<- read_csv("data/Da24counts.txt")
counts24dx<- read_csv("data/Dx24counts.txt")
counts24ep<- read_csv("data/Ep24counts.txt")
counts24mt<- read_csv("data/Mt24counts.txt")
counts24tr<- read_csv("data/TR24counts.txt")


levels(counts24hours$Samples) <- c("Da.24h"="Daunorubicin",
                                           "Dx.24h"="Doxorubicin",
                                           "Ep.24h"="Epirubicin",
                                           "Mt.24h"="Mitoxantrone",
                                         "Tr.24h"="Trastuzumab",
                                         "Ve.24h"="Control") 
 counts24hours %>% dplyr::filter(ENTREZID %in% Response_24list) %>%
  group_by(Samples) %>%
  ggplot(.,aes(x= Samples, y=Counts))+
  geom_boxplot()+
 
  facet_grid(~ENTREZID,labeller = cols= 2, scales ="free")
idx <- match(Emmalist$ENTREZID,counts24hours$ENTREZID)


namelist <- c("1026"="CDKN1A","23411"= "SIRT1","27113"= "BBC3","4193"= "MDM2")
Emmalist <- select(org.Hs.eg.db, 
       keys = Emmalist,
       columns = c("ENTREZID", "SYMBOL"),
       keytype = "SYMBOL")






counts24hours %>% dplyr::filter(ENTREZID %in% Emmalist$ENTREZID) %>%
  group_by(Samples) %>%
  ggplot(.,aes(x= Samples, y=Counts))+
  geom_boxplot(aes(col=Samples))+
  facet_grid(~ENTREZID,labeller=labeller(ENTREZID = namelist))
  
#                       

getBM(attributes=my_attributes,
                     filters ='entrezgene_id',
                     values = Response_24list, 
                     mart = ensembl)

```





