---
title: "GO analysis"
author: "ERM"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# GO Analysis

I have created several files from the RNA analysis that contain the significant genes(determined by adj.P.val \< 0.1) from each Time and Condition. The names of the files are in the following format: 'sigV'+Drug(2 letters)+time.

example: 'sigVDA3.txt' means this file contains the significant DE genes from the Daunorubicin 3 hour compared to Vehicle Control 3 hour analysis

```{r Import libraries, message=FALSE, warning=FALSE}
library(gprofiler2)
library(tidyverse)
library(ggplot2)
library(readr)
library(BiocGenerics)
library(gridExtra)
library(VennDiagram)
library(kableExtra)
library(scales)
library(ggVennDiagram)

```

```{r Import data, echo=FALSE, message=FALSE, warning=FALSE}
 
##note, in this chunk the .rmd file is in the cm/analysis folder.   need to back out with .. 
##this code checks the directory and searches for the pattern, returning the full name of the file
file.names <- list.files(path = "data/", pattern = "sig*", ignore.case = TRUE,full.names = TRUE)

##Next I use the list of names and lapply to read all files into a list
##made the csv file to reimport
#write.csv(filenameonly,"data/filenameonly.txt", col.names = FALSE,row.names = FALSE)
##readin the filenameonly.txt file from data to use for naming and filtering.
filenameonly <- read_csv("data/filenameonly.txt")
#loop through the list of files and make a separate dataframe for each file under the 'real'  name of the data set
for (k in 1:length(file.names)){
  
  assign(paste0(filenameonly$x[k]) , read.csv(file.names[k]))
}

##rename the columns to the previous names

colnames(sigVDA24)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVDX24)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVEP24)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVMT24)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVTR24)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVDA3)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVDX3)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVEP3)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVMT3)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
colnames(sigVTR3)<- c("ENTREZID","SYMBOL","logFC","AveExpr","t","P.Value","adj.P.Val","B")
```

The analysis is based on all genes that passed the rowMeans\>0 from the previous page [link](https://reneeisnowhere.github.io/Cardiotoxicity/RNAseqrun_1_analysis.html)

```{r uploading the background genes, echo=FALSE,  message=FALSE, warning=FALSE, paged.print=FALSE}
#backGL <- efit2$genes   ***making the list

#write.csv(backGL, "data/backGL.txt", row.names = FALSE)
##read the  .csv the render into txt in the data file!
 backGL <- read_csv("data/backGL.txt", 
    col_types = cols(...1 = col_skip()))
```

Below is the analysis of differentially expressed genes for each treatment at 3 hours and 24 hours.

```{r intial analysis, echo=FALSE, message=FALSE, warning=FALSE}
gostres <- gost(query = sigVDA3$SYMBOL,  organism = "hsapiens",
  ordered_query = TRUE,
  domain_scope = "custom",
  measure_underrepresentation = FALSE,
  evcodes = FALSE,
  user_threshold = 0.0001,
  correction_method = c("fdr"),
  custom_bg = backGL$ENTREZID,
  sources=c("GO:BP","GO:MF", "GO:CC"))
p <- gostplot(gostres, capped = FALSE, interactive = TRUE)
p

table1 <- gostres$result %>% 
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) %>% 
   mutate_at(.cols = 6, .funs= scientific_format()) %>% 
  kable(.,) %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover"))
  table1
# publish_gosttable(
#   gostres,
#   highlight_terms = NULL,
#   use_colors = TRUE,
#   show_columns = c("source", "term_name", "term_size", "intersection_size"),
#   filename = "../output/test.pdf",
#   ggplot = TRUE
# )
```

I first looked at the data with all genes from the sigDA3 dataset. I used the list of all genes based on my rowMeans\>0 filtering as background.

### Analysis of Up versus Down

I then separated the VDA3 file by log2 Fold Change to see how the gene sets are enriched. Nothing showed up in the GO-BP/CC/MG-down regulated gene-set at a significant level, p\<0.05.

```{r only the Da3 data3, echo=FALSE, message=FALSE, warning=FALSE}
#results_sig = subset(sigVDA3,adj.P.Val < 0.05)
# get the significant up-regulated genes
  up = subset(sigVDA3, logFC > 0)
# get the significant down-regulated genes
down = subset(sigVDA3, logFC < 0)
gp_up = gost(query = up$ENTREZID,  organism = "hsapiens",
  ordered_query = TRUE,
  domain_scope = "custom",
  measure_underrepresentation = FALSE,
  evcodes = FALSE,
  user_threshold = 0.001,
  correction_method = c("fdr"),
  custom_bg = backGL$ENTREZID,
  sources=c("GO:BP","GO:MF", "GO:CC"))
  
  
#gp_down = gost(query = down$ENTREZID,  organism = "hsapiens",
  # ordered_query = TRUE,
  # domain_scope = "custom",
  # measure_underrepresentation = FALSE,
  # evcodes = FALSE,
  # user_threshold = 0.05,
  # correction_method = c("bonferroni"),
  # custom_bg = backGL$ENTREZID,
  # sources=c("GO:BP","GO:MF", "GO:CC"))
p2_up <- gostplot(gp_up, capped = FALSE, interactive = TRUE)

p2_up #+ ggtitle("Daunorubicin up regulated gene enrichment at 3 hours")

```

Note this is at three hour for Daunorubicin

```{r showing it all, eval = FALSE, echo=FALSE, message=FALSE, warning=FALSE}

#p2_down <- gostplot(gp_down, capped = FALSE, interactive = TRUE)

#p2_down #+ ggtitle("Daunorubicin down regulated gene enrichment at 3 hours")


```

#### I next wanted to see what happened at 24 hours with daunorubicin. I used the sigVDA24 file to do this.

```{r DA24, eval=TRUE, echo=FALSE, message=FALSE, warning=FALSE, paged.print=TRUE}
gostresDA24 <- gost(query = sigVDA24$SYMBOL,  organism = "hsapiens",
  ordered_query = TRUE,
  domain_scope = "custom",
  measure_underrepresentation = FALSE,
  evcodes = FALSE,
  user_threshold = 0.0001,
  correction_method = c("fdr"),
  custom_bg = backGL$ENTREZID,
  sources=c("GO:BP","GO:MF", "GO:CC"))
pDA24 <- gostplot(gostres, capped = FALSE, interactive = TRUE)
pDA24

# (gostres$result$p_value)
table2 <- gostresDA24$result %>%
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) %>% 
  mutate_at(.cols = 6, .funs= scientific_format()) %>% 
  kable(.,) %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover"))
  table2
# publish_gosttable(gostresDA24,
#   highlight_terms = gostresDA24$result,
#   use_colors = TRUE,
#   show_columns = c("source", "term_name", "term_size", "intersection_size"),
#   filename = "../output/test.pdf",
#   ggplot = TRUE
# )

```
Here I was able to visualize the 24 hour GO analysis of daunorubicin.

##Venn Diagrams
```{r venndiagram, echo=FALSE, message=FALSE, warning=FALSE}
total24 <-list(sigVDA24$ENTREZID,sigVDX24$ENTREZID,sigVEP24$ENTREZID,sigVMT24$ENTREZID)
in_common24 <-c(sigVDA24$ENTREZID,sigVDX24$ENTREZID,sigVEP24$ENTREZID,sigVMT24$ENTREZID)

length(unique(in_common24))

ggVennDiagram(total24,
              category.names = c("Daunorubicin","Doxorubicin","Epirubicin","Mitoxantrone"),
              show_intersect = FALSE,
              set_color = "black",
              label = "both",
              label_percent_digit = 1,
              label_size = 3,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = )+
  scale_x_continuous(expand = expansion(mult = .2))+
  scale_fill_gradient(low = "red2", high = "yellow")+
  labs(title = "24 hour comparison of significant genes", caption = "n = 8887 genes")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5))


# total 3 ----------------------------------------------------------------

total3 <- list(sigVDA3$ENTREZID,sigVDX3$ENTREZID,sigVEP3$ENTREZID,sigVMT3$ENTREZID)
in_common3 <- c(sigVDA3$SYMBOL,sigVDX3$SYMBOL,sigVEP3$SYMBOL,sigVMT3$SYMBOL)
length(unique(in_common3))

ggVennDiagram(total3,
              category.names = c("Daunorubicin","Doxorubicin","Epirubicin","Mitoxantrone"),
              show_intersect = FALSE,
              set_color = "black",
              label = "both",
              label_percent_digit = 1,
              label_size = 3.5,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid") +
  scale_x_continuous(expand = expansion(mult = .2))+
  scale_fill_gradient(low = "red2", high = "yellow")+
  labs(title = "3 hour comparison of significant genes", caption = "n = 554 genes")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5))

# Dauno comp --------------------------------------------------------------

Dauncomp <- list(sigVDA24$ENTREZID,sigVDA3$ENTREZID)
in_commonDa <- c(sigVDA24$ENTREZID,sigVDA3$ENTREZID)
length(unique(in_commonDa))
ggVennDiagram(Dauncomp,
              category.names = c("Daunorubicin-24","Daunorubicin-3"),
              show_intersect = FALSE,
              set_color = "black",
              label = "both",
              label_percent_digit = 1,
              label_size = 3,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = )+
  scale_x_continuous(expand = expansion(mult = .2))+
  scale_fill_gradient(low = "light blue", high = "yellow")+
  labs(title = "Comparision of Dauno 3h v 24h", caption = "n = 7732 genes")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5))
#Davenlist <- intersect(Dauncomp[[1]],Dauncomp[[2]])


# Doxocomp ----------------------------------------------------------------
Doxcomp <- list(sigVDX24$ENTREZID,sigVDX3$ENTREZID)
in_commonDx <- c(sigVDX24$ENTREZID,sigVDX3$ENTREZID)
length(unique(in_commonDx))
ggVennDiagram(Doxcomp,
              category.names = c("Doxorubicin-24","Doxorubicin-3"),
              show_intersect = FALSE,
              set_color = "black",
              label = "both",
              label_percent_digit = 1,
              label_size = 3,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = )+
  scale_x_continuous(expand = expansion(mult = .2))+
  scale_fill_gradient(low = "light blue", high = "yellow")+
  labs(title = "Comparision of Doxo 3h v 24h", caption = "n = 6808 genes")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5))
#Dxvenlist <- intersect(Doxcomp[[1]],Doxcomp[[2]])
#length(intersect(Dxvenlist,Davenlist))## 7 of DX are in DA
# Epi Comp ----------------------------------------------------------------

Epicomp <- list(sigVEP24$ENTREZID,sigVEP3$ENTREZID)
in_commonEp <- c(sigVEP24$ENTREZID,sigVEP3$ENTREZID)
length(unique(in_commonEp))

ggVennDiagram(Epicomp,
              category.names = c("Epirubicin-24","Epirubicin-3"),
              show_intersect = FALSE,
              set_color = "black",
              label = "both",
              label_percent_digit = 1,
              label_size = 3,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = )+
  scale_x_continuous(expand = expansion(mult = .2))+
  scale_fill_gradient(low = "light blue", high = "yellow")+
  labs(title = "Comparision of Epi 3h v 24h", caption = "n = 6858 genes")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5))

#Epvenlist <- intersect(Epicomp[[1]],Epicomp[[2]])
# qA <- (intersect(Epvenlist , Davenlist))##76 are are in da
# qB <- (intersect(Dxvenlist, Epvenlist))#y are in Ep from Dx
# #ACintersect <- intersect(qA,qB)##total of 6 are in all 3 comparisons
# #ggVennDiagram(qA,qB)
#               
# qA
# qB
# Mito comp ---------------------------------------------------------------
Mitocomp <- list(sigVMT24$ENTREZID,sigVMT3$ENTREZID)
in_commonMt <-c(sigVMT24$ENTREZID,sigVMT3$ENTREZID)
length(unique(in_commonMt))

ggVennDiagram(Mitocomp,
              category.names = c("Mitoxantrone-24","Mitoxantrone-3"),
              show_intersect = FALSE,
              set_color = "black",
              label = "both",
              label_percent_digit = 1,
              label_size = 3,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = )+
  scale_x_continuous(expand = expansion(mult = .21))+
  scale_fill_gradient(low = "light blue", high = "yellow")+
  labs(title = "Comparision of Mito 3h v 24h", caption = "n = 1251 genes")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5))
#Mtvenlist <- intersect(Mitocomp[[1]],Mitocomp[[2]])
#intersect(Mtvenlist,ACintersect)
#

# AC comparison -----------------------------------------------------------

ACcomp <- list(sigVDA24$ENTREZID,sigVDX24$ENTREZID,sigVEP24$ENTREZID,sigVDA3$ENTREZID,sigVDX3$ENTREZID,sigVEP3$ENTREZID)
in_commonAC <-c(sigVDA24$ENTREZID,sigVDX24$ENTREZID,sigVEP24$ENTREZID,sigVDA3$ENTREZID,sigVDX3$ENTREZID,sigVEP3$ENTREZID)
length(unique(in_commonAC))

ggVennDiagram(ACcomp,
              category.names = c("Daunorubicin-24","Doxorubicin-24", "Epirubicin-24","Daunorubicin-3","Doxorubicin-3", "Epirubicin-3"),
              show_intersect = FALSE,
              set_color = "black",
              label = "both",
              label_percent_digit = 1,
              label_size = 3,
              label_alpha = 0,
              label_color = "black",
              edge_lty = "solid", set_size = )+
  scale_x_continuous(expand = expansion(mult = .21))+
  scale_fill_gradient(low = "red", high = "yellow")+
  labs(title = "Comparision AC 3h v 24h", caption = "n = 8925 genes")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5))



```
Tell me more!  what are those genes at 24 hours??  give me some GO!
# Graphing GO

```{r graphing GO of 24 hour intersect, message=FALSE, warning=FALSE, paged.print=TRUE}
# list24totvenn <- get.venn.partitions(total24)
# all24hour <- listvenn$values[1]

##From Cormotif
response_cluster24h <-  read_csv("data/response_cluster24h.csv", 
    col_types = cols(...1 = col_skip()))
length(response_cluster24h$x)


gostrescorm24 <- gost(query = response_cluster24h$x,  organism = "hsapiens",
  ordered_query = FALSE,
  domain_scope = "custom",
  measure_underrepresentation = FALSE,
  evcodes = FALSE,
  user_threshold = 0.0001,
  correction_method = c("fdr"),
  custom_bg = backGL$ENTREZID,
  sources=c("GO:BP","GO:MF", "GO:CC"))
pDA24 <- gostplot(gostrescorm24, capped = FALSE, interactive = TRUE)
pDA24

# (gostres$result$p_value)
table3 <- gostrescorm24$result %>%
  dplyr::select(c(source, term_id, term_name,intersection_size, term_size, p_value)) %>% 
  mutate_at(.cols = 6, .funs= scientific_format()) %>% 
  kable(.,) %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover"))
  table3
```
  
