---
title: "Knowles 2019 data"
author: "ERM"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE
)
```


###### package loading                              
```{r package loading, echo=FALSE}

library(limma)
library(RColorBrewer)
library(tidyverse)
library(ggsignif)
library(cowplot)
library(ggpubr)
library(scales)
library(sjmisc)
library(kableExtra)
library(broom)
library(ComplexHeatmap)
``` 
## function load
```{r cpmboxplot function, echo= FALSE}
cpm_boxplot <-function(cpmcounts, GOI,brewer_palette, fill_colors, ylab) {
  ##GOI needs to be ENTREZID
  df <- cpmcounts
    df_plot <- df %>% 
      dplyr::filter(rownames(.)==GOI) %>%
      pivot_longer(everything(),
                   names_to = "treatment",
                   values_to = "counts") %>%
      separate(treatment, c("drug","indv","time")) %>%
      mutate(time=factor(time, levels =c("3h", "24h"))) %>%
      mutate(indv=factor(indv, levels = c(1,2,3,4,5,6))) %>%
      mutate(drug =case_match(drug, "Da"~"DNR",
                            "Do"~"DOX",
                            "Ep"~"EPI",
                            "Mi"~"MTX",
                            "Tr"~"TRZ",
                            "Ve"~"VEH", .default = drug)) %>% 
      mutate(drug=factor(drug, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH')))
    plot <- ggplot2::ggplot(df_plot, aes(x=drug, y=counts))+
      geom_boxplot(position="identity",aes(fill=drug))+
      geom_point(aes(col=indv, size=1.5, alpha=0.5))+
      guides(alpha= "none", size= "none")+
      scale_color_brewer(palette = brewer_palette, guide = "none")+
      scale_fill_manual(values=fill_colors)+
      facet_wrap("time", nrow=1, ncol=2)+
      theme_bw()+
      ylab(ylab)+
      xlab("")+
      theme(strip.background = element_rect(fill = "white",linetype=1, linewidth = 0.5),
          plot.title = element_text(size=12,hjust = 0.5,face="bold"),
          axis.title = element_text(size = 10, color = "black"),
          axis.ticks = element_line(linewidth = 1.0),
          axis.line = element_line(linewidth = 1.0),
          axis.text.x = element_blank(),
          strip.text.x = element_text(margin = margin(2,0,2,0, "pt"),face = "bold"))
    print(plot)
}


```
## Knowles data
  The code below is how I wrangled the knowles supplemental lists into entrezgene_ids to overlap with my expressed gene lists
```{r ensembl conversion, eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
library(biomaRt)
ensembl <- useMart("ensembl", dataset="hsapiens_gene_ensembl")
my_chr <- c(1:22, 'M', 'X', 'Y')  ## creates a filter for each
my_attributes <- c('entrezgene_id', 'ensembl_gene_id', 'hgnc_symbol')
knowfig4 <- read.csv("data/knowfig4.csv",row.names = 1)
knowfig5 <- read.csv("data/knowfig5.csv",row.names = 1)

# covert ensg to entrezid -------------------------------------------------

knowles5 <- getBM(attributes=my_attributes,filters ='ensembl_gene_id',
                  values =knowfig5, mart = ensembl)
#377

knowles4 <- getBM(attributes=my_attributes,filters ='ensembl_gene_id',
                  values =knowfig4, mart = ensembl)
#524
notOK <- intersect(knowles4$entrezgene_id,knowles5$entrezgene_id)
##104

knowles4 <-knowles4 %>% 
  distinct(entrezgene_id,.keep_all = TRUE) %>% 
  filter(., !entrezgene_id%in% notOK)
  #521 after filtering out overlap-417 genes
knowles5 <-knowles5 %>% 
  distinct(entrezgene_id,.keep_all = TRUE)%>% 
  filter(., !entrezgene_id%in% notOK) 
#377 after filtering out 273 overlap-genes

# saveRDS(knowles4,"output/knowles4.RDS")
# saveRDS(knowles5,"output/knowles5.RDS")
  #Creating the Dox me/ re QTL data set:


DOXreQTLs <- toplistall %>% 
  dplyr::filter(time =="24_hours") %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>% 
   dplyr::filter(id == "DOX") %>% 
   dplyr::filter(sigcount=='sig') %>% 
   dplyr::filter(reQTL=="a") %>% 
   dplyr::select(time, id, ENTREZID,SYMBOL)

saveRDS(DOXreQTLs,"output/DOXreQTLs.RDS")
DOXmeSNPs <- toplistall %>% 
  dplyr::filter(time =="24_hours") %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>% 
   dplyr::filter(id == "DOX") %>% 
   dplyr::filter(sigcount=='sig') %>% 
   dplyr::filter(meSNP=="y") %>% 
   dplyr::select(time, id, ENTREZID,SYMBOL)
saveRDS(DOXmeSNPs,"output/DOXmeSNPs.RDS")
#Creating the DNR me/re QTL data sets:
DNRreQTLs <- toplistall %>% 
dplyr::filter(time =="24_hours") %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>%
  dplyr::filter(id =="DNR") %>% 
  dplyr::filter(sigcount=='sig') %>% 
   dplyr::filter(reQTL=="a") %>% 
   dplyr::select(time, id, ENTREZID,SYMBOL)
saveRDS(DNRreQTLs,"output/DNRreQTLs.RDS")
DNRmeSNPs <- toplistall %>% 
dplyr::filter(time =="24_hours") %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>%
  dplyr::filter(id =="DNR") %>% 
  dplyr::filter(sigcount=='sig') %>% 
   dplyr::filter(meSNP=="y") %>% 
   dplyr::select(time, id, ENTREZID,SYMBOL)
saveRDS(DNRmeSNPs,"output/DNRmeSNPs.RDS")

#Creating the Epi me/re QTL data sets:
EPIreQTLs <- toplistall %>% 
  dplyr::filter(time =="24_hours") %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>%
   dplyr::filter(id =="EPI") %>% 
   dplyr::filter(sigcount=='sig') %>% 
   dplyr::filter(reQTL=="a") %>% 
   dplyr::select(time, id, ENTREZID,SYMBOL)
saveRDS(EPIreQTLs,"output/EPIreQTLs.RDS")

EPImeSNPs <- toplistall %>% 
  dplyr::filter(time =="24_hours") %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>%
   dplyr::filter(id =="EPI") %>% 
   dplyr::filter(sigcount=='sig') %>% 
    dplyr::filter(meSNP=="y") %>% 
   dplyr::select(time, id, ENTREZID,SYMBOL)
saveRDS(EPImeSNPs,"output/EPImeSNPs.RDS")


#Creating the MTX me/re QTL data sets:

MTXreQTLs <- toplistall %>% 
  dplyr::filter(time =="24_hours") %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>%
   dplyr::filter(id =="MTX") %>% 
   dplyr::filter(sigcount=='sig') %>% 
   dplyr::filter(reQTL=="a") %>% 
   dplyr::select(time, id, ENTREZID,SYMBOL)
saveRDS(MTXreQTLs,"output/MTXreQTLs.RDS")

MTXmeSNPs <- toplistall %>% 
  dplyr::filter(time =="24_hours") %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>%
   dplyr::filter(id =="MTX") %>% 
   dplyr::filter(sigcount=='sig') %>% 
    dplyr::filter(meSNP=="y") %>% 
   dplyr::select(time, id, ENTREZID,SYMBOL)
saveRDS(MTXmeSNPs,"output/MTXmeSNPs.RDS")


```
## Enrichment of Pairwise
A note on Knowles data sets:  the intersection of the Knowles-4 and Knowles-5 sets yielded 104 genes in 

``` {r loading in data sets}

backGL <- read.csv("data/backGL.txt")
drug_palNoVeh <- c("#8B006D","#DF707E","#F1B72B", "#3386DD","#707031")
drug_palc <- c("#8B006D","#DF707E","#F1B72B", "#3386DD","#707031","#41B333")
time <- rep((rep(c("3h", "24h"), c(6,6))), 6)
time <- ordered(time, levels =c("3h", "24h"))
drug <- rep(c("DNR","DOX","EPI","MTX","TRZ", "Vehicle"),12)
mat_drug <- c("DNR","DOX","EPI","MTX")
indv <- as.factor(rep(c(1,2,3,4,5,6), c(12,12,12,12,12,12)))
labeltop <- (interaction(substring(drug, 0, 2), indv, time))
level_order2 <- c('75','87','77','79','78','71')


toplistall <- readRDS("data/toplistall.RDS")
knowles4 <-readRDS("output/knowles4.RDS")
knowles5 <-readRDS("output/knowles5.RDS")
DOXmeSNPs <- readRDS("output/DOXmeSNPs.RDS")
DOXreQTLs <- readRDS("output/DOXreQTLs.RDS")
DNRmeSNPs <- readRDS("output/DNRmeSNPs.RDS")
DNRreQTLs <- readRDS("output/DNRreQTLs.RDS")
EPImeSNPs <- readRDS("output/EPImeSNPs.RDS")
EPIreQTLs <- readRDS("output/EPIreQTLs.RDS")
MTXmeSNPs <- readRDS("output/MTXmeSNPs.RDS")
MTXreQTLs <- readRDS("output/MTXreQTLs.RDS")
cpmcounts <- readRDS("data/cpmcount.RDS")
toplist24hr <-   toplistall %>%
  mutate(id =case_match(id, "Da"~"DNR",
                            "Do"~"DOX",
                            "Ep"~"EPI",
                            "Mi"~"MTX",
                            "Tr"~"TRZ",
                            "Ve"~"VEH", .default = id)) %>% 
      mutate(id=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>% 
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  filter(time=="24_hours")

toplist3hr <-   toplistall %>%
  mutate(id =case_match(id, "Da"~"DNR",
                            "Do"~"DOX",
                            "Ep"~"EPI",
                            "Mi"~"MTX",
                            "Tr"~"TRZ",
                            "Ve"~"VEH", .default = id)) %>% 
      mutate(id=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>% 
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  filter(time=="3_hours")

```
Determining the genetic basis of anthracycline-cardiotoxicity by molecular response QTL mapping in induced cardiomyocytes  
David A Knowles, Courtney K Burrowsâ€ , John D Blischak,
Kristen M Patterson, Daniel J Serie, Nadine Norton, Carole Ober,
Jonathan K Pritchard, Yoav Gilad

Knowles $~~et~ al.~$ eLife 2018;7:e33480. DOI: https://doi.org/10.7554/eLife.33480  

My first question was about transcriptional response at the 24 hour mark with my treatments.   3 hour RNA-seq had low numbers of DEGs,so my initial focus is at 24 hours.  This time also happens to be when the Knowles paper collected their RNA-seq data.  

Supplementary 4 contains a list of 518 SNPs within 1 Mb of TSS, which had a detectable marginal effect on expression (5% FDR). When converted from ensembl gene id to entrez gene id,  my list of genes for this supplement = `r length(knowles4$entrezgene_id)`.  I will call these meSNPS for marginal effect QTLs.  

Supplementary 5 contains a list of 376 response eQTLs (reQTLs).  These are variants that were associated with modulation of transcriptomic response to DOX treatment. After database name conversion, I have `r length(knowles5$entrezgene_id)` genes.
```{r QTL enrichment numbers, message=FALSE, warning=FALSE}

meSNPdf <- toplistall %>% 
   mutate(id=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>% 
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP = if_else( ENTREZID %in% knowles4$entrezgene_id, "y" , "no")) %>% 
  dplyr::select(ENTREZID,id,time,meSNP,sigcount) %>%
  group_by(id,time,meSNP,sigcount) %>% 
  tally() %>% 
  pivot_wider(id_cols = c(id,time,sigcount), names_from = meSNP,names_glue = "meSNP_{meSNP}", values_from = n)

meSNPdf %>% 
  kable(., caption= "Number of genes from minimally expressed SNPs found in DEGs (sig and nonsig) broken down by drug,time, and significance") %>% 
  kable_paper("striped", full_width = FALSE) %>% 
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "80%", height = "400px")



reQTLdf <- toplistall %>% 
   mutate(id=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>% 
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(reQTL = if_else( ENTREZID %in% knowles5$entrezgene_id, "y" , "no")) %>% 
  dplyr::select(ENTREZID,id,time,reQTL,sigcount) %>%
  group_by(id,time,reQTL,sigcount) %>% 
  tally() %>% 
  pivot_wider(id_cols = c(id,time,sigcount), names_from = reQTL,names_glue = "reQTL_{reQTL}", values_from = n)

reQTLdf %>% 
  kable(., caption= "Number of genes from response eQTLS found in DEGs (sig and nonsig) broken down by drug,time, and significance") %>% 
  kable_paper("striped", full_width = FALSE) %>% 
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "80%", height = "400px")


dataframK_45 <- toplistall %>%
  mutate(id =case_match(id, "Da"~"DNR",
                            "Do"~"DOX",
                            "Ep"~"EPI",
                            "Mi"~"MTX",
                            "Tr"~"TRZ",
                            "Ve"~"VEH", .default = id)) %>%
  mutate(id=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>% 
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  mutate(K4 = if_else(ENTREZID %in% knowles4$entrezgene_id,'y','no'))%>%
  mutate(K5 = if_else(ENTREZID %in% knowles5$entrezgene_id,'y','no'))%>%
  filter(adj.P.Val<0.05) %>%
   group_by(time, id) %>%
 dplyr::summarize(n=n(), meSNP=sum(if_else(K4=="y",1,0)), reQTL=sum(if_else(K5=="y",1,0))) %>% 
  as.data.frame()


dataframK_45 %>% 
kable(., caption= "Summary of meSNPs and reQTLs found in my DEGs") %>% 
  kable_paper("striped", full_width = FALSE) %>% 
  kable_styling(font_size = 18) %>% 
  scroll_box(width = "80%", height = "400px")
```
###  Visualization of proportions
 
```{r K4K5 proportions, message=FALSE, warning=FALSE}

toplistall %>%
   mutate(id=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>% 
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>% 
  filter(time =="24_hours") %>% 
  group_by(id,sigcount,meSNP) %>% 
  summarize(K4count=n())%>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(meSNP), values_from=K4count) %>% 
   mutate(SNPprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=SNPprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",SNPprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("non-significant and significant enrichment proportions of Knowles meSNPs ")
toplistall %>%
   mutate(id=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>% 
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>% 
  filter(time =="24_hours") %>% 
  group_by(id,sigcount,reQTL) %>% 
  summarize(K5count=n())%>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(reQTL), values_from=K5count) %>% 
   mutate(QTLprop=(a/(a+b)*100)) %>% 
       ggplot(., aes(x=id, y=QTLprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",QTLprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("non-significant and significant enrichment proportions of Knowles reQTLs ")

chi_squarek4 <- toplistall %>% 
   mutate(id=factor(id, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH'))) %>% 
  dplyr::filter(id!="TRZ") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  group_by(id,time) %>% 
  summarise(pvalue= chisq.test(meSNP, sigcount)$p.value) 

chi_squarek5 <-   toplistall %>% 
   mutate(id=factor(id,levels =c('DOX','EPI','DNR','MTX','TRZ','VEH')))%>% 
  dplyr::filter(id!="TRZ") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>% 
  group_by(id,time) %>% 
  summarise(pvalue= chisq.test(reQTL, sigcount)$p.value) 

chi_squarek4 %>% 
  kable(., caption= "meSNPs (mininmally expressed QTLS) chi-squared pvalues") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "right",bootstrap_options = c("striped"),font_size = 18) %>%   scroll_box(width = "50%", height = "400px")

chi_squarek5 %>% 
  kable(., caption= "reQTLS (response eQTLS) chi-squared pvalues") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "right",bootstrap_options = c("striped"),font_size = 18) %>%   scroll_box(width = "50%", height = "400px")

```

We can "see" the proportions of SNPs are not enriched significantly in the DE genes compared to the non-DE genes (chi square test).  
We do not see significant enrichment of reQTLS in our DE gene sets over the non-DE gene sets.

So what about enrichment of meSNPs in DE compared to reQTLs in significantly DE gene sets?


```{r Knowles comparison between sets, message=FALSE, warning=FALSE}

Knowles_count <- 
  toplistall %>%
  filter(id!='TRZ') %>% 
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  group_by(time, id) %>%
  mutate(K4 = if_else(ENTREZID %in% knowles4$entrezgene_id,1,0))%>%
  mutate(K5 = if_else(ENTREZID %in% knowles5$entrezgene_id,1,0))%>%
  filter(adj.P.Val<0.05) %>%
  dplyr::summarize(n=n(), K4=sum(K4), K5=sum(K5)) %>% 
  as.tibble() %>% 
  dplyr::select(time,id,K4,K5) %>% rename("K4_y"='K4',"K5_y"='K5') %>% 
  mutate(time = case_match(time, '3_hours'~'3 hrs',
                           '24_hours'~'24 hrs',.default = time)) %>% 
  mutate(K4_n= 417-K4_y, K5_n=273-K5_y) %>% 
  pivot_longer(!c(time,id), names_to='QTL',values_to="gene_count") %>%
  separate(QTL,into=c("QTL_type",'group'),sep = '_') %>%  
  mutate(QTL_type =case_match(QTL_type, 'K4'~'meQTLs',
                         'K5'~'reQTLs',.default = QTL_type)) %>% 
  mutate(time=factor(time, levels=c("3 hrs","24 hrs"))) %>% 
  group_by(id,time,QTL_type) %>% 
  mutate(percent=gene_count/sum(gene_count)*100)
  
  ggplot(Knowles_count, aes(x=QTL_type,y=gene_count, group=group,  fill=group))+
    geom_col(position='fill')+
    # guides(fill="none")+
    facet_wrap(time~id,nrow=2,ncol=4)+
    theme_classic()+
   scale_color_manual(values=drug_palNoVeh)+
    scale_fill_manual(values=drug_palNoVeh)+
       scale_y_continuous(label = scales::percent)+
    theme(strip.text=element_text(size=10, face = "bold"),
          strip.text.x = element_text(margin = margin(2,0,2,0, "pt")),
  strip.background = element_rect (linetype=1, linewidth = 0.5))

  ###This works
  chi_frame <- Knowles_count %>%
  pivot_wider(id_cols = c(time, id,QTL_type), names_from = group, values_from = gene_count) 

testDNR3chi <- matrix(c(4,12,413, 261), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c("k4", "k5"),c( "y", "n"))) 
    DNR_3chi <- chisq.test(testDNR3chi,correct=FALSE)$p.value
testDNR24chi <- matrix(c(161,148,256, 125), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c("k4", "k5"),c( "y", "n")))  
    DNR_24chi <- chisq.test(testDNR24chi,correct=FALSE)$p.value

testDOX3chi <- matrix(c(0,0,417, 273), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c("k4", "k5"),c( "y", "n")))  
    DOX_3chi <- chisq.test(testDOX3chi,correct=FALSE)$p.value
testDOX24chi <- matrix(c(142,142,256, 125), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c("k4", "k5"),c( "y", "n"))) 
    DOX_24chi <- chisq.test(testDOX24chi,correct=FALSE)$p.value

testEPI3chi <- matrix(c(3,3,414, 270), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c("k4", "k5"),c( "y", "n"))) 
    EPI_3chi <- chisq.test(testEPI3chi,correct=FALSE)$p.value
testEPI24chi <- matrix(c(128,131,289, 142), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c("k4", "k5"),c( "y", "n"))) 
    EPI_24chi <- chisq.test(testEPI24chi,correct=FALSE)$p.value
  
testMTX3chi <- matrix(c(2,1,415, 272), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c("k4", "k5"),c( "y", "n"))) 
    MTX_3chi <- chisq.test(testMTX3chi,correct=FALSE)$p.value
testMTX24chi <- matrix(c(16,27,401, 246), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c("k4", "k5"),c( "y", "n"))) 
    MTX_24chi <- chisq.test(testMTX24chi,correct=FALSE)$p.value
K4nK5chi <- data.frame(treatment=c('DNR_3','DNR_24','DOX_3','DOX_24','EPI_3','EPI_24','MTX_3','MTX_24'), chi_p.value=c(DNR_3chi,DNR_24chi,DOX_3chi,DOX_24chi,EPI_3chi,EPI_24chi,MTX_3chi,MTX_24chi))

K4nK5chi %>% 
  separate(treatment, into= c('Drug','time')) %>% 
  pivot_wider(id_cols = Drug, names_from = time, values_from = chi_p.value) %>% 
  kable(., caption= "Chi Square p. values from chi-square test between proportions of sig-DE meQTLs and reQTLS by time and treatment") %>% 
  kable_paper("striped", full_width = TRUE) %>%  
  kable_styling(full_width = FALSE, font_size = 16) %>% 
  scroll_box( height = "500px")
 
```

Here we found that 24 hour reQTLs are significantly enriched in sig-DEgens compared to meQTLS, with DNR 3 hour treatment also showing significant enrichment.

```{r abs FC magnitude }
toplistall %>% 
  filter(time=="24_hours") %>% 
  group_by(time, id) %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  ggplot(., aes(x=id, y=abs(logFC)))+
  geom_boxplot(aes(fill=id))+
  ggpubr::fill_palette(palette =drug_palNoVeh)+
  guides(fill=guide_legend(title = "Treatment"))+
  facet_wrap(sigcount~time)+
  theme_bw()+
  xlab("")+
  ylab("abs |Log Fold Change|")+
  theme_bw()+
  facet_wrap(~time)+
  #ggtitle("All QTLs from all expressed genes (n=507)")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        # axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        strip.background = element_rect(fill = "transparent"),
        axis.text = element_text(size = 8, color = "black", angle = 0),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))

  
```

## Dox-reQTLs

When I look ot the reQTLs across treatments at 24 hours, I see that dox has a total of 180 reQTLS.  Because this gene set was specifically about DOX response eQTLs, I wanted to see if these reQTLs were only Dox specific, AC specific,  Top2i specific, or show any type of pattern.

```{r Doxreqtl}

siglist <- readRDS("data/siglist_final.RDS")
list2env(siglist,.GlobalEnv)
reQTLcombine=list(DNRreQTLs$ENTREZID,DOXreQTLs$ENTREZID,EPIreQTLs$ENTREZID,MTXreQTLs$ENTREZID)

length(unique(c(DNRreQTLs$ENTREZID,DOXreQTLs$ENTREZID,EPIreQTLs$ENTREZID,MTXreQTLs$ENTREZID)))

print("number of unique genes expressed in pairwise DE set")
QTLoverlaps <- VennDiagram::get.venn.partitions(reQTLcombine)
DoxonlyQTL <-  QTLoverlaps$..values..[[14]]
     ggVennDiagram::ggVennDiagram(reQTLcombine, category.names = c("DNR-148\nsigDEG","DOX-142 \nsigDEG\n","EPI-131\nsigDEG\n","MTX-27\nsigDEG"),label = "count") +scale_x_continuous(expand = expansion(mult = .2))+scale_y_continuous(expand = expansion(mult = .1))
```
Here, I found 121 reQTLs that were related to all anthracyclines, with 31 reQTLS related to all Top2i drugs at 24 hours.


```{r doxreqtl examine, message=FALSE, warning=FALSE}
 DOXeQTL_table <- toplistall %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(DOXreQTLs=if_else(ENTREZID %in%DOXreQTLs$ENTREZID,"y","no")) %>% 
  dplyr::filter(time =="24_hours") %>% 
  dplyr::select(ENTREZID,id,DOXreQTLs,sigcount) %>%
  group_by(id,DOXreQTLs,sigcount) %>% 
  tally() %>% as.data.frame() %>% 
  pivot_wider(.,id_cols = c(id,DOXreQTLs),names_from = sigcount,values_from = n) %>% 
  dplyr::select(id,DOXreQTLs,sig) 
DOXeQTL_table%>% 
  kable(., caption= "All 24 hour sigDEGs broken down by drug and DOXreQTL status") %>% 
  kable_paper("striped", full_width = FALSE) %>% 
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "80%", height = "400px")

DOXeQTL_table %>%
  add_row(id=c("Dox-specific DEGs","Dox-specific DEGs"),DOXreQTLs=c("no","y"), sig = c(62,1)) %>% 
  dplyr::filter(DOXreQTLs=="y") %>% 
  mutate(opp= 142-sig) %>% 
  filter(id !=c('TRZ','DOX')) %>% 
  rename("yes"=sig, "no"=opp) %>% 
  mutate(y_percent= paste0(sprintf("%2.1f", yes/(yes+no)*100), "%"),n_percent = paste0(sprintf("%2.1f", no/(yes+no)*100),"%")) %>% 
  pivot_longer(!c(id,DOXreQTLs,y_percent,n_percent), names_to="group", values_to = "count") %>% 
  mutate(id=factor(id, levels=c("DNR","EPI", "MTX","Dox-specific DEGs"))) %>% 
ggplot(., aes(y=id,x=count,fill=group))+
     geom_col(position='fill')+
    theme_classic()+
   scale_color_manual(values=c("red4","navy"))+
    scale_fill_manual(values=c("#2297E6","red2"))+
  ggtitle("DOXreQTLs overlaps")+
  scale_y_discrete(limits=rev)+
  geom_text(aes(y=id,x=1, label = n_percent,hjust=.8))+

    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.0),
        axis.line = element_line(linewidth = 1.0),
        axis.text = element_text(size = 10, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))
```

#### DOX specific-Dox dE-reqtl
```{r DOX spec DEreQTL}
# DoxonlyQTL <-  as.numeric(QTLoverlaps$..values..[[14]])
# geneDoxonlyQTL <- getBM(attributes=my_attributes,filters ='entrezgene_id',
#                   values =DoxonlyQTL, mart = ensembl)
# write.csv(geneDoxonlyQTL,"data/geneDoxonlyQTL.csv")

geneDoxonlyQTL <- read.csv("data/geneDoxonlyQTL.csv",row.names = 1)
cpmcounts <- readRDS("data/cpmcount.RDS")

for (g in seq(from=1, to=length(geneDoxonlyQTL$entrezgene_id))){
  a <- geneDoxonlyQTL$hgnc_symbol[g]
  cpm_boxplot(cpmcounts,GOI=geneDoxonlyQTL[g,1],"Dark2",drug_palc,
                ylab=bquote(~italic(.(a))~log[2]~"cpm "))

}

```
### Dox specific DEG examination

```{r Dox specific LFC}

#pull list
total24 <-list(sigVDA24$ENTREZID,sigVDX24$ENTREZID,sigVEP24$ENTREZID,sigVMT24$ENTREZID)
## do venn partition and pull doxspe genes
# total24 <- list(sigVDA24$SYMBOL,sigVDX24$SYMBOL,sigVEP24$SYMBOL,sigVMT24$SYMBOL)

venn_24h <- VennDiagram::get.venn.partitions(total24)
DoxonlyDEG <- venn_24h$..values..[[14]]
EpionlyDEG <- venn_24h$..values..[[12]]
DnronlyDEG <- venn_24h$..values..[[15]]
MtxonlyDEG <- venn_24h$..values..[[8]]
  

intersect(DoxonlyDEG,DOXreQTLs$ENTREZID)
  
Dox24_lfc <- toplist24hr %>% 
  filter(ENTREZID %in% DoxonlyDEG) %>% 
  # group_by(id) %>% 
  # dplyr::filter(adj.P.Val<0.05) #%>% 
  mutate(logFC=logFC*(-1)) %>% 
  ggplot(., aes(x= id, y=logFC))+
  geom_boxplot(aes(fill=id))+
  theme_classic()+
  fill_palette(palette = drug_palNoVeh)+
  ggtitle("LogFC of Dox specific DEGs")+
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5,face = "bold"),
    axis.title = element_text(size = 15, color = "black"),
    axis.ticks = element_line(size = 1.5),
    axis.text = element_text(size = 8, color = "black", angle = 20))
    # strip.text.x = element_text(size = 12, color = "black", face = "italic"))
  
  toplist24hr %>% 
  # filter(ENTREZID %in% DOXdeg_sp$ENTREZID) %>%
  filter(adj.P.Val<0.05) %>%
  ggplot(., aes(x=adj.P.Val))+
  geom_histogram(aes(fill=id,position="dodge"))+
    # geom_density(aes(fill=id))+
    facet_wrap(~id)+
    fill_palette(palette = drug_palNoVeh)
toplist24hr %>% 
  dplyr::filter(ENTREZID %in% DoxonlyDEG) %>%
  ggplot(., aes(x=adj.P.Val))+
  geom_density(aes(fill=id, alpha= 0.8))+
  fill_palette(palette = drug_palNoVeh)

print(Dox24_lfc)


```


## Plots of some genes:


```{r lcpm of some dox spec genes}

### making list
# # 
# Doxonly_deg <- getBM(attributes=my_attributes,filters ='entrezgene_id',
#                   values =DoxonlyDEG, mart = ensembl)
# write.csv(Doxonly_deg,"output/Doxonly_deg.csv")
Doxonly_deg <- read.csv("output/Doxonly_deg.csv", row.names = 1)
set.seed(12345)
sampset <- Doxonly_deg %>% 
  distinct(entrezgene_id,.keep_all = TRUE) %>% 
  sample_n(.,12)
for (g in seq(from=1, to=length(sampset$entrezgene_id))){
  a <- sampset$hgnc_symbol[g]
  cpm_boxplot(cpmcounts,GOI=sampset[g,1],"Dark2",drug_palc,
                ylab=bquote(~italic(.(a))~log[2]~"cpm "))

}
```



## More investigation  of 24 hour specific genes

### DOX
```{r Dox investigation}
DOXdeg_sp <- toplist24hr %>% 
  dplyr::filter(ENTREZID %in% DoxonlyDEG) %>%
  dplyr::filter(adj.P.Val<0.01) %>% 
  filter(id=="DOX") %>% 
  dplyr::select(ENTREZID, SYMBOL)  

DOXdeg_sp %>% 
  kable(., caption= "68 DOX specific genes") %>% 
  kable_paper("striped", full_width = TRUE) %>%  
  kable_styling(full_width = FALSE, font_size = 16) %>% 
  scroll_box( height = "500px")

intersect(DOXreQTLs$ENTREZID, DOXdeg_sp$ENTREZID)
#still jph3
##lfc
toplist24hr %>% 
  group_by(time,id) %>% 
  dplyr::filter(ENTREZID %in% DOXdeg_sp$ENTREZID) %>%
  mutate(logFC=logFC*(-1)) %>% 
  ggplot(., aes(x= id, y=logFC))+
  geom_boxplot(aes(fill=id))+
  theme_classic()+
  fill_palette(palette = drug_palNoVeh)+
  ggtitle("LogFC of n = 68 Dox specific DEGs")+
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5,face = "bold"),
    axis.title = element_text(size = 15, color = "black"),
    axis.ticks = element_line(size = 1.5),
    axis.text = element_text(size = 8, color = "black", angle = 20))
   
##histo check

toplist24hr %>% 
  dplyr::filter(ENTREZID %in% DOXdeg_sp$ENTREZID) %>% 
  dplyr::filter(adj.P.Val <0.05) %>%
  ggplot(., aes(x=adj.P.Val))+
  geom_histogram(aes(fill=id))+
  geom_vline(xintercept=0.01,linetype=2)+
    # geom_density(aes(fill=id))+
    facet_wrap(~id)+
  ggtitle("all Dox specific DEG adj p. value <0.01")+
    fill_palette(palette = drug_palNoVeh)+
  theme_bw()

toplist24hr %>% 
  filter(ENTREZID %in% DoxonlyDEG) %>% 
  ggplot(., aes(x=adj.P.Val))+
  geom_density(aes(fill=id, alpha= 0.8))+
  fill_palette(palette = drug_palNoVeh)+
  theme_bw()




densityDOXsp <- toplist24hr %>% 
  filter(ENTREZID %in% DoxonlyDEG) %>% 
  ggplot(., aes(x=adj.P.Val))+
  geom_density(aes(fill=id, alpha= 0.8))+
  fill_palette(palette = drug_palNoVeh)+
  theme_bw()

densityDOXsp

set.seed(12345)
sampset <- DOXdeg_sp %>% 
  sample_n(.,12)

for (g in seq(from=1, to=length(sampset$ENTREZID))){
  a <- sampset$SYMBOL[g]
  cpm_boxplot(cpmcounts,GOI=sampset[g,1],"Dark2",drug_palc,
                ylab=bquote(~italic(.(a))~log[2]~"cpm "))
}

##most convincing of these is ZNF793-AS1 101927720

```
### MTX
```{r mitoonly}
MTXdeg_sp <- toplist24hr %>% 
   dplyr::filter(ENTREZID %in% MtxonlyDEG) %>%
  dplyr::filter(adj.P.Val<0.01) %>% 
  filter(id=="MTX") %>% 
  dplyr::select(ENTREZID, SYMBOL)  

MTXdeg_sp %>% 
  kable(., caption= "MTX specific genes") %>% 
  kable_paper("striped", full_width = TRUE) %>%  
  kable_styling(full_width = FALSE, font_size = 16) %>% 
  scroll_box( height = "500px")
#intersect(DOXreQTLs$ENTREZID, MTXdeg_sp$ENTREZID)#none

##lfc

toplist24hr %>% 
  group_by(time,id) %>% 
  filter(ENTREZID %in% MtxonlyDEG) %>% 
  mutate(logFC=logFC*(-1)) %>% 
  mutate("treatment"=id)%>% 

  ggplot(., aes(x= treatment, y=logFC))+
  geom_boxplot(aes(fill=id))+
  xlab(" ")+
  theme_classic()+
  fill_palette(palette = drug_palNoVeh)+
  ggtitle("LogFC of MTX specific DEGs (n = 48)")+
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5,face = "bold"),
    axis.title = element_text(size = 15, color = "black"),
    axis.ticks = element_line(size = 1.5),
    axis.text = element_text(size = 8, color = "black", angle = 0))
 
##histo check

toplist24hr %>% 
  dplyr::filter(ENTREZID %in% MTXdeg_sp$ENTREZID) %>% 
  dplyr::filter(adj.P.Val <0.05) %>%
  ggplot(., aes(x=adj.P.Val))+
  geom_histogram(aes(fill=id))+
  geom_vline(xintercept=0.01,linetype=2)+
    # geom_density(aes(fill=id))+
    facet_wrap(~id)+
  ggtitle("all MTX specific DEG adj p. value <0.01")+
    fill_palette(palette = drug_palNoVeh)+
  theme_bw()

toplist24hr %>% 
  filter(ENTREZID %in% MtxonlyDEG) %>% 
  ggplot(., aes(x=adj.P.Val))+
  geom_density(aes(fill=id, alpha= 0.8))+
  fill_palette(palette = drug_palNoVeh)+
  theme_bw()




densityMTXsp <- toplist24hr %>% 
  filter(ENTREZID %in% MtxonlyDEG) %>% 
  ggplot(., aes(x=adj.P.Val))+
  geom_density(aes(fill=id, alpha= 0.8))+
  fill_palette(palette = drug_palNoVeh)+
  theme_bw()

densityMTXsp

set.seed(12345)
sampset <- MTXdeg_sp %>% 
  sample_n(.,12)

for (g in seq(from=1, to=length(sampset$ENTREZID))){
  a <- sampset$SYMBOL[g]
  cpm_boxplot(cpmcounts,GOI=sampset[g,1],"Dark2",drug_palc,
                ylab=bquote(~italic(.(a))~log[2]~"cpm "))
}

##most convincing of these is 54853 WDR55 


```


####  DNR

```{r dnr only}
DNRdeg_sp <- toplist24hr %>% 
  filter(adj.P.Val<0.01) %>% 
  filter(ENTREZID %in% DnronlyDEG) %>% 
  filter(id=="DNR") %>% 
  dplyr::select(ENTREZID, SYMBOL)  

##lfc

toplist24hr %>% 
  filter(ENTREZID %in% DNRdeg_sp$ENTREZID) %>% 
  group_by(time,id) %>% 
  mutate(logFC=logFC*(-1)) %>% 
  mutate("treatment" = id) %>% 
  ggplot(., aes(x= treatment, y=logFC))+
  geom_boxplot(aes(fill=id))+
  xlab(" ")+
  theme_classic()+
  fill_palette(palette = drug_palNoVeh)+
   ggtitle("LogFC of DNR specific DEGs (n = 112)")+
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5,face = "bold"),
    axis.title = element_text(size = 15, color = "black"),
    axis.ticks = element_line(size = 1.5),
    axis.text = element_text(size = 8, color = "black", angle = 0))
 
  
##histo check

toplist24hr %>% 
  dplyr::filter(ENTREZID %in% DNRdeg_sp$ENTREZID) %>% 
  dplyr::filter(adj.P.Val <0.05) %>%
  ggplot(., aes(x=adj.P.Val))+
  geom_histogram(aes(fill=id))+
  geom_vline(xintercept=0.01,linetype=2)+
    # geom_density(aes(fill=id))+
    facet_wrap(~id)+
  ggtitle("all DNR specific DEG adj p. value <0.01")+
    fill_palette(palette = drug_palNoVeh)+
  theme_bw()

toplist24hr %>% 
  filter(ENTREZID %in% DnronlyDEG) %>% 
  ggplot(., aes(x=adj.P.Val))+
  geom_density(aes(fill=id, alpha= 0.8))+
  fill_palette(palette = drug_palNoVeh)+
  theme_bw()


set.seed(12345)
sampset <- DNRdeg_sp %>% 
  sample_n(.,12)

for (g in seq(from=1, to=length(sampset$ENTREZID))){
  a <- sampset$SYMBOL[g]
  cpm_boxplot(cpmcounts,GOI=sampset[g,1],"Dark2",drug_palc,
                ylab=bquote(~italic(.(a))~log[2]~"cpm "))
}

##most convincing of these is 114826 SMYD4

```
#### EPI

```{r epi only}

EPIdeg_sp <- toplist24hr %>% 
  filter(adj.P.Val<0.01) %>% 
  filter(ENTREZID %in% EpionlyDEG) %>% 
  filter(id=="EPI") %>% 
  dplyr::select(ENTREZID, SYMBOL)  

EPIdeg_sp %>% 
  kable(., caption= "EPI specific genes") %>% 
  kable_paper("striped", full_width = TRUE) %>%  
  kable_styling(full_width = FALSE, font_size = 16) %>% 
  scroll_box( height = "500px")


##lfc

toplist24hr %>% 
  filter(ENTREZID %in% EPIdeg_sp$ENTREZID) %>% 
  group_by(time,id) %>% 
  mutate(logFC=logFC*(-1)) %>% 
  mutate("treatment" = id) %>% 

  ggplot(., aes(x= treatment, y=logFC))+
  geom_boxplot(aes(fill=id))+
  xlab(" ")+
  theme_classic()+
  fill_palette(palette = drug_palNoVeh)+
 ggtitle("LogFC of EPI specific DEGs (n = 84) ")+
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5,face = "bold"),
    axis.title = element_text(size = 15, color = "black"),
    axis.ticks = element_line(size = 1.5),
    axis.text = element_text(size = 8, color = "black", angle = 0))
 
  
 
##histo check

toplist24hr %>% 
  dplyr::filter(ENTREZID %in% EPIdeg_sp$ENTREZID) %>% 
  dplyr::filter(adj.P.Val <0.05) %>%
  ggplot(., aes(x=adj.P.Val))+
  geom_histogram(aes(fill=id))+
  geom_vline(xintercept=0.01,linetype=2)+
    # geom_density(aes(fill=id))+
    facet_wrap(~id)+
  ggtitle("all DNR specific DEG adj p. value <0.01")+
    fill_palette(palette = drug_palNoVeh)+
  theme_bw()

densityEPIsp <- toplist24hr %>% 
  filter(ENTREZID %in% EpionlyDEG) %>% 
  ggplot(., aes(x=adj.P.Val))+
  geom_density(aes(fill=id, alpha= 0.8))+
  fill_palette(palette = drug_palNoVeh)+
  theme_bw()


densityEPIsp

set.seed(12345)
sampset <- EPIdeg_sp %>% 
  sample_n(.,12)

for (g in seq(from=1, to=length(sampset$ENTREZID))){
  a <- sampset$SYMBOL[g]
  cpm_boxplot(cpmcounts,GOI=sampset[g,1],"Dark2",drug_palc,
                ylab=bquote(~italic(.(a))~log[2]~"cpm "))
}

##most convincing of these is	220963 SLC16A9


```


#### GO analysis
```{r a little GO analysis, eval=TRUE}
library(gprofiler2)

# 
# DNRdeg_sp had NO enrichment
# EPIdeg_sp had NO enrichment
# MTXdeg_sep had 2, "DNA-templated DNA replication" "nuclear pore localization"     "DNA replication" 

# gostresMTXdeg_sp <- gost(query = c(MTXdeg_sp),
#                       organism = "hsapiens",
#                       ordered_query = FALSE,
#                       domain_scope = "custom",
#                       measure_underrepresentation = FALSE,
#                       evcodes = FALSE,
#                       user_threshold = 0.05,
#                       correction_method = c("fdr"),
#                       custom_bg = backGL$ENTREZID,
#                       sources=c("GO:BP", "KEGG"))
# saveRDS(gostresMTXdeg_sp,"data/DEG-GO/gostresMTXdeg_sp.RDS")
# saveRDS(gostresDOXdeg_sp,"data/DEG-GO/gostresDOXdeg_sp.RDS")
DX_sp_DEGgostres <- readRDS("data/DEG-GO/gostresDOXdeg_sp.RDS")
MT_sp_DEGgostres <- readRDS("data/DEG-GO/gostresMTXdeg_sp.RDS")
DX_spgenes <- gostplot(DX_sp_DEGgostres, capped = FALSE, interactive = TRUE)
DX_spgenes
MT_spgenes <- gostplot(MT_sp_DEGgostres, capped = FALSE, interactive = TRUE)
MT_spgenes

DX_sp_DEGtable <- DX_sp_DEGgostres$result %>%
  dplyr::select(c(source, term_id,term_name,intersection_size, 
                   term_size, p_value))
MT_sp_DEGtable <- MT_sp_DEGgostres$result %>%
  dplyr::select(c(source, term_id,term_name,intersection_size, 
                   term_size, p_value))
  
  
DX_sp_DEGtable  %>% 
    dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = wrap_format(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('DOX specific gene set GO:BP terms') +
    xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
    theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = "black", angle = 0),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))

DX_sp_DEGtable %>% 
  mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(., caption= "Significant (adj. P value of <0.01) DOX specific genes (n = 68) and top 10 enriched GO terms") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")


MT_sp_DEGtable %>% 
 mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(., caption= "Significant (adj. P value of <0.01) DOX specific genes (n = 41) and top 10 enriched GO terms") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")

 
```


##### Grid for supplement
```{r Agrid plot, eval=TRUE,echo=FALSE}


densityDOXsp <- toplist24hr %>% 
  filter(ENTREZID %in% DoxonlyDEG) %>% 
  ggplot(., aes(x=adj.P.Val))+
  geom_density(aes(fill=id, alpha= 0.8))+
  fill_palette(palette = drug_palNoVeh)+
  theme_bw()+
  ylab("DOX-only DEGs\n n = 379")+
  guides(fill=FALSE,alpha=FALSE)+
  theme(
      axis.title = element_text(size = 10, color = "black"), 
      axis.ticks = element_line(size = 1.5),    
      axis.text = element_text(size = 10, color = "black", angle = 0))
    

densityDNRsp <- toplist24hr %>% 
  filter(ENTREZID %in% DnronlyDEG) %>% 
  ggplot(., aes(x=adj.P.Val))+
  geom_density(aes(fill=id, alpha= 0.8))+
  fill_palette(palette = drug_palNoVeh)+
  theme_bw()+
  ylab("DNR-only DEGs\n n = 615")+
  guides(fill=FALSE,alpha=FALSE)+
  theme(
      axis.title = element_text(size = 10, color = "black"), 
      axis.ticks = element_line(size = 1.5),    
      axis.text = element_text(size = 10, color = "black", angle = 0))
    

densityEPIsp <- toplist24hr %>% 
  filter(ENTREZID %in% EpionlyDEG) %>% 
  ggplot(., aes(x=adj.P.Val))+
  geom_density(aes(fill=id, alpha= 0.8))+
  fill_palette(palette = drug_palNoVeh)+
  theme_bw()+
  ylab("EPI-only DEGs\n n = 444")+
  guides(fill=FALSE,alpha=FALSE)+
  theme(axis.title = element_text(size = 10, color = "black"), 
      axis.ticks = element_line(size = 1.5),    
      axis.text = element_text(size = 10, color = "black", angle = 0))
    


densityMTXsp <- toplist24hr %>% 
  filter(ENTREZID %in% MtxonlyDEG) %>% 
  ggplot(., aes(x=adj.P.Val))+
  geom_density(aes(fill=id, alpha= 0.8))+
  fill_palette(palette = drug_palNoVeh)+
  theme_bw()+
  ylab("MTX-only DEGs\n n = 122")+
  guides(fill=FALSE,alpha=FALSE)+
  theme(axis.title = element_text(size = 10, color = "black"), 
      axis.ticks = element_line(size = 1.5),    
      axis.text = element_text(size = 10, color = "black", angle = 0))
    

```



```{r Bgrid plot, eval=TRUE,echo=FALSE, fig.width=14, fig.height=8.5}

 DoxBplot <- toplist24hr %>% 
  group_by(time,id) %>% 
  dplyr::filter(ENTREZID %in% DOXdeg_sp$ENTREZID) %>%
  mutate(logFC=logFC*(-1)) %>% 
  mutate("treatment" = id) %>%
  ggplot(., aes(x= treatment, y=logFC))+
  geom_boxplot(aes(fill=id))+
  xlab(" ")+
  ylab("DOX specific adj.P.V <0.01\n n = 68")+
  theme_classic()+
  fill_palette(palette = drug_palNoVeh)+
  guides(fill= FALSE)+
  theme(
      axis.title = element_text(size = 10, color = "black"), 
      axis.ticks = element_line(size = 1.5),    
      axis.text = element_text(size = 10, color = "black", angle = 0))

    



DnrBplot <- toplist24hr %>% 
  group_by(time,id) %>% 
  dplyr::filter(ENTREZID %in% DNRdeg_sp$ENTREZID) %>% 
  mutate(logFC=logFC*(-1)) %>% 
  mutate("treatment" = id) %>%
  ggplot(., aes(x= treatment, y=logFC))+
  geom_boxplot(aes(fill=id))+
  xlab(" ")+
  ylab("DNR-spec. adj.P.V <0.01\n n = 112")+
  theme_classic()+
  guides(fill= FALSE)+
  fill_palette(palette = drug_palNoVeh)+
 theme(
      axis.title = element_text(size = 10, color = "black"), 
      axis.ticks = element_line(size = 1.5),    
      axis.text = element_text(size = 10, color = "black", angle = 0))
    
EpiBplot <- toplist24hr %>% 
  group_by(time,id) %>% 
  filter(ENTREZID %in% EPIdeg_sp$ENTREZID) %>% 
  mutate(logFC=logFC*(-1)) %>% 
  mutate("treatment" = id) %>%
  ggplot(., aes(x= treatment, y=logFC))+
  geom_boxplot(aes(fill=id))+
  xlab(" ")+
  ylab("EPI-spec.adj.P.V <0.01\n n = 84")+
  theme_classic()+
  guides(fill= FALSE)+
  fill_palette(palette = drug_palNoVeh)+
   theme(axis.title = element_text(size = 10, color = "black"), 
      axis.ticks = element_line(size = 1.5),    
      axis.text = element_text(size = 10, color = "black", angle = 0))
    

MtxBplot <- toplist24hr %>% 
  group_by(time,id) %>% 
  filter(ENTREZID %in% MtxonlyDEG) %>% 
  mutate(logFC=logFC*(-1)) %>% 
  mutate("treatment" = id) %>%
  ggplot(., aes(x= treatment, y=logFC))+
  geom_boxplot(aes(fill=id))+
  xlab(" ")+
  ylab("MTX-spec. adj.P.V <0.01\n n = 41")+
  guides(fill= FALSE)+
  theme_classic()+
  fill_palette(palette = drug_palNoVeh)+
  theme(axis.title = element_text(size = 10, color = "black"), 
      axis.ticks = element_line(size = 1.5),    
      axis.text = element_text(size = 10, color = "black", angle = 0))

Doxgenesp_example <- cpm_boxplot(cpmcounts,GOI=101927720,"Dark2",drug_palc,
                ylab=bquote(~italic("ZNF793-AS1 ")~log[2]~"cpm "))+ 
  theme(axis.title = element_text(size = 10, color = "black"), axis.text = element_text(size = 10, color = "black", angle = 0))

Dnrgenesp_example <- cpm_boxplot(cpmcounts,114826 ,"Dark2",drug_palc,
                ylab=bquote(~italic("SMYD4")~log[2]~"cpm "))+ 
   theme(axis.title = element_text(size = 10, color = "black"), axis.text = element_text(size = 10, color = "black", angle = 0))


Epigenesp_example <- cpm_boxplot(cpmcounts,GOI=220963,"Dark2",drug_palc,
                ylab=bquote(~italic("SLC16A9")~log[2]~"cpm "))+ 
   theme(axis.title = element_text(size = 10, color = "black"), axis.text = element_text(size = 10, color = "black", angle = 0))


Mtxgenesp_example <- cpm_boxplot(cpmcounts,GOI=54853,"Dark2",drug_palc,
                ylab=bquote(~italic("WDR55")~log[2]~"cpm "))+ 
   theme(axis.title = element_text(size = 10, color = "black"), axis.text = element_text(size = 10, color = "black", angle = 0))
    



mitopg <- plot_grid(densityMTXsp, MtxBplot, Mtxgenesp_example, 
                    nrow = 1, 
                    rel_heights = c(.8,2,1), 
                    rel_widths=c(1.2,1,1.5),
                    scale=c(1,0.8,0.8))


Daunpg <- plot_grid(densityDNRsp, DnrBplot, Dnrgenesp_example, nrow = 1, rel_heights = c(.8,2,1), rel_widths=c(1.2,1,1.5),scale=c(1,0.8,0.8))
Doxopg <- plot_grid(densityDOXsp, DoxBplot, Doxgenesp_example, nrow = 1, rel_heights = c(.8,2,1), rel_widths=c(1.2,1,1.5),scale=c(1,0.8,0.8))
Epipg <- plot_grid(densityEPIsp, EpiBplot, Epigenesp_example, nrow = 1, rel_heights = c(.8,2,1), rel_widths=c(1.2,1,1.5),scale=c(1,0.8,0.8))
allfinal <- plot_grid(Doxopg,Epipg,Daunpg,mitopg,nrow=4)
allfinal

# saveRDS(allfinal,"output/allfinal_sup10.RDS")
```
