---
title: "Comparisons with other data sets"
author: "ERM"
date: '`r Sys.Date()`'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message=FALSE, warning=FALSE)
```


```{r loading packages, message=FALSE, warning=FALSE}

library(limma)
library(tidyverse)
library(ggsignif)
library(biomaRt)
library(RColorBrewer)
library(cowplot)
library(ggpubr)
library(scales)
library(sjmisc)
library(kableExtra)
library(broom)
library(ComplexHeatmap)
  


```
##  Data set comparison order:


ArrGWAS  
HFGWAS  
CADGWAS  
  
Seaone 2019  
  Supplemental 1 (408 genes)  
  Supplemental 4 (54 genes)  
  

## GWAS

```{r loading data files, echo=FALSE,eval=TRUE}
# toplistall <- read.csv("output/toplistall.csv", row.names = 1)

backGL <- read.csv("data/backGL.txt",     row.names =1)
 drug_palNoVeh <- c("#8B006D","#DF707E","#F1B72B", "#3386DD","#707031")
drug_palc <- c("#8B006D","#DF707E","#F1B72B", "#3386DD","#707031","#41B333")
time <- rep((rep(c("3h", "24h"), c(6,6))), 6)
time <- ordered(time, levels =c("3h", "24h"))
drug <- rep(c("Daunorubicin","Doxorubicin","Epirubicin","Mitoxantrone","Trastuzumab", "Vehicle"),12)
mat_drug <- c("Daunorubicin","Doxorubicin","Epirubicin","Mitoxantrone")
indv <- as.factor(rep(c(1,2,3,4,5,6), c(12,12,12,12,12,12)))
labeltop <- (interaction(substring(drug, 0, 2), indv, time))
level_order2 <- c('75','87','77','79','78','71')


toplistall <- readRDS("data/toplistall.RDS")

toplist24hr <-   toplistall %>%
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  filter(time=="24_hours")

toplist3hr <-   toplistall %>%
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  filter(time=="3_hours")

```
 


### ArrGWAS to 24 hour DEG genes p < 0.05
#### 24 hour data set
```{r 24 hour ArrGWAS}
# How I did the string split
# Arr_GWAS <- ArrGWAS[,13]
# names(Arr_GWAS) <- "genesplit"
# Arr_GWAS <- Arr_GWAS %>% 
#   separate_longer_delim(genesplit, delim = ",")
#write.csv(Arr_GWAS,"data/Arr_GWAS.txt")
arr_GWAS <- read.csv("data/Arr_GWAS.txt", row.names = 1)
Arr_geneset <- readRDS("data/Arr_geneset.RDS")
# Arr_geneset <- getBM(attributes=my_attributes,filters ='hgnc_symbol',
#                   values = arr_GWAS, mart = ensembl)
# #remove duplicates
# Arr_geneset <- Arr_geneset %>% distinct(entrezgene_id, .keep_all =TRUE)
# saveRDS(Arr_geneset,"data/Arr_geneset.RDS")
#Apply sorting
toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(ARR=if_else(ENTREZID %in%Arr_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,ARR) %>% 
  summarize(ARRcount=n())%>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(ARR), values_from=ARRcount) %>% 
   mutate(ARRprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=ARRprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",ARRprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("24 hour non-significant and significant enrichment proporitions of Arrhythmia GWAS ")

##make table of numbers:


dataframARR <- toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(ARR=if_else(ENTREZID %in%Arr_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,ARR) %>% 
  summarize(ARRcount=n()) %>% 
  as.data.frame()

dataframARR %>% 
  kable(., caption= "Significant (adj. P value of <0.05) and non-sig gene counts in Arrhythmia 24 hour GWAS") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")



```
#### 3 hour data set

```{r three hour ArrGWAS,echo=FALSE}

#Apply sorting
toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(ARR=if_else(ENTREZID %in%Arr_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,ARR) %>% 
  summarize(ARRcount=n())%>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(ARR), values_from=ARRcount) %>% 
   mutate(ARRprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=ARRprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",ARRprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("3 hour non-significant and significant enrichment proporitions of Arrhythmia GWAS ")

##make table of numbers:


dataframARR3 <- toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(ARR=if_else(ENTREZID %in%Arr_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,ARR) %>% 
  summarize(ARRcount=n()) %>% 
  as.data.frame()

dataframARR3 %>% kable(., caption= "Significant (adj. P value of <0.05) and non-sig gene counts in Arrhythmia 3 hour GWAS") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")


```

#### chi square test ARR

```{r chi ARR, echo=TRUE, message=FALSE, warning=FALSE}


chi_funarr <-  toplistall %>% 
  mutate(id = as.factor(id)) %>%
  dplyr::filter(id!="Trastuzumab") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(ARR=if_else(ENTREZID %in%Arr_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,time) %>% 
  summarise(pvalue= chisq.test(ARR, sigcount)$p.value) 


chi_funarr %>% 
  kable(., caption= "after performing chi square test between DEgenes, and non DE genes") %>% 
  kable_paper("striped") %>%  
  kable_styling(full_width = FALSE,font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")


```
```{r ARR heatmap,echo=FALSE}

ARRmat <- chi_funarr %>% ungroup() %>% 
  mutate(ARR= log(pvalue)*(-1)) %>% 
  filter(time=="24_hours") %>% 
  mutate(id =case_match( id, 'Daunorubicin'~'DNR',
                         'Doxorubicin'~'DOX',
                         'Epirubicin'~'EPI',
                         'Mitoxantrone'~'MTX', 
                         .default = id)) %>% 
  mutate(time=case_match(time,"24_hours"~"24_hrs",.default = time)) %>% 
  dplyr::select(id,time,ARR) %>% 
  unite("term_name",id,time,sep="_") %>% 
  column_to_rownames('term_name') 
  


col_fun1 = circlize::colorRamp2(c(-1, 3), c("white", "purple"))
Heatmap( as.matrix(ARRmat), name = "Arrhythmia chi square \n-log (p values)", cluster_rows = FALSE, cluster_columns = FALSE,
         col=col_fun1,
         column_names_rot = 0,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(ARRmat[i, j] > -log(0.05))
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})



```




### HFGWAS
#### 24 hours HF
```{r 24 hours HFGWAS}
##just like ARrGWAS- imported the total csv, the took the "nearest" column and separated out the gene info
# test <- HFGWAS %>% 
#   select(nearest) %>% 
#   separate_wider_delim(nearest, delim = "[", names_sep = "", too_few = "align_start")
# test2 <- str_sub(test$nearest2,0,nchar(test$nearest2)-1)
# Hf_GWAS <- test2
#write.csv(Hf_GWAS, "data/Hf_GWAS.txt")
# HF_GWAS <- read.csv("data/Hf_GWAS.txt", row.names =1)
# 
# HF_geneset <- getBM(attributes=my_attributes,filters ='hgnc_symbol',
#                   values = HF_GWAS, mart = ensembl)
# #remove duplicates
# HF_geneset <- HF_geneset %>% distinct(entrezgene_id, .keep_all =TRUE)
# saveRDS(HF_geneset,"data/HF_geneset.RDS")
HF_geneset <- readRDS("data/HF_geneset.RDS")
#Apply sorting
toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(HF=if_else(ENTREZID %in%HF_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,HF) %>% 
  summarize(HFcount=n())%>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(HF), values_from=HFcount) %>% 
   mutate(HFprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=HFprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",HFprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("non-significant and significant enrichment proporitions of Heart Failure GWAS ")

##make table of numbers:


dataframHF <- toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(HF=if_else(ENTREZID %in%HF_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,HF) %>% 
  summarize(HFcount=n()) %>% 
  as.data.frame()

dataframHF %>% #mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(., caption= "Significant (adj. P value of <0.05) and non-sig gene counts in HFhythmia GWAS") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")


```
### 3 hours HF
```{r 3 hours HFGWAS, echo=FALSE}

#Apply sorting
toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(HF=if_else(ENTREZID %in%HF_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,HF) %>% 
  summarize(HFcount=n())%>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(HF), values_from=HFcount) %>% 
   mutate(HFprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=HFprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",HFprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("non-significant and significant enrichment proportions of Heart Failure GWAS ")

##make table of numbers:


dataframHF3 <- toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(HF=if_else(ENTREZID %in%HF_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,HF) %>% 
  summarize(HFcount=n()) %>% 
  as.data.frame()

dataframHF3 %>%
  kable(., caption= "Significant (adj. P value of <0.05) and non-sig gene counts in Three hour HFhythmia GWAS") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")

```
#### chi square test HF
```{r chiHF,echo=TRUE}


chi_funhf <-  toplistall %>% 
  mutate(id = as.factor(id)) %>%
  dplyr::filter(id!="Trastuzumab") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(HF=if_else(ENTREZID %in%HF_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,time) %>% 
  summarise(pvalue= chisq.test(HF, sigcount)$p.value) 

chi_funhf %>% 
  kable(., caption= "after performing chi square test between DEgenes, and non DE genes") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE,font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")

```

```{r HF heatmap,echo=FALSE}

HFmat <- chi_funhf %>% ungroup() %>% 
  mutate(HF= log(pvalue)*(-1)) %>% 
  filter(time=="24_hours") %>% 
  mutate(id =case_match( id, 'Daunorubicin'~'DNR',
                         'Doxorubicin'~'DOX',
                         'Epirubicin'~'EPI',
                         'Mitoxantrone'~'MTX', 
                         .default = id)) %>% 
  mutate(time=case_match(time,"24_hours"~"24_hrs",.default = time)) %>% 
  dplyr::select(id,time,HF) %>% 
  unite("term_name",id,time,sep="_") %>% 
  column_to_rownames('term_name') 
  
col_fun5 = circlize::colorRamp2(c(0, 5), c("white", "purple"))

Heatmap( as.matrix(HFmat), name = "Heart Failure GWAS\n chi square -log p values", 
         cluster_rows = FALSE, cluster_columns = FALSE,
         col=col_fun1,
         column_names_rot = 0,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(HFmat[i, j] > -log(0.05))
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})



```
## CAD GWAS

### 24 hour data set
```{r CAD 24 GWAS}
# test <- CADGWAS %>% 
#    select(nearest) %>% 
#    separate_wider_delim(nearest, delim = "[", names_sep = "", too_few = "align_start")
#  test2 <- str_sub(test$nearest2,0,nchar(test$nearest2)-1)
# 
# test2[c(32,38,44,74,112,126,191,212)] <- c("TPCN1","C2orf43","FAM222A", "TDRD15"  ,"AGPAT4","SVOP","SVOP","PLG")
#    
#  test2 [c(218,226,228,233,239,245,256,270,281,322,324,332,335,338,347,351,352,358)] <-  c("HPCAL1", "KLHL29"  , "COL4A3BP"  , "ARAP1" ,
#  "VEGFA", "TBPL1","SLC22A3" ,"C19orf38","LPA","VPS29","ATP2A2" ,"ATP2A2","KLHL29","GUCY1A3","KCNE2",  "HOXB9","P2RY2" ,"CTC-236F12.4")
#  
#  CAD_GWAS <- test2
#write.csv(CAD_GWAS, "data/cvd_GWAS.txt")
CAD_GWAS <- read.csv("data/cvd_GWAS.txt", row.names =1)

# CAD_geneset <- getBM(attributes=my_attributes,filters ='hgnc_symbol',
#                   values = CAD_GWAS, mart = ensembl)
# #remove duplicates
# CAD_geneset <- CAD_geneset %>% distinct(entrezgene_id, .keep_all =TRUE)
# 
# saveRDS(CAD_geneset,"data/CAD_geneset.RDS")
CAD_geneset <- readRDS("data/CAD_geneset.RDS")
#Apply sorting



toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(CAD=if_else(ENTREZID %in%CAD_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,CAD) %>% 
  summarize(CADcount=n())%>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(CAD), values_from=CADcount) %>% 
   mutate(CADprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=CADprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",CADprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("non-significant and significant enrichment proporitions of CAD GWAS ")

##make table of numbers:


dataframCAD <- toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(CAD=if_else(ENTREZID %in%CAD_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,CAD) %>% 
  summarize(CADcount=n()) %>% 
  as.data.frame()

dataframCAD %>% #mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(., caption= "Significant (adj. P value of <0.05) and non-sig gene counts in CAD GWAS") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")

```
### 3 hour data set
```{r CAD 3 GWAS,echo=FALSE}

#Apply sorting
toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(CAD=if_else(ENTREZID %in%CAD_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,CAD) %>% 
  summarize(CADcount=n())%>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(CAD), values_from=CADcount) %>% 
   mutate(CADprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=CADprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",CADprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("3 hour non-significant and significant enrichment proporitions of CAD GWAS ")

##make table of numbers:


dataframCAD3 <- toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(CAD=if_else(ENTREZID %in%CAD_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,CAD) %>% 
  summarize(CADcount=n()) %>% 
  as.data.frame()

dataframCAD3 %>% #mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(., caption= "Significant (adj. P value of <0.05) and non-sig gene counts in 3 hour CAD GWAS") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")

```

#### chi square test CAD
```{r chiCAD,echo=TRUE}

chi_funCAD <-  toplistall %>% 
  mutate(id = as.factor(id)) %>%
  dplyr::filter(id!="Trastuzumab") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(CAD=if_else(ENTREZID %in%CAD_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,time) %>% 
  summarise(pvalue= chisq.test(CAD, sigcount)$p.value) 

chi_funCAD %>% 
  kable(., caption= "after performing chi square test between DEgenes, and non DE genes") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE,font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")

```



```{r CAD heatmap,echo=FALSE}

CADmat <- chi_funCAD %>% ungroup() %>% 
  mutate(CAD= log(pvalue)*(-1)) %>% 
  filter(time=="24_hours") %>% 
  mutate(id =case_match( id, 'Daunorubicin'~'DNR',
                         'Doxorubicin'~'DOX',
                         'Epirubicin'~'EPI',
                         'Mitoxantrone'~'MTX', 
                         .default = id)) %>% 
  mutate(time=case_match(time,"24_hours"~"24_hrs",.default = time)) %>% 
  dplyr::select(id,time,CAD) %>% 
  unite("term_name",id,time,sep="_") %>% 
  column_to_rownames('term_name')  

col_fun4 = circlize::colorRamp2(c(-1, 4), c("white", "purple"))

Heatmap( as.matrix(CADmat), name = "Heart Failure GWAS\n chi square -log p values", cluster_rows = FALSE, cluster_columns = FALSE,col =col_fun1,
         column_names_rot = 0,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(CADmat[i, j] > -log(0.05))
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})



GWASmat <- bind_cols(CADmat$CAD,HFmat$HF,ARRmat$ARR) 
colnames(GWASmat) <- c('CAD','HF','ARR')
rownames(GWASmat) <- c('DNR','DOX','EPI','MTX')# %>% tibble() %>% dplyr::select) %>% as.matrix()


# Heatmap(as.matrix(GWASmat), name = "GWAS chi square\n -log p values", cluster_rows = FALSE,column_title = 'This is for GWAS 24 hours -log(chi square pvalue)', cluster_columns = FALSE,row_names_side = 'left', column_names_rot = 0, col_fun1,
#         layer_fun = function(j, i, x, y, w, h, fill) {
#         ind_mat = restore_matrix(j, i, x, y)
#         ind = unique(c(ind_mat[3,3], ind_mat[3, 3]))
#         grid.points(x[ind], y[ind], pch = 8, size = unit(4, "mm"))
#     }
# )

Heatmap(as.matrix(GWASmat), name = "GWAS chi square\n -log p values", cluster_rows = FALSE, column_title = 'This is for GWAS 24 hours -log(chi square pvalue)', cluster_columns = FALSE,row_names_side = 'left', column_names_rot = 0, col_fun1,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(GWASmat[i, j] > -log(0.05))
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})

print('This is for  GWAS 24 hours -log(chi square pvalue)')

```
The star represents chi square p.value < 0.05.
```{r GWAS genes of interest, echo=TRUE, message=FALSE, warning=FALSE}
# GWAS_goi <- c('RARG', 'ITGB7', 'TNS2','ZNF740','SLC28A3','RMI1',
# 'FEDORA' ,'GDF5','FRS2','HDDC2','EEF1B2')
# 
# library(biomaRt)
# ensembl <- useMart("ensembl", dataset="hsapiens_gene_ensembl")
# my_chr <- c(1:22, 'M', 'X', 'Y')  
# my_attributes <- c('entrezgene_id', 'ensembl_gene_id', 'hgnc_symbol')
# 
# 
# GWAS_goi<- getBM(attributes=my_attributes,filters ='hgnc_symbol',
#          values = GWAS_goi, mart = ensembl)
# GWAS_goi<-GWAS_goi %>% distinct(entrezgene_id,.keep_all = TRUE) %>% add_row(entrezgene_id='124903732',ensembl_gene_id='ENSG00000260788', hgnc_symbol="RP11-298D21.1
# ")

  
# write.csv(GWAS_goi,"output/GWAS_goi.csv")
GWAS_goi <- read.csv("output/GWAS_goi.csv")
##get the abs FC of all GOI
GWASabsFCsig <- 
  toplistall %>% 
  # mutate(absFC=abs(logFC)) %>% 
  mutate(id = as.factor(id)) %>%
  filter(id !="Trastuzumab") %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  filter(ENTREZID %in% GWAS_goi$entrezgene_id) %>% 
   filter(time =="24_hours") %>% 
  dplyr::select(ENTREZID ,time, id,logFC, adj.P.Val, SYMBOL) %>% 
  # filter(adj.P.Val <0.05) %>% 
  mutate(id =case_match(id,'Daunorubicin'~'DNR','Doxorubicin'~'DOX','Epirubicin'~'EPI','Mitoxantrone'~'MTX', .default = id)) %>% 
  # mutate(time =case_match(time,"3_hours"~'3_hr',"24_hours"~'24_hr',.default = time)) %>% 
 # unite("trt_time",id:time,sep = '_') %>% 
  pivot_wider(id_cols=id, names_from = SYMBOL, values_from =adj.P.Val)# %>% 
  
gwas_sig_mat <- GWASabsFCsig %>% 
   column_to_rownames(var="id") %>%
  as.matrix()
 

GWASabsFC <- toplistall %>% 
  # mutate(absFC=abs(logFC)) %>% 
  mutate(id = as.factor(id)) %>%
  filter(id !="Trastuzumab") %>% 
  filter(time=="24_hours") %>% 
  # mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  filter(ENTREZID %in% GWAS_goi$entrezgene_id) %>% 
  dplyr::select(SYMBOL ,time, id, logFC) %>% 
  mutate(id =case_match(id,'Daunorubicin'~'DNR','Doxorubicin'~'DOX','Epirubicin'~'EPI','Mitoxantrone'~'MTX', .default = id)) %>% 
  # mutate(time =case_match(time,"3_hours"~'3_hr',"24_hours"~'24_hr',.default = time)) %>%
  # unite("trt_time",id:time,sep = '_') %>% 
  pivot_wider(id_cols=id, names_from = SYMBOL, values_from = logFC) %>% 
  column_to_rownames(var="id") %>%
  as.matrix()

 

Heatmap(GWASabsFC, name = "Fold change\nvalues", 
         cluster_rows = FALSE,
        cluster_columns = FALSE, 
        row_names_side = "left",
        column_title = "Fold change values of GWAS and TWAS genes", 
        column_title_side = "top",
        column_title_gp = gpar(fontsize = 16, fontface = "bold"),
        column_order= c('RARG',
                        'TNS2', 
                        'ZNF740',
                        'SLC28A3',
                        'RMI1',
                        'EEF1B2',
                        'FRS2', 
                        'HDDC2'),
        column_names_rot = 0, 
        column_names_gp = gpar(fontsize = 12),
        column_names_centered = TRUE,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(gwas_sig_mat[i, j] <0.05)
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})

 
```

The stars represent all genes that have an adj. P. value of < 0.05 (significantly differentially expressed)
    
### Seoane 2019

Seoane, Jose  Chromatin gene comparison:   comes from supp data NAT. MED 2019
#### 24 hours in Pairwise with supplemental data 1


```{r gene comparison, eval=TRUE, message=FALSE, warning=FALSE}

# Seoane_chromatinregs <- read_excel("C:/Users/renee/Downloads/Supplements folde manuscriptr/NIHMS1539805-supplement-SuppTables.xlsx", 
#     range = "A13:H469")

#write.csv(Seoane_chromatinregs, "data/Seonane2019supp1.txt")

chrom_reg_Seoane <- read_csv(file = "data/Seonane2019supp1.txt",col_types = cols(...1 = col_skip()))
                            
Seoane_2019 <- chrom_reg_Seoane[,2]
names(Seoane_2019) <- "ENTREZID"
chrom_genes <- (unique(Seoane_2019$ENTREZID))

toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(chrom=if_else(ENTREZID %in%chrom_genes,"y","no")) %>% 
  group_by(id,sigcount,chrom) %>% 
  summarize(chromcount=n()) %>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(chrom), values_from=chromcount) %>% 
   mutate(chromprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=chromprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",chromprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("non-significant and significant enrichment proportions of chromatin gene set ")

dataframchrom <- toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(chrom=if_else(ENTREZID %in%chrom_genes,"y","no")) %>% 
  group_by(id,sigcount,chrom) %>% 
  summarize(chromcount=n()) %>% 
  as.data.frame()

dataframchrom %>%
  pivot_wider(., names_from=c('sigcount','chrom'), values_from = 'chromcount') %>% 
  kable(., caption= "Significant (adj. P value of <0.05) and non-sig gene counts in Seoane geneset") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")

toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
   dplyr::filter(id!="Trastuzumab") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(chrom=if_else(ENTREZID %in%chrom_genes,"y","no")) %>% 
  group_by(id) %>% 
 summarise(pvalue= chisq.test(chrom, sigcount)$p.value) 



```

### 3 hours data Pairwise

```{r three hour seaone,message=FALSE, warning=FALSE, echo=FALSE}
toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(chrom=if_else(ENTREZID %in%chrom_genes,"y","no")) %>% 
  group_by(id,sigcount,chrom) %>% 
  summarize(chromcount=n()) %>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(chrom), values_from=chromcount) %>% 
   mutate(chromprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=chromprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",chromprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("non-significant and significant enrichment proporitions of chromatin gene set ")

dataframchrom3 <- toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(chrom=if_else(ENTREZID %in%chrom_genes,"y","no")) %>% 
  group_by(id,sigcount,chrom) %>% 
  summarize(chromcount=n()) %>% 
  as.data.frame()

dataframchrom3 %>%
  pivot_wider(., names_from=c('sigcount','chrom'), values_from = 'chromcount') %>% 
  kable(., caption= "Significant (adj. P value of <0.05) and non-sig gene counts in 3 hours Seoane geneset") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")
```
#### chi square test Seaone
```{r chi value seoane, echo= TRUE,message=FALSE, warning=FALSE}
##remove Trastuzumab in order to perform chi square tests by time and drug between  DE and non DE enrichment
chi_fun <-  toplistall %>% 
  mutate(id = as.factor(id)) %>%
  dplyr::filter(id!="Trastuzumab") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(chrom=if_else(ENTREZID %in%chrom_genes,"y","no")) %>% 
  group_by(id,time) %>% 
  summarise(pvalue= chisq.test(chrom, sigcount)$p.value) 

chi_fun%>% 
  kable(., caption= "after performing chi square test between DEgenes, and non DE genes") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE,font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")


```



### Supplemental 4 Seoane 3 hours followed by 24 hours 

```{r supp4 seoane, message=FALSE, warning=FALSE}
# library(readxl)
# Sup4seoane <- read_excel("~/Ward Lab/Cardiotoxicity/Manuscript/seoane201941591_2019_638_MOESM3_ESM.xlsx", 
#     sheet = "SupTable4", range = "A16:J567")

#     sheet = "SupTable4", range = "A16:J567")
#write.csv(Sup4seoane, "output/Sup4seoane.csv")
Sup4seoane <- read.csv("output/Sup4seoane.csv", row.names = 1)
Sup4genes <- Sup4seoane  %>% 
  filter(pval.expAnth<0.05) %>% 
  distinct(entrez, .keep_all = TRUE) %>% 
  dplyr::select(entrez)

Sup4seoane  %>% 
  filter(pval.expAnth<0.05) %>% 
  distinct(entrez, .keep_all = TRUE) %>% 
  dplyr::select(entrez,gene,pval.exp,pval.anthr,pval.expAnth,adjpval) %>% 
  kable(., caption = "List of Seoane Supplemental 4 genes") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")

toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(chrom=if_else(ENTREZID %in%Sup4genes$entrez,"y","no")) %>% 
  group_by(id,sigcount,chrom) %>% 
  summarize(chromcount=n()) %>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(chrom), values_from=chromcount) %>% 
   mutate(chromprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=chromprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",chromprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("Seoane supp 4 enrichment proportions found in my pairwise 3 hour data")


toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(chrom=if_else(ENTREZID %in%Sup4genes$entrez,"y","no")) %>% 
  group_by(id,sigcount,chrom) %>% 
  summarize(chromcount=n()) %>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(chrom), values_from=chromcount) %>% 
   mutate(chromprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=chromprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",chromprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("Seoane supp 4 enrichment proportions found in my pairwise 24 hour data")


 chi_fun2 <-  
  toplistall %>% 
  mutate(id = as.factor(id)) %>%
  dplyr::filter(id!="Trastuzumab") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
 mutate(chrom=if_else(ENTREZID %in%Sup4genes$entrez,"y","no")) %>% 
  group_by(id,time) %>% 
  summarise(pvalue= chisq.test(chrom, sigcount)$p.value) 
print("These are the chisquare values from the 54 genes")
chi_fun2


pairS14_mat <- chi_fun2 %>% 
  full_join(chi_fun,by=c("id","time")) %>% 
  mutate("n = 54"=-log(pvalue.x), "n = 408"=-log(pvalue.y)) %>% 
  mutate(time= case_match(time, 
                          '3_hours'~'3_hrs', 
                          '24_hours'~'24_hrs',.default = id)) %>% 
  mutate(id =case_match( id, 
                         'Daunorubicin'~'DNR',   
                         'Doxorubicin'~'DOX' ,
                         'Epirubicin'~'EPI' , 
                         'Mitoxantrone' ~ 'MTX',.default = id)) %>% 
  unite('pairset',time,id ) %>% 
  dplyr::select(!c(pvalue.x,pvalue.y)) %>% 
  column_to_rownames('pairset') %>% 
  as.matrix()
 

Heatmap( pairS14_mat,
         column_title="Pairwise version of Chromatin gene sets \nchi square -log p values", row_order = c(2,4,6,8,1,3,5,7),
         name = "-log p values", 
         cluster_rows = FALSE, 
         cluster_columns = FALSE, 
         column_names_rot = 0,
         col = col_fun5, 
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(pairS14_mat[i, j] > -log(0.05))
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})

```

Stars indicate a chi square pvalue < 0.05



# Using Baysian gene sets to look at enrichment of Seoane data


## Seoane Supplemental 4
```{r loading comotif  data sets, message=FALSE, warning=FALSE}
DEG_cormotif <- readRDS("data/DEG_cormotif.RDS")
list2env(DEG_cormotif,envir=.GlobalEnv)
backGL <- read.csv("data/backGL.txt")
```


```{r Seoane supp4 analysis}
##data sets to compare   Sup4genes$entrez,"y","no"

motifSeoanesummary4 <- toplist24hr %>% 
  distinct(ENTREZID) %>% 
  mutate(ER=if_else(ENTREZID %in% motif_ER,"y","no")) %>% 
  mutate(LR=if_else(ENTREZID %in% motif_LR,"y","no")) %>%
  mutate(TI=if_else(ENTREZID %in% motif_TI,"y","no")) %>%
  mutate(NR=if_else(ENTREZID %in% motif_NR,"y","no")) %>%
  mutate(chrom = if_else(ENTREZID %in% Sup4genes$entrez, "y", "no")) %>% 
  group_by(chrom,ER,TI,LR,NR) %>% 
  dplyr::summarize(n=n()) %>% 
  as.tibble  %>% 
  pivot_wider(id_cols = c(chrom), names_from = c('ER', 'TI', 'LR', 'NR'), values_from= n) %>% 
  rename(.,c("chrom"=chrom,"none"= 2 , "ER" = 3 , "TI" = 4 , "LR" = 5 ,"NR" = 6))
motifSeoanesummary4 %>% kable(., caption= "Summary of genes from Cormotif that are also in Seoane Supp4" )%>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")
    # chisqS4 <- 
##making the matrix

chi_list4 <- toplist24hr %>% 
  distinct(ENTREZID) %>% 
  mutate(ER=if_else(ENTREZID %in%motif_ER,"y","no")) %>% 
  mutate(LR=if_else(ENTREZID %in%motif_LR,"y","no")) %>%
  mutate(TI=if_else(ENTREZID %in%motif_TI,"y","no")) %>%
  mutate(NR=if_else(ENTREZID %in%motif_NR,"y","no")) %>%
  mutate(chrom=if_else(ENTREZID %in%Sup4genes$entrez,"y","no")) %>% 
  group_by(chrom,ER,TI,LR,NR) %>% 
  dplyr::summarize(n=n()) %>% 
  as.tibble  %>% 
  pivot_wider(id_cols = c(chrom), names_from = c('ER', 'TI', 'LR', 'NR'), values_from= n) %>% 
  rename(.,c("chrom"=chrom,"none"= 2 , "ER" = 3 , "TI" = 4 , "LR" = 5 ,"NR" = 6))

chisup4LR <- chisq.test(chi_list4[,c('NR','LR')])
chisup4LR
chisup4ER <- chisq.test(chi_list4[,c('NR','ER')])
chisup4ER
chisup4TI <- chisq.test(chi_list4[,c('NR','TI')])
chisup4TI
chisq.test(chi_list4[,c('ER','TI')])
chisq.test(chi_list4[,c('ER','LR')])
chisq.test(chi_list4[,c('TI','LR')])


```


## Seoane Supplemental 1


```{r supp 1}
 motifSeoanesummary1 <- toplist24hr %>% 
  distinct(ENTREZID) %>% 
  mutate(ER=if_else(ENTREZID %in%motif_ER,"y","no")) %>% 
  mutate(LR=if_else(ENTREZID %in%motif_LR,"y","no")) %>%
  mutate(TI=if_else(ENTREZID %in%motif_TI,"y","no")) %>%
  mutate(NR=if_else(ENTREZID %in%motif_NR,"y","no")) %>%
  mutate(chrom = if_else(ENTREZID %in% chrom_genes, "y", "no")) %>% 
  group_by(chrom,ER,TI,LR,NR) %>% 
  dplyr::summarize(n=n()) %>% 
  as.tibble  %>% 
  pivot_wider(id_cols = c(chrom), names_from = c('ER', 'TI', 'LR', 'NR'), values_from= n) %>% 
  rename(.,c("chrom"=chrom,"none"= 2 , "ER" = 3 , "TI" = 4 , "LR" = 5 ,"NR" = 6))
motifSeoanesummary1 %>% kable(., caption= "Summary of genes from Cormotif that are also in Seoane Supp1" )%>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")
    # chisqS4 <- 
##making the matrix

chi_list1 <- toplist24hr %>% 
  distinct(ENTREZID) %>% 
  mutate(ER=if_else(ENTREZID %in%motif_ER,"y","no")) %>% 
  mutate(LR=if_else(ENTREZID %in%motif_LR,"y","no")) %>%
  mutate(TI=if_else(ENTREZID %in%motif_TI,"y","no")) %>%
  mutate(NR=if_else(ENTREZID %in%motif_NR,"y","no")) %>%
  mutate(chrom=if_else(ENTREZID %in% chrom_genes,"y","no")) %>% 
  group_by(chrom,ER,TI,LR,NR) %>% 
  dplyr::summarize(n=n()) %>% 
  as.tibble  %>% 
  pivot_wider(id_cols = c(chrom), names_from = c('ER', 'TI', 'LR', 'NR'), values_from= n) %>% 
  rename(.,c("chrom"=chrom,"none"= 2 , "ER" = 3 , "TI" = 4 , "LR" = 5 ,"NR" = 6))

chisup1LR <- chisq.test(chi_list1[,c('NR','LR')])
chisup1LR
chisup1ER <- chisq.test(chi_list1[,c('NR','ER')])
chisup1ER
chisup1TI <- chisq.test(chi_list1[,c('NR','TI')])
chisup1TI
chisq.test(chi_list1[,c('ER','TI')])
chisq.test(chi_list1[,c('ER','LR')])
chisq.test(chi_list1[,c('TI','LR')])


```
### chi square heatmap
```{r heatmapsup4n1 cormotif}
supp45X <- NULL
motif <- c('ER','TI','LR')
pval1=c(chisup1ER$p.value,chisup1TI$p.value,chisup1LR$p.value)
pval4=c(chisup4ER$p.value ,chisup4TI$p.value ,chisup4LR$p.value)
supp45X <- tibble("motif"=motif,"pval1"=pval1,"pval4"=pval4)
supp45Xmat <- supp45X %>% 
  mutate(pval1=(-1*log(pval1))) %>% 
  mutate(pval4=(-1*log(pval4))) %>% 
  column_to_rownames('motif') %>% 
  rename(c("n = 408"= pval1, "n = 54"=pval4)) %>% 
  as.matrix()

Heatmap(supp45Xmat,
         column_title="Chromatin gene sets \nchi square -log p values", 
         name = "-log (p value)", 
         cluster_rows = FALSE, 
         cluster_columns = FALSE, 
         column_names_rot = 0,
         col = col_fun5,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(supp45Xmat[i, j] > -log(0.05))
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})



```
Stars represent chi square p values of < 0.05

## Crispr list  


```{r Crispr data analysis}

# Crispr_list <- read_excel("C:/Users/renee/Downloads/41598_2021_92988_MOESM2_ESM.xlsx")
#  View(Crispr_list)
# crispr_genes <- Crispr_list %>% 
#   dplyr::filter(p.value <0.05) %>% 
#   select(GeneName)
  

# crispr_genes <- getBM(attributes=my_attributes,filters ='hgnc_symbol',
                  # values =crispr_genes$GeneName, mart = ensembl)
# write.csv(crispr_genes,'data/crispr_genes.csv')

crispr_genes <- read.csv("data/crispr_genes.csv", row.names = 1)
print(" number of unique crispr_genes after conversion from hgnc symbol to entrezid")
length(unique(crispr_genes$entrezgene_id))
crisprunique <- crispr_genes %>% distinct(entrezgene_id,.keep_all = TRUE)

Doxcrispall <- toplistall %>%
  distinct(ENTREZID,.keep_all = TRUE) %>% 
  dplyr::select(ENTREZID,id,time)
  

crispmotifsummary <- Doxcrispall %>% 
  mutate(ER=if_else(ENTREZID %in% motif_ER,"y","no")) %>% 
  mutate(LR=if_else(ENTREZID %in% motif_LR,"y","no")) %>%
  mutate(TI=if_else(ENTREZID %in% motif_TI,"y","no")) %>%
  mutate(NR=if_else(ENTREZID %in% motif_NR,"y","no")) %>%
  mutate(crisp = if_else(ENTREZID %in% crisprunique$entrezgene_id, "y", "no")) %>% 
  group_by(crisp,ER,TI,LR,NR) %>% 
  dplyr::summarize(n=n()) %>% 
  as.tibble  %>% 
  pivot_wider(id_cols = c(crisp), names_from = c('ER', 'TI', 'LR', 'NR'), values_from= n) %>% 
  rename(.,c("crisp"=crisp,"none"= 2 , "ER" = 3 , "TI" = 4 , "LR" = 5 ,"NR" = 6)) 

cris_mat <- crispmotifsummary %>% dplyr::select(ER:NR) %>% as.matrix()
chicheck <- data.frame(one= c("LR","ER","TI"),two=rep("NR",3),p.value=c("","",""))
  
 chicheck$p.value[1] <- chisq.test(cris_mat[,c('LR','NR')],correct = FALSE)$p.value

chicheck$p.value[2] <- chisq.test(cris_mat[,c('ER','NR')],correct = FALSE)$p.value
chicheck$p.value[3] <- chisq.test(cris_mat[,c('TI','NR')],correct = FALSE)$p.value

chicheck%>% kable(., caption= "chi square test p.values for encrichment of  Doxcrispr gene sets in motif sets" )%>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")

chicheck_1 <- chicheck %>% mutate(p.value=as.numeric(p.value)) %>% 
  mutate(neg.logvalue=(-1*log(p.value))) %>% column_to_rownames('one') %>% dplyr::select(neg.logvalue) %>% as.matrix
col_fun = circlize::colorRamp2(c(0, 2), c("white", "purple"))

Heatmap( chicheck_1, name = "Doxcrispr enrichment \nchi square -log p values", cluster_rows = FALSE, cluster_columns = FALSE, col=col_fun,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(chicheck_1[i, j] > -log(0.05))
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})
```



```{r crisper DEG}


col_fun4 = circlize::colorRamp2(c(0, 5), c("white", "purple"))


pairwisecrispr <- toplistall %>%
  filter(id!='Trastuzumab') %>% 
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(crisp = if_else(ENTREZID %in% crisprunique$entrezgene_id, "y", "no")) %>% 
  group_by(time, id) %>%
  summarise(pvalue= chisq.test(crisp, sigcount, correct=FALSE)$p.value)
 
  
  crisprnumbers <- toplistall %>%
  filter(id!='Trastuzumab') %>% 
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(crisp = if_else(ENTREZID %in% crisprunique$entrezgene_id, "y", "no")) %>% 
  group_by(time, id,sigcount,crisp) %>%
  dplyr::summarize(n=n()) %>% 
  as.tibble() #%>% 
  
crisprnumbers %>% kable(., caption= "Summary of genes found in both sigDE and non sigDE by treatment" )%>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")

pairwisecrispr%>% kable(., caption= "Summary of chisqure values between numbers of sigDE and non sigDE by treatment" )%>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")


crisp_pair_mat <- pairwisecrispr %>%
  mutate(neg.log.pvalue= (-1*log(pvalue))) %>% 
  mutate(time= case_match(time, '3_hours'~'3_hrs', '24_hours'~'24_hrs',.default = id)) %>% 
  mutate(id =case_match( id, 'Daunorubicin'~'DNR',   'Doxorubicin'~'DOX' ,'Epirubicin'~'EPI' , 'Mitoxantrone' ~ 'MTX',.default = id)) %>% 
  unite('pairset',time,id ) %>% 
  column_to_rownames('pairset') %>% dplyr::select(neg.log.pvalue) %>% as.matrix()
    
Heatmap( crisp_pair_mat, name = "Doxcrispr pairwise enrichment \nchi square -log p values", 
         cluster_rows = FALSE, 
         cluster_columns = FALSE, 
         col=col_fun5, column_names_rot = 0,
         cell_fun = function(j, i, x, y, width, height, fill) {
        if(crisp_pair_mat[i, j] > -log(0.05))
            grid.text("*", x, y, gp = gpar(fontsize = 20))
})

```



