---
title: "Comparisons with other data sets"
author: "ERM"
date: '`r Sys.Date()`'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message=FALSE, warning=FALSE)
```


```{r loading packages, message=FALSE, warning=FALSE}

library(limma)
library(tidyverse)
library(ggsignif)
library(biomaRt)
library(RColorBrewer)
library(cowplot)
library(ggpubr)
library(scales)
library(sjmisc)
library(kableExtra)
library(broom)
library(ComplexHeatmap)
  


```
##  Data set comparison order:

Knowles data:  
  meQTLS(aka K4 supp)  
  reQTLS(aka K5 supp)  
ArrGWAS  
HFGWAS  
CADGWAS  
GTEx enrichment  
Seaone 2019  
  Supplemental 1 (408 genes)  
  Supplemental 4 (54 genes)  
  

# Pairwise Data enrichment  

## Knowles Comparison data:  
```{r ensembl conversion, eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
ensembl <- useMart("ensembl", dataset="hsapiens_gene_ensembl")
my_chr <- c(1:22, 'M', 'X', 'Y')  ## creates a filter for each
my_attributes <- c('entrezgene_id', 'ensembl_gene_id', 'hgnc_symbol')
knowfig4 <- read.csv("data/knowfig4.csv",row.names = 1)
knowfig5 <- read.csv("data/knowfig5.csv",row.names = 1)

# covert ensg to entrezid -------------------------------------------------

knowles5 <- getBM(attributes=my_attributes,filters ='ensembl_gene_id',
                  values =knowfig5, mart = ensembl)
#377

knowles4 <- getBM(attributes=my_attributes,filters ='ensembl_gene_id',
                  values =knowfig4, mart = ensembl)
#524

saveRDS(knowles4,"output/knowles4.RDS")
saveRDS(knowles5,"output/knowles5.RDS")

```


```{r loading data files,echo=FALSE,eval=TRUE}
# toplistall <- read.csv("output/toplistall.csv", row.names = 1)

backGL <- read.csv("data/backGL.txt",     row.names =1)
 drug_palNoVeh <- c("#8B006D","#DF707E","#F1B72B", "#3386DD","#707031")
drug_palc <- c("#8B006D","#DF707E","#F1B72B", "#3386DD","#707031","#41B333")
time <- rep((rep(c("3h", "24h"), c(6,6))), 6)
time <- ordered(time, levels =c("3h", "24h"))
drug <- rep(c("Daunorubicin","Doxorubicin","Epirubicin","Mitoxantrone","Trastuzumab", "Vehicle"),12)
mat_drug <- c("Daunorubicin","Doxorubicin","Epirubicin","Mitoxantrone")
indv <- as.factor(rep(c(1,2,3,4,5,6), c(12,12,12,12,12,12)))
labeltop <- (interaction(substring(drug, 0, 2), indv, time))
level_order2 <- c('75','87','77','79','78','71')


toplistall <- readRDS("data/toplistall.RDS")
knowles4 <-readRDS("output/knowles4.RDS")
knowles5 <-readRDS("output/knowles5.RDS")


knowles4 <-knowles4 %>% 
  distinct(entrezgene_id,.keep_all = TRUE)
length(unique(knowles4$entrezgene_id))
knowles5 <-knowles5 %>% 
  distinct(entrezgene_id,.keep_all = TRUE)

# all_QTLSknow<- union(knowles5$entrezgene_id,knowles4$entrezgene_id)

DOXreQTLs <- toplistall %>% 
  dplyr::filter(time =="24_hours") %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>% 
   dplyr::filter(id == "Doxorubicin") %>% 
   dplyr::filter(sigcount=='sig') %>% 
   dplyr::filter(reQTL=="a") %>% 
   dplyr::select(time, id, ENTREZID,SYMBOL)

saveRDS(DOXreQTLs,"output/DOXreQTLs.RDS")
DOXmeQTLs <- toplistall %>% 
  dplyr::filter(time =="24_hours") %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>% 
   dplyr::filter(id == "Doxorubicin") %>% 
   dplyr::filter(sigcount=='sig') %>% 
   dplyr::filter(meSNP=="y") %>% 
   dplyr::select(time, id, ENTREZID,SYMBOL)
saveRDS(DOXmeQTLs,"output/DOXmeQTLs.RDS")
DNRreQTLs <- toplistall %>% 
dplyr::filter(time =="24_hours") %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>%
  dplyr::filter(id =="Daunorubicin") %>% 
  dplyr::filter(sigcount=='sig') %>% 
   dplyr::filter(reQTL=="a") %>% 
   dplyr::select(time, id, ENTREZID,SYMBOL)
saveRDS(DNRreQTLs,"output/DNRreQTLs.RDS")
DNRmeQTLs <- toplistall %>% 
dplyr::filter(time =="24_hours") %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>%
  dplyr::filter(id =="Daunorubicin") %>% 
  dplyr::filter(sigcount=='sig') %>% 
   dplyr::filter(meSNP=="y") %>% 
   dplyr::select(time, id, ENTREZID,SYMBOL)
saveRDS(DNRmeQTLs,"output/DNRmeQTLs.RDS")

EPIreQTLs <- toplistall %>% 
  dplyr::filter(time =="24_hours") %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>%
   dplyr::filter(id =="Epirubicin") %>% 
   dplyr::filter(sigcount=='sig') %>% 
   dplyr::filter(reQTL=="a") %>% 
   dplyr::select(time, id, ENTREZID,SYMBOL)
saveRDS(EPIreQTLs,"output/EPIreQTLs.RDS")

EPImeQTLs <- toplistall %>% 
  dplyr::filter(time =="24_hours") %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>%
   dplyr::filter(id =="Epirubicin") %>% 
   dplyr::filter(sigcount=='sig') %>% 
    dplyr::filter(meSNP=="y") %>% 
   dplyr::select(time, id, ENTREZID,SYMBOL)
saveRDS(EPImeQTLs,"output/EPImeQTLs.RDS")



MTXreQTLs <- toplistall %>% 
  dplyr::filter(time =="24_hours") %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>%
   dplyr::filter(id =="Mitoxantrone") %>% 
   dplyr::filter(sigcount=='sig') %>% 
   dplyr::filter(reQTL=="a") %>% 
   dplyr::select(time, id, ENTREZID,SYMBOL)
saveRDS(MTXreQTLs,"output/MTXreQTLs.RDS")

MTXmeQTLs <- toplistall %>% 
  dplyr::filter(time =="24_hours") %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>%
   dplyr::filter(id =="Mitoxantrone") %>% 
   dplyr::filter(sigcount=='sig') %>% 
    dplyr::filter(meSNP=="y") %>% 
   dplyr::select(time, id, ENTREZID,SYMBOL)
saveRDS(MTXmeQTLs,"output/MTXmeQTLs.RDS")



reQTLcombine=list(DNRreQTLs$ENTREZID,DOXreQTLs$ENTREZID,EPIreQTLs$ENTREZID,MTXreQTLs$ENTREZID)

length(unique(c(DNRreQTLs$ENTREZID,DOXreQTLs$ENTREZID,EPIreQTLs$ENTREZID,MTXreQTLs$ENTREZID)))

print("number of unique genes expressed in pairwise DE set")
QTLoverlaps <- VennDiagram::get.venn.partitions(reQTLcombine)
DoxonlyQTL <-  QTLoverlaps$..values..[[14]]
     ggVennDiagram::ggVennDiagram(reQTLcombine, category.names = c("DNR-187\nsigDEexpressed","DOX-180 \nsigDEexpressed","EPI-176\nsigDEexpressed","MTX-40\nsigDEexpressed"))  

#total 794

```
  
Determining the genetic basis of anthracycline-cardiotoxicity by molecular
response QTL mapping in induced cardiomyocytes
David A Knowles, Courtney K Burrowsâ€ , John D Blischak,
Kristen M Patterson, Daniel J Serie, Nadine Norton, Carole Ober,
Jonathan K Pritchard, Yoav Gilad

Knowles $~~et~ al.~$ eLife 2018;7:e33480. DOI: https://doi.org/10.7554/eLife.33480
My first question was about transcription response at the 24 hour mark with my treatments.   3 hour RNA-seq had low levels of DEGs,so my focus is at 24 hours.  This also happens to be when the Knowles paper collected their RNA-seq data


Supplementary 4 contains a list of 518 SNPs within 1 Mb of TSS, which had a detectable marginal effect on expression (5% FDR). When converted from ensembl gene id to entrez gene id,  my list of unique Entrezgeneids = 521.  I will call these meQTLS for marginal effect QTLs.  

In the meQTLs, 503 are expressed out of 14084.  Using an adj. P value of 0.05, There are 199/6864 in 24 hour daunorubicin, 184/6516 in 24 hour doxorubicin, 182/6202 in 24 hour epirubicin, 30/1327 in 24 hour mitoxantrone and 0 in Trastuzumb.  

Supplementary 5 contains a list of 376 response eQTLs (reQTLs).  These are variants that were associated with modulation of transcriptomic response to doxorubicin treatment. After database name conversion, I have 377 unique Entregene ids.  

Of the reQTLs list, 374 are expressed out of all 14084 expressed genes.  Using an adj. P value of 0.05, There are 187/6864 in 24 hour daunorubicin, 180/6516 in 24 hour doxorubicin, 176/6202 in 24 hour epirubicin, 40/1327 in 24 hour mitoxantrone and 0 in Trastuzumb.

  
```{r  knowles data comparisons}
dataframK_45 <- toplistall %>%
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  filter(time== "24_hours") %>%
  group_by(time, id) %>%
  mutate(K4 = if_else(ENTREZID %in% knowles4$entrezgene_id,1,0))%>%
  mutate(K5 = if_else(ENTREZID %in% knowles5$entrezgene_id,1,0))%>%
  # filter(adj.P.Val<0.05) %>%
 dplyr::summarize(n=n(), K4=sum(K4), K5=sum(K5)) %>% 
  as.data.frame()

sig05_dataframK_45 <- toplistall %>%
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  filter(time== "24_hours") %>%
  group_by(time, id) %>%
  mutate(K4 = if_else(ENTREZID %in% knowles4$entrezgene_id,1,0))%>%
  mutate(K5 = if_else(ENTREZID %in% knowles5$entrezgene_id,1,0))%>%
  filter(adj.P.Val<0.05) %>%
  dplyr::summarize(n=n(), K4=sum(K4), K5=sum(K5)) %>% 
  as.data.frame()

sig05_dataframK_45 %>%
  #mutate_at(.vars = 6, .funs= scientific_format()) %>%
  kable(., caption= "Significant DEG by treatment by (adj. P value of <0.05") %>% 
  kable_paper("striped", full_width = FALSE) %>% 
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "80%", height = "400px")

dataframK_45 %>% 
  kable(., caption= "total number of meQTLS and reQTLS in expressed genes") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "right",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "80%", height = "400px")


chi_squarek4 <- toplistall %>% 
  mutate(id = as.factor(id)) %>%
  dplyr::filter(id!="Trastuzumab") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  group_by(id,time) %>% 
  summarise(pvalue= chisq.test(meSNP, sigcount)$p.value) 

chi_squarek5 <- 
  toplistall %>% 
  mutate(id = as.factor(id)) %>%
  dplyr::filter(id!="Trastuzumab") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>% 
  group_by(id,time) %>% 
  summarise(pvalue= chisq.test(reQTL, sigcount)$p.value) 

chi_squarek4 %>% 
  kable(., caption= "meQTLs (mininmally expressed QTLS) chi-squared pvalues") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "right",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "50%", height = "400px")

chi_squarek5 %>% 
  kable(., caption= "reQTLS in knowles data (reQTLS) chi-squared pvalues") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "right",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "50%", height = "400px")

```


```{r graphs of k4 k5 enrichment and FC}

toplist24hr <-   toplistall %>%
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  filter(time=="24_hours")

toplist3hr <-   toplistall %>%
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  filter(time=="3_hours")


toplist24hr %>% 
  group_by(time, id) %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>% 
  ggplot(., aes(x=id, y=abs(logFC)))+
  geom_boxplot(aes(fill=id))+
  fill_palette(palette =drug_palNoVeh)+
  guides(fill=guide_legend(title = "Treatment"))+
  facet_wrap(~time)+
  theme_bw()+
  xlab("")+
  ylab("abs |Log Fold Change|")+
  theme_bw()+
  facet_wrap(~time)+
  #ggtitle("All QTLs from all expressed genes (n=507)")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        # axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        strip.background = element_rect(fill = "transparent"),
        axis.text = element_text(size = 8, color = "black", angle = 0),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))
  # geom_signif(comparisons = list(c("Daunorubicin", "Doxorubicin"),
  #                                c("Epirubicin", "Doxorubicin"),
  #                                c("Mitoxantrone", "Doxorubicin"),
  #                                  c("Trastuzumab","Doxorubicin"),
  #                                c("Mitoxantrone","Trastuzumab")),
  #                             test = "t.test",
  #             map_signif_level = FALSE,step_increase = 0.1,
  #             textsize = 4)


   toplistall %>%
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>% 
  filter(time =="24_hours") %>% 
  group_by(id,sigcount,meSNP) %>% 
  summarize(K4count=n()) %>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(meSNP), values_from=K4count) %>% 
    mutate(K4prop=(y/(no+y))*100) %>% 
     ggplot(., aes(x=id, y=K4prop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",K4prop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
     ggtitle("non-significant and significant enrichment proportions of Knowles min. eQTLS ")
  
     toplistall %>%
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  mutate(meSNP=if_else(ENTREZID %in%knowles4$entrezgene_id,"y","no")) %>% 
  mutate(reQTL=if_else(ENTREZID %in%knowles5$entrezgene_id,"a","b")) %>% 
  filter(time =="24_hours") %>% 
  group_by(id,sigcount,reQTL) %>% 
  summarize(K5count=n())%>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(reQTL), values_from=K5count) %>% 
   mutate(K5prop=(a/(a+b)*100)) %>% 
       ggplot(., aes(x=id, y=K5prop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",K5prop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("non-significant and significant enrichment proporitions of Knowles reQTLs ")

```

### enrichment comparison
```{r Knowles comparison between sets}

Knowles_count <- toplistall %>%
  filter(id!='Trastuzumab') %>% 
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  group_by(time, id) %>%
  mutate(K4 = if_else(ENTREZID %in% knowles4$entrezgene_id,1,0))%>%
  mutate(K5 = if_else(ENTREZID %in% knowles5$entrezgene_id,1,0))%>%
  filter(adj.P.Val<0.05) %>%
  dplyr::summarize(n=n(), K4=sum(K4), K5=sum(K5)) %>% 
  as.tibble() %>% 
  dplyr::select(time,id,K4,K5) %>% rename("K4_y"='K4',"K5_y"='K5') %>% 
  mutate(time = case_match(time, '3_hours'~'3 hrs',
                           '24_hours'~'24 hrs',.default = time)) %>% 
  mutate(K4_n= 503-K4_y, K5_n=371-K5_y) %>% 
  pivot_longer(!c(time,id), names_to='QTL',values_to="gene_count") %>%
  separate(QTL,into=c("QTL_type",'group'),sep = '_') %>%  
  mutate(QTL_type =case_match(QTL_type, 'K4'~'meQTLs',
                         'K5'~'reQTLs',.default = QTL_type)) %>% 
  mutate(time=factor(time, levels=c("3 hrs","24 hrs"))) %>% 
  group_by(id,time,QTL_type) %>% 
  mutate(percent=gene_count/sum(gene_count)*100)
  
  ggplot(Knowles_count, aes(x=QTL_type,y=gene_count, group=group,  fill=group))+
    geom_col(position='fill')+
    guides(fill="none")+
    facet_wrap(id~time,nrow=2,ncol=4)+
    theme_classic()+
   scale_color_manual(values=drug_palNoVeh)+
    scale_fill_manual(values=drug_palNoVeh)+
       scale_y_continuous(label = scales::percent)

  ###This works
  chi_frame <- Knowles_count %>%
  pivot_wider(id_cols = c(time, id,QTL_type), names_from = group, values_from = gene_count) 

testDNR3chi <- matrix(c(6,13,497, 358), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c("k4", "k5"),c( "y", "n"))) 
    DNR_3chi <- chisq.test(testDNR3chi,correct=FALSE)$p.value
testDNR24chi <- matrix(c(199,187,304, 184), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c("k4", "k5"),c( "y", "n")))  
    DNR_24chi <- chisq.test(testDNR24chi,correct=FALSE)$p.value

testDOX3chi <- matrix(c(0,0,503, 371), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c("k4", "k5"),c( "y", "n")))  
    DOX_3chi <- chisq.test(testDOX3chi,correct=FALSE)$p.value
testDOX24chi <- matrix(c(184,180,318, 191), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c("k4", "k5"),c( "y", "n"))) 
    DOX_24chi <- chisq.test(testDOX24chi,correct=FALSE)$p.value

testEPI3chi <- matrix(c(4,4,499, 367), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c("k4", "k5"),c( "y", "n"))) 
    EPI_3chi <- chisq.test(testEPI3chi,correct=FALSE)$p.value
testEPI24chi <- matrix(c(172,176,331, 195), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c("k4", "k5"),c( "y", "n"))) 
    EPI_24chi <- chisq.test(testEPI24chi,correct=FALSE)$p.value
  
testMTX3chi <- matrix(c(1,0,502, 370), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c("k4", "k5"),c( "y", "n"))) 
    MTX_3chi <- chisq.test(testMTX3chi,correct=FALSE)$p.value
testMTX24chi <- matrix(c(30,40,473, 331), nrow=2, ncol=2, byrow=FALSE,
dimnames=list(c("k4", "k5"),c( "y", "n"))) 
    MTX_24chi <- chisq.test(testMTX24chi,correct=FALSE)$p.value
K4nK5chi <- data.frame(treatment=c('DNR_3','DNR_24','DOX_3','DOX_24','EPI_3','EPI_24','MTX_3','MTX_24'), chi_p.value=c(DNR_3chi,DNR_24chi,DOX_3chi,DOX_24chi,EPI_3chi,EPI_24chi,MTX_3chi,MTX_24chi))

K4nK5chi %>% kable(., caption= "Chi Square p. values between proportions of meQTLs and reQTLS by time and treatment") %>% 
  kable_paper("striped", full_width = TRUE) %>%  
  kable_styling(full_width = FALSE, position = "right",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box( height = "500px")
 
```




## ArrGWAS to 24 hour DEG genes p < 0.05
### 24 hour data set
```{r ArrGWAS}
# How I did the string split
# Arr_GWAS <- ArrGWAS[,13]
# names(Arr_GWAS) <- "genesplit"
# Arr_GWAS <- Arr_GWAS %>% 
#   separate_longer_delim(genesplit, delim = ",")
#write.csv(Arr_GWAS,"data/Arr_GWAS.txt")
toplist24hr <- toplistall %>%
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  filter(time== "24_hours")
arr_GWAS <- read.csv("data/Arr_GWAS.txt", row.names = 1)
Arr_geneset <- readRDS("data/Arr_geneset.RDS")
# Arr_geneset <- getBM(attributes=my_attributes,filters ='hgnc_symbol',
#                   values = arr_GWAS, mart = ensembl)
# #remove duplicates
# Arr_geneset <- Arr_geneset %>% distinct(entrezgene_id, .keep_all =TRUE)
# saveRDS(Arr_geneset,"data/Arr_geneset.RDS")
#Apply sorting
toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(ARR=if_else(ENTREZID %in%Arr_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,ARR) %>% 
  summarize(ARRcount=n())%>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(ARR), values_from=ARRcount) %>% 
   mutate(ARRprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=ARRprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",ARRprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("24 hour non-significant and significant enrichment proporitions of Arrhythmia GWAS ")

##make table of numbers:


dataframARR <- toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(ARR=if_else(ENTREZID %in%Arr_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,ARR) %>% 
  summarize(ARRcount=n()) %>% 
  as.data.frame()

dataframARR %>% 
  kable(., caption= "Significant (adj. P value of <0.05) and non-sig gene counts in Arrhythmia 24 hour GWAS") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")



```
### 3 hour data set

```{r three hour ArrGWAS}

#Apply sorting
toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(ARR=if_else(ENTREZID %in%Arr_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,ARR) %>% 
  summarize(ARRcount=n())%>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(ARR), values_from=ARRcount) %>% 
   mutate(ARRprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=ARRprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",ARRprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("3 hour non-significant and significant enrichment proporitions of Arrhythmia GWAS ")

##make table of numbers:


dataframARR3 <- toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(ARR=if_else(ENTREZID %in%Arr_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,ARR) %>% 
  summarize(ARRcount=n()) %>% 
  as.data.frame()

dataframARR3 %>% kable(., caption= "Significant (adj. P value of <0.05) and non-sig gene counts in Arrhythmia 3 hour GWAS") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")


```

#### chi square test ARR

```{r chi ARR, echo=TRUE, message=FALSE, warning=FALSE}


chi_funarr <-  toplistall %>% 
  mutate(id = as.factor(id)) %>%
  dplyr::filter(id!="Trastuzumab") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(ARR=if_else(ENTREZID %in%Arr_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,time) %>% 
  summarise(pvalue= chisq.test(ARR, sigcount)$p.value) 
print("after performing chi square test between DEgenes, and non DE genes")
chi_funarr


```
```{r ARR heatmap,echo=FALSE}

ARRmat <- chi_funarr %>% ungroup() %>% 
  mutate(ARR= log(pvalue)*(-1)) %>% 
  mutate(id =case_match( id, 'Daunorubicin'~'DNR',   'Doxorubicin'~'DOX' ,'Epirubicin'~'EPI' , 'Mitoxantrone' ~ 'MTX',.default = id)) %>% 
  unite("term_name",id,time,sep="_") %>% 
  column_to_rownames('term_name') 
  


col_fun1 = circlize::colorRamp2(c(-1, 3), c("white", "purple"))
Heatmap( as.matrix(ARRmat), name = "Arrhythmia chi square -log p values", cluster_rows = FALSE, cluster_columns = FALSE,col=col_fun1)



```




## HFGWAS
### 24 hours HF
```{r 24 hours HFGWAS}
##just like ARrGWAS- imported the total csv, the took the "nearest" column and separated out the gene info
# test <- HFGWAS %>% 
#   select(nearest) %>% 
#   separate_wider_delim(nearest, delim = "[", names_sep = "", too_few = "align_start")
# test2 <- str_sub(test$nearest2,0,nchar(test$nearest2)-1)
# Hf_GWAS <- test2
#write.csv(Hf_GWAS, "data/Hf_GWAS.txt")
HF_GWAS <- read.csv("data/Hf_GWAS.txt", row.names =1)
# 
# HF_geneset <- getBM(attributes=my_attributes,filters ='hgnc_symbol',
#                   values = HF_GWAS, mart = ensembl)
# #remove duplicates
# HF_geneset <- HF_geneset %>% distinct(entrezgene_id, .keep_all =TRUE)
# saveRDS(HF_geneset,"data/HF_geneset.RDS")
HF_geneset <- readRDS("data/HF_geneset.RDS")
#Apply sorting
toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(HF=if_else(ENTREZID %in%HF_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,HF) %>% 
  summarize(HFcount=n())%>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(HF), values_from=HFcount) %>% 
   mutate(HFprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=HFprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",HFprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("non-significant and significant enrichment proporitions of Heart Failure GWAS ")

##make table of numbers:


dataframHF <- toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(HF=if_else(ENTREZID %in%HF_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,HF) %>% 
  summarize(HFcount=n()) %>% 
  as.data.frame()

dataframHF %>% #mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(., caption= "Significant (adj. P value of <0.05) and non-sig gene counts in HFhythmia GWAS") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")


```
### 3 hours HF
```{r 3 hours HFGWAS}

#Apply sorting
toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(HF=if_else(ENTREZID %in%HF_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,HF) %>% 
  summarize(HFcount=n())%>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(HF), values_from=HFcount) %>% 
   mutate(HFprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=HFprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",HFprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("non-significant and significant enrichment proporitions of Heart Failure GWAS ")

##make table of numbers:


dataframHF3 <- toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(HF=if_else(ENTREZID %in%HF_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,HF) %>% 
  summarize(HFcount=n()) %>% 
  as.data.frame()

dataframHF3 %>%
  kable(., caption= "Significant (adj. P value of <0.05) and non-sig gene counts in Three hour HFhythmia GWAS") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")

```
#### chi square test HF
```{r chiHF,echo=TRUE}


chi_funhf <-  toplistall %>% 
  mutate(id = as.factor(id)) %>%
  dplyr::filter(id!="Trastuzumab") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(HF=if_else(ENTREZID %in%HF_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,time) %>% 
  summarise(pvalue= chisq.test(HF, sigcount)$p.value) 
print("after performing chi square test between DEgenes, and non DE genes")
chi_funhf

```

```{r HF heatmap,echo=FALSE}
library(ComplexHeatmap)
  
HFmat <- chi_funhf %>% ungroup() %>% 
  mutate("HF"= log(pvalue)*(-1)) %>% 
  mutate(id =case_match( id, 'Daunorubicin'~'DNR',   'Doxorubicin'~'DOX' ,'Epirubicin'~'EPI' , 'Mitoxantrone' ~ 'MTX',.default = id)) %>% 
  unite("term_name",id,time,sep="_") %>% 
  column_to_rownames('term_name') 


Heatmap( as.matrix(HFmat), name = "Heart Failure GWAS\n chi square -log p values", cluster_rows = FALSE, cluster_columns = FALSE,col = col_fun1)



```
## CAD GWAS

### 24 hour data set
```{r CAD 24 GWAS}
# test <- CADGWAS %>% 
#    select(nearest) %>% 
#    separate_wider_delim(nearest, delim = "[", names_sep = "", too_few = "align_start")
#  test2 <- str_sub(test$nearest2,0,nchar(test$nearest2)-1)
# 
# test2[c(32,38,44,74,112,126,191,212)] <- c("TPCN1","C2orf43","FAM222A", "TDRD15"  ,"AGPAT4","SVOP","SVOP","PLG")
#    
#  test2 [c(218,226,228,233,239,245,256,270,281,322,324,332,335,338,347,351,352,358)] <-  c("HPCAL1", "KLHL29"  , "COL4A3BP"  , "ARAP1" ,
#  "VEGFA", "TBPL1","SLC22A3" ,"C19orf38","LPA","VPS29","ATP2A2" ,"ATP2A2","KLHL29","GUCY1A3","KCNE2",  "HOXB9","P2RY2" ,"CTC-236F12.4")
#  
#  CAD_GWAS <- test2
#write.csv(CAD_GWAS, "data/cvd_GWAS.txt")
CAD_GWAS <- read.csv("data/cvd_GWAS.txt", row.names =1)

# CAD_geneset <- getBM(attributes=my_attributes,filters ='hgnc_symbol',
#                   values = CAD_GWAS, mart = ensembl)
# #remove duplicates
# CAD_geneset <- CAD_geneset %>% distinct(entrezgene_id, .keep_all =TRUE)
# 
# saveRDS(CAD_geneset,"data/CAD_geneset.RDS")
CAD_geneset <- readRDS("data/CAD_geneset.RDS")
#Apply sorting



toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(CAD=if_else(ENTREZID %in%CAD_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,CAD) %>% 
  summarize(CADcount=n())%>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(CAD), values_from=CADcount) %>% 
   mutate(CADprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=CADprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",CADprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("non-significant and significant enrichment proporitions of CAD GWAS ")

##make table of numbers:


dataframCAD <- toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(CAD=if_else(ENTREZID %in%CAD_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,CAD) %>% 
  summarize(CADcount=n()) %>% 
  as.data.frame()

dataframCAD %>% #mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(., caption= "Significant (adj. P value of <0.05) and non-sig gene counts in CAD GWAS") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")

```
### 3 hour data set
```{r CAD 3 GWAS}

#Apply sorting
toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(CAD=if_else(ENTREZID %in%CAD_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,CAD) %>% 
  summarize(CADcount=n())%>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(CAD), values_from=CADcount) %>% 
   mutate(CADprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=CADprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",CADprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("3 hour non-significant and significant enrichment proporitions of CAD GWAS ")

##make table of numbers:


dataframCAD3 <- toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(CAD=if_else(ENTREZID %in%CAD_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,sigcount,CAD) %>% 
  summarize(CADcount=n()) %>% 
  as.data.frame()

dataframCAD3 %>% #mutate_at(.vars = 6, .funs= scientific_format()) %>% 
  kable(., caption= "Significant (adj. P value of <0.05) and non-sig gene counts in 3 hour CAD GWAS") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")

```

#### chi square test CAD
```{r chiCAD,echo=TRUE}

chi_funCAD <-  toplistall %>% 
  mutate(id = as.factor(id)) %>%
  dplyr::filter(id!="Trastuzumab") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(CAD=if_else(ENTREZID %in%CAD_geneset$entrezgene_id,"y","no")) %>% 
  group_by(id,time) %>% 
  summarise(pvalue= chisq.test(CAD, sigcount)$p.value) 
print("after performing chi square test between DEgenes, and non DE genes")
chi_funCAD

```
```{r CAD heatmap,echo=FALSE}
library(ComplexHeatmap)
  
CADmat <- chi_funCAD %>% ungroup() %>% 
  mutate("CAD"= log(pvalue)*(-1)) %>% 
  mutate(id =case_match( id, 'Daunorubicin'~'DNR',   'Doxorubicin'~'DOX' ,'Epirubicin'~'EPI' , 'Mitoxantrone' ~ 'MTX',.default = id)) %>% 
  unite("term_name",id,time,sep="_") %>% 
  column_to_rownames('term_name') 

col_fun4 = circlize::colorRamp2(c(-1, 4), c("white", "purple"))

Heatmap( as.matrix(CADmat), name = "Heart Failure GWAS\n chi square -log p values", cluster_rows = FALSE, cluster_columns = FALSE,col_fun4)

GWASmat <- bind_cols(CADmat$CAD,HFmat$HF,ARRmat$ARR) 
colnames(GWASmat) <- c('CAD','HF','ARR')
rownames(GWASmat) <- rownames(CADmat)# %>% tibble() %>% dplyr::select) %>% as.matrix()


Heatmap(as.matrix(GWASmat), name = "GWAS chi square\n -log p values", cluster_rows = FALSE,row_order = c(1,3,5,7,2,4,6,8), cluster_columns = FALSE, column_names_rot = 0,col_fun4)

```
## GTEx V8 heart-left ventrical egenes

```{r GTEx enrichment check}
# heart_gtex <- 
#   readr::read_delim("data/Heart_Left_Ventricle.v8.egenes.txt",  
#                                 delim = "\t", 
#                                 escape_double = FALSE, 
#                                 trim_ws = TRUE)
# heartset <- heart_gtex$gene_name
# GTEX_set <- getBM(attributes=my_attributes,filters = 'hgnc_symbol',
#                   values = heartset, mart = ensembl)
# #remove duplicates
# GTEX_set <- GTEX_set %>% dplyr::distinct(entrezgene_id, .keep_all =TRUE)
# write.csv(GTEX_set,"data/GTEX_setsimple.csv")
GTEX_set <- read.csv("data/GTEX_setsimple.csv", row.names = 1)
 
  toplist24hr %>%
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(gtex=if_else(ENTREZID %in%GTEX_set$entrezgene_id,"y","no")) %>% 
  group_by(id,time,sigcount,gtex) %>% 
  summarize(gtexcount=n())%>% 
    pivot_wider(id_cols = c(id,time,sigcount), names_from=c(gtex), values_from=gtexcount) %>% 
   mutate(gtexprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=gtexprop), group=time) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",gtexprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(time~sigcount)+
       ggtitle("24 hour non-significant and significant enrichment proporitions of GTex eGenes ")
 toplist3hr %>%
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(gtex=if_else(ENTREZID %in%GTEX_set$entrezgene_id,"y","no")) %>% 
  group_by(id,time,sigcount,gtex) %>% 
  summarize(gtexcount=n())%>% 
    pivot_wider(id_cols = c(id,time,sigcount), names_from=c(gtex), values_from=gtexcount) %>% 
   mutate(gtexprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=gtexprop), group=time) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",gtexprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(time~sigcount)+
       ggtitle("3 hour non-significant and significant enrichment proporitions of GTex eGenes ")

```
####chi square
``` {r gtex counts and chi}


dataframgtex <-  toplistall %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(gtex=if_else(ENTREZID %in%GTEX_set$entrezgene_id,"y","no")) %>% 
  group_by(id,time,sigcount,gtex) %>% 
  summarize(gtexcount=n())%>% 
    pivot_wider(id_cols = c(id,time,sigcount), names_from=c(gtex), values_from=gtexcount)# %>% 
  

dataframgtex %>% kable(., caption= "Significant (adj. P value of <0.05) and non-sig gene counts in Arrhythmia 24 hour GWAS") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")



chi_fungtex <-  toplistall %>% 
  mutate(id = as.factor(id)) %>%
  dplyr::filter(id!="Trastuzumab") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
   mutate(gtex=if_else(ENTREZID %in%GTEX_set$entrezgene_id,"y","no")) %>% 
  group_by(id,time) %>% 
  summarise(pvalue= chisq.test(gtex, sigcount)$p.value) 
print("after performing chi square test between DEgenes, and non DE genes")
chi_fungtex

```

## Seaone 2019

Seoane, Jose  Chromatin gene comparison:   comes from supp data NAT. MED 2019
### 24 hours in Pairwise with supplemental data 1


```{r gene comparison, eval=TRUE, message=FALSE, warning=FALSE}
library(readxl)

# Seoane_chromatinregs <- read_excel("C:/Users/renee/Downloads/Supplements folde manuscriptr/NIHMS1539805-supplement-SuppTables.xlsx", 
#     range = "A13:H469")

#write.csv(Seoane_chromatinregs, "data/Seonane2019supp1.txt")

chrom_reg_Seoane <- read_csv(file = "data/Seonane2019supp1.txt",col_types = cols(...1 = col_skip()))
                            
Seoane_2019 <- chrom_reg_Seoane[,2]
names(Seoane_2019) <- "ENTREZID"
chrom_genes <- (unique(Seoane_2019$ENTREZID))

toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(chrom=if_else(ENTREZID %in%chrom_genes,"y","no")) %>% 
  group_by(id,sigcount,chrom) %>% 
  summarize(chromcount=n()) %>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(chrom), values_from=chromcount) %>% 
   mutate(chromprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=chromprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",chromprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("non-significant and significant enrichment proportions of chromatin gene set ")

dataframchrom <- toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(chrom=if_else(ENTREZID %in%chrom_genes,"y","no")) %>% 
  group_by(id,sigcount,chrom) %>% 
  summarize(chromcount=n()) %>% 
  as.data.frame()

dataframchrom %>%
  pivot_wider(., names_from=c('sigcount','chrom'), values_from = 'chromcount') %>% 
  kable(., caption= "Significant (adj. P value of <0.05) and non-sig gene counts in Seoane geneset") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")

toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
   dplyr::filter(id!="Trastuzumab") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(chrom=if_else(ENTREZID %in%chrom_genes,"y","no")) %>% 
  group_by(id) %>% 
 summarise(pvalue= chisq.test(chrom, sigcount)$p.value) 



```

### 3 hours data Pairwise

```{r three hour seaone,message=FALSE, warning=FALSE}
toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(chrom=if_else(ENTREZID %in%chrom_genes,"y","no")) %>% 
  group_by(id,sigcount,chrom) %>% 
  summarize(chromcount=n()) %>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(chrom), values_from=chromcount) %>% 
   mutate(chromprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=chromprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",chromprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("non-significant and significant enrichment proporitions of chromatin gene set ")

dataframchrom3 <- toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(chrom=if_else(ENTREZID %in%chrom_genes,"y","no")) %>% 
  group_by(id,sigcount,chrom) %>% 
  summarize(chromcount=n()) %>% 
  as.data.frame()

dataframchrom3 %>%
  pivot_wider(., names_from=c('sigcount','chrom'), values_from = 'chromcount') %>% 
  kable(., caption= "Significant (adj. P value of <0.05) and non-sig gene counts in 3 hours Seoane geneset") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")
```
#### chi square test Seaone
```{r chi value seoane, echo= TRUE,message=FALSE, warning=FALSE}
##remove Trastuzumab in order to perform chi square tests by time and drug between  DE and non DE enrichment
chi_fun <-  toplistall %>% 
  mutate(id = as.factor(id)) %>%
  dplyr::filter(id!="Trastuzumab") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(chrom=if_else(ENTREZID %in%chrom_genes,"y","no")) %>% 
  group_by(id,time) %>% 
  summarise(pvalue= chisq.test(chrom, sigcount)$p.value) 
print("after performing chi square test between DEgenes, and non DE genes")
chi_fun


```



### Supplemental 4 Seoane 3 hours followed by 24 hours 

```{r supp4 seoane, message=FALSE, warning=FALSE}
# library(readxl)
# Sup4seoane <- read_excel("~/Ward Lab/Cardiotoxicity/Manuscript/seoane201941591_2019_638_MOESM3_ESM.xlsx", 
#     sheet = "SupTable4", range = "A16:J567")

#     sheet = "SupTable4", range = "A16:J567")
#write.csv(Sup4seoane, "output/Sup4seoane.csv")
Sup4seoane <- read.csv("output/Sup4seoane.csv", row.names = 1)
Sup4genes <- Sup4seoane  %>% 
  filter(pval.expAnth<0.05) %>% 
  distinct(entrez, .keep_all = TRUE) %>% 
  dplyr::select(entrez)

Sup4seoane  %>% 
  filter(pval.expAnth<0.05) %>% 
  distinct(entrez, .keep_all = TRUE) %>% 
  dplyr::select(entrez,gene,pval.exp,pval.anthr,pval.expAnth,adjpval) %>% 
  kable(., caption = "List of Seoane Supplemental 4 genes") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped","hover")) %>% 
  scroll_box(width = "100%", height = "400px")

toplist3hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(chrom=if_else(ENTREZID %in%Sup4genes$entrez,"y","no")) %>% 
  group_by(id,sigcount,chrom) %>% 
  summarize(chromcount=n()) %>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(chrom), values_from=chromcount) %>% 
   mutate(chromprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=chromprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",chromprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("Seoane supp 4 enrichment proportions found in my pairwise 3 hour data")


toplist24hr %>% 
  mutate(id = as.factor(id)) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(chrom=if_else(ENTREZID %in%Sup4genes$entrez,"y","no")) %>% 
  group_by(id,sigcount,chrom) %>% 
  summarize(chromcount=n()) %>% 
    pivot_wider(id_cols = c(id,sigcount), names_from=c(chrom), values_from=chromcount) %>% 
   mutate(chromprop=(y/(y+no)*100)) %>% 
       ggplot(., aes(x=id, y=chromprop)) +
       geom_col()+
       geom_text(aes(x=id, label = sprintf("%.2f",chromprop), vjust=-.2))+
       #geom_text(aes(label = expression(paste0("number"~a,"out of",~b))))+
       facet_wrap(~sigcount)+
       ggtitle("Seoane supp 4 enrichment proportions found in my pairwise 24 hour data")


 chi_fun2 <-  
  toplistall %>% 
  mutate(id = as.factor(id)) %>%
  dplyr::filter(id!="Trastuzumab") %>% 
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
 mutate(chrom=if_else(ENTREZID %in%Sup4genes$entrez,"y","no")) %>% 
  group_by(id,time) %>% 
  summarise(pvalue= chisq.test(chrom, sigcount)$p.value) 
print("These are the chisquare values from the 54 genes")
chi_fun2




```

Mitoxantrone is significantly enriched at 24 hours in the 54 genes from supplemental 4 Seoane.




# Using Baysian gene sets to look at enrichment of Seoane data


## Seoane Supplemental 4
```{r loading comotif  data sets }
DEG_cormotif <- readRDS("data/DEG_cormotif.RDS")
list2env(DEG_cormotif,envir=.GlobalEnv)
backGL <- read.csv("data/backGL.txt")
```


```{r Seoane supp4 analysis}
##data sets to compare   Sup4genes$entrez,"y","no"

motifSeoanesummary4 <- toplist24hr %>% 
  distinct(ENTREZID) %>% 
  mutate(ER=if_else(ENTREZID %in% motif_ER,"y","no")) %>% 
  mutate(LR=if_else(ENTREZID %in% motif_LR,"y","no")) %>%
  mutate(TI=if_else(ENTREZID %in% motif_TI,"y","no")) %>%
  mutate(NR=if_else(ENTREZID %in% motif_NR,"y","no")) %>%
  mutate(chrom = if_else(ENTREZID %in% Sup4genes$entrez, "y", "no")) %>% 
  group_by(chrom,ER,TI,LR,NR) %>% 
  dplyr::summarize(n=n()) %>% 
  as.tibble  %>% 
  pivot_wider(id_cols = c(chrom), names_from = c('ER', 'TI', 'LR', 'NR'), values_from= n) %>% 
  rename(.,c("chrom"=chrom,"none"= 2 , "ER" = 3 , "TI" = 4 , "LR" = 5 ,"NR" = 6))
motifSeoanesummary4 %>% kable(., caption= "Summary of genes from Cormotif that are also in Seoane Supp4" )%>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")
    # chisqS4 <- 
##making the matrix

chi_list4 <- toplist24hr %>% 
  distinct(ENTREZID) %>% 
  mutate(ER=if_else(ENTREZID %in%motif_ER,"y","no")) %>% 
  mutate(LR=if_else(ENTREZID %in%motif_LR,"y","no")) %>%
  mutate(TI=if_else(ENTREZID %in%motif_TI,"y","no")) %>%
  mutate(NR=if_else(ENTREZID %in%motif_NR,"y","no")) %>%
  mutate(chrom=if_else(ENTREZID %in%Sup4genes$entrez,"y","no")) %>% 
  group_by(chrom,ER,TI,LR,NR) %>% 
  dplyr::summarize(n=n()) %>% 
  as.tibble  %>% 
  pivot_wider(id_cols = c(chrom), names_from = c('ER', 'TI', 'LR', 'NR'), values_from= n) %>% 
  rename(.,c("chrom"=chrom,"none"= 2 , "ER" = 3 , "TI" = 4 , "LR" = 5 ,"NR" = 6))

chisup4LR <- chisq.test(chi_list4[,c('NR','LR')])
chisup4LR
chisup4ER <- chisq.test(chi_list4[,c('NR','ER')])
chisup4ER
chisup4TI <- chisq.test(chi_list4[,c('NR','TI')])
chisup4TI
chisq.test(chi_list4[,c('ER','TI')])
chisq.test(chi_list4[,c('ER','LR')])
chisq.test(chi_list4[,c('TI','LR')])


```


## Seoane Supplemental 1


```{r supp 1}
 motifSeoanesummary1 <- toplist24hr %>% 
  distinct(ENTREZID) %>% 
  mutate(ER=if_else(ENTREZID %in%motif_ER,"y","no")) %>% 
  mutate(LR=if_else(ENTREZID %in%motif_LR,"y","no")) %>%
  mutate(TI=if_else(ENTREZID %in%motif_TI,"y","no")) %>%
  mutate(NR=if_else(ENTREZID %in%motif_NR,"y","no")) %>%
  mutate(chrom = if_else(ENTREZID %in% chrom_genes, "y", "no")) %>% 
  group_by(chrom,ER,TI,LR,NR) %>% 
  dplyr::summarize(n=n()) %>% 
  as.tibble  %>% 
  pivot_wider(id_cols = c(chrom), names_from = c('ER', 'TI', 'LR', 'NR'), values_from= n) %>% 
  rename(.,c("chrom"=chrom,"none"= 2 , "ER" = 3 , "TI" = 4 , "LR" = 5 ,"NR" = 6))
motifSeoanesummary1 %>% kable(., caption= "Summary of genes from Cormotif that are also in Seoane Supp1" )%>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")
    # chisqS4 <- 
##making the matrix

chi_list1 <- toplist24hr %>% 
  distinct(ENTREZID) %>% 
  mutate(ER=if_else(ENTREZID %in%motif_ER,"y","no")) %>% 
  mutate(LR=if_else(ENTREZID %in%motif_LR,"y","no")) %>%
  mutate(TI=if_else(ENTREZID %in%motif_TI,"y","no")) %>%
  mutate(NR=if_else(ENTREZID %in%motif_NR,"y","no")) %>%
  mutate(chrom=if_else(ENTREZID %in% chrom_genes,"y","no")) %>% 
  group_by(chrom,ER,TI,LR,NR) %>% 
  dplyr::summarize(n=n()) %>% 
  as.tibble  %>% 
  pivot_wider(id_cols = c(chrom), names_from = c('ER', 'TI', 'LR', 'NR'), values_from= n) %>% 
  rename(.,c("chrom"=chrom,"none"= 2 , "ER" = 3 , "TI" = 4 , "LR" = 5 ,"NR" = 6))

chisup1LR <- chisq.test(chi_list1[,c('NR','LR')])
chisup1LR
chisup1ER <- chisq.test(chi_list1[,c('NR','ER')])
chisup1ER
chisup1TI <- chisq.test(chi_list1[,c('NR','TI')])
chisup1TI
chisq.test(chi_list1[,c('ER','TI')])
chisq.test(chi_list1[,c('ER','LR')])
chisq.test(chi_list1[,c('TI','LR')])


```
### chi square heatmap
```{r heatmapsup4n1 cormotif}
supp45X <- NULL
motif <- c('ER','TI','LR')
pval1=c(chisup1ER$p.value,chisup1TI$p.value,chisup1LR$p.value)
pval4=c(chisup4ER$p.value ,chisup4TI$p.value ,chisup4LR$p.value)
supp45X <- tibble("motif"=motif,"pval1"=pval1,"pval4"=pval4)
supp45Xmat <- supp45X %>% 
  mutate(pval1=(-1*log(pval1))) %>% 
  mutate(pval4=(-1*log(pval4))) %>% 
  column_to_rownames('motif') %>% 
  rename(c("n = 408"= pval1, "n = 54"=pval4))

Heatmap( as.matrix(supp45Xmat), name = "Seoane CAG sets \nchi square -log p values", cluster_rows = FALSE, cluster_columns = FALSE)



```


## Crispr list  


```{r Crispr data analysis}
library(readxl)
# Crispr_list <- read_excel("C:/Users/renee/Downloads/41598_2021_92988_MOESM2_ESM.xlsx")
#  View(Crispr_list)
# crispr_genes <- Crispr_list %>% 
#   dplyr::filter(p.value <0.05) %>% 
#   select(GeneName)
  

# crispr_genes <- getBM(attributes=my_attributes,filters ='hgnc_symbol',
                  # values =crispr_genes$GeneName, mart = ensembl)
# write.csv(crispr_genes,'data/crispr_genes.csv')

crispr_genes <- read.csv("data/crispr_genes.csv", row.names = 1)
print(" number of unique crispr_genes after conversion from hgnc symbol to entrezid")
length(unique(crispr_genes$entrezgene_id))
crisprunique <- crispr_genes %>% distinct(entrezgene_id,.keep_all = TRUE)

Doxcrispall <- toplistall %>%
  distinct(ENTREZID,.keep_all = TRUE) %>% 
  dplyr::select(ENTREZID,id,time)
  

crispmotifsummary <- Doxcrispall %>% 
  mutate(ER=if_else(ENTREZID %in% motif_ER,"y","no")) %>% 
  mutate(LR=if_else(ENTREZID %in% motif_LR,"y","no")) %>%
  mutate(TI=if_else(ENTREZID %in% motif_TI,"y","no")) %>%
  mutate(NR=if_else(ENTREZID %in% motif_NR,"y","no")) %>%
  mutate(crisp = if_else(ENTREZID %in% crisprunique$entrezgene_id, "y", "no")) %>% 
  group_by(crisp,ER,TI,LR,NR) %>% 
  dplyr::summarize(n=n()) %>% 
  as.tibble  %>% 
  pivot_wider(id_cols = c(crisp), names_from = c('ER', 'TI', 'LR', 'NR'), values_from= n) %>% 
  rename(.,c("crisp"=crisp,"none"= 2 , "ER" = 3 , "TI" = 4 , "LR" = 5 ,"NR" = 6)) 

cris_mat <- crispmotifsummary %>% dplyr::select(ER:NR) %>% as.matrix()
chicheck <- data.frame(one= c("LR","ER","TI"),two=rep("NR",3),p.value=c("","",""))
  
 chicheck$p.value[1] <- chisq.test(cris_mat[,c('LR','NR')],correct = FALSE)$p.value

chicheck$p.value[2] <- chisq.test(cris_mat[,c('ER','NR')],correct = FALSE)$p.value
chicheck$p.value[3] <- chisq.test(cris_mat[,c('TI','NR')],correct = FALSE)$p.value

chicheck%>% kable(., caption= "chi square test p.values for encrichment of  Doxcrispr gene sets in motif sets" )%>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")

chicheck_1 <- chicheck %>% mutate(p.value=as.numeric(p.value)) %>% 
  mutate(neg.logvalue=(-1*log(p.value))) %>% column_to_rownames('one') %>% dplyr::select(neg.logvalue) %>% as.matrix
col_fun = circlize::colorRamp2(c(0, 2), c("white", "purple"))

Heatmap( chicheck_1, name = "Doxcrispr enrichment \nchi square -log p values", cluster_rows = FALSE, cluster_columns = FALSE, col=col_fun)
```



```{r crisper DEG}


col_fun4 = circlize::colorRamp2(c(0, 5), c("white", "purple"))


pairwisecrispr <- toplistall %>%
  filter(id!='Trastuzumab') %>% 
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(crisp = if_else(ENTREZID %in% crisprunique$entrezgene_id, "y", "no")) %>% 
  group_by(time, id) %>%
  summarise(pvalue= chisq.test(crisp, sigcount, correct=FALSE)$p.value)
 
  
  crisprnumbers <- toplistall %>%
  filter(id!='Trastuzumab') %>% 
  mutate(id = as.factor(id)) %>%
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>%
  mutate(sigcount = if_else(adj.P.Val <0.05,'sig','notsig'))%>%
  mutate(crisp = if_else(ENTREZID %in% crisprunique$entrezgene_id, "y", "no")) %>% 
  group_by(time, id,sigcount,crisp) %>%
  dplyr::summarize(n=n()) %>% 
  as.tibble() #%>% 
  
crisprnumbers %>% kable(., caption= "Summary of genes found in both sigDE and non sigDE by treatment" )%>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")

pairwisecrispr%>% kable(., caption= "Summary of chisqure values between numbers of sigDE and non sigDE by treatment" )%>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE, position = "left",bootstrap_options = c("striped"),font_size = 18) %>% 
  scroll_box(width = "60%", height = "400px")


crisp_pair_mat <- pairwisecrispr %>%
  mutate(neg.log.pvalue= (-1*log(pvalue))) %>% 
  mutate(time= case_match(time, '3_hours'~'3_hrs', '24_hours'~'24_hrs',.default = id)) %>% 
  mutate(id =case_match( id, 'Daunorubicin'~'DNR',   'Doxorubicin'~'DOX' ,'Epirubicin'~'EPI' , 'Mitoxantrone' ~ 'MTX',.default = id)) %>% 
  unite('pairset',time,id ) %>% 
  column_to_rownames('pairset') %>% dplyr::select(neg.log.pvalue) %>% as.matrix()
    
Heatmap( crisp_pair_mat, name = "Doxcrispr pairwise enrichment \nchi square -log p values", cluster_rows = FALSE, cluster_columns = FALSE, col=col_fun)

```



