---
title: "Inital RNA-seq analysis, Pairwise"
author: "ERM"
date: "20230414"
output: 
  html_document: 
    toc: yes
    theme: flatly
    df_print: kable
editor_options: 
  markdown: 
    wrap: 72
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE
)
```

```{css, echo=FALSE}
pre {
  max-height: 400px;
  overflow-y: auto;
}

pre[class] {
  max-height: 200px;
}
```
This starts the documentation of the RNA-seq cardiotoxicity analysis for my cardiotoxicity data.
```{r library loading, warning=FALSE, echo=TRUE, message=FALSE}

library(edgeR)#
library(limma)#
library(RColorBrewer)
library(gridExtra)#
library(reshape2)#
library(data.table)
library(tidyverse)
library(scales)
library(biomaRt)#
library(cowplot)#
library(ggrepel)#
library(corrplot)
library(Hmisc)
library(ggpubr)
```


```{r functions for graphs, warning=FALSE, echo=TRUE, message=FALSE}
pca_plot <-
  function(df,
           col_var = NULL,
           shape_var = NULL,
           title = "") {
    ggplot(df) + geom_point(aes_string(
      x = "PC1",
      y = "PC2",
      color = col_var,
      shape = shape_var
    ),
    size = 5) +
      labs(title = title, x = "PC 1", y = "PC2") +
      scale_color_manual(values = c(
        "#8B006D",
        "#DF707E",
        "#F1B72B",
        "#3386DD",
        "#707031",
        "#41B333"
      ))
  }

pca_var_plot <- function(pca) {
  # x: class == prcomp
  pca.var <- pca$sdev ^ 2
  pca.prop <- pca.var / sum(pca.var)
  var.plot <-
    qplot(PC, prop, data = data.frame(PC = 1:length(pca.prop),
                                      prop = pca.prop)) +
    labs(title = 'Variance contributed by each PC',
         x = 'PC', y = 'Proportion of variance')
}

calc_pca <- function(x) {
  # Performs principal components analysis with prcomp
  # x: a sample-by-gene numeric matrix
  prcomp(x, scale. = TRUE, retx = TRUE)
}

get_regr_pval <- function(mod) {
  # Returns the p-value for the Fstatistic of a linear model
  # mod: class lm
  stopifnot(class(mod) == "lm")
  fstat <- summary(mod)$fstatistic
  pval <- 1 - pf(fstat[1], fstat[2], fstat[3])
  return(pval)
}

plot_versus_pc <- function(df, pc_num, fac) {
  # df: data.frame
  # pc_num: numeric, specific PC for plotting
  # fac: column name of df for plotting against PC
  pc_char <- paste0("PC", pc_num)
  # Calculate F-statistic p-value for linear model
  pval <- get_regr_pval(lm(df[, pc_char] ~ df[, fac]))
  if (is.numeric(df[, f])) {
    ggplot(df, aes_string(x = f, y = pc_char)) + geom_point() +
      geom_smooth(method = "lm") + labs(title = sprintf("p-val: %.2f", pval))
  } else {
    ggplot(df, aes_string(x = f, y = pc_char)) + geom_boxplot() +
      labs(title = sprintf("p-val: %.2f", pval))
  }
}
x_axis_labels = function(labels, every_nth = 1, ...) {
  axis(side = 1,
       at = seq_along(labels),
       labels = F)
  text(
    x = (seq_along(labels))[seq_len(every_nth) == 1],
    y = par("usr")[3] - 0.075 * (par("usr")[4] - par("usr")[3]),
    labels = labels[seq_len(every_nth) == 1],
    xpd = TRUE,
    ...
  )
}
```



```{r file loading, eval=TRUE, include=FALSE}

design <- read.csv("data/data_outline.txt", row.names = 1)
mymatrix <- readRDS("data/allmatrix.RDS")
seq_info <-read.csv("output/sequencing_info.txt", row.names=1)

drug_palc <-
  c("#8B006D",
    "#DF707E",
    "#F1B72B",
    "#3386DD",
    "#707031",
    "#41B333")

time <- rep((rep(c("3h", "24h"), c(6, 6))), 6)
time <- ordered(time, levels = c("3h", "24h"))
mymatrix$samples$time <- time

indv <- as.factor(rep(c(1, 2, 3, 4, 5, 6), c(12, 12, 12, 12, 12, 12)))
mymatrix$samples$indv <- indv

group <-
  (rep(c(
    "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12"
  ), 6))
group <- factor(group, levels = c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12))

drug <-
  rep(
    c(
      "DNR",
      "DOX",
      "EPI",
      "MTX",
      "TRZ",
      "VEH"
    ),
    12
  )
mymatrix$samples$drug <- drug

RIN <-
  c(9.3,
    9.8,
    9.8,
    10.0,
    9.6,
    9.9,  
    7.8,
    8.7,
    8.6,
    9.1,
    9.4,
    9.3,
    9.4,
    9.6,
    9.9,
    9.4,
    9.9,
    9.8,
    9.5,
    9.9,
    9.6,
    9.9,
    8.8,
    9.5,
    9.3,
    10.0,
    9.3,
    9.6,
    9.7,
    9.4,
    8.9,
    7.5,
    8.6,
    8.5,
    8.6,
    9.3,
    9.6,
    9.6,
    9.4,
    9.7,
    9.9,
    10.0,
    9.4,
    9.6,
    9.4,
    9.5,
    9.8,
    9.5,
    9.6,
    9.7,
    9.6,
    9.9,
    9.6,
    10.0,
    8.3,
    8.4,
    9.4,
    9.7,
    10.0,
    10.0,
    9.6,
    9.6,
    9.7,
    9.5,
    9.8,
    9.5,
    9.8,
    8.6,
    8.9,
    9.1,
    9.3,
    9.6
  )
mymatrix$samples$RIN <- RIN

samplenames <-
  c(
    "MCW_RM_R_11",
    "MCW_RM_R_12",
    "MCW_RM_R_13",
    "MCW_RM_R_14",
    "MCW_RM_R_15",
    "MCW_RM_R_16",
    "MCW_RM_R_17",
    "MCW_RM_R_18",
    "MCW_RM_R_19",
    "MCW_RM_R_20",
    "MCW_RM_R_21",
    "MCW_RM_R_22",
    "MCW_RM_R_23",
    "MCW_RM_R_24",
    "MCW_RM_R_25",
    "MCW_RM_R_26",
    "MCW_RM_R_27",
    "MCW_RM_R_28",
    "MCW_RM_R_29",
    "MCW_RM_R_30",
    "MCW_RM_R_31",
    "MCW_RM_R_32",
    "MCW_RM_R_33",
    "MCW_RM_R_34",
    "MCW_RM_R_35",
    "MCW_RM_R_36",
    "MCW_RM_R_37",
    "MCW_RM_R_38",
    "MCW_RM_R_39",
    "MCW_RM_R_40",
    "MCW_RM_R_41",
    "MCW_RM_R_42",
    "MCW_RM_R_43",
    "MCW_RM_R_44",
    "MCW_RM_R_45",
    "MCW_RM_R_46",
    "MCW_RM_R_47",
    "MCW_RM_R_48",
    "MCW_RM_R_49",
    "MCW_RM_R_50",
    "MCW_RM_R_51",
    "MCW_RM_R_52",
    "MCW_RM_R_53",
    "MCW_RM_R_54",
    "MCW_RM_R_55",
    "MCW_RM_R_56",
    "MCW_RM_R_57",
    "MCW_RM_R_58",
    "MCW_RM_R_59",
    "MCW_RM_R_60",
    "MCW_RM_R_61",
    "MCW_RM_R_62",
    "MCW_RM_R_63",
    "MCW_RM_R_64",
    "MCW_RM_R_65",
    "MCW_RM_R_66",
    "MCW_RM_R_67",
    "MCW_RM_R_68",
    "MCW_RM_R_69",
    "MCW_RM_R_70",
    "MCW_RM_R_71",
    "MCW_RM_R_72",
    "MCW_RM_R_73",
    "MCW_RM_R_74",
    "MCW_RM_R_75",
    "MCW_RM_R_76",
    "MCW_RM_R_77",
    "MCW_RM_R_78",
    "MCW_RM_R_79",
    "MCW_RM_R_80",
    "MCW_RM_R_81",
    "MCW_RM_R_82"
  )

anno <- (cbind(samplenames, indv, drug, time, RIN, group))
anno <- as.data.frame(anno)
# saveRDS(anno,"data/annotation_data_frame.RDS")

```

```{r gene id infromation, eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
 
 ###now we add genenames to the geneid###
 
geneid <- rownames(mymatrix) ### pulls the names we have in the counts file
genes <- select(Homo.sapiens, keys=geneid, columns=c("SYMBOL"),
  keytype="ENTREZID")
genes <- genes[!duplicated(genes$ENTREZID),]
mymatrix$genes <- genes

```
# Initial RNA-seq quality checks

```{r seq quals, echo=TRUE, message=FALSE, warning=FALSE}

drug_palc <- c("#8B006D","#DF707E","#F1B72B", "#3386DD","#707031","#41B333")
seq_info %>% 
  filter(type=="Total_reads") %>% 
  ggplot(., aes (x =samplenames, y=count, fill = drug, group_by=indv))+
  geom_col()+
 geom_hline(aes(yintercept=20000000))+
 scale_fill_manual(values=drug_palc)+
  ggtitle(expression("Total number of reads by sample"))+
  xlab("")+
  ylab(expression("RNA -sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

seq_info %>% 
  filter(type=="Total_reads") %>% 
  ggplot(., aes (x =drug, y=count, fill = drug))+
  geom_boxplot()+
 scale_fill_manual(values=drug_palc)+
  ggtitle(expression("Total number of reads by treatment"))+
  xlab("")+
  ylab(expression("RNA -sequencing reads"))+
  theme_bw()+
  
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))
seq_info %>% 
  filter(type=="Total_reads") %>% 
  ggplot(., aes (x =as.factor(indv), y=count))+
  geom_boxplot(aes(fill=as.factor(indv)))+
 scale_fill_brewer(palette = "Dark2", name = "Individual")+
  ggtitle(expression("Total number of reads by individual"))+
  xlab("")+
  ylab(expression("RNA -sequencing reads"))+
  theme_bw()+

  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 0, hjust = 1),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

seq_info %>% 
  filter(type=="Mapped_reads") %>% 
  ggplot(., aes (x =samplenames, y=count, fill = drug, group_by=indv))+
  geom_col()+
  geom_hline(aes(yintercept=20000000))+
 scale_fill_manual(values=drug_palc)+
  ggtitle(expression("Total number of mapped reads by sample"))+
  xlab("")+
  ylab(expression("RNA -sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


seq_info %>% 
  filter(type=="Mapped_reads") %>% 
  ggplot(., aes (x =as.factor(indv), y=count))+
  geom_boxplot(aes(fill=as.factor(indv)))+
 scale_fill_brewer(palette = "Dark2", name = "Individual")+
  ggtitle(expression("Total mapped reads by individual"))+
  xlab("")+
  ylab(expression("Number of reads"))+
  theme_bw()+

  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 0, hjust = 0.5),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


seq_info %>% 
  filter(type=="Mapped_reads") %>% 
  ggplot(., aes (x =drug, y=count))+
  geom_boxplot(aes(fill=drug))+
 scale_fill_manual(values=drug_palc)+
  ggtitle(expression("Total mapped reads by treatment"))+
  xlab("")+
  ylab(expression("Number of reads"))+
  theme_bw()+

  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 0, hjust = 0.5),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

```

## Initial histograms from count matrix

```{r readinfo,echo=TRUE, message=FALSE, warning=FALSE}

cpm <- cpm(mymatrix)
lcpm <- cpm(mymatrix, log=TRUE)  ### for determining the basic cutoffs
dim(cpm)

row_means <- rowMeans(lcpm)
x <- mymatrix[row_means > 0,]
dim(x)

filcpm_counts <- cpm(x$counts, log = TRUE)


label <- (interaction(drug, indv, time))
colnames(filcpm_counts) <- label

hist(lcpm,  main = "Histogram of total counts (unfiltered)", 
     xlab =expression("Log"[2]*" counts-per-million"), col =4 )

hist(cpm(x$counts, log = TRUE), main = "Histogram of filtered counts using rowMeans > 0 method", 
     xlab =expression("Log"[2]*" counts-per-million"), col =2 ) 


boxplot(lcpm, main = "Boxplots of log cpm per sample",xaxt = "n", xlab= "")
x_axis_labels(labels = label, every_nth = 1, adj=0.7, srt =90, cex =0.4)

boxplot(filcpm_counts, main ="boxplots of log cpm per sample filtered",xaxt = "n", xlab="")
x_axis_labels(labels = label, every_nth = 1, adj=0.7, srt =90, cex =0.4)



```

## PCA by treatment and as a whole


```{r individual PCA by drug and time steps, echo=FALSE, message=FALSE, warning=FALSE}

label <- (interaction(drug, indv, time))
smlabel <- (interaction(drug, time))
x <- calcNormFactors(x, method = "TMM")
normalized_lib_size <- x$samples$lib.size * x$samples$norm.factors
dat_cpm <- cpm(x$counts, lib.size = normalized_lib_size, log=TRUE)

colnames(dat_cpm) <- label
# saveRDS(dat_cpm,"data/dat_cpm.RDS")
anno$time <- ordered(time, levels =c("3h", "24h"))
anno$group <- group
cpm_per_sample <- cbind(anno, t(dat_cpm)) 
treattype <-c("VEH","DNR","DOX","EPI","MTX","TRZ")

for (b in treattype) { 
  dat <- dat_cpm[,anno$drug %in% c("none", b)] 
  cat("\n\n### ", b, "\n\n") 
  # PCA s
  pca <- calc_pca(t(dat))$x
  pca <- data.frame(anno[anno$drug %in% c("none", b), c("drug", "time","indv")], pca)
  pca <- droplevels(pca) 
  for (cat_var in c("drug", "time", "indv")) {
    assign(paste0(cat_var, "_pca"), arrangeGrob(pca_plot(pca, cat_var)))
  } 
  drug_time_pca <- pca_plot(pca, "drug", "time")
 
  grid.arrange(drug_time_pca, indv_pca,  nrow =2, top =paste(b)) 
  }
# write.csv(x$counts, "data/cpmnorm_counts.csv")

pca_all <- calc_pca(t(dat_cpm))
pca_all_anno <- data.frame(anno, pca_all$x)
varpals <- (pca_var_plot(pca_all))
print(varpals)



head(pca_all_anno)[,1:12]
drug_palc <- c("#8B006D","#DF707E","#F1B72B", "#3386DD","#707031","#41B333")

pca_all_anno %>%
  ggplot(.,aes(x = PC1, y = PC2, col=drug, shape=time))+
  geom_point(size= 5)+
  scale_color_manual(values=drug_palc)+
   ggrepel::geom_text_repel(label=indv)+
  #scale_shape_manual(name = "Time",values= c("3h"=0,"24h"=1))+
  ggtitle(expression("PCA of log"[2]*"(cpm)"))+
  theme_bw()+
  guides(col="none", size =4)+
  labs(y = "PC 2 (15.76%)", x ="PC 1 (29.06%)")+
  theme(plot.title=element_text(size= 14,hjust = 0.5),
        axis.title = element_text(size = 12, color = "black"))


pca_all_anno %>%
  ggplot(.,aes(x = PC2, y = PC3, col=drug, shape=time))+
  geom_point(size= 5)+
  scale_color_manual(values=drug_palc)+
  ggrepel::geom_text_repel(label=indv)+
  ggtitle(expression("PCA of log"[2]*"(cpm)"))+
  theme_bw()+
  guides( size =4)+
  labs(x = "PC 2 (15.76%)", y ="PC 3 (8.3%)")+
  theme(plot.title=element_text(size= 14,hjust = 0.5),
        axis.title = element_text(size = 12, color = "black"))
prdat_cpm <- calc_pca(t(dat_cpm)) 
storepca <- summary(prdat_cpm)
head(storepca$importance[,1:7])

```




## Typical genes expressed in iPSC-CMS


```{r typical gene expression, echo=TRUE, message=FALSE, warning=FALSE}

genecardiccheck <- c("MYH7", "TNNT2","MYH6","ACTN2","BMP3","TNNI3","RYR2","CACNA1C","KCNQ1", "HCN1", "ADRB1", "ADRB2")
#ensembl <- useMart("ensembl", dataset="hsapiens_gene_ensembl")
#saveRDS(ensembl, "data/ensembl_backup.RDS")
#ensemble <- readRDS("data/ensembl_backup.RDS")
#my_chr <- c(1:22, 'M', 'X', 'Y')  ## creates a filter for each database

#my_attributes <- c('entrezgene_id', 'ensembl_gene_id', 'hgnc_symbol')

#heartgenes <- getBM(attributes=my_attributes,filters ='hgnc_symbol',
                # values = genecardiccheck, mart = ensembl)
#write.csv(heartgenes, "data/heartgenes.csv")
heartgenes <-read.csv("data/heartgenes.csv")

fungraph <- as.data.frame(filcpm_counts[rownames(filcpm_counts) %in% heartgenes$entrezgene_id,])


fungraph %>% 
  rownames_to_column("entrezgene_id") %>% 
  pivot_longer(-entrezgene_id, names_to = "samples",values_to = "counts") %>% 
  mutate(gene = case_match(entrezgene_id,"88"~"ACTN2","153"~"ADRB1",
  "154"~"ADRB2","651"~"BMP3","775"~"CACNA1C", "100874369"~"CACNA1C","348980"~"HCN1",
                           "3784"~"KCNQ1", "4624"~"MYH6","4625"~"MYH7","6262"~"RYR2",
                           "7137"~"TNNI3","7139"~"TNNT2",.default = entrezgene_id)) %>% 
  ggplot(., aes(x=reorder(gene,counts,decreasing=TRUE), y=counts))+
  geom_boxplot()+
  ggtitle(expression("Expression of typical cardiac tissue genes"))+
  xlab("")+
  ylab(expression("log"[2]~"cpm"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size =10, color = "black", angle = 0),
        strip.text.y = element_text(color = "white"))
```


## correlation heatmap of counts matrix
```{R pheatmap clustering of gene counts, echo=TRUE}

colnames(filcpm_counts) <- label
# saveRDS(filcpm_counts,"data/filcpm_counts.RDS")
mcort <- cor(filcpm_counts)
pheatmap::pheatmap(mcort , cluster_rows = TRUE, cluster_cols = TRUE)
```



### now to get the counts set for DEG!!

```{r counts to DEG set up, echo= TRUE}
group1 <- interaction(drug,time)
mm <- model.matrix(~0 + group1)
colnames(mm) <- c("A3", "X3", "E3","M3","T3", "V3","A24", "X24", "E24","M24","T24", "V24")
y <- voom(x, mm,plot =TRUE)

corfit <- duplicateCorrelation(y, mm, block = indv)

v <- voom(x, mm, block = indv, correlation = corfit$consensus)

fit <- lmFit(v, mm, block = indv, correlation = corfit$consensus)

cm <- makeContrasts(
  V.DA = V3 - A3,
  V.DX = V3 - X3,
  V.EP = V3 - E3,
  V.MT = V3 - M3,
  V.TR = V3 - T3,
  V.DA24 = V24-A24,
  V.DX24= V24-X24,
  V.EP24= V24-E24,
  V.MT24= V24-M24,
  V.TR24= V24-T24,
  levels = mm)

vfit <- lmFit(y, mm)

vfit<- contrasts.fit(vfit, contrasts=cm)

efit2 <- eBayes(vfit)
# saveRDS(efit2,"data/efit2_final.RDS")


```

```{r Differential expression, echo=FALSE}
efit2 <- readRDS("data/efit2_final.RDS")

V.DA.top= topTable(efit2, coef=1, adjust.method="BH", number=Inf, sort.by="p")
V.DX.top= topTable(efit2, coef=2, adjust.method="BH", number=Inf, sort.by="p")
V.EP.top= topTable(efit2, coef=3, adjust.method="BH", number=Inf, sort.by="p")
V.MT.top= topTable(efit2, coef=4, adjust.method="BH", number=Inf, sort.by="p")
V.TR.top= topTable(efit2, coef=5, adjust.method="BH", number=Inf, sort.by="p")
V.DA24.top= topTable(efit2, coef=6, adjust.method="BH", number=Inf, sort.by="p")
V.DX24.top= topTable(efit2, coef=7, adjust.method="BH", number=Inf, sort.by="p")
V.EP24.top= topTable(efit2, coef=8, adjust.method="BH", number=Inf, sort.by="p")
V.MT24.top= topTable(efit2, coef=9, adjust.method="BH", number=Inf, sort.by="p")
V.TR24.top= topTable(efit2, coef=10, adjust.method="BH", number=Inf, sort.by="p")

sigVDA3 = V.DA.top[V.DA.top$adj.P.Val < 0.05 , ]
sigVDX3 = V.DX.top[V.DX.top$adj.P.Val < 0.05 , ]
sigVEP3 = V.EP.top[V.EP.top$adj.P.Val < 0.05 , ]
sigVMT3 = V.MT.top[V.MT.top$adj.P.Val < 0.05 , ]
sigVTR3 = V.TR.top[V.TR.top$adj.P.Val < 0.05 , ]
sigVDA24 = V.DA24.top[V.DA24.top$adj.P.Val < 0.05 , ]
sigVDX24 = V.DX24.top[V.DX24.top$adj.P.Val < 0.05 , ]
sigVEP24 = V.EP24.top[V.EP24.top$adj.P.Val < 0.05 , ]
sigVMT24 = V.MT24.top[V.MT24.top$adj.P.Val < 0.05 , ]
sigVTR24 = V.TR24.top[V.TR24.top$adj.P.Val < 0.05 , ]

siglist <- list(sigVDA3,
             sigVDX3,
             sigVEP3,
             sigVMT3,
             sigVTR3,
             sigVDA24,
             sigVDX24,
             sigVEP24,
             sigVMT24,
             sigVTR24)
names(siglist) <- c("sigVDA3",
             "sigVDX3",
             "sigVEP3",
             "sigVMT3",
             "sigVTR3",
             "sigVDA24",
             "sigVDX24",
             "sigVEP24",
             "sigVMT24",
             "sigVTR24")
# siglist <- lapply(siglist, function(x) {colnames(x) <- gsub(pattern = "^siglist\\.", replacement = "", x = colnames(x))
# return(x)})
plot_filenames <- c("V.DA.top","V.DX.top","V.EP.top","V.MT.top","V.TR.top","V.DA24.top",
"V.DX24.top","V.EP24.top","V.MT24.top","V.TR24.top")
toptablebydrug <- list(V.DA.top,V.DX.top , V.EP.top , V.MT.top , V.TR.top , V.DA24.top ,
 V.DX24.top , V.EP24.top , V.MT24.top , V.TR24.top )
names(toptablebydrug) <- plot_filenames
# saveRDS(toptablebydrug,"output/toptablebydrug.RDS")

 for(i in seq_along(siglist)) {
 fnl <- paste0("data/",names(siglist)[i],".txt")
    write.csv(siglist[i], fnl, col.names = TRUE, row.names =FALSE, sep="")
 }
# 
# saveRDS(siglist,"data/siglist_final.RDS")


```


# DEG analysis
## summary
```{r summary of DEG}
efit2 <- readRDS("data/efit2_final.RDS")
sum <- summary(decideTests(efit2))

sum

```


## Volcano plots from pairwise gene analysis 


```{r Volcanoplots, echo=TRUE, message=FALSE, warning=FALSE}
library(cowplot)
siglist_final <- readRDS("data/siglist_final.RDS")
list2env(siglist_final,envir = .GlobalEnv)
volcanosig <- function(df, psig.lvl,topg) {
    df <- df %>% 
    mutate(threshold = ifelse(adj.P.Val > psig.lvl, "A", ifelse(adj.P.Val <= psig.lvl & logFC<=0,"B","C")))
    
  ggplot(df, aes(x=logFC, y=-log10(adj.P.Val))) + 
    geom_point(aes(color=threshold))+
    xlab(expression("Log"[2]*" FC"))+
    ylim(0,30)+
    ylab(expression("-log"[10]*"P Value"))+
    scale_color_manual(values = c("black", "red","blue"))+
    theme_cowplot()+
    theme(legend.position = "none",
          plot.title = element_text(size = rel(0.8), hjust = 0.5),
          axis.title = element_text(size = rel(0.8))) 
}

v1 <- volcanosig(V.DA.top, 0.01,0)+ ggtitle("Daunorubicin\n 3 hour")
v2 <- volcanosig(V.DA24.top, 0.01,0)+ ggtitle("Daunorubicin\n 24 hour")
v3 <- volcanosig(V.DX.top, 0.01,0)+ ggtitle("Doxorubicin\n 3 hour")+ylab("")
v4 <- volcanosig(V.DX24.top, 0.01,0)+ ggtitle("Doxorubicin\n 24 hour")+ylab("")
v5 <- volcanosig(V.EP.top, 0.01,0)+ ggtitle("Epirubicin\n 3 hour")+ylab("")
v6 <- volcanosig(V.EP24.top, 0.01,0)+ ggtitle("Epirubicin\n 24 hour")+ylab("")
v7 <- volcanosig(V.MT.top, 0.01,0)+ ggtitle("Mitoxatrone\n 3 hour")+ylab("")
v8 <- volcanosig(V.MT24.top, 0.01,0)+ ggtitle("Mitoxatrone\n 24 hour")+ylab("")
v9 <- volcanosig(V.TR.top, 0.01,0)+ ggtitle("Trastuzumab\n 3 hour")+ylab("")
v10 <- volcanosig(V.TR24.top, 0.01,0)+ ggtitle("Trastuzumab\n 24 hour")+ylab("")

Volcanoplots <- plot_grid(v1,v3,v5,v7,v9,v2,v4,v6,v8,v10, nrow = 2, ncol = 5)
Volcanoplots
# saveRDS(Volcanoplots,"output/Volcanoplot_10.RDS")
```




