---
title: "Supplementary figures"
author: "ERM"
date: "`r Sys.Date()`"
output: 
  html_document:
  keep_md: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,message=FALSE, warning=FALSE,
                      dev = c("svg","png","pdf"),
                      dpi = 300)
```

```{r packages, message=FALSE, warning=FALSE}
library(tidyverse)
library(ggpubr)
library(rstatix)
library(zoo)
library(ggsignif)
library(RColorBrewer)
library(kableExtra)
library(ComplexHeatmap)
library(gridExtra)
library(cowplot)

```
### Fig. S1
```{r Fig S1}

ctnnt <- read.csv("data/ctnnt_results.txt", row.names = 1)
ctnnt %>% 
  mutate(Individual=fct_inorder(Individual)) %>% 
  ggplot(., aes(Individual,Percent , fill=Individual))+
  geom_boxplot()+
  geom_point()+
  geom_hline(yintercept =70,linetype="dashed", alpha=0.75)+###adds a line indicating high positivity +
coord_cartesian(ylim = c(0,105))+ ##set those limits
  theme_bw()+  ##white background
  labs(title="Cardiomyocyte Purity")+ #subtitle = "from  n>3 differentiations")+
  geom_boxplot(color="black",alpha =0.2, fill=NA, fatten=0, show.legend = FALSE)+
  scale_fill_brewer(palette = "Dark2",name="" )+
  xlab(NULL)+ 
  ylab("% cTNNT+ ")+
  guides(fill = NULL)+
  theme(plot.title = element_text(hjust = 0.5, size =20, face= "bold"), 
        axis.title.x=element_blank(),
        axis.text.x=element_blank(),###removes all axis names and tick names etc.####
        axis.ticks.x=element_blank(),
        # legend.text=element_text(size=15), 
        axis.title.y=element_text(size=15),
        axis.ticks.y=element_line(size =2),
        axis.text.y=element_text(size=10, face = "bold"),
       panel.grid.major = element_line(colour = 'grey'),
       panel.border=element_rect(fill = NA, size = 3),
       plot.subtitle=element_text(size=18, hjust=0.5, face="italic", color="black")) 
(summary(ctnnt)) %>% 
  kable(., caption= "Stats summary of cTNNT+ FACs readings") %>% 
  kable_paper("striped", full_width = FALSE) %>%  
  kable_styling(full_width = FALSE,font_size = 18) #%>% 
  # scroll_box(width = "60%", height = "400px")

```

### Fig. S2


```{r Fig S2}


ld50_table <- read.csv("data/ld50_table.csv",row.names = 1)
ld_corr <- cor(ld50_table)
col_fun1 = circlize::colorRamp2(c(-1, 1), c("white", "purple"))

Heatmap(ld_corr, cluster_rows = FALSE,cluster_columns = FALSE, col = col_fun1,column_title = expression("LD "[50]*" correlation"),row_title=" ",row_split = factor(rep(c("1","2","3","4","5","6"),each=2)), column_split = 
    factor(rep(c("1","2","3","4","5","6"),each=2)))

Heatmap(ld_corr, cluster_rows = FALSE,cluster_columns = FALSE, column_title = expression("LD "[50]*" correlation"),row_title=" ",row_split = factor(rep(c("1","2","3","4","5","6"),each=2)), column_split = 
    factor(rep(c("1","2","3","4","5","6"),each=2)))

```
### Fig. S3


```{r Fig S3}
drug_pal_fact <- c("#8B006D","#DF707E","#F1B72B", "#3386DD","#707031","#41B333")
viability <- readRDS("data/viability.RDS")
norm_LDH48 <- readRDS("data/supp_normLDH48.RDS")
viability %>% 
  full_join(., norm_LDH48,by = c("indv","Drug","Conc")) %>% 
  ggplot(., aes(x=per.live, y=ldh))+
  geom_point(aes(col=indv))+
  geom_smooth(method="lm")+
  facet_wrap("Drug")+
  theme_bw()+
  xlab("Average viability of cardiomyocytes/100") +
  ylab("Average LDH") +
  ggtitle("Relative viability and relative LDH release at 48 hours")+
  scale_color_brewer(palette = "Dark2",name = "Individual", label = c("1","2","3","4","5","6"))+
  stat_cor(method="pearson",aes(label = paste(..r.label.., ..p.label.., sep = "~`,`~")),
           color = "red")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5,face = "bold"),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(size = 1.5),
        axis.text = element_text(size = 8, color = "black", angle = 20),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.background = element_rect(fill = "white")) 

```

### Fig. S4 

```{r Fig S4}
viabilitytable <- readRDS("data/averageviabilitytable.RDS")
viabilitytable %>% 
  ungroup() %>% 
    mutate(indv=substr(SampleID,4,4)) %>% 
    mutate(indv=factor(as.numeric(indv))) %>%
    filter(Conc <5) %>% 
    mutate(Conc= factor(as.numeric(Conc))) %>% 
    group_by(indv,Drug,Conc,sDrug) %>% 
    dplyr::summarize(Viability=mean(Mean)) %>%
    ungroup() %>% 
    ggplot(.,  aes(x=sDrug, y= Viability*100 )) +
    geom_boxplot(position="dodge", outlier.colour = "transparent",
                 aes(fill=Drug))+
    geom_point(aes(color=indv))+
    guides(alpha = "none")+
    ylim(0,150.5)+
    scale_color_brewer(palette = "Dark2",
                       guide="legend",
                       name ="Individual", 
                       labels(c(1,2,3,4,5,6)))+
    scale_fill_manual(values=drug_pal_fact)+
    theme_classic() +
    # geom_hline(yintercept = 1,lty = 4)+
    ylab("Viability") +
  xlab("Treatment")+
    facet_wrap(~Conc)+ 
    ggtitle("Viablity across concentrations at 48 hours")+
    theme(axis.title=element_text(size=10),
          axis.ticks=element_line(size =2),
          axis.text=element_text(size=9, face = "bold"),
          panel.grid.major = element_line(colour = 'darkgrey'),
          panel.border=element_rect(fill = NA, size = 2),
          plot.title = element_text(hjust = 0.5, size =15, face = "bold"))
   

```

### Fig. S5

```{r Fig S5}
library(limma)
library(edgeR)
library(cowplot)
x <- readRDS("data/filtermatrix_x.RDS")
ggplot(x$samples, aes(x = as.factor(time), y = RIN)) +
  geom_boxplot(aes(fill=as.factor(time)))+ 
  theme_bw()+
  ylim(c(0,10))+
  labs(x= "", fill ="Time in hours",y ="RNA Integrity Number")+ 
  ggtitle("Boxplot of RIN by time and drug")+
  facet_wrap(~drug)+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 0, hjust = 1, vjust = 0.2),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.background = element_rect(fill = "white"))



seq_info <-read.csv("output/sequencing_info.txt", row.names = 1)
seq_info %>% 
  filter(type=="Total_reads") %>% 
  ggplot(., aes (x =drug, y=Total.Sequences, fill = drug))+
  geom_boxplot()+
  scale_fill_manual(values=drug_pal_fact)+
  ggtitle(expression("Total number of reads by treatment"))+
  xlab(" ")+
  ylab(expression("RNA -sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 12, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "white"),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

seq_info %>% 
  separate(samplenames, into=c(NA,NA,NA,"samplenames")) %>% 
  mutate(shortnames = paste("Sample",str_trim(samplenames))) %>% 
  filter(type=="Total_reads") %>% 
  ggplot(., aes (x =shortnames, y=Total.Sequences, fill = drug, group_by=indv))+
  geom_col()+
 geom_hline(aes(yintercept=20000000))+
 scale_fill_manual(values=drug_pal_fact)+
  ggtitle(expression("Total number of reads by sample"))+
  xlab("")+
  ylab(expression("RNA -sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =6, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


```
### Fig. S6


```{r Fig S6}
filcpm_matrix <- readRDS("data/filcpm_counts.RDS")

mcor <- cor(filcpm_matrix) 
# pheatmap::pheatmap(mcor)
Heatmap(mcor)


```
heatmap is pending a few changes!  just not my focus today.

### Fig. S7


```{r Fig S7}
pca_all_anno <- readRDS("data/supp_pca_all_anno.RDS")
pca_all_anno <- pca_all_anno %>% 
  mutate(drug = case_match(drug, "Daunorubicin"~"DNR","Doxorubicin"~"DOX", "Epirubicin"~"EPI","Mitoxantrone"~"MTX","Trastuzumab"~"TRX","Vehicle"~"VEH", .default = drug))


facs <- c("indv", "drug", "time")
names(facs) <- c("Individual", "Treatment", "Time")

get_regr_pval <- function(mod) {
  # Returns the p-value for the Fstatistic of a linear model
  # mod: class lm
  stopifnot(class(mod) == "lm")
  fstat <- summary(mod)$fstatistic
  pval <- 1 - pf(fstat[1], fstat[2], fstat[3])
  return(pval)
}

plot_versus_pc <- function(df, pc_num, fac) {
  # df: data.frame
  # pc_num: numeric, specific PC for plotting
  # fac: column name of df for plotting against PC
  pc_char <- paste0("PC", pc_num)
  # Calculate F-statistic p-value for linear model
  pval <- get_regr_pval(lm(df[, pc_char] ~ df[, fac]))
  if (is.numeric(df[, f])) {
    ggplot(df, aes_string(x = f, y = pc_char)) + geom_point() +
      geom_smooth(method = "lm") + labs(title = sprintf("p-val: %.2f", pval))
  } else {
    ggplot(df, aes_string(x = f, y = pc_char)) + geom_boxplot() +
      labs(title = sprintf("p-val: %.2f", pval))
  }
}

for (f in facs) {
    # Plot f versus PC1 and PC2
  f_v_pc1 <- arrangeGrob(plot_versus_pc(pca_all_anno, 1, f)+theme_bw())
  f_v_pc2 <- arrangeGrob(plot_versus_pc(pca_all_anno, 2, f)+theme_bw())
  grid.arrange(f_v_pc1, f_v_pc2, ncol = 2, top = names(facs)[which(facs == f)])
}

```

### Fig. S8
```{r Fig S8 volcano, fig.width=10}

Volcanoplots <- readRDS("output/Volcanoplot_10.RDS")
Volcanoplots
```

### Fig. S9

```{r Fig S9}
toplistall <- readRDS("data/toplistall.RDS")
toplistall %>% 
  group_by(time, id) %>% 
  mutate(time=factor(time, levels=c("3_hours","24_hours"))) %>% 
  # mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  ggplot(., aes(x=id, y=logFC))+
  geom_boxplot(aes(fill=id))+
  ggpubr::fill_palette(palette =drug_pal_fact)+
  guides(fill=guide_legend(title = "Treatment"))+
  facet_wrap(sigcount~time)+
  theme_bw()+
  xlab("")+
  ylab("Log Fold Change")+
  theme_bw()+
  facet_wrap(~time)+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        # axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        strip.background = element_rect(fill = "transparent"),
        axis.text.x = element_text(size = 8, color = "white", angle = 0),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))


```
### Fig. S10
```{r Fig S10 Drug-specific genes and pathways,fig.width=14, fig.height=8.5}


allfinal <- readRDS("output/allfinal_sup10.RDS")
allfinal

```
### Fig. S11

```{r Fig venntimepoints}
DNRvenn<- readRDS ("output/DNRvenn.RDS")
DOXvenn<- readRDS ("output/DOXvenn.RDS")
EPIvenn<- readRDS ("output/EPIvenn.RDS")
MTXvenn<- readRDS ("output/MTXvenn.RDS")

plot_grid(DNRvenn,DOXvenn,EPIvenn,MTXvenn,nrow=2, ncol = 2)
```

### Fig. S12
```{r Fig S12 Cormotif}

cormotif_initial <- readRDS("data/cormotif_initialall.RDS")
Cormotif::plotIC(cormotif_initial)
```

```{r Fig S12 Cormotif genes, fig.height=10, fig.width=8,out.width ="80%"}
motif_NRrep <-  readRDS("output/motif_NRrep.RDS")
motif_ERrep <-  readRDS("output/motif_ERrep.RDS")
motif_TIrep <-  readRDS("output/motif_TI_rep.RDS")
motif_LRrep <-  readRDS("output/motif_LRrep.RDS")
# motif_NRrep <- motif_NRrep+theme(axis.title.y = element_text(size=1))

plot_grid(motif_ERrep,motif_LRrep,motif_TIrep,motif_NRrep,nrow = 4,ncol = 1)
motif_NRrep 
motif_ERrep 
motif_TIrep
motif_LRrep

```

```{r Fig S12 GO Cormotif, echo=FALSE, fig.height=15, fig.width=8, out.height="80%"}
motif_list_GO <- readRDS("output/supplementary_motif_list_GO.RDS")
list2env(motif_list_GO,envir=.GlobalEnv)

GOmotiflong <-list("Early Response"=ER3f,"Late Response"=LR3f,"Time-Independent\nResponse"=TI3f, "No Response"=NR1f)
GOBPlong <- data.table::rbindlist(GOmotiflong, idcol = "motif")



p <- GOBPlong %>% 
  mutate(motif=fct_inorder(motif)) %>% 
  ggplot(., aes(x = log_val, y= reorder(term_name, order_val, desc=FALSE),col=motif)) +
  geom_point(aes(size = intersection_size, col=motif)) +
  scale_y_discrete(labels = scales::wrap_format(40))+
  facet_grid(motif~., scales = "free_y")+
  guides(col=guide_legend(title="Motif group"), 
         size=guide_legend(title = "# of intersected \n terms"))+
  ggtitle('Top 10 enriched GO:BP terms for each motif') +
  xlab(expression("-log"[10]~"(p-value)"))+
  ylab("GO:BP term")+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 8, color = "black", angle = 0),
        strip.text.x = element_text(size = 15, color = "black", face = "bold"))

#https://stackoverflow.com/questions/41631806/change-facet-label-text-and-background-colour/60046113#60046113


g <- ggplot_gtable(ggplot_build(p))
stripr <- which(grepl('strip-r', g$layout$name))
fills <- c("#F8766D", "#7CAE00", "#00BFC4", "#C77CFF")
k <- 1
for (i in stripr) {
  j <- which(grepl('rect', g$grobs[[i]]$grobs[[1]]$childrenOrder))
  g$grobs[[i]]$grobs[[1]]$children[[j]]$gp$fill <- fills[k]
  k <- k+1
}
# print(grid::grid.draw(g))
grid.draw(g)

```


