---
title: "Supplementary figures"
author: "ERM"
date: "`r Sys.Date()`"
output:
  pdf_document: default
  keep_md: yes
  html_document:
    df_print: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE
)
```

```{css, echo=FALSE}
pre {
  max-height: 400px;
  overflow-y: auto;
}

pre[class] {
  max-height: 200px;
}
```
```{r packages, message=FALSE, warning=FALSE, error=TRUE}
library(tidyverse)
library(ggpubr)
library(rstatix)
library(zoo)
library(ggsignif)
library(RColorBrewer)
library(grid)
library(scales)
library(ComplexHeatmap)
library(gridExtra)
library(cowplot)
library(drc)

library(kableExtra)
library(broom)
library(ggVennDiagram)

```

#### Function to graph gene expression across treatments
```{r functions}


cpm_boxplot_time <-function(cpmcounts,timex, GOI,brewer_palette, fill_colors, ylab) {
  ##GOI needs to be ENTREZID
  df <- cpmcounts
    df_plot <- df %>% 
      dplyr::filter(rownames(.)==GOI) %>%
      pivot_longer(everything(),
                   names_to = "treatment",
                   values_to = "counts") %>%
      separate(treatment, c("drug","indv","time")) %>%
      dplyr::filter(time == paste(timex)) %>% 
      mutate(time=factor(time, levels =c("3h", "24h"), labels=c("3 hours", "24 hours"))) %>%
      
      mutate(indv=factor(indv, levels = c(1,2,3,4,5,6))) %>%
      mutate(drug =case_match(drug, "Da"~"DNR",
                            "Do"~"DOX",
                            "Ep"~"EPI",
                            "Mi"~"MTX",
                            "Tr"~"TRZ",
                            "Ve"~"VEH", .default = drug)) %>% 
      mutate(drug=factor(drug, levels = c('DOX','EPI','DNR','MTX','TRZ','VEH')))
    plot <- ggplot2::ggplot(df_plot, aes(x=drug, y=counts))+
      geom_boxplot(position="identity",aes(fill=drug))+
      geom_point(aes(col=indv, size=1.5, alpha=0.5))+
      guides(alpha= "none", size= "none")+
      scale_color_brewer(palette = brewer_palette, guide = "none")+
      scale_fill_manual(values=fill_colors)+
      # try(facet_wrap("time", nrow=1, ncol=2))+
      theme_bw()+
      ylab(ylab)+
      xlab("")+
      theme(strip.background = element_rect(fill = "white",linetype=1, linewidth = 0.5),
          plot.title = element_text(size=10,hjust = 0.5,face="bold"),
          axis.title = element_text(size = 10, color = "black"),
          axis.ticks = element_line(linewidth = 1.0),
          axis.line = element_line(linewidth = 1.0),
          axis.text.x = element_blank(),
          strip.text.x = element_text(margin = margin(2,0,2,0, "pt"),face = "bold"))
    print(plot)
}

drug_pal <- c("#707031","#41B333")
# use "Dark2" for individuals, and drug_pal for drug fill

cpm_TRZ <-
  function(cpmcounts,
           GOI,
           ylab,
           titlename) {
    ##GOI needs to be ENTREZID
    df <- trz_veh_count %>% 
      column_to_rownames("ENTREZID")
    df_plot <- df %>%
      dplyr::filter(rownames(.) == GOI) %>%
      pivot_longer(everything(),
                   names_to = "treatment",
                   values_to = "counts") %>%
      separate(treatment, c("drug", "indv", "time")) %>%
      mutate(time = factor(time, levels = c("3h", "24h"))) %>%
      mutate(indv = factor(indv, levels = c(1, 2, 3, 4, 5, 6))) %>%
      mutate(
        drug = case_match(
          drug,
          "Da" ~ "DNR",
          "Do" ~ "DOX",
          "Ep" ~ "EPI",
          "Mi" ~ "MTX",
          "Tr" ~ "TRZ",
          "Ve" ~ "VEH",
          .default = drug
        )
      ) %>%
      mutate(drug = factor(drug, levels = c('DOX', 'EPI', 'DNR', 'MTX', 'TRZ', 'VEH')))
    plot <-
    ggplot2::ggplot(df_plot, aes(x = drug, y = counts)) +
      geom_boxplot(position = "identity", aes(fill = drug)) +
      geom_point(aes(
        col = indv,
        size = 1.5,
        alpha = 0.5
      )) +
      guides(alpha = "none", size = "none") +
      # scale_color_brewer(palette = "Dark2", guide = "none") +
      scale_fill_manual(values = drug_pal) +
      facet_wrap("time", nrow = 1, ncol = 2) +
      theme_bw() +
      ylab(ylab) +
      xlab("") +
      ggtitle(titlename)+
      theme(
        strip.background = element_rect(
          fill = "white",
          linetype = 1,
          linewidth = 0.5
        ),
        plot.title = element_text(
          size = 10,
          hjust = 0.5,
          face = "bold"
        ),
        axis.title = element_text(size = 10, color = "black"),
        axis.ticks = element_line(linewidth = 1.0),
        axis.line = element_line(linewidth = 1.0),
        axis.text.x = element_blank(),
        strip.text.x = element_text(margin = margin(2, 0, 2, 0, "pt"), face = "bold")
      )
    print(plot)
  }



```


## S1 Fig: Cardiomyocytes can be generated at high purity across six individuals.

### S1 FigA Flow cytometry results 
Please see the paper for this figure.  It was generated using FloJo.


### S1 FigB 
```{r Fig S1 troponinT}

ctnnt <- read.csv("data/ctnnt_results.txt", row.names = 1)
ctnnt %>% 
  mutate(Individual=fct_inorder(Individual)) %>% 
  ggplot(., aes(Individual,Percent , fill=Individual))+
  geom_boxplot()+
  geom_point()+
  geom_hline(yintercept =70,linetype="dashed", alpha=0.75)+###adds a line indicating high positivity +
coord_cartesian(ylim = c(0,105))+ ##set those limits
  theme_bw()+  ##white background
  labs(title="Cardiomyocyte Purity")+ #subtitle = "from  n>3 differentiations")+
  geom_boxplot(color="black",alpha =0.2, fill=NA, fatten=0, show.legend = FALSE)+
  scale_fill_brewer(palette = "Dark2",name="" )+
  xlab(NULL)+ 
  ylab("% TNNT2+ ")+
  guides(fill = NULL)+
  theme(plot.title = element_text(hjust = 0.5, size =20, face= "bold"), 
        axis.title.x=element_blank(),
        axis.text.x=element_blank(),###removes all axis names and tick names etc.####
        axis.ticks.x=element_blank(),
        # legend.text=element_text(size=15), 
        axis.title.y=element_text(size=15),
        axis.ticks.y=element_line(size =2),
        axis.text.y=element_text(size=10, face = "bold"),
       panel.grid.major = element_line(colour = 'grey'),
       panel.border=element_rect(fill = NA, size = 3),
       plot.subtitle=element_text(size=18, hjust=0.5, face="italic", color="black")) 
# ctnnt %>%
#   summary() %>%
#   kable(., caption= "Stats summary of cTNNT+ FACs readings") %>%
#   kable_paper("striped", full_width = FALSE) #%>%
#   kable_styling(full_width = FALSE,font_size = 18) #%>%
#   # scroll_box(width = "60%", height = "400px")

```

## S2 Fig: Dose-response curves are reproducible across replicate cardiomyocyte differentiations from the same individual.

```{r Fig S2 DRC reps}
drug_pal_fact <- c("#8B006D","#DF707E","#F1B72B", "#3386DD","#707031","#41B333")
library(data.table)
conf_int <- readRDS("data/plot_intv_list.RDS")
DRC_list <- readRDS("data/plot_list_DRC.RDS")
pull_drc2 <- data.frame("ind1", "ind2","ind3","ind4","ind5","ind6")
doubl_plot <- data.frame("ind3a", "ind3b", "ind5a", "ind5b")
color_order <- c('1','2','5','4','3','6')
intervals <- rbindlist(conf_int,idcol="trt")
drug_list <- c("DNR","DOX","EPI","MTX", "TRZ", "VEH")

for (each in 1:6){
  newdata <- intervals %>% 
    separate("trt", into=c("sDrug",NA)) %>% 
    dplyr::filter(sDrug ==drug_list[each]) %>% 
    mutate(SampleID=indv) %>% 
    mutate(indv=substr(indv,4,4)) %>% 
    mutate(indv=factor(indv, levels=color_order)) %>% 
    dplyr::filter(SampleID %in% doubl_plot)
    
drug_plot <- DRC_list[[each]]
     f <-
       drug_plot %>% 
         filter(SampleID %in% doubl_plot) %>% 
         ggplot(., aes(x=Conc, y= Percent, group=SampleID,linetype=SampleID, color= indv,alpha =0.6 )) +
         guides(color="none", alpha = "none")+
         stat_smooth(method = "drm",
                     method.args = list(fct = L.4(c(NA,NA,1,NA))),
                     se = FALSE)+
         geom_ribbon(data = newdata, 
                     aes(x = Conc, y = Prediction, 
                         ymin = Lower, 
                         ymax = Upper, 
                         fill=indv),
                         alpha = 0.1, 
                         color = "transparent")+
         # ylim(-.2,1.5)+
        coord_cartesian(ylim = c(-0.1, 1.5)) +
         scale_linetype_manual(values = c("dotted","solid","dotted","solid"),
                               name="replicate",
                               labels=c("Rep 1", "Rep 2","Rep 1", "Rep 2"))+
         scale_color_brewer(palette = "Dark2")+
         scale_fill_brewer(palette = "Dark2")+
         scale_x_log10() +  # Change the x-axis scale to log 10 scale
         theme_classic() +
         xlab(NULL)+
         ylab(NULL)+
        # scale_y_continuous(oob=scales::rescale_max, limits = -.4, 1.5)+
        ggtitle(drug_list[each])+
      theme(plot.title = element_text(hjust = 0.5, size =15, face ="bold"),
            axis.title=element_text(size=10),
            axis.ticks=element_line(linewidth = 2),
            axis.text=element_text(size=10, face = "bold", color="black"),
            panel.grid.major = element_line(colour = 'lightgrey'),
            panel.border=element_rect(fill = NA, linewidth = 2),
            plot.background = element_rect(fill = "white", colour = NA))
       
  print(f)
} 


```


## S3 Fig: Cancer drugs that decrease cardiomyocyte viability induce cellular stress.



```{r Fig S3, message=FALSE, warning=FALSE, eval=TRUE,echo=TRUE}

viability <- readRDS("data/viability.RDS")
norm_LDH48 <- readRDS("data/supp_normLDH48.RDS")
viability %>% 
  left_join(., norm_LDH48,by = c("indv","Drug","Conc")) %>% 
  ggplot(., aes(x=per.live, y=ldh))+
  geom_point(aes(col=indv))+
  geom_smooth(method="lm")+
  facet_wrap(~Drug)+
  theme_bw()+
  xlab("Average viability of cardiomyocytes/100") +
  ylab("Average LDH") +
  ggtitle("Cancer drugs that decrease cardiomyocyte viability induce cellular stress")+
  scale_color_brewer(palette = "Dark2",name = "Individual", label = c("1","2","3","4","5","6"))+
  ggpubr::stat_cor(method="pearson",
                   aes(label = paste(..r.label.., ..p.label.., sep = "*`,`~")),
           color = "red")+
  
  theme(plot.title = element_text(size = rel(1), hjust = 0.5,face = "bold"),
        axis.title = element_text(size = 12, color = "black"),
        axis.ticks = element_line(size = 1.5),
        axis.text = element_text(size = 8, color = "black", angle = 0),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"),
        strip.background = element_rect(fill = "transparent")) 

```


## S4 Fig: Treatment with ACs at a dose of 0.5 $\mu$M for 48 hours induces effects on cardiomyocyte viability.


```{r Fig S4}


viabilitytable <- readRDS("data/averageviabilitytable.RDS")


viabilitytable %>% 
  ungroup() %>% 
  mutate(indv=substr(SampleID,4,4)) %>% 
  mutate(indv=factor(indv, levels= c('1','2','3','4','5','6'))) %>% 
  dplyr::filter(Conc <5) %>%
  mutate(Conc= factor(as.numeric(Conc))) %>%
  group_by(indv,sDrug,Conc) %>% 
  dplyr::summarize(Viability=mean(Mean)) %>%
  ggplot(.,  aes(x=sDrug, y= Viability*100 )) +
  geom_boxplot(position="dodge", 
                 aes(fill=sDrug))+
  geom_point(aes(color=indv))+
  guides(alpha = "none")+
  ylim(0,150.5)+
  scale_color_brewer(palette = "Dark2",
                       guide="legend",
                       name ="individual", 
                       labels(c(1,2,3,4,5,6)))+
  scale_fill_manual(values=drug_pal_fact, name ="treatment")+
  theme_classic() +
  xlab("")+
  ylab("% Viability") +
  facet_wrap(~Conc)+ 
  theme(axis.title=element_text(size=10),
        axis.ticks=element_line(size =2),
        axis.text.y=element_text(size=9, face = "bold"),
        axis.text.x=element_blank(),
        panel.grid.major = element_line(colour = 'darkgrey'),
        panel.border=element_rect(fill = NA, size = 2),
        plot.title = element_text(hjust = 0.5, size =15, face = "bold"))
   

```

## S5 Fig: RNA-seq sample quality is equivalent across individuals, treatments, and time points.
### S5 FigA  RIN vs treatment time by drug

```{r Fig S5A}
library(limma)
library(edgeR)
library(cowplot)
filcpm_matrix <- readRDS("data/filcpm_counts.RDS")


x <- readRDS("data/filtermatrix_x.RDS")
x$samples %>%
  mutate(
    drug = case_match(
      drug,
      "Daunorubicin" ~ "DNR",
      "Doxorubicin" ~ "DOX",
      "Epirubicin" ~ "EPI",
      "Mitoxantrone" ~ "MTX",
      "Trastuzumab" ~ "TRZ",
      "Vehicle" ~ "VEH",
      .default = drug
    )
  ) %>%
  mutate(drug = factor(drug, levels = c('DOX', 'EPI', 'DNR', 'MTX', 'TRZ', 'VEH'))) %>%
  mutate(time = factor(time, labels = c("3 hours", "24 hours"))) %>%
  ggplot(., aes(x = as.factor(time), y = RIN)) +
  geom_boxplot(aes(fill = as.factor(time))) +
  theme_bw() +
  ylim(c(0, 10)) +
  labs(x = "", fill = "Time in hours", y = "RNA Integrity Number") +
  ggtitle("Boxplot of RIN by time and drug") +
  facet_wrap( ~ drug) +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.text.y = element_text(
      size = 10,
      color = "black",
      angle = 0,
      hjust = 0.8,
      vjust = 0.5
    ),
    axis.text.x = element_text(
      size = 10,
      color = "black",
      angle = 0,
      hjust = 1,
      vjust = 0.2
    ),
    strip.text.x = element_text(
      size = 15,
      color = "black",
      face = "bold"
    ),
    strip.background = element_rect(fill = "white")
  )


```

### S5 FigB  Total number of reads by treatment

```{r S5B}
seq_info <- read.csv("output/sequencing_info.txt", row.names = 1)
seq_info %>%
  
  filter(type == "Total_reads") %>%
  mutate(
    drug = case_match(
      drug,
      "Daunorubicin" ~ "DNR",
      "Doxorubicin" ~ "DOX",
      "Epirubicin" ~ "EPI",
      "Mitoxantrone" ~ "MTX",
      "Trastuzumab" ~ "TRZ",
      "Vehicle" ~ "VEH",
      .default = drug
    )
  ) %>%
  mutate(drug = factor(drug, levels = c('DOX', 'EPI', 'DNR', 'MTX', 'TRZ', 'VEH'))) %>%
  ggplot(., aes (x = drug, y = count, fill = drug)) +
  geom_boxplot() +
  scale_fill_manual(values = drug_pal_fact) +
  ggtitle(expression("Total number of reads by treatment")) +
  xlab(" ") +
  ylab(expression("RNA -sequencing reads")) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 12, color = "black"),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(
      size = 10,
      color = "black",
      angle = 0,
      hjust = 0.8,
      vjust = 0.5
    ),
    axis.text.x = element_text(size = 10, color = "white"),
    #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
    strip.text.y = element_text(color = "white")
  )
```


### S5 FigC  Total number of reads by sample
```{r S5C, fig.width=8}
seq_info %>%
  mutate(
    drug = case_match(
      drug,
      "Daunorubicin" ~ "DNR",
      "Doxorubicin" ~ "DOX",
      "Epirubicin" ~ "EPI",
      "Mitoxantrone" ~ "MTX",
      "Trastuzumab" ~ "TRZ",
      "Vehicle" ~ "VEH",
      .default = drug
    )
  ) %>%
  mutate(drug = factor(drug, levels = c('DOX', 'EPI', 'DNR', 'MTX', 'TRZ', 'VEH'))) %>%
  # separate(samplenames, into=c(NA,NA,NA,"samplenames")) %>%
  # mutate(shortnames = paste("Sample",str_trim(samplenames))) %>%
  filter(type == "Total_reads") %>%
  mutate(sampleID = colnames(filcpm_matrix)) %>%
  ggplot(., aes (
    x = sampleID,
    y = count,
    fill = drug,
    group_by = indv
  )) +
  geom_col() +
  geom_hline(aes(yintercept = 20000000)) +
  scale_fill_manual(values = drug_pal_fact) +
  ggtitle(expression("Total number of reads by sample")) +
  xlab("") +
  ylab(expression("RNA -sequencing reads")) +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(2), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text.y = element_text(
      size = 10,
      color = "black",
      angle = 0,
      hjust = 0.8,
      vjust = 0.5
    ),
    axis.text.x = element_text(
      size = 6,
      color = "black",
      angle = 90,
      hjust = 1,
      vjust = 0.2
    ),
    #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
    strip.text.y = element_text(color = "white")
  )

```

To see more RNA-seq related analysis [click here](https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/run_all_analysis.html)



## S6 Fig: RNA-seq samples cluster by treatment type, timepoint, and individual. Pearson correlation of log2 cpm values across all pairs of samples.


```{r Fig S6, fig.width=10, fig.height=10}
filcpm_matrix <- readRDS("data/filcpm_counts.RDS")

mcor <- cor(filcpm_matrix) 

filmat_groupmat_col <- data.frame(timeset = colnames(filcpm_matrix))
counts_corr_mat <- filmat_groupmat_col %>%
  separate(timeset, into = c("drug", "indv", "time")) %>%
  mutate(class = if_else(drug == "DNR", "AC", if_else(
    drug == "DOX", "AC", if_else(drug == "EPI", "AC", "nAC")
  ))) %>%
  mutate(TOP2i = if_else(drug == "DNR", "yes", if_else(
    drug == "DOX", "yes", if_else(drug == "EPI", "yes", if_else(drug == "MTX", "yes", "no"))
  ))) 
                         
 mat_colors <- list( 
   drug= c("#F1B72B","#8B006D","#DF707E","#3386DD","#707031","#41B333"),
   indv=c("#1B9E77", "#D95F02" ,"#7570B3", "#E7298A" ,"#66A61E", "#E6AB02"),
   time=c("pink", "chocolate4"),
   class=c("yellow1","darkorange1"), 
   TOP2i =c("darkgreen","lightgreen"))                        
                         
names(mat_colors$drug)   <- unique(counts_corr_mat$drug)                      
names(mat_colors$indv) <- unique(counts_corr_mat$indv)
names(mat_colors$time) <- unique(counts_corr_mat$time)
names(mat_colors$class) <- unique(counts_corr_mat$class)
names(mat_colors$TOP2i) <- unique(counts_corr_mat$TOP2i)


ComplexHeatmap::pheatmap(mcor,
                         # column_title=(paste0("RNA-seq log"[2]~"cpm correlation")),
        annotation_col = counts_corr_mat,
        annotation_colors = mat_colors,
        heatmap_legend_param = mat_colors,
        fontsize=10,
        fontsize_row = 8,
        angle_col="90",
        treeheight_row=25,
        fontsize_col = 8,
        treeheight_col = 20)
```
To see more RNA-seq related analysis [click here](https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/run_all_analysis.html#correlation_heatmap_of_counts_matrix)

## S7 Fig: PC1 associates with drug treatment and treatment time, while PC2 associates with individual.

```{r Fig S7, message=FALSE, warning=FALSE}
pca_all_anno <- readRDS("data/supp_pca_all_anno.RDS")
pca_all_anno <- pca_all_anno %>%
  mutate(
    drug = case_match(
      drug,
      "Daunorubicin" ~ "DNR",
      "Doxorubicin" ~ "DOX",
      "Epirubicin" ~ "EPI",
      "Mitoxantrone" ~ "MTX",
      "Trastuzumab" ~ "TRX",
      "Vehicle" ~ "VEH",
      .default = drug
    )
  )
facs <- c("indv", "drug", "time")
names(facs) <- c("Individual", "Treatment", "Time")

get_regr_pval <- function(mod) {
  # Returns the p-value for the Fstatistic of a linear model
  # mod: class lm
  stopifnot(class(mod) == "lm")
  fstat <- summary(mod)$fstatistic
  pval <- 1 - pf(fstat[1], fstat[2], fstat[3])
  return(pval)
}

plot_versus_pc <- function(df, pc_num, fac) {
  # df: data.frame
  # pc_num: numeric, specific PC for plotting
  # fac: column name of df for plotting against PC
  pc_char <- paste0("PC", pc_num)
  # Calculate F-statistic p-value for linear model
  pval <- get_regr_pval(lm(df[, pc_char] ~ df[, fac]))
  if (is.numeric(df[, f])) {
    ggplot(df, aes_string(x = f, y = pc_char)) + geom_point() +
      geom_smooth(method = "lm") + labs(title = sprintf("p-val: %.2f", pval))
  } else {
    ggplot(df, aes_string(x = f, y = pc_char)) + geom_boxplot() +
      labs(title = sprintf("p-val: %.2f", pval))
  }
}

for (f in facs) {
  # Plot f versus PC1 and PC2
  f_v_pc1 <-
    arrangeGrob(plot_versus_pc(pca_all_anno, 1, f) + theme_bw())
  f_v_pc2 <-
    arrangeGrob(plot_versus_pc(pca_all_anno, 2, f) + theme_bw())
  grid.arrange(f_v_pc1, f_v_pc2, ncol = 2, top = names(facs)[which(facs == f)])
}

```
To see more RNA-seq related analysis [click here](9)https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/run_all_analysis.html)


## S8 Fig: Thousands of gene expression changes are induced in response to TOP2i treatment over 24 hours.
```{r Fig S8 volcano, fig.width=10}

Volcanoplots <- readRDS("output/Volcanoplot_10.RDS")
Volcanoplots
```
(To see more RNA-seq related analysis [click here](https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/run_all_analysis.html#Volcano_plots_from_pairwise_gene_analysis)



## S9 Fig: ACs affect expression of nearly half of all expressed genes after 24 hours of treatment.
### S9 FigA  Proportion of genes DE in response to drug treatment
```{r Fig S9}
toplistall <- readRDS("data/toplistall.RDS")
drug_pal_fact <- c("#8B006D","#DF707E","#F1B72B", "#3386DD","#707031","#41B333")
toplistall %>%
    mutate(id=factor(id, levels = c('DOX', 'EPI', 'DNR', 'MTX', 'TRZ','VEH'))) %>% 
  mutate(time= factor(time,
     levels=c("3_hours","24_hours"),
              labels=c("3 hours","24 hours"))) %>%
  group_by(time, id) %>% 
  mutate(sigcount = if_else(adj.P.Val < 0.05,'sig','notsig'))%>%
  count(sigcount) %>% 
  pivot_wider(id_cols = c(time,id), names_from=sigcount, values_from=n) %>% 
  mutate(prop = sig/(sig+notsig)*100) %>% 
  mutate(prop=if_else(is.na(prop),0,prop)) %>% 
  ggplot(., aes(x=id, y= prop))+
  geom_col(aes(fill=id))+
  geom_text(aes(label = sprintf("%.2f",prop)),
            position=position_dodge(0.9),vjust=-.2 )+
  scale_fill_manual(values =drug_pal_fact)+
  guides(fill=guide_legend(title = "Treatment"))+
  facet_wrap(~time)+#labeller = (time = facettimelabel) )+
  theme_bw()+
  xlab("")+
  ylab("Percentage of expressed genes")+
  theme_bw()+
  ggtitle("Percent DEGs (adj. P value <0.05)")+
  scale_y_continuous(expand=expansion(c(0.02,.2)))+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        # axis.ticks = element_line(linewidth = 1.5),
        # axis.line = element_line(linewidth = 1.5),
        strip.background = element_rect(fill = "transparent"),
        axis.text.x = element_text(size = 8, color = "white", angle = 0),
        axis.text.y = element_text(size = 8, color = "black", angle = 0),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))

```

### S9 FigB Magnitude of response to drug treatment
```{r FIGS9B}

toplistall %>% 
  group_by(time, id) %>% 
  mutate(id=factor(id, levels = c('DOX', 'EPI', 'DNR', 'MTX', 'TRZ','VEH'))) %>% 
  mutate(time= factor(time,
     levels=c("3_hours","24_hours"),
              labels=c("3 hours","24 hours"))) %>%
  ggplot(., aes(x=id, y=logFC))+
  geom_boxplot(aes(fill=id))+
  ggpubr::fill_palette(palette =drug_pal_fact)+
  guides(fill=guide_legend(title = "Treatment"))+
  # facet_wrap(sigcount~time)+
  theme_bw()+
  xlab("")+
  ylab(expression("Log"[2]*" fold change"))+
  theme_bw()+
  facet_wrap(~time)+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        # axis.ticks = element_line(linewidth = 1.5),
        # axis.line = element_line(linewidth = 1.5),
        strip.background = element_rect(fill = "transparent"),
        axis.text.x = element_blank(),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))


```

## S10 Fig: A small number of genes respond to a single drug only.

### 3 hours  

```{r Fig S10 3hr Drug specific genes,  fig.height=8.5, fig.width=12, message=FALSE, warning=FALSE}

supp10_3hlist <- readRDS("data/supp10_3hlist.RDS")
list2env(supp10_3hlist, envir = .GlobalEnv)

mito3pg <- plot_grid(densityMTX3sp, MTX3Bplot, MTX3genesp_example, 
                    nrow = 1, 
                    rel_heights = c(.8,2,1), 
                    rel_widths=c(1.2,1,1.5),
                    scale=c(1,0.8,0.8))


Daun3pg <- plot_grid(densityDNR3sp, DNR3Bplot, DNR3genesp_example, nrow = 1, rel_heights = c(.8,2,1), rel_widths=c(1.2,1,1.5),scale=c(1,0.8,0.8))
Doxo3pg <- plot_grid(NA, NA, NA, nrow = 1, rel_heights = c(.8,2,1), rel_widths=c(1.2,1,1.5),scale=c(1,0.8,0.8))
Epi3pg <- plot_grid(densityEPI3sp, NA, NA, nrow = 1, rel_heights = c(.8,2,1), rel_widths=c(1.2,1,1.5),scale=c(1,0.8,0.8))

allfinal3hour <- plot_grid(Doxo3pg,Epi3pg,Daun3pg,mito3pg,nrow=4, rel_heights = c(1,1,1,1))
# allfinal3hour <- readRDS("data/allfinal3hour.RDS")

allfinal3hour
```

### 24 hours  

```{r Fig S1024hr Drug specific genes,  fig.height=8.5, fig.width=12, message=FALSE, warning=FALSE}

supp10_24hlist <- readRDS("data/supp10_24hlist.RDS")
list2env(supp10_24hlist, envir = .GlobalEnv)
mitopg <- plot_grid(densityMTXsp, MtxBplot, Mtxgenesp_example, 
                    nrow = 1, 
                    rel_heights = c(.8,2,1), 
                    rel_widths=c(1.2,1,1.5),
                    scale=c(1,0.8,0.8))


Daunpg <- plot_grid(densityDNRsp, DnrBplot, Dnrgenesp_example, nrow = 1, rel_heights = c(.8,2,1), rel_widths=c(1.2,1,1.5),scale=c(1,0.8,0.8))
Doxopg <- plot_grid(densityDOXsp, DoxBplot, Doxgenesp_example, nrow = 1, rel_heights = c(.8,2,1), rel_widths=c(1.2,1,1.5),scale=c(1,0.8,0.8))
Epipg <- plot_grid(densityEPIsp, EpiBplot, Epigenesp_example, nrow = 1, rel_heights = c(.8,2,1), rel_widths=c(1.2,1,1.5),scale=c(1,0.8,0.8))
allfinal <- plot_grid(Doxopg,Epipg,Daunpg,mitopg,nrow=4, labels = "AUTO")


# allfinal <- readRDS("output/allfinal_sup10.RDS") 
plot(allfinal)

```

## S11 Fig: Stringently-identified drug-specific response genes are enriched in biological processes.


### 3 hours
```{r Fig S11 stringent, echo= TRUE, include=TRUE}
gostres3Dnrdeg_sp <- readRDS("data/DEG-GO/gostres3Dnrdeg_sp.RDS")

Dnr3_sp_DEGtable <- gostres3Dnrdeg_sp$result %>%
  dplyr::select(c(source, term_id,term_name,intersection_size, 
                   term_size, p_value))


Dnr3_sp_DEGtable  %>% 
    dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels =scales::label_wrap(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('DNR 3 hour specific(stringent)\n gene set GO:BP terms') +
    xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
    theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = "black", angle = 0),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))

gostres3Mtxdeg_sp <- readRDS("data/DEG-GO/gostres3Mtxdeg_sp.RDS")
Mtx3_sp_DEGtable <- gostres3Mtxdeg_sp$result %>%
  dplyr::select(c(source, term_id,term_name,intersection_size, 
                   term_size, p_value))
Mtx3_sp_DEGtable  %>% 
    dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=5,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = scales::label_wrap(30))+
  geom_vline(xintercept = (-log10(0.05)))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('MTX 3 hour specific(stringent)\n gene set GO:BP terms') +
    xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
    theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = "black", angle = 0),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))

```

### 24 hours
```{r FigS11 24h- drug spec path}

DX_sp_DEGgostres <- readRDS("data/DEG-GO/gostresDOXdeg_sp.RDS")
MT_sp_DEGgostres <- readRDS("data/DEG-GO/gostresMTXdeg_sp.RDS")

DX_sp_DEGtable <- DX_sp_DEGgostres$result %>%
  dplyr::select(c(source, term_id,term_name,intersection_size, 
                   term_size, p_value))
MT_sp_DEGtable <- MT_sp_DEGgostres$result %>%
  dplyr::select(c(source, term_id,term_name,intersection_size, 
                   term_size, p_value))
  
  
DX_sp_DEGtable  %>% 
    dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = scales::wrap_format(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('DOX specific 24 hour gene set GO:BP terms') +
    xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
    theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = "black", angle = 0),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))



MT_sp_DEGtable  %>% 
    dplyr::filter(source=="GO:BP") %>% 
    dplyr::select(p_value,term_name,intersection_size) %>%
    slice_min(., n=10 ,order_by=p_value) %>%
    mutate(log_val = -log10(p_value)) %>%
   # slice_max(., n=10,order_by = p_value) %>%
   ggplot(., aes(x = log_val, y =reorder(term_name,p_value), col= intersection_size)) +
    geom_point(aes(size = intersection_size)) +
    scale_y_discrete(labels = scales::wrap_format(30))+
    guides(col="none", size=guide_legend(title = "# of intersected \n terms"))+
    ggtitle('MTX specific 24 hour gene set GO:BP terms') +
    xlab(expression(" -"~log[10]~("adj. p-value")))+
    ylab("GO: BP term")+
    theme_bw()+
    theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, color = "black", angle = 0),
        strip.text.x = element_text(size = 12, color = "black", face = "bold"))





```
## S12 Fig: Nominally significant TRZ response genes show enrichment in cancer-related pathways. 


```{r TRZ expression}
FC_mine <- readRDS("data/toplistall.RDS")

TRZ_list <- FC_mine %>% 
  dplyr::filter(id=="TRZ",P.Value < 0.05) 

TRZ_list %>% 
  group_by(time) %>% 
 count() %>% 
  mutate(time = factor(time, levels = c("3_hours", "24_hours"), labels = c("3 hours", "24 hours")))%>% 
  
  ggplot(., aes(x=time, y=n))+
  geom_col()+
   theme_bw() +
  ggtitle("Number of nominal pvalue TRZ genes by time"
  )+
  ylab(expression("count")) +
  theme(
    strip.background = element_rect(
      fill = "white",
      linetype = 1,
      linewidth = 0.5
    ),
    plot.title = element_text(size = 12, hjust = 0.5, face = "bold"),
    axis.title = element_text(size = 10, color = "black"),
    axis.ticks = element_line(linewidth = 1.0),
    panel.background = element_rect(colour = "black", size = 1),
    # axis.text.x = element_blank(),
    strip.text.x = element_text(margin = margin(2, 0, 2, 0, "mm"), face = "bold")
  )

```

```{r  cpmTRZ}
all_cpmcount <-   readRDS("data/cpmcount.RDS") %>% rownames_to_column(var = "ENTREZID")
##pulled all my logcpm, now choosing only TRZ and VEH for plotting

trz_veh_count <- all_cpmcount %>% 
  dplyr::select(ENTREZID, starts_with("Tr"), starts_with("Ve"))



sampset <- readRDS("data/sampsettrz.RDS")
sub_samp <- sampset %>% 
  dplyr::filter(SYMBOL=="PHLDA1"|SYMBOL== "ANKRD2")
#  <- TRZ_list %>% 
#   distinct(ENTREZID,.keep_all = TRUE) %>% 
#   sample_n(.,6)
for (g in seq(from=1, to=length(sub_samp$ENTREZID))){
  a <- sub_samp$SYMBOL[g]
  cpm_TRZ(trz_veh_count,GOI=sub_samp[g,3],
          ylab=bquote(~italic(.(a))~log[2]~"cpm "), 
          titlename=bquote(~italic(.(a))~"gene "))

}

```

More on TRZ expression [click here](https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/after_comments.html)

## S13 Fig: Four gene expression signatures capture the response to TOP2i over time.
### S13 FigA
```{r Fig S13A Cormotif}

cormotif_initial <- readRDS("data/cormotif_initialall.RDS")
Cormotif::plotIC(cormotif_initial)
```

### S13 FigB

```{r Fig S13B Cormotif genes, fig.height=10, fig.width=6,out.width ="80%"}
motif_NRrep <-  readRDS("output/motif_NRrep.RDS")
motif_ERrep <-  readRDS("output/motif_ERrep.RDS")
motif_TIrep <-  readRDS("output/motif_TI_rep.RDS")
motif_LRrep <-  readRDS("output/motif_LRrep.RDS")
motif_NRrep <- motif_NRrep +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1))+
  theme(legend.position = "NULL", axis.title.y =element_text(size =12))

motif_ERrep <- motif_ERrep+
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1))+
  theme(legend.position = "NULL", axis.title.y =element_text(size =12))

motif_TIrep <- motif_TIrep+
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1))+ 
  theme(legend.position = "NULL", axis.title.y =element_text(size =12))

motif_LRrep <- motif_LRrep+
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1))+
  theme(legend.position = "NULL", axis.title.y =element_text(size =12))
plot_grid(motif_ERrep,motif_TIrep,motif_LRrep,motif_NRrep,nrow = 4,ncol = 1)


```


### S13 FigC


```{r Fig S13C GO Cormotif, echo=TRUE, fig.height=14, fig.width=8, out.height="80%"}
motif_list_GO <- readRDS("output/supplementary_motif_list_GO.RDS")
list2env(motif_list_GO, envir = .GlobalEnv)
motifcol <-  c("#F8766D", "#00BFC4", "#7CAE00",  "#C77CFF")
GOmotiflong <- list(
  "EAR" = ER3f,
  "ESR" = TI3f,
  "LR" = LR3f,
  "NR" = NR1f
)
GOBPlong <- data.table::rbindlist(GOmotiflong, idcol = "motif")
p <-
  GOBPlong %>% mutate(motif = factor(motif, levels = c("EAR", "ESR", "LR", "NR"))) %>%
  
  ggplot(., aes(
    x = log_val,
    y = reorder(term_name, order_val, desc = FALSE),
    col = motif
  )) +
  geom_point(aes(size = intersection_size, col = motif)) +
  scale_y_discrete(labels = scales::label_wrap(40)) +
  facet_grid(motif ~ ., scales = "free_y") +
  scale_color_discrete(type = motifcol) +
  guides(col = guide_legend(title = "Motif"),
         size = guide_legend(title = "# of intersected \n terms")) +
  ggtitle('Top 10 enriched GO:BP terms for each motif') +
  xlab(expression("-log"[10] ~ "(p-value)")) +
  ylab("GO:BP term") +
  theme_bw() +
  theme(
    plot.title = element_text(size = rel(1.5), hjust = 0.5),
    axis.title = element_text(size = 15, color = "black"),
    axis.ticks = element_line(linewidth = 1.5),
    axis.line = element_line(linewidth = 1.5),
    axis.text = element_text(
      size = 8,
      color = "black",
      angle = 0
    ),
    strip.text.x = element_text(
      size = 15,
      color = "black",
      face = "bold"
    )
  )

g <- ggplot_gtable(ggplot_build(p))
stripr <- which(grepl('strip-r', g$layout$name))
fills <- c("#F8766D", "#00BFC4", "#7CAE00",  "#C77CFF")
k <- 1
for (i in stripr) {
  j <- which(grepl('rect', g$grobs[[i]]$grobs[[1]]$childrenOrder))
  g$grobs[[i]]$grobs[[1]]$children[[j]]$gp$fill <- fills[k]
  k <- k + 1
}

grid.draw(g)

```

More on Baysian gene analysis [click here](https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/Cormotifcluster_analysis.html).


## S14 Fig: Gene expression variance across 45 individuals increases in iPSC-CMs treated with DOX. 

```{r Knowlesvariance}
store_var <- readRDS("data/Knowles_variation_data.RDS")

knowlesdrug<- store_var %>% 
  dplyr::select("ESGN","mean_DOX","var_DOX","mean_NT", "var_NT") %>% 
  pivot_longer(cols = !"ESGN", names_to = "short", values_to = "values") %>% 
  separate(short, into=c("calc","treatment")) #%>% 
knowlesdrug %>% 
  as.data.frame() %>% 
  dplyr::filter(calc == "mean") %>% 
  ggplot(., aes(x= treatment, y=values))+
  geom_boxplot()+
  ggtitle("Knowles Means across all genes")+
  geom_signif(
    comparisons = list(
      c("DOX", "NT")),
    test = "t.test",
    tip_length = 0.01,
    map_signif_level = FALSE
    # textsize = 4,
    # # y_position = 11,
    # step_increase = 0.05
  )+
  theme_bw() +
  # xlab(" ") +
  ylab(expression("counts log"[2] ~ "cpm ")) +
  theme(
    strip.background = element_rect(
      fill = "white",
      linetype = 1,
      linewidth = 0.5
    ),
    plot.title = element_text(size = 12, hjust = 0.5, face = "bold"),
    axis.title = element_text(size = 10, color = "black"),
    axis.ticks = element_line(linewidth = 1.0),
    panel.background = element_rect(colour = "black", size = 1),
    # axis.text.x = element_blank(),
    strip.text.x = element_text(margin = margin(2, 0, 2, 0, "mm"), face = "bold")
  )

knowlesdrug %>% 
  as.data.frame() %>% 
  dplyr::filter(calc == "var") %>% 
  ggplot(., aes(x= treatment, y=values))+
  geom_boxplot(outlier.shape= NA)+
  ggtitle(" Knowles Variance across all genes")+
  geom_signif(
    comparisons = list(
      c("DOX", "NT")),
    test = "var.test",
    tip_length = 0.01,
    y_position = 0.5,
    # vjust=1,
    map_signif_level = FALSE)+
   coord_cartesian(ylim = c(NA, 1.5), expand = TRUE)+
theme_bw() +
  # xlab(" ") +
  ylab(expression("counts log"[2] ~ "cpm ")) +
  theme(
    strip.background = element_rect(
      fill = "white",
      linetype = 1,
      linewidth = 0.5
    ),
    plot.title = element_text(size = 12, hjust = 0.5, face = "bold"),
    axis.title = element_text(size = 10, color = "black"),
    axis.ticks = element_line(linewidth = 1.0),
    panel.background = element_rect(colour = "black", size = 1),
    # axis.text.x = element_blank(),
    strip.text.x = element_text(margin = margin(2, 0, 2, 0, "mm"), face = "bold")
  )

```

More on Knowles gene analysis [click here](https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/https://mward-lab.github.io/Matthews_TOP2i_cardiotox_2023/Knowels_trop_analysis.html#Replication_of_variance_figures_using_Knowels_data).



## S15 Fig: Replication of gene expression response in AC-induced cardiotoxicity loci. 


```{r Fig S15,fig.height=12, fig.width=12,out.width="80%"}
DOX_de_genes_interest <- read.csv("output/DOX_de_goi.csv", row.names = 1)


all_cpmcounts <-  read_table("data/Counts_RNA_ERMatthews.txt")
##all my count data
Knowles_log2cpm <- readRDS("data/Knowles_log2cpm_real.RDS") 
##all Knowles count data, down with my method in analysis/after_comments.Rmd
store_box <- Knowles_log2cpm%>% 
  dplyr::select( 'ESGN',ends_with(c('0.625', '0'))) %>%
  dplyr::filter(ESGN %in% DOX_de_genes_interest$ensembl_gene_id) %>% 
  pivot_longer(cols=!ESGN, names_to = "ind", values_to = "counts") %>% 
  separate(ind,into=c("cell_line","dosage"), sep = ":")%>%
  left_join(., DOX_de_genes_interest, by = c("ESGN" = "ensembl_gene_id")) %>% 
  mutate(expr="K")


lcpm_24h <- all_cpmcounts %>% 
  column_to_rownames("ENTREZID") %>% 
  cpm(., log=TRUE) %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "ENTREZID") %>% 
  # dplyr::select(ENTREZID, all_of(starts_with(c("DOX","VEH")))) %>%
  dplyr::select(ENTREZID, all_of(ends_with("24h")))  %>% 
  dplyr::filter(ENTREZID %in% DOX_de_genes_interest$ENTREZID) %>% 
  pivot_longer(cols=!ENTREZID, names_to = "ind", values_to = "counts") %>%   mutate(ENTREZID = as.numeric(ENTREZID)) %>% 
  full_join(., DOX_de_genes_interest, by = c("ENTREZID"="ENTREZID")) %>% 
  mutate(expr="ME") %>% 
  rename("ESGN"="ensembl_gene_id") %>% 
  separate(ind, into = c("dosage","cell_line",NA)) %>% 
  mutate(dosage=case_match(dosage,"DOX"~"0.5", .default = dosage)) 

my_fill_colors <-  c("#41B333","#8B006D","#41B333","#8B006D","#DF707E","#F1B72B", "#3386DD","#707031")

lcpm_24h %>% 
  rbind(.,store_box) %>% 
  mutate(dosage=factor(dosage, levels=c('0','0.625',"1.25","2.5","5","VEH","0.5","EPI","DNR", "MTX","TRZ"))) %>% 
  ggplot(., aes(x=dosage,y=counts))+
  geom_boxplot(aes(fill=dosage))+
  facet_wrap(~hgnc_symbol, scales="free_y", nrow=10, ncol = 3 )+
  theme_bw()+
  scale_fill_manual(values=my_fill_colors)






```
